---
title: "WGS_PoolSeq_Clines&Fst - main analyses & figures, Supporting Info"
output:
  word_document: default
  html_document: default
date: "2025-03-04"
editor_options:
  chunk_output_type: console
---


## Table of Contents
- [Input Data sets](#input-data-sets)
- [Workflow](#workflow)
  - [(i) Data, functions, parameters](#i-data-functions-parameters)
  - [(ii) Genome scan sliding window](#ii-genome-scan-sliding-window)
  - [(iii) FastClines geographic](#iii-fastclines-geographic)
  - [(iv) Post processing sliding window and FastClines outputs](#iv-post-processing-sliding-window-and-fastclines-outputs)
  - [(v) RNAseq Differential Gene Expression (DGE)](#v-rnaseq-differential-gene-expression-dge)
  - [(vi) Genotype-Phenotype associations](#vi-genotype-phenotype-associations)
  - [(vii) MLE Cline fitting](#vii-mle-cline-fitting)
  - [(viii) FastClines simulations](#viii-fastclines-simulations)

## Input Data sets

- Antirrhinum Reference genome v3.0: (online link)
- Antirrhinum Annotated genome: GWHBJVT00000000.gff (online link) 
- Whole Genome PoolSeq dataset: SRA under accession number SUB15081578
- Whole Genome PoolSeq dataset sync files: 
- RNAseq dataset: SRA under accession number SUB15081578
- RNAseq candidate genes Chr 5: `candidate_genes_Chr5_RNAseq_Rimport.csv` (Dryad link) 
- Hybrid zone plant genotypes and spatial dataset: `` (Dryad link)
- Hybrid zone plant colour phenotype dataset: `` (Dryad link)

## Scripts

Analyses and figures for main text and supplementary materials are conducted in a series of R markdown scripts contained in main folder of _https://github.com/dfield007/genome_wide_clines_

- FastClines_Amajus_main.Rmd
- FastClines_Amajus_RNAseq.Rmd
- FastClines_Amajus_GenoPheno.Rmd
- FastClines_Amajus_MLEclines.Rmd
- FastClines_Amajus_simulations.Rmd

and also a number of Python scripts called within the markdowns 

- SlidingWindows.py
- FastClines.py

further custom script functions called from R markdowns contained in `genome_wide_clines/myRfunctions`.

Most of the main analyses and generation of figures are conducted in the script `FastClines_Amajus_main.Rmd`

## Workflow pipeline summary

### (i) Data, functions, parameters

Input datasets:

- Antirrhinum Reference genome 
- Antirrhinum Annotated genome GWHBJVT00000000.gff
- Whole Genome PoolSeq dataset
- RNAseq dataset
- Hybrid zone plant genotypes and spatial dataset
- Hybrid zone plant colour phenotype dataset
- KASP loci details: Amajus_shortSummarySNPsAdj.csv

Functions: see `Source-my-functions` code chunk in `FastClines_Amajus_main.Rmd` to call all functions from `genome_wide_clines/myRfunctions`

### (ii) Genome scan sliding window

- Genome scan sliding windows (Fst, Dxy, pi) performed in `FastClines_Amajus_main.Rmd` 
- splits original sync files into separate chromosome chunks using BASH scripts
- Calls python script `SlidingWindows_v1.4.py` available here: _https://github.com/dfield007/slidingWindows_

### (iii) FastClines geographic

- Geographic cline analyses (cline centre and width) performed in `FastClines_Amajus_main.Rmd` 
- Calls python script `FastClines.py`available here: _https://github.com/dfield007/fastClines_

### (iv) post processing sliding window and FastClines outputs

- Performed by `FastClines_Amajus_main.Rmd` (unless stated otherwise)
- imports individual SlidingWindow outputs and concatenates chromosome outputs
- combine fastClines with SlidngWindows for permutation tests (_python code_)
- combining flower colour gene locations with sliding window and clines data
- Fig 1. map of hybrid zone sampling
- Fig 3a. genome wide cline width by cline centre
- Fig 4. summary box plots of cline width in different genomic regions
- Fig 5. Genome scans for cline properties, frequency and diversity and differentiation sliding window analyses
- Fig 6. 
- Fig 8a,b. Genome scans of Chromosome 5 and RUBIA region (Fst, Dxy, pi, cline width)
- Fig S5 - Fig S13

### (v) RNAseq Differential Gene Expression (DGE)

- main bioinformatics pipeline performed by `FastClines_Amajus_RNAseq.Rmd`
- post analyses and figures in`FastClines_Amajus_main.Rmd`
- Fig 8c. Differential gene expression across RUBIA region

### (vi) Genotype-Phenotype associations

- main analyses run by `FastClines_Amajus_GenoPheno.Rmd`
- Fig 8d. ROS/RUB interaction
- Fig S15 - Fig S17, Table S5

### (vii) MLE Cline fitting

- main analyses run by `FastClines_Amajus_MLEclines.Rmd`
- Fig 8d. ROS/RUB interaction
- Fig S3, Fig S4

### (viii) FastClines simulations

- main analyses run by `FastClines_Amajus_MLEclines.Rmd`
- creates figures for supplementary Fig S1 and Fig S2

```{r setup, include=FALSE}
rm(list = ls())
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(engine.opts = list(bash = "-l"))
knitr::opts_chunk$set(tidy.opts=list(width.cutoff=90),tidy=TRUE)
par(mfrow=c(1,1))
```

```{r load_Packages}
library(reshape2)
library(mapplots)
library(plyr)
library(dplyr)
library(latticeExtra)
library(raster)
library(rasterVis)
library(MASS)
library(RColorBrewer)
library(ggplot2)
library(ggExtra)
library(gplots)
library(tidyverse)
library(devtools)
library(mapplots)
library(viridis)
library(wesanderson)
library(Rmpfr)
library(gmp)
library(openxlsx)
library(cowplot)
library(renv)

```

```{r start BiocManager, eval=FALSE}

# in 2022 on my MAC this version required
if (!requireNamespace("BiocManager", quietly=TRUE))
  install.packages("BiocManager")
BiocManager::install(version="3.20")
#BiocManager::install(version="3.17")

# then run all these lines if new R installation
# for plotting annotation
# BiocManager::install("BiocParallel")
# BiocManager::install("Rhtslib")
# BiocManager::install("DelayedArray")
# BiocManager::install("Biostrings")
# BiocManager::install("Rsamtools")
# BiocManager::install("rtracklayer")
library(rtracklayer)
library(Rsamtools)
library(Biostrings)
library(BiocParallel)
library(DelayedArray)

```

```{r working directories, echo=F}
# MAC
workingDir <- '/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/poolSeq/version3.5/postAnalyses_R/'

setwd(workingDir)

# outputDir_SlidingWindows <- '/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/poolSeq/version3.5/slidingWindows/output/'
# 
# outputDir_fastClines <- '/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/poolSeq/version3.5/fastClines/output/'
# 
# outputDir_PostAnalyses <- '/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/poolSeq/version3.5/PostAnalyses_R/'
# 
# outputDir_RNAseqData <- '/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/RNAseq/'
# 
# setwd('/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/poolSeq/version3.5/postAnalyses_R')

```

```{r Source-my-functions, include=F,echo=F}
# load all of my functions
sourceDir <- function(path, trace = TRUE, ...) {
  # path <- folder_myFunctions
    for (nm in list.files(path, pattern = "[.][RrSsQq]$")) {
       if(trace) cat(nm,":")
       source(file.path(path, nm), ...)
       if(trace) cat("\n")
    }
 }
sourceDir(as.vector("/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/poolSeq/version3.5/postAnalyses_R/myRfunctions"))

```

```{r colour scheme to use through all main figures}

#table(allChr_Clines_filterDpth15[,"ColourGene"])
colourChoice <- matrix(-9,14,4)
colnames(colourChoice) <- c("Region","pointColour","lineColour","histogramColour")
colourChoice[,1] <- c("background","CREMOSA","CREMOSAlinked","FLAVIA","FLAVIAlinked","AURINA","AURINAlinked","SULF","SULFlinked","RUBIA","RUBIAlinked","ROS","ELUTA","ROSelLinked")

# scatterplot used for plotting KASP allele frequencies, width x centre on scatterplot and Genome scans
# lineColour used for plotting allele frequencies at lots of loci with poolSeq data (Figure 2)
# histogram for two side plots on width x centre scatterplot (Figure 2)

colourChoice[colourChoice[,"Region"]=="background",2] <- c((rgb(220,220,220,50,maxColorValue=255)))
colourChoice[colourChoice[,"Region"]=="background",3] <- c((rgb(110,110,110,50,maxColorValue=255)))
colourChoice[colourChoice[,"Region"]=="background",4] <- 'white'
colourChoice[colourChoice[,"Region"]=="CREMOSA",2:4] <- c((rgb(0,255,0,255,maxColorValue=255)))

colourChoice[colourChoice[,"Region"]=="CREMOSAlinked",2] <- c((rgb(0,128,0,200,maxColorValue=255)))
colourChoice[colourChoice[,"Region"]=="CREMOSAlinked",3] <- c((rgb(0,100,0,170,maxColorValue=255)))
colourChoice[colourChoice[,"Region"]=="CREMOSAlinked",4] <- c((rgb(0,128,0,200,maxColorValue=255)))

# previous for lines c((rgb(0,128,0,80,maxColorValue=255)))
# for points c((rgb(60,179,113,255,maxColorValue=255)))
colourChoice[colourChoice[,"Region"]=="FLAVIA",2:4] <- c((rgb(255,127,36,255,maxColorValue=255)))
# previous c((rgb(255,140,0,255,maxColorValue=255)))
colourChoice[colourChoice[,"Region"]=="FLAVIAlinked",2] <- c((rgb(255,195,77,160,maxColorValue=255)))
colourChoice[colourChoice[,"Region"]=="FLAVIAlinked",3] <- c((rgb(255,195,77,160,maxColorValue=255)))
colourChoice[colourChoice[,"Region"]=="FLAVIAlinked",4] <- c((rgb(255,195,77,255,maxColorValue=255)))

# previous c((rgb(255,140,0,80,maxColorValue=255)))
colourChoice[colourChoice[,"Region"]=="AURINA",2] <- 'purple'
colourChoice[colourChoice[,"Region"]=="AURINA",3] <- 'purple'
colourChoice[colourChoice[,"Region"]=="AURINA",4] <- 'purple'

colourChoice[colourChoice[,"Region"]=="AURINAlinked",2] <- c((rgb(210,145,255,120,maxColorValue=255)))
colourChoice[colourChoice[,"Region"]=="AURINAlinked",3] <- c((rgb(210,145,255,120,maxColorValue=255)))
colourChoice[colourChoice[,"Region"]=="AURINAlinked",4] <- c((rgb(210,145,255,120,maxColorValue=255)))

# previous c((rgb(245,222,179,110,maxColorValue=255)))
colourChoice[colourChoice[,"Region"]=="SULF",2:4] <- 'yellow'
colourChoice[colourChoice[,"Region"]=="SULFlinked",2:4] <- 'yellow'
# previously tried this c((rgb(244,164,96,190,maxColorValue=255)))
# for points c((rgb(255,228,196,255,maxColorValue=255)))
colourChoice[colourChoice[,"Region"]=="RUBIAlinked",2] <- c((rgb(135,206,235,255,maxColorValue=255))) # sky blue
colourChoice[colourChoice[,"Region"]=="RUBIAlinked",3] <- c((rgb(135,206,235,160,maxColorValue=255)))
colourChoice[colourChoice[,"Region"]=="RUBIAlinked",4] <- c((rgb(135,206,235,255,maxColorValue=255)))
# previously tried this c((rgb(32,178,170,190,maxColorValue=255)))
colourChoice[colourChoice[,"Region"]=="RUBIA",2] <- c((rgb(65,105,225,255,maxColorValue=255))) # royal blue
colourChoice[colourChoice[,"Region"]=="RUBIA",3] <- c((rgb(0,0,139,255,maxColorValue=255))) # dark blue
colourChoice[colourChoice[,"Region"]=="RUBIA",4] <- c((rgb(0,0,139,255,maxColorValue=255))) #

# previously tried this c((rgb(32,178,170,80,maxColorValue=255)))
colourChoice[colourChoice[,"Region"]=="ROS",2] <- c((rgb(220,0,0,255,maxColorValue=255)))
colourChoice[colourChoice[,"Region"]=="ROS",3] <- c((rgb(220,0,0,150,maxColorValue=255)))
colourChoice[colourChoice[,"Region"]=="ROS",4] <- c((rgb(220,20,60,255,maxColorValue=255)))
colourChoice[colourChoice[,"Region"]=="ELUTA",2] <- c((rgb(119,21,21,255,maxColorValue=255)))
colourChoice[colourChoice[,"Region"]=="ELUTA",3] <- c((rgb(119,21,21,255,maxColorValue=255)))
colourChoice[colourChoice[,"Region"]=="ELUTA",4] <- c((rgb(119,21,21,255,maxColorValue=255)))
colourChoice[colourChoice[,"Region"]=="ROSelLinked",2:4] <- c((rgb(248,212,212,200,maxColorValue=255)))

# Daniels alternative colour scheme
library(RColorBrewer)
 
palette <- brewer.pal(7, "Set1")
palette <- palette[-c(6)]
palette[1] <- "deeppink2"

colourChoice[colourChoice[,"Region"]=="ROS",2] <- "#A65628"
colourChoice[colourChoice[,"Region"]=="ROS",3] <- "#A65628"
colourChoice[colourChoice[,"Region"]=="ROS",4] <- "#A65628"
colourChoice[colourChoice[,"Region"]=="ELUTA",2] <- "#A65628"
colourChoice[colourChoice[,"Region"]=="ELUTA",3] <- "#A65628"
colourChoice[colourChoice[,"Region"]=="ELUTA",4] <- "#A65628"
colourChoice[colourChoice[,"Region"]=="ROSelLinked",2:4] <- "#A65628"
colourChoice[colourChoice[,"Region"]=="RUBIA",2] <- "#E41A1C"
colourChoice[colourChoice[,"Region"]=="RUBIA",3] <- "#E41A1C"
colourChoice[colourChoice[,"Region"]=="RUBIA",4] <- "#E41A1C"
colourChoice[colourChoice[,"Region"]=="RUBIAlinked",2] <- "#E41A1C"
colourChoice[colourChoice[,"Region"]=="RUBIAlinked",3] <- "#E41A1C"
colourChoice[colourChoice[,"Region"]=="RUBIAlinked",4] <- "#E41A1C"

colourChoice[colourChoice[,"Region"]=="SULF",2:4] <- "#984EA3"
colourChoice[colourChoice[,"Region"]=="SULFlinked",2:4] <- "#984EA3"
colourChoice[colourChoice[,"Region"]=="AURINAlinked",2] <- "#377EB8"
colourChoice[colourChoice[,"Region"]=="AURINAlinked",3] <- "#377EB8"
colourChoice[colourChoice[,"Region"]=="AURINAlinked",4] <- "#377EB8"
colourChoice[colourChoice[,"Region"]=="AURINA",2:4] <- "#377EB8"

colourChoice[colourChoice[,"Region"]=="FLAVIA",2] <- "#4DAF4A"
colourChoice[colourChoice[,"Region"]=="FLAVIA",3] <- "#4DAF4A"
colourChoice[colourChoice[,"Region"]=="FLAVIA",4] <- "#4DAF4A"
colourChoice[colourChoice[,"Region"]=="FLAVIAlinked",2] <- "#4DAF4A"
colourChoice[colourChoice[,"Region"]=="FLAVIAlinked",3] <- "#4DAF4A"
colourChoice[colourChoice[,"Region"]=="FLAVIAlinked",4] <- "#4DAF4A"
colourChoice[colourChoice[,"Region"]=="CREMOSA",2:4] <- "deeppink2"
colourChoice[colourChoice[,"Region"]=="CREMOSAlinked",2:4] <- "deeppink2"
colourChoice[colourChoice[,"Region"]=="background",2:4] <- c((rgb(220,220,220,50,maxColorValue=255)))

# update 2024 final?
colourChoice[colourChoice[,"Region"]=="ROS",2] <- "#A65628"
colourChoice[colourChoice[,"Region"]=="ROS",3] <- "#A65628"
colourChoice[colourChoice[,"Region"]=="ROS",4] <- "#A65628"
colourChoice[colourChoice[,"Region"]=="ELUTA",2] <- "#A65628"
colourChoice[colourChoice[,"Region"]=="ELUTA",3] <- "#A65628"
colourChoice[colourChoice[,"Region"]=="ELUTA",4] <- "#A65628"
colourChoice[colourChoice[,"Region"]=="ROSelLinked",2:4] <- "#A65628"
colourChoice[colourChoice[,"Region"]=="RUBIA",2] <- "#377EB8"
colourChoice[colourChoice[,"Region"]=="RUBIA",3] <- "#377EB8"
colourChoice[colourChoice[,"Region"]=="RUBIA",4] <- "#377EB8"
colourChoice[colourChoice[,"Region"]=="RUBIAlinked",2] <- "#377EB8"
colourChoice[colourChoice[,"Region"]=="RUBIAlinked",3] <- "#377EB8"
colourChoice[colourChoice[,"Region"]=="RUBIAlinked",4] <- "#377EB8"

colourChoice[colourChoice[,"Region"]=="SULF",2:4] <- "#984EA3"
colourChoice[colourChoice[,"Region"]=="SULFlinked",2:4] <- "#984EA3"
colourChoice[colourChoice[,"Region"]=="AURINAlinked",2] <- "#FF7F00"
colourChoice[colourChoice[,"Region"]=="AURINAlinked",3] <- "#FF7F00"
colourChoice[colourChoice[,"Region"]=="AURINAlinked",4] <- "#FF7F00"
colourChoice[colourChoice[,"Region"]=="AURINA",2:4] <- "#FF7F00"

colourChoice[colourChoice[,"Region"]=="FLAVIA",2] <- "#4DAF4A"
colourChoice[colourChoice[,"Region"]=="FLAVIA",3] <- "#4DAF4A"
colourChoice[colourChoice[,"Region"]=="FLAVIA",4] <- "#4DAF4A"
colourChoice[colourChoice[,"Region"]=="FLAVIAlinked",2] <- "#4DAF4A"
colourChoice[colourChoice[,"Region"]=="FLAVIAlinked",3] <- "#4DAF4A"
colourChoice[colourChoice[,"Region"]=="FLAVIAlinked",4] <- "#4DAF4A"
colourChoice[colourChoice[,"Region"]=="CREMOSA",2:4] <- "deeppink2"
colourChoice[colourChoice[,"Region"]=="CREMOSAlinked",2:4] <- "deeppink2"
colourChoice[colourChoice[,"Region"]=="background",2:4] <- c((rgb(220,220,220,50,maxColorValue=255)))
```

# (ii) Genome scan sliding window

```{r basic stats on size of reference genome and sync data genome coverage}
# original sync files
Chr1_sync <- 67180736 
Chr2_sync <- 71922798 
Chr3_sync <- 59606420 
Chr4_sync <- 50790506 
Chr5_sync <- 67336222 
Chr6_sync <- 53548017 
Chr7_sync <- 52080256 
Chr8_sync <- 54167825 

totalLength_Sync <- Chr1_sync + Chr2_sync + Chr3_sync + Chr4_sync + Chr5_sync +Chr6_sync + Chr7_sync + Chr8_sync

# reference genome 
# numbers from following linux command line:
# faidx GWHBJVT00000000.genome.fasta -i chromsizes > sizes.genome

GWHBJVT00000001 <- 71919034
GWHBJVT00000002 <- 77118269
GWHBJVT00000003 <- 65231163
GWHBJVT00000004 <- 54887108
GWHBJVT00000005 <- 71106538
GWHBJVT00000006 <- 55699338
GWHBJVT00000007 <- 55564713
GWHBJVT00000008 <- 57431585

totalLength_ref <- GWHBJVT00000001 + GWHBJVT00000002 + GWHBJVT00000003 + GWHBJVT00000004 + GWHBJVT00000005 + GWHBJVT00000006 + GWHBJVT00000007 + GWHBJVT00000008

totalLength_Sync/totalLength_ref
```

```{bash running SlidingWindows python script}
#################################
# NEW Sliding windows analyses 
# Ref genome v3.5
#################################

# Sliding Window analysis Python 3+
# SlidingWindows_v1.14.py 

# unzip all the files individually

gunzip Chr1.sync.gz

# or use loop
for i in *.sync.gz                                                                       
do
base = $(basename ${i} .sync.gz) 
gunzip $i ${base}.sync.gz
done


# Data in sync files for each chromosome
# big file still, splitting Chr1 into sections so progress is saved more regularly and in case of memory issues
# number of lines
wc -l Chr1.sync

# 67 Million

# one way to split files up: but its a bit tedious
#sed -n '1,83000p' Six_HZ_pools_Genome_V3_Chr1.masked.sync > Six_HZ_pools_Genome_V3_Chr1.a.masked.sync

# this will split the file into roughly 17 parts (4 million lines in each). Named xaa, xab,..
# Chr1
split -l 4000000 Chr1.sync
wc -l xaa
# remember always to check how many files are split off as it may vary among chromosomes
cp xaa Chr1_xaa.sync
cp xab Chr1_xab.sync
cp xac Chr1_xac.sync
cp xad Chr1_xad.sync
cp xae Chr1_xae.sync
cp xaf Chr1_xaf.sync
cp xag Chr1_xag.sync
cp xah Chr1_xah.sync
cp xai Chr1_xai.sync
cp xaj Chr1_xaj.sync
cp xak Chr1_xak.sync
cp xal Chr1_xal.sync
cp xam Chr1_xam.sync
cp xan Chr1_xan.sync
cp xao Chr1_xao.sync
cp xap Chr1_xap.sync
cp xaq Chr1_xaq.sync

rm xaa
rm xab
rm xac
rm xad
rm xae
rm xaf
rm xag
rm xah
rm xai
rm xaj
rm xak
rm xal
rm xam
rm xan
rm xao
rm xap
rm xaq

# run split version Chr1- trying a file that runs just the first 3 above

python SlidingWindows_v1.14.py scaffold_Chr1_v3_5_split1_3.txt popDetails_Chr1_v3_5.txt Chr1_split_1_3 15 300 2 2 10000 0 1 0 0 1

# part 2
python SlidingWindows_v1.14.py scaffold_Chr1_v3_5_split4_17.txt popDetails_Chr1_v3_5.txt Chr1_split_4_17 15 300 2 2 10000 0 1 0 0 1

```

# (iii) FastClines geographic cline estimation

```{bash running FastClines python script}

# See FastClines notes regarding files required

#### Splitting sync files to run chunks of chromsomes separately

# Chr1
split -l 4000000 Chr1.sync
wc -l xaa
# remember always to check how many files are split off as it may vary among chromosomes
cp xaa Chr1_xaa.sync
cp xab Chr1_xab.sync
cp xac Chr1_xac.sync
cp xad Chr1_xad.sync
cp xae Chr1_xae.sync
cp xaf Chr1_xaf.sync
cp xag Chr1_xag.sync
cp xah Chr1_xah.sync
cp xai Chr1_xai.sync
cp xaj Chr1_xaj.sync
cp xak Chr1_xak.sync
cp xal Chr1_xal.sync
cp xam Chr1_xam.sync
cp xan Chr1_xan.sync
cp xao Chr1_xao.sync
cp xap Chr1_xap.sync
cp xaq Chr1_xaq.sync

rm xaa
rm xab
rm xac
rm xad
rm xae
rm xaf
rm xag
rm xah
rm xai
rm xaj
rm xak
rm xal
rm xam
rm xan
rm xao
rm xap
rm xaq

# Chr2
split -l 4000000 Chr2.sync
wc -l xaa
# remember always to check how many files are split off as it may vary among chromosomes
cp xaa Chr2_xaa.sync
cp xab Chr2_xab.sync
cp xac Chr2_xac.sync
cp xad Chr2_xad.sync
cp xae Chr2_xae.sync
cp xaf Chr2_xaf.sync
cp xag Chr2_xag.sync
cp xah Chr2_xah.sync
cp xai Chr2_xai.sync
cp xaj Chr2_xaj.sync
cp xak Chr2_xak.sync
cp xal Chr2_xal.sync
cp xam Chr2_xam.sync
cp xan Chr2_xan.sync
cp xao Chr2_xao.sync
cp xap Chr2_xap.sync
cp xaq Chr2_xaq.sync
cp xar Chr2_xar.sync

rm xaa
rm xab
rm xac
rm xad
rm xae
rm xaf
rm xag
rm xah
rm xai
rm xaj
rm xak
rm xal
rm xam
rm xan
rm xao
rm xap
rm xaq
rm xar
# Chr 3
split -l 4000000 Chr3.sync
wc -l xaa
# remember always to check how many files are split off as it may vary among chromosomes
cp xaa Chr3_xaa.sync
cp xab Chr3_xab.sync
cp xac Chr3_xac.sync
cp xad Chr3_xad.sync
cp xae Chr3_xae.sync
cp xaf Chr3_xaf.sync
cp xag Chr3_xag.sync
cp xah Chr3_xah.sync
cp xai Chr3_xai.sync
cp xaj Chr3_xaj.sync
cp xak Chr3_xak.sync
cp xal Chr3_xal.sync
cp xam Chr3_xam.sync
cp xan Chr3_xan.sync
cp xao Chr3_xao.sync

rm xaa
rm xab
rm xac
rm xad
rm xae
rm xaf
rm xag
rm xah
rm xai
rm xaj
rm xak
rm xal
rm xam
rm xan
rm xao
# Chr 4
split -l 4000000 Chr4.sync
wc -l xaa
# remember always to check how many files are split off as it may vary among chromosomes
cp xaa Chr4_xaa.sync
cp xab Chr4_xab.sync
cp xac Chr4_xac.sync
cp xad Chr4_xad.sync
cp xae Chr4_xae.sync
cp xaf Chr4_xaf.sync
cp xag Chr4_xag.sync
cp xah Chr4_xah.sync
cp xai Chr4_xai.sync
cp xaj Chr4_xaj.sync
cp xak Chr4_xak.sync
cp xal Chr4_xal.sync
cp xam Chr4_xam.sync

rm xaa
rm xab
rm xac
rm xad
rm xae
rm xaf
rm xag
rm xah
rm xai
rm xaj
rm xak
rm xal
rm xam
# Chr6
split -l 4000000 Chr6.sync
wc -l xaa
# remember always to check how many files are split off as it may vary among chromosomes
cp xaa Chr6_xaa.sync
cp xab Chr6_xab.sync
cp xac Chr6_xac.sync
cp xad Chr6_xad.sync
cp xae Chr6_xae.sync
cp xaf Chr6_xaf.sync
cp xag Chr6_xag.sync
cp xah Chr6_xah.sync
cp xai Chr6_xai.sync
cp xaj Chr6_xaj.sync
cp xak Chr6_xak.sync
cp xal Chr6_xal.sync
cp xam Chr6_xam.sync
cp xan Chr6_xan.sync


rm xaa
rm xab
rm xac
rm xad
rm xae
rm xaf
rm xag
rm xah
rm xai
rm xaj
rm xak
rm xal
rm xam
rm xan

# Chr7
split -l 4000000 Chr7.sync
wc -l xaa
# remember always to check how many files are split off as it may vary among chromosomes
cp xaa Chr7_xaa.sync
cp xab Chr7_xab.sync
cp xac Chr7_xac.sync
cp xad Chr7_xad.sync
cp xae Chr7_xae.sync
cp xaf Chr7_xaf.sync
cp xag Chr7_xag.sync
cp xah Chr7_xah.sync
cp xai Chr7_xai.sync
cp xaj Chr7_xaj.sync
cp xak Chr7_xak.sync
cp xal Chr7_xal.sync
cp xam Chr7_xam.sync
cp xan Chr7_xan.sync

rm xaa
rm xab
rm xac
rm xad
rm xae
rm xaf
rm xag
rm xah
rm xai
rm xaj
rm xak
rm xal
rm xam
rm xan


# Chr8
split -l 4000000 Chr8.sync
wc -l xaa
# remember always to check how many files are split off as it may vary among chromosomes
cp xaa Chr8_xaa.sync
cp xab Chr8_xab.sync
cp xac Chr8_xac.sync
cp xad Chr8_xad.sync
cp xae Chr8_xae.sync
cp xaf Chr8_xaf.sync
cp xag Chr8_xag.sync
cp xah Chr8_xah.sync
cp xai Chr8_xai.sync
cp xaj Chr8_xaj.sync
cp xak Chr8_xak.sync
cp xal Chr8_xal.sync
cp xam Chr8_xam.sync
cp xan Chr8_xan.sync

rm xaa
rm xab
rm xac
rm xad
rm xae
rm xaf
rm xag
rm xah
rm xai
rm xaj
rm xak
rm xal
rm xam
rm xan

#### FastClines runs per chromosome
# Chr 1
python3 fastClines_v1.4.py scaffold_Chr1_v3_5.txt popDetails.txt _clines_demeOpt1 10 300 2 6 0.8 1 0 1
# Chr 2
python3 fastClines_v1.4.py scaffold_Chr2_v3_5.txt popDetails.txt _clines_demeOpt1 10 300 2 6 0.8 1 0 1
# Chr 3
python3 fastClines_v1.4.py scaffold_Chr3_v3_5.txt popDetails.txt _clines_demeOpt1 10 300 2 6 0.8 1 0 1
# Chr 4
python3 fastClines_v1.4.py scaffold_Chr4_v3_5.txt popDetails.txt _clines_demeOpt1 10 300 2 6 0.8 1 0 1
# Chr 5
python3 fastClines_v1.4.py scaffold_Chr5_v3_5.txt popDetails.txt _clines_demeOpt1 10 300 2 6 0.8 1 0 1
# Chr 6
python3 fastClines_v1.4.py scaffold_Chr6_v3_5.txt popDetails.txt _clines_demeOpt1 10 300 2 6 0.8 1 0 1
# Chr 7
python3 fastClines_v1.4.py scaffold_Chr7_v3_5.txt popDetails.txt _clines_demeOpt1 10 300 2 6 0.8 1 0 1
# Chr 8
python3 fastClines_v1.4.py scaffold_Chr8_v3_5.txt popDetails.txt _clines_demeOpt1 10 300 2 6 0.8 1 0 1
```

# (iv) Importing, merging and post processing sliding window and FastClines outputs

```{r import sliding windows outputs first time, eval=F}
runMe = F
if (runMe == T) {
  # First time importing for merging split files
# Chr1
Chr1_xaa <- read.table(paste0(outputDir_SlidingWindows,'Chr1_xaa_OutputWindows_Chr1_split_1_3.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xab <- read.table(paste0(outputDir_SlidingWindows,'Chr1_xab_OutputWindows_Chr1_split_1_3.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xac <- read.table(paste0(outputDir_SlidingWindows,'Chr1_xac_OutputWindows_Chr1_split_1_3.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xad <- read.table(paste0(outputDir_SlidingWindows,'Chr1_xad_OutputWindows_Chr1_split_4_17.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xae <- read.table(paste0(outputDir_SlidingWindows,'Chr1_xae_OutputWindows_Chr1_split_4_17.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xaf <- read.table(paste0(outputDir_SlidingWindows,'Chr1_xaf_OutputWindows_Chr1_split_4_17.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xag <- read.table(paste0(outputDir_SlidingWindows,'Chr1_xag_OutputWindows_Chr1_split_4_17.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xah <- read.table(paste0(outputDir_SlidingWindows,'Chr1_xah_OutputWindows_Chr1_split_4_17.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xai <- read.table(paste0(outputDir_SlidingWindows,'Chr1_xai_OutputWindows_Chr1_split_4_17.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xaj <- read.table(paste0(outputDir_SlidingWindows,'Chr1_xaj_OutputWindows_Chr1_split_4_17.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xak <- read.table(paste0(outputDir_SlidingWindows,'Chr1_xak_OutputWindows_Chr1_split_4_17.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xal <- read.table(paste0(outputDir_SlidingWindows,'Chr1_xal_OutputWindows_Chr1_split_4_17.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xam <- read.table(paste0(outputDir_SlidingWindows,'Chr1_xam_OutputWindows_Chr1_split_4_17.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xan <- read.table(paste0(outputDir_SlidingWindows,'Chr1_xan_OutputWindows_Chr1_split_4_17.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xao <- read.table(paste0(outputDir_SlidingWindows,'Chr1_xao_OutputWindows_Chr1_split_4_17.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xap <- read.table(paste0(outputDir_SlidingWindows,'Chr1_xap_OutputWindows_Chr1_split_4_17.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xaq <- read.table(paste0(outputDir_SlidingWindows,'Chr1_xaq_OutputWindows_Chr1_split_4_17.txt'), header=T,fill=T,na.strings = "NA")
Chr1 <- rbind(Chr1_xaa,Chr1_xab,Chr1_xac,Chr1_xad,Chr1_xae,Chr1_xaf,Chr1_xag,Chr1_xah,Chr1_xai,Chr1_xaj,Chr1_xak,Chr1_xal,Chr1_xam,Chr1_xan,Chr1_xao,Chr1_xap,Chr1_xaq)
rm(Chr1_xaa,Chr1_xab,Chr1_xac,Chr1_xad,Chr1_xae,Chr1_xaf,Chr1_xag,Chr1_xah,Chr1_xai,Chr1_xaj,Chr1_xak,Chr1_xal,Chr1_xam,Chr1_xan,Chr1_xao,Chr1_xap,Chr1_xaq)

#Chr2
Chr2_xaa <- read.table(paste0(outputDir_SlidingWindows,'Chr2_xaa_OutputWindows_Chr2_split.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xab <- read.table(paste0(outputDir_SlidingWindows,'Chr2_xab_OutputWindows_Chr2_split.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xac <- read.table(paste0(outputDir_SlidingWindows,'Chr2_xac_OutputWindows_Chr2_split.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xad <- read.table(paste0(outputDir_SlidingWindows,'Chr2_xad_OutputWindows_Chr2_split.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xae <- read.table(paste0(outputDir_SlidingWindows,'Chr2_xae_OutputWindows_Chr2_split.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xaf <- read.table(paste0(outputDir_SlidingWindows,'Chr2_xaf_OutputWindows_Chr2_split.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xag <- read.table(paste0(outputDir_SlidingWindows,'Chr2_xag_OutputWindows_Chr2_split.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xah <- read.table(paste0(outputDir_SlidingWindows,'Chr2_xah_OutputWindows_Chr2_split.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xai <- read.table(paste0(outputDir_SlidingWindows,'Chr2_xai_OutputWindows_Chr2_split.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xaj <- read.table(paste0(outputDir_SlidingWindows,'Chr2_xaj_OutputWindows_Chr2_split.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xak <- read.table(paste0(outputDir_SlidingWindows,'Chr2_xak_OutputWindows_Chr2_split.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xal <- read.table(paste0(outputDir_SlidingWindows,'Chr2_xal_OutputWindows_Chr2_split.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xam <- read.table(paste0(outputDir_SlidingWindows,'Chr2_xam_OutputWindows_Chr2_split.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xan <- read.table(paste0(outputDir_SlidingWindows,'Chr2_xan_OutputWindows_Chr2_split.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xao <- read.table(paste0(outputDir_SlidingWindows,'Chr2_xao_OutputWindows_Chr2_split.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xap <- read.table(paste0(outputDir_SlidingWindows,'Chr2_xap_OutputWindows_Chr2_split.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xaq <- read.table(paste0(outputDir_SlidingWindows,'Chr2_xaq_OutputWindows_Chr2_split.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xar <- read.table(paste0(outputDir_SlidingWindows,'Chr2_xar_OutputWindows_Chr2_split.txt'), header=T,fill=T,na.strings = "NA")

Chr2 <- rbind(Chr2_xaa,Chr2_xab,Chr2_xac,Chr2_xad,Chr2_xae,Chr2_xaf,Chr2_xag,Chr2_xah,Chr2_xai,Chr2_xaj,Chr2_xak,Chr2_xal,Chr2_xam,Chr2_xan,Chr2_xao,Chr2_xap,Chr2_xaq,Chr2_xar)
rm(Chr2_xaa,Chr2_xab,Chr2_xac,Chr2_xad,Chr2_xae,Chr2_xaf,Chr2_xag,Chr2_xah,Chr2_xai,Chr2_xaj,Chr2_xak,Chr2_xal,Chr2_xam,Chr2_xan,Chr2_xao,Chr2_xap,Chr2_xaq,Chr2_xar)

#Chr3
# Note, some files had a window with blank values. Replaced manually (find->replace) with NaN
Chr3_xaa <- read.table(paste0(outputDir_SlidingWindows,'Chr3_xaa_OutputWindows_Chr3_split.txt'), header=T,fill=T,na.strings = "NA")
Chr3_xab <- read.table(paste0(outputDir_SlidingWindows,'Chr3_xab_OutputWindows_Chr3_split.txt'), header=T,fill=T,na.strings = "NA")
Chr3_xac <- read.table(paste0(outputDir_SlidingWindows,'Chr3_xac_OutputWindows_Chr3_split.txt'), header=T,fill=T,na.strings = "NA")
Chr3_xad <- read.table(paste0(outputDir_SlidingWindows,'Chr3_xad_OutputWindows_Chr3_split.txt'), header=T,fill=T,na.strings = "NA")
Chr3_xae <- read.table(paste0(outputDir_SlidingWindows,'Chr3_xae_OutputWindows_Chr3_split.txt'), header=T,fill=T,na.strings = "NA")
Chr3_xaf <- read.table(paste0(outputDir_SlidingWindows,'Chr3_xaf_OutputWindows_Chr3_split.txt'), header=T,fill=T,na.strings = "NA")
Chr3_xag <- read.table(paste0(outputDir_SlidingWindows,'Chr3_xag_OutputWindows_Chr3_split.txt'), header=T,fill=T,na.strings = "NA")
Chr3_xah <- read.table(paste0(outputDir_SlidingWindows,'Chr3_xah_OutputWindows_Chr3_split.txt'), header=T,fill=T,na.strings = "NA")
Chr3_xai <- read.table(paste0(outputDir_SlidingWindows,'Chr3_xai_OutputWindows_Chr3_split.txt'), header=T,fill=T,na.strings = "NA")
Chr3_xaj <- read.table(paste0(outputDir_SlidingWindows,'Chr3_xaj_OutputWindows_Chr3_split.txt'), header=T,fill=T,na.strings = "NA")
Chr3_xak <- read.table(paste0(outputDir_SlidingWindows,'Chr3_xak_OutputWindows_Chr3_split.txt'), header=T,fill=T,na.strings = "NA")
Chr3_xal <- read.table(paste0(outputDir_SlidingWindows,'Chr3_xal_OutputWindows_Chr3_split.txt'), header=T,fill=T,na.strings = "NA")
Chr3_xam <- read.table(paste0(outputDir_SlidingWindows,'Chr3_xam_OutputWindows_Chr3_split.txt'), header=T,fill=T,na.strings = "NA")
Chr3_xan <- read.table(paste0(outputDir_SlidingWindows,'Chr3_xan_OutputWindows_Chr3_split.txt'), header=T,fill=T,na.strings = "NA")

Chr3 <- rbind(Chr3_xaa,Chr3_xab,Chr3_xac,Chr3_xad,Chr3_xae,Chr3_xaf,Chr3_xag,Chr3_xah,Chr3_xai,Chr3_xaj,Chr3_xak,Chr3_xal,Chr3_xam,Chr3_xan)
rm(Chr3_xaa,Chr3_xab,Chr3_xac,Chr3_xad,Chr3_xae,Chr3_xaf,Chr3_xag,Chr3_xah,Chr3_xai,Chr3_xaj,Chr3_xak,Chr3_xal,Chr3_xam,Chr3_xan)

#Chr4
Chr4_xaa <- read.table(paste0(outputDir_SlidingWindows,'Chr4_xaa_OutputWindows_Chr4_split.txt'), header=T,fill=T,na.strings = "NA")
Chr4_xab <- read.table(paste0(outputDir_SlidingWindows,'Chr4_xab_OutputWindows_Chr4_split.txt'), header=T,fill=T,na.strings = "NA")
Chr4_xac <- read.table(paste0(outputDir_SlidingWindows,'Chr4_xac_OutputWindows_Chr4_split.txt'), header=T,fill=T,na.strings = "NA")
Chr4_xad <- read.table(paste0(outputDir_SlidingWindows,'Chr4_xad_OutputWindows_Chr4_split.txt'), header=T,fill=T,na.strings = "NA")
Chr4_xae <- read.table(paste0(outputDir_SlidingWindows,'Chr4_xae_OutputWindows_Chr4_split.txt'), header=T,fill=T,na.strings = "NA")
Chr4_xaf <- read.table(paste0(outputDir_SlidingWindows,'Chr4_xaf_OutputWindows_Chr4_split.txt'), header=T,fill=T,na.strings = "NA")
Chr4_xag <- read.table(paste0(outputDir_SlidingWindows,'Chr4_xag_OutputWindows_Chr4_split.txt'), header=T,fill=T,na.strings = "NA")
Chr4_xah <- read.table(paste0(outputDir_SlidingWindows,'Chr4_xah_OutputWindows_Chr4_split.txt'), header=T,fill=T,na.strings = "NA")
Chr4_xai <- read.table(paste0(outputDir_SlidingWindows,'Chr4_xai_OutputWindows_Chr4_split.txt'), header=T,fill=T,na.strings = "NA")
Chr4_xaj <- read.table(paste0(outputDir_SlidingWindows,'Chr4_xaj_OutputWindows_Chr4_split.txt'), header=T,fill=T,na.strings = "NA")
Chr4_xak <- read.table(paste0(outputDir_SlidingWindows,'Chr4_xak_OutputWindows_Chr4_split.txt'), header=T,fill=T,na.strings = "NA")
Chr4_xal <- read.table(paste0(outputDir_SlidingWindows,'Chr4_xal_OutputWindows_Chr4_split.txt'), header=T,fill=T,na.strings = "NA")
Chr4_xam <- read.table(paste0(outputDir_SlidingWindows,'Chr4_xam_OutputWindows_Chr4_split.txt'), header=T,fill=T,na.strings = "NA")

Chr4 <- rbind(Chr4_xaa,Chr4_xab,Chr4_xac,Chr4_xad,Chr4_xae,Chr4_xaf,Chr4_xag,Chr4_xah,Chr4_xai,Chr4_xaj,Chr4_xak,Chr4_xal,Chr4_xam)
rm(Chr4_xaa,Chr4_xab,Chr4_xac,Chr4_xad,Chr4_xae,Chr4_xaf,Chr4_xag,Chr4_xah,Chr4_xai,Chr4_xaj,Chr4_xak,Chr4_xal,Chr4_xam)

# Chr5
Chr5_xaa <- read.table(paste0(outputDir_SlidingWindows,'Chr5_xaa_OutputWindows_Chr5_split.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xab <- read.table(paste0(outputDir_SlidingWindows,'Chr5_xab_OutputWindows_Chr5_split.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xac <- read.table(paste0(outputDir_SlidingWindows,'Chr5_xac_OutputWindows_Chr5_split.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xad <- read.table(paste0(outputDir_SlidingWindows,'Chr5_xad_OutputWindows_Chr5_split.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xae <- read.table(paste0(outputDir_SlidingWindows,'Chr5_xae_OutputWindows_Chr5_split.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xaf <- read.table(paste0(outputDir_SlidingWindows,'Chr5_xaf_OutputWindows_Chr5_split.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xag <- read.table(paste0(outputDir_SlidingWindows,'Chr5_xag_OutputWindows_Chr5_split.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xah <- read.table(paste0(outputDir_SlidingWindows,'Chr5_xah_OutputWindows_Chr5_split.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xai <- read.table(paste0(outputDir_SlidingWindows,'Chr5_xai_OutputWindows_Chr5_split.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xaj <- read.table(paste0(outputDir_SlidingWindows,'Chr5_xaj_OutputWindows_Chr5_split.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xak <- read.table(paste0(outputDir_SlidingWindows,'Chr5_xak_OutputWindows_Chr5_split.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xal <- read.table(paste0(outputDir_SlidingWindows,'Chr5_xal_OutputWindows_Chr5_split.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xam <- read.table(paste0(outputDir_SlidingWindows,'Chr5_xam_OutputWindows_Chr5_split.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xan <- read.table(paste0(outputDir_SlidingWindows,'Chr5_xan_OutputWindows_Chr5_split.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xao <- read.table(paste0(outputDir_SlidingWindows,'Chr5_xao_OutputWindows_Chr5_split.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xap <- read.table(paste0(outputDir_SlidingWindows,'Chr5_xap_OutputWindows_Chr5_split.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xaq <- read.table(paste0(outputDir_SlidingWindows,'Chr5_xaq_OutputWindows_Chr5_split.txt'), header=T,fill=T,na.strings = "NA")

Chr5 <- rbind(Chr5_xaa,Chr5_xab,Chr5_xac,Chr5_xad,Chr5_xae,Chr5_xaf,Chr5_xag,Chr5_xah,Chr5_xai,Chr5_xaj,Chr5_xak,Chr5_xal,Chr5_xam,Chr5_xan,Chr5_xao,Chr5_xap,Chr5_xaq)
rm(Chr5_xaa,Chr5_xab,Chr5_xac,Chr5_xad,Chr5_xae,Chr5_xaf,Chr5_xag,Chr5_xah,Chr5_xai,Chr5_xaj,Chr5_xak,Chr5_xal,Chr5_xam,Chr5_xan,Chr5_xao,Chr5_xap,Chr5_xaq)

# Chr6
Chr6_xaa <- read.table(paste0(outputDir_SlidingWindows,'Chr6_xaa_OutputWindows_Chr6_split.txt'), header=T,fill=T,na.strings = "NA")
Chr6_xab <- read.table(paste0(outputDir_SlidingWindows,'Chr6_xab_OutputWindows_Chr6_split.txt'), header=T,fill=T,na.strings = "NA")
Chr6_xac <- read.table(paste0(outputDir_SlidingWindows,'Chr6_xac_OutputWindows_Chr6_split.txt'), header=T,fill=T,na.strings = "NA")
Chr6_xad <- read.table(paste0(outputDir_SlidingWindows,'Chr6_xad_OutputWindows_Chr6_split.txt'), header=T,fill=T,na.strings = "NA")
Chr6_xae <- read.table(paste0(outputDir_SlidingWindows,'Chr6_xae_OutputWindows_Chr6_split.txt'), header=T,fill=T,na.strings = "NA")
Chr6_xaf <- read.table(paste0(outputDir_SlidingWindows,'Chr6_xaf_OutputWindows_Chr6_split.txt'), header=T,fill=T,na.strings = "NA")
Chr6_xag <- read.table(paste0(outputDir_SlidingWindows,'Chr6_xag_OutputWindows_Chr6_split.txt'), header=T,fill=T,na.strings = "NA")
Chr6_xah <- read.table(paste0(outputDir_SlidingWindows,'Chr6_xah_OutputWindows_Chr6_split.txt'), header=T,fill=T,na.strings = "NA")
Chr6_xai <- read.table(paste0(outputDir_SlidingWindows,'Chr6_xai_OutputWindows_Chr6_split.txt'), header=T,fill=T,na.strings = "NA")
Chr6_xaj <- read.table(paste0(outputDir_SlidingWindows,'Chr6_xaj_OutputWindows_Chr6_split.txt'), header=T,fill=T,na.strings = "NA")
Chr6_xak <- read.table(paste0(outputDir_SlidingWindows,'Chr6_xak_OutputWindows_Chr6_split.txt'), header=T,fill=T,na.strings = "NA")
Chr6_xal <- read.table(paste0(outputDir_SlidingWindows,'Chr6_xal_OutputWindows_Chr6_split.txt'), header=T,fill=T,na.strings = "NA")
Chr6_xam <- read.table(paste0(outputDir_SlidingWindows,'Chr6_xam_OutputWindows_Chr6_split.txt'), header=T,fill=T,na.strings = "NA")
Chr6_xan <- read.table(paste0(outputDir_SlidingWindows,'Chr6_xan_OutputWindows_Chr6_split.txt'), header=T,fill=T,na.strings = "NA")

Chr6 <- rbind(Chr6_xaa,Chr6_xab,Chr6_xac,Chr6_xad,Chr6_xae,Chr6_xaf,Chr6_xag,Chr6_xah,Chr6_xai,Chr6_xaj,Chr6_xak,Chr6_xal,Chr6_xam,Chr6_xan)
rm(Chr6_xaa,Chr6_xab,Chr6_xac,Chr6_xad,Chr6_xae,Chr6_xaf,Chr6_xag,Chr6_xah,Chr6_xai,Chr6_xaj,Chr6_xak,Chr6_xal,Chr6_xam,Chr6_xan)

#Chr7
Chr7_xaa <- read.table(paste0(outputDir_SlidingWindows,'Chr7_xaa_OutputWindows_Chr7_split.txt'), header=T,fill=T,na.strings = "NA")
Chr7_xab <- read.table(paste0(outputDir_SlidingWindows,'Chr7_xab_OutputWindows_Chr7_split.txt'), header=T,fill=T,na.strings = "NA")
Chr7_xac <- read.table(paste0(outputDir_SlidingWindows,'Chr7_xac_OutputWindows_Chr7_split.txt'), header=T,fill=T,na.strings = "NA")
Chr7_xad <- read.table(paste0(outputDir_SlidingWindows,'Chr7_xad_OutputWindows_Chr7_split.txt'), header=T,fill=T,na.strings = "NA")
Chr7_xae <- read.table(paste0(outputDir_SlidingWindows,'Chr7_xae_OutputWindows_Chr7_split.txt'), header=T,fill=T,na.strings = "NA")
Chr7_xaf <- read.table(paste0(outputDir_SlidingWindows,'Chr7_xaf_OutputWindows_Chr7_split.txt'), header=T,fill=T,na.strings = "NA")
Chr7_xag <- read.table(paste0(outputDir_SlidingWindows,'Chr7_xag_OutputWindows_Chr7_split.txt'), header=T,fill=T,na.strings = "NA")
Chr7_xah <- read.table(paste0(outputDir_SlidingWindows,'Chr7_xah_OutputWindows_Chr7_split.txt'), header=T,fill=T,na.strings = "NA")
Chr7_xai <- read.table(paste0(outputDir_SlidingWindows,'Chr7_xai_OutputWindows_Chr7_split.txt'), header=T,fill=T,na.strings = "NA")
Chr7_xaj <- read.table(paste0(outputDir_SlidingWindows,'Chr7_xaj_OutputWindows_Chr7_split.txt'), header=T,fill=T,na.strings = "NA")
Chr7_xak <- read.table(paste0(outputDir_SlidingWindows,'Chr7_xak_OutputWindows_Chr7_split.txt'), header=T,fill=T,na.strings = "NA")
Chr7_xal <- read.table(paste0(outputDir_SlidingWindows,'Chr7_xal_OutputWindows_Chr7_split.txt'), header=T,fill=T,na.strings = "NA")
Chr7_xam <- read.table(paste0(outputDir_SlidingWindows,'Chr7_xam_OutputWindows_Chr7_split.txt'), header=T,fill=T,na.strings = "NA")
Chr7_xan <- read.table(paste0(outputDir_SlidingWindows,'Chr7_xan_OutputWindows_Chr7_split.txt'), header=T,fill=T,na.strings = "NA")

Chr7 <- rbind(Chr7_xaa,Chr7_xab,Chr7_xac,Chr7_xad,Chr7_xae,Chr7_xaf,Chr7_xag,Chr7_xah,Chr7_xai,Chr7_xaj,Chr7_xak,Chr7_xal,Chr7_xam,Chr7_xan)
rm(Chr7_xaa,Chr7_xab,Chr7_xac,Chr7_xad,Chr7_xae,Chr7_xaf,Chr7_xag,Chr7_xah,Chr7_xai,Chr7_xaj,Chr7_xak,Chr7_xal,Chr7_xam,Chr7_xan)

#Chr8
Chr8_xaa <- read.table(paste0(outputDir_SlidingWindows,'Chr8_xaa_OutputWindows_Chr8_split.txt'), header=T,fill=T,na.strings = "NA")
Chr8_xab <- read.table(paste0(outputDir_SlidingWindows,'Chr8_xab_OutputWindows_Chr8_split.txt'), header=T,fill=T,na.strings = "NA")
Chr8_xac <- read.table(paste0(outputDir_SlidingWindows,'Chr8_xac_OutputWindows_Chr8_split.txt'), header=T,fill=T,na.strings = "NA")
Chr8_xad <- read.table(paste0(outputDir_SlidingWindows,'Chr8_xad_OutputWindows_Chr8_split.txt'), header=T,fill=T,na.strings = "NA")
Chr8_xae <- read.table(paste0(outputDir_SlidingWindows,'Chr8_xae_OutputWindows_Chr8_split.txt'), header=T,fill=T,na.strings = "NA")
Chr8_xaf <- read.table(paste0(outputDir_SlidingWindows,'Chr8_xaf_OutputWindows_Chr8_split.txt'), header=T,fill=T,na.strings = "NA")
Chr8_xag <- read.table(paste0(outputDir_SlidingWindows,'Chr8_xag_OutputWindows_Chr8_split.txt'), header=T,fill=T,na.strings = "NA")
Chr8_xah <- read.table(paste0(outputDir_SlidingWindows,'Chr8_xah_OutputWindows_Chr8_split.txt'), header=T,fill=T,na.strings = "NA")
Chr8_xai <- read.table(paste0(outputDir_SlidingWindows,'Chr8_xai_OutputWindows_Chr8_split.txt'), header=T,fill=T,na.strings = "NA")
Chr8_xaj <- read.table(paste0(outputDir_SlidingWindows,'Chr8_xaj_OutputWindows_Chr8_split.txt'), header=T,fill=T,na.strings = "NA")
Chr8_xak <- read.table(paste0(outputDir_SlidingWindows,'Chr8_xak_OutputWindows_Chr8_split.txt'), header=T,fill=T,na.strings = "NA")
Chr8_xal <- read.table(paste0(outputDir_SlidingWindows,'Chr8_xal_OutputWindows_Chr8_split.txt'), header=T,fill=T,na.strings = "NA")
Chr8_xam <- read.table(paste0(outputDir_SlidingWindows,'Chr8_xam_OutputWindows_Chr8_split.txt'), header=T,fill=T,na.strings = "NA")
Chr8_xan <- read.table(paste0(outputDir_SlidingWindows,'Chr8_xan_OutputWindows_Chr8_split.txt'), header=T,fill=T,na.strings = "NA")

Chr8 <- rbind(Chr8_xaa,Chr8_xab,Chr8_xac,Chr8_xad,Chr8_xae,Chr8_xaf,Chr8_xag,Chr8_xah,Chr8_xai,Chr8_xaj,Chr8_xak,Chr8_xal,Chr8_xam,Chr8_xan)
rm(Chr8_xaa,Chr8_xab,Chr8_xac,Chr8_xad,Chr8_xae,Chr8_xaf,Chr8_xag,Chr8_xah,Chr8_xai,Chr8_xaj,Chr8_xak,Chr8_xal,Chr8_xam,Chr8_xan)


# Write each Chromosome to file

write.csv(Chr1,"Chr1_SlidingWindows.csv",row.names = F)
write.csv(Chr2,"Chr2_SlidingWindows.csv",row.names = F)
write.csv(Chr3,"Chr3_SlidingWindows.csv",row.names = F)
write.csv(Chr4,"Chr4_SlidingWindows.csv",row.names = F)
write.csv(Chr5,"Chr5_SlidingWindows.csv",row.names = F)
write.csv(Chr6,"Chr6_SlidingWindows.csv",row.names = F)
write.csv(Chr7,"Chr7_SlidingWindows.csv",row.names = F)
write.csv(Chr8,"Chr8_SlidingWindows.csv",row.names = F)
  
}


```

```{r import fastClines outputs first time, eval=F}
runMe <- F

if (runMe==T) {
  # First time importing for merging split files
# Chr1
Chr1_xaa <- read.table(paste0(outputDir_fastClines,'Chr1_xaa_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xab <- read.table(paste0(outputDir_fastClines,'Chr1_xab_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xac <- read.table(paste0(outputDir_fastClines,'Chr1_xac_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xad <- read.table(paste0(outputDir_fastClines,'Chr1_xad_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xae <- read.table(paste0(outputDir_fastClines,'Chr1_xae_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xaf <- read.table(paste0(outputDir_fastClines,'Chr1_xaf_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xag <- read.table(paste0(outputDir_fastClines,'Chr1_xag_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xah <- read.table(paste0(outputDir_fastClines,'Chr1_xah_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xai <- read.table(paste0(outputDir_fastClines,'Chr1_xai_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xaj <- read.table(paste0(outputDir_fastClines,'Chr1_xaj_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xak <- read.table(paste0(outputDir_fastClines,'Chr1_xak_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xal <- read.table(paste0(outputDir_fastClines,'Chr1_xal_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xam <- read.table(paste0(outputDir_fastClines,'Chr1_xam_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xan <- read.table(paste0(outputDir_fastClines,'Chr1_xan_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xao <- read.table(paste0(outputDir_fastClines,'Chr1_xao_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xap <- read.table(paste0(outputDir_fastClines,'Chr1_xap_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr1_xaq <- read.table(paste0(outputDir_fastClines,'Chr1_xaq_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")

Chr1_clines <- rbind(Chr1_xaa,Chr1_xab,Chr1_xac,Chr1_xad,Chr1_xae,Chr1_xaf,Chr1_xag,Chr1_xah,Chr1_xai,Chr1_xaj,Chr1_xak,Chr1_xal,Chr1_xam,Chr1_xan,Chr1_xao,Chr1_xap,Chr1_xaq)
rm(Chr1_xaa,Chr1_xab,Chr1_xac,Chr1_xad,Chr1_xae,Chr1_xaf,Chr1_xag,Chr1_xah,Chr1_xai,Chr1_xaj,Chr1_xak,Chr1_xal,Chr1_xam,Chr1_xan,Chr1_xao,Chr1_xap,Chr1_xaq)

#Chr2
Chr2_xaa <- read.table(paste0(outputDir_fastClines,'Chr2_xaa_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xab <- read.table(paste0(outputDir_fastClines,'Chr2_xab_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xac <- read.table(paste0(outputDir_fastClines,'Chr2_xac_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xad <- read.table(paste0(outputDir_fastClines,'Chr2_xad_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xae <- read.table(paste0(outputDir_fastClines,'Chr2_xae_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xaf <- read.table(paste0(outputDir_fastClines,'Chr2_xaf_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xag <- read.table(paste0(outputDir_fastClines,'Chr2_xag_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xah <- read.table(paste0(outputDir_fastClines,'Chr2_xah_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xai <- read.table(paste0(outputDir_fastClines,'Chr2_xai_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xaj <- read.table(paste0(outputDir_fastClines,'Chr2_xaj_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xak <- read.table(paste0(outputDir_fastClines,'Chr2_xak_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xal <- read.table(paste0(outputDir_fastClines,'Chr2_xal_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xam <- read.table(paste0(outputDir_fastClines,'Chr2_xam_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xan <- read.table(paste0(outputDir_fastClines,'Chr2_xan_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xao <- read.table(paste0(outputDir_fastClines,'Chr2_xao_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xap <- read.table(paste0(outputDir_fastClines,'Chr2_xap_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xaq <- read.table(paste0(outputDir_fastClines,'Chr2_xaq_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr2_xar <- read.table(paste0(outputDir_fastClines,'Chr2_xar_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")

Chr2_clines <- rbind(Chr2_xaa,Chr2_xab,Chr2_xac,Chr2_xad,Chr2_xae,Chr2_xaf,Chr2_xag,Chr2_xah,Chr2_xai,Chr2_xaj,Chr2_xak,Chr2_xal,Chr2_xam,Chr2_xan,Chr2_xao,Chr2_xap,Chr2_xaq,Chr2_xar)
rm(Chr2_xaa,Chr2_xab,Chr2_xac,Chr2_xad,Chr2_xae,Chr2_xaf,Chr2_xag,Chr2_xah,Chr2_xai,Chr2_xaj,Chr2_xak,Chr2_xal,Chr2_xam,Chr2_xan,Chr2_xao,Chr2_xap,Chr2_xaq,Chr2_xar)

#Chr3
Chr3_xaa <- read.table(paste0(outputDir_fastClines,'Chr3_xaa_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr3_xab <- read.table(paste0(outputDir_fastClines,'Chr3_xab_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr3_xac <- read.table(paste0(outputDir_fastClines,'Chr3_xac_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr3_xad <- read.table(paste0(outputDir_fastClines,'Chr3_xad_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr3_xae <- read.table(paste0(outputDir_fastClines,'Chr3_xae_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr3_xaf <- read.table(paste0(outputDir_fastClines,'Chr3_xaf_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr3_xag <- read.table(paste0(outputDir_fastClines,'Chr3_xag_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr3_xah <- read.table(paste0(outputDir_fastClines,'Chr3_xah_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr3_xai <- read.table(paste0(outputDir_fastClines,'Chr3_xai_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr3_xaj <- read.table(paste0(outputDir_fastClines,'Chr3_xaj_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr3_xak <- read.table(paste0(outputDir_fastClines,'Chr3_xak_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr3_xal <- read.table(paste0(outputDir_fastClines,'Chr3_xal_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr3_xam <- read.table(paste0(outputDir_fastClines,'Chr3_xam_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr3_xan <- read.table(paste0(outputDir_fastClines,'Chr3_xan_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr3_xao <- read.table(paste0(outputDir_fastClines,'Chr3_xao_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")

Chr3_clines <- rbind(Chr3_xaa,Chr3_xab,Chr3_xac,Chr3_xad,Chr3_xae,Chr3_xaf,Chr3_xag,Chr3_xah,Chr3_xai,Chr3_xaj,Chr3_xak,Chr3_xal,Chr3_xam,Chr3_xan,Chr3_xao)
rm(Chr3_xaa,Chr3_xab,Chr3_xac,Chr3_xad,Chr3_xae,Chr3_xaf,Chr3_xag,Chr3_xah,Chr3_xai,Chr3_xaj,Chr3_xak,Chr3_xal,Chr3_xam,Chr3_xan,Chr3_xao)

#Chr4
Chr4_xaa <- read.table(paste0(outputDir_fastClines,'Chr4_xaa_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr4_xab <- read.table(paste0(outputDir_fastClines,'Chr4_xab_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr4_xac <- read.table(paste0(outputDir_fastClines,'Chr4_xac_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr4_xad <- read.table(paste0(outputDir_fastClines,'Chr4_xad_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr4_xae <- read.table(paste0(outputDir_fastClines,'Chr4_xae_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr4_xaf <- read.table(paste0(outputDir_fastClines,'Chr4_xaf_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr4_xag <- read.table(paste0(outputDir_fastClines,'Chr4_xag_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr4_xah <- read.table(paste0(outputDir_fastClines,'Chr4_xah_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr4_xai <- read.table(paste0(outputDir_fastClines,'Chr4_xai_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr4_xaj <- read.table(paste0(outputDir_fastClines,'Chr4_xaj_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr4_xak <- read.table(paste0(outputDir_fastClines,'Chr4_xak_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr4_xal <- read.table(paste0(outputDir_fastClines,'Chr4_xal_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr4_xam <- read.table(paste0(outputDir_fastClines,'Chr4_xam_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")

Chr4_clines <- rbind(Chr4_xaa,Chr4_xab,Chr4_xac,Chr4_xad,Chr4_xae,Chr4_xaf,Chr4_xag,Chr4_xah,Chr4_xai,Chr4_xaj,Chr4_xak,Chr4_xal,Chr4_xam)
rm(Chr4_xaa,Chr4_xab,Chr4_xac,Chr4_xad,Chr4_xae,Chr4_xaf,Chr4_xag,Chr4_xah,Chr4_xai,Chr4_xaj,Chr4_xak,Chr4_xal,Chr4_xam)

# Chr5
Chr5_xaa <- read.table(paste0(outputDir_fastClines,'Chr5_xaa_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xab <- read.table(paste0(outputDir_fastClines,'Chr5_xab_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xac <- read.table(paste0(outputDir_fastClines,'Chr5_xac_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xad <- read.table(paste0(outputDir_fastClines,'Chr5_xad_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xae <- read.table(paste0(outputDir_fastClines,'Chr5_xae_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xaf <- read.table(paste0(outputDir_fastClines,'Chr5_xaf_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xag <- read.table(paste0(outputDir_fastClines,'Chr5_xag_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xah <- read.table(paste0(outputDir_fastClines,'Chr5_xah_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xai <- read.table(paste0(outputDir_fastClines,'Chr5_xai_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xaj <- read.table(paste0(outputDir_fastClines,'Chr5_xaj_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xak <- read.table(paste0(outputDir_fastClines,'Chr5_xak_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xal <- read.table(paste0(outputDir_fastClines,'Chr5_xal_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xam <- read.table(paste0(outputDir_fastClines,'Chr5_xam_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xan <- read.table(paste0(outputDir_fastClines,'Chr5_xan_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xao <- read.table(paste0(outputDir_fastClines,'Chr5_xao_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xap <- read.table(paste0(outputDir_fastClines,'Chr5_xap_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr5_xaq <- read.table(paste0(outputDir_fastClines,'Chr5_xaq_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")

Chr5_clines <- rbind(Chr5_xaa,Chr5_xab,Chr5_xac,Chr5_xad,Chr5_xae,Chr5_xaf,Chr5_xag,Chr5_xah,Chr5_xai,Chr5_xaj,Chr5_xak,Chr5_xal,Chr5_xam,Chr5_xan,Chr5_xao,Chr5_xap,Chr5_xaq)
rm(Chr5_xaa,Chr5_xab,Chr5_xac,Chr5_xad,Chr5_xae,Chr5_xaf,Chr5_xag,Chr5_xah,Chr5_xai,Chr5_xaj,Chr5_xak,Chr5_xal,Chr5_xam,Chr5_xan,Chr5_xao,Chr5_xap,Chr5_xaq)

# Chr6
Chr6_xaa <- read.table(paste0(outputDir_fastClines,'Chr6_xaa_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr6_xab <- read.table(paste0(outputDir_fastClines,'Chr6_xab_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr6_xac <- read.table(paste0(outputDir_fastClines,'Chr6_xac_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr6_xad <- read.table(paste0(outputDir_fastClines,'Chr6_xad_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr6_xae <- read.table(paste0(outputDir_fastClines,'Chr6_xae_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr6_xaf <- read.table(paste0(outputDir_fastClines,'Chr6_xaf_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr6_xag <- read.table(paste0(outputDir_fastClines,'Chr6_xag_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr6_xah <- read.table(paste0(outputDir_fastClines,'Chr6_xah_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr6_xai <- read.table(paste0(outputDir_fastClines,'Chr6_xai_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr6_xaj <- read.table(paste0(outputDir_fastClines,'Chr6_xaj_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr6_xak <- read.table(paste0(outputDir_fastClines,'Chr6_xak_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr6_xal <- read.table(paste0(outputDir_fastClines,'Chr6_xal_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr6_xam <- read.table(paste0(outputDir_fastClines,'Chr6_xam_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr6_xan <- read.table(paste0(outputDir_fastClines,'Chr6_xan_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")

# checking original sync files for missing loci
# ROS el positions - SNP [A/G] at 52,889,078 & SNP [A/T] at 52,889,080 are missing
# it was a filtering issue in FastClines. Pool 20 has depth of 11 and 12 at those sites which resulted in the whole locus dropped
# Chr6_xam[Chr6_xam$LG==6 & Chr6_xam$position>52889000 & Chr6_xam$position<52889737,"position"]

Chr6_clines <- rbind(Chr6_xaa,Chr6_xab,Chr6_xac,Chr6_xad,Chr6_xae,Chr6_xaf,Chr6_xag,Chr6_xah,Chr6_xai,Chr6_xaj,Chr6_xak,Chr6_xal,Chr6_xam,Chr6_xan)
rm(Chr6_xaa,Chr6_xab,Chr6_xac,Chr6_xad,Chr6_xae,Chr6_xaf,Chr6_xag,Chr6_xah,Chr6_xai,Chr6_xaj,Chr6_xak,Chr6_xal,Chr6_xam,Chr6_xan)

#Chr7
Chr7_xaa <- read.table(paste0(outputDir_fastClines,'Chr7_xaa_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr7_xab <- read.table(paste0(outputDir_fastClines,'Chr7_xab_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr7_xac <- read.table(paste0(outputDir_fastClines,'Chr7_xac_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr7_xad <- read.table(paste0(outputDir_fastClines,'Chr7_xad_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr7_xae <- read.table(paste0(outputDir_fastClines,'Chr7_xae_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr7_xaf <- read.table(paste0(outputDir_fastClines,'Chr7_xaf_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr7_xag <- read.table(paste0(outputDir_fastClines,'Chr7_xag_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr7_xah <- read.table(paste0(outputDir_fastClines,'Chr7_xah_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr7_xai <- read.table(paste0(outputDir_fastClines,'Chr7_xai_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr7_xaj <- read.table(paste0(outputDir_fastClines,'Chr7_xaj_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr7_xak <- read.table(paste0(outputDir_fastClines,'Chr7_xak_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr7_xal <- read.table(paste0(outputDir_fastClines,'Chr7_xal_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr7_xam <- read.table(paste0(outputDir_fastClines,'Chr7_xam_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr7_xan <- read.table(paste0(outputDir_fastClines,'Chr7_xan_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")

Chr7_clines <- rbind(Chr7_xaa,Chr7_xab,Chr7_xac,Chr7_xad,Chr7_xae,Chr7_xaf,Chr7_xag,Chr7_xah,Chr7_xai,Chr7_xaj,Chr7_xak,Chr7_xal,Chr7_xam,Chr7_xan)
rm(Chr7_xaa,Chr7_xab,Chr7_xac,Chr7_xad,Chr7_xae,Chr7_xaf,Chr7_xag,Chr7_xah,Chr7_xai,Chr7_xaj,Chr7_xak,Chr7_xal,Chr7_xam,Chr7_xan)

#Chr8 - note Chr8_xaa empty, no clines
#Chr8_xaa <- read.table(paste0(outputDir_fastClines,'Chr8_xaa_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr8_xab <- read.table(paste0(outputDir_fastClines,'Chr8_xab_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr8_xac <- read.table(paste0(outputDir_fastClines,'Chr8_xac_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr8_xad <- read.table(paste0(outputDir_fastClines,'Chr8_xad_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr8_xae <- read.table(paste0(outputDir_fastClines,'Chr8_xae_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr8_xaf <- read.table(paste0(outputDir_fastClines,'Chr8_xaf_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr8_xag <- read.table(paste0(outputDir_fastClines,'Chr8_xag_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr8_xah <- read.table(paste0(outputDir_fastClines,'Chr8_xah_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr8_xai <- read.table(paste0(outputDir_fastClines,'Chr8_xai_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr8_xaj <- read.table(paste0(outputDir_fastClines,'Chr8_xaj_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr8_xak <- read.table(paste0(outputDir_fastClines,'Chr8_xak_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr8_xal <- read.table(paste0(outputDir_fastClines,'Chr8_xal_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr8_xam <- read.table(paste0(outputDir_fastClines,'Chr8_xam_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")
Chr8_xan <- read.table(paste0(outputDir_fastClines,'Chr8_xan_clines_demeOpt1.txt'), header=T,fill=T,na.strings = "NA")

Chr8_clines <- rbind(Chr8_xab,Chr8_xac,Chr8_xad,Chr8_xae,Chr8_xaf,Chr8_xag,Chr8_xah,Chr8_xai,Chr8_xaj,Chr8_xak,Chr8_xal,Chr8_xam,Chr8_xan)
rm(Chr8_xab,Chr8_xac,Chr8_xad,Chr8_xae,Chr8_xaf,Chr8_xag,Chr8_xah,Chr8_xai,Chr8_xaj,Chr8_xak,Chr8_xal,Chr8_xam,Chr8_xan)


# Write each Chromosome to file
write.csv(Chr1_clines,"Chr1_clines.csv",row.names = F)
write.csv(Chr2_clines,"Chr2_clines.csv",row.names = F)
write.csv(Chr3_clines,"Chr3_clines.csv",row.names = F)
write.csv(Chr4_clines,"Chr4_clines.csv",row.names = F)
write.csv(Chr5_clines,"Chr5_clines.csv",row.names = F)
write.csv(Chr6_clines,"Chr6_clines.csv",row.names = F)
write.csv(Chr7_clines,"Chr7_clines.csv",row.names = F)
write.csv(Chr8_clines,"Chr8_clines.csv",row.names = F)
  
}


```

```{r import concatenated sliding windows outputs}
Chr1_Windows <- read.csv(paste0(workingDir,'Chr1_SlidingWindows.csv'), header=T,fill=T,na.strings = "NA")
Chr2_Windows <- read.csv(paste0(workingDir,'Chr2_SlidingWindows.csv'), header=T,fill=T,na.strings = "NA")
Chr3_Windows <- read.csv(paste0(workingDir,'Chr3_SlidingWindows.csv'), header=T,fill=T,na.strings = "NA")
Chr4_Windows <- read.csv(paste0(workingDir,'Chr4_SlidingWindows.csv'), header=T,fill=T,na.strings = "NA")
Chr5_Windows <- read.csv(paste0(workingDir,'Chr5_SlidingWindows.csv'), header=T,fill=T,na.strings = "NA")
Chr6_Windows <- read.csv(paste0(workingDir,'Chr6_SlidingWindows.csv'), header=T,fill=T,na.strings = "NA")
Chr7_Windows <- read.csv(paste0(workingDir,'Chr7_SlidingWindows.csv'), header=T,fill=T,na.strings = "NA")
Chr8_Windows <- read.csv(paste0(workingDir,'Chr8_SlidingWindows.csv'), header=T,fill=T,na.strings = "NA")

#colnames(Chr1_Windows)
#par(mfrow=c(1,1),mar=c(5,5,4,4))
#plot(Chr1_Windows[,"windowStart"],Chr1_Windows$FstfromMeanPiAdj_15_20)
#plot(Chr1_Windows[,"windowStart"],Chr1_Windows$FstfromMeanPiAdj_16_17)

# Merged version
allChr_Windows <- rbind(Chr1_Windows,Chr2_Windows,Chr3_Windows,Chr4_Windows,Chr5_Windows,Chr6_Windows,Chr7_Windows,Chr8_Windows)
rm(Chr1_Windows,Chr2_Windows,Chr3_Windows,Chr4_Windows,Chr5_Windows,Chr6_Windows,Chr7_Windows,Chr8_Windows)



nrow(allChr_Windows)*10000
```

```{r import concatenated fastClines outputs }
Chr1_clines <- read.csv(paste0(workingDir,'Chr1_clines.csv'), header=T,fill=T,na.strings = "NA")
Chr2_clines <- read.csv(paste0(workingDir,'Chr2_clines.csv'), header=T,fill=T,na.strings = "NA")
Chr3_clines <- read.csv(paste0(workingDir,'Chr3_clines.csv'), header=T,fill=T,na.strings = "NA")
Chr4_clines <- read.csv(paste0(workingDir,'Chr4_clines.csv'), header=T,fill=T,na.strings = "NA")
Chr5_clines <- read.csv(paste0(workingDir,'Chr5_clines.csv'), header=T,fill=T,na.strings = "NA")
Chr6_clines <- read.csv(paste0(workingDir,'Chr6_clines.csv'), header=T,fill=T,na.strings = "NA")
Chr7_clines <- read.csv(paste0(workingDir,'Chr7_clines.csv'), header=T,fill=T,na.strings = "NA")
Chr8_clines <- read.csv(paste0(workingDir,'Chr8_clines.csv'), header=T,fill=T,na.strings = "NA")

# Merged version
allChr_Clines <- rbind(Chr1_clines,Chr2_clines,Chr3_clines,Chr4_clines,Chr5_clines,Chr6_clines,Chr7_clines,Chr8_clines)
rm(Chr1_clines,Chr2_clines,Chr3_clines,Chr4_clines,Chr5_clines,Chr6_clines,Chr7_clines,Chr8_clines)

# checking ROSel positions
# allChr_Clines[allChr_Clines$LG==6 & allChr_Clines$position>52889000 & allChr_Clines$position<52889737,"position"]

```

```{r basic window stats coverage and polymorphism plus first filtering of slidingWindows data}
nrow(allChr_Windows)
nrow(allChr_Windows) * 10000 
# 491 Mbp, higher than amount estimated in original sync files. 
# but coverage varies a lot, so this will come substantially down
par(mfrow=c(1,1))
par(mar=c(4,4,4,4))
hist(allChr_Windows[,"coverage"],xlab="window coverage",ylab="frequency",main="")
# median 0.797
# mean 0.682
median(allChr_Windows[,"coverage"]) 
mean(allChr_Windows[,"coverage"]) 

sum(allChr_Windows[,"coverage"]>0.1)/nrow(allChr_Windows)

# estimate of coverage with min depth = 335.6 Mbp (65.9% of 509Mbp genome)
sum(allChr_Windows[,"coverage"]*rep(10000,nrow(allChr_Windows)))

#table(allChr_Windows[,"scaffold"])
#sum(table(allChr_Windows[,"scaffold"]))

# check a missing window example near CREMOSA clines to see when it gets filtered out
colnames(allChr_Windows)[1:15]
allChr_Windows[1:20,1:6]
plot(allChr_Windows[,"windowStart"])
# there are a few jumps (missing windows)
# plot(allChr_Windows[allChr_Windows[,"LG"]==1,"windowStart"])
# plot(allChr_Windows[allChr_Windows[,"LG"]==2,"windowStart"])
# plot(allChr_Windows[allChr_Windows[,"LG"]==3,"windowStart"])
# plot(allChr_Windows[allChr_Windows[,"LG"]==4,"windowStart"])
# plot(allChr_Windows[allChr_Windows[,"LG"]==5,"windowStart"])
# plot(allChr_Windows[allChr_Windows[,"LG"]==6,"windowStart"])
# plot(allChr_Windows[allChr_Windows[,"LG"]==7,"windowStart"])
# plot(allChr_Windows[allChr_Windows[,"LG"]==8,"windowStart"])
# 
# range(allChr_Windows[allChr_Windows[,"LG"]==1,"windowStart"])
# range(allChr_Windows[allChr_Windows[,"LG"]==2,"windowStart"])
# range(allChr_Windows[allChr_Windows[,"LG"]==3,"windowStart"])
# range(allChr_Windows[allChr_Windows[,"LG"]==4,"windowStart"])
# range(allChr_Windows[allChr_Windows[,"LG"]==5,"windowStart"])

# NaN NAs from sliding windows
# removing minus Fst
#create temporary variable where NAs are removed

allChr_Windows <- allChr_Windows[which(!is.na(allChr_Windows$FstfromMeanPiAdj_1_8)),]
#remove windows with lots of missing data (low coverage)
#temp.fst10 <- temp.fst10[which(temp.fst10$coverage>0.1),]
# set any Fst <0 to zero
allChr_Windows[which(allChr_Windows$FstfromMeanPiAdj_1_8<0),"FstfromMeanPiAdj_1_8"] <- 0
# any(allChr_Windows$FstfromMeanPiAdj_1_8=="NaN")
#hist(as.numeric(temp.fst10$FstfromMeanPiAdj_1_8))
# range(as.numeric(allChr_Windows$FstfromMeanPiAdj_1_8))


# Update - not removing right now, need them all for proportion analysis of clines and windows
# remove low coverage <0.1 windows
allChr_Windows <- allChr_Windows[which(allChr_Windows$coverage>0.1),]

# polymorphism
summary(allChr_Windows[,"poly_pool_15"])
summary(allChr_Windows[,"poly_pool_16"])
summary(allChr_Windows[,"poly_pool_17"])
summary(allChr_Windows[,"poly_pool_18"])
summary(allChr_Windows[,"poly_pool_19"])
summary(allChr_Windows[,"poly_pool_20"])
 
# pi_w
colnames(allChr_Windows)
summary(allChr_Windows[,"piAdj_15"])
summary(allChr_Windows[,"piAdj_16"])
summary(allChr_Windows[,"piAdj_17"])
summary(allChr_Windows[,"piAdj_18"])
summary(allChr_Windows[,"piAdj_19"])
summary(allChr_Windows[,"piAdj_20"])
#allChr_Windows[1,1:26]

# average depth & SE
summary(allChr_Windows[,"dpthAdj_15"])
sd(allChr_Windows[,"dpthAdj_15"])
sd(allChr_Windows[,"dpthAdj_15"])/sqrt(mean(summary(allChr_Windows[,"dpthAdj_15"])))

summary(allChr_Windows[,"dpthAdj_16"])
sd(allChr_Windows[,"dpthAdj_16"])
sd(allChr_Windows[,"dpthAdj_16"])/sqrt(mean(summary(allChr_Windows[,"dpthAdj_16"])))

summary(allChr_Windows[,"dpthAdj_17"])
sd(allChr_Windows[,"dpthAdj_17"])
sd(allChr_Windows[,"dpthAdj_17"])/sqrt(mean(summary(allChr_Windows[,"dpthAdj_17"])))

summary(allChr_Windows[,"dpthAdj_18"])
sd(allChr_Windows[,"dpthAdj_18"])
sd(allChr_Windows[,"dpthAdj_18"])/sqrt(mean(summary(allChr_Windows[,"dpthAdj_18"])))

summary(allChr_Windows[,"dpthAdj_19"])
sd(allChr_Windows[,"dpthAdj_19"])
sd(allChr_Windows[,"dpthAdj_19"])/sqrt(mean(summary(allChr_Windows[,"dpthAdj_19"])))

summary(allChr_Windows[,"dpthAdj_20"])
sd(allChr_Windows[,"dpthAdj_20"])
sd(allChr_Windows[,"dpthAdj_20"])/sqrt(mean(summary(allChr_Windows[,"dpthAdj_20"])))

# This many sites with delta p > 0.9 
# (3847/(2.1*(10^7)))*100

# This many sites with delta p = 1 
# (2319/(2.1*(10^7)))*100

```

```{r making ploting position for windows across Chromosomes}
# x-value approach (to make gaps between LG)
#colnames(allChr_Windows)
#unique(allChr_Windows$scaffold)
#unique(allChr_Windows$LG)

midPointWindow <- 5000
currentScaff <- as.vector(allChr_Windows[1,"scaffold"])
currentLG <- as.vector(allChr_Windows[1,"LG"])

graphPos <- matrix(0,nrow(allChr_Windows),8)
colnames(graphPos) <- c("LG","scaff","windowStart","windowEnd","windowMid","windowStart_graph","windowEnd_graph","windowMid_graph")
graphPos[,] <- -9
graphPos <- as.data.frame(graphPos)
windowAdd <- 0
for (thisWindow in 1:nrow(allChr_Windows)) {
  # thisWindow <- 1
  graphPos[thisWindow,"LG"] <- as.vector(allChr_Windows[thisWindow,"LG"])
  graphPos[thisWindow,"scaff"] <- as.vector(allChr_Windows[thisWindow,"scaffold"])
  if (allChr_Windows[thisWindow,"LG"]==currentLG) {
    graphPos[thisWindow,"windowStart"] <- allChr_Windows[thisWindow,"windowStart"]
    graphPos[thisWindow,"windowEnd"] <- allChr_Windows[thisWindow,"windowEnd"]
    graphPos[thisWindow,"windowMid"] <- allChr_Windows[thisWindow,"windowStart"]+midPointWindow
    graphPos[thisWindow,"windowStart_graph"] <- graphPos[thisWindow,"windowStart"]+windowAdd
    graphPos[thisWindow,"windowEnd_graph"] <- graphPos[thisWindow,"windowEnd"]+windowAdd
    graphPos[thisWindow,"windowMid_graph"] <- graphPos[thisWindow,"windowStart_graph"]+midPointWindow
  }
  if (allChr_Windows[thisWindow,"LG"]!=currentLG) {
    # change current LG
    currentLG <- as.vector(allChr_Windows[thisWindow,"LG"])
    # add size of previous LG to new one
    windowAdd <- graphPos[thisWindow-1,"windowEnd"]+windowAdd
    graphPos[thisWindow,"windowStart"] <- allChr_Windows[thisWindow,"windowStart"]
    graphPos[thisWindow,"windowEnd"] <- allChr_Windows[thisWindow,"windowEnd"]
    graphPos[thisWindow,"windowMid"] <- allChr_Windows[thisWindow,"windowStart"]+midPointWindow
    graphPos[thisWindow,"windowStart_graph"] <- graphPos[thisWindow,"windowStart"]+windowAdd
    graphPos[thisWindow,"windowEnd_graph"] <- graphPos[thisWindow,"windowEnd"]+windowAdd
    graphPos[thisWindow,"windowMid_graph"] <- graphPos[thisWindow,"windowStart_graph"]+midPointWindow
  }
}
 # check it worked
#plot(graphPos[,"windowMid_graph"],allChr_Windows$FstfromMeanPiAdj_15_20,ylim=(c(0,0.8)))

#all(graphPos[,"scaff"]==as.vector(allChr_Windows[,"scaffold"]))
#all(graphPos[,"windowStart"]==as.vector(allChr_Windows[,"windowStart"]))
#all(graphPos[,"LG"]==as.vector(allChr_Windows[,"LG"]))

# Add a gap between each linkage group
graphPos[graphPos[,"LG"]==2,6:8] <- graphPos[graphPos[,"LG"]==2,6:8]+10000000
graphPos[graphPos[,"LG"]==3,6:8] <- graphPos[graphPos[,"LG"]==3,6:8]+20000000
graphPos[graphPos[,"LG"]==4,6:8] <- graphPos[graphPos[,"LG"]==4,6:8]+30000000
graphPos[graphPos[,"LG"]==5,6:8] <- graphPos[graphPos[,"LG"]==5,6:8]+40000000
graphPos[graphPos[,"LG"]==6,6:8] <- graphPos[graphPos[,"LG"]==6,6:8]+50000000
graphPos[graphPos[,"LG"]==7,6:8] <- graphPos[graphPos[,"LG"]==7,6:8]+60000000
graphPos[graphPos[,"LG"]==8,6:8] <- graphPos[graphPos[,"LG"]==8,6:8]+70000000

# check gaps are present
#plot(graphPos[,"windowMid_graph"],allChr_Windows$FstfromMeanPiAdj_15_20,ylim=(c(0,0.8)),xlab="Fst",ylab="Genome position (bp)")
# combine with main data
allChr_Windows <- cbind(allChr_Windows,graphPos[,5:8])
```

```{r combining flower colour locations with windows data}
# import in flower gene location file
colourGenes <- read.csv(paste0(workingDir,'ColourGenes_refGenomes_Rimport.csv'), header=T,fill=T,na.strings = "NA")

colourGene_add <- matrix(-9,nrow(allChr_Windows),3)
colnames(colourGene_add) <- c("ColourGene","Position_start","Position_end")

# CREMOSA_1 & 2
# note, because CRE1 and CRE2 are in the same window, I have combined the start positions
# CREMOSA_1: [928205:930025,] is in window 84 & 85
# CREMOSA_2: [921293:925659,] is in window 84

allChr_Windows_LG1rows <- allChr_Windows[,"LG"]==1
thisColourGene_start <- 928205
startProx <- allChr_Windows[allChr_Windows_LG1rows & which.min(abs(allChr_Windows[,"windowStart"] - thisColourGene_start)),"windowStart"]
colourGene_add[84,"ColourGene"] <- "CREMOSA_1&2"
colourGene_add[85,"ColourGene"] <- "CREMOSA_1&2"
colourGene_add[84,"Position_start"] <- 921293
colourGene_add[84,"Position_end"] <- 930000
colourGene_add[85,"Position_start"] <- 930000
colourGene_add[85,"Position_end"] <- 930025

# adding linked windows +/- 30kb
colourGene_add[80:83,"ColourGene"]<- "CREMOSA_1&2_linked"
colourGene_add[86:88,"ColourGene"]<- "CREMOSA_1&2_linked"

# CREMOSA_3
# CREMOSA_3: [994948:995277,] is in row 91
thisLG <- 1
allChr_Windows_LG1rows <- allChr_Windows[,"LG"]==thisLG
thisColourGene_start <- 994948
CRE3_windows <- allChr_Windows %>%
                filter(LG==thisLG) %>%
                filter(thisColourGene_start>windowStart & thisColourGene_start<windowEnd)
CRE3_row <- which(allChr_Windows[,"LG"]==thisLG & allChr_Windows[,"windowStart"]==CRE3_windows[,"windowStart"])
colourGene_add[CRE3_row,"ColourGene"] <- "CREMOSA_3"
colourGene_add[CRE3_row,"Position_start"] <- 994948
colourGene_add[CRE3_row,"Position_end"] <- 995277

# adding linked windows +/- 30kb
colourGene_add[89:90,"ColourGene"]<- "CREMOSA_3_linked"
colourGene_add[92:94,"ColourGene"]<- "CREMOSA_3_linked"

# AURINA
# AURINA LG2: [1051685:1053653,] is in row 6697
allChr_Windows_LG2rows <- allChr_Windows[,"LG"]==2
thisColourGene_start <- 1051685
AURINA_windows <- allChr_Windows %>%
                filter(LG==2) %>%
                filter(thisColourGene_start>windowStart & thisColourGene_start<windowEnd)
AURINA_row <- which(allChr_Windows[,"LG"]==2 & allChr_Windows[,"windowStart"]==AURINA_windows[,"windowStart"])
colourGene_add[AURINA_row,"ColourGene"] <- "AURINA"
colourGene_add[AURINA_row,"Position_start"] <- 1051685
colourGene_add[AURINA_row,"Position_end"] <- 1053653

# adding linked windows +/- 30kb
colourGene_add[6693:6696,"ColourGene"]<- "AURINA_linked"
colourGene_add[6698:6700,"ColourGene"]<- "AURINA_linked"

# FLAVIA
# FLAVIA LG2: [53468062:53469435,] is in row 11388
allChr_Windows_LG2rows <- allChr_Windows[,"LG"]==2
thisColourGene_start <- 53468062
FLAVIA_windows <- allChr_Windows %>%
                filter(LG==2) %>%
                filter(thisColourGene_start>windowStart & thisColourGene_start<windowEnd)
FLAVIA_row <- which(allChr_Windows[,"LG"]==2 & allChr_Windows[,"windowStart"]==FLAVIA_windows[,"windowStart"])
colourGene_add[FLAVIA_row,"ColourGene"] <- "FLAVIA"
colourGene_add[FLAVIA_row,"Position_start"] <- 53468062
colourGene_add[FLAVIA_row,"Position_end"] <- 53469435

# because FLAVIA an area of low recombination, adding linked windows +/- 200kb
colourGene_add[11367:11387,"ColourGene"]<- "FLAVIA_linked"
colourGene_add[11389:11409,"ColourGene"]<- "FLAVIA_linked"


# SULF
# SULF LG4: [38348572: 38350336,] is in row 22444 & 22445
# note sulf goes over two windows, so the information has been split between them (Postion_start to end of 1st window: start of 2nd window to gene end)
allChr_Windows_LG4rows <- allChr_Windows[,"LG"]==4
thisColourGene_start <- 38348572
SULF_windows <- allChr_Windows %>%
                filter(LG==4) %>%
                filter(thisColourGene_start>windowStart & thisColourGene_start<windowEnd)
SULF_row <- which(allChr_Windows[,"LG"]==4 & allChr_Windows[,"windowStart"]==SULF_windows[,"windowStart"])
allChr_Windows[SULF_row,"windowStart"]; allChr_Windows[SULF_row,"windowEnd"]
colourGene_add[SULF_row,"ColourGene"] <- "SULF"
colourGene_add[SULF_row+1,"ColourGene"] <- "SULF"
colourGene_add[SULF_row,"Position_start"] <- 38348572
colourGene_add[SULF_row,"Position_end"] <- 38350000
colourGene_add[SULF_row+1,"Position_start"] <- 38350000
colourGene_add[SULF_row+1,"Position_end"] <- 38350336

# adding linked windows +/- 30kb
colourGene_add[22441:22443,"ColourGene"]<- "SULF_linked"
colourGene_add[22446:22449,"ColourGene"]<- "SULF_linked"

# RUBIA
# RUBIA LG5: [6269966: 6272151,] is in row 24622
allChr_Windows_LG5rows <- allChr_Windows[,"LG"]==5
thisColourGene_start <- 6269966
RUBIA_windows <- allChr_Windows %>%
                filter(LG==5) %>%
                filter(thisColourGene_start>windowStart & thisColourGene_start<windowEnd)
RUBIA_row <- which(allChr_Windows[,"LG"]==5 & allChr_Windows[,"windowStart"]==RUBIA_windows[,"windowStart"])
colourGene_add[RUBIA_row,"ColourGene"] <- "RUBIA"
colourGene_add[RUBIA_row,"Position_start"] <- 6269966
colourGene_add[RUBIA_row,"Position_end"] <- 6272151

# adding linked windows +/- 30kb
colourGene_add[24619:24621,"ColourGene"]<- "RUBIA_linked"
colourGene_add[24623:24625,"ColourGene"]<- "RUBIA_linked"

# ROS1
# ROS1 LG6: [52889323: 52889737,] is in row 35540
allChr_Windows_LG6rows <- allChr_Windows[,"LG"]==6
thisColourGene_start <- 52889323
ROS1_windows <- allChr_Windows %>%
                filter(LG==6) %>%
                filter(thisColourGene_start>windowStart & thisColourGene_start<windowEnd)
ROS1_row <- which(allChr_Windows[,"LG"]==6 & allChr_Windows[,"windowStart"]==ROS1_windows[,"windowStart"])
allChr_Windows[ROS1_row,"windowStart"]
colourGene_add[ROS1_row,"ColourGene"] <- "ROS_1"
colourGene_add[ROS1_row,"Position_start"] <- 52889323
colourGene_add[ROS1_row,"Position_end"] <- 52889737

# adding linked windows - 30kb and +20kb until it hits ROS2
colourGene_add[35537:35539,"ColourGene"]<- "ROS1_linked"
colourGene_add[35541:35542,"ColourGene"]<- "ROS1_linked"

# ROS2
# ROS2 LG6: [52,912,351:52,912,777,] is in row 35543
allChr_Windows_LG6rows <- allChr_Windows[,"LG"]==6
thisColourGene_start <- 52912351
ROS2_windows <- allChr_Windows %>%
                filter(LG==6) %>%
                filter(thisColourGene_start>windowStart & thisColourGene_start<windowEnd)
ROS2_row <- which(allChr_Windows[,"LG"]==6 & allChr_Windows[,"windowStart"]==ROS2_windows[,"windowStart"])
allChr_Windows[ROS2_row,"windowStart"]
colourGene_add[ROS2_row,"ColourGene"] <- "ROS_2"
colourGene_add[ROS2_row,"Position_start"] <- 52912351
colourGene_add[ROS2_row,"Position_end"] <- 52912777

# we cant put ROS2 linked, because its right next to ROS_3 on one side and only a few windows to ROS_1 on the other

# ROS3
# ROS3 LG6: [52,919,558:52,922,055,] is in row 35543
# Note ROS3 starts in same window as ROS2 but finishes in the next window
# I shifted ROS3 to the next window. Justified on the basis that the majority of the gene is on the next window
allChr_Windows_LG6rows <- allChr_Windows[,"LG"]==6
thisColourGene_start <- 52919558
ROS3_windows <- allChr_Windows %>%
                filter(LG==6) %>%
                filter(thisColourGene_start>windowStart & thisColourGene_start<windowEnd)
ROS3_row <- which(allChr_Windows[,"LG"]==6 & allChr_Windows[,"windowStart"]==ROS3_windows[,"windowStart"])
allChr_Windows[ROS3_row,"windowStart"]
colourGene_add[ROS3_row+1,"ColourGene"] <- "ROS_3"
#colourGene_add[ROS3_row,"Position_start"] <- 52919558
colourGene_add[ROS3_row+1,"Position_start"] <- 52920000
colourGene_add[ROS3_row+1,"Position_end"] <- 52922055

# adding linked windows linked to all windows between ROS3 and ELUTA
colourGene_add[35545:35557,"ColourGene"]<- "ROS_EL_barrier"

# ELUTA
# ELUTA LG6: [53060761:53062505,] is in row 35558
allChr_Windows_LG6rows <- allChr_Windows[,"LG"]==6
thisColourGene_start <- 53060761
EL_windows <- allChr_Windows %>%
                filter(LG==6) %>%
                filter(thisColourGene_start>windowStart & thisColourGene_start<windowEnd)
EL_row <- which(allChr_Windows[,"LG"]==6 & allChr_Windows[,"windowStart"]==EL_windows[,"windowStart"])
allChr_Windows[EL_row,"windowStart"]
colourGene_add[EL_row,"ColourGene"] <- "ELUTA"
colourGene_add[EL_row,"Position_start"] <- 53060761
colourGene_add[EL_row,"Position_end"] <- 53062505

# adding linked windows downstream of EL +30kb
colourGene_add[35559:35561,"ColourGene"]<- "EL_linked"

allChr_Windows <- cbind(allChr_Windows,colourGene_add)

plotMe = F
if (plotMe==T) {
  plot(allChr_Windows[,"windowMid_graph"],allChr_Windows$FstfromMeanPiAdj_15_20,ylim=(c(0,0.8)),ylab="Fst",xlab="Genome position (bp)")
  points(x=allChr_Windows[EL_row,"windowMid_graph"],y=allChr_Windows[EL_row,"FstfromMeanPiAdj_15_20"],pch=19,col='pink')
  points(x=allChr_Windows[ROS2_row,"windowMid_graph"],y=allChr_Windows[ROS2_row,"FstfromMeanPiAdj_15_20"],pch=19,col='red')
  points(x=allChr_Windows[ROS1_row,"windowMid_graph"],y=allChr_Windows[ROS1_row,"FstfromMeanPiAdj_15_20"],pch=19,col='red')
  points(x=allChr_Windows[RUBIA_row,"windowMid_graph"],y=allChr_Windows[RUBIA_row,"FstfromMeanPiAdj_15_20"],pch=19,col='blue')
  points(x=allChr_Windows[FLAVIA_row,"windowMid_graph"],y=allChr_Windows[FLAVIA_row,"FstfromMeanPiAdj_15_20"],pch=19,col='yellow')
  points(x=allChr_Windows[SULF_row,"windowMid_graph"],y=allChr_Windows[SULF_row,"FstfromMeanPiAdj_15_20"],pch=19,col='orange')
  points(x=allChr_Windows[SULF_row,"windowMid_graph"],y=allChr_Windows[SULF_row,"FstfromMeanPiAdj_15_20"],pch=19,col='orange')
  points(x=allChr_Windows[84,"windowMid_graph"],y=allChr_Windows[84,"FstfromMeanPiAdj_15_20"],pch=19,col='green')
  points(x=allChr_Windows[85,"windowMid_graph"],y=allChr_Windows[85,"FstfromMeanPiAdj_15_20"],pch=19,col='green')
  points(x=allChr_Windows[91,"windowMid_graph"],y=allChr_Windows[91,"FstfromMeanPiAdj_15_20"],pch=19,col='green')
  
  # adding some linked windows to FLAVIA
  points(x=allChr_Windows[11367:11387,"windowMid_graph"],y=allChr_Windows[11367:11387,"FstfromMeanPiAdj_15_20"],pch=19,col='gold')
  points(x=allChr_Windows[11389:11409,"windowMid_graph"],y=allChr_Windows[11389:11409,"FstfromMeanPiAdj_15_20"],pch=19,col='gold')
  
  # adding some linked windows to FLAVIA
  points(x=allChr_Windows[11367:11387,"windowMid_graph"],y=allChr_Windows[11367:11387,"FstfromMeanPiAdj_15_20"],pch=19,col='gold')
  points(x=allChr_Windows[11389:11409,"windowMid_graph"],y=allChr_Windows[11389:11409,"FstfromMeanPiAdj_15_20"],pch=19,col='gold')
  
}

# checking if EL still present
tail(colnames(allChr_Windows))
table(allChr_Windows[,"ColourGene"])

```

```{r making new plotting position for clines across Chromosomes}
graphPos_clines <- matrix(0,nrow(allChr_Clines),4)
colnames(graphPos_clines) <- c("LG","scaff","position","newPlotPosCline")
graphPos_clines[,] <- -9
graphPos_clines <- as.data.frame(graphPos_clines)
windowAdd <- 0
currentLG <- 1
allChr_Windows_LG <- allChr_Windows[allChr_Windows[,"LG"]==currentLG,]

for (thisRow in 1:nrow(allChr_Clines)) {
  # thisRow <- 1
  thisPos <- allChr_Clines[thisRow,"position"]
  thisLG <- allChr_Clines[thisRow,"LG"]
  if (allChr_Clines[thisRow,"LG"]!=currentLG) {
    allChr_Windows_LG <- allChr_Windows[allChr_Windows[,"LG"]==thisLG,]
    currentLG <- thisLG
  }
  closestWindow <- allChr_Windows_LG[which.min(abs(allChr_Windows_LG[,"windowMid"] - thisPos)),"windowMid"]
  if (thisPos < closestWindow) {
    differenceWind <- abs(closestWindow-thisPos)
    newPlotPos <- allChr_Windows_LG[allChr_Windows_LG[,"windowMid"]==closestWindow,"windowMid_graph"]-differenceWind
    graphPos_clines[thisRow,"newPlotPosCline"] <- newPlotPos
  }
  if (thisPos > closestWindow) {
    differenceWind <- abs(closestWindow-thisPos)
    newPlotPos <- allChr_Windows_LG[allChr_Windows_LG[,"windowMid"]==closestWindow,"windowMid_graph"]+differenceWind
    graphPos_clines[thisRow,"newPlotPosCline"] <- newPlotPos
  }
}

# combine with main data
allChr_Clines <- cbind(allChr_Clines,graphPos_clines[,4])
colnames(allChr_Clines)[ncol(allChr_Clines)] <- "position_graph"
allChr_Clines <- allChr_Clines %>% relocate(position_graph, .after=position)
# colnames(allChr_Clines)

# checking Ros1 marker: ros_assembly_543443

# BLASTING positions from Daniel
# Chr6: 52,889,323 -  52,889,737

# lifting positions from Frank & Arka:
#Chr6	52889073	52889074	ros_assembly_543443	1000	+

# SNP loci in the region from pool Seq:
# SNP loci in PoolSeq data at:
#allChr_Clines[allChr_Clines$LG==6 & allChr_Clines$position>52888073 & allChr_Clines$position<52889737,"position"]
# within ROS1
#allChr_Clines[allChr_Clines$LG==6 & allChr_Clines$position>52889323 & allChr_Clines$position<52889737,"position"]

# no loci close to the positions from Frank & Arka, but there are SNP loci in pool seq within the ROS1 region from Daniel

# revisiting from Daniels email 8/3/23
# A/G SNP at 52889078
# A/T SNP at 52889080
#allChr_Clines[allChr_Clines$LG==6 & allChr_Clines$position>52889000 & allChr_Clines$position<52889737,"position"]

# problem solved. It was an issue with FastClines filtering. Those KASP clines has <15 depth in pool 20 so they were cut. Note current run of FastClines with lower filtering (10 depth min) and I will apply stricter filtering here in post analysis.

```

```{r combining flower colour locations with clines data}
# this code links up the colour genes to clines. Produces the outputs that describe whether the position is 'inGene', 'up30kb', 'down30kb' etc..

# import in flower gene location file

# range(allChr_Clines[,"position"])

colourGene_add <- matrix(-9,nrow(allChr_Clines),2)
colnames(colourGene_add) <- c("ColourGene","distanceToGene")

# CREMOSA_1 
CREMOSA_1_gene <- c(928205,930025)
#allChr_Windows_LG1rows <- allChr_Windows[,"LG"]==1
# within gene - NOTE there are no loci with clines within the coding sequences of CRE_1
CREMOSA_1 <- which(allChr_Clines[,"LG"]==1 & allChr_Clines[,"position"] > CREMOSA_1_gene[1] & allChr_Clines[,"position"] < CREMOSA_1_gene[2])
# +/- 30kb & 100kb
CREMOSA_1_up100 <- which(allChr_Clines[,"LG"]==1 & abs(allChr_Clines[,"position"] - CREMOSA_1_gene[1]) < 100000 & abs(allChr_Clines[,"position"] - CREMOSA_1_gene[1]) > 30000 & allChr_Clines[,"position"] < CREMOSA_1_gene[1])
CREMOSA_1_up30 <- which(allChr_Clines[,"LG"]==1 & abs(allChr_Clines[,"position"] - CREMOSA_1_gene[1]) < 30000 & allChr_Clines[,"position"] < CREMOSA_1_gene[1])
CREMOSA_1_down100 <- which(allChr_Clines[,"LG"]==1 & abs(allChr_Clines[,"position"] - CREMOSA_1_gene[2]) < 100000 & abs(allChr_Clines[,"position"] - CREMOSA_1_gene[2]) > 30000 & allChr_Clines[,"position"] > CREMOSA_1_gene[2])
CREMOSA_1_down30 <- which(allChr_Clines[,"LG"]==1 & abs(allChr_Clines[,"position"] - CREMOSA_1_gene[2]) < 30000 & allChr_Clines[,"position"] > CREMOSA_1_gene[2])

colourGene_add[CREMOSA_1_up100,"ColourGene"] <- "CREMOSA"
colourGene_add[CREMOSA_1_down100,"ColourGene"] <- "CREMOSA"
colourGene_add[CREMOSA_1_down30,"ColourGene"] <- "CREMOSA"

colourGene_add[CREMOSA_1_up100,"distanceToGene"] <- "up30_100kb"
colourGene_add[CREMOSA_1_down100,"distanceToGene"] <- "down30_100kb"
colourGene_add[CREMOSA_1_down30,"distanceToGene"] <- "down30kb"

# CREMOSA_2
CREMOSA_2_gene <- c(921293,925659)
# within gene - no clines within coding 
CREMOSA_2 <- which(allChr_Clines[,"LG"]==1 & allChr_Clines[,"position"] > CREMOSA_2_gene[1] & allChr_Clines[,"position"] < CREMOSA_2_gene[2])
# +/- 30kb & 30-100kb
CREMOSA_2_up100 <- which(allChr_Clines[,"LG"]==1 & abs(allChr_Clines[,"position"] - CREMOSA_2_gene[1]) < 100000 & abs(allChr_Clines[,"position"] - CREMOSA_2_gene[1]) > 30000 & allChr_Clines[,"position"] < CREMOSA_2_gene[1])
CREMOSA_2_up30 <- which(allChr_Clines[,"LG"]==1 & abs(allChr_Clines[,"position"] - CREMOSA_2_gene[1]) < 30000 & allChr_Clines[,"position"] < CREMOSA_2_gene[1])
CREMOSA_2_down100 <- which(allChr_Clines[,"LG"]==1 & abs(allChr_Clines[,"position"] - CREMOSA_2_gene[2]) < 100000 & abs(allChr_Clines[,"position"] - CREMOSA_2_gene[2]) > 30000 & allChr_Clines[,"position"] > CREMOSA_2_gene[2])
CREMOSA_2_down30 <- which(allChr_Clines[,"LG"]==1 & abs(allChr_Clines[,"position"] - CREMOSA_2_gene[2]) < 30000 & allChr_Clines[,"position"] > CREMOSA_2_gene[2])

# note many overlap with cremosa 1 , so we cannot differentiate
# colourGene_add[CREMOSA_2_up100,"ColourGene"] <- "CREMOSA"
# colourGene_add[CREMOSA_2_down100,"ColourGene"] <- "CREMOSA"
# colourGene_add[CREMOSA_1_down30,"ColourGene"] <- "CREMOSA"
# 
# colourGene_add[CREMOSA_1_up100,"distanceToGene"] <- "up30_100kb"
# colourGene_add[CREMOSA_1_down100,"distanceToGene"] <- "down30_100kb"
# colourGene_add[CREMOSA_1_down30,"distanceToGene"] <- "down30kb"

# CREMOSA_3
# CREMOSA_3: [994948:995277,] 
CREMOSA_3_gene <- c(994948,995277)

# within gene - note, no clines within coding sequence
CREMOSA_3 <- which(allChr_Clines[,"LG"]==1 & allChr_Clines[,"position"] > CREMOSA_3_gene[1] & allChr_Clines[,"position"] < CREMOSA_3_gene[2])
# +/- 30kb & 30-100kb
CREMOSA_3_up100 <- which(allChr_Clines[,"LG"]==1 & abs(allChr_Clines[,"position"] - CREMOSA_3_gene[1]) < 100000 & abs(allChr_Clines[,"position"] - CREMOSA_3_gene[1]) > 30000 & allChr_Clines[,"position"] < CREMOSA_3_gene[1])
CREMOSA_3_up30 <- which(allChr_Clines[,"LG"]==1 & abs(allChr_Clines[,"position"] - CREMOSA_3_gene[1]) < 30000 & allChr_Clines[,"position"] < CREMOSA_3_gene[1])
CREMOSA_3_down100 <- which(allChr_Clines[,"LG"]==1 & abs(allChr_Clines[,"position"] - CREMOSA_3_gene[2]) < 100000 &  abs(allChr_Clines[,"position"] - CREMOSA_3_gene[2]) > 30000 & allChr_Clines[,"position"] > CREMOSA_3_gene[2])
CREMOSA_3_down30 <- which(allChr_Clines[,"LG"]==1 & abs(allChr_Clines[,"position"] - CREMOSA_3_gene[2]) < 30000 & allChr_Clines[,"position"] > CREMOSA_3_gene[2])

# no clines that dont overlap with those linked to CRE 1 & 2

# AURINA
# AURINA LG2: [1051685:1053653,] 
AURINA_gene <- c(1051685,1053653)

# within gene - note, no clines within coding sequence
AURINA <- which(allChr_Clines[,"LG"]==2 & allChr_Clines[,"position"] > AURINA_gene[1] & allChr_Clines[,"position"] < AURINA_gene[2])
# +/- 30kb & 30-100kb
AURINA_up100 <- which(allChr_Clines[,"LG"]==2 & abs(allChr_Clines[,"position"] - AURINA_gene[1]) < 100000 & abs(allChr_Clines[,"position"] - AURINA_gene[1]) > 30000 & allChr_Clines[,"position"] < AURINA_gene[1])
AURINA_up30 <- which(allChr_Clines[,"LG"]==2 & abs(allChr_Clines[,"position"] - AURINA_gene[1]) < 30000 & allChr_Clines[,"position"] < AURINA_gene[1])
AURINA_down100 <- which(allChr_Clines[,"LG"]==2 & abs(allChr_Clines[,"position"] - AURINA_gene[2]) < 100000 &  abs(allChr_Clines[,"position"] - AURINA_gene[2]) > 30000 & allChr_Clines[,"position"] > AURINA_gene[2])
AURINA_down30 <- which(allChr_Clines[,"LG"]==2 & abs(allChr_Clines[,"position"] - AURINA_gene[2]) < 30000 & allChr_Clines[,"position"] > AURINA_gene[2])

colourGene_add[AURINA_up100,"ColourGene"] <- "AURINA"
colourGene_add[AURINA_down30,"ColourGene"] <- "AURINA"
colourGene_add[AURINA_up100,"distanceToGene"] <- "up30_100kb"
colourGene_add[AURINA_down30,"distanceToGene"] <- "down30kb"

# FLAVIA
# FLAVIA LG2: [53468062:53469435,] 
FLAVIA_gene <- c(53468062,53469435)
# within gene - 5 loci with clines within coding region
FLAVIA <- which(allChr_Clines[,"LG"]==2 & allChr_Clines[,"position"] > FLAVIA_gene[1] & allChr_Clines[,"position"] < FLAVIA_gene[2])
# +/- 30kb & 30-400kb (seems to platue off after 400kb)
FLAVIA_up400 <- which(allChr_Clines[,"LG"]==2 & abs(allChr_Clines[,"position"] - FLAVIA_gene[1]) < 400000 & abs(allChr_Clines[,"position"] - FLAVIA_gene[1]) > 30000 & allChr_Clines[,"position"] < FLAVIA_gene[1])
FLAVIA_up30 <- which(allChr_Clines[,"LG"]==2 & abs(allChr_Clines[,"position"] - FLAVIA_gene[1]) < 30000 & allChr_Clines[,"position"] < FLAVIA_gene[1])
FLAVIA_down400 <- which(allChr_Clines[,"LG"]==2 & abs(allChr_Clines[,"position"] - FLAVIA_gene[2]) < 400000 &  abs(allChr_Clines[,"position"] - FLAVIA_gene[2]) > 30000 & allChr_Clines[,"position"] > FLAVIA_gene[2])
FLAVIA_down30 <- which(allChr_Clines[,"LG"]==2 & abs(allChr_Clines[,"position"] - FLAVIA_gene[2]) < 30000 & allChr_Clines[,"position"] > FLAVIA_gene[2])

# because FLAVIA an area of low recombination, adding linked windows +/- 200kb

colourGene_add[FLAVIA_up400,"ColourGene"] <- "FLAVIA"
colourGene_add[FLAVIA_up30,"ColourGene"] <- "FLAVIA"
colourGene_add[FLAVIA_down400,"ColourGene"] <- "FLAVIA"
colourGene_add[FLAVIA_down30,"ColourGene"] <- "FLAVIA"
colourGene_add[FLAVIA,"ColourGene"] <- "FLAVIA"

colourGene_add[FLAVIA_up400,"distanceToGene"] <- "up30_400kb"
colourGene_add[FLAVIA_down400,"distanceToGene"] <- "down30_400kb"
colourGene_add[FLAVIA_up30,"distanceToGene"] <- "up30kb"
colourGene_add[FLAVIA_down30,"distanceToGene"] <- "down30kb"
colourGene_add[FLAVIA,"distanceToGene"] <- "inGene"


# SULF
# SULF LG4: [38348572: 38350336,] is in row 22444 & 22445

SULF_gene <- c(38348572,38350336)
# within gene - no loci with clines within coding region
SULF <- which(allChr_Clines[,"LG"]==4 & allChr_Clines[,"position"] > SULF_gene[1] & allChr_Clines[,"position"] < SULF_gene[2])
# +/- 30kb & 30-100kb 
SULF_up100 <- which(allChr_Clines[,"LG"]==4 & abs(allChr_Clines[,"position"] - SULF_gene[1]) < 100000 & abs(allChr_Clines[,"position"] - SULF_gene[1]) > 30000 & allChr_Clines[,"position"] < SULF_gene[1])
SULF_up30 <- which(allChr_Clines[,"LG"]==4 & abs(allChr_Clines[,"position"] - SULF_gene[1]) < 30000 & allChr_Clines[,"position"] < SULF_gene[1])
SULF_down100 <- which(allChr_Clines[,"LG"]==4 & abs(allChr_Clines[,"position"] - SULF_gene[2]) < 100000 &  abs(allChr_Clines[,"position"] - SULF_gene[2]) > 30000 & allChr_Clines[,"position"] > SULF_gene[2])
SULF_down30 <- which(allChr_Clines[,"LG"]==4 & abs(allChr_Clines[,"position"] - SULF_gene[2]) < 30000 & allChr_Clines[,"position"] > SULF_gene[2])

colourGene_add[SULF_up100,"ColourGene"] <- "SULF"
colourGene_add[SULF_up100,"distanceToGene"] <- "up30_100kb"
colourGene_add[SULF_down100,"ColourGene"] <- "SULF"
colourGene_add[SULF_down100,"distanceToGene"] <- "down30_100kb"
colourGene_add[SULF_up30,"ColourGene"] <- "SULF"
colourGene_add[SULF_up30,"distanceToGene"] <- "up30kb"
colourGene_add[SULF_down30,"ColourGene"] <- "SULF"
colourGene_add[SULF_down30,"distanceToGene"] <- "down30kb"

# repeat for special rerun of Chr4 5/12/2024
# colourGene_add_SULFspecial <- matrix(-9,nrow(Chr4_clinesNEW),2)
# colnames(colourGene_add_SULFspecial) <- c("ColourGene","distanceToGene")
# 
# # within gene - no loci with clines within coding region
# SULF <- which(Chr4_clinesNEW[,"LG"]==4 & Chr4_clinesNEW[,"position"] > SULF_gene[1] & Chr4_clinesNEW[,"position"] < SULF_gene[2])
# 
# # +/- 30kb & 30-100kb 
# SULF_up100 <- which(Chr4_clinesNEW[,"LG"]==4 & abs(Chr4_clinesNEW[,"position"] - SULF_gene[1]) < 100000 & abs(Chr4_clinesNEW[,"position"] - SULF_gene[1]) > 30000 & Chr4_clinesNEW[,"position"] < SULF_gene[1])
# SULF_up30 <- which(Chr4_clinesNEW[,"LG"]==4 & abs(Chr4_clinesNEW[,"position"] - SULF_gene[1]) < 30000 & Chr4_clinesNEW[,"position"] < SULF_gene[1])
# SULF_down100 <- which(Chr4_clinesNEW[,"LG"]==4 & abs(Chr4_clinesNEW[,"position"] - SULF_gene[2]) < 100000 &  abs(Chr4_clinesNEW[,"position"] - SULF_gene[2]) > 30000 & Chr4_clinesNEW[,"position"] > SULF_gene[2])
# SULF_down30 <- which(Chr4_clinesNEW[,"LG"]==4 & abs(Chr4_clinesNEW[,"position"] - SULF_gene[2]) < 30000 & Chr4_clinesNEW[,"position"] > SULF_gene[2])
# 
# 
# colourGene_add_SULFspecial[SULF_up100,"ColourGene"] <- "SULF"
# colourGene_add_SULFspecial[SULF_up100,"distanceToGene"] <- "up30_100kb"
# colourGene_add_SULFspecial[SULF_down100,"ColourGene"] <- "SULF"
# colourGene_add_SULFspecial[SULF_down100,"distanceToGene"] <- "down30_100kb"
# colourGene_add_SULFspecial[SULF_up30,"ColourGene"] <- "SULF"
# colourGene_add_SULFspecial[SULF_up30,"distanceToGene"] <- "up30kb"
# colourGene_add_SULFspecial[SULF_down30,"ColourGene"] <- "SULF"
# colourGene_add_SULFspecial[SULF_down30,"distanceToGene"] <- "down30kb"

# RUBIA
# RUBIA LG5: [6269966: 6272151,] is in row 24622
RUBIA_gene <- c(6269966,6272151)
# within gene - 8 loci with clines within coding region
RUBIA <- which(allChr_Clines[,"LG"]==5 & allChr_Clines[,"position"] > RUBIA_gene[1] & allChr_Clines[,"position"] < RUBIA_gene[2])
# +/- 30kb & 30-100kb
RUBIA_up100 <- which(allChr_Clines[,"LG"]==5 & abs(allChr_Clines[,"position"] - RUBIA_gene[1]) < 100000 & abs(allChr_Clines[,"position"] - RUBIA_gene[1]) > 30000 & allChr_Clines[,"position"] < RUBIA_gene[1])
RUBIA_up30 <- which(allChr_Clines[,"LG"]==5 & abs(allChr_Clines[,"position"] - RUBIA_gene[1]) < 30000 & allChr_Clines[,"position"] < RUBIA_gene[1])
RUBIA_down100 <- which(allChr_Clines[,"LG"]==5 & abs(allChr_Clines[,"position"] - RUBIA_gene[2]) < 100000 &  abs(allChr_Clines[,"position"] - RUBIA_gene[2]) > 30000 & allChr_Clines[,"position"] > RUBIA_gene[2])
RUBIA_down30 <- which(allChr_Clines[,"LG"]==5 & abs(allChr_Clines[,"position"] - RUBIA_gene[2]) < 30000 & allChr_Clines[,"position"] > RUBIA_gene[2])

# because RUBIA an area of low recombination, adding linked windows +/- 200kb

colourGene_add[RUBIA_up100,"ColourGene"] <- "RUBIA"
colourGene_add[RUBIA_up30,"ColourGene"] <- "RUBIA"
colourGene_add[RUBIA_down100,"ColourGene"] <- "RUBIA"
colourGene_add[RUBIA_down30,"ColourGene"] <- "RUBIA"
colourGene_add[RUBIA,"ColourGene"] <- "RUBIA"

colourGene_add[RUBIA_up100,"distanceToGene"] <- "up30_100kb"
colourGene_add[RUBIA_down100,"distanceToGene"] <- "down30_100kb"
colourGene_add[RUBIA_up30,"distanceToGene"] <- "up30kb"
colourGene_add[RUBIA_down30,"distanceToGene"] <- "down30kb"
colourGene_add[RUBIA,"distanceToGene"] <- "inGene"


# ROS1 - note KASP marker at 52889078
# ROS1 LG6: [52889323:52889737,] is in row 35540
ROS1_gene <- c(52887400,52889737)
# within gene - 2 loci with clines within coding region
ROS1 <- which(allChr_Clines[,"LG"]==6 & allChr_Clines[,"position"] > ROS1_gene[1] & allChr_Clines[,"position"] < ROS1_gene[2])
# +/- 30kb & 30-100kb
ROS1_up100 <- which(allChr_Clines[,"LG"]==6 & abs(allChr_Clines[,"position"] - ROS1_gene[1]) < 100000 & abs(allChr_Clines[,"position"] - ROS1_gene[1]) > 30000 & allChr_Clines[,"position"] < ROS1_gene[1])
ROS1_up30 <- which(allChr_Clines[,"LG"]==6 & abs(allChr_Clines[,"position"] - ROS1_gene[1]) < 30000 & allChr_Clines[,"position"] < ROS1_gene[1])
ROS1_down100 <- which(allChr_Clines[,"LG"]==6 & abs(allChr_Clines[,"position"] - ROS1_gene[2]) < 100000 &  abs(allChr_Clines[,"position"] - ROS1_gene[2]) > 30000 & allChr_Clines[,"position"] > ROS1_gene[2])
ROS1_down30 <- which(allChr_Clines[,"LG"]==6 & abs(allChr_Clines[,"position"] - ROS1_gene[2]) < 30000 & allChr_Clines[,"position"] > ROS1_gene[2])

colourGene_add[ROS1_up100,"ColourGene"] <- "ROSEA"
colourGene_add[ROS1_up30,"ColourGene"] <- "ROSEA"
colourGene_add[ROS1_down100,"ColourGene"] <- "ROSEA"
colourGene_add[ROS1_down30,"ColourGene"] <- "ROSEA"
colourGene_add[ROS1,"ColourGene"] <- "ROS1"

colourGene_add[ROS1_up100,"distanceToGene"] <- "up30_100kb"
colourGene_add[ROS1_down100,"distanceToGene"] <- "down30_100kb"
colourGene_add[ROS1_up30,"distanceToGene"] <- "up30kb"
colourGene_add[ROS1_down30,"distanceToGene"] <- "down30kb"
colourGene_add[ROS1,"distanceToGene"] <- "inGene"
allChr_Clines[ROS1,"position"]
#allChr_Clines[ROS1,]
#allChr_Clines[ROS1_up30,]
#allChr_Clines[ROS1_down30,]


# ROS2
# ROS2 LG6: [52912351:52912777,] 
ROS2_gene <- c(52912351,52912777)
# within gene - 7 loci with clines within coding region
ROS2 <- which(allChr_Clines[,"LG"]==6 & allChr_Clines[,"position"] > ROS2_gene[1] & allChr_Clines[,"position"] < ROS2_gene[2])
# +/- 30kb & 30-100kb
ROS2_up100 <- which(allChr_Clines[,"LG"]==6 & abs(allChr_Clines[,"position"] - ROS2_gene[1]) < 100000 & abs(allChr_Clines[,"position"] - ROS2_gene[1]) > 30000 & allChr_Clines[,"position"] < ROS2_gene[1])
ROS2_up30 <- which(allChr_Clines[,"LG"]==6 & abs(allChr_Clines[,"position"] - ROS2_gene[1]) < 30000 & allChr_Clines[,"position"] < ROS2_gene[1])
ROS2_down100 <- which(allChr_Clines[,"LG"]==6 & abs(allChr_Clines[,"position"] - ROS2_gene[2]) < 100000 &  abs(allChr_Clines[,"position"] - ROS2_gene[2]) > 30000 & allChr_Clines[,"position"] > ROS2_gene[2])
ROS2_down30 <- which(allChr_Clines[,"LG"]==6 & abs(allChr_Clines[,"position"] - ROS2_gene[2]) < 30000 & allChr_Clines[,"position"] > ROS2_gene[2])

colourGene_add[ROS2_up100,"ColourGene"] <- "ROSEA"
colourGene_add[ROS2_up30,"ColourGene"] <- "ROSEA"
colourGene_add[ROS2_down100,"ColourGene"] <- "ROSEA"
colourGene_add[ROS2_down30,"ColourGene"] <- "ROSEA"
colourGene_add[ROS2,"ColourGene"] <- "ROS2"

colourGene_add[ROS2_up100,"distanceToGene"] <- "up30_100kb"
colourGene_add[ROS2_down100,"distanceToGene"] <- "down30_100kb"
colourGene_add[ROS2_up30,"distanceToGene"] <- "up30kb"
colourGene_add[ROS2_down30,"distanceToGene"] <- "down30kb"
colourGene_add[ROS2,"distanceToGene"] <- "inGene"



# we cant put ROS2 linked, because its right next to ROS_3 on one side and only a few windows to ROS_1 on the other

# ROS3
# ROS3 LG6: [52919558:52922055,] is in row 35543
# Note ROS3 starts in same window as ROS2 but finishes in the next window
# I shifted ROS3 to the next window. Justified on the basis that the majority of the gene is on the next window
ROS3_gene <- c(52919558,52922055)
# within gene - no loci with clines within coding region
ROS3 <- which(allChr_Clines[,"LG"]==6 & allChr_Clines[,"position"] > ROS3_gene[1] & allChr_Clines[,"position"] < ROS3_gene[2])
# +/- 30kb & 30-100kb
ROS3_up100 <- which(allChr_Clines[,"LG"]==6 & abs(allChr_Clines[,"position"] - ROS3_gene[1]) < 100000 & abs(allChr_Clines[,"position"] - ROS3_gene[1]) > 30000 & allChr_Clines[,"position"] < ROS3_gene[1])
ROS3_up30 <- which(allChr_Clines[,"LG"]==6 & abs(allChr_Clines[,"position"] - ROS3_gene[1]) < 30000 & allChr_Clines[,"position"] < ROS3_gene[1])
ROS3_down100 <- which(allChr_Clines[,"LG"]==6 & abs(allChr_Clines[,"position"] - ROS3_gene[2]) < 100000 &  abs(allChr_Clines[,"position"] - ROS3_gene[2]) > 30000 & allChr_Clines[,"position"] > ROS3_gene[2])
ROS3_down30 <- which(allChr_Clines[,"LG"]==6 & abs(allChr_Clines[,"position"] - ROS3_gene[2]) < 30000 & allChr_Clines[,"position"] > ROS3_gene[2])

colourGene_add[ROS3_up100,"ColourGene"] <- "ROSEA"
colourGene_add[ROS3_up30,"ColourGene"] <- "ROSEA"
colourGene_add[ROS3_down100,"ColourGene"] <- "ROSEA"
colourGene_add[ROS3_down30,"ColourGene"] <- "ROSEA"
# colourGene_add[ROS3,"ColourGene"] <- "ROS3"

colourGene_add[ROS3_up100,"distanceToGene"] <- "up30_100kb"
colourGene_add[ROS3_down100,"distanceToGene"] <- "down30_100kb"
colourGene_add[ROS3_up30,"distanceToGene"] <- "up30kb"
colourGene_add[ROS3_down30,"distanceToGene"] <- "down30kb"
# colourGene_add[ROS3,"distanceToGene"] <- "inGene"

# Have to re-enter hits within the genes, because ROS3 and ROS2 100kb overlaps with ROS1 and vice versa. All others now called "ROSEA" as its tricky to discriminate amongst ROS1, ROS2 and ROS3 linked when they are all very close to each other.
colourGene_add[ROS1,"ColourGene"] <- "ROS1"
colourGene_add[ROS1,"distanceToGene"] <- "inGene"
colourGene_add[ROS2,"ColourGene"] <- "ROS2"
colourGene_add[ROS2,"distanceToGene"] <- "inGene"

# adding linked windows linked to all windows between ROS3 and ELUTA
# colourGene_add[35545:35557,"ColourGene"]<- "ROS_EL_barrier"

# ELUTA
# ELUTA LG6: [53060761:53062505,] 

ELUTA_gene <- c(53060761,53062505)
# within gene - 10 loci with clines within coding region
ELUTA <- which(allChr_Clines[,"LG"]==6 & allChr_Clines[,"position"] > ELUTA_gene[1] & allChr_Clines[,"position"] < ELUTA_gene[2])
# +/- 30kb & 30-100kb
ELUTA_up100 <- which(allChr_Clines[,"LG"]==6 & abs(allChr_Clines[,"position"] - ELUTA_gene[1]) < 100000 & abs(allChr_Clines[,"position"] - ELUTA_gene[1]) > 30000 & allChr_Clines[,"position"] < ELUTA_gene[1])
ELUTA_up30 <- which(allChr_Clines[,"LG"]==6 & abs(allChr_Clines[,"position"] - ELUTA_gene[1]) < 30000 & allChr_Clines[,"position"] < ELUTA_gene[1])
ELUTA_down100 <- which(allChr_Clines[,"LG"]==6 & abs(allChr_Clines[,"position"] - ELUTA_gene[2]) < 100000 &  abs(allChr_Clines[,"position"] - ELUTA_gene[2]) > 30000 & allChr_Clines[,"position"] > ELUTA_gene[2])
ELUTA_down30 <- which(allChr_Clines[,"LG"]==6 & abs(allChr_Clines[,"position"] - ELUTA_gene[2]) < 30000 & allChr_Clines[,"position"] > ELUTA_gene[2])

colourGene_add[ELUTA_up100,"ColourGene"] <- "ELUTA"
colourGene_add[ELUTA_up30,"ColourGene"] <- "ELUTA"
#colourGene_add[ELUTA_down100,"ColourGene"] <- "ELUTA"
colourGene_add[ELUTA_down30,"ColourGene"] <- "ELUTA"
colourGene_add[ELUTA,"ColourGene"] <- "ELUTA"

colourGene_add[ELUTA_up100,"distanceToGene"] <- "up30_100kb"
#colourGene_add[ELUTA_down100,"distanceToGene"] <- "down30_100kb"
colourGene_add[ELUTA_up30,"distanceToGene"] <- "up30kb"
colourGene_add[ELUTA_down30,"distanceToGene"] <- "down30kb"
colourGene_add[ELUTA,"distanceToGene"] <- "inGene"

allChr_Clines <- cbind(allChr_Clines,colourGene_add)
#Chr4_clinesNEW <- cbind(Chr4_clinesNEW,colourGene_add_SULFspecial)
#allChr_Clines[allChr_Clines[,"position"]==52889078,]

```

# FastClines setup and cline esimates

## Deme locations along transect

We used the whole genome poolSeq data and the transect positioning through the hybrid zone as described in Tavares et al 2018 [ref]. The geographic positions of each pool were located along this transect and collapsed to one dimensional geographic distance (in kilometers) along the transect (Fig S.\ref{fig:HZtransect}). The total geographic distance along the transect was set by the larger SNP KASP genotype data () with demes from 0 km to 25.176km. The average Easting and Northing of individuals in each pool was used to set the poolSeq demes along this transect [pool 1: 0.382km, pool2: 11.227, pool3: 11.404, pool4: 13.767, pool5: 14.424, pool6: 20.670km].

## Deme Spans

In the fastClines method, we need to define the geographic span of each deme. If the demes are equally spaced across a transect, one approach is to use half the distance either side of the geographic centre of each deme as the span of each deme. However, the patchiness of individual samples in natural populations, including the snapdragon system, often preclude setting equal distances between pools. Therefore, to account for possible influence of deme span selection on cline properties, we examined three approaches: 

(1)	Deme Spacing 1 (DS1)- midpoint edges: where the outer edges (pool 1 and 6) were defined by the dimensions of the KASP sample demes (0 to 25.176km) and all internal edges set as the midpoint distances between each pool along the transect [deme edges: 0, 5.423, 11.316, 12.586, 14.095,  17.548, 25.172km] from which the geographic span of each deme was calculated [pool 1:  5.805km, pool2: 5.893, pool3: 1.270, pool4: 1.500, pool5: 3.453, pool6: 7.251km] (Fig Sxa).

(2)	Deme Spacing 2 (DS2) – equal spans: outer edges as above, but internal edges chosen to produce equally sized deme spans either side of the same edge between pool 3 and 4 (across centre of phenotype clines) [pool 1:  4.195, pool2: 4.195, pool3: 4.195, pool4: 4.195, pool5: 4.195, pool6: 4.195km] (Fig Sxb)*3

(3)	Deme Spacing 3 (DS3)– extended outer pools: outer edges as in (1) and (2), but to account for the extended gaps in the available plants between pool 1 and pool 6 to internal pools, we extend the span of these outer pools inwards and force pool 2 to have the same span as pool 3 as well as pool 4 and 5 sharing the same span [pool 1:  10.046km, pool2: 1.270, pool3: 1.270, pool4: 1.500, pool5: 1.500, pool6: 9.586km] (Fig Sxc).

```{r new columns for depth in pools for clines}
# total clines
nrow(allChr_Clines)
depth15min <- allChr_Clines[,15:20]>=15
depth20min <- allChr_Clines[,15:20]>=20

allChr_Clines_filter <- cbind(allChr_Clines,apply(depth15min,1,sum),apply(depth20min,1,sum))
colnames(allChr_Clines_filter)[ncol(allChr_Clines_filter)-1] <- "poolsPass_Depth15"
colnames(allChr_Clines_filter)[ncol(allChr_Clines_filter)] <- "poolsPass_Depth20"

# Fast clines run with very low filtering threshold to begin analysis (min depth = 10)
# shows that many loci failed filtering
depth15table <- table(allChr_Clines_filter[,"poolsPass_Depth15"]) # sum(depth15table[6:7])
depth20table <- table(allChr_Clines_filter[,"poolsPass_Depth20"]) # sum(depth20table[6:7])

```

```{r basic stats on fixed differences and deltaP from fastClines output across genome}
# colnames(allChr_Clines_filter)

# 3847 loci with delta P> 0.9 and min depth 15 in at least 5 of 6 pools
sum(allChr_Clines_filter[,"poolsPass_Depth15"]>4 & allChr_Clines_filter[,"AFD_15_20"]>=0.9)

sum(allChr_Clines_filter[,"poolsPass_Depth15"]>4 & allChr_Clines_filter[,"AFD_15_20"]>=0.9)/sum(allChr_Clines_filter[,"poolsPass_Depth15"]>4)

hist(allChr_Clines_filter[allChr_Clines_filter[,"poolsPass_Depth15"]>4,"AFD_15_20"],xlab="delta p",ylab="frequency",main="")

# 2317 fixed differences
colnames(allChr_Clines_filter)
which(colnames(allChr_Clines_filter)=="AFD_15_20")

allChr_Clines_filter_deltaP_10 <- allChr_Clines_filter[allChr_Clines_filter[,"poolsPass_Depth15"]>4 & allChr_Clines_filter[,"AFD_15_20"]==1,]
nrow(allChr_Clines_filter_deltaP_10)
  # which chromosomes
table(allChr_Clines_filter_deltaP_10[,"LG"])
rbind(table(allChr_Clines_filter_deltaP_10[,"LG"]),table(allChr_Clines_filter_deltaP_10[,"LG"])/sum(table(allChr_Clines_filter_deltaP_10[,"LG"])))


# 1421 deltaP > 0.9 but <1.0
allChr_Clines_filter_deltaP_09_10 <- allChr_Clines_filter %>% filter(poolsPass_Depth15>4 & AFD_15_20>0.9 & AFD_15_20<1)
nrow(allChr_Clines_filter_deltaP_09_10)

allChr_Clines_filter_deltaP_grt09 <- allChr_Clines_filter %>% filter(poolsPass_Depth15>4 & AFD_15_20>0.9 )
nrow(allChr_Clines_filter_deltaP_grt09)

# 7070 fixed differences 0.8-1.0
allChr_Clines_filter_deltaP_08_09 <- allChr_Clines_filter %>% filter(poolsPass_Depth15>4 & AFD_15_20>0.8 & AFD_15_20<0.9)
nrow(allChr_Clines_filter_deltaP_08_09)


# 7,790 loci with delta P> 0.8 and min depth 15 in at least all 6 pools
sum(allChr_Clines_filter[,"poolsPass_Depth15"]>5)
hist(allChr_Clines_filter[allChr_Clines_filter[,"poolsPass_Depth15"]>5,"AFD_15_20"],xlab="delta p",ylab="frequency",main="")

(allChr_Clines_filter[allChr_Clines_filter[,"poolsPass_Depth15"]>4,"AFD_15_20"])

# loci fixed in one species and polymorphic in the other
allChr_Clines_filter_deltaP_fixedY_polyM <- allChr_Clines_filter %>% filter(poolsPass_Depth15>4 & AFD_15_20>=0.8 & AFD_15_20<1 & p_15==0)
#nrow(allChr_Clines_filter_deltaP_fixedY_polyM)

allChr_Clines_filter_deltaP_fixedM_polyY <- allChr_Clines_filter %>% filter(poolsPass_Depth15>4 & AFD_15_20>=0.8 & AFD_15_20<1 & p_20==1)
#nrow(allChr_Clines_filter_deltaP_fixedM_polyY)
#allChr_Clines_filter_deltaP_fixedM_polyY[,c(1:5,9,14)]
```

# Re-running FastCline estimates with Pmin adjustment and deme span options 1-3

```{r Supporting Info re-doing FastCline estimates with three deme span options}

# original
clineCentre <- function(p,deltaPthresh,demeOrder,demeSpans) {
  # 6-(sum(c(0,0.25,0.45,0.55,0.75,1))+0.5)
  # deltaPthresh <- 0.8
  # p <- c(0,0.25,0.45,0.55,0.75,1)
  # demeOrder <- seq(1,6); demeSpans <- rep(1000,6)
  # demeSpans <- demeSpacing_Opt1
  p <- as.vector(unlist(p))
  # reordering demes based on position along transect (if required)
  p <- p[demeOrder]
  p0 <- p[1]
  p1 <- p[length(p)]
  deltaP <- abs(p0-p1)
  d_i = demeSpans 
  # reorder the spans - not required (spans fixed, just flipping p between pool 2 & 3)
  # d_i <- d_i[demeOrder]
  if (deltaP < deltaPthresh) {
    return (-9)
  }
  if (deltaP >= deltaPthresh) {
    numerCentre = p-rep(p0,length(p))
    denomCentre = rep(p1-p0,length(p))
    #sum(d_i) - sum((numerCentre/denomCentre)*rep(d_i))
    centre_output = sum(d_i) - sum((numerCentre/denomCentre) * d_i)
    return(centre_output)
  }
}

clineWidth <- function(p,deltaPthresh,demeOrder,demeSpans) {
  # deltaPthresh <- 0.8
  # p <- this_p
  # demeSpans <- demeSpacing_Opt1
  p <- as.vector(unlist(p))
  # reordering demes based on position along transect (if required)
  p <- p[demeOrder]
  p0 <- p[1]
  p1 <- p[length(p)]
  deltaP <- abs(p0-p1)
  d_i = demeSpans 
  # reorder the spans (not required here)
  # d_i <- d_i[demeOrder]
  if (deltaP < deltaPthresh) {
    return (-9)
  }
  if (deltaP >= deltaPthresh) {
    numerWidth = (p-rep(p0,length(p)))*(rep(p1,length(p))-p)
    denomWidth = rep((p1-p0)^2,length(p))
    width_output = 4*(sum((numerWidth/denomWidth)*rep(d_i)))
    return(width_output)
  }
}

# pmin adjustment Feb 2025
clineCentre <- function(p,deltaPthresh,demeOrder,demeSpans) {
  # 6-(sum(c(0,0.25,0.45,0.55,0.75,1))+0.5)
  # deltaPthresh <- 0.8
  # p <- c(0,0.25,0.45,0.55,0.75,1)
  # demeOrder <- seq(1,6); demeSpans <- rep(1000,6)
  # demeSpans <- demeSpacing_Opt1
  p <- as.vector(unlist(p))
  # reordering demes based on position along transect (if required)
  p <- p[demeOrder]
  p0 <- p[1]
  p1 <- p[length(p)]
  # pmin adjustment 2025
  p0 <- min(p[1:3])
  p1 <- max(p[(length(p)-2):length(p)])
  deltaP <- abs(p0-p1)
  d_i = demeSpans 
  # reorder the spans - not required (spans fixed, just flipping p between pool 2 & 3)
  # d_i <- d_i[demeOrder]
  if (deltaP < deltaPthresh) {
    return (-9)
  }
  if (deltaP >= deltaPthresh) {
    numerCentre = p-rep(p0,length(p))
    denomCentre = rep(p1-p0,length(p))
    #sum(d_i) - sum((numerCentre/denomCentre)*rep(d_i))
    centre_output = sum(d_i) - sum((numerCentre/denomCentre) * d_i)
    return(centre_output)
  }
}

clineWidth <- function(p,deltaPthresh,demeOrder,demeSpans) {
  # deltaPthresh <- 0.8
  # p <- this_p
  # demeSpans <- demeSpacing_Opt1
  p <- as.vector(unlist(p))
  # reordering demes based on position along transect (if required)
  p <- p[demeOrder]
  p0 <- p[1]
  p1 <- p[length(p)]
  # pmin adjustment 2025
  p0 <- min(p[1:3])
  p1 <- max(p[(length(p)-2):length(p)])
  deltaP <- abs(p0-p1)
  d_i = demeSpans 
  # reorder the spans (not required here)
  # d_i <- d_i[demeOrder]
  if (deltaP < deltaPthresh) {
    return (-9)
  }
  if (deltaP >= deltaPthresh) {
    numerWidth = (p-rep(p0,length(p)))*(rep(p1,length(p))-p)
    denomWidth = rep((p1-p0)^2,length(p))
    width_output = 4*(sum((numerWidth/denomWidth)*rep(d_i)))
    return(width_output)
  }
}

# testing functions
demeSpacingTest <- rep(1000,6)
clineCentre(p=c(0,0.25,0.45,0.55,0.75,1),1,seq(1,6),rep(1000,6))
clineWidth(p=c(0,0.25,0.45,0.55,0.75,1),1,seq(1,6),rep(1000,6))

# testin 2025 pmin adjustment
clineWidth(p=c(0.14,0,0.45,0.9,0.95,1),1,seq(1,6),rep(1000,6))

# deme spacing original
demeSpacing_Original <- c(6000,5000,1500,1400,3500,6000)
# deme spacing 1
demeSpacing_Opt1 <- c(5805,5893,1270,1500,3453,7251)
# deme spacing 2
demeSpacing_Opt2 <- c(4195,4195,4195,4195,4195,4195)
# deme spacing 3
demeSpacing_Opt3 <- c(10046,1270,1270,1500,1500,9586)

# check ROS markers first
# anyDuplicated(names(allChr_Clines))
allChr_Clines_ROS1 <- allChr_Clines %>%
                      filter(LG==6) %>%
                      filter(position > 52889323 & position < 52889737)
clineEstCheck <- matrix(-9,nrow(allChr_Clines_ROS1),4*2)
colnames(clineEstCheck) <- c("centre_Original","width_Original","centre_Opt1","width_Opt1","centre_Opt2","width_Opt2","centre_Opt3","width_Opt3")


for (thisRow in 1:nrow(allChr_Clines_ROS1)) {
  # thisRow <- 1
  this_p <- allChr_Clines_ROS1[thisRow,9:14]
  clineEstCheck[thisRow,"centre_Original"] <- clineCentre(this_p,0.8,c(1,3,2,4,5,6),demeSpacing_Original)
  clineEstCheck[thisRow,"width_Original"] <- clineWidth(this_p,0.8,c(1,3,2,4,5,6),demeSpacing_Original)
  clineEstCheck[thisRow,"centre_Opt1"] <- clineCentre(this_p,0.8,c(1,3,2,4,5,6),demeSpacing_Opt1)
  clineEstCheck[thisRow,"width_Opt1"] <- clineWidth(this_p,0.8,c(1,3,2,4,5,6),demeSpacing_Opt1)
  clineEstCheck[thisRow,"centre_Opt2"] <- clineCentre(this_p,0.8,c(1,3,2,4,5,6),demeSpacing_Opt2)
  clineEstCheck[thisRow,"width_Opt2"] <- clineWidth(this_p,0.8,c(1,3,2,4,5,6),demeSpacing_Opt2)
  clineEstCheck[thisRow,"centre_Opt3"] <- clineCentre(this_p,0.8,c(1,3,2,4,5,6),demeSpacing_Opt3)
  clineEstCheck[thisRow,"width_Opt3"] <- clineWidth(this_p,0.8,c(1,3,2,4,5,6),demeSpacing_Opt3)
}

ROScheck <- cbind(allChr_Clines_ROS1[,"centre"],allChr_Clines_ROS1[,"width"],clineEstCheck)
# Fast cline estimates (first two columns) are identical to centre_Opt1 and width_Opt1
# all the data

clineEstCheck <- matrix(-9,nrow(allChr_Clines),4*2)
colnames(clineEstCheck) <- c("centre_Original","width_Original","centre_Opt1","width_Opt1","centre_Opt2","width_Opt2","centre_Opt3","width_Opt3")


for (thisRow in 1:nrow(allChr_Clines)) {
  # thisRow <- 1
  this_p <- allChr_Clines[thisRow,9:14]
  clineEstCheck[thisRow,"centre_Original"] <- clineCentre(this_p,0.8,c(1,3,2,4,5,6),demeSpacing_Original)
  clineEstCheck[thisRow,"width_Original"] <- clineWidth(this_p,0.8,c(1,3,2,4,5,6),demeSpacing_Original)
  clineEstCheck[thisRow,"centre_Opt1"] <- clineCentre(this_p,0.8,c(1,3,2,4,5,6),demeSpacing_Opt1)
  clineEstCheck[thisRow,"width_Opt1"] <- clineWidth(this_p,0.8,c(1,3,2,4,5,6),demeSpacing_Opt1)
  clineEstCheck[thisRow,"centre_Opt2"] <- clineCentre(this_p,0.8,c(1,3,2,4,5,6),demeSpacing_Opt2)
  clineEstCheck[thisRow,"width_Opt2"] <- clineWidth(this_p,0.8,c(1,3,2,4,5,6),demeSpacing_Opt2)
  clineEstCheck[thisRow,"centre_Opt3"] <- clineCentre(this_p,0.8,c(1,3,2,4,5,6),demeSpacing_Opt3)
  clineEstCheck[thisRow,"width_Opt3"] <- clineWidth(this_p,0.8,c(1,3,2,4,5,6),demeSpacing_Opt3)
}

allChr_Clines_chk <- cbind(allChr_Clines_filter,clineEstCheck)

# still some negatives
range(allChr_Clines_chk[,"width_Original"])

```

```{r filtering for depth and mean p in magenta outer and yellow inner pools}

# keep only depth 15 at 5 pools
nrow(allChr_Clines_chk)
allChr_Clines_Dpth15 <- allChr_Clines_chk %>% filter(poolsPass_Depth15>4)

allChr_Clines_Dpth15_delta9 <- allChr_Clines_Dpth15 %>% filter(AFD_15_20>=0.9)
nrow(allChr_Clines_Dpth15_delta9)
table(allChr_Clines_Dpth15_delta9[,"LG"])
linkedAreas <- c("down30kb","down30_100kb","down30_400kb","up30_100kb","up30kb")

# Second filter for unusual non-clinal SNPs
# Using average p of two outer magenta pools against two inner yellow pools as a filter
# Average p for 2 Magenta pools must be higher than 2 inner yellow pools
colnames(allChr_Clines_Dpth15)
YPouterMinusMPouter<- round(apply((cbind(allChr_Clines_Dpth15[,"p_19"],allChr_Clines_Dpth15[,"p_20"])),1,mean) - apply((cbind(allChr_Clines_Dpth15[,"p_16"],allChr_Clines_Dpth15[,"p_17"])),1,mean),3)

# 8.7% of clines dont pass the filter - keep them to plot
length(which(YPouterMinusMPouter<0))/nrow(allChr_Clines_Dpth15)
allChr_Clines_Dpth15_MeanP_fail <- allChr_Clines_Dpth15[which(YPouterMinusMPouter<0),]
allChr_Clines_Dpth15_MeanP_pass <- allChr_Clines_Dpth15[which(YPouterMinusMPouter>0),]
nrow(allChr_Clines_Dpth15_MeanP_fail) # n = 954 loci
nrow(allChr_Clines_Dpth15_MeanP_pass) # n = 8939 loci

# where are the failed ones located compared to pass
table(allChr_Clines_Dpth15_MeanP_fail[,"LG"])
table(allChr_Clines_Dpth15_MeanP_pass[,"LG"])

table(allChr_Clines_Dpth15_MeanP_fail[,"LG"])/table(allChr_Clines_Dpth15[,"LG"])

```

```{r separating out each Chr for numbers at different deltaP }


################### 
# Chr 1 & CREMOSA #
###################
allChr_Clines_Chr1 <- allChr_Clines_Dpth15 %>% filter(LG==1)
allChr_Clines_Chr1_deltaP9 <- allChr_Clines_Dpth15 %>% filter(LG==1 & AFD_15_20>=0.9)
allChr_Clines_Chr1_deltaP1 <- allChr_Clines_Dpth15 %>% filter(LG==1 & AFD_15_20>=1.0)

# Delta p 0.8 has 387 clines, 0 in CRE, 79 linked CRE, 308 elsewhere
Chr1_delta8_Table <- table(allChr_Clines_Chr1[,"ColourGene"],allChr_Clines_Chr1[,"distanceToGene"])
Chr1_delta8_Table
nrow(allChr_Clines_Chr1)
# Delta p 0.9 has 90 clines, 0 in CRE,  41 linked CRE, 49 elsewhere
Chr1_delta9_Table <- table(allChr_Clines_Chr1_deltaP9[,"ColourGene"],allChr_Clines_Chr1_deltaP9[,"distanceToGene"])
Chr1_delta9_Table
nrow(allChr_Clines_Chr1_deltaP9)
# Diagnostic has 45 clines, 0 in FLA,  26 linked CRE, 19 elsewhere
Chr1_delta1_Table <- table(allChr_Clines_Chr1_deltaP1[,"ColourGene"],allChr_Clines_Chr1_deltaP1[,"distanceToGene"])
Chr1_delta1_Table
nrow(allChr_Clines_Chr1_deltaP1)

# 
allChr_Clines_Chr1_elsewhere <- allChr_Clines_Dpth15 %>% filter(LG==1 & ColourGene==-9)
allChr_Clines_Chr1_elsewhere_deltaP_09 <- allChr_Clines_Dpth15 %>% filter(LG==1 & ColourGene==-9 & AFD_15_20>=0.9)

allChr_Clines_CREMOSA <- allChr_Clines_Dpth15 %>% filter(ColourGene=="CREMOSA" & distanceToGene =="inGene")
allChr_Clines_CREMOSAlinked <- allChr_Clines_Dpth15 %>% filter(ColourGene=="CREMOSA") %>% filter(distanceToGene %in% linkedAreas) 

allChr_Clines_Chr1clines <- rbind(allChr_Clines_Chr1_elsewhere,allChr_Clines_CREMOSAlinked)

allChr_Clines_Chr1clines_deltaP_09 <- rbind(allChr_Clines_Chr1_elsewhere_deltaP_09,allChr_Clines_CREMOSAlinked)


##################
# Chr 2 & FLAVIA #
##################

allChr_Clines_Chr2 <- allChr_Clines_Dpth15 %>% filter(LG==2)
allChr_Clines_Chr2_deltaP9 <- allChr_Clines_Dpth15 %>% filter(LG==2 & AFD_15_20>=0.9)
allChr_Clines_Chr2_deltaP1 <- allChr_Clines_Dpth15 %>% filter(LG==2 & AFD_15_20>=1.0)

# Delta p 0.8 has 7622 clines, 5 in FLA, 0 in AUR, 3639 linked FLA, 29 linked AUR, 3949 elsewhere
Chr2_delta8_Table <- table(allChr_Clines_Chr2[,"ColourGene"],allChr_Clines_Chr2[,"distanceToGene"])
Chr2_delta8_Table <- Chr2_delta8_Table[,c(1,2,3,4,7,5,6)]
Chr2_delta8_Table
nrow(allChr_Clines_Chr2)
# Delta p 0.9 has 3159 clines, 4 in FLA, 0 in AUR, 2584 linked FLA, 6 linked AUR, 565 elsewhere
Chr2_delta9_Table <- table(allChr_Clines_Chr2_deltaP9[,"ColourGene"],allChr_Clines_Chr2_deltaP9[,"distanceToGene"])
Chr2_delta9_Table <- Chr2_delta9_Table[,c(1,2,3,4,6,5)]
Chr2_delta9_Table
nrow(allChr_Clines_Chr2_deltaP9)
# Diagnostic has 118 clines, 0 in FLA, 0 in AUR, 1834 linked FLA, 0 linked AUR, 161 elsewhere
Chr2_delta1_Table <- table(allChr_Clines_Chr2_deltaP1[,"ColourGene"],allChr_Clines_Chr2_deltaP1[,"distanceToGene"])
Chr2_delta1_Table <- Chr2_delta1_Table[,c(1,2,3,5,4)]
nrow(allChr_Clines_Chr2_deltaP1)

allChr_Clines_Chr2_elsewhere <- allChr_Clines_Dpth15 %>% filter(LG==2 & ColourGene==-9)
allChr_Clines_Chr2_elsewhere_deltaP_09 <- allChr_Clines_Dpth15 %>% filter(LG==2 & ColourGene==-9 & AFD_15_20>=0.9)

nrow(allChr_Clines_Chr2_elsewhere)

allChr_Clines_FLAVIA <- allChr_Clines_Dpth15 %>% filter(ColourGene=="FLAVIA" & distanceToGene =="inGene")
allChr_Clines_FLAVIAlinked <- allChr_Clines_Dpth15 %>% filter(ColourGene=="FLAVIA") %>% filter(distanceToGene %in% linkedAreas) 
allChr_Clines_AURINAlinked <- allChr_Clines_Dpth15 %>% filter(ColourGene=="AURINA") %>% filter(distanceToGene %in% linkedAreas) 
allChr_Clines_Chr2clines <- rbind(allChr_Clines_Chr2_elsewhere,allChr_Clines_FLAVIAlinked,allChr_Clines_AURINAlinked,allChr_Clines_FLAVIA)

allChr_Clines_Chr2clines_deltaP_09 <- rbind(allChr_Clines_Chr2_elsewhere_deltaP_09,allChr_Clines_FLAVIAlinked,allChr_Clines_AURINAlinked,allChr_Clines_FLAVIA)


# FLAVIA - 0 diagnostic, 4 with delta p > 0.9, 1 with delta p > 0.8
table(allChr_Clines_FLAVIA[,"ColourGene"],allChr_Clines_FLAVIA[,"AFD_15_20"])
# FLAVIA linked - 1971 diagnostic, 712 delta p > 0.8
table(allChr_Clines_FLAVIAlinked[,"ColourGene"],allChr_Clines_FLAVIAlinked[,"AFD_15_20"])
nrow(allChr_Clines_FLAVIAlinked)
# elsewhere - 270 diagnostic, 4326 delta p > 0.8
table(allChr_Clines_Chr2_elsewhere[,"ColourGene"],allChr_Clines_Chr2_elsewhere[,"AFD_15_20"])
nrow(allChr_Clines_Chr2_elsewhere)

#########
# Chr 3 #
#########

allChr_Clines_Chr3 <- allChr_Clines_Dpth15 %>% filter(LG==3)
allChr_Clines_Chr3_deltaP9 <- allChr_Clines_Dpth15 %>% filter(LG==3 & AFD_15_20>=0.9)
allChr_Clines_Chr3_deltaP1 <- allChr_Clines_Dpth15 %>% filter(LG==3 & AFD_15_20>=1.0)

# Delta p 0.8 has 256 clines
Chr3_delta8_Table <- table(allChr_Clines_Chr3[,"ColourGene"],allChr_Clines_Chr3[,"distanceToGene"])
Chr3_delta8_Table
nrow(allChr_Clines_Chr3)
# Delta p 0.9 has 41 clines
Chr3_delta9_Table <- table(allChr_Clines_Chr3_deltaP9[,"ColourGene"],allChr_Clines_Chr3_deltaP9[,"distanceToGene"])
Chr3_delta9_Table
nrow(allChr_Clines_Chr3_deltaP9)
# Diagnostic has 16 clines, 
Chr3_delta1_Table <- table(allChr_Clines_Chr3_deltaP1[,"ColourGene"],allChr_Clines_Chr3_deltaP1[,"distanceToGene"])
nrow(allChr_Clines_Chr3_deltaP1)

allChr_Clines_Chr3_elsewhere <- allChr_Clines_Dpth15 %>% filter(LG==3 & ColourGene==-9)
nrow(allChr_Clines_Chr3_elsewhere)

allChr_Clines_Chr3clines <- allChr_Clines_Chr3_elsewhere

allChr_Clines_Chr3clines_deltaP_09 <- allChr_Clines_Dpth15 %>% filter(LG==3 & ColourGene==-9 & AFD_15_20>=0.9)

#################
# Chr 4 & SULF  #
#################

allChr_Clines_Chr4 <- allChr_Clines_Dpth15 %>% filter(LG==4)
allChr_Clines_Chr4_deltaP9 <- allChr_Clines_Dpth15 %>% filter(LG==4 & AFD_15_20>=0.9)
allChr_Clines_Chr4_deltaP1 <- allChr_Clines_Dpth15 %>% filter(LG==4 & AFD_15_20>=1.0)

# Delta p 0.8 has 960 clines, 0 in SULF, 16 linked to SULF, 944 elsewhere
Chr4_delta8_Table <- table(allChr_Clines_Chr4[,"ColourGene"],allChr_Clines_Chr4[,"distanceToGene"])
Chr4_delta8_Table
nrow(allChr_Clines_Chr4)
# Delta p 0.8 has 144 clines, 0 in SULF, 4 linked to SULF, 140 elsewhere
Chr4_delta9_Table <- table(allChr_Clines_Chr4_deltaP9[,"ColourGene"],allChr_Clines_Chr4_deltaP9[,"distanceToGene"])
Chr4_delta9_Table
nrow(allChr_Clines_Chr4_deltaP9)
# Diagnostic has 71 clines, 0 in SULF, 2 linked to SULF, 69 elsewhere
Chr4_delta1_Table <- table(allChr_Clines_Chr4_deltaP1[,"ColourGene"],allChr_Clines_Chr4_deltaP1[,"distanceToGene"])
Chr4_delta1_Table
nrow(allChr_Clines_Chr4_deltaP1)

# separate out clines for in genes, linked and elsewhere
allChr_Clines_Chr4_elsewhere <- allChr_Clines_Dpth15 %>% filter(LG==4 & ColourGene==-9)
allChr_Clines_Chr4_elsewhere_deltaP_09 <- allChr_Clines_Dpth15 %>% filter(LG==4 & ColourGene==-9 & AFD_15_20>=0.9)

allChr_Clines_SULF <- allChr_Clines_Dpth15 %>% filter(ColourGene=="SULF" & distanceToGene =="inGene")
allChr_Clines_SULFlinked <- allChr_Clines_Dpth15 %>% filter(ColourGene=="SULF") %>% filter(distanceToGene %in% linkedAreas) 

allChr_Clines_Chr4clines <- rbind(allChr_Clines_Chr4_elsewhere,allChr_Clines_SULFlinked)
allChr_Clines_Chr4clines_deltaP_09 <- rbind(allChr_Clines_Chr4_elsewhere_deltaP_09,allChr_Clines_SULFlinked)

#################
# Chr 5 & RUBIA #
#################

allChr_Clines_Chr5 <- allChr_Clines_Dpth15 %>% filter(LG==5)
allChr_Clines_Chr5_deltaP9 <- allChr_Clines_Dpth15 %>% filter(LG==5 & AFD_15_20>=0.9)
allChr_Clines_Chr5_deltaP1 <- allChr_Clines_Dpth15 %>% filter(LG==5 & AFD_15_20>=1.0)

# Delta p 0.8 has 926 clines, 7 in RUBIA, 198 linked RUBIA, 721 elsewhere
Chr5_delta8_Table <- table(allChr_Clines_Chr5[,"ColourGene"],allChr_Clines_Chr5[,"distanceToGene"])
Chr5_delta8_Table <- Chr5_delta8_Table[,c(1,2,3,5,4)]
Chr5_delta8_Table
nrow(allChr_Clines_Chr5)
# Delta p 0.9 has 161 clines, 4 in RUBIA, 83 linked RUBIA, 74 elsewhere
Chr5_delta9_Table <- table(allChr_Clines_Chr5_deltaP9[,"ColourGene"],allChr_Clines_Chr5_deltaP9[,"distanceToGene"])
Chr5_delta9_Table <- Chr5_delta9_Table[,c(1,2,3,5,4)]
nrow(allChr_Clines_Chr5_deltaP9)
# Diagnostic has 60 clines, 1 in RUBIA, 24 linked RUBIA, 35 elsewhere
Chr5_delta1_Table <- table(allChr_Clines_Chr5_deltaP1[,"ColourGene"],allChr_Clines_Chr5_deltaP1[,"distanceToGene"])
Chr5_delta1_Table <- Chr5_delta1_Table[,c(1,2,3,5,4)]
nrow(allChr_Clines_Chr5_deltaP1)

allChr_Clines_Chr5_elsewhere <- allChr_Clines_Dpth15 %>% filter(LG==5 & ColourGene==-9)
nrow(allChr_Clines_Chr5_elsewhere)

allChr_Clines_Chr5_elsewhere_deltaP_09 <- allChr_Clines_Dpth15 %>% filter(LG==5 & ColourGene==-9 & AFD_15_20>=0.9)

allChr_Clines_RUBIA <- allChr_Clines_Dpth15 %>% filter(ColourGene=="RUBIA" & distanceToGene =="inGene")
allChr_Clines_RUBIAlinked <- allChr_Clines_Dpth15 %>% filter(ColourGene=="RUBIA") %>% filter(distanceToGene %in% linkedAreas)
allChr_Clines_Chr5clines <- rbind(allChr_Clines_Chr5_elsewhere,allChr_Clines_RUBIAlinked,allChr_Clines_RUBIA)
allChr_Clines_Chr5clines_deltaP_09 <- rbind(allChr_Clines_Chr5_elsewhere_deltaP_09,allChr_Clines_RUBIAlinked,allChr_Clines_RUBIA)

####################
# Chr 6 and ROS EL #
####################

allChr_Clines_Chr6 <- allChr_Clines_Dpth15 %>% filter(LG==6)
allChr_Clines_Chr6_deltaP9 <- allChr_Clines_Dpth15 %>% filter(LG==6 & AFD_15_20>=0.9)
allChr_Clines_Chr6_deltaP1 <- allChr_Clines_Dpth15 %>% filter(LG==6 & AFD_15_20>=1.0)

# Delta p 0.8 has 461 clines, 11 in ROS1 linked, 7 in ROS2 linked, 11 in ELUTA gene, 88 linked to ROS1/ROS2, 217 linked EL 127 elsewhere
Chr6_delta8_Table <- table(allChr_Clines_Chr6[,"ColourGene"],allChr_Clines_Chr6[,"distanceToGene"])
Chr6_delta8_Table <- Chr6_delta8_Table[,c(1,2,3,5,4)]
Chr6_delta8_Table
nrow(allChr_Clines_Chr6)
# Delta p 0.8 has 461 clines, 11 in ROS1 , 7 in ROS2 linked, 10 in ELUTA gene, 56 linked to ROS1/ROS2, 96 linked EL, 17 elsewhere
Chr6_delta9_Table <- table(allChr_Clines_Chr6_deltaP9[,"ColourGene"],allChr_Clines_Chr6_deltaP9[,"distanceToGene"])
Chr6_delta9_Table <- Chr6_delta9_Table[,c(1,2,3,5,4)]
Chr6_delta9_Table
nrow(allChr_Clines_Chr6_deltaP9)
# Diagnostic has 118 clines, 9 in ROS1, 7 in ROS2, 0 in ELUTA gene, 49 linked to ROS1/ROS2, 43 linked EL, 3 elsewhere
Chr6_delta1_Table <- table(allChr_Clines_Chr6_deltaP1[,"ColourGene"],allChr_Clines_Chr6_deltaP1[,"distanceToGene"])
Chr6_delta1_Table <- Chr6_delta1_Table[,c(1,2,3,5,4)]
nrow(allChr_Clines_Chr6_deltaP1)

# separate out clines for in genes, linked and elsewhere
allChr_Clines_Chr6_elsewhere <- allChr_Clines_Dpth15 %>% filter(LG==6 & ColourGene==-9)
allChr_Clines_Chr6_elsewhere_deltaP_09 <- allChr_Clines_Dpth15 %>% filter(LG==6 & ColourGene==-9 & AFD_15_20>=0.9)

allChr_Clines_ROS1 <- allChr_Clines_Dpth15 %>% filter(ColourGene=="ROS1" & distanceToGene =="inGene")
allChr_Clines_ROS2 <- allChr_Clines_Dpth15 %>% filter(ColourGene=="ROS2" & distanceToGene =="inGene")
# allChr_Clines_ROS3 <- allChr_Clines_Dpth15 %>% filter(ColourGene=="ROS3" & distanceToGene =="inGene") # none in ROS3
allChr_Clines_EL <- allChr_Clines_Dpth15 %>% filter(ColourGene=="ELUTA" & distanceToGene =="inGene")
allChr_Clines_ROSelLinked <- allChr_Clines_Dpth15 %>% filter(ColourGene=="ROSEA" | ColourGene=="ROS1" | ColourGene=="ROS2" | ColourGene=="ELUTA") %>% filter(distanceToGene %in% linkedAreas) 

allChr_Clines_Chr6clines <- rbind(allChr_Clines_Chr6_elsewhere,allChr_Clines_ROSelLinked,allChr_Clines_EL,allChr_Clines_ROS1,allChr_Clines_ROS2)

allChr_Clines_Chr6clines_deltaP_09 <- rbind(allChr_Clines_Chr6_elsewhere_deltaP_09,allChr_Clines_ROSelLinked,allChr_Clines_EL,allChr_Clines_ROS1,allChr_Clines_ROS2)


#########
# Chr 7 #
#########

allChr_Clines_Chr7 <- allChr_Clines_Dpth15 %>% filter(LG==7)
allChr_Clines_Chr7_deltaP9 <- allChr_Clines_Dpth15 %>% filter(LG==7 & AFD_15_20>=0.9)
allChr_Clines_Chr7_deltaP1 <- allChr_Clines_Dpth15 %>% filter(LG==7 & AFD_15_20>=1.0)

# Delta p 0.8 has 256 clines
Chr7_delta8_Table <- table(allChr_Clines_Chr7[,"ColourGene"],allChr_Clines_Chr7[,"distanceToGene"])
Chr7_delta8_Table
nrow(allChr_Clines_Chr7)
# Delta p 0.9 has 41 clines
Chr7_delta9_Table <- table(allChr_Clines_Chr7_deltaP9[,"ColourGene"],allChr_Clines_Chr7_deltaP9[,"distanceToGene"])
Chr7_delta9_Table
nrow(allChr_Clines_Chr7_deltaP9)
# Diagnostic has 16 clines, 
Chr7_delta1_Table <- table(allChr_Clines_Chr7_deltaP1[,"ColourGene"],allChr_Clines_Chr7_deltaP1[,"distanceToGene"])
nrow(allChr_Clines_Chr7_deltaP1)

allChr_Clines_Chr7_elsewhere <- allChr_Clines_Dpth15 %>% filter(LG==7 & ColourGene==-9)
nrow(allChr_Clines_Chr7_elsewhere)
allChr_Clines_Chr7clines <- allChr_Clines_Chr7_elsewhere

allChr_Clines_Chr7_elsewhere_deltaP_09 <- allChr_Clines_Dpth15 %>% filter(LG==7 & ColourGene==-9 & AFD_15_20>=0.9)

#########
# Chr 8 #
#########
allChr_Clines_Dpth15[,"ColourGene"]
allChr_Clines_Chr8 <- allChr_Clines_Dpth15 %>% filter(LG==8)
allChr_Clines_Chr8_deltaP9 <- allChr_Clines_Dpth15 %>% filter(LG==8 & AFD_15_20>=0.9)
allChr_Clines_Chr8_deltaP1 <- allChr_Clines_Dpth15 %>% filter(LG==8 & AFD_15_20>=1.0)

# Delta p 0.8 has 256 clines
Chr8_delta8_Table <- table(allChr_Clines_Chr8[,"ColourGene"],allChr_Clines_Chr8[,"distanceToGene"])
Chr8_delta8_Table
nrow(allChr_Clines_Chr8)
# Delta p 0.9 has 41 clines
Chr8_delta9_Table <- table(allChr_Clines_Chr8_deltaP9[,"ColourGene"],allChr_Clines_Chr8_deltaP9[,"distanceToGene"])
Chr8_delta9_Table
nrow(allChr_Clines_Chr8_deltaP9)
# Diagnostic has 16 clines, 
Chr8_delta1_Table <- table(allChr_Clines_Chr8_deltaP1[,"ColourGene"],allChr_Clines_Chr8_deltaP1[,"distanceToGene"])
nrow(allChr_Clines_Chr8_deltaP1)

allChr_Clines_Chr8_elsewhere <- allChr_Clines_Dpth15 %>% filter(LG==8 & ColourGene==-9)
nrow(allChr_Clines_Chr8_elsewhere)
allChr_Clines_Chr8_elsewhere_deltaP_09 <- allChr_Clines_Dpth15 %>% filter(LG==8 & ColourGene==-9 & AFD_15_20>=0.9)

allChr_Clines_Chr8clines <- allChr_Clines_Chr8_elsewhere

```

```{r summary table}
table(allChr_Clines[,"ColourGene"],allChr_Clines[,"distanceToGene"])

```

# Checking positions

```{r Supporting Info plotting of transect for WGSPoolSeq in relation to KASP 50m demes, echo=F, include=T,eval=T,warning=F,width = 14.5, height = 7, pointsize=18,fig.cap="\\textbf{\\label{fig:HZtransect}Transect through hybrid zone for ROS1 KASP marker with allele frequencies in 2D followed by plot of allele frequencies collapsed to 1D}."}

# folder structure file system pedigree
#folderLoc <- "/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/Pedigree_DF"
#folderStructure <- read.table(paste0(folderLoc,"/Scripts/R/","folderStructure_MQ.txt"),header=TRUE,fill=FALSE,na.strings = "?")

# Import some additional data
#GenoEcol <- read.csv("/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/Pedigree_DF/PostProcessedData/FinalGenoEcol_short_int.csv")

GenoEcol <- read.csv("/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/Pedigree_DF/PostProcessedData/Amajus_KASP_publishV2.csv")

SNPlociListUpdated <- read.csv("/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/Pedigree_DF/RawDataFiles/genetic/Amajus_shortSummarySNPsV2.csv")

# Pick out Planoles and parapatric pops using Easting and Northing
GenoEcol_Planoles <- GenoEcol %>% 
  dplyr::filter(Easting>410000 & Easting<435000 & Northing>4684000 & Northing<4692000)

#PedigreeLoci_strict <- readMyFile(folderStructure,"folder_base_genetic","PedigreeLoci_strict.csv")
#PedigreeLoci_strict[,1] <- as.vector(PedigreeLoci_strict[,1])

# KASP markers
# CREMOSA (near UDP) Chr 1 "s1187_290152" 
# Flavia Chr 2 - 43.175, "s316_93292", "s316_257789", 
# SULF Chr 4 "s91_39699", 
# RUBIA Chr 5 (near FLS) "s261_720757"
# ROS EL Chr 6 "ros_assembly_543443","ros_assembly_715015"

# make matrix for poolSeq information

poolLoc <- as.data.frame(matrix(0,7,8))
colnames(poolLoc) <- c("pool","Easting","Northing","XPos_deme","YPos_deme","X_adj","Y_adj","DistAlongTransect")
poolLoc[1:7,1] <- c("DemeKASP_Start","YP4","YP2","YP1","MP2","MP4","MP11")
# new positions 2023
poolLoc[2,2:3] <- c(411635.27,4690296.66) # YP4
poolLoc[3,2:3] <- c(422120.82,4686387.46) # YP2
poolLoc[4,2:3] <- c(421968.27,4686511.65) # YP1
poolLoc[5,2:3] <- c(424440.68,4686082.79) # MP2
poolLoc[6,2:3] <- c(425130.38,4685954.23) # MP4
poolLoc[7,2:3] <- c(431641.97,4686865.38) # MP11
poolLoc[,2] <- as.numeric(as.vector(poolLoc[,2]))
poolLoc[,3] <- as.numeric(as.vector(poolLoc[,3]))
poolLoc[,4] <- as.numeric(as.vector(poolLoc[,4]))
poolLoc[,5] <- as.numeric(as.vector(poolLoc[,5]))
poolLoc[,6] <- as.numeric(as.vector(poolLoc[,6]))
poolLoc[,7] <- as.numeric(as.vector(poolLoc[,7]))


# try plot with the same locus only (KASP ros_assembly_543443) in WGS pools then change to this below
# allChr_Clines_ROSEL[allChr_Clines_ROSEL[,"position"]==52889078,]

HZ_50mDemes <- DemeSetup2D_PoolSeq(HZdata=GenoEcol_Planoles,lociNames="ros_assembly_543443",demeSize=50,minSample=10,removePhenoMiss=F,transectGradient = -0.20, transectIntercept = 4900,  adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",poolSeqPlot=T,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=T,FigName="Chr1",plotCentreLine=T,poolSeqClines=allChr_Clines_Chr6clines,poolSeqStats=poolLoc,colourTable=colourChoice)
# str(HZ_50mDemes)

# positions of poolSeq demes along transect
poolSeqStats <- HZ_50mDemes$ros_assembly_543443$poolSeqStats
# pdf version

poolSeqStatsKeep <- poolSeqStats
write.csv(poolSeqStatsKeep,"poolSeqStatsKeep.csv", row.names = FALSE)
```

# Interesting subset
locusSet_interesting <- c("s1187_290152", # near UDP LG1
              "s316_93292", "s316_257789", # Flavia LG 2 - 43.175
              "s91_39699", # LG 4 - sulf
              "s261_720757", # LG 5 - FLS
              "ros_assembly_543443","ros_assembly_715015", # ros & el
              "RosInferred")

For SlowCline fitting for 2025 papers, look further below in script


# WGS poolSeq figures showing allele frequencies for all clinal loci
## Generates Figs S5 - Fig S11

```{r plotting lots of little cline plots for LG1 for deltaP_0_9 }
table(allChr_Clines_Dpth15$LG,allChr_Clines_Dpth15$ColourGene)

# comparing to some random clines with positive values in the middle position
nrow(allChr_Clines_Dpth15)
thisLG <- 1
clines_thisLG <- allChr_Clines_Dpth15 %>% filter(LG == thisLG)
numClines_LG1 <- nrow(clines_thisLG)
table_LG1 <- table(clines_thisLG$ColourGene,clines_thisLG$distanceToGene)

clines_thisLG <- clines_thisLG %>% filter(AFD_15_20>= 0.9)
numClines_LG1_deltaP9 <- nrow(clines_thisLG)
table_LG1_deltaP9 <- table(clines_thisLG$ColourGene,clines_thisLG$distanceToGene)

# colour code clines by:
#   centre orientated
#   left leaning
#   right leaning
#   negative widths

par(mfrow=c(10,10))
par(mar=c(1,1,1,1))
par(oma=c(1,1,1,1))
PoolSeq_summary <- poolSeqStats
# PoolSeq_summary[, "DistAlongTransect"]

for (thisRow in 1:nrow(clines_thisLG)) {
  # thisRow <- 1
  PoolSeq_summary <- cbind(poolSeqStats[2:7,],c(0,0.3,0.1,0.8,0.85,1.0))
  colnames(PoolSeq_summary)[ncol(PoolSeq_summary)] <- "p"
  PoolSeq_summary[,"p"] <- as.vector(unlist(clines_thisLG[thisRow,c(9:14)]))
  PoolSeq_summary <- cbind(PoolSeq_summary,(2*PoolSeq_summary[,"p"]*(1-PoolSeq_summary[,"p"])))
  colnames(PoolSeq_summary)[ncol(PoolSeq_summary)] <- "2pq"
  PoolSeq_summary <- PoolSeq_summary[order(PoolSeq_summary[,"DistAlongTransect"]),]
  plot(x=PoolSeq_summary[,"DistAlongTransect"],y=PoolSeq_summary[,"p"],bg = "black",xaxs = "i", yaxs = "i",xlab="", ylab="",main="",xlim=c(0,24),ylim=c(-0.1,1.1),type='l',cex.main=0.9,bty="n",yaxt="n",xaxt="n")
  axis(1,at=seq(0,24,1),labels=NA,col.axis="black",las=2,cex.axis=1.1,lwd=1.6,pos=-0.01, tck = -.02)
  axis(1,at=seq(0,24+1,2),labels=seq(0,24000,2000)/1000,col.axis="black",tck=-0.3,las=1,cex.axis=1.1,lwd=0,pos=0.1)
  axis(2,at=seq(-0.01,1.01,0.1),labels=NA,col.axis="black",las=1,cex.axis=1.1,lwd=1.6,pos=0,tck = -.02)
  axis(2,at=seq(0,1.0,0.2),labels=seq(0,1,0.2),col.axis="black",tck=-.03,las=1,cex.axis=1.1,lwd=0,pos=3)
  lines(x=c(24,24),y=c(0,1),lwd=1.8)
  lines(x=c(0,24),y=c(1,1),lwd=1.8)
  title(paste(paste("LG",clines_thisLG[thisRow,"LG"]),clines_thisLG[thisRow,"position"],sep="- "), line=0, cex.main=0.8)
  lines(x = c(clines_thisLG[thisRow,'centre']/1000,clines_thisLG[thisRow,'centre']/1000), y = c(0,1),lty="dotted",lwd=1.3,col='black')
  lines(x = c(12.580,12.580), y = c(0,1),lty="dotted",lwd=1.3,col='red')

  points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"])
  lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1)
 thisWidth <- as.vector(unlist(clines_thisLG[thisRow,"width"]))
  text(x = 24, y = 0.28, labels = "w = ", cex = 1, col = "black", adj = 1)
  text(x = 23, y = 0.1, labels = sprintf("%.2f", as.numeric(thisWidth/1000)), cex = 1, col = "black", adj = 1)
    if (clines_thisLG[thisRow,"ColourGene"]!="-9") {
    # table(allChr_Clines_Dpth15[,"ColourGene"])
    # table(allChr_Clines_Dpth15[,"distanceToGene"])
    if (clines_thisLG[thisRow,"ColourGene"]=="CREMOSA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col="deeppink2")
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col="deeppink2",bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col="deeppink2")
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col="deeppink2",bg="deeppink2")
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="ELUTA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='pink')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='pink',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='pink')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='pink',bg='pink')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="ROSEA" | clines_thisLG[thisRow,"ColourGene"]=="ROS1" | clines_thisLG[thisRow,"ColourGene"]=="ROS2") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='red')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='red',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='red')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='red',bg='red')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="RUBIA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='blue')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='blue',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='blue')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='blue',bg='blue')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="FLAVIA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='yellow')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='yellow',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='yellow')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='yellow',bg='yellow')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="SULF") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='orange')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='orange',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='orange')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='orange',bg='orange')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="AURINA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='wheat')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='wheat',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='wheat')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='wheat',bg='wheat')
      }
    }
  }
}

cat('Number of clines LG1 deltaP >0.8 : ',numClines_LG1) 
cat('Number of clines LG1 deltaP >0.9 : ',numClines_LG1_deltaP9) 
cat('Gene breakdown LG1 deltaP >0.8 : ') 
table_LG1
cat('Gene breakdown LG1 deltaP >0.9 : ')
table_LG1_deltaP9

```

```{r plotting lots of little cline plots for LG1 for deltaP_0_9 that passes meanP check}
table(allChr_Clines_Dpth15_MeanP_pass$LG,allChr_Clines_Dpth15_MeanP_pass$ColourGene)

# comparing to some random clines with positive values in the middle position
nrow(allChr_Clines_Dpth15_MeanP_pass)
thisLG <- 1
clines_thisLG <- allChr_Clines_Dpth15_MeanP_pass %>% filter(LG == thisLG)
numClines_LG1 <- nrow(clines_thisLG)
table_LG1 <- table(clines_thisLG$ColourGene,clines_thisLG$distanceToGene)

clines_thisLG <- clines_thisLG %>% filter(AFD_15_20>= 0.9)
numClines_LG1_deltaP9 <- nrow(clines_thisLG)
table_LG1_deltaP9 <- table(clines_thisLG$ColourGene,clines_thisLG$distanceToGene)

# colour code clines by:
#   centre orientated
#   left leaning
#   right leaning
#   negative widths

par(mfrow=c(10,10))
par(mar=c(1,1,1,1))
par(oma=c(1,1,1,1))
PoolSeq_summary <- poolSeqStats
# PoolSeq_summary[, "DistAlongTransect"]

for (thisRow in 1:nrow(clines_thisLG)) {
  # thisRow <- 1
  PoolSeq_summary <- cbind(poolSeqStats[2:7,],c(0,0.3,0.1,0.8,0.85,1.0))
  colnames(PoolSeq_summary)[ncol(PoolSeq_summary)] <- "p"
  PoolSeq_summary[,"p"] <- as.vector(unlist(clines_thisLG[thisRow,c(9:14)]))
  PoolSeq_summary <- cbind(PoolSeq_summary,(2*PoolSeq_summary[,"p"]*(1-PoolSeq_summary[,"p"])))
  colnames(PoolSeq_summary)[ncol(PoolSeq_summary)] <- "2pq"
  PoolSeq_summary <- PoolSeq_summary[order(PoolSeq_summary[,"DistAlongTransect"]),]
  plot(x=PoolSeq_summary[,"DistAlongTransect"],y=PoolSeq_summary[,"p"],bg = "black",xaxs = "i", yaxs = "i",xlab="", ylab="",main="",xlim=c(0,24),ylim=c(-0.1,1.1),type='l',cex.main=0.9,bty="n",yaxt="n",xaxt="n")
  axis(1,at=seq(0,24,1),labels=NA,col.axis="black",las=2,cex.axis=1.1,lwd=1.6,pos=-0.01, tck = -.02)
  axis(1,at=seq(0,24+1,2),labels=seq(0,24000,2000)/1000,col.axis="black",tck=-0.3,las=1,cex.axis=1.1,lwd=0,pos=-0.001)
  axis(2,at=seq(-0.01,1.01,0.1),labels=NA,col.axis="black",las=1,cex.axis=1.1,lwd=1.6,pos=0,tck = -.02)
  axis(2,at=seq(0,1.0,0.2),labels=seq(0,1,0.2),col.axis="black",tck=-.03,las=1,cex.axis=1.1,lwd=0,pos=5)
  lines(x=c(24,24),y=c(0,1),lwd=1.8)
  lines(x=c(0,24),y=c(1,1),lwd=1.8)
  title(paste(paste("LG",clines_thisLG[thisRow,"LG"]),clines_thisLG[thisRow,"position"],sep="- "), line=0, cex.main=0.8)
  lines(x = c(12.580,12.580), y = c(0,1),lty="dotted",lwd=1.3,col='red')
  points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"])
  lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1)
    if (clines_thisLG[thisRow,"ColourGene"]!="-9") {
    # table(allChr_Clines_Dpth15[,"ColourGene"])
    # table(allChr_Clines_Dpth15[,"distanceToGene"])
    if (clines_thisLG[thisRow,"ColourGene"]=="CREMOSA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='green')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='green',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='green')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='green',bg='green')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="ELUTA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='pink')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='pink',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='pink')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='pink',bg='pink')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="ROSEA" | clines_thisLG[thisRow,"ColourGene"]=="ROS1" | clines_thisLG[thisRow,"ColourGene"]=="ROS2") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='red')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='red',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='red')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='red',bg='red')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="RUBIA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='blue')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='blue',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='blue')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='blue',bg='blue')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="FLAVIA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='yellow')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='yellow',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='yellow')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='yellow',bg='yellow')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="SULF") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='orange')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='orange',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='orange')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='orange',bg='orange')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="AURINA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='wheat')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='wheat',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='wheat')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='wheat',bg='wheat')
      }
    }
  }
}

cat('Number of clines LG1 deltaP >0.8 : ',numClines_LG1) 
cat('Number of clines LG1 deltaP >0.9 : ',numClines_LG1_deltaP9) 
cat('Gene breakdown LG1 deltaP >0.8 : ') 
table_LG1
cat('Gene breakdown LG1 deltaP >0.9 : ')
table_LG1_deltaP9

```

```{r plotting lots of little cline plots for LG1 that failed mean p for deltaP_0_9}
table(allChr_Clines_Dpth15_MeanP_fail$LG,allChr_Clines_Dpth15_MeanP_fail$ColourGene)

# comparing to some random clines with positive values in the middle position
nrow(allChr_Clines_Dpth15_MeanP_fail)
thisLG <- 1
clines_thisLG_fail <- allChr_Clines_Dpth15_MeanP_fail %>% filter(LG == thisLG)
numClines_LG1_fail <- nrow(clines_thisLG_fail)
table_LG1_fail <- table(clines_thisLG_fail$ColourGene,clines_thisLG_fail$distanceToGene)

clines_thisLG_fail <- clines_thisLG_fail %>% filter(AFD_15_20>= 0.9)
numClines_LG1_deltaP9 <- nrow(clines_thisLG_fail)
table_LG1_fail_deltaP9 <- table(clines_thisLG_fail$ColourGene,clines_thisLG_fail$distanceToGene)

# colour code clines by:
#   centre orientated
#   left leaning
#   right leaning
#   negative widths

par(mfrow=c(5,5))
par(mar=c(1,1,1,1))
par(oma=c(1,1,1,1))
PoolSeq_summary <- poolSeqStats
# PoolSeq_summary[, "DistAlongTransect"]

for (thisRow in 1:nrow(clines_thisLG_fail)) {
  # thisRow <- 1
  PoolSeq_summary <- cbind(poolSeqStats[2:7,],c(0,0.3,0.1,0.8,0.85,1.0))
  colnames(PoolSeq_summary)[ncol(PoolSeq_summary)] <- "p"
  PoolSeq_summary[,"p"] <- as.vector(unlist(clines_thisLG_fail[thisRow,c(9:14)]))
  PoolSeq_summary <- cbind(PoolSeq_summary,(2*PoolSeq_summary[,"p"]*(1-PoolSeq_summary[,"p"])))
  colnames(PoolSeq_summary)[ncol(PoolSeq_summary)] <- "2pq"
  PoolSeq_summary <- PoolSeq_summary[order(PoolSeq_summary[,"DistAlongTransect"]),]
  plot(x=PoolSeq_summary[,"DistAlongTransect"],y=PoolSeq_summary[,"p"],bg = "black",xaxs = "i", yaxs = "i",xlab="", ylab="",main="",xlim=c(0,24),ylim=c(-0.1,1.1),type='l',cex.main=0.9,bty="n",yaxt="n",xaxt="n")
  axis(1,at=seq(0,24,1),labels=NA,col.axis="black",las=2,cex.axis=1.1,lwd=1.6,pos=-0.01, tck = -.02)
  axis(1,at=seq(0,24+1,2),labels=seq(0,24000,2000)/1000,col.axis="black",tck=-0.3,las=1,cex.axis=1.1,lwd=0,pos=-0.001)
  axis(2,at=seq(-0.01,1.01,0.1),labels=NA,col.axis="black",las=1,cex.axis=1.1,lwd=1.6,pos=0,tck = -.02)
  axis(2,at=seq(0,1.0,0.2),labels=seq(0,1,0.2),col.axis="black",tck=-.03,las=1,cex.axis=1.1,lwd=0,pos=5)
  lines(x=c(24,24),y=c(0,1),lwd=1.8)
  lines(x=c(0,24),y=c(1,1),lwd=1.8)
  title(paste(paste("LG",clines_thisLG_fail[thisRow,"LG"]),clines_thisLG_fail[thisRow,"position"],sep="- "), line=0, cex.main=0.8)
  lines(x = c(12.580,12.580), y = c(0,1),lty="dotted",lwd=1.3,col='red')
  points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"])
  lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1)
    if (clines_thisLG_fail[thisRow,"ColourGene"]!="-9") {
    # table(allChr_Clines_Dpth15[,"ColourGene"])
    # table(allChr_Clines_Dpth15[,"distanceToGene"])
    if (clines_thisLG_fail[thisRow,"ColourGene"]=="CREMOSA") {
      if (clines_thisLG_fail[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='green')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='green',bg='white')
      }
      if (clines_thisLG_fail[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='green')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='green',bg='green')
      }
    }
    if (clines_thisLG_fail[thisRow,"ColourGene"]=="ELUTA") {
      if (clines_thisLG_fail[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='pink')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='pink',bg='white')
      }
      if (clines_thisLG_fail[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='pink')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='pink',bg='pink')
      }
    }
    if (clines_thisLG_fail[thisRow,"ColourGene"]=="ROSEA" | clines_thisLG_fail[thisRow,"ColourGene"]=="ROS1" | clines_thisLG_fail[thisRow,"ColourGene"]=="ROS2") {
      if (clines_thisLG_fail[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='red')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='red',bg='white')
      }
      if (clines_thisLG_fail[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='red')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='red',bg='red')
      }
    }
    if (clines_thisLG_fail[thisRow,"ColourGene"]=="RUBIA") {
      if (clines_thisLG_fail[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='blue')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='blue',bg='white')
      }
      if (clines_thisLG_fail[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='blue')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='blue',bg='blue')
      }
    }
    if (clines_thisLG_fail[thisRow,"ColourGene"]=="FLAVIA") {
      if (clines_thisLG_fail[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='yellow')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='yellow',bg='white')
      }
      if (clines_thisLG_fail[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='yellow')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='yellow',bg='yellow')
      }
    }
    if (clines_thisLG_fail[thisRow,"ColourGene"]=="SULF") {
      if (clines_thisLG_fail[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='orange')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='orange',bg='white')
      }
      if (clines_thisLG_fail[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='orange')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='orange',bg='orange')
      }
    }
    if (clines_thisLG_fail[thisRow,"ColourGene"]=="AURINA") {
      if (clines_thisLG_fail[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='wheat')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='wheat',bg='white')
      }
      if (clines_thisLG_fail[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='wheat')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='wheat',bg='wheat')
      }
    }
  }
}

cat('Number of clines LG1 deltaP >0.8 : ',numClines_LG1) 
cat('Number of clines LG1 deltaP >0.9 : ',numClines_LG1_deltaP9) 
cat('Gene breakdown LG1 deltaP >0.8 : ') 
table_LG1
cat('Gene breakdown LG1 deltaP >0.9 : ')
table_LG1_deltaP9

```

Ultimately, there is not a lot of difference in the look of the clines in whether we used this meanP filter (i.e. mean of 2 outer magentas must be higher than mean of two inner yellow - ignoring effects of over the pass). Therefore, we continue with the basic filtering of depth15 and delta p from the outer most pools as before.

* warning LG 2 has thousands, turn off for pdf markdown because too many loci or select subset

```{r plotting lots of little cline plots for LG2 for deltaP_0_9}
table(allChr_Clines_Dpth15$LG,allChr_Clines_Dpth15$ColourGene)

#poolSeqStats <- demeData$ros_assembly_543443$poolSeqStats
# comparing to some random clines with positive values in the middle position
allChr_Clines_Dpth15 <- allChr_Clines_filter[allChr_Clines_filter[,"poolsPass_Depth15"]>=5,]
nrow(allChr_Clines_Dpth15)
thisLG <- 2
clines_thisLG <- allChr_Clines_Dpth15 %>%
                    filter(LG == thisLG)
numClines_LG2 <- nrow(clines_thisLG)
table_LG2 <- table(clines_thisLG$ColourGene,clines_thisLG$distanceToGene)

clines_thisLG <- clines_thisLG %>%
                    filter(AFD_15_20>= 0.9)
numClines_LG2_deltaP9 <- nrow(clines_thisLG)
table_LG2_deltaP9 <- table(clines_thisLG$ColourGene,clines_thisLG$distanceToGene)

# colour code clines by:
#   centre orientated
#   left leaning
#   right leaning
#   negative widths

par(mfrow=c(10,10))
par(mar=c(1,1,1,1))
par(oma=c(1,1,1,1))

for (thisRow in 1:nrow(clines_thisLG)) {
  # thisRow <- 2500
  PoolSeq_summary <- cbind(poolSeqStats[2:7,],c(0,0.3,0.1,0.8,0.85,1.0))
  colnames(PoolSeq_summary)[ncol(PoolSeq_summary)] <- "p"
  PoolSeq_summary[,"p"] <- as.vector(unlist(clines_thisLG[thisRow,c(9:14)]))
  PoolSeq_summary <- cbind(PoolSeq_summary,(2*PoolSeq_summary[,"p"]*(1-PoolSeq_summary[,"p"])))
  colnames(PoolSeq_summary)[ncol(PoolSeq_summary)] <- "2pq"
  PoolSeq_summary <- PoolSeq_summary[order(PoolSeq_summary[,"DistAlongTransect"]),]
  plot(x=PoolSeq_summary[,"DistAlongTransect"],y=PoolSeq_summary[,"p"],bg = "black",xaxs = "i", yaxs = "i",xlab="", ylab="",main="",xlim=c(0,24),ylim=c(-0.1,1.1),type='l',cex.main=0.9,bty="n",yaxt="n",xaxt="n")
  axis(1,at=seq(0,24,1),labels=NA,col.axis="black",las=2,cex.axis=1.1,lwd=1.6,pos=-0.01, tck = -.02)
  axis(1,at=seq(0,24+1,2),labels=seq(0,24000,2000)/1000,col.axis="black",tck=-0.3,las=1,cex.axis=1.1,lwd=0,pos=0.1)
  axis(2,at=seq(-0.01,1.01,0.1),labels=NA,col.axis="black",las=1,cex.axis=1.1,lwd=1.6,pos=0,tck = -.02)
  axis(2,at=seq(0,1.0,0.2),labels=seq(0,1,0.2),col.axis="black",tck=-.03,las=1,cex.axis=1.1,lwd=0,pos=3)
  lines(x=c(24,24),y=c(0,1),lwd=1.8)
  lines(x=c(0,24),y=c(1,1),lwd=1.8)
  title(paste(paste("LG",clines_thisLG[thisRow,"LG"]),clines_thisLG[thisRow,"position"],sep="- "), line=0, cex.main=0.8)
   lines(x = c(clines_thisLG[thisRow,'centre']/1000,clines_thisLG[thisRow,'centre']/1000), y = c(0,1),lty="dotted",lwd=1.3,col='black')
  lines(x = c(12.580,12.580), y = c(0,1),lty="dotted",lwd=1.3,col='red')
  points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"])
  lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1)
  thisWidth <- as.vector(unlist(clines_thisLG[thisRow,"width"]))
  text(x = 24, y = 0.28, labels = "w = ", cex = 1, col = "black", adj = 1)
  text(x = 23, y = 0.1, labels = sprintf("%.2f", as.numeric(thisWidth/1000)), cex = 1, col = "black", adj = 1)
    if (clines_thisLG[thisRow,"ColourGene"]!="-9") {
    # table(allChr_Clines_Dpth15[,"ColourGene"])
    # table(allChr_Clines_Dpth15[,"distanceToGene"])
    if (clines_thisLG[thisRow,"ColourGene"]=="CREMOSA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='green')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='green',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='green')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='green',bg='green')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="ELUTA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='pink')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='pink',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='pink')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='pink',bg='pink')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="ROSEA" | clines_thisLG[thisRow,"ColourGene"]=="ROS1" | clines_thisLG[thisRow,"ColourGene"]=="ROS2") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='red')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='red',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='red')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='red',bg='red')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="RUBIA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='blue')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='blue',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='blue')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='blue',bg='blue')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="FLAVIA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col="#4DAF4A")
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col="#4DAF4A",bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col="#4DAF4A")
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col="#4DAF4A",bg="#4DAF4A")
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="SULF") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='orange')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='orange',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='orange')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='orange',bg='orange')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="AURINA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='wheat')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='wheat',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='wheat')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='wheat',bg='wheat')
      }
    }
  }
}

cat('Number of clines LG2 deltaP >0.8 : ',numClines_LG2) 
cat('Number of clines LG2 deltaP >0.9 : ',numClines_LG2_deltaP9) 
cat('Gene breakdown LG2 deltaP >0.8 : ') 
table_LG2
cat('Gene breakdown LG2 deltaP >0.9 : ')
table_LG2_deltaP9

```

```{r plotting lots of little cline plots for LG3 for deltaP_0_9}
table(allChr_Clines_Dpth15$LG,allChr_Clines_Dpth15$ColourGene)

# comparing to some random clines with positive values in the middle position
allChr_Clines_Dpth15 <- allChr_Clines_filter[allChr_Clines_filter[,"poolsPass_Depth15"]>=5,]
nrow(allChr_Clines_Dpth15)
thisLG <- 3
clines_thisLG <- allChr_Clines_Dpth15 %>%
                    filter(LG == thisLG)
numClines_LG3 <- nrow(clines_thisLG)
table_LG3 <- table(clines_thisLG$ColourGene,clines_thisLG$distanceToGene)

clines_thisLG <- clines_thisLG %>%
                    filter(AFD_15_20>= 0.9)
numClines_LG3_deltaP9 <- nrow(clines_thisLG)
table_LG3_deltaP9 <- table(clines_thisLG$ColourGene,clines_thisLG$distanceToGene)

# colour code clines by:
#   centre orientated
#   left leaning
#   right leaning
#   negative widths

par(mfrow=c(10,10))
par(mar=c(1,1,1,1))
par(oma=c(1,1,1,1))

for (thisRow in 1:nrow(clines_thisLG)) {
  # thisRow <- 1
  PoolSeq_summary <- cbind(poolSeqStats[2:7,],c(0,0.3,0.1,0.8,0.85,1.0))
  colnames(PoolSeq_summary)[ncol(PoolSeq_summary)] <- "p"
  PoolSeq_summary[,"p"] <- as.vector(unlist(clines_thisLG[thisRow,c(9:14)]))
  PoolSeq_summary <- cbind(PoolSeq_summary,(2*PoolSeq_summary[,"p"]*(1-PoolSeq_summary[,"p"])))
  colnames(PoolSeq_summary)[ncol(PoolSeq_summary)] <- "2pq"
  PoolSeq_summary <- PoolSeq_summary[order(PoolSeq_summary[,"DistAlongTransect"]),]
  plot(x=PoolSeq_summary[,"DistAlongTransect"],y=PoolSeq_summary[,"p"],bg = "black",xaxs = "i", yaxs = "i",xlab="", ylab="",main="",xlim=c(0,24),ylim=c(-0.1,1.1),type='l',cex.main=0.9,bty="n",yaxt="n",xaxt="n")
  axis(1,at=seq(0,24,1),labels=NA,col.axis="black",las=2,cex.axis=1.1,lwd=1.6,pos=-0.01, tck = -.02)
  axis(1,at=seq(0,24+1,2),labels=seq(0,24000,2000)/1000,col.axis="black",tck=-0.3,las=1,cex.axis=1.1,lwd=0,pos=0.1)
  axis(2,at=seq(-0.01,1.01,0.1),labels=NA,col.axis="black",las=1,cex.axis=1.1,lwd=1.6,pos=0,tck = -.02)
  axis(2,at=seq(0,1.0,0.2),labels=seq(0,1,0.2),col.axis="black",tck=-.03,las=1,cex.axis=1.1,lwd=0,pos=3)
  lines(x=c(24,24),y=c(0,1),lwd=1.8)
  lines(x=c(0,24),y=c(1,1),lwd=1.8)
  title(paste(paste("LG",clines_thisLG[thisRow,"LG"]),clines_thisLG[thisRow,"position"],sep="- "), line=0, cex.main=0.8)
   lines(x = c(clines_thisLG[thisRow,'centre']/1000,clines_thisLG[thisRow,'centre']/1000), y = c(0,1),lty="dotted",lwd=1.3,col='black')
  lines(x = c(12.580,12.580), y = c(0,1),lty="dotted",lwd=1.3,col='red')
  points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"])
  lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1)
thisWidth <- as.vector(unlist(clines_thisLG[thisRow,"width"]))
  text(x = 24, y = 0.28, labels = "w = ", cex = 1, col = "black", adj = 1)
  text(x = 23, y = 0.1, labels = sprintf("%.2f", as.numeric(thisWidth/1000)), cex = 1, col = "black", adj = 1)
    if (clines_thisLG[thisRow,"ColourGene"]!="-9") {
    # table(allChr_Clines_Dpth15[,"ColourGene"])
    # table(allChr_Clines_Dpth15[,"distanceToGene"])
    if (clines_thisLG[thisRow,"ColourGene"]=="CREMOSA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='green')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='green',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='green')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='green',bg='green')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="ELUTA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='pink')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='pink',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='pink')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='pink',bg='pink')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="ROSEA" | clines_thisLG[thisRow,"ColourGene"]=="ROS1" | clines_thisLG[thisRow,"ColourGene"]=="ROS2") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='red')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='red',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='red')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='red',bg='red')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="RUBIA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='blue')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='blue',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='blue')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='blue',bg='blue')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="FLAVIA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='yellow')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='yellow',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='yellow')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='yellow',bg='yellow')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="SULF") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='orange')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='orange',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='orange')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='orange',bg='orange')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="AURINA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='wheat')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='wheat',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='wheat')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='wheat',bg='wheat')
      }
    }
  }
}

cat('Number of clines LG3 deltaP >0.8 : ',numClines_LG3) 
cat('Number of clines LG3 deltaP >0.9 : ',numClines_LG3_deltaP9) 
cat('Gene breakdown LG3 deltaP >0.8 : ') 
table_LG3
cat('Gene breakdown LG3 deltaP >0.9 : ')
table_LG3_deltaP9

```

```{r plotting lots of little cline plots for LG4 for deltaP_0_9}
table(allChr_Clines_Dpth15$LG,allChr_Clines_Dpth15$ColourGene)

# comparing to some random clines with positive values in the middle position
allChr_Clines_Dpth15 <- allChr_Clines_filter[allChr_Clines_filter[,"poolsPass_Depth15"]>=5,]
nrow(allChr_Clines_Dpth15)
thisLG <- 4
clines_thisLG <- allChr_Clines_Dpth15 %>%
                    filter(LG == thisLG)
numClines_LG4 <- nrow(clines_thisLG)
table_LG4 <- table(clines_thisLG$ColourGene,clines_thisLG$distanceToGene)

clines_thisLG <- clines_thisLG %>%
                    filter(AFD_15_20>= 0.9)
numClines_LG4_deltaP9 <- nrow(clines_thisLG)
table_LG4_deltaP9 <- table(clines_thisLG$ColourGene,clines_thisLG$distanceToGene)

# colour code clines by:
#   centre orientated
#   left leaning
#   right leaning
#   negative widths

par(mfrow=c(10,10))
par(mar=c(1,1,1,1))
par(oma=c(1,1,1,1))

for (thisRow in 1:nrow(clines_thisLG)) {
  # thisRow <- 1
  PoolSeq_summary <- cbind(poolSeqStats[2:7,],c(0,0.3,0.1,0.8,0.85,1.0))
  colnames(PoolSeq_summary)[ncol(PoolSeq_summary)] <- "p"
  PoolSeq_summary[,"p"] <- as.vector(unlist(clines_thisLG[thisRow,c(9:14)]))
  thisWidth <- as.vector(unlist(clines_thisLG[thisRow,"width"]))
  PoolSeq_summary <- cbind(PoolSeq_summary,(2*PoolSeq_summary[,"p"]*(1-PoolSeq_summary[,"p"])))
  colnames(PoolSeq_summary)[ncol(PoolSeq_summary)] <- "2pq"
  PoolSeq_summary <- PoolSeq_summary[order(PoolSeq_summary[,"DistAlongTransect"]),]
  plot(x=PoolSeq_summary[,"DistAlongTransect"],y=PoolSeq_summary[,"p"],bg = "black",xaxs = "i", yaxs = "i",xlab="", ylab="",main="",xlim=c(0,24),ylim=c(-0.1,1.1),type='l',cex.main=0.9,bty="n",yaxt="n",xaxt="n")
  axis(1,at=seq(0,24,1),labels=NA,col.axis="black",las=2,cex.axis=1.1,lwd=1.6,pos=-0.01, tck = -.02)
  axis(1,at=seq(0,24+1,2),labels=seq(0,24000,2000)/1000,col.axis="black",tck=-0.3,las=1,cex.axis=1.1,lwd=0,pos=0.1)
  axis(2,at=seq(-0.01,1.01,0.1),labels=NA,col.axis="black",las=1,cex.axis=1.1,lwd=1.6,pos=0,tck = -.02)
  axis(2,at=seq(0,1.0,0.2),labels=seq(0,1,0.2),col.axis="black",tck=-.03,las=1,cex.axis=1.1,lwd=0,pos=3)
  lines(x=c(24,24),y=c(0,1),lwd=1.8)
  lines(x=c(0,24),y=c(1,1),lwd=1.8)
  title(paste(paste("LG",clines_thisLG[thisRow,"LG"]),clines_thisLG[thisRow,"position"],sep="- "), line=0, cex.main=0.8)
   lines(x = c(clines_thisLG[thisRow,'centre']/1000,clines_thisLG[thisRow,'centre']/1000), y = c(0,1),lty="dotted",lwd=1.3,col='black')
  lines(x = c(12.580,12.580), y = c(0,1),lty="dotted",lwd=1.3,col='red')
  points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"])
  lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1)
thisWidth <- as.vector(unlist(clines_thisLG[thisRow,"width"]))
  text(x = 24, y = 0.28, labels = "w = ", cex = 1, col = "black", adj = 1)
  text(x = 23, y = 0.1, labels = sprintf("%.2f", as.numeric(thisWidth/1000)), cex = 1, col = "black", adj = 1)
    if (clines_thisLG[thisRow,"ColourGene"]!="-9") {
    # table(allChr_Clines_Dpth15[,"ColourGene"])
    # table(allChr_Clines_Dpth15[,"distanceToGene"])
    if (clines_thisLG[thisRow,"ColourGene"]=="CREMOSA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='green')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='green',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='green')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='green',bg='green')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="ELUTA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='pink')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='pink',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='pink')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='pink',bg='pink')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="ROSEA" | clines_thisLG[thisRow,"ColourGene"]=="ROS1" | clines_thisLG[thisRow,"ColourGene"]=="ROS2") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='red')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='red',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='red')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='red',bg='red')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="RUBIA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='blue')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='blue',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='blue')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='blue',bg='blue')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="FLAVIA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='yellow')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='yellow',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='yellow')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='yellow',bg='yellow')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="SULF") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col="#984EA3")
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col="#984EA3",bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col="#984EA3")
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col="#984EA3",bg="#984EA3")
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="AURINA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='wheat')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='wheat',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='wheat')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='wheat',bg='wheat')
      }
    }
  }
}

cat('Number of clines LG4 deltaP >0.8 : ',numClines_LG4) 
cat('Number of clines LG4 deltaP >0.9 : ',numClines_LG4_deltaP9) 
cat('Gene breakdown LG4 deltaP >0.8 : ') 
table_LG4
cat('Gene breakdown LG4 deltaP >0.9 : ')
table_LG4_deltaP9

```

```{r plotting lots of little cline plots for LG5 for deltaP_0_9}
table(allChr_Clines_Dpth15$LG,allChr_Clines_Dpth15$ColourGene)

# comparing to some random clines with positive values in the middle position
allChr_Clines_Dpth15 <- allChr_Clines_filter[allChr_Clines_filter[,"poolsPass_Depth15"]>=5,]
nrow(allChr_Clines_Dpth15)
thisLG <- 5
clines_thisLG <- allChr_Clines_Dpth15 %>%
                    filter(LG == thisLG)
numClines_LG5 <- nrow(clines_thisLG)
table_LG5 <- table(clines_thisLG$ColourGene,clines_thisLG$distanceToGene)

clines_thisLG <- clines_thisLG %>%
                    filter(AFD_15_20>= 0.9)
numClines_LG5_deltaP9 <- nrow(clines_thisLG)
table_LG5_deltaP9 <- table(clines_thisLG$ColourGene,clines_thisLG$distanceToGene)

# colour code clines by:
#   centre orientated
#   left leaning
#   right leaning
#   negative widths

par(mfrow=c(10,10))
par(mar=c(1,1,1,1))
par(oma=c(1,1,1,1))

for (thisRow in 1:nrow(clines_thisLG)) {
  # thisRow <- 1
  PoolSeq_summary <- cbind(poolSeqStats[2:7,],c(0,0.3,0.1,0.8,0.85,1.0))
  colnames(PoolSeq_summary)[ncol(PoolSeq_summary)] <- "p"
  PoolSeq_summary[,"p"] <- as.vector(unlist(clines_thisLG[thisRow,c(9:14)]))
  
  PoolSeq_summary <- cbind(PoolSeq_summary,(2*PoolSeq_summary[,"p"]*(1-PoolSeq_summary[,"p"])))
  colnames(PoolSeq_summary)[ncol(PoolSeq_summary)] <- "2pq"
  PoolSeq_summary <- PoolSeq_summary[order(PoolSeq_summary[,"DistAlongTransect"]),]
  plot(x=PoolSeq_summary[,"DistAlongTransect"],y=PoolSeq_summary[,"p"],bg = "black",xaxs = "i", yaxs = "i",xlab="", ylab="",main="",xlim=c(0,24),ylim=c(-0.1,1.1),type='l',cex.main=0.9,bty="n",yaxt="n",xaxt="n")
 axis(1,at=seq(0,24,1),labels=NA,col.axis="black",las=2,cex.axis=1.1,lwd=1.6,pos=-0.01, tck = -.02)
  axis(1,at=seq(0,24+1,2),labels=seq(0,24000,2000)/1000,col.axis="black",tck=-0.3,las=1,cex.axis=1.1,lwd=0,pos=0.1)
  axis(2,at=seq(-0.01,1.01,0.1),labels=NA,col.axis="black",las=1,cex.axis=1.1,lwd=1.6,pos=0,tck = -.02)
  axis(2,at=seq(0,1.0,0.2),labels=seq(0,1,0.2),col.axis="black",tck=-.03,las=1,cex.axis=1.1,lwd=0,pos=3)
  lines(x=c(24,24),y=c(0,1),lwd=1.8)
  lines(x=c(0,24),y=c(1,1),lwd=1.8)
  title(paste(paste("LG",clines_thisLG[thisRow,"LG"]),clines_thisLG[thisRow,"position"],sep="- "), line=0, cex.main=0.8)
  lines(x = c(clines_thisLG[thisRow,'centre']/1000,clines_thisLG[thisRow,'centre']/1000), y = c(0,1),lty="dotted",lwd=1.3,col='black')
  lines(x = c(12.580,12.580), y = c(0,1),lty="dotted",lwd=1.3,col='red')
  points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"])
  lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1)
  thisWidth <- as.vector(unlist(clines_thisLG[thisRow,"width"]))
  text(x = 24, y = 0.28, labels = "w = ", cex = 1, col = "black", adj = 1)
  text(x = 23, y = 0.1, labels = sprintf("%.2f", as.numeric(thisWidth/1000)), cex = 1, col = "black", adj = 1)

    if (clines_thisLG[thisRow,"ColourGene"]!="-9") {
    # table(allChr_Clines_Dpth15[,"ColourGene"])
    # table(allChr_Clines_Dpth15[,"distanceToGene"])
    if (clines_thisLG[thisRow,"ColourGene"]=="CREMOSA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='green')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='green',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='green')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='green',bg='green')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="ELUTA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='pink')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='pink',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='pink')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='pink',bg='pink')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="ROSEA" | clines_thisLG[thisRow,"ColourGene"]=="ROS1" | clines_thisLG[thisRow,"ColourGene"]=="ROS2") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='red')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='red',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='red')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='red',bg='red')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="RUBIA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col="#E41A1C")
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col="#E41A1C",bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col="#E41A1C")
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col="#E41A1C",bg="#E41A1C")
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="FLAVIA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='yellow')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='yellow',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='yellow')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='yellow',bg='yellow')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="SULF") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='orange')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='orange',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='orange')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='orange',bg='orange')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="AURINA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='wheat')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='wheat',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='wheat')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='wheat',bg='wheat')
      }
    }
  }
}

cat('Number of clines LG5 deltaP >0.8 : ',numClines_LG5) 
cat('Number of clines LG5 deltaP >0.9 : ',numClines_LG5_deltaP9) 
cat('Gene breakdown LG5 deltaP >0.8 : ') 
table_LG5
cat('Gene breakdown LG5 deltaP >0.9 : ')
table_LG5_deltaP9

```

```{r plotting lots of little cline plots for LG6 for deltaP_0_9}
table(allChr_Clines_Dpth15$LG,allChr_Clines_Dpth15$ColourGene)

# comparing to some random clines with positive values in the middle position
allChr_Clines_Dpth15 <- allChr_Clines_filter[allChr_Clines_filter[,"poolsPass_Depth15"]>=5,]
nrow(allChr_Clines_Dpth15)
thisLG <- 6
clines_thisLG <- allChr_Clines_Dpth15 %>%
                    filter(LG == thisLG)
numClines_LG6 <- nrow(clines_thisLG)
table_LG6 <- table(clines_thisLG$ColourGene,clines_thisLG$distanceToGene)

clines_thisLG <- clines_thisLG %>%
                    filter(AFD_15_20>= 0.9)
numClines_LG6_deltaP9 <- nrow(clines_thisLG)
table_LG6_deltaP9 <- table(clines_thisLG$ColourGene,clines_thisLG$distanceToGene)

# colour code clines by:
#   centre orientated
#   left leaning
#   right leaning
#   negative widths

par(mfrow=c(10,10))
par(mar=c(1,1,1,1))
par(oma=c(1,1,1,1))

for (thisRow in 1:nrow(clines_thisLG)) {
  # thisRow <- 1
  PoolSeq_summary <- cbind(poolSeqStats[2:7,],c(0,0.3,0.1,0.8,0.85,1.0))
  colnames(PoolSeq_summary)[ncol(PoolSeq_summary)] <- "p"
  PoolSeq_summary[,"p"] <- as.vector(unlist(clines_thisLG[thisRow,c(9:14)]))
  PoolSeq_summary <- cbind(PoolSeq_summary,(2*PoolSeq_summary[,"p"]*(1-PoolSeq_summary[,"p"])))
  colnames(PoolSeq_summary)[ncol(PoolSeq_summary)] <- "2pq"
  PoolSeq_summary <- PoolSeq_summary[order(PoolSeq_summary[,"DistAlongTransect"]),]
  plot(x=PoolSeq_summary[,"DistAlongTransect"],y=PoolSeq_summary[,"p"],bg = "black",xaxs = "i", yaxs = "i",xlab="", ylab="",main="",xlim=c(0,24),ylim=c(-0.1,1.1),type='l',cex.main=0.9,bty="n",yaxt="n",xaxt="n")
  axis(1,at=seq(0,24,1),labels=NA,col.axis="black",las=2,cex.axis=1.1,lwd=1.6,pos=-0.01, tck = -.02)
  axis(1,at=seq(0,24+1,2),labels=seq(0,24000,2000)/1000,col.axis="black",tck=-0.3,las=1,cex.axis=1.1,lwd=0,pos=0.1)
  axis(2,at=seq(-0.01,1.01,0.1),labels=NA,col.axis="black",las=1,cex.axis=1.1,lwd=1.6,pos=0,tck = -.02)
  axis(2,at=seq(0,1.0,0.2),labels=seq(0,1,0.2),col.axis="black",tck=-.03,las=1,cex.axis=1.1,lwd=0,pos=3)
  lines(x=c(24,24),y=c(0,1),lwd=1.8)
  lines(x=c(0,24),y=c(1,1),lwd=1.8)
  title(paste(paste("LG",clines_thisLG[thisRow,"LG"]),clines_thisLG[thisRow,"position"],sep="- "), line=0, cex.main=0.8)
  lines(x = c(clines_thisLG[thisRow,'centre']/1000,clines_thisLG[thisRow,'centre']/1000), y = c(0,1),lty="dotted",lwd=1.3,col='black')
  lines(x = c(12.580,12.580), y = c(0,1),lty="dotted",lwd=1.3,col='red')
  points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"])
  lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1)
  thisWidth <- as.vector(unlist(clines_thisLG[thisRow,"width"]))
  text(x = 24, y = 0.28, labels = "w = ", cex = 1, col = "black", adj = 1)
  text(x = 23, y = 0.1, labels = sprintf("%.2f", as.numeric(thisWidth/1000)), cex = 1, col = "black", adj = 1)
    if (clines_thisLG[thisRow,"ColourGene"]!="-9") {
    # table(allChr_Clines_Dpth15[,"ColourGene"])
    # table(allChr_Clines_Dpth15[,"distanceToGene"])
    if (clines_thisLG[thisRow,"ColourGene"]=="CREMOSA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='green')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='green',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='green')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='green',bg='green')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="ELUTA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='pink')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='pink',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='pink')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='pink',bg='pink')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="ROSEA" | clines_thisLG[thisRow,"ColourGene"]=="ROS1" | clines_thisLG[thisRow,"ColourGene"]=="ROS2") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col="#A65628")
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col="#A65628",bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col="#A65628")
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col="#A65628",bg="#A65628")
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="RUBIA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='blue')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='blue',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='blue')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='blue',bg='blue')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="FLAVIA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='yellow')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='yellow',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='yellow')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='yellow',bg='yellow')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="SULF") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='orange')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='orange',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='orange')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='orange',bg='orange')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="AURINA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='wheat')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='wheat',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='wheat')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='wheat',bg='wheat')
      }
    }
  }
}

cat('Number of clines LG6 deltaP >0.8 : ',numClines_LG6) 
cat('Number of clines LG6 deltaP >0.9 : ',numClines_LG6_deltaP9) 
cat('Gene breakdown LG6 deltaP >0.8 : ') 
table_LG6
cat('Gene breakdown LG6 deltaP >0.9 : ')
table_LG6_deltaP9

```

```{r plotting lots of little cline plots for LG7 for deltaP_0_9}
table(allChr_Clines_Dpth15$LG,allChr_Clines_Dpth15$ColourGene)

# comparing to some random clines with positive values in the middle position
allChr_Clines_Dpth15 <- allChr_Clines_filter[allChr_Clines_filter[,"poolsPass_Depth15"]>=5,]
nrow(allChr_Clines_Dpth15)
thisLG <- 7
clines_thisLG <- allChr_Clines_Dpth15 %>%
                    filter(LG == thisLG)
numClines_LG7 <- nrow(clines_thisLG)
table_LG7 <- table(clines_thisLG$ColourGene,clines_thisLG$distanceToGene)

clines_thisLG <- clines_thisLG %>%
                    filter(AFD_15_20>= 0.9)
numClines_LG7_deltaP9 <- nrow(clines_thisLG)
table_LG7_deltaP9 <- table(clines_thisLG$ColourGene,clines_thisLG$distanceToGene)

# colour code clines by:
#   centre orientated
#   left leaning
#   right leaning
#   negative widths

par(mfrow=c(10,10))
par(mar=c(1,1,1,1))
par(oma=c(1,1,1,1))

for (thisRow in 1:nrow(clines_thisLG)) {
  # thisRow <- 1
  PoolSeq_summary <- cbind(poolSeqStats[2:7,],c(0,0.3,0.1,0.8,0.85,1.0))
  colnames(PoolSeq_summary)[ncol(PoolSeq_summary)] <- "p"
  PoolSeq_summary[,"p"] <- as.vector(unlist(clines_thisLG[thisRow,c(9:14)]))
  PoolSeq_summary <- cbind(PoolSeq_summary,(2*PoolSeq_summary[,"p"]*(1-PoolSeq_summary[,"p"])))
  colnames(PoolSeq_summary)[ncol(PoolSeq_summary)] <- "2pq"
  PoolSeq_summary <- PoolSeq_summary[order(PoolSeq_summary[,"DistAlongTransect"]),]
  plot(x=PoolSeq_summary[,"DistAlongTransect"],y=PoolSeq_summary[,"p"],bg = "black",xaxs = "i", yaxs = "i",xlab="", ylab="",main="",xlim=c(0,24),ylim=c(-0.1,1.1),type='l',cex.main=0.9,bty="n",yaxt="n",xaxt="n")
 axis(1,at=seq(0,24,1),labels=NA,col.axis="black",las=2,cex.axis=1.1,lwd=1.6,pos=-0.01, tck = -.02)
  axis(1,at=seq(0,24+1,2),labels=seq(0,24000,2000)/1000,col.axis="black",tck=-0.3,las=1,cex.axis=1.1,lwd=0,pos=0.1)
  axis(2,at=seq(-0.01,1.01,0.1),labels=NA,col.axis="black",las=1,cex.axis=1.1,lwd=1.6,pos=0,tck = -.02)
  axis(2,at=seq(0,1.0,0.2),labels=seq(0,1,0.2),col.axis="black",tck=-.03,las=1,cex.axis=1.1,lwd=0,pos=3)
  lines(x=c(24,24),y=c(0,1),lwd=1.8)
  lines(x=c(0,24),y=c(1,1),lwd=1.8)
  title(paste(paste("LG",clines_thisLG[thisRow,"LG"]),clines_thisLG[thisRow,"position"],sep="- "), line=0, cex.main=0.8)
  lines(x = c(clines_thisLG[thisRow,'centre']/1000,clines_thisLG[thisRow,'centre']/1000), y = c(0,1),lty="dotted",lwd=1.3,col='black')
  lines(x = c(12.580,12.580), y = c(0,1),lty="dotted",lwd=1.3,col='red')
  points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"])
  lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1)
thisWidth <- as.vector(unlist(clines_thisLG[thisRow,"width"]))
  text(x = 24, y = 0.28, labels = "w = ", cex = 1, col = "black", adj = 1)
  text(x = 23, y = 0.1, labels = sprintf("%.2f", as.numeric(thisWidth/1000)), cex = 1, col = "black", adj = 1)
    if (clines_thisLG[thisRow,"ColourGene"]!="-9") {
    # table(allChr_Clines_Dpth15[,"ColourGene"])
    # table(allChr_Clines_Dpth15[,"distanceToGene"])
    if (clines_thisLG[thisRow,"ColourGene"]=="CREMOSA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='green')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='green',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='green')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='green',bg='green')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="ELUTA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='pink')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='pink',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='pink')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='pink',bg='pink')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="ROSEA" | clines_thisLG[thisRow,"ColourGene"]=="ROS1" | clines_thisLG[thisRow,"ColourGene"]=="ROS2") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='red')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='red',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='red')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='red',bg='red')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="RUBIA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='blue')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='blue',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='blue')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='blue',bg='blue')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="FLAVIA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='yellow')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='yellow',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='yellow')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='yellow',bg='yellow')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="SULF") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='orange')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='orange',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='orange')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='orange',bg='orange')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="AURINA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='wheat')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='wheat',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='wheat')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='wheat',bg='wheat')
      }
    }
  }
}

cat('Number of clines LG7 deltaP >0.8 : ',numClines_LG7) 
cat('Number of clines LG7 deltaP >0.9 : ',numClines_LG7_deltaP9) 
cat('Gene breakdown LG7 deltaP >0.8 : ') 
table_LG7
cat('Gene breakdown LG7 deltaP >0.9 : ')
table_LG7_deltaP9

```

```{r plotting lots of little cline plots for LG8 for deltaP_0_9}
table(allChr_Clines_Dpth15$LG,allChr_Clines_Dpth15$ColourGene)

# comparing to some random clines with positive values in the middle position
allChr_Clines_Dpth15 <- allChr_Clines_filter[allChr_Clines_filter[,"poolsPass_Depth15"]>=5,]
nrow(allChr_Clines_Dpth15)
thisLG <- 8
clines_thisLG <- allChr_Clines_Dpth15 %>%
                    filter(LG == thisLG)
numClines_LG8 <- nrow(clines_thisLG)
table_LG8 <- table(clines_thisLG$ColourGene,clines_thisLG$distanceToGene)

clines_thisLG <- clines_thisLG %>%
                    filter(AFD_15_20>= 0.9)
numClines_LG8_deltaP9 <- nrow(clines_thisLG)
table_LG8_deltaP9 <- table(clines_thisLG$ColourGene,clines_thisLG$distanceToGene)

# colour code clines by:
#   centre orientated
#   left leaning
#   right leaning
#   negative widths

par(mfrow=c(10,10))
par(mar=c(1,1,1,1))
par(oma=c(1,1,1,1))

for (thisRow in 1:nrow(clines_thisLG)) {
  # thisRow <- 1
  PoolSeq_summary <- cbind(poolSeqStats[2:7,],c(0,0.3,0.1,0.8,0.85,1.0))
  colnames(PoolSeq_summary)[ncol(PoolSeq_summary)] <- "p"
  PoolSeq_summary[,"p"] <- as.vector(unlist(clines_thisLG[thisRow,c(9:14)]))
  PoolSeq_summary <- cbind(PoolSeq_summary,(2*PoolSeq_summary[,"p"]*(1-PoolSeq_summary[,"p"])))
  colnames(PoolSeq_summary)[ncol(PoolSeq_summary)] <- "2pq"
  PoolSeq_summary <- PoolSeq_summary[order(PoolSeq_summary[,"DistAlongTransect"]),]
  plot(x=PoolSeq_summary[,"DistAlongTransect"],y=PoolSeq_summary[,"p"],bg = "black",xaxs = "i", yaxs = "i",xlab="", ylab="",main="",xlim=c(0,24),ylim=c(-0.1,1.1),type='l',cex.main=0.9,bty="n",yaxt="n",xaxt="n")
  axis(1,at=seq(0,24,1),labels=NA,col.axis="black",las=2,cex.axis=1.1,lwd=1.6,pos=-0.01, tck = -.02)
  axis(1,at=seq(0,24+1,2),labels=seq(0,24000,2000)/1000,col.axis="black",tck=-0.3,las=1,cex.axis=1.1,lwd=0,pos=0.1)
  axis(2,at=seq(-0.01,1.01,0.1),labels=NA,col.axis="black",las=1,cex.axis=1.1,lwd=1.6,pos=0,tck = -.02)
  axis(2,at=seq(0,1.0,0.2),labels=seq(0,1,0.2),col.axis="black",tck=-.03,las=1,cex.axis=1.1,lwd=0,pos=3)
  lines(x=c(24,24),y=c(0,1),lwd=1.8)
  lines(x=c(0,24),y=c(1,1),lwd=1.8)
  title(paste(paste("LG",clines_thisLG[thisRow,"LG"]),clines_thisLG[thisRow,"position"],sep="- "), line=0, cex.main=0.8)
  lines(x = c(clines_thisLG[thisRow,'centre']/1000,clines_thisLG[thisRow,'centre']/1000), y = c(0,1),lty="dotted",lwd=1.3,col='black')
  lines(x = c(12.580,12.580), y = c(0,1),lty="dotted",lwd=1.3,col='red')
  points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"])
  lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1)
 thisWidth <- as.vector(unlist(clines_thisLG[thisRow,"width"]))
  text(x = 24, y = 0.28, labels = "w = ", cex = 1, col = "black", adj = 1)
  text(x = 23, y = 0.1, labels = sprintf("%.2f", as.numeric(thisWidth/1000)), cex = 1, col = "black", adj = 1)
    if (clines_thisLG[thisRow,"ColourGene"]!="-9") {
    # table(allChr_Clines_Dpth15[,"ColourGene"])
    # table(allChr_Clines_Dpth15[,"distanceToGene"])
    if (clines_thisLG[thisRow,"ColourGene"]=="CREMOSA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='green')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='green',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='green')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='green',bg='green')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="ELUTA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='pink')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='pink',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='pink')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='pink',bg='pink')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="ROSEA" | clines_thisLG[thisRow,"ColourGene"]=="ROS1" | clines_thisLG[thisRow,"ColourGene"]=="ROS2") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='red')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='red',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='red')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='red',bg='red')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="RUBIA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='blue')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='blue',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='blue')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='blue',bg='blue')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="FLAVIA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='yellow')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='yellow',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='yellow')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='yellow',bg='yellow')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="SULF") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='orange')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='orange',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='orange')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='orange',bg='orange')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="AURINA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='wheat')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='wheat',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='wheat')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='wheat',bg='wheat')
      }
    }
  }
}

cat('Number of clines LG8 deltaP >0.8 : ',numClines_LG8) 
cat('Number of clines LG8 deltaP >0.9 : ',numClines_LG8_deltaP9) 
cat('Gene breakdown LG8 deltaP >0.8 : ') 
table_LG8
cat('Gene breakdown LG8 deltaP >0.9 : ')
table_LG8_deltaP9

```

```{r Supporting Info plotting of transect for WGSPoolSeq in relation to KASP 50m demes, echo=F, include=T,eval=T,warning=F,width = 14.5, height = 7, pointsize=18,fig.cap="\\textbf{\\label{fig:HZtransect}Transect through hybrid zone for RUBIA KASP marker with allele frequencies in 2D followed by plot of allele frequencies collapsed to 1D}."}
# # folder structure file system pedigree
# folderLoc <- "/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/Pedigree_DF/"
# folderStructure <- read.table(paste0(folderLoc,"/Scripts/R/","folderStructure_MQ.txt"),header=TRUE,fill=FALSE,na.strings = "?")
# 
# setwd("/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/Pedigree_DF/PostProcessedData/")
# 
# # Import some additional data
# GenoEcol <- read.csv("/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/Pedigree_DF/PostProcessedData/FinalGenoEcol_short_int.csv")
# 
# # Import KASP data
# GenoEcol <- read.csv("FinalGenoEcol_short_int_publish.csv")
# 
# SNPlociListUpdated <- read.csv("shortSummarySNPsAdj.csv")
# 
# # Pick out Planoles and parapatric pops using Easting and Northing
# GenoEcol_Planoles <- GenoEcol %>% 
#   dplyr::filter(Easting>410000 & Easting<435000 & Northing>4684000 & Northing<4692000)
# 
# PedigreeLoci_strict <- readMyFile(folderStructure,"folder_base_genetic","PedigreeLoci_strict.csv")
# PedigreeLoci_strict[,1] <- as.vector(PedigreeLoci_strict[,1])

# KASP markers
# CREMOSA (near UDP) Chr 1 "s1187_290152" 
# Flavia Chr 2 - 43.175, "s316_93292", "s316_257789", 
# SULF Chr 4 "s91_39699", 
# RUBIA Chr 5 (near FLS) "s261_720757"
# ROS EL Chr 6 "ros_assembly_543443","ros_assembly_715015"

# make matrix for poolSeq information
# 
# poolLoc <- as.data.frame(matrix(0,7,8))
# colnames(poolLoc) <- c("pool","Easting","Northing","XPos_deme","YPos_deme","X_adj","Y_adj","DistAlongTransect")
# poolLoc[1:7,1] <- c("DemeKASP_Start","YP4","YP2","YP1","MP2","MP4","MP11")
# # new positions 2023
# poolLoc[2,2:3] <- c(411635.27,4690296.66) # YP4
# poolLoc[3,2:3] <- c(422120.82,4686387.46) # YP2
# poolLoc[4,2:3] <- c(421968.27,4686511.65) # YP1
# poolLoc[5,2:3] <- c(424440.68,4686082.79) # MP2
# poolLoc[6,2:3] <- c(425130.38,4685954.23) # MP4
# poolLoc[7,2:3] <- c(431641.97,4686865.38) # MP11
# poolLoc[,2] <- as.numeric(as.vector(poolLoc[,2]))
# poolLoc[,3] <- as.numeric(as.vector(poolLoc[,3]))
# poolLoc[,4] <- as.numeric(as.vector(poolLoc[,4]))
# poolLoc[,5] <- as.numeric(as.vector(poolLoc[,5]))
# poolLoc[,6] <- as.numeric(as.vector(poolLoc[,6]))
# poolLoc[,7] <- as.numeric(as.vector(poolLoc[,7]))


# try plot with the same locus only (KASP ros_assembly_543443) in WGS pools then change to this below
# allChr_Clines_ROSEL[allChr_Clines_ROSEL[,"position"]==52889078,]

HZ_50mDemes <- DemeSetup2D_PoolSeq(HZdata=GenoEcol_Planoles,lociNames="s1187_290152",demeSize=50,minSample=10,removePhenoMiss=F,transectGradient = -0.20, transectIntercept = 4900,  adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",poolSeqPlot=T,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=T,FigName="Chr1",plotCentreLine=T,poolSeqClines=allChr_Clines_Chr6clines,poolSeqStats=poolLoc,colourTable=colourChoice)
# str(HZ_50mDemes)

# positions of poolSeq demes along transect
poolSeqStats <- HZ_50mDemes$s1187_290152$poolSeqStats
# pdf version


```

# Interesting subset
locusSet_interesting <- c("s1187_290152", # near UDP LG1
              "s316_93292", "s316_257789", # Flavia LG 2 - 43.175
              "s91_39699", # LG 4 - sulf
              "s261_720757", # LG 5 - FLS
              "ros_assembly_543443","ros_assembly_715015", # ros & el
              "RosInferred")
    
# Not used in manuscript

multipanel plot with one for each chromosome showing allele frequencies across the transect in 1D

Plot shows the poolSeq allele frequencies, and on some chromosomes, an example KASP locus with frequencies
              
Chromosome 1
        
```{r Chr1 clines and KASP 100m demes plot for KASP CREMOSA}
HZ_200mDemes <- DemeSetup2D_PoolSeq(HZdata=GenoEcol_Planoles,lociNames="s1187_290152",demeSize=50,minSample=10,removePhenoMiss=F,transectGradient = -0.20, transectIntercept = 4900,  adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",poolSeqPlot=T,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=T,FigName="Chr1",plotCentreLine=T,poolSeqClines=allChr_Clines_Chr1clines,poolSeqStats=poolSeqStats,colourTable=colourChoice)
allChr_Clines_Chr1clines
table(allChr_Clines_Chr1clines[,"ColourGene"],allChr_Clines_Chr1clines[,"distanceToGene"])
```

```{r Chr1 clines and KASP 100m demes plot for KASP CREMOSA deltaP_09}
HZ_100mDemes <- DemeSetup2D(HZdata=GenoEcol_Planoles,lociNames="s1187_290152",demeSize=100,minSample=10,removePhenoMiss=F,transectGradient = -0.20, transectIntercept = 4900,  adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",poolSeqPlot=T,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=T,FigName="Chr1_deltaP09",plotCentreLine=T,poolSeqClines=allChr_Clines_Chr1clines_deltaP_09,poolSeqStats=poolSeqStats,colourTable=colourChoice)


```

Chromosome 2 FLA down

```{r Chr2 clines and KASP 100m demes plot for KASP FLAVIA down}
colourChoice[colourChoice[,"Region"]=="FLAVIA",2:4] <- c((rgb(255,158,15,255,maxColorValue=255)))

HZ_100mDemes <- DemeSetup2D(HZdata=GenoEcol_Planoles,lociNames=c("s316_93292"),demeSize=100,minSample=10,removePhenoMiss=F,transectGradient = -0.20, transectIntercept = 4900,  adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",poolSeqPlot=T,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=T,FigName="Chr2",plotCentreLine=T,poolSeqClines=allChr_Clines_Chr2clines,poolSeqStats=poolLoc,colourTable=colourChoice)
```

```{r Chr2 clines and KASP 100m demes plot for KASP FLAVIA down but no poolSeq}
colourChoice[colourChoice[,"Region"]=="FLAVIA",2:4] <- c((rgb(232,119,34,255,maxColorValue=255)))

HZ_100mDemes <- DemeSetup2D(HZdata=GenoEcol_Planoles,lociNames=c("s316_93292"),demeSize=100,minSample=10,removePhenoMiss=F,transectGradient = -0.20, transectIntercept = 4900,  adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",poolSeqPlot=T,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=T,FigName="Chr2kasp",plotCentreLine=T,poolSeqClines=allChr_Clines_Chr2clines[1,],poolSeqStats=poolLoc,colourTable=colourChoice)
```

Chromosome 2 FLA up

```{r Chr2 clines and KASP 100m demes plot for KASP FLAVIA up}
colourChoice[colourChoice[,"Region"]=="FLAVIA",2:4] <- c((rgb(205,149,12,255,maxColorValue=255)))

HZ_100mDemes <- DemeSetup2D(HZdata=GenoEcol_Planoles,lociNames="s316_257789",demeSize=100,minSample=10,removePhenoMiss=F,transectGradient = -0.20, transectIntercept = 4900,  adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",poolSeqPlot=T,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=T,FigName="Chr2",plotCentreLine=T,poolSeqClines=allChr_Clines_Chr2clines,poolSeqStats=poolLoc,colourTable=colourChoice)
```

```{r Chr2 clines and KASP 100m demes plot for KASP FLAVIA up deltaP_09}
colourChoice[colourChoice[,"Region"]=="FLAVIA",2:4] <- c((rgb(205,149,12,255,maxColorValue=255)))

HZ_100mDemes <- DemeSetup2D(HZdata=GenoEcol_Planoles,lociNames="s316_257789",demeSize=100,minSample=10,removePhenoMiss=F,transectGradient = -0.20, transectIntercept = 4900,  adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",poolSeqPlot=T,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=T,FigName="Chr2_delta_09",plotCentreLine=T,poolSeqClines=allChr_Clines_Chr2clines_deltaP_09,poolSeqStats=poolLoc,colourTable=colourChoice)
```

Chromosome 3

```{r Chr3 clines and KASP 100m demes plot}
HZ_100mDemes <- DemeSetup2D(HZdata=GenoEcol_Planoles,lociNames="ros_assembly_543443",demeSize=100,minSample=10,removePhenoMiss=F,transectGradient = -0.20, transectIntercept = 4900,  adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",poolSeqPlot=T,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=F,FigName="Chr3",plotCentreLine=T,poolSeqClines=allChr_Clines_Chr3clines,poolSeqStats=poolLoc,colourTable=colourChoice)
```

```{r Chr3 clines and KASP 100m demes plot deltaP_09}
HZ_100mDemes <- DemeSetup2D(HZdata=GenoEcol_Planoles,lociNames="ros_assembly_543443",demeSize=100,minSample=10,removePhenoMiss=F,transectGradient = -0.20, transectIntercept = 4900,  adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",poolSeqPlot=T,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=F,FigName="Chr3_deltaP09",plotCentreLine=T,poolSeqClines=allChr_Clines_Chr3clines_deltaP_09,poolSeqStats=poolLoc,colourTable=colourChoice)
```

Chromosome 4

```{r Chr4 clines and KASP 100m demes plot for KASP SULF}
HZ_100mDemes <- DemeSetup2D(HZdata=GenoEcol_Planoles,lociNames="s91_39699",demeSize=100,minSample=10,removePhenoMiss=F,transectGradient = -0.20, transectIntercept = 4900,  adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",poolSeqPlot=T,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=T,FigName="Chr4",plotCentreLine=T,poolSeqClines=allChr_Clines_Chr4clines,poolSeqStats=poolLoc,colourTable=colourChoice)
```

```{r Chr4 clines and KASP 100m demes plot for KASP SULF deltaP09}
HZ_100mDemes <- DemeSetup2D(HZdata=GenoEcol_Planoles,lociNames="s91_39699",demeSize=100,minSample=10,removePhenoMiss=F,transectGradient = -0.20, transectIntercept = 4900,  adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",poolSeqPlot=T,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=T,FigName="Chr4_deltaP09",plotCentreLine=T,poolSeqClines=allChr_Clines_Chr4clines_deltaP_09,poolSeqStats=poolLoc,colourTable=colourChoice)
```

Chromosome 5

```{r Chr5 clines and KASP 100m demes plot for KASP RUBIA}
table(allChr_Clines_Chr5clines[,"ColourGene"],allChr_Clines_Chr5clines[,"distanceToGene"])

# Convert 'distanceToGene' column to a factor with the desired order
allChr_Clines_Chr5clines$distanceToGene <- factor(
    allChr_Clines_Chr5clines$distanceToGene, 
    levels = c("-9", "up30_100kb", "down30kb", "up30kb", "inGene")
    )
# Reorder the dataframe by the 'distanceToGene' factor levels
allChr_Clines_Chr5clines <- allChr_Clines_Chr5clines[order(allChr_Clines_Chr5clines$distanceToGene), ]

HZ_100mDemes <- DemeSetup2D_PoolSeq(HZdata=GenoEcol_Planoles,lociNames="s261_720757",demeSize=100,minSample=10,removePhenoMiss=F,transectGradient = -0.20, transectIntercept = 4900,  adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",poolSeqPlot=T,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=T,FigName="Chr5",plotCentreLine=T,poolSeqClines=allChr_Clines_Chr5clines,poolSeqStats=poolLoc,plotKASPcurve=T,colourTable=colourChoice)
```

```{r Chr5 clines and KASP 100m demes plot for KASP RUBIA deltaP09}

HZ_100mDemes <- DemeSetup2D(HZdata=GenoEcol_Planoles,lociNames="s261_720757",demeSize=100,minSample=10,removePhenoMiss=F,transectGradient = -0.20, transectIntercept = 4900,  adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",poolSeqPlot=T,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=T,FigName="Chr5_DeltaP_09",plotCentreLine=T,poolSeqClines=allChr_Clines_Chr5clines_deltaP_09,poolSeqStats=poolLoc,colourTable=colourChoice)
```

Chromosome 6

```{r Chr6 clines and KASP 100m demes plot for ROSel}
HZ_100mDemes <- DemeSetup2D_PoolSeq(HZdata=GenoEcol_Planoles,lociNames="ros_assembly_543443",demeSize=100,minSample=10,removePhenoMiss=F,transectGradient = -0.20, transectIntercept = 4900,  adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",poolSeqPlot=T,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=T,FigName="Chr6",plotCentreLine=T,poolSeqClines=allChr_Clines_Chr6clines,poolSeqStats=poolLoc,colourTable=colourChoice)
```

```{r Chr6 clines and KASP 100m demes plot for ROSel deltaP_09}
HZ_100mDemes <- DemeSetup2D(HZdata=GenoEcol_Planoles,lociNames="ros_assembly_543443",demeSize=100,minSample=10,removePhenoMiss=F,transectGradient = -0.20, transectIntercept = 4900,  adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",poolSeqPlot=T,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=T,FigName="Chr6",plotCentreLine=T,poolSeqClines=allChr_Clines_Chr6clines_deltaP_09,poolSeqStats=poolLoc,colourTable=colourChoice)
```

```{r Chr6 clines and KASP 100m demes plot for ROSel but with just a few ROS Poolseq}
HZ_100mDemes <- DemeSetup2D(HZdata=GenoEcol_Planoles,lociNames="ros_assembly_543443",demeSize=100,minSample=10,removePhenoMiss=F,transectGradient = -0.20, transectIntercept = 4900,  adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",poolSeqPlot=T,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=T,FigName="Chr6",plotCentreLine=T,poolSeqClines=allChr_Clines_Chr6clines,poolSeqStats=poolLoc,colourTable=colourChoice)
```

Chromosome 7

```{r Chr7 clines and KASP 100m demes plot for ROSel as yardstick}
HZ_100mDemes <- DemeSetup2D(HZdata=GenoEcol_Planoles,lociNames="ros_assembly_543443",demeSize=100,minSample=10,removePhenoMiss=F,transectGradient = -0.20, transectIntercept = 4900,  adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",poolSeqPlot=T,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=F,FigName="Chr7",plotCentreLine=T,poolSeqClines=allChr_Clines_Chr7clines,poolSeqStats=poolLoc,colourTable=colourChoice)
```

```{r Chr7 clines and KASP 100m demes plot for ROSel as yardstick}
HZ_100mDemes <- DemeSetup2D(HZdata=GenoEcol_Planoles,lociNames="ros_assembly_543443",demeSize=100,minSample=10,removePhenoMiss=F,transectGradient = -0.20, transectIntercept = 4900,  adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",poolSeqPlot=T,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=F,FigName="Chr7_deltaP_09",plotCentreLine=T,poolSeqClines=allChr_Clines_Chr7_elsewhere_deltaP_09,poolSeqStats=poolLoc,colourTable=colourChoice)
```

Chromosome 8

```{r Chr8 Fast clines and KASP 100m demes plot for ROSel as yardstick}
HZ_100mDemes <- DemeSetup2D(HZdata=GenoEcol_Planoles,lociNames="ros_assembly_543443",demeSize=100,minSample=10,removePhenoMiss=F,transectGradient = -0.20, transectIntercept = 4900,  adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",poolSeqPlot=T,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=F,FigName="Chr8",plotCentreLine=T,poolSeqClines=allChr_Clines_Chr8clines,poolSeqStats=poolLoc,colourTable=colourChoice)
```

```{r Chr8 Fast clines and KASP 100m demes plot for ROSel as yardstick deltaP_09}
HZ_100mDemes <- DemeSetup2D(HZdata=GenoEcol_Planoles,lociNames="ros_assembly_543443",demeSize=100,minSample=10,removePhenoMiss=F,transectGradient = -0.20, transectIntercept = 4900,  adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",poolSeqPlot=T,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=F,FigName="Chr8_deltaP_09",plotCentreLine=T,poolSeqClines=allChr_Clines_Chr8_elsewhere_deltaP_09,poolSeqStats=poolLoc,colourTable=colourChoice)
```

# Exploratory look at the effect of different deme span options

```{r functions for Supporting Info demonstrative histograms of three deme span options together WG clines}

demeSpanPlot <- function(clinesData,clinesDataFiltered,locusLG,locusPosition,locusColour,demeData,demeSpanOpt) {
  # clinesData <- allChr_Clines_chk; clinesDataFiltered <- allChr_Clines_filter
  # locusLG <-6 ; locusPosition <- 52988204 ; locusColour <- "blue"  # ELUTA linked example
  # demeData <- HZ_50mDemes; demeSpanOpt <- "Opt1"
PoolSeq_summary <- cbind(demeData$ros_assembly_543443$poolSeqStats[2:7,],c(0,0.3,0.1,0.8,0.85,1.0))
colnames(PoolSeq_summary)[ncol(PoolSeq_summary)] <- "p"
PoolSeq_summary[,"p"] <- as.vector(unlist(clinesData[clinesData[,"position"]==locusPosition & clinesData[,"LG"]==locusLG,c(9:14)]))
PoolSeq_summary <- cbind(PoolSeq_summary,(2*PoolSeq_summary[,"p"]*(1-PoolSeq_summary[,"p"])))
colnames(PoolSeq_summary)[ncol(PoolSeq_summary)] <- "2pq"
PoolSeq_summary <- PoolSeq_summary[order(PoolSeq_summary[,"DistAlongTransect"]),]

if (demeSpanOpt == "Opt1") {
  width <- round(as.vector(unlist(clinesData[clinesData[,"position"]==locusPosition & clinesData[,"LG"]==locusLG,"width_Opt1"])),0)
  centre <- round(as.vector(unlist(clinesData[clinesData[,"position"]==locusPosition & clinesData[,"LG"]==locusLG,"centre_Opt1"])),0)
  chartLabel1 <- paste0("width DS1 = ",width)
  chartLabel2 <- paste0("centre DS1 = ",centre)
}
if (demeSpanOpt == "Opt2") {
  width <- round(as.vector(unlist(clinesData[clinesData[,"position"]==locusPosition & clinesData[,"LG"]==locusLG,"width_Opt2"])),0)
  centre <- round(as.vector(unlist(clinesData[clinesData[,"position"]==locusPosition & clinesData[,"LG"]==locusLG,"centre_Opt2"])),0)
  chartLabel1 <- paste0("width DS2 = ",width)
  chartLabel2 <- paste0("centre DS2 = ",centre)
}

if (demeSpanOpt == "Opt3") {
  width <- round(as.vector(unlist(clinesData[clinesData[,"position"]==locusPosition & clinesData[,"LG"]==locusLG,"width_Opt3"])),0)
  centre <- round(as.vector(unlist(clinesData[clinesData[,"position"]==locusPosition & clinesData[,"LG"]==locusLG,"centre_Opt3"])),0)
  chartLabel1 <- paste0("width DS3 = ",width)
  chartLabel2 <- paste0("centre DS3 = ",centre)
}

plot(PoolSeq_summary[,"DistAlongTransect"],PoolSeq_summary[,"p"],bg = "black",xaxs = "i", yaxs = "i",
             xlab="",ylab="",main="",xlim=c(0,30000),
           ylim=c(-0.05,1.09),type='n',cex.main=1.3,bty="n",yaxt="n",xaxt="n")
title(paste(chartLabel1,chartLabel2,sep="\n"),line=0.1,cex.main=0.9)
  
axis(1,at=seq(0,26000,2000),labels=NA,col.axis="black",las=2,cex.axis=1.1,lwd=1.6,pos=-0.01, tck = -.02)
axis(1,at=seq(0,26000,4000),labels=seq(0,26000,4000)/1000,col.axis="black",tck=-0.3,las=1,cex.axis=1,lwd=0,pos=-0.02)
# allele frequency
axis(2,at=seq(-0.01,1.01,0.1),labels=NA,col.axis="black",las=1,cex.axis=1.1,lwd=1.6,pos=0,tck = -.02)
axis(2,at=seq(0,1.0,0.2),labels=seq(0,1,0.2),col.axis="black",tck=-.03,las=1,cex.axis=1,lwd=0,pos=1.02)
# 2pq
axis(2,at=seq(-0.01,1.01,0.1),labels=NA,col.axis="black",las=1,cex.axis=1.1,lwd=1.6,pos=26000,tck = -.02)
axis(2,at=seq(0,1.0,0.2),labels=seq(0,1,0.2),col.axis="black",tck=-.03,las=1,cex.axis=1.1,lwd=0,pos=29500)
lines(x=c(26000,26000),y=c(0,1),lwd=1.6)
lines(x=c(0,26000),y=c(1,1),lwd=1.6)
mtext("Distance along transect (km)",side=1,cex=1,line=2)
mtext("Allele frequency (p)",side=2,cex=1,line=2.2)
mtext("2pq",side=4,cex=1,line=-0.5)

# DS 1 histograms

if (demeSpanOpt=="Opt1") {
  rect(xleft=(0),xright=(5423),ybottom=0,ytop=PoolSeq_summary[1,"2pq"], col='lightgrey',lty = "solid", lwd=2, border='black')
  rect(xleft=(5423),xright=(11316),ybottom=0,ytop=PoolSeq_summary[2,"2pq"], col='lightgrey',lty = "solid", lwd=2, border='black')
  rect(xleft=(11316),xright=(12586),ybottom=0,ytop=PoolSeq_summary[3,"2pq"], col='lightgrey',lty = "solid", lwd=2, border='black')
  rect(xleft=(12586),xright=(14095),ybottom=0,ytop=PoolSeq_summary[4,"2pq"], col='lightgrey',lty = "solid", lwd=2, border='black')
  rect(xleft=(14095),xright=(17548),ybottom=0,ytop=PoolSeq_summary[5,"2pq"], col='lightgrey',lty = "solid", lwd=2, border='black')
  rect(xleft=(17548),xright=(25176),ybottom=0,ytop=PoolSeq_summary[6,"2pq"], col='lightgrey',lty = "solid", lwd=2, border='black')
  lines(x = c(12580,12580), y = c(0,1),lty=5,lwd=2,col='red')
}
if (demeSpanOpt=="Opt2") {
  rect(xleft=(0),xright=(4195),ybottom=0,ytop=PoolSeq_summary[1,"2pq"], col='lightgrey',lty = "solid", lwd=2, border='black')
  rect(xleft=(4195),xright=(8390),ybottom=0,ytop=PoolSeq_summary[2,"2pq"], col='lightgrey',lty = "solid", lwd=2, border='black')
  rect(xleft=(8390),xright=(12586),ybottom=0,ytop=PoolSeq_summary[3,"2pq"], col='lightgrey',lty = "solid", lwd=2, border='black')
  rect(xleft=(12586),xright=(16781),ybottom=0,ytop=PoolSeq_summary[4,"2pq"], col='lightgrey',lty = "solid", lwd=2, border='black')
  rect(xleft=(16781),xright=(20976),ybottom=0,ytop=PoolSeq_summary[5,"2pq"], col='lightgrey',lty = "solid", lwd=2, border='black')
  rect(xleft=(20976),xright=(25176),ybottom=0,ytop=PoolSeq_summary[6,"2pq"], col='lightgrey',lty = "solid", lwd=2, border='black')
  lines(x = c(12580,12580), y = c(0,1),lty=5,lwd=2,col='red')
}
if (demeSpanOpt=="Opt3") {
  rect(xleft=(0),xright=(10046),ybottom=0,ytop=PoolSeq_summary[1,"2pq"], col='lightgrey',lty = "solid", lwd=2, border='black')
  rect(xleft=(10046),xright=(11316),ybottom=0,ytop=PoolSeq_summary[2,"2pq"], col='lightgrey',lty = "solid", lwd=2, border='black')
  rect(xleft=(11316),xright=(12586),ybottom=0,ytop=PoolSeq_summary[3,"2pq"], col='lightgrey',lty = "solid", lwd=2, border='black')
  rect(xleft=(12586),xright=(14086),ybottom=0,ytop=PoolSeq_summary[4,"2pq"], col='lightgrey',lty = "solid", lwd=2, border='black')
  rect(xleft=(14086),xright=(15586),ybottom=0,ytop=PoolSeq_summary[5,"2pq"], col='lightgrey',lty = "solid", lwd=2, border='black')
  rect(xleft=(15586),xright=(25176),ybottom=0,ytop=PoolSeq_summary[6,"2pq"], col='lightgrey',lty = "solid", lwd=2, border='black')
  lines(x = c(12580,12580), y = c(0,1),lty=5,lwd=2,col='red')
}

lines(PoolSeq_summary[,"DistAlongTransect"],PoolSeq_summary[,"p"],lty=1,lwd=2.5,col='black')
points(PoolSeq_summary[,"DistAlongTransect"],PoolSeq_summary[,"p"],pch=23,col='black',bg=locusColour,lwd=1.5,cex=1.6)
}

clineCentreWidthPlotbasic <- function(clinesData,clinesDataFiltered,locusLG,locusPosition,locusColour,demeData,demeSpanOpt) {
  # negative cline widths coloured differently
  
  if (demeSpanOpt =="Opt1") {
     negativeClines <- clinesDataFiltered[which(clinesDataFiltered$width_Opt1<0),]
     positiveClines <- clinesDataFiltered[which(clinesDataFiltered$width_Opt1>=0),]
     columnWidth <- "width_Opt1"
     columnCentre <- "centre_Opt1"
     
     plot(positiveClines[positiveClines[,"poolsPass_Depth15"]>=5,"centre_Opt1"],positiveClines[positiveClines[,"poolsPass_Depth15"]>=5,"width_Opt1"],bg = "black",xaxs = "i", type='n',cex.main=1.3,bty="n",yaxt="n",xaxt="n", yaxs = "i",main="",xlim=c(0,28000),ylim=c(-11000,20000),xlab="",ylab="",cex.axis=1.2)
     axis(1,at=seq(0,26000,2000),labels=NA,col.axis="black",las=2,cex.axis=1.1,lwd=1.6,pos=-10000.01, tck = -.02)
     axis(1,at=seq(0,26000,4000),labels=seq(0,26000,4000)/1000,col.axis="black",tck=-0.3,las=1,cex.axis=1,lwd=0,pos=-10000.02)
     axis(2,at=seq(-10000,20000,2000),labels=NA,col.axis="black",las=2,cex.axis=1.1,lwd=1.6,pos=-0.01, tck = -.02)
     axis(2,at=seq(-10000,20000,4000),labels=seq(-10000,20000,4000)/1000,col.axis="black",tck=-0.3,las=1,cex.axis=1,lwd=0,pos=-0.02)
     lines(x = c(12580,12580), y = c(-10000,20000),lty=5,lwd=2,col='red')
     mtext("cline centre (km)",side=1,cex=1,line=1.5)
     mtext("cline width (km)",side=2,cex=1,line=2)
     points(positiveClines[positiveClines[,"poolsPass_Depth15"]>=5,"centre_Opt1"],positiveClines[positiveClines[,"poolsPass_Depth15"]>=5,"width_Opt1"],col='black')
  points(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5,"centre_Opt1"],negativeClines[negativeClines[,"poolsPass_Depth15"]>=5,"width_Opt1"],col='grey')
  # highlight locus of choice
  thisWidth <- allChr_Clines_chk[allChr_Clines_chk[,"position"]==locusPosition & allChr_Clines_chk[,"LG"]==locusLG,"width_Opt1"] 
  thisCentre <- allChr_Clines_chk[allChr_Clines_chk[,"position"]==locusPosition & allChr_Clines_chk[,"LG"]==locusLG,"centre_Opt1"] 
  cat(c(", width = "),thisWidth)
  cat(c(", centre = "),thisCentre)
  points(x=thisCentre,y=thisWidth,bg=locusColour,col=locusColour,pch=21,cex=1.5)
  }
  if (demeSpanOpt =="Opt2") {
     negativeClines <- clinesDataFiltered[which(clinesDataFiltered$width_Opt2<0),]
     positiveClines <- clinesDataFiltered[which(clinesDataFiltered$width_Opt2>=0),]
     columnWidth <- "width_Opt2"
     columnCentre <- "centre_Opt2"
     plot(positiveClines[positiveClines[,"poolsPass_Depth15"]>=5,"centre_Opt2"],positiveClines[positiveClines[,"poolsPass_Depth15"]>=5,"width_Opt2"],bg = "black",xaxs = "i", type='n',cex.main=1.3,bty="n",yaxt="n",xaxt="n", yaxs = "i",main="",xlim=c(0,28000),ylim=c(-11000,20000),xlab="",ylab="",cex.axis=1.2)
     axis(1,at=seq(0,26000,2000),labels=NA,col.axis="black",las=2,cex.axis=1.1,lwd=1.6,pos=-10000.01, tck = -.02)
     axis(1,at=seq(0,26000,4000),labels=seq(0,26000,4000)/1000,col.axis="black",tck=-0.3,las=1,cex.axis=1,lwd=0,pos=-10000.02)
     axis(2,at=seq(-10000,20000,2000),labels=NA,col.axis="black",las=2,cex.axis=1.1,lwd=1.6,pos=-0.01, tck = -.02)
     axis(2,at=seq(-10000,20000,4000),labels=seq(-10000,20000,4000)/1000,col.axis="black",tck=-0.3,las=1,cex.axis=1,lwd=0,pos=-0.02)
     lines(x = c(12580,12580), y = c(-10000,20000),lty=5,lwd=2,col='red')
     mtext("cline centre (km)",side=1,cex=1,line=2.5)
     mtext("cline width (km)",side=2,cex=1,line=2.5)
     points(positiveClines[positiveClines[,"poolsPass_Depth15"]>=5,"centre_Opt2"],positiveClines[positiveClines[,"poolsPass_Depth15"]>=5,"width_Opt2"],col='black')
  points(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5,"centre_Opt2"],negativeClines[negativeClines[,"poolsPass_Depth15"]>=5,"width_Opt2"],col='grey')
  # highlight locus of choice
  thisWidth <- allChr_Clines_chk[allChr_Clines_chk[,"position"]==locusPosition & allChr_Clines_chk[,"LG"]==locusLG,"width_Opt2"] 
  thisCentre <- allChr_Clines_chk[allChr_Clines_chk[,"position"]==locusPosition & allChr_Clines_chk[,"LG"]==locusLG,"centre_Opt2"] 
  cat(c(", width = "),thisWidth)
  cat(c(", centre = "),thisCentre)
  points(x=thisCentre,y=thisWidth,bg=locusColour,col=locusColour,pch=21,cex=1.5)
  }
   if (demeSpanOpt =="Opt3") {
     negativeClines <- clinesDataFiltered[which(clinesDataFiltered$width_Opt3<0),]
     positiveClines <- clinesDataFiltered[which(clinesDataFiltered$width_Opt3>=0),]
     columnWidth <- "width_Opt3"
     columnCentre <- "centre_Opt3"
     plot(positiveClines[positiveClines[,"poolsPass_Depth15"]>=5,"centre_Opt3"],positiveClines[positiveClines[,"poolsPass_Depth15"]>=5,"width_Opt3"],bg = "black",xaxs = "i", type='n',cex.main=1.3,bty="n",yaxt="n",xaxt="n", yaxs = "i",main="",xlim=c(0,28000),ylim=c(-11000,20000),xlab="",ylab="",cex.axis=1.2)
    axis(1,at=seq(0,26000,2000),labels=NA,col.axis="black",las=2,cex.axis=1.1,lwd=1.6,pos=-10000.01, tck = -.02)
     axis(1,at=seq(0,26000,4000),labels=seq(0,26000,4000)/1000,col.axis="black",tck=-0.3,las=1,cex.axis=1,lwd=0,pos=-10000.02)
     axis(2,at=seq(-10000,20000,2000),labels=NA,col.axis="black",las=2,cex.axis=1.1,lwd=1.6,pos=-0.01, tck = -.02)
     axis(2,at=seq(-10000,20000,4000),labels=seq(-10000,20000,4000)/1000,col.axis="black",tck=-0.3,las=1,cex.axis=1,lwd=0,pos=-0.02)
     lines(x = c(12580,12580), y = c(-10000,20000),lty=5,lwd=2,col='red')
     mtext("cline centre (km)",side=1,cex=1,line=2.5)
     mtext("cline width (km)",side=2,cex=1,line=2.5)
     points(positiveClines[positiveClines[,"poolsPass_Depth15"]>=5,"centre_Opt3"],positiveClines[positiveClines[,"poolsPass_Depth15"]>=5,"width_Opt3"],col='black')
  points(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5,"centre_Opt3"],negativeClines[negativeClines[,"poolsPass_Depth15"]>=5,"width_Opt3"],col='grey')
  # highlight locus of choice
  thisWidth <- allChr_Clines_chk[allChr_Clines_chk[,"position"]==locusPosition & allChr_Clines_chk[,"LG"]==locusLG,"width_Opt3"] 
  thisCentre <- allChr_Clines_chk[allChr_Clines_chk[,"position"]==locusPosition & allChr_Clines_chk[,"LG"]==locusLG,"centre_Opt3"] 
  cat(c(", width = "),thisWidth)
  cat(c(", centre = "),thisCentre)
  points(x=thisCentre,y=thisWidth,bg=locusColour,col=locusColour,pch=21,cex=1.5)
  }
  # all clines before any filtering for depth
  # plot(positiveClines[,"centre"],positiveClines[,"width"],xlim=c(2000,20000),ylim=c(-10000,15000),xlab="cline centre",ylab="cline width")
  # title("all clinal loci")
  # points(negativeClines[,"centre"],negativeClines[,"width"],col='grey')
  
   sum(positiveClines[,"poolsPass_Depth15"]>=5) # 10663 positive cline width have depth min 15 in 5 pools
  sum(negativeClines[,"poolsPass_Depth15"]>=5) # 249 negative cline width have depth min 15 in 5 pools
  sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5,"centre"]<8000) # 129 are located far left
  sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5,"centre"]>15000) # 103 are located far right
  
}
  
```

```{r Supporting Info demonstrative histograms of three deme span options together WG clines for ROS1}

# KASP SNPs Chr6	52889078	

allChr_Clines_chk_delta9 <- allChr_Clines_chk %>% filter(AFD_15_20>=0.9)
clinesDataFiltered_delta9 <- allChr_Clines_filter %>% filter(AFD_15_20>=0.9)
nrow(allChr_Clines_chk_delta9)
par(mfrow=c(3,2))
par(mar=c(3.5,3.5,3.5,2))
# ROS1 marker

demeSpanPlot(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=6,locusPosition=52889078,locusColour = 'red',demeData=HZ_50mDemes,demeSpanOpt="Opt1")
 clineCentreWidthPlotbasic(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=6,locusPosition=52889078,locusColour = 'red',demeData=HZ_50mDemes,demeSpanOpt="Opt1")
 
demeSpanPlot(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=6,locusPosition=52889078,locusColour = 'red',demeData=HZ_50mDemes,demeSpanOpt="Opt2")
 clineCentreWidthPlotbasic(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=6,locusPosition=52889078,locusColour = 'red',demeData=HZ_50mDemes,demeSpanOpt="Opt2")
 
demeSpanPlot(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=6,locusPosition=52889078,locusColour = 'red',demeData=HZ_50mDemes,demeSpanOpt="Opt3")
 clineCentreWidthPlotbasic(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=6,locusPosition=52889078,locusColour = 'red',demeData=HZ_50mDemes,demeSpanOpt="Opt3")
 
 # pdf version
pdf(paste0("DemeSpacingDemo_ROS1",".pdf"),width = 11, height = 12,pointsize=16)
par(mfrow=c(3,2))
par(mar=c(3.5,3.5,3.5,2))
# ELUTA marker
demeSpanPlot(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=6,locusPosition=52889078,locusColour = 'red',demeData=HZ_50mDemes,demeSpanOpt="Opt1")
 clineCentreWidthPlotbasic(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=6,locusPosition=52889078,locusColour = 'red',demeData=HZ_50mDemes,demeSpanOpt="Opt1")
demeSpanPlot(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=6,locusPosition=52889078,locusColour = 'red',demeData=HZ_50mDemes,demeSpanOpt="Opt2")
 clineCentreWidthPlotbasic(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=6,locusPosition=52889078,locusColour = 'red',demeData=HZ_50mDemes,demeSpanOpt="Opt2")
demeSpanPlot(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=6,locusPosition=52889078,locusColour = 'red',demeData=HZ_50mDemes,demeSpanOpt="Opt3")
 clineCentreWidthPlotbasic(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=6,locusPosition=52889078,locusColour = 'red',demeData=HZ_50mDemes,demeSpanOpt="Opt3")
dev.off()
```

```{r Supporting Info demonstrative histograms of three deme span options together WG clines for ELUTA}

allChr_Clines_chk_delta9 <- allChr_Clines_chk %>% filter(AFD_15_20>=0.9)
clinesDataFiltered_delta9 <- allChr_Clines_filter %>% filter(AFD_15_20>=0.9)

par(mfrow=c(3,2))
par(mar=c(3.5,3.5,3.5,2))
# ELUTA marker
demeSpanPlot(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=6,locusPosition=52988204,locusColour = 'pink',demeData=HZ_50mDemes,demeSpanOpt="Opt1")
 clineCentreWidthPlotbasic(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=6,locusPosition=52988204,locusColour = 'pink',demeData=HZ_50mDemes,demeSpanOpt="Opt1")
 
demeSpanPlot(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=6,locusPosition=52988204,locusColour = 'pink',demeData=HZ_50mDemes,demeSpanOpt="Opt2")
 clineCentreWidthPlotbasic(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=6,locusPosition=52988204,locusColour = 'pink',demeData=HZ_50mDemes,demeSpanOpt="Opt2")
 
demeSpanPlot(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=6,locusPosition=52988204,locusColour = 'pink',demeData=HZ_50mDemes,demeSpanOpt="Opt3")
 clineCentreWidthPlotbasic(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=6,locusPosition=52988204,locusColour = 'pink',demeData=HZ_50mDemes,demeSpanOpt="Opt3")
 
 # pdf version
pdf(paste0("DemeSpacingDemo_ELUTA",".pdf"),width = 11, height = 12,pointsize=16)
par(mfrow=c(3,2))
par(mar=c(3.5,3.5,3.5,2))
# ELUTA marker
demeSpanPlot(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=6,locusPosition=52988204,locusColour = 'pink',demeData=HZ_50mDemes,demeSpanOpt="Opt1")
 clineCentreWidthPlotbasic(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=6,locusPosition=52988204,locusColour = 'pink',demeData=HZ_50mDemes,demeSpanOpt="Opt1")
demeSpanPlot(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=6,locusPosition=52988204,locusColour = 'pink',demeData=HZ_50mDemes,demeSpanOpt="Opt2")
 clineCentreWidthPlotbasic(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=6,locusPosition=52988204,locusColour = 'pink',demeData=HZ_50mDemes,demeSpanOpt="Opt2")
demeSpanPlot(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=6,locusPosition=52988204,locusColour = 'pink',demeData=HZ_50mDemes,demeSpanOpt="Opt3")
 clineCentreWidthPlotbasic(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=6,locusPosition=52988204,locusColour = 'pink',demeData=HZ_50mDemes,demeSpanOpt="Opt3")
dev.off()
```

```{r Supporting Info demonstrative histograms of three deme span options together WG clines for FLAVIA}

allChr_Clines_chk_delta9 <- allChr_Clines_chk %>% filter(AFD_15_20>=0.9)
clinesDataFiltered_delta9 <- allChr_Clines_filter %>% filter(AFD_15_20>=0.9)

par(mfrow=c(3,2))
par(mar=c(3.5,3.5,3.5,2))
# FLAVIA downstream marker

allChr_Clines_chk[allChr_Clines_chk[,"ColourGene"] == "FLAVIA" & allChr_Clines_chk[,"centre"] <= 8000,"position"]
allChr_Clines_chk[allChr_Clines_chk[,"ColourGene"] == "FLAVIA" & allChr_Clines_chk[,"centre"] >= 16000,"position"]
allChr_Clines_chk[allChr_Clines_chk[,"position"]==53494658 & allChr_Clines_chk[,"LG"]==2,]
allChr_Clines_chk[allChr_Clines_chk[,"position"]==53494658 & allChr_Clines_chk[,"LG"]==2,]

# FLAVIA marker > 16000 cetnre
demeSpanPlot(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=2,locusPosition=53494658,locusColour = 'wheat',demeData=HZ_50mDemes,demeSpanOpt="Opt1")
 clineCentreWidthPlotbasic(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=2,locusPosition=53494658,locusColour = 'wheat',demeData=HZ_50mDemes,demeSpanOpt="Opt1")
demeSpanPlot(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=2,locusPosition=53494658,locusColour = 'wheat',demeData=HZ_50mDemes,demeSpanOpt="Opt2")
 clineCentreWidthPlotbasic(clinesData=allChr_Clines_chk,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=2,locusPosition=53494658,locusColour = 'wheat',demeData=HZ_50mDemes,demeSpanOpt="Opt2")
demeSpanPlot(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=2,locusPosition=53494658,locusColour = 'wheat',demeData=HZ_50mDemes,demeSpanOpt="Opt3")
 clineCentreWidthPlotbasic(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=2,locusPosition=53494658,locusColour = 'wheat',demeData=HZ_50mDemes,demeSpanOpt="Opt3")
 
 # pdf version
pdf(paste0("DemeSpacingDemo_FLAVIA",".pdf"),width = 11, height = 12,pointsize=16)
par(mfrow=c(3,2))
par(mar=c(3,4,3,2))
# FLAVIA marker > 16000 cetnre
demeSpanPlot(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=2,locusPosition=53494658,locusColour = 'wheat',demeData=HZ_50mDemes,demeSpanOpt="Opt1")
 clineCentreWidthPlotbasic(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=2,locusPosition=53494658,locusColour = 'wheat',demeData=HZ_50mDemes,demeSpanOpt="Opt1")
demeSpanPlot(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=2,locusPosition=53494658,locusColour = 'wheat',demeData=HZ_50mDemes,demeSpanOpt="Opt2")
 clineCentreWidthPlotbasic(clinesData=allChr_Clines_chk,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=2,locusPosition=53494658,locusColour = 'wheat',demeData=HZ_50mDemes,demeSpanOpt="Opt2")
demeSpanPlot(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=2,locusPosition=53494658,locusColour = 'wheat',demeData=HZ_50mDemes,demeSpanOpt="Opt3")
 clineCentreWidthPlotbasic(clinesData=allChr_Clines_chk_delta9,clinesDataFiltered=clinesDataFiltered_delta9,locusLG=2,locusPosition=53494658,locusColour = 'wheat',demeData=HZ_50mDemes,demeSpanOpt="Opt3")
 dev.off()
```

```{r Supporting Info pair-wise comparison of FastCline deme span options}

# deme original compared to mid point (opt 1)
#plot(allChr_Clines_chk[,"centre"],allChr_Clines_chk[,"centre_Opt1"])
#plot(allChr_Clines_chk[,"width"],allChr_Clines_chk[,"width_Opt1"])

#colnames(allChr_Clines_chk)
# table(allChr_Clines_chk[,"ColourGene"],allChr_Clines_chk[,"distanceToGene"])

allChr_Clines_chk_delta9 <- allChr_Clines_chk %>%
              filter(AFD_15_20 >=0.9)
#nrow(allChr_Clines_chk_delta9)
#nrow(allChr_Clines_chk)

pairwiseClineEst <- function(clineData,deltaP,Xval,Yval,Xlabel,Ylabel,colourGenesCoding,colourGenesLinked) {
  # clineData <- allChr_Clines_chk
  # Xval <- "width_Opt1"; Yval <- "width_Opt2" ; Xlabel<- "cline width - Opt1"; Ylabel<- "cline width - Opt2"
  # colourGenesCoding <- T; colourGenesLinked <- T
  clineData <- clineData %>%
              filter(AFD_15_20 >=deltaP)
  plot(clineData[,Xval]/1000,clineData[,Yval]/1000,cex.axis=1.3,cex.lab=1.3, xlim = c(0,20000/1000),   ylim=c(0,20000/1000),type='n',xlab=Xlabel,ylab=Ylabel)
      points(x=clineData[,Xval]/1000,y=clineData[,Yval]/1000,pch=19,col='black')
      lines(x = c(0,20000/1000), y = c(0,20000/1000),lty=5,lwd=2,col='darkgrey')

    if (colourGenesLinked==T) {
    clineData_EL <- clineData %>%
      filter(ColourGene=="ELUTA" & distanceToGene %in% c("up30_100kb","up30_400kb","up30kb","down30kb","down30_400kb","down30_100kb"))
    if (nrow(clineData_EL)>0) {
      points(x=clineData_EL[,Xval]/1000,y=clineData_EL[,Yval]/1000,pch=1,col='pink')
    }
    clineData_ROS1 <- clineData %>%
                    filter(ColourGene=="ROS1" & distanceToGene %in% c("up30_100kb","up30_400kb","up30kb","down30kb","down30_400kb","down30_100kb"))
    if (nrow(clineData_ROS1)>0) {
      points(x=clineData_ROS1[,Xval]/1000,y=clineData_ROS1[,Yval]/1000,pch=1,col='red')
    }
    clineData_ROS2 <- clineData %>%
                    filter(ColourGene=="ROS2" & distanceToGene %in% c("up30_100kb","up30_400kb","up30kb","down30kb","down30_400kb","down30_100kb"))
    if (nrow(clineData_ROS2)>0) {
      points(x=clineData_ROS2[,Xval]/1000,y=clineData_ROS2[,Yval]/1000,pch=1,col='red')
    }
    clineData_RUBIA <- clineData %>%
                    filter(ColourGene=="RUBIA" & distanceToGene %in% c("up30_100kb","up30_400kb","up30kb","down30kb","down30_400kb","down30_100kb"))
    if (nrow(clineData_RUBIA)>0) {
      points(x=clineData_RUBIA[,Xval]/1000,y=clineData_RUBIA[,Yval]/1000,pch=1,col='red')
    }
    clineData_RUBIA <- clineData %>%
                    filter(ColourGene=="RUBIA" & distanceToGene %in% c("up30_100kb","up30_400kb","up30kb","down30kb","down30_400kb","down30_100kb"))
    if (nrow(clineData_RUBIA)>0) {
      points(x=clineData_RUBIA[,Xval]/1000,y=clineData_RUBIA[,Yval]/1000,pch=1,col='blue')
    }
    clineData_SULF <- clineData %>%
                    filter(ColourGene=="SULF" & distanceToGene %in% c("up30_100kb","up30_400kb","up30kb","down30kb","down30_400kb","down30_100kb"))
    if (nrow(clineData_SULF)>0) {
      points(x=clineData_SULF[,Xval]/1000,y=clineData_SULF[,Yval]/1000,pch=1,col='orange')
    }
    clineData_CREMOSA <- clineData %>%
                    filter(ColourGene=="CREMOSA" & distanceToGene %in% c("up30_100kb","up30_400kb","up30kb","down30kb","down30_400kb","down30_100kb"))
    if (nrow(clineData_CREMOSA)>0) {
      points(x=clineData_CREMOSA[,Xval]/1000,y=clineData_CREMOSA[,Yval]/1000,pch=1,col='green')
    }
    clineData_AURINA <- clineData %>%
                    filter(ColourGene=="AURINA" & distanceToGene %in% c("up30_100kb","up30_400kb","up30kb","down30kb","down30_400kb","down30_100kb"))
    if (nrow(clineData_AURINA)>0) {
      points(x=clineData_AURINA[,Xval]/1000,y=clineData_AURINA[,Yval]/1000,pch=1,col='wheat')
    }
    clineData_FLAVIA <- clineData %>%
                    filter(ColourGene=="FLAVIA" & distanceToGene %in% c("up30_100kb","up30_400kb","up30kb","down30kb","down30_400kb","down30_100kb"))
    if (nrow(clineData_FLAVIA)>0) {
      points(x=clineData_FLAVIA[,Xval]/1000,y=clineData_FLAVIA[,Yval]/1000,pch=1,col='yellow')
    }
    }
  
    if (colourGenesCoding==T) {
    clineData_EL <- clineData %>%
                    filter(ColourGene=="ELUTA" & distanceToGene =="inGene")
    if (nrow(clineData_EL)>0) {
      points(x=clineData_EL[,Xval]/1000,y=clineData_EL[,Yval]/1000,pch=19,col='pink')
    }
    clineData_ROS1 <- clineData %>%
                    filter(ColourGene=="ROS1" & distanceToGene =="inGene")
    if (nrow(clineData_ROS1)>0) {
      points(x=clineData_ROS1[,Xval]/1000,y=clineData_ROS1[,Yval]/1000,pch=19,col='red')
    }
    clineData_ROS2 <- clineData %>%
                    filter(ColourGene=="ROS2" & distanceToGene =="inGene")
    if (nrow(clineData_ROS2)>0) {
      points(x=clineData_ROS2[,Xval]/1000,y=clineData_ROS2[,Yval]/1000,pch=19,col='red')
    }
    clineData_RUBIA <- clineData %>%
                    filter(ColourGene=="RUBIA" & distanceToGene =="inGene")
    if (nrow(clineData_RUBIA)>0) {
      points(x=clineData_RUBIA[,Xval]/1000,y=clineData_RUBIA[,Yval]/1000,pch=19,col='red')
    }
    clineData_RUBIA <- clineData %>%
                    filter(ColourGene=="RUBIA" & distanceToGene =="inGene")
    if (nrow(clineData_RUBIA)>0) {
      points(x=clineData_RUBIA[,Xval]/1000,y=clineData_RUBIA[,Yval]/1000,pch=19,col='blue')
    }
    clineData_SULF <- clineData %>%
                    filter(ColourGene=="SULF" & distanceToGene =="inGene")
    if (nrow(clineData_SULF)>0) {
      points(x=clineData_SULF[,Xval]/1000,y=clineData_SULF[,Yval]/1000,pch=19,col='orange')
    }
    clineData_CREMOSA <- clineData %>%
                    filter(ColourGene=="CREMOSA" & distanceToGene =="inGene")
    if (nrow(clineData_CREMOSA)>0) {
      points(x=clineData_CREMOSA[,Xval]/1000,y=clineData_CREMOSA[,Yval]/1000,pch=19,col='green')
    }
    clineData_AURINA <- clineData %>%
                    filter(ColourGene=="AURINA" & distanceToGene =="inGene")
    if (nrow(clineData_AURINA)>0) {
      points(x=clineData_AURINA[,Xval]/1000,y=clineData_AURINA[,Yval]/1000,pch=19,col='wheat')
    }
    clineData_FLAVIA <- clineData %>%
                    filter(ColourGene=="FLAVIA" & distanceToGene =="inGene")
    if (nrow(clineData_FLAVIA)>0) {
      points(x=clineData_FLAVIA[,Xval]/1000,y=clineData_FLAVIA[,Yval]/1000,pch=19,col='yellow')
    }
  }
}

par(mfrow=c(3,2))
par(mar=c(4,4,4,4))

pairwiseClineEst(allChr_Clines_chk,0.9,"centre_Opt1","centre_Opt2","cline centre - DS1","cline centre - DS2",T,F)
pairwiseClineEst(allChr_Clines_chk,0.9,"width_Opt1","width_Opt2","cline width - DS1","cline width - DS2",T,F)
pairwiseClineEst(allChr_Clines_chk,0.9,"centre_Opt1","centre_Opt3","cline centre - DS1","cline centre - DS3",T,F)
pairwiseClineEst(allChr_Clines_chk,0.9,"width_Opt1","width_Opt3","cline width - DS1","cline width - DS3",T,F)
pairwiseClineEst(allChr_Clines_chk,0.9,"centre_Opt2","centre_Opt3","cline centre - DS2","cline centre - DS3",T,F)
pairwiseClineEst(allChr_Clines_chk,0.9,"width_Opt2","width_Opt3","cline width - DS2","cline width - DS3",T,F)

# pdf version
pdf(paste0("DemeSpacingPairwise",".pdf"),width = 9, height = 13,pointsize=1)
par(mfrow=c(3,2))
par(mar=c(4,4,4,4))

pairwiseClineEst(allChr_Clines_chk,0.9,"centre_Opt1","centre_Opt2","cline centre - DS1","cline centre - DS2",T,F)
pairwiseClineEst(allChr_Clines_chk,0.9,"width_Opt1","width_Opt2","cline width - DS1","cline width - DS2",T,F)
pairwiseClineEst(allChr_Clines_chk,0.9,"centre_Opt1","centre_Opt3","cline centre - DS1","cline centre - DS3",T,F)
pairwiseClineEst(allChr_Clines_chk,0.9,"width_Opt1","width_Opt3","cline width - DS1","cline width - DS3",T,F)
pairwiseClineEst(allChr_Clines_chk,0.9,"centre_Opt2","centre_Opt3","cline centre - DS2","cline centre - DS3",T,F)
pairwiseClineEst(allChr_Clines_chk,0.9,"width_Opt2","width_Opt3","cline width - DS2","cline width - DS3",T,F)
dev.off()

```

```{r Supporting Info basic filtering stats of clines}

# Initially before any post FastClines filtering n = 12,936 (i.e. depth x = 10 in at least 5 pools, deltaP>0.8)

allChr_Clines_chk_delta9 <- allChr_Clines_chk %>% filter(AFD_15_20>=0.9)
clinesDataFiltered_delta9 <- allChr_Clines_chk %>% filter(AFD_15_20>=0.9)

# shows that many loci failed filtering
depth15table <- table(allChr_Clines_chk[,"poolsPass_Depth15"]) # sum(depth15table[6:7])


sum(allChr_Clines_chk[,"poolsPass_Depth15"]>4)
sum(allChr_Clines_chk[,"poolsPass_Depth20"]>4)

depth20table <- table(allChr_Clines_chk[,"poolsPass_Depth20"]) # sum(depth20table[6:7])
nrow(allChr_Clines_chk_delta9)

```

Using the Deme Span 1 option, initial analysis using FastClines found n = 12,936 clinal loci with a low filtering threshold (min depth = 10). However, this was reduced to 10,912 and 7,271 loci when increasing filtering to a minimum of 15 and 20 depth, respectively (in at least 5 of 6 pools) (Fig Sx). 

Considering only loci with at least 15 depth at 5 or more pools, we found n = 249 loci of 10,912 (2.3%) exhibit negative cline widths. Of these negative cline widths, n = 232 (93.2%) were centred to the extreme left or right of the transect. These clines likely represent artefacts due to the limited number of pools which draws widths downward as centres move to the edges. The remaining n = 16 negative cline widths positioned near the hybrid zone centre tended to display allele frequency reversals in the outer first pool (Fig Sx). 

```{r Supporting Info basic filtering stats & main filtering of clines}
# Initially before any post FastClines filtering n = 12,936 (i.e. depth x = 10 in at least 5 pools, deltaP>0.8)
nrow(allChr_Clines_chk) #12936

allChr_Clines_filterDpth15 <- allChr_Clines_chk[allChr_Clines_chk[,"poolsPass_Depth15"]>=5,]
nrow(allChr_Clines_filterDpth15) #10912

allChr_Clines_filterDpth20 <- allChr_Clines_chk[allChr_Clines_chk[,"poolsPass_Depth20"]>=5,]
nrow(allChr_Clines_filterDpth20) #7271

```

DeltaP has a greater influence on the number of negative cline widths.

Note greek letter function (for delta) now not working. May need to re-install package

```{r fastClines and negative cline widths by depth and deltaP, echo=FALSE}
# plot centre by width
range(allChr_Clines_chk[,"centre"])
range(allChr_Clines_chk[,"width"])
par(mfrow=c(3,2))
par(mar=c(3.8,3.8,3.8,2.5))

# (a) all loci deltaP > 0.8
# negative cline widths coloured differently

plot(allChr_Clines_chk$width_Original,allChr_Clines_chk$width)

# from FastClines script
negativeClines <- allChr_Clines_chk[which(allChr_Clines_chk$width<0),]
positiveClines <- allChr_Clines_chk[which(allChr_Clines_chk$width>=0),]

# new adjustment 2025
negativeClines <- allChr_Clines_chk[which(allChr_Clines_chk$width_Original<0),]
positiveClines <- allChr_Clines_chk[which(allChr_Clines_chk$width_Original>=0),]

# ~290 in old script, with adjument 23 negative cline widths

# still negative cline widths with pmin adjustment 2025
# recalculate some cline widths
negativeClines[,]
this_p <- negativeClines[1,9:14]

# original
clineWidthOLD <- function(p,deltaPthresh,demeOrder,demeSpans) {
  # deltaPthresh <- 0.8
  # p <- this_p
  # demeSpans <- demeSpacing_Opt1
  p <- as.vector(unlist(p))
  # reordering demes based on position along transect (if required)
  p <- p[demeOrder]
  p0 <- p[1]
  p1 <- p[length(p)]
  deltaP <- abs(p0-p1)
  d_i = demeSpans 
  # reorder the spans (not required here)
  # d_i <- d_i[demeOrder]
  if (deltaP < deltaPthresh) {
    return (-9)
  }
  if (deltaP >= deltaPthresh) {
    numerWidth = (p-rep(p0,length(p)))*(rep(p1,length(p))-p)
    denomWidth = rep((p1-p0)^2,length(p))
    width_output = 4*(sum((numerWidth/denomWidth)*rep(d_i)))
    return(width_output)
  }
}
clineWidthOLD(this_p,0.8,c(1,3,2,4,5,6),demeSpacing_Original)

# pmin adjustment Feb 2025
clineWidthNew <- function(p,deltaPthresh,demeOrder,demeSpans) {
  # deltaPthresh <- 0.8
  # p <- this_p
  # demeSpans <- demeSpacing_Opt1
  p <- as.vector(unlist(p))
  # reordering demes based on position along transect (if required)
  p <- p[demeOrder]
  p0 <- p[1]
  p1 <- p[length(p)]
  # pmin adjustment 2025
  p0 <- min(p[1:3])
  p1 <- max(p[(length(p)-2):length(p)])
  deltaP <- abs(p0-p1)
  d_i = demeSpans 
  # reorder the spans (not required here)
  # d_i <- d_i[demeOrder]
  if (deltaP < deltaPthresh) {
    return (-9)
  }
  if (deltaP >= deltaPthresh) {
    numerWidth = (p-rep(p0,length(p)))*(rep(p1,length(p))-p)
    denomWidth = rep((p1-p0)^2,length(p))
    width_output = 4*(sum((numerWidth/denomWidth)*rep(d_i)))
    return(width_output)
  }
}
clineWidthNew(this_p,0.8,c(1,3,2,4,5,6),demeSpacing_Original)

pdf("clineWidthCentre_NegativeClinesfiltering_adjustedP.pdf",width = 9, height = 12,pointsize=16)
par(mfrow=c(3,2))
par(mar=c(3.8,3.8,3.8,2.5))


# adjusted p values
# (a) all loci deltaP > 0.8
# negative cline widths coloured differently
negativeClines <- allChr_Clines_chk[which(allChr_Clines_chk$width_Original<0),]
positiveClines <- allChr_Clines_chk[which(allChr_Clines_chk$width_Original>=0),]
# all clines before any filtering for depth
plot(positiveClines[,"centre_Original"],positiveClines[,"width_Original"],xlim=c(2000,20000),ylim=c(-10000,15000),xlab="",ylab="")
title(paste0("adjusted p, depth > 10, ","AFD > 0.8"))
#mtext("cline centre (km)",side=1,cex=0.8,line=1.9)
mtext("cline width (km)",side=2,cex=0.9,line=2)
lines(x = c(12580,12580), y = c(-10000,20000),lty=5,lwd=2,col='red')
points(negativeClines[,"centre_Original"],negativeClines[,"width_Original"],col='grey')

# (b) all loci deltaP > 0.9

plot(positiveClines[positiveClines[,"AFD_15_20"]>=0.9,"centre_Original"],positiveClines[positiveClines[,"AFD_15_20"]>=0.9,"width_Original"],xlim=c(2000,20000),ylim=c(-10000,15000),xlab="",ylab="")
title(paste0("adjusted p, depth > 10, ","AFD > 0.9"))
lines(x = c(12580,12580), y = c(-10000,20000),lty=5,lwd=2,col='red')
points(negativeClines[negativeClines[,"AFD_15_20"]>=0.9,"centre_Original"],negativeClines[negativeClines[,"AFD_15_20"]>=0.9,"width_Original"],col='grey')

sum(positiveClines[,"AFD_15_20"]>=0.9) # 4637 positive cline width have depth min 15 in 5 pools
sum(negativeClines[,"AFD_15_20"]>=0.9) # 2 negative cline width have depth min 15 in 5 pools
sum(negativeClines[negativeClines[,"AFD_15_20"]>=0.9,"centre"]<8000) # 0 are located far left
sum(negativeClines[negativeClines[,"AFD_15_20"]>=0.9,"centre"]>15000) # 0 are located far right


# (c) min depth 15 in at least 5 of 6 pools & deltaP > 0.8
plot(positiveClines[positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.8 ,"centre_Original"],positiveClines[positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.8,"width_Original"],xlim=c(2000,20000),ylim=c(-10000,15000),xlab="",ylab="")
title(paste0("depth > 15, ","AFD > 0.8"))
lines(x = c(12580,12580), y = c(-10000,20000),lty=5,lwd=2,col='red')
points(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre_Original"],negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"width_Original"],col='grey')
#mtext("cline centre (km)",side=1,cex=0.8,line=1.9)
mtext("cline width (km)",side=2,cex=0.9,line=2)

sum(positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.8) # 10663 positive cline width have depth min 15 in 5 pools
sum(negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8) # 249 negative cline width have depth min 15 in 5 pools
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre"]<8000) # 129 are located far left
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre"]>15000) # 103 are located far right

# (d) min depth 15 in at least 5 of 6 pools & deltaP > 0.9
plot(positiveClines[positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.9 ,"centre_Original"],positiveClines[positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.9,"width_Original"],xlim=c(2000,20000),ylim=c(-10000,15000),xlab="",ylab="")
title(paste0("depth > 15, ","AFD > 0.9"))
lines(x = c(12580,12580), y = c(-10000,20000),lty=5,lwd=2,col='red')
points(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre_Original"],negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"width_Original"],col='grey')

sum(positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.9) # 10663 positive cline width have depth min 15 in 5 pools
sum(negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9) # 249 negative cline width have depth min 15 in 5 pools
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre"]<8000) # 129 are located far left
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre"]>15000) # 103 are located far right


# (e) min depth 20 in at least 5 of 6 pools & deltaP > 0.8
plot(positiveClines[positiveClines[,"poolsPass_Depth20"]>=5 & positiveClines[,"AFD_15_20"]>=0.8 ,"centre_Original"],positiveClines[positiveClines[,"poolsPass_Depth20"]>=5 & positiveClines[,"AFD_15_20"]>=0.8,"width_Original"],xlim=c(2000,20000),ylim=c(-10000,15000),xlab="",ylab="")
title(paste0("depth > 20, ","AFD > 0.8"))
lines(x = c(12580,12580), y = c(-10000,20000),lty=5,lwd=2,col='red')
points(negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre_Original"],negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"width_Original"],col='grey')
mtext("cline centre (km)",side=1,cex=0.9,line=2)
mtext("cline width (km)",side=2,cex=0.9,line=2)
sum(positiveClines[,"poolsPass_Depth20"]>=5 & positiveClines[,"AFD_15_20"]>=0.8) # 10663 positive cline width have depth min 15 in 5 pools
sum(negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.8) # 249 negative cline width have depth min 15 in 5 pools
sum(negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre"]<8000) # 129 are located far left
sum(negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre"]>15000) # 103 are located far right

# (f) min depth 15 in at least 5 of 6 pools & deltaP > 0.9
plot(positiveClines[positiveClines[,"poolsPass_Depth20"]>=5 & positiveClines[,"AFD_15_20"]>=0.9 ,"centre"],positiveClines[positiveClines[,"poolsPass_Depth20"]>=5 & positiveClines[,"AFD_15_20"]>=0.9,"width_Original"],xlim=c(2000,20000),ylim=c(-10000,15000),xlab="",ylab="")
title(paste0("depth > 20, ","AFD > 0.9"))
lines(x = c(12580,12580), y = c(-10000,20000),lty=5,lwd=2,col='red')
points(negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre_Original"],negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"width_Original"],col='grey')
mtext("cline centre (km)",side=1,cex=0.9,line=2)
#mtext("cline width (km)",side=2,cex=0.9,line=2)
sum(positiveClines[,"poolsPass_Depth20"]>=5 & positiveClines[,"AFD_15_20"]>=0.9) # 10663 positive cline width have depth min 15 in 5 pools
sum(negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.9) # 249 negative cline width have depth min 15 in 5 pools
sum(negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre"]<8000) # 129 are located far left
sum(negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre"]>15000) # 103 are located far right

dev.off()

# Non- adjusted
# pdf version
pdf("clineWidthCentre_NegativeClinesfiltering.pdf",width = 9, height = 12,pointsize=16)
par(mfrow=c(3,2))
par(mar=c(3.8,3.8,3.8,2.5))

# (a) all loci deltaP > 0.8
# negative cline widths coloured differently
negativeClines <- allChr_Clines_chk[which(allChr_Clines_chk$width<0),]
positiveClines <- allChr_Clines_chk[which(allChr_Clines_chk$width>=0),]
# all clines before any filtering for depth
plot(positiveClines[,"centre"],positiveClines[,"width"],xlim=c(2000,20000),ylim=c(-10000,15000),xlab="",ylab="")
title(paste0("depth > 10, ","AFD > 0.8"))
#mtext("cline centre (km)",side=1,cex=0.8,line=1.9)
mtext("cline width (km)",side=2,cex=0.9,line=2)
lines(x = c(12580,12580), y = c(-10000,20000),lty=5,lwd=2,col='red')
points(negativeClines[,"centre"],negativeClines[,"width"],col='grey')

# (b) all loci deltaP > 0.9

plot(positiveClines[positiveClines[,"AFD_15_20"]>=0.9,"centre"],positiveClines[positiveClines[,"AFD_15_20"]>=0.9,"width"],xlim=c(2000,20000),ylim=c(-10000,15000),xlab="",ylab="")
title(paste0("depth > 10, ","AFD > 0.9"))
lines(x = c(12580,12580), y = c(-10000,20000),lty=5,lwd=2,col='red')
points(negativeClines[negativeClines[,"AFD_15_20"]>=0.9,"centre"],negativeClines[negativeClines[,"AFD_15_20"]>=0.9,"width"],col='grey')

sum(positiveClines[,"AFD_15_20"]>=0.9) # 10663 positive cline width have depth min 15 in 5 pools
sum(negativeClines[,"AFD_15_20"]>=0.9) # 249 negative cline width have depth min 15 in 5 pools
sum(negativeClines[negativeClines[,"AFD_15_20"]>=0.9,"centre"]<8000) # 129 are located far left
sum(negativeClines[negativeClines[,"AFD_15_20"]>=0.9,"centre"]>15000) # 103 are located far right


# (c) min depth 15 in at least 5 of 6 pools & deltaP > 0.8
plot(positiveClines[positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.8 ,"centre"],positiveClines[positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.8,"width"],xlim=c(2000,20000),ylim=c(-10000,15000),xlab="",ylab="")
title(paste0("depth > 15, ","AFD > 0.8"))
lines(x = c(12580,12580), y = c(-10000,20000),lty=5,lwd=2,col='red')
points(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre"],negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"width"],col='grey')
#mtext("cline centre (km)",side=1,cex=0.8,line=1.9)
mtext("cline width (km)",side=2,cex=0.9,line=2)

sum(positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.8) # 10663 positive cline width have depth min 15 in 5 pools
sum(negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8) # 249 negative cline width have depth min 15 in 5 pools
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre"]<8000) # 129 are located far left
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre"]>15000) # 103 are located far right

# (d) min depth 15 in at least 5 of 6 pools & deltaP > 0.9
plot(positiveClines[positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.9 ,"centre"],positiveClines[positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.9,"width"],xlim=c(2000,20000),ylim=c(-10000,15000),xlab="",ylab="")
title(paste0("depth > 15, ","AFD > 0.9"))
lines(x = c(12580,12580), y = c(-10000,20000),lty=5,lwd=2,col='red')
points(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre"],negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"width"],col='grey')

sum(positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.9) # 10663 positive cline width have depth min 15 in 5 pools
sum(negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9) # 249 negative cline width have depth min 15 in 5 pools
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre"]<8000) # 129 are located far left
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre"]>15000) # 103 are located far right


# (e) min depth 20 in at least 5 of 6 pools & deltaP > 0.8
plot(positiveClines[positiveClines[,"poolsPass_Depth20"]>=5 & positiveClines[,"AFD_15_20"]>=0.8 ,"centre"],positiveClines[positiveClines[,"poolsPass_Depth20"]>=5 & positiveClines[,"AFD_15_20"]>=0.8,"width"],xlim=c(2000,20000),ylim=c(-10000,15000),xlab="",ylab="")
title(paste0("depth > 20, ","AFD > 0.8"))
lines(x = c(12580,12580), y = c(-10000,20000),lty=5,lwd=2,col='red')
points(negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre"],negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"width"],col='grey')
mtext("cline centre (km)",side=1,cex=0.9,line=2)
mtext("cline width (km)",side=2,cex=0.9,line=2)
sum(positiveClines[,"poolsPass_Depth20"]>=5 & positiveClines[,"AFD_15_20"]>=0.8) # 10663 positive cline width have depth min 15 in 5 pools
sum(negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.8) # 249 negative cline width have depth min 15 in 5 pools
sum(negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre"]<8000) # 129 are located far left
sum(negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre"]>15000) # 103 are located far right

# (f) min depth 15 in at least 5 of 6 pools & deltaP > 0.9
plot(positiveClines[positiveClines[,"poolsPass_Depth20"]>=5 & positiveClines[,"AFD_15_20"]>=0.9 ,"centre"],positiveClines[positiveClines[,"poolsPass_Depth20"]>=5 & positiveClines[,"AFD_15_20"]>=0.9,"width"],xlim=c(2000,20000),ylim=c(-10000,15000),xlab="",ylab="")
title(paste0("depth > 20, ","AFD > 0.9"))
lines(x = c(12580,12580), y = c(-10000,20000),lty=5,lwd=2,col='red')
points(negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre"],negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"width"],col='grey')
mtext("cline centre (km)",side=1,cex=0.9,line=2)
#mtext("cline width (km)",side=2,cex=0.9,line=2)
sum(positiveClines[,"poolsPass_Depth20"]>=5 & positiveClines[,"AFD_15_20"]>=0.9) # 10663 positive cline width have depth min 15 in 5 pools
sum(negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.9) # 249 negative cline width have depth min 15 in 5 pools
sum(negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre"]<8000) # 129 are located far left
sum(negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre"]>15000) # 103 are located far right
dev.off()
```

```{r fastClines and negative cline widths by deme span method, echo=FALSE}

# plot centre by width
range(allChr_Clines_chk[,"centre"])
range(allChr_Clines_chk[,"width"])
par(mfrow=c(3,2))
par(mar=c(3.8,3.8,3.8,2.5))

# (a) DS 1, min depth 15 in at least 5 of 6 pools & deltaP > 0.8
negativeClines <- allChr_Clines_chk[which(allChr_Clines_chk$width_Opt1<0),]
positiveClines <- allChr_Clines_chk[which(allChr_Clines_chk$width_Opt1>=0),]

# original
clineWidthOLD(this_p,0.8,c(1,3,2,4,5,6),demeSpacing_Original)

# pmin adjustment Feb 2025
clineWidthNew(this_p,0.8,c(1,3,2,4,5,6),demeSpacing_Original)

plot(positiveClines[positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.8 ,"centre"],positiveClines[positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.8,"width"],xlim=c(2000,20000),ylim=c(-10000,20000),xlab="",ylab="")
title(paste0("DS1, ",paste0(greeks("Delta"),"p"," > 0.8")))
lines(x = c(12580,12580), y = c(-10000,20000),lty=5,lwd=2,col='red')
points(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre"],negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"width"],col='grey')
#mtext("cline centre (km)",side=1,cex=0.8,line=1.9)
mtext("cline width (km)",side=2,cex=0.9,line=2)

sum(positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.8) # 10663 positive cline width have depth min 15 in 5 pools
sum(negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8) # 249 negative cline width have depth min 15 in 5 pools
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre"]<8000) # 129 are located far left
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre"]>15000) # 103 are located far right

# (b) DS 1, min depth 15 in at least 5 of 6 pools & deltaP > 0.9
plot(positiveClines[positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.9 ,"centre"],positiveClines[positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.9,"width"],xlim=c(2000,20000),ylim=c(-10000,20000),xlab="",ylab="")
title(paste0("DS1, ",paste0(greeks("Delta"),"p"," > 0.9")))
lines(x = c(12580,12580), y = c(-10000,20000),lty=5,lwd=2,col='red')
points(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre"],negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"width"],col='grey')

sum(positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.9) # 3832 positive cline width have depth min 15 in 5 pools
sum(negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9) # 10 negative cline width have depth min 15 in 5 pools
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre"]<8000) # 129 are located far left
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre"]>15000) # 103 are located far right

# (c) DS 2, min depth 15 in at least 5 of 6 pools & deltaP > 0.8
negativeClines <- allChr_Clines_chk[which(allChr_Clines_chk$width_Opt2<0),]
positiveClines <- allChr_Clines_chk[which(allChr_Clines_chk$width_Opt2>=0),]

plot(positiveClines[positiveClines[,"poolsPass_Depth20"]>=5 & positiveClines[,"AFD_15_20"]>=0.8 ,"centre_Opt2"],positiveClines[positiveClines[,"poolsPass_Depth20"]>=5 & positiveClines[,"AFD_15_20"]>=0.8,"width_Opt2"],xlim=c(2000,20000),ylim=c(-10000,20000),xlab="",ylab="")
title(paste0("DS2, ",paste0(greeks("Delta"),"p"," > 0.8")))
lines(x = c(12580,12580), y = c(-10000,20000),lty=5,lwd=2,col='red')
points(negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre_Opt2"],negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"width_Opt2"],col='grey')
#mtext("cline centre (km)",side=1,cex=0.9,line=2)
mtext("cline width (km)",side=2,cex=0.9,line=2)
sum(positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.8) # 10759 positive cline width have depth min 15 in 5 pools
sum(negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8) # 153 negative cline width have depth min 15 in 5 pools
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre"]<8000) # 129 are located far left
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre"]>15000) # 103 are located far right

# (d) DS 2, min depth 15 in at least 5 of 6 pools & deltaP > 0.9
plot(positiveClines[positiveClines[,"poolsPass_Depth20"]>=5 & positiveClines[,"AFD_15_20"]>=0.9 ,"centre_Opt2"],positiveClines[positiveClines[,"poolsPass_Depth20"]>=5 & positiveClines[,"AFD_15_20"]>=0.9,"width_Opt2"],xlim=c(2000,20000),ylim=c(-10000,20000),xlab="",ylab="")
title(paste0("DS2, ",paste0(greeks("Delta"),"p"," > 0.9")))
lines(x = c(12580,12580), y = c(-10000,20000),lty=5,lwd=2,col='red')
points(negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre_Opt2"],negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"width_Opt2"],col='grey')
#mtext("cline centre (km)",side=1,cex=0.9,line=2)
#mtext("cline width (km)",side=2,cex=0.9,line=2)
sum(positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.9) # 3832 positive cline width have depth min 15 in 5 pools
sum(negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9) # 10 negative cline width have depth min 15 in 5 pools
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre"]<8000) # 129 are located far left
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre"]>15000) # 103 are located far right

# (e) DS 3, min depth 15 in at least 5 of 6 pools & deltaP > 0.8
negativeClines <- allChr_Clines_chk[which(allChr_Clines_chk$width_Opt3<0),]
positiveClines <- allChr_Clines_chk[which(allChr_Clines_chk$width_Opt3>=0),]

plot(positiveClines[positiveClines[,"poolsPass_Depth20"]>=5 & positiveClines[,"AFD_15_20"]>=0.8 ,"centre_Opt3"],positiveClines[positiveClines[,"poolsPass_Depth20"]>=5 & positiveClines[,"AFD_15_20"]>=0.8,"width_Opt3"],xlim=c(2000,20000),ylim=c(-10000,20000),xlab="",ylab="")
title(paste0("DS3, ",paste0(greeks("Delta"),"p"," > 0.8")))
lines(x = c(12580,12580), y = c(-10000,20000),lty=5,lwd=2,col='red')
points(negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre_Opt3"],negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"width_Opt3"],col='grey')
#mtext("cline centre (km)",side=1,cex=0.9,line=2)
mtext("cline width (km)",side=2,cex=0.9,line=2)
sum(positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.8) # 10759 positive cline width have depth min 15 in 5 pools
sum(negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8) # 153 negative cline width have depth min 15 in 5 pools
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre"]<8000) # 129 are located far left
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre"]>15000) # 103 are located far right

# (f) DS 3, min depth 15 in at least 5 of 6 pools & deltaP > 0.9
plot(positiveClines[positiveClines[,"poolsPass_Depth20"]>=5 & positiveClines[,"AFD_15_20"]>=0.9 ,"centre_Opt3"],positiveClines[positiveClines[,"poolsPass_Depth20"]>=5 & positiveClines[,"AFD_15_20"]>=0.9,"width_Opt3"],xlim=c(2000,20000),ylim=c(-10000,20000),xlab="",ylab="")
title(paste0("DS3, ",paste0(greeks("Delta"),"p"," > 0.9")))
lines(x = c(12580,12580), y = c(-10000,20000),lty=5,lwd=2,col='red')
points(negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre_Opt3"],negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"width_Opt3"],col='grey')
#mtext("cline centre (km)",side=1,cex=0.9,line=2)
#mtext("cline width (km)",side=2,cex=0.9,line=2)
sum(positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.9) # 3832 positive cline width have depth min 15 in 5 pools
sum(negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9) # 10 negative cline width have depth min 15 in 5 pools
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre"]<8000) # 129 are located far left
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre"]>15000) # 103 are located far right

# pdf version
pdf("clineWidthCentre_NegativeClinesByDemeSpanfiltering.pdf",width = 9, height = 12,pointsize=16)
par(mfrow=c(3,2))
par(mar=c(3.8,3.8,3.8,2.5))
# (a) DS 1, min depth 15 in at least 5 of 6 pools & deltaP > 0.8
negativeClines <- allChr_Clines_filter[which(allChr_Clines_filter$width_Opt1<0),]
positiveClines <- allChr_Clines_filter[which(allChr_Clines_filter$width_Opt1>=0),]

plot(positiveClines[positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.8 ,"centre"],positiveClines[positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.8,"width"],xlim=c(2000,20000),ylim=c(-10000,20000),xlab="",ylab="")
title(paste0("DS1, ","AFD > 0.8"))
lines(x = c(12580,12580), y = c(-10000,20000),lty=5,lwd=2,col='red')
points(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre"],negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"width"],col='grey')
#mtext("cline centre (km)",side=1,cex=0.8,line=1.9)
mtext("cline width (km)",side=2,cex=0.9,line=2)

sum(positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.8) # 10663 positive cline width have depth min 15 in 5 pools
sum(negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8) # 249 negative cline width have depth min 15 in 5 pools
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre"]<8000) # 129 are located far left
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre"]>15000) # 103 are located far right

# (b) DS 1, min depth 15 in at least 5 of 6 pools & deltaP > 0.9
plot(positiveClines[positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.9 ,"centre"],positiveClines[positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.9,"width"],xlim=c(2000,20000),ylim=c(-10000,20000),xlab="",ylab="")
title(paste0("DS1, ","AFD > 0.9"))
lines(x = c(12580,12580), y = c(-10000,20000),lty=5,lwd=2,col='red')
points(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre"],negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"width"],col='grey')

sum(positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.9) # 10663 positive cline width have depth min 15 in 5 pools
sum(negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9) # 249 negative cline width have depth min 15 in 5 pools
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre"]<8000) # 129 are located far left
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre"]>15000) # 103 are located far right



# (c) DS 2, min depth 15 in at least 5 of 6 pools & deltaP > 0.8
negativeClines <- allChr_Clines_filter[which(allChr_Clines_filter$width_Opt2<0),]
positiveClines <- allChr_Clines_filter[which(allChr_Clines_filter$width_Opt2>=0),]

plot(positiveClines[positiveClines[,"poolsPass_Depth20"]>=5 & positiveClines[,"AFD_15_20"]>=0.8 ,"centre_Opt2"],positiveClines[positiveClines[,"poolsPass_Depth20"]>=5 & positiveClines[,"AFD_15_20"]>=0.8,"width_Opt2"],xlim=c(2000,20000),ylim=c(-10000,20000),xlab="",ylab="")
title(paste0("DS2, ","AFD > 0.8"))
lines(x = c(12580,12580), y = c(-10000,20000),lty=5,lwd=2,col='red')
points(negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre_Opt2"],negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"width_Opt2"],col='grey')
#mtext("cline centre (km)",side=1,cex=0.9,line=2)
mtext("cline width (km)",side=2,cex=0.9,line=2)
sum(positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.8) # 10663 positive cline width have depth min 15 in 5 pools
sum(negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8) # 249 negative cline width have depth min 15 in 5 pools
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre"]<8000) # 129 are located far left
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre"]>15000) # 103 are located far right

# (d) DS 2, min depth 15 in at least 5 of 6 pools & deltaP > 0.9
plot(positiveClines[positiveClines[,"poolsPass_Depth20"]>=5 & positiveClines[,"AFD_15_20"]>=0.9 ,"centre_Opt2"],positiveClines[positiveClines[,"poolsPass_Depth20"]>=5 & positiveClines[,"AFD_15_20"]>=0.9,"width_Opt2"],xlim=c(2000,20000),ylim=c(-10000,20000),xlab="",ylab="")
title(paste0("DS2, ","AFD > 0.9"))
lines(x = c(12580,12580), y = c(-10000,20000),lty=5,lwd=2,col='red')
points(negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre_Opt2"],negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"width_Opt2"],col='grey')
#mtext("cline centre (km)",side=1,cex=0.9,line=2)
#mtext("cline width (km)",side=2,cex=0.9,line=2)
sum(positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.9) # 10663 positive cline width have depth min 15 in 5 pools
sum(negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9) # 249 negative cline width have depth min 15 in 5 pools
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre"]<8000) # 129 are located far left
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre"]>15000) # 103 are located far right

# (e) DS 3, min depth 15 in at least 5 of 6 pools & deltaP > 0.8
negativeClines <- allChr_Clines_filter[which(allChr_Clines_filter$width_Opt3<0),]
positiveClines <- allChr_Clines_filter[which(allChr_Clines_filter$width_Opt3>=0),]

plot(positiveClines[positiveClines[,"poolsPass_Depth20"]>=5 & positiveClines[,"AFD_15_20"]>=0.8 ,"centre_Opt3"],positiveClines[positiveClines[,"poolsPass_Depth20"]>=5 & positiveClines[,"AFD_15_20"]>=0.8,"width_Opt3"],xlim=c(2000,20000),ylim=c(-10000,20000),xlab="",ylab="")
title(paste0("DS3, ","AFD > 0.8"))
lines(x = c(12580,12580), y = c(-10000,20000),lty=5,lwd=2,col='red')
points(negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre_Opt3"],negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"width_Opt3"],col='grey')
mtext("cline centre (km)",side=1,cex=0.9,line=2)
mtext("cline width (km)",side=2,cex=0.9,line=2)
sum(positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.8) # 10663 positive cline width have depth min 15 in 5 pools
sum(negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8) # 249 negative cline width have depth min 15 in 5 pools
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre"]<8000) # 129 are located far left
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.8,"centre"]>15000) # 103 are located far right

# (f) DS 3, min depth 15 in at least 5 of 6 pools & deltaP > 0.9
plot(positiveClines[positiveClines[,"poolsPass_Depth20"]>=5 & positiveClines[,"AFD_15_20"]>=0.9 ,"centre_Opt3"],positiveClines[positiveClines[,"poolsPass_Depth20"]>=5 & positiveClines[,"AFD_15_20"]>=0.9,"width_Opt3"],xlim=c(2000,20000),ylim=c(-10000,20000),xlab="",ylab="")
title(paste0("DS3, ","AFD > 0.9"))
lines(x = c(12580,12580), y = c(-10000,20000),lty=5,lwd=2,col='red')
points(negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre_Opt3"],negativeClines[negativeClines[,"poolsPass_Depth20"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"width_Opt3"],col='grey')
mtext("cline centre (km)",side=1,cex=0.9,line=2)
#mtext("cline width (km)",side=2,cex=0.9,line=2)
sum(positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"AFD_15_20"]>=0.9) # 10663 positive cline width have depth min 15 in 5 pools
sum(negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9) # 249 negative cline width have depth min 15 in 5 pools
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre"]<8000) # 129 are located far left
sum(negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"AFD_15_20"]>=0.9,"centre"]>15000) # 103 are located far right

dev.off()
```

```{r detailed look at negative cline widths in centre of hybrid zone}

# par(mfrow=c(2,1))
# par(mar=c(4,4,4,4))
# hist(allChr_Clines[,"width"],xlab="cline width",main="all cline widths")
# hist(negativeClines[,"width"],xlab="cline width",main="negative cline widths")
# 
# # plotting some of them
# par(mfrow=c(1,1))
# par(mar=c(4,4,4,4))
# par(oma=c(4,4,4,4))
# plot(c(0.382,11.227,11.404,13.767,14.424,20.670),rep(0,6),bg = "black",xaxs = "i", yaxs = "i",xlab="",ylab="",main="",xlim=c(0,24),
#            ylim=c(-0.1,1.1),type='n',cex.main=0.9,bty="n",yaxt="n",xaxt="n")
# axis(1,at=seq(0,24,1),labels=NA,col.axis="black",las=2,cex.axis=1.1,lwd=1.6,pos=-0.01, tck = -.02)
# axis(1,at=seq(0,24+1,2),labels=seq(0,24000,2000)/1000,col.axis="black",tck=-0.3,las=1,cex.axis=1.1,lwd=0,pos=-0.02)
# axis(2,at=seq(-0.01,1.01,0.1),labels=NA,col.axis="black",las=1,cex.axis=1.1,lwd=1.6,pos=0,tck = -.02)
# axis(2,at=seq(0,1.0,0.2),labels=seq(0,1,0.2),col.axis="black",tck=-.03,las=1,cex.axis=1.1,lwd=0,pos=0)
# lines(x=c(24,24),y=c(0,1),lwd=1.8)
# lines(x=c(0,24),y=c(1,1),lwd=1.8)
# title("Negative cline width examples",line=1,cex.main=1)
# mtext("Distance along transect (m)",side=1,cex=1,line=2)
# mtext("Allele frequency (p)",side=2,cex=1,line=3.5)
# for (thisRow in 1:nrow(negativeClines)) {
#           # thisRow <- 1
#           poolSeqClineThisLocus <- cbind(c(0.382,11.227,11.404,13.767,14.424,20.670),unlist(negativeClines[thisRow,c(8,10,9,11,12,13)]))
#           colnames(poolSeqClineThisLocus) <- c("DistAlongTransect","p")
#           poolSeqClineThisLocus <- as.data.frame(poolSeqClineThisLocus)
#           poolSeqClineThisLocus[,"DistAlongTransect"] <- as.numeric(poolSeqClineThisLocus[,"DistAlongTransect"])
#           poolSeqClineThisLocus[,"p"] <- as.numeric(poolSeqClineThisLocus[,"p"])
#           lines(x=poolSeqClineThisLocus[,"DistAlongTransect"],y=poolSeqClineThisLocus[,"p"],lty=1,lwd=1.2,col='lightgrey')
# }    

# alternative with lots of small panels
# plotting some of them

# just those with negative values positioned near centre of hybrid zone
# negative cline widths coloured differently
negativeClines <- allChr_Clines_chk[which(allChr_Clines_chk$width<0),]
positiveClines <- allChr_Clines_chk[which(allChr_Clines_chk$width>=0),]

negativeClines_HZcentre <- negativeClines[negativeClines[,"poolsPass_Depth15"]>=5 & negativeClines[,"centre"]>10000 & negativeClines[,"centre"]<15000,]
  nrow(negativeClines_HZcentre)
par(mfrow=c(4,4))
par(mar=c(1.2,1.2,1.2,1.2))
par(oma=c(1.2,1.2,1.2,1.2))
for (thisRow in 1:nrow(negativeClines_HZcentre)) {
  # thisRow <- 1
  poolSeqClineThisLocus <- cbind(c(0.382,11.227,11.404,13.767,14.424,20.670),unlist(negativeClines_HZcentre[thisRow,c(9,11,10,12,13,14)]))
  #negativeClines_HZcentre[thisRow,]
  thisWidth_OPT1 <- round(negativeClines_HZcentre[thisRow,"width"],0)
  thisWidth_OPT2 <- round(negativeClines_HZcentre[thisRow,"width_Opt2"],0)
  thisWidth_OPT3 <- round(negativeClines_HZcentre[thisRow,"width_Opt3"],0)
  subTitle <- paste(paste(paste0("w_opt1: ",thisWidth_OPT1),paste0("w_opt2: ",thisWidth_OPT2),sep=", "),paste0("w_opt3: ",thisWidth_OPT3),sep=", ")
  colnames(poolSeqClineThisLocus) <- c("DistAlongTransect","p")
  poolSeqClineThisLocus <- as.data.frame(poolSeqClineThisLocus)
  poolSeqClineThisLocus[,"DistAlongTransect"] <- as.numeric(poolSeqClineThisLocus[,"DistAlongTransect"])
  poolSeqClineThisLocus[,"p"] <- as.numeric(poolSeqClineThisLocus[,"p"])
  plot(x=poolSeqClineThisLocus[,"DistAlongTransect"],y=poolSeqClineThisLocus[,"p"],bg = "black",xaxs = "i", yaxs = "i",xlab="",ylab="",main="",xlim=c(0,24),ylim=c(-0.1,1.1),type='l',cex.main=0.9,bty="n",yaxt="n",xaxt="n")
  axis(1,at=seq(0,24,1),labels=NA,col.axis="black",las=2,cex.axis=1.1,lwd=1.6,pos=-0.01, tck = -.02)
  axis(1,at=seq(0,24+1,2),labels=seq(0,24000,2000)/1000,col.axis="black",tck=-0.3,las=1,cex.axis=1.1,lwd=0,pos=-0.001)
  axis(2,at=seq(-0.01,1.01,0.1),labels=NA,col.axis="black",las=1,cex.axis=1.1,lwd=1.6,pos=0,tck = -.02)
  axis(2,at=seq(0,1.0,0.2),labels=seq(0,1,0.2),col.axis="black",tck=-.03,las=1,cex.axis=1.1,lwd=0,pos=0)
  lines(x=c(24,24),y=c(0,1),lwd=1.8)
  lines(x=c(0,24),y=c(1,1),lwd=1.8)
  title(paste(paste(paste("LG",negativeClines_HZcentre[thisRow,"LG"]),negativeClines_HZcentre[thisRow,"position"],sep="- "),subTitle,sep="\n"),line=-0.6,cex.main=1)
      #mtext("Distance along transect (m)",side=1,cex=1,line=2)
      #mtext("Allele frequency (p)",side=2,cex=1,line=3.5)
}

# pdf version
pdf("alleleFrequencies_negativeWidthExample.pdf",width = 9, height = 10,pointsize=16)
par(mfrow=c(4,4))
par(mar=c(1.2,1.2,1.2,1.2))
par(oma=c(1.2,1.2,1.2,1.2))
for (thisRow in 1:nrow(negativeClines_HZcentre)) {
  # thisRow <- 1
  poolSeqClineThisLocus <- cbind(c(0.382,11.227,11.404,13.767,14.424,20.670),unlist(negativeClines_HZcentre[thisRow,c(9,11,10,12,13,14)]))
  #negativeClines_HZcentre[thisRow,]
  thisWidth_OPT1 <- round(negativeClines_HZcentre[thisRow,"width"],0)
  thisWidth_OPT2 <- round(negativeClines_HZcentre[thisRow,"width_Opt2"],0)
  thisWidth_OPT3 <- round(negativeClines_HZcentre[thisRow,"width_Opt3"],0)
  subTitle <- paste(paste(paste0("w_opt1: ",thisWidth_OPT1),paste0("w_opt2: ",thisWidth_OPT2),sep=", "),paste0("w_opt3: ",thisWidth_OPT3),sep=", ")
  colnames(poolSeqClineThisLocus) <- c("DistAlongTransect","p")
  poolSeqClineThisLocus <- as.data.frame(poolSeqClineThisLocus)
  poolSeqClineThisLocus[,"DistAlongTransect"] <- as.numeric(poolSeqClineThisLocus[,"DistAlongTransect"])
  poolSeqClineThisLocus[,"p"] <- as.numeric(poolSeqClineThisLocus[,"p"])
  plot(x=poolSeqClineThisLocus[,"DistAlongTransect"],y=poolSeqClineThisLocus[,"p"],bg = "black",xaxs = "i", yaxs = "i",xlab="",ylab="",main="",xlim=c(0,24),ylim=c(-0.1,1.1),type='l',cex.main=0.9,bty="n",yaxt="n",xaxt="n")
  axis(1,at=seq(0,24,1),labels=NA,col.axis="black",las=2,cex.axis=1.1,lwd=1.6,pos=-0.01, tck = -.02)
  axis(1,at=seq(0,24+1,2),labels=seq(0,24000,2000)/1000,col.axis="black",tck=-0.3,las=1,cex.axis=1.1,lwd=0,pos=-0.001)
  axis(2,at=seq(-0.01,1.01,0.1),labels=NA,col.axis="black",las=1,cex.axis=1.1,lwd=1.6,pos=0,tck = -.02)
  axis(2,at=seq(0,1.0,0.2),labels=seq(0,1,0.2),col.axis="black",tck=-.03,las=1,cex.axis=1.1,lwd=0,pos=0.8)
  lines(x=c(24,24),y=c(0,1),lwd=1.8)
  lines(x=c(0,24),y=c(1,1),lwd=1.8)
  title(paste(paste(paste("LG",negativeClines_HZcentre[thisRow,"LG"]),negativeClines_HZcentre[thisRow,"position"],sep="- "),subTitle,sep="\n"),line=-0.6,cex.main=0.7)
      #mtext("Distance along transect (m)",side=1,cex=1,line=2)
      #mtext("Allele frequency (p)",side=2,cex=1,line=3.5)
}

dev.off()



# comparing to some random clines with positive values in the middle position
positiveClines_HZcentre <- positiveClines[positiveClines[,"poolsPass_Depth15"]>=5 & positiveClines[,"centre"]>10000 & positiveClines[,"centre"]<15000,]
  nrow(positiveClines_HZcentre)
par(mfrow=c(4,4))
par(mar=c(1.2,1.2,1.2,1.2))
par(oma=c(1.2,1.2,1.2,1.2))
for (thisRow in 1:16) {
    poolSeqClineThisLocus <- cbind(c(0.382,11.227,11.404,13.767,14.424,20.670),unlist(positiveClines_HZcentre[thisRow,c(9,11,10,12,13,14)]))
    colnames(poolSeqClineThisLocus) <- c("DistAlongTransect","p")
    poolSeqClineThisLocus <- as.data.frame(poolSeqClineThisLocus)
    poolSeqClineThisLocus[,"DistAlongTransect"] <- as.numeric(poolSeqClineThisLocus[,"DistAlongTransect"])
    poolSeqClineThisLocus[,"p"] <- as.numeric(poolSeqClineThisLocus[,"p"])
    plot(x=poolSeqClineThisLocus[,"DistAlongTransect"],y=poolSeqClineThisLocus[,"p"],bg = "black",xaxs = "i", yaxs = "i",xlab="", ylab="",main="",xlim=c(0,24),ylim=c(-0.1,1.1),type='l',cex.main=0.9,bty="n",yaxt="n",xaxt="n")
    axis(1,at=seq(0,24,1),labels=NA,col.axis="black",las=2,cex.axis=1.1,lwd=1.6,pos=-0.01, tck = -.02)
    axis(1,at=seq(0,24+1,2),labels=seq(0,24000,2000)/1000,col.axis="black",tck=-0.3,las=1,cex.axis=1.1,lwd=0,pos=-0.001)
    axis(2,at=seq(-0.01,1.01,0.1),labels=NA,col.axis="black",las=1,cex.axis=1.1,lwd=1.6,pos=0,tck = -.02)
    axis(2,at=seq(0,1.0,0.2),labels=seq(0,1,0.2),col.axis="black",tck=-.03,las=1,cex.axis=1.1,lwd=0,pos=0)
    lines(x=c(24,24),y=c(0,1),lwd=1.8)
    lines(x=c(0,24),y=c(1,1),lwd=1.8)
    title(paste(paste("LG",positiveClines_HZcentre[thisRow,"LG"]),positiveClines_HZcentre[thisRow,"position"],sep="- "), line=0.2, cex.main=1)
    #mtext("Distance along transect (m)",side=1,cex=1,line=2)
    #mtext("Allele frequency (p)",side=2,cex=1,line=3.5)
}
#allChr_Clines[which(allChr_Clines$width<0),"width"] <- 0
#colnames(allChr_Clines)

#sum(allChr_Clines[,"dpthAdj_15"]<15)/nrow(allChr_Clines)
```

```{r combine fastClines with SlidngWindows version for permutation for depth15, include=F,eval=F}
clinesWindows <- matrix(-9,nrow(allChr_Clines_filterDpth15),23)
colnames(clinesWindows) <- c("Fst_15_20_95Q","Fst_16_19_95Q","Fst_17_18_95Q","Fst_15_16_95Q","Fst_16_17_95Q","Fst_18_19_95Q","Fst_19_20_95Q","AFD_1_Clines","AFD_90_Clines","AFD_80_clines","AFD_80to90_clines","AFD_1_NumClines","AFD_90_NumClines","AFD_80_NumClines","AFD_80to90_NumClines","AFD_1_MeanWidth","AFD_90_MeanWidth","AFD_80_MeanWidth","AFD_80to90_MeanWidth","AFD_1_MeanCentre","AFD_90_MeanCentre","AFD_80_MeanCentre","AFD_80to90_MeanCentre")
clinesWindows <- as.data.frame(clinesWindows)
clinesWindows[,"AFD_80to90_clines"] <- 0
clinesWindows[,"AFD_80_clines"] <- 0
clinesWindows[,"AFD_90_clines"] <- 0
clinesWindows[,"AFD_1_clines"] <- 0
clinesWindows[,"AFD_80to90_NumClines"] <- 0
clinesWindows[,"AFD_80_NumClines"] <- 0
clinesWindows[,"AFD_90_NumClines"] <- 0
clinesWindows[,"AFD_1_NumClines"] <- 0

# scan through clines to find which windows they belong to
for (thisRow in 1:nrow(allChr_Windows)) {
  # thisRow <- 85
  thisScaff <- as.vector(allChr_Windows[thisRow,"scaffold"])
  thisLG <- as.vector(allChr_Windows[thisRow,"LG"])
  thisWindowStart <- as.vector(allChr_Windows[thisRow,"windowStart"])
  thisWindowEnd <- as.vector(allChr_Windows[thisRow,"windowEnd"])
  # look in clines data for which ones are in this current window
  allChr_Clines_thisWindow <- allChr_Clines_filterDpth15 %>%
                filter(LG==thisLG) %>%
                filter(position>thisWindowStart & position<thisWindowEnd)
  if (nrow(allChr_Clines_thisWindow)==0) {
    next
  }
  if (nrow(allChr_Clines_thisWindow)>0) {
    #print(thisRow)
    # deltaP 1
    allChr_Clines_thisWindow_delta1 <- allChr_Clines_thisWindow %>%
                filter(AFD_15_20==1)
    if (nrow(allChr_Clines_thisWindow_delta1)>0) {
      print(paste0("thisRow: ",thisRow,", LG: ",thisLG,", window: ",thisWindowStart,": deltap1"))  
      clinesWindows[thisRow,"AFD_1_clines"] <- 1
      clinesWindows[thisRow,"AFD_1_NumClines"] <- nrow(allChr_Clines_thisWindow_delta1)
      clinesWindows[thisRow,"AFD_1_MeanWidth"] <- mean(allChr_Clines_thisWindow_delta1[,"width"])
      clinesWindows[thisRow,"AFD_1_MeanCentre"] <- mean(allChr_Clines_thisWindow_delta1[,"centre"])
    }
    # deltaP 0.9
    allChr_Clines_thisWindow_delta90 <- allChr_Clines_thisWindow %>%
                filter(AFD_15_20>=0.9)
    if (nrow(allChr_Clines_thisWindow_delta90)>0) {
      print(paste0("thisRow: ",thisRow,", LG: ",thisLG,", window: ",thisWindowStart,": deltap09"))  
      clinesWindows[thisRow,"AFD_90_clines"] <- 1
      clinesWindows[thisRow,"AFD_90_NumClines"] <- nrow(allChr_Clines_thisWindow_delta90)
      clinesWindows[thisRow,"AFD_90_MeanWidth"] <- mean(allChr_Clines_thisWindow_delta90[,"width"])
      clinesWindows[thisRow,"AFD_90_MeanCentre"] <- mean(allChr_Clines_thisWindow_delta90[,"centre"])
    }
    # deltaP 0.8
    allChr_Clines_thisWindow_delta80 <- allChr_Clines_thisWindow %>%
                filter(AFD_15_20>=0.8)
    if (nrow(allChr_Clines_thisWindow_delta80)>0) {
      print(paste0("thisRow: ",thisRow,", LG: ",thisLG,", window: ",thisWindowStart,": deltap08"))  
      clinesWindows[thisRow,"AFD_80_clines"] <- 1
      clinesWindows[thisRow,"AFD_80_NumClines"] <- nrow(allChr_Clines_thisWindow_delta80)
      clinesWindows[thisRow,"AFD_80_MeanWidth"] <- mean(allChr_Clines_thisWindow_delta80[,"width"])
      clinesWindows[thisRow,"AFD_80_MeanCentre"] <- mean(allChr_Clines_thisWindow_delta80[,"centre"])
    }
    # deltaP 0.8 to 0.9
    allChr_Clines_thisWindow_delta80to90 <- allChr_Clines_thisWindow %>%
                filter(AFD_15_20>=0.8 & AFD_15_20<0.9)
    if (nrow(allChr_Clines_thisWindow_delta80to90)>0) {
      clinesWindows[thisRow,"AFD_80to90_clines"] <- 1
      clinesWindows[thisRow,"AFD_80to90_NumClines"] <- nrow(allChr_Clines_thisWindow_delta80to90)
      clinesWindows[thisRow,"AFD_80to90_MeanWidth"] <- mean(allChr_Clines_thisWindow_delta80to90[,"width"])
      clinesWindows[thisRow,"AFD_80to90_MeanCentre"] <- mean(allChr_Clines_thisWindow_delta80to90[,"centre"])
    }
  } 
} 


# Outlier Fst Islands - 95% Quantiles
Fst_15_20_95Q <- quantile(allChr_Windows[,"FstfromMeanPiAdj_15_20"], na.rm=T, c(0.5, 0.95))
clinesWindows[which(allChr_Windows[,"FstfromMeanPiAdj_15_20"]>=as.numeric(Fst_15_20_95Q[2])),"Fst_15_20_95Q"] <- 1
clinesWindows[which(allChr_Windows[,"FstfromMeanPiAdj_15_20"]<as.numeric(Fst_15_20_95Q[2])),"Fst_15_20_95Q"] <- 0
clinesWindows[which(clinesWindows[,"Fst_15_20_95Q"]==-9),"Fst_15_20_95Q"] <- 0

Fst_16_19_95Q <- quantile(allChr_Windows[,"FstfromMeanPiAdj_16_19"], na.rm=T, c(0.5, 0.95))
clinesWindows[which(allChr_Windows[,"FstfromMeanPiAdj_16_19"]>=as.numeric(Fst_16_19_95Q[2])),"Fst_16_19_95Q"] <- 1
clinesWindows[which(allChr_Windows[,"FstfromMeanPiAdj_16_19"]<as.numeric(Fst_16_19_95Q[2])),"Fst_16_19_95Q"] <- 0
clinesWindows[which(clinesWindows[,"Fst_16_19_95Q"]==-9),"Fst_16_19_95Q"] <- 0

Fst_17_18_95Q <- quantile(allChr_Windows[,"FstfromMeanPiAdj_17_18"], na.rm=T, c(0.5, 0.95))
clinesWindows[which(allChr_Windows[,"FstfromMeanPiAdj_17_18"]>=as.numeric(Fst_17_18_95Q[2])),"Fst_17_18_95Q"] <- 1
clinesWindows[which(allChr_Windows[,"FstfromMeanPiAdj_17_18"]<as.numeric(Fst_17_18_95Q[2])),"Fst_17_18_95Q"] <- 0
clinesWindows[which(clinesWindows[,"Fst_17_18_95Q"]==-9),"Fst_17_18_95Q"] <- 0

Fst_15_16_95Q <- quantile(allChr_Windows[,"FstfromMeanPiAdj_15_16"], na.rm=T, c(0.5, 0.95))
clinesWindows[which(allChr_Windows[,"FstfromMeanPiAdj_15_16"]>=as.numeric(Fst_15_16_95Q[2])),"Fst_15_16_95Q"] <- 1
clinesWindows[which(allChr_Windows[,"FstfromMeanPiAdj_15_16"]<as.numeric(Fst_15_16_95Q[2])),"Fst_15_16_95Q"] <- 0
clinesWindows[which(clinesWindows[,"Fst_15_16_95Q"]==-9),"Fst_15_16_95Q"] <- 0

Fst_16_17_95Q <- quantile(allChr_Windows[,"FstfromMeanPiAdj_16_17"], na.rm=T, c(0.5, 0.95))
clinesWindows[which(allChr_Windows[,"FstfromMeanPiAdj_16_17"]>=as.numeric(Fst_16_17_95Q[2])),"Fst_16_17_95Q"] <- 1
clinesWindows[which(allChr_Windows[,"FstfromMeanPiAdj_16_17"]<as.numeric(Fst_16_17_95Q[2])),"Fst_16_17_95Q"] <- 0
clinesWindows[which(clinesWindows[,"Fst_16_17_95Q"]==-9),"Fst_16_17_95Q"] <- 0

Fst_18_19_95Q <- quantile(allChr_Windows[,"FstfromMeanPiAdj_18_19"], na.rm=T, c(0.5, 0.95))
clinesWindows[which(allChr_Windows[,"FstfromMeanPiAdj_18_19"]>=as.numeric(Fst_18_19_95Q[2])),"Fst_18_19_95Q"] <- 1
clinesWindows[which(allChr_Windows[,"FstfromMeanPiAdj_18_19"]<as.numeric(Fst_15_16_95Q[2])),"Fst_18_19_95Q"] <- 0
clinesWindows[which(clinesWindows[,"Fst_18_19_95Q"]==-9),"Fst_18_19_95Q"] <- 0

Fst_19_20_95Q <- quantile(allChr_Windows[,"FstfromMeanPiAdj_19_20"], na.rm=T, c(0.5, 0.95))
clinesWindows[which(allChr_Windows[,"FstfromMeanPiAdj_19_20"]>=as.numeric(Fst_19_20_95Q[2])),"Fst_19_20_95Q"] <- 1
clinesWindows[which(allChr_Windows[,"FstfromMeanPiAdj_19_20"]<as.numeric(Fst_19_20_95Q[2])),"Fst_19_20_95Q"] <- 0
clinesWindows[which(clinesWindows[,"Fst_19_20_95Q"]==-9),"Fst_19_20_95Q"] <- 0

table(clinesWindows[,"Fst_15_20_95Q"])
table(clinesWindows[,"Fst_16_19_95Q"])
table(clinesWindows[,"Fst_17_18_95Q"])
table(clinesWindows[,"Fst_15_20_95Q"])
table(clinesWindows[,"Fst_15_20_95Q"])

range(clinesWindows[,"AFD_1_MeanWidth"])
range(clinesWindows[,"AFD_1_MeanCentre"])
range(clinesWindows[,"AFD_90_MeanCentre"])
range(clinesWindows[,"AFD_80_MeanCentre"])
range(clinesWindows[,"AFD_80to90_MeanCentre"])
range(clinesWindows[,"AFD_80to90_MeanWidth"])
range(clinesWindows[,"AFD_80_MeanWidth"])
range(clinesWindows[,"AFD_1_MeanWidth"])

range(clinesWindows[,"AFD_1_clines"])
range(clinesWindows[,"AFD_1_Clines"])


# combine
# One off, accidently added colour genes columns twice
# tail(colnames(allChr_Windows))
# ncol(allChr_Windows)
# allChr_Windows<- allChr_Windows[-c(3154:3156)]
# tail(colnames(allChr_Windows))


clinesWindows_combined <- cbind(allChr_Windows,clinesWindows)
#write.csv(clinesWindows_combined,"clinesWindows_combined.csv",row.names = F)

clinesWindows_combined <- as.data.frame(clinesWindows_combined)


colnames(clinesWindows_combined)
clinesWindows_combined_short <- clinesWindows_combined %>% 
  dplyr::select(-(poly_pool_1:poly_pool_20)) %>% 
  dplyr::select(-(dpthPass_1:pi_20)) %>% 
  dplyr::select(-(dpthPolyPass_1_2:piBar_19_20)) %>%
  dplyr::select(-(piT_1_2:dXYraw_19_20)) %>%
  dplyr::select(-(FstPi_1_2:FstPi_19_20)) %>%
  dplyr::select(-(FstfromDxy_1_2:FstfromMeanPi_19_20)) %>%
  dplyr::select(-(FstPiAdj_1_2:FstPiAdj_19_20)) %>%
  dplyr::select(-(dXYfromFstadj_1_2:dXYfromFstadj_19_20)) %>%
  dplyr::select(-(dXYfromMeanFst_1_2:dXYfromMeanFst_19_20)) %>%
  dplyr::select(-(dAfromMeanFst_1_2:dAfromMeanFst_19_20)) %>%
  dplyr::select(-(AFD_1_Clines)) %>%
  dplyr::select(-(AFD_90_Clines))

# colnames(clinesWindows_combined_short)
# check ELUTA is still there
#colnames(clinesWindows_combined_short)
table(clinesWindows_combined_short[,"ColourGene"])
write.csv(clinesWindows_combined_short,"clinesWindows_combined_short_dpth15.csv",row.names = F)
```

```{r combine fastClines with SlidngWindows version for permutation for depth20, include=F,eval=F}
clinesWindows <- matrix(-9,nrow(allChr_Clines_filterDpth20),23)
colnames(clinesWindows) <- c("Fst_15_20_95Q","Fst_16_19_95Q","Fst_17_18_95Q","Fst_15_16_95Q","Fst_16_17_95Q","Fst_18_19_95Q","Fst_19_20_95Q","AFD_1_Clines","AFD_90_Clines","AFD_80_clines","AFD_80to90_clines","AFD_1_NumClines","AFD_90_NumClines","AFD_80_NumClines","AFD_80to90_NumClines","AFD_1_MeanWidth","AFD_90_MeanWidth","AFD_80_MeanWidth","AFD_80to90_MeanWidth","AFD_1_MeanCentre","AFD_90_MeanCentre","AFD_80_MeanCentre","AFD_80to90_MeanCentre")
clinesWindows <- as.data.frame(clinesWindows)
clinesWindows[,"AFD_80to90_clines"] <- 0
clinesWindows[,"AFD_80_clines"] <- 0
clinesWindows[,"AFD_90_clines"] <- 0
clinesWindows[,"AFD_1_clines"] <- 0
clinesWindows[,"AFD_80to90_NumClines"] <- 0
clinesWindows[,"AFD_80_NumClines"] <- 0
clinesWindows[,"AFD_90_NumClines"] <- 0
clinesWindows[,"AFD_1_NumClines"] <- 0

# scan through clines to find which windows they belong to
for (thisRow in 1:nrow(allChr_Windows)) {
  # thisRow <- 85
  thisScaff <- as.vector(allChr_Windows[thisRow,"scaffold"])
  thisLG <- as.vector(allChr_Windows[thisRow,"LG"])
  thisWindowStart <- as.vector(allChr_Windows[thisRow,"windowStart"])
  thisWindowEnd <- as.vector(allChr_Windows[thisRow,"windowEnd"])
  # look in clines data for which ones are in this current window
  allChr_Clines_thisWindow <- allChr_Clines_filterDpth20 %>%
                filter(LG==thisLG) %>%
                filter(position>thisWindowStart & position<thisWindowEnd)
  if (nrow(allChr_Clines_thisWindow)==0) {
    next
  }
  if (nrow(allChr_Clines_thisWindow)>0) {
    #print(thisRow)
    # deltaP 1
    allChr_Clines_thisWindow_delta1 <- allChr_Clines_thisWindow %>%
                filter(AFD_15_20==1)
    if (nrow(allChr_Clines_thisWindow_delta1)>0) {
      print(paste0("thisRow: ",thisRow,", LG: ",thisLG,", window: ",thisWindowStart,": deltap1"))  
      clinesWindows[thisRow,"AFD_1_clines"] <- 1
      clinesWindows[thisRow,"AFD_1_NumClines"] <- nrow(allChr_Clines_thisWindow_delta1)
      clinesWindows[thisRow,"AFD_1_MeanWidth"] <- mean(allChr_Clines_thisWindow_delta1[,"width"])
      clinesWindows[thisRow,"AFD_1_MeanCentre"] <- mean(allChr_Clines_thisWindow_delta1[,"centre"])
    }
    # deltaP 0.9
    allChr_Clines_thisWindow_delta90 <- allChr_Clines_thisWindow %>%
                filter(AFD_15_20>=0.9)
    if (nrow(allChr_Clines_thisWindow_delta90)>0) {
      print(paste0("thisRow: ",thisRow,", LG: ",thisLG,", window: ",thisWindowStart,": deltap09"))  
      clinesWindows[thisRow,"AFD_90_clines"] <- 1
      clinesWindows[thisRow,"AFD_90_NumClines"] <- nrow(allChr_Clines_thisWindow_delta90)
      clinesWindows[thisRow,"AFD_90_MeanWidth"] <- mean(allChr_Clines_thisWindow_delta90[,"width"])
      clinesWindows[thisRow,"AFD_90_MeanCentre"] <- mean(allChr_Clines_thisWindow_delta90[,"centre"])
    }
    # deltaP 0.8
    allChr_Clines_thisWindow_delta80 <- allChr_Clines_thisWindow %>%
                filter(AFD_15_20>=0.8)
    if (nrow(allChr_Clines_thisWindow_delta80)>0) {
      print(paste0("thisRow: ",thisRow,", LG: ",thisLG,", window: ",thisWindowStart,": deltap08"))  
      clinesWindows[thisRow,"AFD_80_clines"] <- 1
      clinesWindows[thisRow,"AFD_80_NumClines"] <- nrow(allChr_Clines_thisWindow_delta80)
      clinesWindows[thisRow,"AFD_80_MeanWidth"] <- mean(allChr_Clines_thisWindow_delta80[,"width"])
      clinesWindows[thisRow,"AFD_80_MeanCentre"] <- mean(allChr_Clines_thisWindow_delta80[,"centre"])
    }
    # deltaP 0.8 to 0.9
    allChr_Clines_thisWindow_delta80to90 <- allChr_Clines_thisWindow %>%
                filter(AFD_15_20>=0.8 & AFD_15_20<0.9)
    if (nrow(allChr_Clines_thisWindow_delta80to90)>0) {
      clinesWindows[thisRow,"AFD_80to90_clines"] <- 1
      clinesWindows[thisRow,"AFD_80to90_NumClines"] <- nrow(allChr_Clines_thisWindow_delta80to90)
      clinesWindows[thisRow,"AFD_80to90_MeanWidth"] <- mean(allChr_Clines_thisWindow_delta80to90[,"width"])
      clinesWindows[thisRow,"AFD_80to90_MeanCentre"] <- mean(allChr_Clines_thisWindow_delta80to90[,"centre"])
    }
  } 
} 


# Outlier Fst Islands - 95% Quantiles
Fst_15_20_95Q <- quantile(allChr_Windows[,"FstfromMeanPiAdj_15_20"], na.rm=T, c(0.5, 0.95))
clinesWindows[which(allChr_Windows[,"FstfromMeanPiAdj_15_20"]>=as.numeric(Fst_15_20_95Q[2])),"Fst_15_20_95Q"] <- 1
clinesWindows[which(allChr_Windows[,"FstfromMeanPiAdj_15_20"]<as.numeric(Fst_15_20_95Q[2])),"Fst_15_20_95Q"] <- 0
clinesWindows[which(clinesWindows[,"Fst_15_20_95Q"]==-9),"Fst_15_20_95Q"] <- 0

Fst_16_19_95Q <- quantile(allChr_Windows[,"FstfromMeanPiAdj_16_19"], na.rm=T, c(0.5, 0.95))
clinesWindows[which(allChr_Windows[,"FstfromMeanPiAdj_16_19"]>=as.numeric(Fst_16_19_95Q[2])),"Fst_16_19_95Q"] <- 1
clinesWindows[which(allChr_Windows[,"FstfromMeanPiAdj_16_19"]<as.numeric(Fst_16_19_95Q[2])),"Fst_16_19_95Q"] <- 0
clinesWindows[which(clinesWindows[,"Fst_16_19_95Q"]==-9),"Fst_16_19_95Q"] <- 0

Fst_17_18_95Q <- quantile(allChr_Windows[,"FstfromMeanPiAdj_17_18"], na.rm=T, c(0.5, 0.95))
clinesWindows[which(allChr_Windows[,"FstfromMeanPiAdj_17_18"]>=as.numeric(Fst_17_18_95Q[2])),"Fst_17_18_95Q"] <- 1
clinesWindows[which(allChr_Windows[,"FstfromMeanPiAdj_17_18"]<as.numeric(Fst_17_18_95Q[2])),"Fst_17_18_95Q"] <- 0
clinesWindows[which(clinesWindows[,"Fst_17_18_95Q"]==-9),"Fst_17_18_95Q"] <- 0

Fst_15_16_95Q <- quantile(allChr_Windows[,"FstfromMeanPiAdj_15_16"], na.rm=T, c(0.5, 0.95))
clinesWindows[which(allChr_Windows[,"FstfromMeanPiAdj_15_16"]>=as.numeric(Fst_15_16_95Q[2])),"Fst_15_16_95Q"] <- 1
clinesWindows[which(allChr_Windows[,"FstfromMeanPiAdj_15_16"]<as.numeric(Fst_15_16_95Q[2])),"Fst_15_16_95Q"] <- 0
clinesWindows[which(clinesWindows[,"Fst_15_16_95Q"]==-9),"Fst_15_16_95Q"] <- 0

Fst_16_17_95Q <- quantile(allChr_Windows[,"FstfromMeanPiAdj_16_17"], na.rm=T, c(0.5, 0.95))
clinesWindows[which(allChr_Windows[,"FstfromMeanPiAdj_16_17"]>=as.numeric(Fst_16_17_95Q[2])),"Fst_16_17_95Q"] <- 1
clinesWindows[which(allChr_Windows[,"FstfromMeanPiAdj_16_17"]<as.numeric(Fst_16_17_95Q[2])),"Fst_16_17_95Q"] <- 0
clinesWindows[which(clinesWindows[,"Fst_16_17_95Q"]==-9),"Fst_16_17_95Q"] <- 0

Fst_18_19_95Q <- quantile(allChr_Windows[,"FstfromMeanPiAdj_18_19"], na.rm=T, c(0.5, 0.95))
clinesWindows[which(allChr_Windows[,"FstfromMeanPiAdj_18_19"]>=as.numeric(Fst_18_19_95Q[2])),"Fst_18_19_95Q"] <- 1
clinesWindows[which(allChr_Windows[,"FstfromMeanPiAdj_18_19"]<as.numeric(Fst_15_16_95Q[2])),"Fst_18_19_95Q"] <- 0
clinesWindows[which(clinesWindows[,"Fst_18_19_95Q"]==-9),"Fst_18_19_95Q"] <- 0

Fst_19_20_95Q <- quantile(allChr_Windows[,"FstfromMeanPiAdj_19_20"], na.rm=T, c(0.5, 0.95))
clinesWindows[which(allChr_Windows[,"FstfromMeanPiAdj_19_20"]>=as.numeric(Fst_19_20_95Q[2])),"Fst_19_20_95Q"] <- 1
clinesWindows[which(allChr_Windows[,"FstfromMeanPiAdj_19_20"]<as.numeric(Fst_19_20_95Q[2])),"Fst_19_20_95Q"] <- 0
clinesWindows[which(clinesWindows[,"Fst_19_20_95Q"]==-9),"Fst_19_20_95Q"] <- 0

table(clinesWindows[,"Fst_15_20_95Q"])
table(clinesWindows[,"Fst_16_19_95Q"])
table(clinesWindows[,"Fst_17_18_95Q"])
table(clinesWindows[,"Fst_15_20_95Q"])
table(clinesWindows[,"Fst_15_20_95Q"])

range(clinesWindows[,"AFD_1_MeanWidth"])
range(clinesWindows[,"AFD_1_MeanCentre"])
range(clinesWindows[,"AFD_90_MeanCentre"])
range(clinesWindows[,"AFD_80_MeanCentre"])
range(clinesWindows[,"AFD_80to90_MeanCentre"])
range(clinesWindows[,"AFD_80to90_MeanWidth"])
range(clinesWindows[,"AFD_80_MeanWidth"])
range(clinesWindows[,"AFD_1_MeanWidth"])

range(clinesWindows[,"AFD_1_clines"])
range(clinesWindows[,"AFD_1_Clines"])


# combine
# One off, accidently added colour genes columns twice
# tail(colnames(allChr_Windows))
# ncol(allChr_Windows)
# allChr_Windows<- allChr_Windows[-c(3154:3156)]
# tail(colnames(allChr_Windows))


clinesWindows_combined <- cbind(allChr_Windows,clinesWindows)
#write.csv(clinesWindows_combined,"clinesWindows_combined.csv",row.names = F)

clinesWindows_combined <- as.data.frame(clinesWindows_combined)

colnames(clinesWindows_combined)
clinesWindows_combined_short <- clinesWindows_combined %>% 
  dplyr::select(-(poly_pool_1:poly_pool_20)) %>% 
  dplyr::select(-(dpthPass_1:pi_20)) %>% 
  dplyr::select(-(dpthPolyPass_1_2:piBar_19_20)) %>%
  dplyr::select(-(piT_1_2:dXYraw_19_20)) %>%
  dplyr::select(-(FstPi_1_2:FstPi_19_20)) %>%
  dplyr::select(-(FstfromDxy_1_2:FstfromMeanPi_19_20)) %>%
  dplyr::select(-(FstPiAdj_1_2:FstPiAdj_19_20)) %>%
  dplyr::select(-(dXYfromFstadj_1_2:dXYfromFstadj_19_20)) %>%
  dplyr::select(-(dXYfromMeanFst_1_2:dXYfromMeanFst_19_20)) %>%
  dplyr::select(-(dAfromMeanFst_1_2:dAfromMeanFst_19_20)) %>%
  dplyr::select(-(AFD_1_Clines)) %>%
  dplyr::select(-(AFD_90_Clines))

#colnames(clinesWindows_combined_short)
table(clinesWindows_combined_short[,"ColourGene"])
write.csv(clinesWindows_combined_short,"clinesWindows_combined_short_dpth20.csv",row.names = F)
```

# note permutation tests run in external python script: 

```{r special update on cline counts and averages for SULF region}
# Step 4: Remove rows with NA values in the 'width' column (if any)
Chr4_clinesNEWfilter <- Chr4_clinesNEW[Chr4_clinesNEW[,"width"]>0,]
table(Chr4_clinesNEW[,"distanceToGene"])
# Step 5: Summarise data for annotations
summary_stats_by_gene <- Chr4_clinesNEWfilter %>%
  group_by(ColourGene, distanceToGene) %>%
  summarise(
    avg_width = mean(width, na.rm = TRUE),
    sd_width = sd(width, na.rm = TRUE),
    min_width = min(width, na.rm = TRUE),
    max_width = max(width, na.rm = TRUE),
    n = n(),
    max_width = max(width, na.rm = TRUE)  # Capture max width for dynamic positioning
  )

```

```{r plotting lots of little cline plots for LG4 for deltaP_0_9 and lower deltaP 06}
table(Chr4_clinesNEWfilter$LG,Chr4_clinesNEWfilter$ColourGene)

colnames(Chr4_clinesNEWfilter)
Chr4_clinesNEWfilter[Chr4_clinesNEWfilter[,"position"]>38310000,colnames(Chr4_clinesNEWfilter)%in% c("position","dpthAdj_15","dpthAdj_16","dpthAdj_17","dpthAdj_18","dpthAdj_19","dpthAdj_20","width","ColourGene","distanceToGene")]

Chr4_clinesNEWfilter_nearSULF <- Chr4_clinesNEWfilter[Chr4_clinesNEWfilter$ColourGene=="SULF",]

Chr4_clinesNEWfilter_nearSULF[,colnames(Chr4_clinesNEWfilter_nearSULF)%in% c("position","width","ColourGene","distanceToGene")]
# comparing to some random clines with positive values in the middle position
thisLG <- 4
clines_thisLG <- Chr4_clinesNEWfilter %>%
                    filter(LG == thisLG)
numClines_LG4 <- nrow(clines_thisLG)
table_LG4 <- table(clines_thisLG$ColourGene,clines_thisLG$distanceToGene)

clines_thisLG <- clines_thisLG %>%
                    filter(AFD_15_20>= 0.9)
numClines_LG4_deltaP9 <- nrow(clines_thisLG)
table_LG4_deltaP9 <- table(clines_thisLG$ColourGene,clines_thisLG$distanceToGene)

# colour code clines by:
#   centre orientated
#   left leaning
#   right leaning
#   negative widths

par(mfrow=c(10,10))
par(mar=c(1,1,1,1))
par(oma=c(1,1,1,1))

for (thisRow in 1:nrow(clines_thisLG)) {
  # thisRow <- 1
  PoolSeq_summary <- cbind(poolSeqStats[2:7,],c(0,0.3,0.1,0.8,0.85,1.0))
  colnames(PoolSeq_summary)[ncol(PoolSeq_summary)] <- "p"
  PoolSeq_summary[,"p"] <- as.vector(unlist(clines_thisLG[thisRow,c(9:14)]))
  PoolSeq_summary <- cbind(PoolSeq_summary,(2*PoolSeq_summary[,"p"]*(1-PoolSeq_summary[,"p"])))
  colnames(PoolSeq_summary)[ncol(PoolSeq_summary)] <- "2pq"
  PoolSeq_summary <- PoolSeq_summary[order(PoolSeq_summary[,"DistAlongTransect"]),]
  plot(x=PoolSeq_summary[,"DistAlongTransect"],y=PoolSeq_summary[,"p"],bg = "black",xaxs = "i", yaxs = "i",xlab="", ylab="",main="",xlim=c(0,24),ylim=c(-0.1,1.1),type='l',cex.main=0.9,bty="n",yaxt="n",xaxt="n")
  axis(1,at=seq(0,24,1),labels=NA,col.axis="black",las=2,cex.axis=1.1,lwd=1.6,pos=-0.01, tck = -.02)
  axis(1,at=seq(0,24+1,2),labels=seq(0,24000,2000)/1000,col.axis="black",tck=-0.3,las=1,cex.axis=1.1,lwd=0,pos=-0.001)
  axis(2,at=seq(-0.01,1.01,0.1),labels=NA,col.axis="black",las=1,cex.axis=1.1,lwd=1.6,pos=0,tck = -.02)
  axis(2,at=seq(0,1.0,0.2),labels=seq(0,1,0.2),col.axis="black",tck=-.03,las=1,cex.axis=1.1,lwd=0,pos=5)
  lines(x=c(24,24),y=c(0,1),lwd=1.8)
  lines(x=c(0,24),y=c(1,1),lwd=1.8)
  title(paste(paste("LG",clines_thisLG[thisRow,"LG"]),clines_thisLG[thisRow,"position"],sep="- "), line=0, cex.main=0.8)
   lines(x = c(clines_thisLG[thisRow,'centre']/1000,clines_thisLG[thisRow,'centre']/1000), y = c(0,1),lty="dotted",lwd=1.3,col='black')
  lines(x = c(12.580,12.580), y = c(0,1),lty="dotted",lwd=1.3,col='red')
  points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"])
  lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1)
  
  thisWidth <- as.vector(unlist(clines_thisLG[thisRow,"width"]))
  text(x = 24, y = 0.28, labels = "w = ", cex = 1, col = "black", adj = 1)
  text(x = 23, y = 0.1, labels = sprintf("%.2f", as.numeric(thisWidth/1000)), cex = 1, col = "black", adj = 1)
    if (clines_thisLG[thisRow,"ColourGene"]!="-9") {
    # table(allChr_Clines_Dpth15[,"ColourGene"])
    # table(allChr_Clines_Dpth15[,"distanceToGene"])
    if (clines_thisLG[thisRow,"ColourGene"]=="CREMOSA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='green')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='green',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='green')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='green',bg='green')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="ELUTA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='pink')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='pink',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='pink')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='pink',bg='pink')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="ROSEA" | clines_thisLG[thisRow,"ColourGene"]=="ROS1" | clines_thisLG[thisRow,"ColourGene"]=="ROS2") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='red')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='red',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='red')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='red',bg='red')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="RUBIA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='blue')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='blue',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='blue')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='blue',bg='blue')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="FLAVIA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='yellow')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='yellow',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='yellow')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='yellow',bg='yellow')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="SULF") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='orange')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='orange',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='orange')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='orange',bg='orange')
      }
    }
    if (clines_thisLG[thisRow,"ColourGene"]=="AURINA") {
      if (clines_thisLG[thisRow,"distanceToGene"]!="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1,col='wheat')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='wheat',bg='white')
      }
      if (clines_thisLG[thisRow,"distanceToGene"]=="inGene") {
        lines(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],lwd=1.5,col='wheat')
        points(x=PoolSeq_summary[,"DistAlongTransect"]/1000,y=PoolSeq_summary[,"p"],pch=21,col='wheat',bg='wheat')
      }
    }
  }
}

cat('Number of clines LG4 deltaP >0.8 : ',numClines_LG4) 
cat('Number of clines LG4 deltaP >0.9 : ',numClines_LG4_deltaP9) 
cat('Gene breakdown LG4 deltaP >0.8 : ') 
table_LG4
cat('Gene breakdown LG4 deltaP >0.9 : ')
table_LG4_deltaP9

```

# Figures for final manuscript (March 2025 version)

Below are most of the figures for the manuscript, some designed with based R and custom functions. For some figures, a pdf automatically produced, for others figure appears in window and needs to be manually saved.

# Figure 1. map and clines
Sampling locations and allele frequencies for Antirrhinum majus hybrid zone that occurs from west (A. m. striatum) to east (A. m. pseudomajus) in the Spanish Pyrenees.

Panel produced below (Fig1 topography_map_ALL_years_Ros1) script dealing with KASP SNP genotype data. Then some manual editing in Inkscape  

Figure 1. Sampling locations and allele frequencies for Antirrhinum majus majus hybrid zone that occurs from varieties A. m. m. var. striatum (yellow) and A. m. m. var. pseudomajus (magenta) in the Spanish Pyrenees. Sampling locations of six poolSeq whole genomes (white diamonds) and allele frequencies at ROS from denser sampling of individuals where each pie diagram shows the proportion of the ROSps allele from pseudomajus (red) compared to the rosst allele from striatum (yellow) alleles at the ROS locus within a 200 metre diameter (see methods). Red arrow indicates the approximate centre of the phenotype cline. 

```{r Fig1 topography_map_ALL_years_Ros1,eval=F, include=F,fig.width=12, fig.height=16}
# note: figure has to be manually edited Inkscape/Illustrator to finalise

#Issues with package rgdal. Required the following steps:
# some relevant instructions here: https://candid.technology/configure-error-gdal-config-not-found-or-not-executable/

# jdk-18_macos-x64_bin.dmg (installation) OR
# brew install openjdk
# 1. brew install laszip
# 2. brew install --build-from-source gcc
# 3. xcode-select --install
# 4. brew install gdal
# 5. install devtools in R studio
# 6. install package 'remotes'
# 6. install_version("rgdal", version = "1.4-8", repos = "http://cran.us.r-project.org")

# when sourcing rgdal answer No to the question below:
#  There is a binary version available but the source version is later:
#      binary source needs_compilation
#rgdal 1.5-30 1.5-32              TRUE
# Do you want to install from sources the package which needs compilation? (Yes/no/cancel) no

# (Option a)
DEM_Baga <- raster(paste0("/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/manuscripts/Clines_Antirrhinum_WGS/PlosBiol submission/Github/InputData/","Baga_15_15m.txt"),header=TRUE,fill=FALSE)
DEM_Puigcerda <- raster(paste0("/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/manuscripts/Clines_Antirrhinum_WGS/PlosBiol submission/Github/InputData/","Puigcerda_15_15m.txt"),header=TRUE,fill=FALSE)
DEM_Ribes <- raster(paste0("/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/manuscripts/Clines_Antirrhinum_WGS/PlosBiol submission/Github/InputData/","Ribes_15_15m.txt"),header=TRUE,fill=FALSE)
DEM_Querabls <- raster(paste0("/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/manuscripts/Clines_Antirrhinum_WGS/PlosBiol submission/Github/InputData/","Querables_15_15m.txt"),header=TRUE,fill=FALSE)

DEM_merged <- merge(DEM_Baga,DEM_Puigcerda,DEM_Ribes,DEM_Querabls,filename='DEM_merged.grd',overwrite=TRUE)
rm(DEM_Baga,DEM_Puigcerda,DEM_Ribes,DEM_Querabls)

land.pal <- colorRampPalette(c("#336600", "#F3CA89", "#D9A627", "#A49019", "#9F7B0D", "#996600", 
                               "#B27676", "#C2B0B0", "#E5E5E5","#FFFFFF"))

# assumes HZ_50mDemes has been generated above for ros_assembly_543443
plotTerrainPiePlot(DataSetName = "Clines_Planoles",DEM_merged,SNPsFullData=GenoEcol, 
                   DemeData=HZ_50mDemes,locusToPlot="ros_assembly_543443",SNPlociListUpdated,minNum=5,
                   xRange=c(410000,435572.1),yRange=c(4684000,4691800),alphaTerm=1,land.pal,
                   colPalleteNum=40,poolsAdd=T,demeSize=100,plotPies=T,pieRadius=80,pointsInd="none",
                   ptSize=1.5,gridAdd=F,landMarks=T,fileFormat="pdf",markRoads=F)

```

# Figure 3a. Cline width x centre for all loci.
 
DeltaP 0.9 for main text Fig3a

```{r Fig 3a FINAL version in here -cline centre width scatter plot ggplot function }
# - colour scheme 3 (ROS/EL & FLA final manuscript version with histogram plot) 
#centreWidthScatterPlot_ggplot <- function(clineData,pool_deltaP, deltaP,ClineEst,plotQuantiles,filterNegativeClines,trimValuesCentre,colourTable, linkedAlpha) { #
  
  # colnames(clineData)
  # clineData <- allChr_Clines_filterDpth15; deltaP <- 0.9; pool_deltaP <- "AFD_15_20"; ClineEst = c("centre","width")
  # trimQuantiles=T; filterNegativeClines = T; plotQuantiles = T; trimValuesCentre <- c(7000,17000); 
  # colourTable <- colourChoice
  
  deltaP <- 0.9; filterNegativeClines <- T; pool_deltaP <- "AFD_15_20"; ClineEst = c('centre','width')
  trimValuesCentre <- c(7000,17000)
  cat("\nclinal loci start n=",nrow(allChr_Clines_Dpth15_delta9))
  
  clineData_filter <- allChr_Clines_Dpth15_delta9[allChr_Clines_Dpth15_delta9[,pool_deltaP]>=deltaP,]
  cat("\nclinal loci after delta p filter n=",nrow(clineData_filter))

  # nrow(clineData_filter)
  # remove negative cline widths & those >12km wide
  if (filterNegativeClines==T) {
      clineData_filter <- clineData_filter[clineData_filter[,ClineEst[2]]>0 & clineData_filter[,ClineEst[2]]<12000,]
  }
  cat("\nclinal loci after removing negative cline widths n=",nrow(clineData_filter))

  # trim cline centres
  if (!is.na(trimValuesCentre[1])) {
          clineData_filter_trimmed <- clineData_filter[clineData_filter[,ClineEst[1]]<=trimValuesCentre[1] | clineData_filter[,ClineEst[1]]>=trimValuesCentre[2],]
      clineData_filter <- clineData_filter[clineData_filter[,ClineEst[1]]>trimValuesCentre[1] & clineData_filter[,ClineEst[1]]<trimValuesCentre[2],]
  }

  cat("\nclinal loci after trim filter step changing loci n=",nrow(clineData_filter))
  range(clineData_filter[,ClineEst[2]])
  # quantiles
  quantiles_centre <- quantile(clineData_filter[,ClineEst[1]]/1000, na.rm=T, c(0.5, 0.05))
  quantiles_width <- quantile(clineData_filter[,ClineEst[2]]/1000, na.rm=T, c(0.5, 0.05))

  # separating each colour gene regions and background
  linkedAreas <- c("down30kb","down30_100kb","down30_400kb","up30_100kb","up30kb")

  # trimmed background
  
  # background loci
  allChr_Clines_WG_elsewhere <- clineData_filter %>% filter(ColourGene==-9)
  allChr_Clines_WG_elsewhere <- cbind(allChr_Clines_WG_elsewhere,rep("background",nrow(allChr_Clines_WG_elsewhere)))
  colnames(allChr_Clines_WG_elsewhere)[[ncol(allChr_Clines_WG_elsewhere)]] <- "colGene_plotDetails"
  
  nrow(allChr_Clines_WG_elsewhere) 
  
  # Chr 1 & CREMOSA 
  allChr_Clines_CREMOSA <- clineData_filter %>% filter(ColourGene=="CREMOSA" & distanceToGene =="inGene")
  allChr_Clines_CREMOSAlinked <- clineData_filter %>% filter(ColourGene=="CREMOSA") %>% filter(distanceToGene %in% linkedAreas)
  allChr_Clines_CREMOSA <- cbind(allChr_Clines_CREMOSA,rep("CREMOSA",nrow(allChr_Clines_CREMOSA)))
  colnames(allChr_Clines_CREMOSA)[[ncol(allChr_Clines_CREMOSA)]] <- "colGene_plotDetails"
  allChr_Clines_CREMOSAlinked <- cbind(allChr_Clines_CREMOSAlinked,rep("CREMOSAlinked",nrow(allChr_Clines_CREMOSAlinked)))
  colnames(allChr_Clines_CREMOSAlinked)[[ncol(allChr_Clines_CREMOSAlinked)]] <- "colGene_plotDetails"
  
  # Chr 2 & FLAVIA/Aurina 
  allChr_Clines_FLAVIA <- clineData_filter %>% filter(ColourGene=="FLAVIA" & distanceToGene =="inGene")
  allChr_Clines_FLAVIAlinked <- clineData_filter %>% filter(ColourGene=="FLAVIA") %>% filter(distanceToGene %in% linkedAreas) 
  nrow(allChr_Clines_FLAVIA)
  allChr_Clines_FLAVIA_cluster1 <- subset(allChr_Clines_FLAVIA, centre / 1000 < 10)  # First cluster: centre < 10
  allChr_Clines_FLAVIA_cluster2 <- subset(allChr_Clines_FLAVIA, centre / 1000 >= 10)  # Second cluster: centre >= 10

  allChr_Clines_AURINAlinked <- clineData_filter %>% filter(ColourGene=="AURINA") %>% filter(distanceToGene %in% linkedAreas) 
  allChr_Clines_FLAVIA <- cbind(allChr_Clines_FLAVIA,rep("FLAVIA",nrow(allChr_Clines_FLAVIA)))
  colnames(allChr_Clines_FLAVIA)[[ncol(allChr_Clines_FLAVIA)]] <- "colGene_plotDetails"
  allChr_Clines_FLAVIAlinked <- cbind(allChr_Clines_FLAVIAlinked,rep("FLAVIA",nrow(allChr_Clines_FLAVIAlinked)))
  colnames(allChr_Clines_FLAVIAlinked)[[ncol(allChr_Clines_FLAVIAlinked)]] <- "colGene_plotDetails"
  allChr_Clines_AURINAlinked <- cbind(allChr_Clines_AURINAlinked,rep("AURINAlinked",nrow(allChr_Clines_AURINAlinked)))
  colnames(allChr_Clines_AURINAlinked)[[ncol(allChr_Clines_AURINAlinked)]] <- "colGene_plotDetails"
  
  allChr_Clines_FLAVIA_cluster1 <- subset(allChr_Clines_FLAVIA, centre / 1000 < 10)  # First cluster: centre < 10
  allChr_Clines_FLAVIA_cluster2 <- subset(allChr_Clines_FLAVIA, centre / 1000 >= 10)  # Second cluster: centre >= 10
  allChr_Clines_FLAVIAlinked_cluster1 <- subset(allChr_Clines_FLAVIAlinked, centre / 1000 < 10)  # First cluster: centre < 10
  allChr_Clines_FLAVIAlinked_cluster2 <- subset(allChr_Clines_FLAVIAlinked, centre / 1000 >= 10)  # Second cluster: centre >= 10
  allChr_Clines_FLAVIA_cluster1[,"colGene_plotDetails"] <- "FLAVIA_cluster1"
  allChr_Clines_FLAVIA_cluster2[,"colGene_plotDetails"] <- "FLAVIA_cluster2"
  allChr_Clines_FLAVIAlinked_cluster1[,"colGene_plotDetails"] <- "FLAVIA_cluster1"
  allChr_Clines_FLAVIAlinked_cluster2[,"colGene_plotDetails"] <- "FLAVIA_cluster2"

  # Chr 3 
  allChr_Clines_Chr3 <- clineData_filter %>% filter(LG==3)
  allChr_Clines_Chr3_elsewhere <- clineData_filter %>% filter(LG==3 & ColourGene==-9)
  
  # Chr 4 & SULF  
  allChr_Clines_Chr4_elsewhere <- clineData_filter %>% filter(LG==4 & ColourGene==-9)
  allChr_Clines_SULF <- clineData_filter %>% filter(ColourGene=="SULF" & distanceToGene =="inGene")
  allChr_Clines_SULFlinked <- clineData_filter %>% filter(ColourGene=="SULF") %>% filter(distanceToGene %in% linkedAreas) 
  allChr_Clines_SULF <- cbind(allChr_Clines_SULF,rep("SULF",nrow(allChr_Clines_SULF)))
  colnames(allChr_Clines_SULF)[[ncol(allChr_Clines_SULF)]] <- "colGene_plotDetails"
  allChr_Clines_SULFlinked <- cbind(allChr_Clines_SULFlinked,rep("SULFlinked",nrow(allChr_Clines_SULFlinked)))
  colnames(allChr_Clines_SULFlinked)[[ncol(allChr_Clines_SULFlinked)]] <- "colGene_plotDetails"
  # Chr 5 & RUBIA 
  # 721 clines outside FLAVIA
  allChr_Clines_Chr5_elsewhere <- clineData_filter %>% filter(LG==5 & ColourGene==-9)
  allChr_Clines_RUBIA <- clineData_filter %>% filter(ColourGene=="RUBIA" & distanceToGene =="inGene")
  allChr_Clines_RUBIAlinked <- clineData_filter %>% filter(ColourGene=="RUBIA") %>% filter(distanceToGene %in% linkedAreas)
  allChr_Clines_RUBIA <- cbind(allChr_Clines_RUBIA,rep("RUBIA",nrow(allChr_Clines_RUBIA)))
  colnames(allChr_Clines_RUBIA)[[ncol(allChr_Clines_RUBIA)]] <- "colGene_plotDetails"
  allChr_Clines_RUBIAlinked <- cbind(allChr_Clines_RUBIAlinked,rep("RUBIAlinked",nrow(allChr_Clines_RUBIAlinked)))
  colnames(allChr_Clines_RUBIAlinked)[[ncol(allChr_Clines_RUBIAlinked)]] <- "colGene_plotDetails"
  # Chr 6 and ROS EL 
  allChr_Clines_Chr6_elsewhere <- clineData_filter %>% filter(LG==6 & ColourGene==-9)
  
  allChr_Clines_ROS1 <- clineData_filter %>% filter(ColourGene=="ROS1" & distanceToGene =="inGene")
  allChr_Clines_ROS2 <- clineData_filter %>% filter(ColourGene=="ROS2" & distanceToGene =="inGene")
  # allChr_Clines_ROS3 <- clineData_filter %>% filter(ColourGene=="ROS3" & distanceToGene =="inGene") # none in ROS3
  allChr_Clines_EL <- clineData_filter %>% filter(ColourGene=="ELUTA" & distanceToGene =="inGene")
  allChr_Clines_ROSelLinked <- clineData_filter %>% filter(ColourGene=="ROSEA" | ColourGene=="ROS1" | ColourGene=="ROS2" | ColourGene=="ELUTA") %>% filter(distanceToGene %in% linkedAreas) 
  
  allChr_Clines_ROS1 <- cbind(allChr_Clines_ROS1,rep("ROS",nrow(allChr_Clines_ROS1)))
  colnames(allChr_Clines_ROS1)[[ncol(allChr_Clines_ROS1)]] <- "colGene_plotDetails"
  allChr_Clines_ROS2 <- cbind(allChr_Clines_ROS2,rep("ROS",nrow(allChr_Clines_ROS2)))
  colnames(allChr_Clines_ROS2)[[ncol(allChr_Clines_ROS2)]] <- "colGene_plotDetails"
  allChr_Clines_EL <- cbind(allChr_Clines_EL,rep("EL",nrow(allChr_Clines_EL)))
  colnames(allChr_Clines_EL)[[ncol(allChr_Clines_EL)]] <- "colGene_plotDetails"
  allChr_Clines_ROSelLinked <- cbind(allChr_Clines_ROSelLinked,rep("ROSelLinked",nrow(allChr_Clines_ROSelLinked)))
  colnames(allChr_Clines_ROSelLinked)[[ncol(allChr_Clines_ROSelLinked)]] <- "colGene_plotDetails"
  
  # Chr 7 
  allChr_Clines_Chr7_elsewhere <- clineData_filter %>% filter(LG==7 & ColourGene==-9)

  # Chr 8 
  allChr_Clines_Chr8 <- clineData_filter %>% filter(LG==8)
  # combine all the 'elsewhere' loci not linked to known colour genes
  
  # combine all again
  allChr_Clines_newFiltered <- rbind(allChr_Clines_WG_elsewhere,allChr_Clines_CREMOSAlinked,allChr_Clines_CREMOSAlinked,allChr_Clines_FLAVIA_cluster1,allChr_Clines_FLAVIA_cluster2,allChr_Clines_FLAVIA_cluster1,allChr_Clines_FLAVIAlinked_cluster2,allChr_Clines_AURINAlinked,allChr_Clines_SULF,allChr_Clines_SULFlinked,allChr_Clines_RUBIA,allChr_Clines_RUBIAlinked,allChr_Clines_ROS1,allChr_Clines_ROS2,allChr_Clines_EL,allChr_Clines_ROSelLinked)
  
  unique(allChr_Clines_newFiltered[,"colGene_plotDetails"])
  allChr_Clines_newFiltered[,"colGene_plotDetails"] <- as.factor(allChr_Clines_newFiltered[,"colGene_plotDetails"])
  
  range(allChr_Clines_newFiltered[,"width"])
  range(allChr_Clines_newFiltered[,"centre"])
  unique_groups <- unique(allChr_Clines_newFiltered$colGene_plotDetails)
  custom_colors <- c(
  "allChr_Clines_WG_elsewhere" = "#E0E0E07F",  # lighter grey
  "CREMOSAlinked" = "deeppink2",
  "FLAVIA_cluster1" = "#4DAF4A",
  "FLAVIA_cluster2" = "#4DAF4A7F",
  "AURINAlinked" = "#FF7F00",
  "SULFlinked" = "#984EA3",
  "RUBIA" = "#377EB8",
  "RUBIAlinked" = rgb(228, 26, 28, maxColorValue = 255, alpha = 80),
  "ROS" = "#A65628",
  "EL" = "#A65628",
  "ROSelLinked" = "#A6562880"
)

# Main scatter plot
scatter_clines <- ggplot(allChr_Clines_newFiltered, aes(x = centre / 1000, y = width / 1000, color = colGene_plotDetails)) +
  geom_point(size = 1.5, alpha = 0.3, stroke = 1.4) +  # Thicker outlines using stroke
  geom_density_2d(aes(size = colGene_plotDetails), alpha = 0.6) +
  
  # Specific clusters with distinct dashed lines
  geom_density_2d(data = allChr_Clines_FLAVIAlinked_cluster1, color = "#4DAF4A", alpha = 0.9, size = 1, linetype = "solid") +
  geom_density_2d(data = allChr_Clines_FLAVIAlinked_cluster2, color = "#4DAF4A", alpha = 0.9, size = 1, linetype = "solid") +

  # Apply custom color and size scales
  scale_color_manual(values = custom_colors) +
  scale_size_manual(values = c(
    "FLAVIA" = 1, "RUBIA" = 1, "ROS" = 1,
    "EL" = 0.3, "CREMOSAlinked" = 1, "SULFlinked" = 1,
    "RUBIAlinked" = 0.01, "AURINAlinked" = 1,
    "ROSelLinked" = 0.4, "other_group" = 0.4
  )) +

  # Reverse the y-axis and customize tick labels to integers
  scale_y_reverse(
    breaks = scales::pretty_breaks(n = 10),
    labels = function(x) round(x)
  ) +

  # Customize x-axis limits and breaks
  scale_x_continuous(
    limits = c(7, 18),
    breaks = seq(7, 18, by = 1)
  ) +

  # Customize theme
  theme_minimal() +
  theme(
    axis.text = element_text(size = 24, color = "black"),
    axis.title = element_text(size = 20),
    axis.title.x = element_text(margin = margin(t = 10)),
    legend.title = element_blank(),
    legend.position = "bottom",
    legend.box.spacing = unit(1, "cm"),
    plot.margin = margin(10, 10, 50, 10),
    axis.line = element_line(color = "black", size = 0.5)
  ) +
  
  labs(
    title = "",
    x = "Cline Centre",
    y = "Cline Width"
  ) +
  
  geom_vline(xintercept = 12.5, linetype = "dashed", color = "black", size = 1)



pdf("CentreWidthScatterPlot_deltaP9_ggplot.pdf",width = 9, height = 9, pointsize = 12)
# Add marginal histograms
# Add marginal density plot
ggMarginal(scatter_clines, type = "density", groupColour = TRUE, groupFill = TRUE, bins = 20, na.rm = TRUE, size = 5)
dev.off()

```

```{r Fig 3a FINAL version in here -cline centre width scatter plot ggplot function modified version with parental adj}
# - colour scheme 3 (ROS/EL & FLA final manuscript version with histogram plot) 
#centreWidthScatterPlot_ggplot <- function(clineData,pool_deltaP, deltaP,ClineEst,plotQuantiles,filterNegativeClines,trimValuesCentre,colourTable, linkedAlpha) { #
  
  # colnames(clineData)
  # clineData <- allChr_Clines_filterDpth15; deltaP <- 0.9; pool_deltaP <- "AFD_15_20"; ClineEst = c("centre","width")
  # trimQuantiles=T; filterNegativeClines = T; plotQuantiles = T; trimValuesCentre <- c(7000,17000); 
  # colourTable <- colourChoice
  
  deltaP <- 0.9; filterNegativeClines <- T; pool_deltaP <- "AFD_15_20"; ClineEst = c('centre_Original','width_Original')
  trimValuesCentre <- c(7000,17000)
  cat("\nclinal loci start n=",nrow(allChr_Clines_Dpth15_delta9))
  
  clineData_filter <- allChr_Clines_Dpth15_delta9[allChr_Clines_Dpth15_delta9[,pool_deltaP]>=deltaP,]
  cat("\nclinal loci after delta p filter n=",nrow(clineData_filter))

  # nrow(clineData_filter)
  # remove negative cline widths & those >12km wide
  if (filterNegativeClines==T) {
      clineData_filter <- clineData_filter[clineData_filter[,ClineEst[2]]>0 & clineData_filter[,ClineEst[2]]<12000,]
  }
  cat("\nclinal loci after removing negative cline widths n=",nrow(clineData_filter))

  # trim cline centres
  if (!is.na(trimValuesCentre[1])) {
          clineData_filter_trimmed <- clineData_filter[clineData_filter[,ClineEst[1]]<=trimValuesCentre[1] | clineData_filter[,ClineEst[1]]>=trimValuesCentre[2],]
      clineData_filter <- clineData_filter[clineData_filter[,ClineEst[1]]>trimValuesCentre[1] & clineData_filter[,ClineEst[1]]<trimValuesCentre[2],]
  }

  cat("\nclinal loci after trim filter step changing loci n=",nrow(clineData_filter))
  range(clineData_filter[,ClineEst[2]])
  # quantiles
  quantiles_centre <- quantile(clineData_filter[,ClineEst[1]]/1000, na.rm=T, c(0.5, 0.05))
  quantiles_width <- quantile(clineData_filter[,ClineEst[2]]/1000, na.rm=T, c(0.5, 0.05))

  # separating each colour gene regions and background
  linkedAreas <- c("down30kb","down30_100kb","down30_400kb","up30_100kb","up30kb")

  # trimmed background
  
  # background loci
  allChr_Clines_WG_elsewhere <- clineData_filter %>% filter(ColourGene==-9)
  allChr_Clines_WG_elsewhere <- cbind(allChr_Clines_WG_elsewhere,rep("background",nrow(allChr_Clines_WG_elsewhere)))
  colnames(allChr_Clines_WG_elsewhere)[[ncol(allChr_Clines_WG_elsewhere)]] <- "colGene_plotDetails"
  
  nrow(allChr_Clines_WG_elsewhere) 
  
  # Chr 1 & CREMOSA 
  allChr_Clines_CREMOSA <- clineData_filter %>% filter(ColourGene=="CREMOSA" & distanceToGene =="inGene")
  allChr_Clines_CREMOSAlinked <- clineData_filter %>% filter(ColourGene=="CREMOSA") %>% filter(distanceToGene %in% linkedAreas)
  allChr_Clines_CREMOSA <- cbind(allChr_Clines_CREMOSA,rep("CREMOSA",nrow(allChr_Clines_CREMOSA)))
  colnames(allChr_Clines_CREMOSA)[[ncol(allChr_Clines_CREMOSA)]] <- "colGene_plotDetails"
  allChr_Clines_CREMOSAlinked <- cbind(allChr_Clines_CREMOSAlinked,rep("CREMOSAlinked",nrow(allChr_Clines_CREMOSAlinked)))
  colnames(allChr_Clines_CREMOSAlinked)[[ncol(allChr_Clines_CREMOSAlinked)]] <- "colGene_plotDetails"
  
  # Chr 2 & FLAVIA/Aurina 
  allChr_Clines_FLAVIA <- clineData_filter %>% filter(ColourGene=="FLAVIA" & distanceToGene =="inGene")
  allChr_Clines_FLAVIAlinked <- clineData_filter %>% filter(ColourGene=="FLAVIA") %>% filter(distanceToGene %in% linkedAreas) 
  nrow(allChr_Clines_FLAVIA)
  allChr_Clines_FLAVIA_cluster1 <- subset(allChr_Clines_FLAVIA, centre / 1000 < 10)  # First cluster: centre < 10
  allChr_Clines_FLAVIA_cluster2 <- subset(allChr_Clines_FLAVIA, centre / 1000 >= 10)  # Second cluster: centre >= 10

  allChr_Clines_AURINAlinked <- clineData_filter %>% filter(ColourGene=="AURINA") %>% filter(distanceToGene %in% linkedAreas) 
  allChr_Clines_FLAVIA <- cbind(allChr_Clines_FLAVIA,rep("FLAVIA",nrow(allChr_Clines_FLAVIA)))
  colnames(allChr_Clines_FLAVIA)[[ncol(allChr_Clines_FLAVIA)]] <- "colGene_plotDetails"
  allChr_Clines_FLAVIAlinked <- cbind(allChr_Clines_FLAVIAlinked,rep("FLAVIA",nrow(allChr_Clines_FLAVIAlinked)))
  colnames(allChr_Clines_FLAVIAlinked)[[ncol(allChr_Clines_FLAVIAlinked)]] <- "colGene_plotDetails"
  allChr_Clines_AURINAlinked <- cbind(allChr_Clines_AURINAlinked,rep("AURINAlinked",nrow(allChr_Clines_AURINAlinked)))
  colnames(allChr_Clines_AURINAlinked)[[ncol(allChr_Clines_AURINAlinked)]] <- "colGene_plotDetails"
  
  allChr_Clines_FLAVIA_cluster1 <- subset(allChr_Clines_FLAVIA, centre / 1000 < 10)  # First cluster: centre < 10
  allChr_Clines_FLAVIA_cluster2 <- subset(allChr_Clines_FLAVIA, centre / 1000 >= 10)  # Second cluster: centre >= 10
  allChr_Clines_FLAVIAlinked_cluster1 <- subset(allChr_Clines_FLAVIAlinked, centre / 1000 < 10)  # First cluster: centre < 10
  allChr_Clines_FLAVIAlinked_cluster2 <- subset(allChr_Clines_FLAVIAlinked, centre / 1000 >= 10)  # Second cluster: centre >= 10
  allChr_Clines_FLAVIA_cluster1[,"colGene_plotDetails"] <- "FLAVIA_cluster1"
  allChr_Clines_FLAVIA_cluster2[,"colGene_plotDetails"] <- "FLAVIA_cluster2"
  allChr_Clines_FLAVIAlinked_cluster1[,"colGene_plotDetails"] <- "FLAVIA_cluster1"
  allChr_Clines_FLAVIAlinked_cluster2[,"colGene_plotDetails"] <- "FLAVIA_cluster2"

  # Chr 3 
  allChr_Clines_Chr3 <- clineData_filter %>% filter(LG==3)
  allChr_Clines_Chr3_elsewhere <- clineData_filter %>% filter(LG==3 & ColourGene==-9)
  
  # Chr 4 & SULF  
  allChr_Clines_Chr4_elsewhere <- clineData_filter %>% filter(LG==4 & ColourGene==-9)
  allChr_Clines_SULF <- clineData_filter %>% filter(ColourGene=="SULF" & distanceToGene =="inGene")
  allChr_Clines_SULFlinked <- clineData_filter %>% filter(ColourGene=="SULF") %>% filter(distanceToGene %in% linkedAreas) 
  allChr_Clines_SULF <- cbind(allChr_Clines_SULF,rep("SULF",nrow(allChr_Clines_SULF)))
  colnames(allChr_Clines_SULF)[[ncol(allChr_Clines_SULF)]] <- "colGene_plotDetails"
  allChr_Clines_SULFlinked <- cbind(allChr_Clines_SULFlinked,rep("SULFlinked",nrow(allChr_Clines_SULFlinked)))
  colnames(allChr_Clines_SULFlinked)[[ncol(allChr_Clines_SULFlinked)]] <- "colGene_plotDetails"
  # Chr 5 & RUBIA 
  # 721 clines outside FLAVIA
  allChr_Clines_Chr5_elsewhere <- clineData_filter %>% filter(LG==5 & ColourGene==-9)
  allChr_Clines_RUBIA <- clineData_filter %>% filter(ColourGene=="RUBIA" & distanceToGene =="inGene")
  allChr_Clines_RUBIAlinked <- clineData_filter %>% filter(ColourGene=="RUBIA") %>% filter(distanceToGene %in% linkedAreas)
  allChr_Clines_RUBIA <- cbind(allChr_Clines_RUBIA,rep("RUBIA",nrow(allChr_Clines_RUBIA)))
  colnames(allChr_Clines_RUBIA)[[ncol(allChr_Clines_RUBIA)]] <- "colGene_plotDetails"
  allChr_Clines_RUBIAlinked <- cbind(allChr_Clines_RUBIAlinked,rep("RUBIAlinked",nrow(allChr_Clines_RUBIAlinked)))
  colnames(allChr_Clines_RUBIAlinked)[[ncol(allChr_Clines_RUBIAlinked)]] <- "colGene_plotDetails"
  # Chr 6 and ROS EL 
  allChr_Clines_Chr6_elsewhere <- clineData_filter %>% filter(LG==6 & ColourGene==-9)
  
  allChr_Clines_ROS1 <- clineData_filter %>% filter(ColourGene=="ROS1" & distanceToGene =="inGene")
  allChr_Clines_ROS2 <- clineData_filter %>% filter(ColourGene=="ROS2" & distanceToGene =="inGene")
  # allChr_Clines_ROS3 <- clineData_filter %>% filter(ColourGene=="ROS3" & distanceToGene =="inGene") # none in ROS3
  allChr_Clines_EL <- clineData_filter %>% filter(ColourGene=="ELUTA" & distanceToGene =="inGene")
  allChr_Clines_ROSelLinked <- clineData_filter %>% filter(ColourGene=="ROSEA" | ColourGene=="ROS1" | ColourGene=="ROS2" | ColourGene=="ELUTA") %>% filter(distanceToGene %in% linkedAreas) 
  
  allChr_Clines_ROS1 <- cbind(allChr_Clines_ROS1,rep("ROS",nrow(allChr_Clines_ROS1)))
  colnames(allChr_Clines_ROS1)[[ncol(allChr_Clines_ROS1)]] <- "colGene_plotDetails"
  allChr_Clines_ROS2 <- cbind(allChr_Clines_ROS2,rep("ROS",nrow(allChr_Clines_ROS2)))
  colnames(allChr_Clines_ROS2)[[ncol(allChr_Clines_ROS2)]] <- "colGene_plotDetails"
  allChr_Clines_EL <- cbind(allChr_Clines_EL,rep("EL",nrow(allChr_Clines_EL)))
  colnames(allChr_Clines_EL)[[ncol(allChr_Clines_EL)]] <- "colGene_plotDetails"
  allChr_Clines_ROSelLinked <- cbind(allChr_Clines_ROSelLinked,rep("ROSelLinked",nrow(allChr_Clines_ROSelLinked)))
  colnames(allChr_Clines_ROSelLinked)[[ncol(allChr_Clines_ROSelLinked)]] <- "colGene_plotDetails"
  
  # Chr 7 
  allChr_Clines_Chr7_elsewhere <- clineData_filter %>% filter(LG==7 & ColourGene==-9)

  # Chr 8 
  allChr_Clines_Chr8 <- clineData_filter %>% filter(LG==8)
  # combine all the 'elsewhere' loci not linked to known colour genes
  
  # combine all again
  allChr_Clines_newFiltered <- rbind(allChr_Clines_WG_elsewhere,allChr_Clines_CREMOSAlinked,allChr_Clines_CREMOSAlinked,allChr_Clines_FLAVIA_cluster1,allChr_Clines_FLAVIA_cluster2,allChr_Clines_FLAVIA_cluster1,allChr_Clines_FLAVIAlinked_cluster2,allChr_Clines_AURINAlinked,allChr_Clines_SULF,allChr_Clines_SULFlinked,allChr_Clines_RUBIA,allChr_Clines_RUBIAlinked,allChr_Clines_ROS1,allChr_Clines_ROS2,allChr_Clines_EL,allChr_Clines_ROSelLinked)
  
  unique(allChr_Clines_newFiltered[,"colGene_plotDetails"])
  allChr_Clines_newFiltered[,"colGene_plotDetails"] <- as.factor(allChr_Clines_newFiltered[,"colGene_plotDetails"])
  
  range(allChr_Clines_newFiltered[,"width_Original"])
  range(allChr_Clines_newFiltered[,"centre_Original"])
  unique_groups <- unique(allChr_Clines_newFiltered$colGene_plotDetails)
  custom_colors <- c(
  "allChr_Clines_WG_elsewhere" = "#E0E0E07F",  # lighter grey
  "CREMOSAlinked" = "deeppink2",
  "FLAVIA_cluster1" = "#4DAF4A",
  "FLAVIA_cluster2" = "#4DAF4A7F",
  "AURINAlinked" = "#FF7F00",
  "SULFlinked" = "#984EA3",
  "RUBIA" = "#377EB8",
  "RUBIAlinked" = rgb(228, 26, 28, maxColorValue = 255, alpha = 80),
  "ROS" = "#A65628",
  "EL" = "#A65628",
  "ROSelLinked" = "#A6562880"
)

# Main scatter plot
scatter_clines <- ggplot(allChr_Clines_newFiltered, aes(x = centre / 1000, y = width / 1000, color = colGene_plotDetails)) +
  geom_point(size = 1.5, alpha = 0.3, stroke = 1.4) +  # Thicker outlines using stroke
  geom_density_2d(aes(size = colGene_plotDetails), alpha = 0.6) +
  
  # Specific clusters with distinct dashed lines
  geom_density_2d(data = allChr_Clines_FLAVIAlinked_cluster1, color = "#4DAF4A", alpha = 0.9, size = 1, linetype = "solid") +
  geom_density_2d(data = allChr_Clines_FLAVIAlinked_cluster2, color = "#4DAF4A", alpha = 0.9, size = 1, linetype = "solid") +

  # Apply custom color and size scales
  scale_color_manual(values = custom_colors) +
  scale_size_manual(values = c(
    "FLAVIA" = 1, "RUBIA" = 1, "ROS" = 1,
    "EL" = 0.3, "CREMOSAlinked" = 1, "SULFlinked" = 1,
    "RUBIAlinked" = 0.01, "AURINAlinked" = 1,
    "ROSelLinked" = 0.4, "other_group" = 0.4
  )) +

  # Reverse the y-axis and customize tick labels to integers
  scale_y_reverse(
    breaks = scales::pretty_breaks(n = 10),
    labels = function(x) round(x)
  ) +

  # Customize x-axis limits and breaks
  scale_x_continuous(
    limits = c(7, 18),
    breaks = seq(7, 18, by = 1)
  ) +

  # Customize theme
  theme_minimal() +
  theme(
    axis.text = element_text(size = 24, color = "black"),
    axis.title = element_text(size = 20),
    axis.title.x = element_text(margin = margin(t = 10)),
    legend.title = element_blank(),
    legend.position = "bottom",
    legend.box.spacing = unit(1, "cm"),
    plot.margin = margin(10, 10, 50, 10),
    axis.line = element_line(color = "black", size = 0.5)
  ) +
  
  labs(
    title = "",
    x = "Cline Centre",
    y = "Cline Width"
  ) +
  
  geom_vline(xintercept = 12.5, linetype = "dashed", color = "black", size = 1)



pdf("CentreWidthScatterPlot_deltaP9_ggplot_padj.pdf",width = 9, height = 9, pointsize = 12)
# Add marginal histograms
# Add marginal density plot
ggMarginal(scatter_clines, type = "density", groupColour = TRUE, groupFill = TRUE, bins = 20, na.rm = TRUE, size = 5)
dev.off()

```

- note Fig 3b and 3c in SlowClines section and put together in inkscape manually

# Fig 4 - summary box plots of cline width in different genomic regions

```{r Fig 4a Fig box plot summary of cline properties in all geographic regions cline summaries colour }

# All located
poolSeqClines_All <- allChr_Clines_Dpth15_delta9

nrow(poolSeqClines_All)

## Step 1: Determine the 5th percentile of 'width'
lower_95_cutoff <- quantile(poolSeqClines_All$width, probs = 0.05)

# Step 2: Filter rows below the 5th percentile
lower_95_data <- poolSeqClines_All %>%
  filter(width <= lower_95_cutoff)

# Step 1: Rename '-9' to 'background'
poolSeqClines_All <- poolSeqClines_All %>%
  mutate(distanceToGene = recode(distanceToGene, "-9" = "background"))

# Step 2: Reclassify 'distanceToGene' based on your grouping criteria
poolSeqClines_All <- poolSeqClines_All %>%
  mutate(distanceToGene = case_when(
    distanceToGene %in% c("down30kb", "up30kb") ~ "<30kb",
    distanceToGene %in% c("down30_100kb", "up30_100kb") ~ "30 to 100kb",
    distanceToGene %in% c("down30_400kb", "up30_400kb") ~ "30 to 400kb",
    distanceToGene == "background" ~ "background",
    TRUE ~ distanceToGene
  )) %>%
  mutate(distanceToGene = factor(distanceToGene, levels = c("inGene", "<30kb", "30 to 100kb", "30 to 400kb", "background")))

# Step 3: Reorder the 'ColourGene' categories
poolSeqClines_All <- poolSeqClines_All %>%
  mutate(ColourGene = factor(ColourGene, levels = c("CREMOSA", "AURINA", "FLAVIA", "SULF", "RUBIA", "ROS1", "ROS2", "ROSEA", "ELUTA", "background")))

# Step 4: Remove rows with NA values in the 'width' column (if any)
poolSeqClines_All <- poolSeqClines_All %>%
  filter(!is.na(width))

# Step 5: Summarise data for annotations
summary_stats_by_gene <- poolSeqClines_All %>%
  group_by(ColourGene, distanceToGene) %>%
  summarise(
    avg_width = mean(width, na.rm = TRUE),
    sd_width = sd(width, na.rm = TRUE),
    n = n(),
    max_width = max(width, na.rm = TRUE)  # Capture max width for dynamic positioning
  )

ggplot(poolSeqClines_All, aes(x = ColourGene, y = width, fill = distanceToGene)) +
  geom_boxplot(outlier.colour = "red", outlier.shape = 16, outlier.size = 2) +

  # Add sample size annotations (without 'n=')
  geom_text(data = summary_stats_by_gene,
            aes(x = ColourGene, y = max_width + 5, label = n), # Just the number
            position = position_dodge(width = 0.75), # Stagger labels for groups
            vjust = -0.5, # Adjust position above the box
            size = 5, color = "black") +

  labs(
    title = "(a) all clines 7.5 - 16.5 km",
    x = "ColourGene",
    y = "Width (km)",
    fill = "Distance to Gene"
  ) +
  
  # Add vertical lines to separate ColourGene groups
  geom_vline(xintercept = 1:9 + 0.5, linetype = "dashed", color = "gray", alpha = 0.5, linewidth = 0.5) +
  
  # Customize axis, font sizes, and add axis lines
  theme_minimal() +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1, size = 16),
    axis.text.y = element_text(size = 16),
    axis.title.x = element_text(size = 18),
    axis.title.y = element_text(size = 18),
    legend.title = element_text(size = 18),
    legend.text = element_text(size = 16),
    axis.line.y = element_line(colour = "black", linewidth = 0.8),
    axis.line.x = element_line(colour = "black", linewidth = 0.8),
    axis.ticks.y = element_line(colour = "black", linewidth = 0.8),
    axis.ticks.x = element_line(colour = "black", linewidth = 0.8),
    panel.grid.major.y = element_blank(),
    panel.grid.major.x = element_blank()
  ) +

  # Adjust y-axis ticks and labels
  scale_y_continuous(
    limits = c(0, NA),
    breaks = scales::pretty_breaks(n = 10),
    labels = function(x) x / 1000
  ) +

  # Apply the Cividis color scale
  scale_fill_viridis_d(option = "cividis", direction = -1)

print(summary_stats_by_gene)




# : Summarise data for annotations LG
summary_stats_by_gene_LG <- poolSeqClines_All %>%
  group_by(LG, ColourGene) %>%
  summarise(
    avg_width = mean(width, na.rm = TRUE),
    sd_width = sd(width, na.rm = TRUE),
    n = n()
  )

print(summary_stats_by_gene_LG)
write.xlsx(summary_stats_by_gene_LG, "summary_stats_by_gene_LG.xlsx")

# make pdf 9 x 6
```

```{r Fig 4a Fig box plot summary of cline properties in all geographic regions cline summaries colour version with parental allele freq adj}

# All located
poolSeqClines_All <- allChr_Clines_Dpth15_delta9

nrow(poolSeqClines_All)

## Step 1: Determine the 5th percentile of 'width'
lower_95_cutoff <- quantile(poolSeqClines_All$width_Original, probs = 0.05)

# Step 2: Filter rows below the 5th percentile
lower_95_data <- poolSeqClines_All %>%
  filter(width_Original <= lower_95_cutoff)

# Step 1: Rename '-9' to 'background'
poolSeqClines_All <- poolSeqClines_All %>%
  mutate(distanceToGene = recode(distanceToGene, "-9" = "background"))

# Step 2: Reclassify 'distanceToGene' based on your grouping criteria
poolSeqClines_All <- poolSeqClines_All %>%
  mutate(distanceToGene = case_when(
    distanceToGene %in% c("down30kb", "up30kb") ~ "<30kb",
    distanceToGene %in% c("down30_100kb", "up30_100kb") ~ "30 to 100kb",
    distanceToGene %in% c("down30_400kb", "up30_400kb") ~ "30 to 400kb",
    distanceToGene == "background" ~ "background",
    TRUE ~ distanceToGene
  )) %>%
  mutate(distanceToGene = factor(distanceToGene, levels = c("inGene", "<30kb", "30 to 100kb", "30 to 400kb", "background")))

# Step 3: Reorder the 'ColourGene' categories
poolSeqClines_All <- poolSeqClines_All %>%
  mutate(ColourGene = factor(ColourGene, levels = c("CREMOSA", "AURINA", "FLAVIA", "SULF", "RUBIA", "ROS1", "ROS2", "ROSEA", "ELUTA", "background")))

# Step 4: Remove rows with NA values in the 'width' column (if any)
poolSeqClines_All <- poolSeqClines_All %>%
  filter(!is.na(width_Original))

# Step 5: Summarise data for annotations
summary_stats_by_gene <- poolSeqClines_All %>%
  group_by(ColourGene, distanceToGene) %>%
  summarise(
    avg_width = mean(width_Original, na.rm = TRUE),
    sd_width = sd(width_Original, na.rm = TRUE),
    n = n(),
    max_width = max(width_Original, na.rm = TRUE)  # Capture max width for dynamic positioning
  )

ggplot(poolSeqClines_All, aes(x = ColourGene, y = width_Original, fill = distanceToGene)) +
  geom_boxplot(outlier.colour = "red", outlier.shape = 16, outlier.size = 2) +

  # Add sample size annotations (without 'n=')
  geom_text(data = summary_stats_by_gene,
            aes(x = ColourGene, y = max_width + 5, label = n), # Just the number
            position = position_dodge(width = 0.75), # Stagger labels for groups
            vjust = -0.5, # Adjust position above the box
            size = 5, color = "black") +

  labs(
    title = "(a) all clines 7.5 - 16.5 km",
    x = "ColourGene",
    y = "Width (km)",
    fill = "Distance to Gene"
  ) +
  
  # Add vertical lines to separate ColourGene groups
  geom_vline(xintercept = 1:9 + 0.5, linetype = "dashed", color = "gray", alpha = 0.5, linewidth = 0.5) +
  
  # Customize axis, font sizes, and add axis lines
  theme_minimal() +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1, size = 16),
    axis.text.y = element_text(size = 16),
    axis.title.x = element_text(size = 18),
    axis.title.y = element_text(size = 18),
    legend.title = element_text(size = 18),
    legend.text = element_text(size = 16),
    axis.line.y = element_line(colour = "black", linewidth = 0.8),
    axis.line.x = element_line(colour = "black", linewidth = 0.8),
    axis.ticks.y = element_line(colour = "black", linewidth = 0.8),
    axis.ticks.x = element_line(colour = "black", linewidth = 0.8),
    panel.grid.major.y = element_blank(),
    panel.grid.major.x = element_blank()
  ) +

  # Adjust y-axis ticks and labels
  scale_y_continuous(
    limits = c(0, NA),
    breaks = scales::pretty_breaks(n = 10),
    labels = function(x) x / 1000
  ) +

  # Apply the Cividis color scale
  scale_fill_viridis_d(option = "cividis", direction = -1)

print(summary_stats_by_gene)




# : Summarise data for annotations LG
summary_stats_by_gene_LG <- poolSeqClines_All %>%
  group_by(LG, ColourGene) %>%
  summarise(
    avg_width = mean(width, na.rm = TRUE),
    sd_width = sd(width, na.rm = TRUE),
    n = n()
  )

print(summary_stats_by_gene_LG)
write.xlsx(summary_stats_by_gene_LG, "summary_stats_by_gene_LG.xlsx")

# make pdf 9 x 6
# new sizing 5 x 6

```

```{r Fig 4b box plotsummary of cline properties in different regions central area}

allChr_Clines_Dpth15_delta9
# centrally located
poolSeqClines_centralClines <- allChr_Clines_Dpth15_delta9[allChr_Clines_Dpth15_delta9[,"centre"]>11000 & allChr_Clines_Dpth15_delta9[,"centre"]<14000,]

nrow(poolSeqClines_centralClines)

## Step 1: Determine the 5th percentile of 'width'
lower_95_cutoff <- quantile(poolSeqClines_centralClines$width, probs = 0.05)

# Step 2: Filter rows below the 5th percentile
lower_95_data <- poolSeqClines_centralClines %>%
  filter(width <= lower_95_cutoff)

# Step 1: Rename '-9' to 'background'
poolSeqClines_centralClines <- poolSeqClines_centralClines %>%
  mutate(distanceToGene = recode(distanceToGene, "-9" = "background"))

# Step 2: Reclassify 'distanceToGene' based on your grouping criteria
poolSeqClines_centralClines <- poolSeqClines_centralClines %>%
  mutate(distanceToGene = case_when(
    distanceToGene %in% c("down30kb", "up30kb") ~ "<30kb",
    distanceToGene %in% c("down30_100kb", "up30_100kb") ~ "30 to 100kb",
    distanceToGene %in% c("down30_400kb", "up30_400kb") ~ "30 to 400kb",
    distanceToGene == "background" ~ "background",
    TRUE ~ distanceToGene
  )) %>%
  mutate(distanceToGene = factor(distanceToGene, levels = c("inGene", "<30kb", "30 to 100kb", "30 to 400kb", "background")))

# Step 3: Reorder the 'ColourGene' categories
poolSeqClines_centralClines <- poolSeqClines_centralClines %>%
  mutate(ColourGene = factor(ColourGene, levels = c("CREMOSA", "AURINA", "FLAVIA", "SULF", "RUBIA", "ROS1", "ROS2", "ROSEA", "ELUTA", "background")))

# Step 4: Remove rows with NA values in the 'width' column (if any)
poolSeqClines_centralClines <- poolSeqClines_centralClines %>%
  filter(!is.na(width))

# Step 5: Summarise data for annotations
summary_stats_by_gene <- poolSeqClines_centralClines %>%
  group_by(ColourGene, distanceToGene) %>%
  summarise(
    avg_width = mean(width, na.rm = TRUE),
    sd_width = sd(width, na.rm = TRUE),
    n = n(),
    max_width = max(width, na.rm = TRUE)  # Capture max width for dynamic positioning
  )

ggplot(poolSeqClines_centralClines, aes(x = ColourGene, y = width, fill = distanceToGene)) +
  geom_boxplot(outlier.colour = "red", outlier.shape = 16, outlier.size = 2) +

  # Add sample size annotations (without 'n=')
  geom_text(data = summary_stats_by_gene,
            aes(x = ColourGene, y = max_width + 5, label = n), # Just the number
            position = position_dodge(width = 0.75), # Stagger labels for groups
            vjust = -0.5, # Adjust position above the box
            size = 4.8, color = "black") +

  labs(
    title = "(b) central clines 11-14 km",
    x = "ColourGene",
    y = "Width (km)",
    fill = "Distance to Gene"
  ) +
  
  # Add vertical lines to separate ColourGene groups
  geom_vline(xintercept = 1:9 + 0.5, linetype = "dashed", color = "gray", alpha = 0.5, linewidth = 0.5) +
  
  # Customize axis, font sizes, and add axis lines
  theme_minimal() +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1, size = 22),  # ⬆️ Increased x-axis tick label font size
    axis.text.y = element_text(size = 22),  # ⬆️ Increased y-axis tick label font size
    axis.title.x = element_text(size = 24),  
    axis.title.y = element_text(size = 24),  
    legend.title = element_text(size = 20),  
    legend.text = element_text(size = 16),  
    axis.line.y = element_line(colour = "black", linewidth = 0.8),  
    axis.line.x = element_line(colour = "black", linewidth = 0.8),  
    axis.ticks.y = element_line(colour = "black", linewidth = 0.8),  
    axis.ticks.x = element_line(colour = "black", linewidth = 0.8),  
    panel.grid.major.y = element_blank(),  
    panel.grid.major.x = element_blank()
  ) +

  # Adjust y-axis ticks and labels
  scale_y_continuous(
    limits = c(0, NA),
    breaks = scales::pretty_breaks(n = 10),
    labels = function(x) x / 1000
  ) +

  # Apply the Cividis color scale
  scale_fill_viridis_d(option = "cividis", direction = -1)

print(summary_stats_by_gene)

```

I havent updated chunk below yet with parental allele freq adj, but still working as without adj. Little difference in summary stats

```{r Fig 4b box plotsummary of cline properties in different regions central area adj parental freq}

allChr_Clines_Dpth15_delta9
# centrally located
poolSeqClines_centralClines <- allChr_Clines_Dpth15_delta9[allChr_Clines_Dpth15_delta9[,"centre_Original"]>11000 & allChr_Clines_Dpth15_delta9[,"centre_Original"]<14000,]

nrow(poolSeqClines_centralClines)

## Step 1: Determine the 5th percentile of 'width'
lower_95_cutoff <- quantile(poolSeqClines_centralClines$width_Original, probs = 0.05)

# Step 2: Filter rows below the 5th percentile
lower_95_data <- poolSeqClines_centralClines %>%
  filter(width_Original <= lower_95_cutoff)

# Step 1: Rename '-9' to 'background'
poolSeqClines_centralClines <- poolSeqClines_centralClines %>%
  mutate(distanceToGene = recode(distanceToGene, "-9" = "background"))

# Step 2: Reclassify 'distanceToGene' based on your grouping criteria
poolSeqClines_centralClines <- poolSeqClines_centralClines %>%
  mutate(distanceToGene = case_when(
    distanceToGene %in% c("down30kb", "up30kb") ~ "<30kb",
    distanceToGene %in% c("down30_100kb", "up30_100kb") ~ "30 to 100kb",
    distanceToGene %in% c("down30_400kb", "up30_400kb") ~ "30 to 400kb",
    distanceToGene == "background" ~ "background",
    TRUE ~ distanceToGene
  )) %>%
  mutate(distanceToGene = factor(distanceToGene, levels = c("inGene", "<30kb", "30 to 100kb", "30 to 400kb", "background")))

# Step 3: Reorder the 'ColourGene' categories
poolSeqClines_centralClines <- poolSeqClines_centralClines %>%
  mutate(ColourGene = factor(ColourGene, levels = c("CREMOSA", "AURINA", "FLAVIA", "SULF", "RUBIA", "ROS1", "ROS2", "ROSEA", "ELUTA", "background")))

# Step 4: Remove rows with NA values in the 'width' column (if any)
poolSeqClines_centralClines <- poolSeqClines_centralClines %>%
  filter(!is.na(width_Original))

# Step 5: Summarise data for annotations
summary_stats_by_gene <- poolSeqClines_centralClines %>%
  group_by(ColourGene, distanceToGene) %>%
  summarise(
    avg_width = mean(width_Original, na.rm = TRUE),
    sd_width = sd(width_Original, na.rm = TRUE),
    n = n(),
    max_width = max(width_Original, na.rm = TRUE)  # Capture max width for dynamic positioning
  )

ggplot(poolSeqClines_centralClines, aes(x = ColourGene, y = width_Original, fill = distanceToGene)) +
  geom_boxplot(outlier.colour = "red", outlier.shape = 16, outlier.size = 2) +

  # Add sample size annotations (without 'n=')
  geom_text(data = summary_stats_by_gene,
            aes(x = ColourGene, y = max_width + 5, label = n), # Just the number
            position = position_dodge(width = 0.75), # Stagger labels for groups
            vjust = -0.5, # Adjust position above the box
            size = 4.8, color = "black") +

  labs(
    title = "(b) central clines 11-14 km",
    x = "ColourGene",
    y = "Width (km)",
    fill = "Distance to Gene"
  ) +
  
  # Add vertical lines to separate ColourGene groups
  geom_vline(xintercept = 1:9 + 0.5, linetype = "dashed", color = "gray", alpha = 0.5, linewidth = 0.5) +
  
  # Customize axis, font sizes, and add axis lines
  theme_minimal() +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1, size = 22),  # ⬆️ Increased x-axis tick label font size
    axis.text.y = element_text(size = 22),  # ⬆️ Increased y-axis tick label font size
    axis.title.x = element_text(size = 24),  
    axis.title.y = element_text(size = 24),  
    legend.title = element_text(size = 20),  
    legend.text = element_text(size = 16),  
    axis.line.y = element_line(colour = "black", linewidth = 0.8),  
    axis.line.x = element_line(colour = "black", linewidth = 0.8),  
    axis.ticks.y = element_line(colour = "black", linewidth = 0.8),  
    axis.ticks.x = element_line(colour = "black", linewidth = 0.8),  
    panel.grid.major.y = element_blank(),  
    panel.grid.major.x = element_blank()
  ) +

  # Adjust y-axis ticks and labels
  scale_y_continuous(
    limits = c(0, NA),
    breaks = scales::pretty_breaks(n = 10),
    labels = function(x) x / 1000
  ) +

  # Apply the Cividis color scale
  scale_fill_viridis_d(option = "cividis", direction = -1)

print(summary_stats_by_gene)

```

```{r Fig 4c box plotsummary of cline properties in different regions yellow flank area}

# left located
poolSeqClines_leftClines <- allChr_Clines_Dpth15_delta9[allChr_Clines_Dpth15_delta9[,"centre"]<11000 & allChr_Clines_Dpth15_delta9[,"centre"]>8000 ,]
nrow(poolSeqClines_leftClines)

## Step 1: Determine the 5th percentile of 'width'
lower_95_cutoff <- quantile(poolSeqClines_leftClines$width, probs = 0.05)

# Step 2: Filter rows below the 5th percentile
lower_95_data <- poolSeqClines_leftClines %>%
  filter(width <= lower_95_cutoff)

# Step 2: Reclassify 'distanceToGene' based on your grouping criteria
poolSeqClines_leftClines <- poolSeqClines_leftClines %>%
  mutate(distanceToGene = case_when(
    distanceToGene %in% c("down30kb", "up30kb") ~ "<30kb",
    distanceToGene %in% c("down30_100kb", "up30_100kb") ~ "30 to 100kb",
    distanceToGene %in% c("down30_400kb", "up30_400kb") ~ "30 to 400kb",
    distanceToGene == "-9" ~ "background",
    TRUE ~ distanceToGene
  ))

# Step 3: Set custom order for distanceToGene
poolSeqClines_leftClines <- poolSeqClines_leftClines %>%
  mutate(distanceToGene = factor(distanceToGene, levels = c("inGene", "<30kb", "30 to 100kb", "30 to 400kb", "background")))

# Step 3: Reorder the 'ColourGene' categories
poolSeqClines_leftClines <- poolSeqClines_leftClines %>%
  mutate(ColourGene = factor(ColourGene, levels = c("AURINA", "FLAVIA", "SULF", "RUBIA", "ROS1", "ROS2", "ROSEA", "ELUTA", "background")))

# Step 4: Remove rows with NA values in the 'width' column (if any)
poolSeqClines_leftClines <- poolSeqClines_leftClines %>%
  filter(!is.na(width))

# Step 4: Summarise data for annotations
summary_stats_by_gene <- poolSeqClines_leftClines %>%
  group_by(ColourGene, distanceToGene) %>%
  summarise(
    avg_width = mean(width, na.rm = TRUE),
    sd_width = sd(width, na.rm = TRUE),
    n = n()
  )

ggplot(poolSeqClines_leftClines, aes(x = ColourGene, y = width, fill = distanceToGene)) +
  geom_boxplot(outlier.colour = "red", outlier.shape = 16, outlier.size = 2) +

  # Add sample size annotations (without 'n=')
  geom_text(data = summary_stats_by_gene,
            aes(x = ColourGene, label = n), # Just the number
            position = position_dodge(width = 0.75), # Stagger labels for groups
            vjust = -0.5, # Adjust position above the box
            size = 4.8, color = "black") +

  labs(
    title = "(c) yellow flank 8-11 km",
    x = "ColourGene",
    y = "Width (km)",
    fill = "Distance to Gene"
  ) +
  
  # Add vertical lines to separate ColourGene groups
  geom_vline(xintercept = 1:9 + 0.5, linetype = "dashed", color = "gray", alpha = 0.5, linewidth = 0.5) +
  
  # Customize axis, font sizes, and add axis lines
  theme_minimal() +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1, size = 22),  # ⬆️ Increased x-axis tick label font size
    axis.text.y = element_text(size = 22),  # ⬆️ Increased y-axis tick label font size
    axis.title.x = element_text(size = 24),  
    axis.title.y = element_text(size = 24),  
    legend.title = element_text(size = 20),  
    legend.text = element_text(size = 16),  
    axis.line.y = element_line(colour = "black", linewidth = 0.8),  
    axis.line.x = element_line(colour = "black", linewidth = 0.8),  
    axis.ticks.y = element_line(colour = "black", linewidth = 0.8),  
    axis.ticks.x = element_line(colour = "black", linewidth = 0.8),  
    panel.grid.major.y = element_blank(),  
    panel.grid.major.x = element_blank()
  ) +

  # Adjust y-axis ticks and labels
  scale_y_continuous(
    limits = c(0, NA),
    breaks = scales::pretty_breaks(n = 10),
    labels = function(x) x / 1000
  ) +

  # Apply the Cividis color scale
  scale_fill_viridis_d(option = "cividis", direction = -1)

print(summary_stats_by_gene)

# : Summarise data for annotations LG
summary_stats_by_gene_LG_left <- poolSeqClines_leftClines %>%
  group_by(LG, ColourGene) %>%
  summarise(
    avg_width = mean(width, na.rm = TRUE),
    sd_width = sd(width, na.rm = TRUE),
    n = n()
  )
```

```{r Fig 4d box plotsummary of cline properties in different regions magenta flank area}

# left located
poolSeqClines_rightClines <- allChr_Clines_Dpth15_delta9[allChr_Clines_Dpth15_delta9[,"centre"]>14000 & allChr_Clines_Dpth15_delta9[,"centre"]<17000 ,]
nrow(poolSeqClines_rightClines)

## Step 1: Determine the 5th percentile of 'width'
lower_95_cutoff <- quantile(poolSeqClines_rightClines$width, probs = 0.05)

# Step 2: Filter rows below the 5th percentile
lower_95_data <- poolSeqClines_rightClines %>%
  filter(width <= lower_95_cutoff)

# Step 2: Reclassify 'distanceToGene' based on your grouping criteria
poolSeqClines_rightClines <- poolSeqClines_rightClines %>%
  mutate(distanceToGene = case_when(
    distanceToGene %in% c("down30kb", "up30kb") ~ "<30kb",
    distanceToGene %in% c("down30_100kb", "up30_100kb") ~ "30 to 100kb",
    distanceToGene %in% c("down30_400kb", "up30_400kb") ~ "30 to 400kb",
    distanceToGene == "-9" ~ "background",
    TRUE ~ distanceToGene
  ))

# Step 3: Set custom order for distanceToGene
poolSeqClines_rightClines <- poolSeqClines_rightClines %>%
  mutate(distanceToGene = factor(distanceToGene, levels = c("inGene", "<30kb", "30 to 100kb", "30 to 400kb", "background")))

# Step 3: Reorder the 'ColourGene' categories
poolSeqClines_rightClines <- poolSeqClines_rightClines %>%
  mutate(ColourGene = factor(ColourGene, levels = c("AURINA", "FLAVIA", "SULF", "RUBIA", "ROS1", "ROS2", "ROSEA", "ELUTA", "background")))

# Step 4: Remove rows with NA values in the 'width' column (if any)
poolSeqClines_rightClines <- poolSeqClines_rightClines %>%
  filter(!is.na(width))

# Step 4: Summarise data for annotations
summary_stats_by_gene <- poolSeqClines_rightClines %>%
  group_by(ColourGene, distanceToGene) %>%
  summarise(
    avg_width = mean(width, na.rm = TRUE),
    sd_width = sd(width, na.rm = TRUE),
    n = n()
  )


ggplot(poolSeqClines_rightClines, aes(x = ColourGene, y = width, fill = distanceToGene)) +
  geom_boxplot(outlier.colour = "red", outlier.shape = 16, outlier.size = 2) +
  labs(
    title = "(d) magenta flank clines 14-17 km",
    x = "ColourGene",
    y = "Width (km)",
    fill = "Distance to Gene"
  ) +
  
 # Add sample size annotations (without 'n=')
  geom_text(data = summary_stats_by_gene,
            aes(x = ColourGene, label = n), # Just the number
            position = position_dodge(width = 0.75), # Stagger labels for groups
            vjust = -0.5, # Adjust position above the box
            size = 4.8, color = "black") +

  labs(
    title = "(c) yellow flank 8-11 km",
    x = "ColourGene",
    y = "Width (km)",
    fill = "Distance to Gene"
  ) +
  
  # Add vertical lines to separate ColourGene groups
  geom_vline(xintercept = 1:9 + 0.5, linetype = "dashed", color = "gray", alpha = 0.5, linewidth = 0.5) +
  
  # Customize axis, font sizes, and add axis lines
  theme_minimal() +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1, size = 22),  # ⬆️ Increased x-axis tick label font size
    axis.text.y = element_text(size = 22),  # ⬆️ Increased y-axis tick label font size
    axis.title.x = element_text(size = 24),  
    axis.title.y = element_text(size = 24),  
    legend.title = element_text(size = 20),  
    legend.text = element_text(size = 16),  
    axis.line.y = element_line(colour = "black", linewidth = 0.8),  
    axis.line.x = element_line(colour = "black", linewidth = 0.8),  
    axis.ticks.y = element_line(colour = "black", linewidth = 0.8),  
    axis.ticks.x = element_line(colour = "black", linewidth = 0.8),  
    panel.grid.major.y = element_blank(),  
    panel.grid.major.x = element_blank()
  ) +

  # Adjust y-axis ticks and labels
  scale_y_continuous(
    limits = c(0, NA),
    breaks = scales::pretty_breaks(n = 10),
    labels = function(x) x / 1000
  ) +

  # Apply the Cividis color scale
  scale_fill_viridis_d(option = "cividis", direction = -1)

print(summary_stats_by_gene)



```

# Figure 5. Genome scans 

Summary 
a) cline width
b) cline centre
c) proportion of windows with clines (in 1Mbp or 500kb blocks) 
d) Fst YP4 x MP11
e) Fst YP2 x MP4
f) Fst YP2 x YP1
g) Fst MP4 x MP11

Figure 5. Genome scans between whole genome pools of A. m. m. var. striatum (yellow) and A. m. m var. pseudomajus (magenta). Panels show (A) cline width with FastClines, (B) cline centre with FastClines (C) cline frequency in windows (from FastClines) in 10kb windows, (D) FST pools YP4 x MP11 in 10kb windows (different colours and over mountain pass), (E) FST pools YP2 x MP4 (different colours and no mountain pass), (F) FST pools YP2 x YP1 (same colours and no mountain pass), (G) FST pools MP4 x MP11 (same colours and no mountain pass). Dashed vertical lines indicate phenotype cline centre (red), and 99th% and 95th% quantiles for FST panels (black dashed). Insert in panel (D) indicates relative positions of the six pools along the transect (in km). Window FST within 100kb of colour loci are coloured separately for CRE (pink), AUN (orange), FLA (green), SULF (purple), RUB (blue), ROS and EL (brown).

```{r proportion of windows with clines function}
# import Seans analysis

ClinesProportionsWindows <- read.csv(paste0('/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/manuscripts/Clines_Antirrhinum_WGS/PlosBiol submission/Github/InputData/','prop_clines_along_genome.csv'), header=T,fill=T,na.strings = "NA")

(nrow(ClinesProportionsWindows[ClinesProportionsWindows[,"sums"]>0,])/nrow(ClinesProportionsWindows))*100
ClinesProportionsWindows_nonZero<- ClinesProportionsWindows[ClinesProportionsWindows[,"sums"]>0,]
mean(ClinesProportionsWindows_nonZero[,"sums"])

genomeScanClineProportions <- function(clineData,ClinePropData,WindowData,deltaP,pool_deltaP,ClineEst,filterPass,filterNegativeClines, plotPos1,startPlot,colourTable) {
  #
  # clineData <- allChr_Clines_filterDpth15; ClinePropData <- ClinesProportionsWindows; WindowData <- allChr_Windows; deltaP <- 0.9; pool_deltaP <- "AFD_15_20"; startPlot = T
  # plotPos1 = c(0.0, 1.0, 0.85, 1)
  # ClineEst = c('centre','width'); filterNegativeClines = T; filterPass = T; colourTable=colourChoice

  #cat("\nclinal loci start n=",nrow(clineData))

  #clineData_filter <- clineData[clineData[,pool_deltaP]>=deltaP,]
  #cat("\nclinal loci after delta p filter n=",nrow(clineData_filter))

    # filter pass
  nrow(ClinePropData)
  if (filterPass==T) {
    ClinePropData <- ClinePropData %>% filter(less_300kb=="Pass")
  }
  
  # commence plot
  par(mar=c(3,3,4,4))
  par(oma=c(1,1,1,1))
  par(fig = plotPos1, mar=c(1,1,1,1),new=((startPlot==TRUE)==FALSE))
  plot(NA, xlab="", xaxt="n", ylab="", main="", las=2, bty="n",cex=0.5,yaxt="n",xaxt="n",type='n',
       xlim=c(-5500000,max(range(clineData[,"position_graph"]))+100000), ylim=c(-0.1,1.1),
       cex.axis=1.4,cex.lab=1.4,col=rgb(100,100,100,120,maxColorValue=255), pch=16)
  axis(2, at=c(0,0.1,0.2,0.3,0.4,0.5,0.6,0.7,0.8,0.9,1.0), labels=NA,col.axis="black", las=1,cex.axis=1.1,lwd=1.6,pos=-3000000, tck = -.03)
  axis(2, at=c(0,0.5,1.0), labels=c(0,0.5,1.0),col.axis="black", las=1,cex.axis=1.1,lwd=0,pos=-100000)
  makeBaseLG_ClineProp <- function(graphPos) {
    lines(x=c(range(graphPos[graphPos[,"LG"]==1,"windowMid_graph"])[1], range(graphPos[graphPos[,"LG"]==1,"windowMid_graph"])[2]), y=c(-0.02,-0.02), col = "black", lwd=4.5,lty = 1)
    lines(x=c(range(graphPos[graphPos[,"LG"]==2,"windowMid_graph"])[1], range(graphPos[graphPos[,"LG"]==2,"windowMid_graph"])[2]), y=c(-0.02,-0.02), col = "black", lwd=4.5,lty = 1)
    lines(x=c(range(graphPos[graphPos[,"LG"]==3,"windowMid_graph"])[1], range(graphPos[graphPos[,"LG"]==3,"windowMid_graph"])[2]), y=c(-0.02,-0.02), col = "black", lwd=4.5,lty = 1)
    lines(x=c(range(graphPos[graphPos[,"LG"]==4,"windowMid_graph"])[1], range(graphPos[graphPos[,"LG"]==4,"windowMid_graph"])[2]),y=c(-0.02,-0.02),col = "black", lwd=4.5,lty = 1)
    lines(x=c(range(graphPos[graphPos[,"LG"]==5,"windowMid_graph"])[1], range(graphPos[graphPos[,"LG"]==5,"windowMid_graph"])[2]),y=c(-0.02,-0.02),col = "black", lwd=4.5,lty = 1)
    lines(x=c(range(graphPos[graphPos[,"LG"]==6,"windowMid_graph"])[1], range(graphPos[graphPos[,"LG"]==6,"windowMid_graph"])[2]), y=c(-0.02,-0.02),col = "black", lwd=4.5,lty = 1)
    lines(x=c(range(graphPos[graphPos[,"LG"]==7,"windowMid_graph"])[1], range(graphPos[graphPos[,"LG"]==7,"windowMid_graph"])[2]),y=c(-0.02,-0.02),col = "black", lwd=4.5,lty = 1)
    lines(x=c(range(graphPos[graphPos[,"LG"]==8,"windowMid_graph"])[1], range(graphPos[graphPos[,"LG"]==8,"windowMid_graph"])[2]),y=c(-0.02,-0.02),col = "black", lwd=4.5,lty = 1)
  }
  makeBaseLG_ClineProp(WindowData)
  mtext("cline windows",side=2,cex=1.3,line=0.5)
  
  # add histograms/lines
  
  points(ClinePropData$block_mid,ClinePropData$sums/10, type="h", lwd=3, col="black")
  
}

```

```{r cline width and centre genome scan whole genome functions}
genomeScanClineWidth <- function(clineData,WindowData,deltaP,pool_deltaP,ClineEst,filterNegativeClines, plotPos1,startPlot,colourTable,lwdPoints,sizePoints) {
  #
  # clineData <- allChr_Clines_filterDpth15; WindowData <- allChr_Windows; deltaP <- 0.9; pool_deltaP <- "AFD_15_20"; startPlot = T
  # plotPos1 <- c(0.03, 1.0, 0.69, 0.85)
  # ClineEst = c('centre',width'); filterNegativeClines = T; colourTable=colourChoice

  cat("\nclinal loci start n=",nrow(clineData))

  clineData_filter <- clineData[clineData[,pool_deltaP]>=deltaP,]
  cat("\nclinal loci after delta p filter n=",nrow(clineData_filter))

  # nrow(clineData_filter)
  # remove negative cline widths & those >12km wide
  if (filterNegativeClines==T) {
      clineData_filter <- clineData_filter[clineData_filter[,ClineEst[2]]>0 & clineData_filter[,ClineEst[2]]<12000,]
  }
  clineData_filter[,ClineEst[2]] <- as.numeric(clineData_filter[,ClineEst[2]])/1000

  # WidthQuantiles <- quantile(clineData[,"width"], na.rm=T, c(0.5, 0.05))
  # lines(x=c(0,max(range(graphPos[,"windowMid_graph"]))),y=c(as.numeric(WidthQuantiles[1]),as.numeric(WidthQuantiles[1]))/1000,col = "black", lwd=1.5,lty = 3)
  # lines(x=c(0,max(range(graphPos[,"windowMid_graph"]))),y=c(as.numeric(WidthQuantiles[2]),as.numeric(WidthQuantiles[2]))/1000,col = "black", lwd=1.5,lty = 3)
  
  # background loci
  allChr_Clines_WG_elsewhere <- clineData_filter %>% filter(ColourGene==-9)
  
  # Chr 1 & CREMOSA 
  allChr_Clines_CREMOSA <- clineData_filter %>% filter(ColourGene=="CREMOSA" & distanceToGene =="inGene")
  allChr_Clines_CREMOSAlinked <- clineData_filter %>% filter(ColourGene=="CREMOSA") %>% filter(distanceToGene %in% linkedAreas)
  
  # Chr 2 & FLAVIA/Aurina 
  allChr_Clines_FLAVIA <- clineData_filter %>% filter(ColourGene=="FLAVIA" & distanceToGene =="inGene")
  allChr_Clines_FLAVIAlinked <- clineData_filter %>% filter(ColourGene=="FLAVIA") %>% filter(distanceToGene %in% linkedAreas) 
  allChr_Clines_AURINAlinked <- clineData_filter %>% filter(ColourGene=="AURINA") %>% filter(distanceToGene %in% linkedAreas) 

  # Chr 3 
  allChr_Clines_Chr3 <- clineData_filter %>% filter(LG==3)
  allChr_Clines_Chr3_elsewhere <- clineData_filter %>% filter(LG==3 & ColourGene==-9)
  
  # Chr 4 & SULF  
  allChr_Clines_Chr4_elsewhere <- clineData_filter %>% filter(LG==4 & ColourGene==-9)
  allChr_Clines_SULF <- clineData_filter %>% filter(ColourGene=="SULF" & distanceToGene =="inGene")
  allChr_Clines_SULFlinked <- clineData_filter %>% filter(ColourGene=="SULF") %>% filter(distanceToGene %in% linkedAreas) 
  
  # Chr 5 & RUBIA 
  # 721 clines outside FLAVIA
  allChr_Clines_Chr5_elsewhere <- clineData_filter %>% filter(LG==5 & ColourGene==-9)
  allChr_Clines_RUBIA <- clineData_filter %>% filter(ColourGene=="RUBIA" & distanceToGene =="inGene")
  allChr_Clines_RUBIAlinked <- clineData_filter %>% filter(ColourGene=="RUBIA") %>% filter(distanceToGene %in% linkedAreas)

  # Chr 6 and ROS EL 
  allChr_Clines_Chr6_elsewhere <- clineData_filter %>% filter(LG==6 & ColourGene==-9)
  
  allChr_Clines_ROS1 <- clineData_filter %>% filter(ColourGene=="ROS1" & distanceToGene =="inGene")
  allChr_Clines_ROS2 <- clineData_filter %>% filter(ColourGene=="ROS2" & distanceToGene =="inGene")
  # allChr_Clines_ROS3 <- clineData_filter %>% filter(ColourGene=="ROS3" & distanceToGene =="inGene") # none in ROS3
  allChr_Clines_EL <- clineData_filter %>% filter(ColourGene=="ELUTA" & distanceToGene =="inGene")
  allChr_Clines_ROSelLinked <- clineData_filter %>% filter(ColourGene=="ROSEA" | ColourGene=="ROS1" | ColourGene=="ROS2" | ColourGene=="ELUTA") %>% filter(distanceToGene %in% linkedAreas) 
  
  # Chr 7 
  allChr_Clines_Chr7_elsewhere <- clineData_filter %>% filter(LG==7 & ColourGene==-9)

  # Chr 8 
  allChr_Clines_Chr8 <- clineData_filter %>% filter(LG==8)
  # combine all the 'elsewhere' loci not linked to known colour genes
  
  # commence plot
  par(mar=c(3,3,4,4))
  par(oma=c(1,1,1,1))
  par(fig = plotPos1, mar=c(1,1,1,1),new=((startPlot==TRUE)==FALSE))
  plot(NA, xlab="", xaxt="n", ylab="", main="", las=2, bty="n",cex=0.5,yaxt="n",xaxt="n",type='n',
       xlim=c(-5500000,max(range(clineData[,"position_graph"]))+100000), ylim=rev(c(-1.2,12.2)),
       cex.axis=1.4,cex.lab=1.4,col=rgb(100,100,100,120,maxColorValue=255), pch=16)
  axis(2, at=rev(c(0,2,4,6,8,10,12)), labels=NA,col.axis="black", las=1,cex.axis=1.1,lwd=1.6,pos=-3000000, tck = -.03)
  axis(2, at=rev(c(0,2,4,6,8,10,12)), labels=rev(c(0,2,4,6,8,10,12)), col.axis="black", las=1, cex.axis=1.1, lwd=0, pos=-100000)
  makeBaseLG_ClineWidth <- function(graphPos) {
    lines(x=c(range(graphPos[graphPos[,"LG"]==1,"windowMid_graph"])[1], range(graphPos[graphPos[,"LG"]==1,"windowMid_graph"])[2]), y=c(12.1,12.1), col = "black", lwd=4.5,lty = 1)
    lines(x=c(range(graphPos[graphPos[,"LG"]==2,"windowMid_graph"])[1], range(graphPos[graphPos[,"LG"]==2,"windowMid_graph"])[2]), y=c(12.1,12.1), col = "black", lwd=4.5,lty = 1)
    lines(x=c(range(graphPos[graphPos[,"LG"]==3,"windowMid_graph"])[1], range(graphPos[graphPos[,"LG"]==3,"windowMid_graph"])[2]), y=c(12.1,12.1), col = "black", lwd=4.5,lty = 1)
    lines(x=c(range(graphPos[graphPos[,"LG"]==4,"windowMid_graph"])[1], range(graphPos[graphPos[,"LG"]==4,"windowMid_graph"])[2]),y=c(12.1,12.1),col = "black", lwd=4.5,lty = 1)
    lines(x=c(range(graphPos[graphPos[,"LG"]==5,"windowMid_graph"])[1], range(graphPos[graphPos[,"LG"]==5,"windowMid_graph"])[2]),y=c(12.1,12.1),col = "black", lwd=4.5,lty = 1)
    lines(x=c(range(graphPos[graphPos[,"LG"]==6,"windowMid_graph"])[1], range(graphPos[graphPos[,"LG"]==6,"windowMid_graph"])[2]), y=c(12.1,12.1),col = "black", lwd=4.5,lty = 1)
    lines(x=c(range(graphPos[graphPos[,"LG"]==7,"windowMid_graph"])[1], range(graphPos[graphPos[,"LG"]==7,"windowMid_graph"])[2]),y=c(12.1,12.1),col = "black", lwd=4.5,lty = 1)
    lines(x=c(range(graphPos[graphPos[,"LG"]==8,"windowMid_graph"])[1], range(graphPos[graphPos[,"LG"]==8,"windowMid_graph"])[2]),y=c(12.1,12.1),col = "black", lwd=4.5,lty = 1)
  }
  makeBaseLG_ClineWidth(WindowData)
  mtext("width",side=2,cex=1.3,line=0.5)
  
  #range(WindowData[WindowData[,"LG"]==1,"windowMid_graph"])
  #range(WindowData[WindowData[,"LG"]==2,"windowMid_graph"])

  # add points
  # all windows
  #points(allChr_Clines_WG_elsewhere[,"position_graph"], allChr_Clines_WG_elsewhere[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="background","pointColour"], lwd=1, pch=21,cex=0.7)
  # points(SNPfreqsPool_allElse_0.9DeltaP_Fstats[,"newPlotPosCline"], 
  #        as.numeric(SNPfreqsPool_allElse_0.9DeltaP_Fstats[,"width_6pool_est3"])/1000,
  #        col=rgb(100,100,100,120,maxColorValue=255), lwd=1, pch=16,cex=0.8)
  
# add points
 
  # background
   points(allChr_Clines_WG_elsewhere[,"position_graph"], allChr_Clines_WG_elsewhere[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="background","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
  
  # CRE and linked
   points(allChr_Clines_CREMOSAlinked[,"position_graph"], allChr_Clines_CREMOSAlinked[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="CREMOSAlinked","pointColour"], lwd=0.2, pch=21,cex=0.7)
    points(allChr_Clines_CREMOSA[,"position_graph"], allChr_Clines_CREMOSA[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="CREMOSA","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
  
  #  FLAVIA/Aurina 
   points(allChr_Clines_FLAVIAlinked[,"position_graph"], allChr_Clines_FLAVIAlinked[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="FLAVIAlinked","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
   
   points(allChr_Clines_AURINAlinked[,"position_graph"], allChr_Clines_AURINAlinked[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="AURINAlinked","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
   
    points(allChr_Clines_FLAVIA[,"position_graph"], allChr_Clines_FLAVIA[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="FLAVIA","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
  
    # SULF
     points(allChr_Clines_SULFlinked[,"position_graph"], allChr_Clines_SULFlinked[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="SULFlinked","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
    
     # RUBIA
   points(allChr_Clines_RUBIAlinked[,"position_graph"], allChr_Clines_RUBIAlinked[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="RUBIAlinked","pointColour"], lwd=0.2, pch=21,cex=0.7)
   points(allChr_Clines_RUBIA[,"position_graph"], allChr_Clines_RUBIA[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="RUBIA","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
   
      # ROS EL
   points(allChr_Clines_ROSelLinked[,"position_graph"], allChr_Clines_ROSelLinked[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="ROSelLinked","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
         points(allChr_Clines_EL[,"position_graph"], allChr_Clines_EL[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="ELUTA","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
            points(allChr_Clines_ROS1[,"position_graph"], allChr_Clines_ROS1[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="ROS","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
      points(allChr_Clines_ROS2[,"position_graph"], allChr_Clines_ROS2[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="ROS","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
}

genomeScanClineCentre <- function(clineData,WindowData,deltaP,pool_deltaP,ClineEst,filterNegativeClines, plotPos1,startPlot,colourTable,lwdPoints,sizePoints) {

  # plotPos1 <- c(0.03, 1.0, 0.54, 0.70); 
  # clineData <- allChr_Clines_filterDpth15; WindowData <- allChr_Windows; deltaP <- 0.9; pool_deltaP <- "AFD_15_20"; startPlot = F
  # ClineEst = c('centre','width'); filterNegativeClines = T; colourTable=colourChoice

  cat("\nclinal loci start n=",nrow(clineData))

  clineData_filter <- clineData[clineData[,pool_deltaP]>=deltaP,]
  cat("\nclinal loci after delta p filter n=",nrow(clineData_filter))

  # nrow(clineData_filter)
  # remove negative cline widths & those >12km wide
  if (filterNegativeClines==T) {
      clineData_filter <- clineData_filter[clineData_filter[,ClineEst[2]]>0 & clineData_filter[,ClineEst[2]]<12000,]
  }
  clineData_filter[,ClineEst[1]] <- as.numeric(clineData_filter[,ClineEst[1]])/1000

  # WidthQuantiles <- quantile(clineData[,"width"], na.rm=T, c(0.5, 0.05))
  # lines(x=c(0,max(range(graphPos[,"windowMid_graph"]))),y=c(as.numeric(WidthQuantiles[1]),as.numeric(WidthQuantiles[1]))/1000,col = "black", lwd=1.5,lty = 3)
  # lines(x=c(0,max(range(graphPos[,"windowMid_graph"]))),y=c(as.numeric(WidthQuantiles[2]),as.numeric(WidthQuantiles[2]))/1000,col = "black", lwd=1.5,lty = 3)
  
  # background loci
  allChr_Clines_WG_elsewhere <- clineData_filter %>% filter(ColourGene==-9)
  
  # Chr 1 & CREMOSA 
  allChr_Clines_CREMOSA <- clineData_filter %>% filter(ColourGene=="CREMOSA" & distanceToGene =="inGene")
  allChr_Clines_CREMOSAlinked <- clineData_filter %>% filter(ColourGene=="CREMOSA") %>% filter(distanceToGene %in% linkedAreas)
  
  # Chr 2 & FLAVIA/Aurina 
  allChr_Clines_FLAVIA <- clineData_filter %>% filter(ColourGene=="FLAVIA" & distanceToGene =="inGene")
  allChr_Clines_FLAVIAlinked <- clineData_filter %>% filter(ColourGene=="FLAVIA") %>% filter(distanceToGene %in% linkedAreas) 
  allChr_Clines_AURINAlinked <- clineData_filter %>% filter(ColourGene=="AURINA") %>% filter(distanceToGene %in% linkedAreas) 

  # Chr 3 
  allChr_Clines_Chr3 <- clineData_filter %>% filter(LG==3)
  allChr_Clines_Chr3_elsewhere <- clineData_filter %>% filter(LG==3 & ColourGene==-9)
  
  # Chr 4 & SULF  
  allChr_Clines_Chr4_elsewhere <- clineData_filter %>% filter(LG==4 & ColourGene==-9)
  allChr_Clines_SULF <- clineData_filter %>% filter(ColourGene=="SULF" & distanceToGene =="inGene")
  allChr_Clines_SULFlinked <- clineData_filter %>% filter(ColourGene=="SULF") %>% filter(distanceToGene %in% linkedAreas) 
  
  # Chr 5 & RUBIA 
  # 721 clines outside FLAVIA
  allChr_Clines_Chr5_elsewhere <- clineData_filter %>% filter(LG==5 & ColourGene==-9)
  allChr_Clines_RUBIA <- clineData_filter %>% filter(ColourGene=="RUBIA" & distanceToGene =="inGene")
  allChr_Clines_RUBIAlinked <- clineData_filter %>% filter(ColourGene=="RUBIA") %>% filter(distanceToGene %in% linkedAreas)

  # Chr 6 and ROS EL 
  allChr_Clines_Chr6_elsewhere <- clineData_filter %>% filter(LG==6 & ColourGene==-9)
  
  allChr_Clines_ROS1 <- clineData_filter %>% filter(ColourGene=="ROS1" & distanceToGene =="inGene")
  allChr_Clines_ROS2 <- clineData_filter %>% filter(ColourGene=="ROS2" & distanceToGene =="inGene")
  # allChr_Clines_ROS3 <- clineData_filter %>% filter(ColourGene=="ROS3" & distanceToGene =="inGene") # none in ROS3
  allChr_Clines_EL <- clineData_filter %>% filter(ColourGene=="ELUTA" & distanceToGene =="inGene")
  allChr_Clines_ROSelLinked <- clineData_filter %>% filter(ColourGene=="ROSEA" | ColourGene=="ROS1" | ColourGene=="ROS2" | ColourGene=="ELUTA") %>% filter(distanceToGene %in% linkedAreas) 
  
  # Chr 7 
  allChr_Clines_Chr7_elsewhere <- clineData_filter %>% filter(LG==7 & ColourGene==-9)

  # Chr 8 
  allChr_Clines_Chr8 <- clineData_filter %>% filter(LG==8)
  # combine all the 'elsewhere' loci not linked to known colour genes
  
  # commence plot
  par(mar=c(3,3,4,4))
  par(oma=c(1,1,1,1))
  par(fig = plotPos1, mar=c(1,1,1,1),new=((startPlot==TRUE)==FALSE))
  plot(NA, xlab="", xaxt="n", ylab="", main="", las=2, bty="n",cex=0.5,yaxt="n",xaxt="n",type='n',
       xlim=c(-5500000,max(range(clineData[,"position_graph"]))+100000), ylim=c(3.8,19),
       cex.axis=1.4,cex.lab=1.4,col=rgb(100,100,100,120,maxColorValue=255), pch=16)
  axis(2, at=c(4,6,8,10,12,14,16,18), labels=NA,col.axis="black", las=1,cex.axis=1.1,lwd=1.6,pos=-3000000, tck = -.03)
  axis(2, at=c(4,6,8,10,12,14,16,18), labels=c(4,6,8,10,12,14,16,18),col.axis="black", las=1,cex.axis=1.1,lwd=0,pos=-100000)
  
  makeBaseLG_ClineCentre <- function(graphPos) {
    lines(x=c(range(graphPos[graphPos[,"LG"]==1,"windowMid_graph"])[1],range(graphPos[graphPos[,"LG"]==1,"windowMid_graph"])[2]),y=c(3.5,3.5),col = "black", lwd=4.5,lty = 1)
    lines(x = c(range(graphPos[graphPos[,"LG"]==1,"windowMid_graph"])[1],range(graphPos[graphPos[,"LG"]==1,"windowMid_graph"])[2]), y = c(12.580,12.580), lty=5,lwd=2,col='red')
    textPos1 <- (range(graphPos[graphPos[,"LG"]==1,"windowMid_graph"])[2]-range(graphPos[graphPos[,"LG"]==1,"windowMid_graph"])[1])/2
    #text(x=textPos1,y=2.0,"LG 1",cex=1.3)
    lines(x=c(range(graphPos[graphPos[,"LG"]==2,"windowMid_graph"])[1],range(graphPos[graphPos[,"LG"]==2,"windowMid_graph"])[2]),y=c(3.5,3.5),col = "black", lwd=4.5,lty = 1)
    lines(x = c(range(graphPos[graphPos[,"LG"]==2,"windowMid_graph"])[1],range(graphPos[graphPos[,"LG"]==2,"windowMid_graph"])[2]), y = c(12.580,12.580), lty=5,lwd=2,col='red')

    textPos2 <- ((range(graphPos[graphPos[,"LG"]==2,"windowMid_graph"])[2]-range(graphPos[graphPos[,"LG"]==2,"windowMid_graph"])[1])/2)+range(graphPos[graphPos[,"LG"]==2,"windowMid_graph"])[1]
    #text(x=textPos2,y=2.0,"LG 2",cex=1.3)
    lines(x=c(range(graphPos[graphPos[,"LG"]==3,"windowMid_graph"])[1],range(graphPos[graphPos[,"LG"]==3,"windowMid_graph"])[2]),y=c(3.5,3.5),col = "black", lwd=4.5,lty = 1)
        lines(x = c(range(graphPos[graphPos[,"LG"]==3,"windowMid_graph"])[1],range(graphPos[graphPos[,"LG"]==3,"windowMid_graph"])[2]), y = c(12.580,12.580), lty=5,lwd=2,col='red')

    textPos3 <- ((range(graphPos[graphPos[,"LG"]==3,"windowMid_graph"])[2]-range(graphPos[graphPos[,"LG"]==3,"windowMid_graph"])[1])/2)+range(graphPos[graphPos[,"LG"]==3,"windowMid_graph"])[1]
    #text(x=textPos3,y=2.0,"LG 3",cex=1.3)
    lines(x=c(range(graphPos[graphPos[,"LG"]==4,"windowMid_graph"])[1],range(graphPos[graphPos[,"LG"]==4,"windowMid_graph"])[2]),y=c(3.5,3.5),col = "black", lwd=4.5,lty = 1)
        lines(x = c(range(graphPos[graphPos[,"LG"]==4,"windowMid_graph"])[1],range(graphPos[graphPos[,"LG"]==4,"windowMid_graph"])[2]), y = c(12.580,12.580), lty=5,lwd=2,col='red')

    textPos4 <- ((range(graphPos[graphPos[,"LG"]==4,"windowMid_graph"])[2]-range(graphPos[graphPos[,"LG"]==4,"windowMid_graph"])[1])/2)+range(graphPos[graphPos[,"LG"]==4,"windowMid_graph"])[1]
    #text(x=textPos4,y=2.0,"LG 4",cex=1.3)
    lines(x=c(range(graphPos[graphPos[,"LG"]==5,"windowMid_graph"])[1],range(graphPos[graphPos[,"LG"]==5,"windowMid_graph"])[2]),y=c(3.5,3.5),col = "black", lwd=4.5,lty = 1)
        lines(x = c(range(graphPos[graphPos[,"LG"]==5,"windowMid_graph"])[1],range(graphPos[graphPos[,"LG"]==5,"windowMid_graph"])[2]), y = c(12.580,12.580), lty=5,lwd=2,col='red')

    textPos5 <- ((range(graphPos[graphPos[,"LG"]==5,"windowMid_graph"])[2]-range(graphPos[graphPos[,"LG"]==5,"windowMid_graph"])[1])/2)+range(graphPos[graphPos[,"LG"]==5,"windowMid_graph"])[1]
    #text(x=textPos5,y=2.0,"LG 5",cex=1.3)
    lines(x=c(range(graphPos[graphPos[,"LG"]==6,"windowMid_graph"])[1],range(graphPos[graphPos[,"LG"]==6,"windowMid_graph"])[2]),y=c(3.5,3.5),col = "black", lwd=4.5,lty = 1)
        lines(x = c(range(graphPos[graphPos[,"LG"]==6,"windowMid_graph"])[1],range(graphPos[graphPos[,"LG"]==6,"windowMid_graph"])[2]), y = c(12.580,12.580), lty=5,lwd=2,col='red')

    textPos6 <- ((range(graphPos[graphPos[,"LG"]==6,"windowMid_graph"])[2]-range(graphPos[graphPos[,"LG"]==6,"windowMid_graph"])[1])/2)+range(graphPos[graphPos[,"LG"]==6,"windowMid_graph"])[1]
    #text(x=textPos6,y=2.0,"LG 6",cex=1.3)
    lines(x=c(range(graphPos[graphPos[,"LG"]==7,"windowMid_graph"])[1],range(graphPos[graphPos[,"LG"]==7,"windowMid_graph"])[2]),y=c(3.5,3.5),col = "black", lwd=4.5,lty = 1)
        lines(x = c(range(graphPos[graphPos[,"LG"]==7,"windowMid_graph"])[1],range(graphPos[graphPos[,"LG"]==7,"windowMid_graph"])[2]), y = c(12.580,12.580), lty=5,lwd=2,col='red')

    textPos7 <- ((range(graphPos[graphPos[,"LG"]==7,"windowMid_graph"])[2]-range(graphPos[graphPos[,"LG"]==7,"windowMid_graph"])[1])/2)+range(graphPos[graphPos[,"LG"]==7,"windowMid_graph"])[1]
    #text(x=textPos7,y=2.0,"LG 7",cex=1.3)
    lines(x=c(range(graphPos[graphPos[,"LG"]==8,"windowMid_graph"])[1],range(graphPos[graphPos[,"LG"]==8,"windowMid_graph"])[2]), y=c(3.5,3.5),col = "black", lwd=4.5,lty = 1)
        lines(x = c(range(graphPos[graphPos[,"LG"]==8,"windowMid_graph"])[1],range(graphPos[graphPos[,"LG"]==8,"windowMid_graph"])[2]), y = c(12.580,12.580), lty=5,lwd=2,col='red')

    textPos8 <- ((range(graphPos[graphPos[,"LG"]==8,"windowMid_graph"])[2]-range(graphPos[graphPos[,"LG"]==8,"windowMid_graph"])[1])/2)+range(graphPos[graphPos[,"LG"]==8,"windowMid_graph"])[1]
    #text(x=textPos8,y=2.0,"LG 8",cex=1.3)
  }
  makeBaseLG_ClineCentre(graphPos)
  mtext("centre",side=2,cex=1.3,line=0.5)

# add points
 
  # background
   points(allChr_Clines_WG_elsewhere[,"position_graph"], allChr_Clines_WG_elsewhere[,ClineEst[1]],       col='black',bg=colourTable[colourTable[,"Region"]=="background","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
  
  # CRE and linked
   points(allChr_Clines_CREMOSAlinked[,"position_graph"], allChr_Clines_CREMOSAlinked[,ClineEst[1]],       col='black',bg=colourTable[colourTable[,"Region"]=="CREMOSAlinked","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
    points(allChr_Clines_CREMOSA[,"position_graph"], allChr_Clines_CREMOSA[,ClineEst[1]],       col='black',bg=colourTable[colourTable[,"Region"]=="CREMOSA","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
  
  #  FLAVIA/Aurina 
   points(allChr_Clines_FLAVIAlinked[,"position_graph"], allChr_Clines_FLAVIAlinked[,ClineEst[1]],       col='black',bg=colourTable[colourTable[,"Region"]=="FLAVIAlinked","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
   points(allChr_Clines_AURINAlinked[,"position_graph"], allChr_Clines_AURINAlinked[,ClineEst[1]],       col='black',bg=colourTable[colourTable[,"Region"]=="AURINAlinked","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
    points(allChr_Clines_FLAVIA[,"position_graph"], allChr_Clines_FLAVIA[,ClineEst[1]],       col='black',bg=colourTable[colourTable[,"Region"]=="FLAVIA","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
  
    # SULF
     points(allChr_Clines_SULFlinked[,"position_graph"], allChr_Clines_SULFlinked[,ClineEst[1]],       col='black',bg=colourTable[colourTable[,"Region"]=="SULFlinked","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
    
     # RUBIA
   points(allChr_Clines_RUBIAlinked[,"position_graph"], allChr_Clines_RUBIAlinked[,ClineEst[1]],       col='black',bg=colourTable[colourTable[,"Region"]=="RUBIAlinked","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
   points(allChr_Clines_RUBIA[,"position_graph"], allChr_Clines_RUBIA[,ClineEst[1]],       col='black',bg=colourTable[colourTable[,"Region"]=="RUBIA","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
   
      # ROS EL
   points(allChr_Clines_ROSelLinked[,"position_graph"], allChr_Clines_ROSelLinked[,ClineEst[1]],       col='black',bg=colourTable[colourTable[,"Region"]=="ROSelLinked","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
         points(allChr_Clines_EL[,"position_graph"], allChr_Clines_EL[,ClineEst[1]],       col='black',bg=colourTable[colourTable[,"Region"]=="ELUTA","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
            points(allChr_Clines_ROS1[,"position_graph"], allChr_Clines_ROS1[,ClineEst[1]],       col='black',bg=colourTable[colourTable[,"Region"]=="ROS","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
      points(allChr_Clines_ROS2[,"position_graph"], allChr_Clines_ROS2[,ClineEst[1]],       col='black',bg=colourTable[colourTable[,"Region"]=="ROS","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
}


```

```{r Fst genome scan pairwise whole genome function}
genomeScanFst <- function(clineData,WindowData,dataPlot,plotPos1,startPlot,colourTable,lwdPoints,sizePoints) {

  # plotPos1 <- c(0.0, 1.0, 0.4, 0.54)
  # clineData <- allChr_Clines_filterDpth15; WindowData <- allChr_Windows; deltaP <- 0.9; pool_deltaP <- "AFD_15_20"; startPlot = F
  # colourTable = colourChoice
  # dataPlot <- "FstfromMeanPiAdj_15_20"
  
  # remove any NAs and NaN
  WindowData <- WindowData[which(!is.na(WindowData[,dataPlot])),]
  WindowData <- WindowData[which(WindowData[,dataPlot]!="NaN"),]
  # remove any zero Fst
  WindowData[WindowData[,dataPlot]<0,dataPlot] <- 0

  par(mar=c(3,3,4,4))
  par(oma=c(1,1,1,1))
  par(fig = plotPos1, mar=c(1,1,1,1),new=((startPlot==TRUE)==FALSE))
  #Make the first plot (using YP1 and MP4. Other pools can be used)   mtext(expression(pi['b']),side=1,line=3,cex=1.8)
  plot(WindowData[,"windowMid_graph"], as.numeric(WindowData[,dataPlot]), xlab="",  
       ylab="Fst", main="", las=2, bty="n",cex=0.5,yaxt="n",xaxt="n",type='n',
       xlim=c(-5500000,max(range(WindowData[,"windowMid_graph"]))), ylim=c(-0.15,1.25),
       cex.axis=1.4,cex.lab=1.4,col=rgb(100,100,100,120,maxColorValue=255), pch=16)
  axis(2, at=c(0,0.2,0.4,0.6,0.8,1.0), labels=NA,col.axis="black", las=1,cex.axis=1.1,lwd=1.6,pos=-3000000, tck = -.03)
  axis(2, at=c(0,0.2,0.4,0.6,0.8,1.0), labels=c(0,0.2,0.4,0.6,0.8,1.0),col.axis="black", las=1,cex.axis=1.1,lwd=0,pos=-100000)
  makeBaseLG <- function(graphPos) {
    lines(x=c(range(graphPos[graphPos[,"LG"]==1,"windowMid_graph"])[1],range(graphPos[graphPos[,"LG"]==1,"windowMid_graph"])[2]),y=c(-0.06,-0.06),col = "black", lwd=4,lty = 1)
    lines(x=c(range(graphPos[graphPos[,"LG"]==2,"windowMid_graph"])[1],range(graphPos[graphPos[,"LG"]==2,"windowMid_graph"])[2]),y=c(-0.06,-0.06),col = "black", lwd=4,lty = 1)
    lines(x=c(range(graphPos[graphPos[,"LG"]==3,"windowMid_graph"])[1],range(graphPos[graphPos[,"LG"]==3,"windowMid_graph"])[2]),y=c(-0.06,-0.06),col = "black", lwd=4,lty = 1)
    lines(x=c(range(graphPos[graphPos[,"LG"]==4,"windowMid_graph"])[1],range(graphPos[graphPos[,"LG"]==4,"windowMid_graph"])[2]),y=c(-0.06,-0.06),col = "black", lwd=4,lty = 1)
    lines(x=c(range(graphPos[graphPos[,"LG"]==5,"windowMid_graph"])[1],range(graphPos[graphPos[,"LG"]==5,"windowMid_graph"])[2]),y=c(-0.06,-0.06),col = "black", lwd=4,lty = 1)
    lines(x=c(range(graphPos[graphPos[,"LG"]==6,"windowMid_graph"])[1],range(graphPos[graphPos[,"LG"]==6,"windowMid_graph"])[2]),y=c(-0.06,-0.06),col = "black", lwd=4,lty = 1)
    lines(x=c(range(graphPos[graphPos[,"LG"]==7,"windowMid_graph"])[1],range(graphPos[graphPos[,"LG"]==7,"windowMid_graph"])[2]),y=c(-0.06,-0.06),col = "black", lwd=4,lty = 1)
    lines(x=c(range(graphPos[graphPos[,"LG"]==8,"windowMid_graph"])[1],range(graphPos[graphPos[,"LG"]==8,"windowMid_graph"])[2]),y=c(-0.06,-0.06),col = "black", lwd=4,lty = 1)
  }
  makeBaseLG(graphPos)
  
  mtext(expression('F'['ST']),side=2,cex=1.3,line=0.5)
  #mtext("YP1 x MP6",side=2,cex=1.2,line=0.92)
  thisMedian <- round(median(as.numeric(WindowData[,dataPlot])),3)
  FstMedian=substitute(paste('median'~'F'['ST']~'=',nn), list(nn=thisMedian))
  #text(FstMedian,x = max(range(temp.fst10[,"windowMid_graph"]))-30000000,y = 0.85,cex=1)
  
  #Find quantiles
  FstQuantiles <- quantile(WindowData[,dataPlot], na.rm=T, c(0.5, 0.95))
  WindowData_outliers <- WindowData[WindowData[,dataPlot]>=as.numeric(FstQuantiles[2]),]
  
  FstQuantiles_99Q <- quantile(WindowData[,dataPlot], na.rm=T, c(0.5, 0.99))
  WindowData_outliers_99Q <- WindowData[WindowData[,dataPlot]>=as.numeric(FstQuantiles_99Q[2]),]
 
  nrow(WindowData_outliers) 
  scaffs_outliers <- table(as.vector(WindowData_outliers[,"scaffold"]))
  scaffs_outliers_LG <- table(as.vector(WindowData_outliers[,"LG"]),as.vector(WindowData_outliers[,"cM"]))
  
  scaffs_outliers <- sort(scaffs_outliers[WindowData_outliers!=0])
  # plotting dashed horizontal quantile lines
  lines(x=c(0,max(range(WindowData[,"windowMid_graph"]))),y=c(as.numeric(FstQuantiles_99Q[2]),as.numeric(FstQuantiles_99Q[2])),col = "black", lwd=1.5,lty = 3)
  lines(x=c(0,max(range(WindowData[,"windowMid_graph"]))),y=c(as.numeric(FstQuantiles[2]),as.numeric(FstQuantiles[2])),col = "black", lwd=1.5,lty = 3)
  
  
  # WidthQuantiles <- quantile(clineData[,"width"], na.rm=T, c(0.5, 0.05))
  # lines(x=c(0,max(range(graphPos[,"windowMid_graph"]))),y=c(as.numeric(WidthQuantiles[1]),as.numeric(WidthQuantiles[1]))/1000,col = "black", lwd=1.5,lty = 3)
  # lines(x=c(0,max(range(graphPos[,"windowMid_graph"]))),y=c(as.numeric(WidthQuantiles[2]),as.numeric(WidthQuantiles[2]))/1000,col = "black", lwd=1.5,lty = 3)
  

  #tail(colnames(WindowData))
  #table(WindowData[,"ColourGene"])
  # background loci
  allChr_Fst_WG_elsewhere <- WindowData %>% filter(ColourGene==-9)
  
  # Chr 1 & CREMOSA 
  allChr_Fst_CREMOSA <- WindowData %>% filter(ColourGene=="CREMOSA1&2" | ColourGene=="CREMOSA_3")
  allChr_Fst_CREMOSAlinked <- WindowData %>% filter(ColourGene=="CREMOSA_1&2_linked" | ColourGene=="CREMOSA_3_linked") 
  
  # Chr 2 & FLAVIA/Aurina 
  allChr_Fst_Chr3 <- WindowData %>% filter(LG==2)
  allChr_Fst_FLAVIA <- WindowData %>% filter(ColourGene=="FLAVIA")
  allChr_Fst_AURINA <- WindowData %>% filter(ColourGene=="AURINA")
  allChr_Fst_FLAVIAlinked <- WindowData %>% filter(ColourGene=="FLAVIA_linked") 
  allChr_Fst_AURINAlinked <- WindowData %>% filter(ColourGene=="AURINA_linked")

  # Chr 3 
  allChr_Fst_Chr3 <- WindowData %>% filter(LG==3)
  allChr_Fst_Chr3_elsewhere <- WindowData %>% filter(LG==3 & ColourGene==-9)
  
  # Chr 4 & SULF  
  allChr_Fst_Chr4_elsewhere <- WindowData %>% filter(LG==4 & ColourGene==-9)
  allChr_Fst_SULF <- WindowData %>% filter(ColourGene=="SULF")
  allChr_Fst_SULFlinked <- WindowData %>% filter(ColourGene=="SULF_linked") 
  
  # Chr 5 & RUBIA 
  # 721 clines outside FLAVIA
  allChr_Fst_Chr5_elsewhere <- WindowData %>% filter(LG==5 & ColourGene==-9)
  allChr_Fst_RUBIA <- WindowData %>% filter(ColourGene=="RUBIA")
  allChr_Fst_RUBIAlinked <- WindowData %>% filter(ColourGene=="RUBIA_linked")

  # Chr 6 and ROS EL 
  allChr_Fst_Chr6_elsewhere <- WindowData %>% filter(LG==6 & ColourGene==-9)

  allChr_Fst_ROS1 <- WindowData %>% filter(ColourGene=="ROS_1")
  allChr_Fst_ROS2 <- WindowData %>% filter(ColourGene=="ROS_2")
  allChr_Fst_ROS3 <- WindowData %>% filter(ColourGene=="ROS_3")
  allChr_Fst_EL <- WindowData %>% filter(ColourGene=="ELUTA")
  allChr_Fst_ROSelLinked <- WindowData %>% filter(ColourGene=="ROS1_linked" | ColourGene=="ROS_EL_barrier" | ColourGene=="EL_linked")
  
  # Chr 7 
  allChr_Fst_Chr7_elsewhere <- WindowData %>% filter(LG==7 & ColourGene==-9)

  # Chr 8 
  allChr_Fst_Chr8 <- WindowData %>% filter(LG==8)
  # combine all the 'elsewhere' loci not linked to known colour genes
  
  # plot points
  # background
   points(allChr_Fst_WG_elsewhere[,"windowMid_graph"], allChr_Fst_WG_elsewhere[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="background","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
  
  # CRE and linked
   points(allChr_Fst_CREMOSAlinked[,"windowMid_graph"], allChr_Fst_CREMOSAlinked[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="CREMOSAlinked","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
    points(allChr_Fst_CREMOSA[,"windowMid_graph"], allChr_Fst_CREMOSA[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="CREMOSA","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
  
  #  FLAVIA/Aurina 
   points(allChr_Fst_FLAVIAlinked[,"windowMid_graph"], allChr_Fst_FLAVIAlinked[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="FLAVIAlinked","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
   
   points(allChr_Fst_AURINAlinked[,"windowMid_graph"], allChr_Fst_AURINAlinked[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="AURINAlinked","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
   
    points(allChr_Fst_FLAVIA[,"windowMid_graph"], allChr_Fst_FLAVIA[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="FLAVIA","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
    
   points(allChr_Fst_AURINA[,"windowMid_graph"], allChr_Fst_AURINA[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="AURINA","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
    
   # SULF
     points(allChr_Fst_SULFlinked[,"windowMid_graph"], allChr_Fst_SULFlinked[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="SULFlinked","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
    points(allChr_Fst_SULF[,"windowMid_graph"], allChr_Fst_SULF[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="SULFlinked","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
     
    # RUBIA
   points(allChr_Fst_RUBIAlinked[,"windowMid_graph"], allChr_Fst_RUBIAlinked[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="RUBIAlinked","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
  points(allChr_Fst_RUBIA[,"windowMid_graph"], allChr_Fst_RUBIA[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="RUBIAlinked","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
   
      # ROS EL
   points(allChr_Fst_ROSelLinked[,"windowMid_graph"], allChr_Fst_ROSelLinked[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="ROSelLinked","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
         points(allChr_Fst_EL[,"windowMid_graph"], allChr_Fst_EL[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="ELUTA","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
            points(allChr_Fst_ROS1[,"windowMid_graph"], allChr_Fst_ROS1[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="ROS","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
      points(allChr_Fst_ROS2[,"windowMid_graph"], allChr_Fst_ROS2[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="ROS","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
  points(allChr_Fst_ROS3[,"windowMid_graph"], allChr_Fst_ROS3[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="ROS","pointColour"], lwd=lwdPoints, pch=21,cex=sizePoints)
  
} 
```

```{r Fig 5 genome scans plots pairwise,eval=F,include=F}

# a) cline proportions - proportion of 10kb windows in 100kb, that have at least 1 cline
genomeScanClineProportions(clineData = allChr_Clines_Dpth15 ,ClinePropData=ClinesProportionsWindows,WindowData = allChr_Windows, deltaP = 0.9, pool_deltaP = "AFD_15_20", ClineEst = c('centre','width'), filterPass=T, filterNegativeClines= T, plotPos1 = c(0.0, 1.0, 0.85, 1),startPlot =T,colourTable = colourChoice)
# b) cline width
genomeScanClineWidth(clineData = allChr_Clines_Dpth15 ,WindowData = allChr_Windows, deltaP = 0.9, pool_deltaP = "AFD_15_20", ClineEst = c('centre','width'), filterNegativeClines= T, plotPos1 = c(0.0, 1.0, 0.69, 0.85), startPlot =F,colourTable = colourChoice,lwdPoints=0.3,sizePoints=0.7)
#c) cline centre
genomeScanClineCentre(clineData = allChr_Clines_Dpth15 ,WindowData = allChr_Windows, deltaP = 0.9, pool_deltaP = "AFD_15_20", ClineEst = c('centre','width'), filterNegativeClines= T, plotPos1 = c(0.0, 1.0, 0.54, 0.70),startPlot =F,colourTable = colourChoice,lwdPoints=0.3,sizePoints=0.7)
# d) Fst - pass and colour
genomeScanFst(clineData = allChr_Clines_Dpth15 ,WindowData = allChr_Windows,dataPlot="FstfromMeanPiAdj_15_20", plotPos1=c(0.0, 1.0, 0.42, 0.58),startPlot=F,colourTable=colourChoice,lwdPoints=0.3,sizePoints=0.7)
# e) Fst - no pass yet colour
genomeScanFst(clineData = allChr_Clines_Dpth15,WindowData = allChr_Windows,dataPlot="FstfromMeanPiAdj_16_19", plotPos1=c(0.0, 1.0, 0.30, 0.46),startPlot=F,colourTable=colourChoice,lwdPoints=0.3,sizePoints=0.7)

# f) Fst - no pass no colour (Y,Y)
genomeScanFst(clineData = allChr_Clines_Dpth15,WindowData = allChr_Windows,dataPlot="FstfromMeanPiAdj_16_17", plotPos1=c(0.0, 1.0, 0.18, 0.34),startPlot=F,colourTable=colourChoice,lwdPoints=0.3,sizePoints=0.7)

# g) Fst - no pass no colour (M,M)
genomeScanFst(clineData = allChr_Clines_Dpth15,WindowData =  allChr_Windows, dataPlot="FstfromMeanPiAdj_19_20", plotPos1=c(0.0, 1.0, 0.06, 0.22),startPlot=F,colourTable=colourChoice,lwdPoints=0.3,sizePoints=0.7)


```

```{r Fig 5 genome scans plots pairwise PDF version,eval=F,include=F}

pdf("Fig4_GenomeScanPanel.pdf",width = 18, height = 25, pointsize = 25)

# a) cline proportions - proportion of 10kb windows in 100kb, that have at least 1 cline
genomeScanClineProportions(clineData = allChr_Clines_Dpth15 ,ClinePropData=ClinesProportionsWindows,WindowData = allChr_Windows, deltaP = 0.9, pool_deltaP = "AFD_15_20", ClineEst = c('centre','width'), filterPass=T, filterNegativeClines= T, plotPos1 = c(0.0, 1.0, 0.85, 1),startPlot =T,colourTable = colourChoice)
# b) cline width
genomeScanClineWidth(clineData = allChr_Clines_Dpth15 ,WindowData = allChr_Windows, deltaP = 0.9, pool_deltaP = "AFD_15_20", ClineEst = c('centre','width'), filterNegativeClines= T, plotPos1 = c(0.0, 1.0, 0.69, 0.85), startPlot =F,colourTable = colourChoice,lwdPoints=0.3,sizePoints=0.7)
#c) cline centre
genomeScanClineCentre(clineData = allChr_Clines_Dpth15 ,WindowData = allChr_Windows, deltaP = 0.9, pool_deltaP = "AFD_15_20", ClineEst = c('centre','width'), filterNegativeClines= T, plotPos1 = c(0.0, 1.0, 0.54, 0.70),startPlot =F,colourTable = colourChoice,lwdPoints=0.3,sizePoints=0.7)
# d) Fst - pass and colour
genomeScanFst(clineData = allChr_Clines_Dpth15 ,WindowData = allChr_Windows,dataPlot="FstfromMeanPiAdj_15_20", plotPos1=c(0.0, 1.0, 0.42, 0.58),startPlot=F,colourTable=colourChoice,lwdPoints=0.3,sizePoints=0.7)
# e) Fst - no pass yet colour
genomeScanFst(clineData = allChr_Clines_Dpth15,WindowData = allChr_Windows,dataPlot="FstfromMeanPiAdj_16_19", plotPos1=c(0.0, 1.0, 0.30, 0.46),startPlot=F,colourTable=colourChoice,lwdPoints=0.3,sizePoints=0.7)

# f) Fst - no pass no colour (Y,Y)
genomeScanFst(clineData = allChr_Clines_Dpth15,WindowData = allChr_Windows,dataPlot="FstfromMeanPiAdj_16_17", plotPos1=c(0.0, 1.0, 0.18, 0.34),startPlot=F,colourTable=colourChoice,lwdPoints=0.3,sizePoints=0.7)

# g) Fst - no pass no colour (M,M)
genomeScanFst(clineData = allChr_Clines_Dpth15,WindowData =  allChr_Windows, dataPlot="FstfromMeanPiAdj_19_20", plotPos1=c(0.0, 1.0, 0.06, 0.22),startPlot=F,colourTable=colourChoice,lwdPoints=0.3,sizePoints=0.7)

dev.off()
```

# Figure 6. Fst outliers and colour genes all pair-wise 99% quantile

Figure 6. All pair-wise population comparisons of FST and the number of 10kb windows in colour genes and linked regions along the genome. All pair-wise population pairs shown in rows, with the main colour of the varieties in each pair indicated (left colour bars) highlighting comparisons between yellow and magenta or within variety outliers. Columns indicated the number of windows in the 99th % quantile for FST at each of the colour genes and also regions closely linked (<100kb). For the FLA gene, closely linked regions are larger (<400kb) due to this being an area of low recombination in the genome. Numbers in each cell indicate how many 10kb windows were detected in the 99th % quantile. Legend displays the colour match for number of outlier windows.

```{r Fig 6 Fst outliers and colour genes all pair-wise 99% quantile}
# Generate all pairwise combinations of numbers 15 to 20
pairwise_comparisons <- combn(15:20, 2)

# Transpose to make each pair a row
pairwise_transposed <- t(pairwise_comparisons)

# Convert to a data frame for easier handling
pairwise_df <- as.data.frame(pairwise_transposed)
colnames(pairwise_df) <- c("Number1", "Number2")

# Define group labels
get_group <- function(num) {
  if (num %in% 15:17) {
    "Yellow"
  } else if (num %in% 18:20) {
    "Magenta"
  } else {
    "Unknown"
  }
}

# Add group columns for each number
pairwise_df$Group1 <- sapply(pairwise_df$Number1, get_group)
pairwise_df$Group2 <- sapply(pairwise_df$Number2, get_group)

# Define population names
population_names <- c("15" = "YP4", "16" = "YP2", "17" = "YP1", 
                      "18" = "MP2", "19" = "MP4", "20" = "MP11")
# Add population columns for each number
pairwise_df$Pop1 <- sapply(pairwise_df$Number1, function(x) population_names[as.character(x)])
pairwise_df$Pop2 <- sapply(pairwise_df$Number2, function(x) population_names[as.character(x)])

# Add the new column with the desired format
pairwise_df$FstfromMeanPiAdj <- paste0("FstfromMeanPiAdj_", pairwise_df$Number1, "_", pairwise_df$Number2)
# Add a sorting column to prioritize Yellow-Yellow rows
pairwise_df$SortOrder <- ifelse(pairwise_df$Group1 == "Yellow" & pairwise_df$Group2 == "Yellow", 1, 2)

# Sort the data frame by the SortOrder column
pairwise_df <- pairwise_df[order(pairwise_df$SortOrder), ]

# View the final data frame
pairwise_df

# total outlierWindows
outlierWinds <- matrix(0,nrow(pairwise_df),20)
colnames(outlierWinds) <- c("windows",names(table(allChr_Windows[,"ColourGene"]))[-1])
pairwise_df <- cbind(pairwise_df,outlierWinds)
colourGenes <- names(table(allChr_Windows[,"ColourGene"]))[-1]

# looping
for (thisRow in 1:nrow(pairwise_df)) {
  # thisRow <- 1
  dataPlot <- pairwise_df[thisRow,"FstfromMeanPiAdj"]
  FstQuantiles <- quantile(allChr_Windows[,dataPlot], na.rm=T, c(0.5, 0.99))
  WindowData_outliers <- allChr_Windows[allChr_Windows[,dataPlot]>=as.numeric(FstQuantiles[2]),]
  pairwise_df[thisRow,"windows"] <- nrow(WindowData_outliers)
  colourGenesFound <- WindowData_outliers[,"ColourGene"]
  
  colourGenesFound[is.na(colourGenesFound)] <- "none"
  colourGenesFound[colourGenesFound=="-9"] <- "none"
  
  # checking rubia
  # within FLS/rubia # Fst = 0.175
  # allChr_Windows[allChr_Windows[,"ColourGene"]=="RUBIA","FstfromMeanPiAdj_15_20"] 
  # linked to FLS/rubida # Fst 0.17 - 0.369, with 2 windows above 99% quantile of Fst 0.301
  # allChr_Windows[allChr_Windows[,"ColourGene"]=="RUBIA_linked","FstfromMeanPiAdj_15_20"] 
  thisTable <- table(WindowData_outliers[,"ColourGene"])
  
  # Count occurrences of each factor in the outlier data
  for (gene in colourGenes) {
    # gene <- "AURINA" 
    # Count occurrences of the factor
    count <- sum(colourGenesFound == gene)
    
    # Add the count to the appropriate column for all rows
    pairwise_df[thisRow,gene] <- count
  }
}

# export to Excel
# Create a new workbook
wb <- createWorkbook()

# Add a worksheet and write the data frame
addWorksheet(wb, "Pairwise Table")
writeData(wb, "Pairwise Table", pairwise_df)

# Save the workbook
saveWorkbook(wb, "pairwise_table.xlsx", overwrite = TRUE)

# trying coloured version



# re-import edited table

pairwise_df <- read.csv("Table1_FstwindowEdits.csv")

library(pheatmap)

# Ensure correct column selection for ColourGene regions
#colour_gene_cols <- grep("AURINA|CREMOSA|ELUTA|FLAVIA|ROS|RUBIA|SULF", names(pairwise_df), value = TRUE)

# Subset dataframe for pairwise comparisons and ColourGene data
heatmap_data <- pairwise_df[, c(1, 2, 5:ncol(pairwise_df))]

# Set row names as 'Number1 vs Number2' for better visualization
rownames(heatmap_data) <- paste(heatmap_data$Pop1, heatmap_data$Pop2, sep = " vs ")

# Convert to matrix (excluding non-numeric columns) and round values
heatmap_matrix <- round(as.matrix(heatmap_data[, -c(1, 2)]))  # Exclude 'Number1' and 'Number2'

# Define a custom color palette
custom_palette <- colorRampPalette(c("lightgray", "yellow", "red"))(50)

# Define breaks to ensure strong contrast for nonzero values
breaks <- c(seq(0, 1, length.out = 10), seq(2, max(heatmap_matrix, na.rm = TRUE), length.out = 40))

# Generate heatmap without tree structures (dendrograms)
pheatmap(heatmap_matrix,
         cluster_rows = FALSE,      # Disable clustering for rows
         cluster_cols = FALSE,      # Disable clustering for columns
         color = custom_palette,    # Use custom colors
         breaks = breaks,           # Ensure emphasis on nonzero values
         display_numbers = TRUE,    # Show actual values
         number_format = "%.0f",    # Force integer format
         fontsize_number = 12,      # Adjust text size
         cellwidth = 25,            # Adjust cell size
         cellheight = 25,
         main = "Pairwise Comparisons Across ColourGene Regions (Rounded)",
         border_color = "gray",     # Border color for cells
         legend = TRUE,             # Add a legend for the color scale
         show_rownames = TRUE,      # Show row names
         show_colnames = TRUE)      # Show column names

```

```{r Fig 6 Fst outliers and colour genes all pair-wise 95% quantile}
# Generate all pairwise combinations of numbers 15 to 20
pairwise_comparisons <- combn(15:20, 2)

# Transpose to make each pair a row
pairwise_transposed <- t(pairwise_comparisons)

# Convert to a data frame for easier handling
pairwise_df <- as.data.frame(pairwise_transposed)
colnames(pairwise_df) <- c("Number1", "Number2")

# Define group labels
get_group <- function(num) {
  if (num %in% 15:17) {
    "Yellow"
  } else if (num %in% 18:20) {
    "Magenta"
  } else {
    "Unknown"
  }
}

# Add group columns for each number
pairwise_df$Group1 <- sapply(pairwise_df$Number1, get_group)
pairwise_df$Group2 <- sapply(pairwise_df$Number2, get_group)

# Define population names
population_names <- c("15" = "YP4", "16" = "YP2", "17" = "YP1", 
                      "18" = "MP2", "19" = "MP4", "20" = "MP11")
# Add population columns for each number
pairwise_df$Pop1 <- sapply(pairwise_df$Number1, function(x) population_names[as.character(x)])
pairwise_df$Pop2 <- sapply(pairwise_df$Number2, function(x) population_names[as.character(x)])

# Add the new column with the desired format
pairwise_df$FstfromMeanPiAdj <- paste0("FstfromMeanPiAdj_", pairwise_df$Number1, "_", pairwise_df$Number2)
# Add a sorting column to prioritize Yellow-Yellow rows
pairwise_df$SortOrder <- ifelse(pairwise_df$Group1 == "Yellow" & pairwise_df$Group2 == "Yellow", 1, 2)

# Sort the data frame by the SortOrder column
pairwise_df <- pairwise_df[order(pairwise_df$SortOrder), ]

# View the final data frame
pairwise_df

# total outlierWindows
outlierWinds <- matrix(0,nrow(pairwise_df),20)
colnames(outlierWinds) <- c("windows",names(table(allChr_Windows[,"ColourGene"]))[-1])
pairwise_df <- cbind(pairwise_df,outlierWinds)
colourGenes <- names(table(allChr_Windows[,"ColourGene"]))[-1]

# looping
for (thisRow in 1:nrow(pairwise_df)) {
  # thisRow <- 1
  dataPlot <- pairwise_df[thisRow,"FstfromMeanPiAdj"]
  FstQuantiles <- quantile(allChr_Windows[,dataPlot], na.rm=T, c(0.5, 0.95))
  WindowData_outliers <- allChr_Windows[allChr_Windows[,dataPlot]>=as.numeric(FstQuantiles[2]),]
  pairwise_df[thisRow,"windows"] <- nrow(WindowData_outliers)
  colourGenesFound <- WindowData_outliers[,"ColourGene"]
  
  colourGenesFound[is.na(colourGenesFound)] <- "none"
  colourGenesFound[colourGenesFound=="-9"] <- "none"
  
  # checking rubia
  # within FLS/rubia # Fst = 0.175
  # allChr_Windows[allChr_Windows[,"ColourGene"]=="RUBIA","FstfromMeanPiAdj_15_20"] 
  # linked to FLS/rubida # Fst 0.17 - 0.369, with 2 windows above 99% quantile of Fst 0.301
  # allChr_Windows[allChr_Windows[,"ColourGene"]=="RUBIA_linked","FstfromMeanPiAdj_15_20"] 
  thisTable <- table(WindowData_outliers[,"ColourGene"])
  
  # Count occurrences of each factor in the outlier data
  for (gene in colourGenes) {
    # gene <- "AURINA" 
    # Count occurrences of the factor
    count <- sum(colourGenesFound == gene)
    
    # Add the count to the appropriate column for all rows
    pairwise_df[thisRow,gene] <- count
  }
}

# export to Excel
# Create a new workbook
wb <- createWorkbook()

# Add a worksheet and write the data frame
addWorksheet(wb, "Pairwise Table")
writeData(wb, "Pairwise Table", pairwise_df)

# Save the workbook
saveWorkbook(wb, "pairwise_table_Q95.xlsx", overwrite = TRUE)

# trying coloured version



# re-import edited table

pairwise_df <- read.csv(paste0(workingDir,"Table1_FstwindowsQ95_edits.csv"))

library(pheatmap)

# Ensure correct column selection for ColourGene regions
#colour_gene_cols <- grep("AURINA|CREMOSA|ELUTA|FLAVIA|ROS|RUBIA|SULF", names(pairwise_df), value = TRUE)

# Subset dataframe for pairwise comparisons and ColourGene data
heatmap_data <- pairwise_df[, c(1, 2, 5:ncol(pairwise_df))]

# Set row names as 'Number1 vs Number2' for better visualization
rownames(heatmap_data) <- paste(heatmap_data$Pop1, heatmap_data$Pop2, sep = " vs ")

# Convert to matrix (excluding non-numeric columns) and round values
heatmap_matrix <- round(as.matrix(heatmap_data[, -c(1, 2)]))  # Exclude 'Number1' and 'Number2'

# Define a custom color palette
custom_palette <- colorRampPalette(c("lightgray", "yellow", "red"))(50)

# Define breaks to ensure strong contrast for nonzero values
breaks <- c(seq(0, 1, length.out = 10), seq(2, max(heatmap_matrix, na.rm = TRUE), length.out = 40))

# Generate heatmap without tree structures (dendrograms)
pheatmap(heatmap_matrix,
         cluster_rows = FALSE,      # Disable clustering for rows
         cluster_cols = FALSE,      # Disable clustering for columns
         color = custom_palette,    # Use custom colors
         breaks = breaks,           # Ensure emphasis on nonzero values
         display_numbers = TRUE,    # Show actual values
         number_format = "%.0f",    # Force integer format
         fontsize_number = 12,      # Adjust text size
         cellwidth = 25,            # Adjust cell size
         cellheight = 25,
         main = "Pairwise Comparisons Across ColourGene Regions (Rounded)",
         border_color = "gray",     # Border color for cells
         legend = TRUE,             # Add a legend for the color scale
         show_rownames = TRUE,      # Show row names
         show_colnames = TRUE)      # Show column names

```

# Figure 7. Rubia plots
(a) Genome scan chromosome 5
(b) zoom in for region around RUBIA
(c) differential gene expression
(d) phenotype x genotype 
(e) example photos 

# Zooming in on all key clinal regions Combined Clines, Fst, Dxy plot

```{r import genome annotations and lifted coordinates for DGE}
# annotated genome is v3.5

# technically mapped to v3.0, but coordinates lifted to v3.5 working version so it aligns with Fst and clines

# note - change this to user working Directoryr

gff.path <- "/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/referenceGenomes/version3.5/GWHBJVT00000000.gff"

gff.data <- rtracklayer::readGFF(gff.path)
gff.by.gene <- split(gff.data, gff.data$type)
gene.len.long <- lapply(gff.by.gene, function(x) x[order(x$end - x$start),][1,])

# but RNAseq data was mapped to v3
#  therefore importing data set with lifted coordinates (v3 -> v3.5) with functional gene annotations (from Dasha)

RNAseqData <- read.csv(paste0('/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/RNAseq/','candidate_genes_Chr5_RNAseq_Rimport.csv'),header=T)

```

```{r function for genome scans zoomed plots Fst and clines with gene annotation}
genomeScanZoomFstClineWidth <- function(clineData,WindowData,thisLG,WindowCut,dataPlot,FstRange,lwdFst,DxyPlot,piWPlot,deltaP,pool_deltaP,ClineEst, filterNegativeClines,gffdata,gffPlot,DFgeneExp,plotColGeneRegions,colourTable,startPlot,plotPos1,clineCexBackground,clineCexGene,KASPpos) {
  # par(mfrow=c(1,1))
  # clineData <- allChr_Clines_filterDpth15; WindowData <- allChr_Windows; deltaP <- 0.9; pool_deltaP <- "AFD_15_20"; startPlot = T; gffdata <- gff.data
  # plotPos1 <- c(0.03, 0.97, 0.70, 1) ; thisLG <- 5; WindowCut=c(6100000,6400000); FstRange <- c(0,0.5); dataPlot <- "FstfromMeanPiAdj_15_20"
  # DxyPlot <- "dXYfromMeanFstAdj_15_20"
  # piWPlot <- "piBarAdj_15_20"
  # ClineEst <- c('centre','width'); filterNegativeClines = T; colourTable=colourChoice
  # clineCexBackground <- 1.2; clineCexGene <- 1.3; gffPlot =T
  # cat("\nclinal loci start n=",nrow(clineData))
  # DFgeneExp <- RNAseqData
  # KASPpos contains two numbers, x-axis position on chromosome, y-axis cline width from KASP MLE fits
  # KASPpos <- c(6286923,7600)
  # note that the windowCut object is the range of positions to zoom in on. It has to be based on original positions on each LG and not on the graphing positions, so it lines up with gene annotations
  # colnames(allChr_Windows)[1:12]
  # allChr_Fst_RUBIA[,3145:3153]
  # allChr_Fst_RUBIAlinked[,3145:3153]
  # WindowCut=c(6100000,6400000) for RUBIA
  # WindowCut=c(0,80000000) for LG 2
  # other Fst island Chr 5 WindowCut=c(25000000,30000000)
  # plotting Fst
  par(mar=c(2,2,1,1))
  par(oma=c(1,1,1,1))
  par(fig = plotPos1, mar=c(2,2,1,1),new=((startPlot==TRUE)==FALSE))
  plot(WindowData[,"windowMid_graph"]/1000, as.numeric(WindowData[,dataPlot]), xlab="",  
       ylab="", main="", las=1, bty="n",cex=0.5,type='n',yaxt='n',lwd=2,xaxt='n',
       xlim=c((WindowCut[1]/1000),(WindowCut[2]/1000)), ylim=c(-0.01,FstRange[2]),
       cex.axis=1.4,cex.lab=1.4,col=rgb(100,100,100,120,maxColorValue=255), pch=16)
  if ((WindowCut[2]-WindowCut[1])>10000000) {
     axis(1, at=seq(WindowCut[1]/1000,WindowCut[2]/1000,10000), labels=NA,col.axis="black", las=1,cex.axis=1.1,lwd=4,pos=0, tck = -.03)
   axis(1,at=seq(WindowCut[1]/1000,WindowCut[2]/1000,10000), labels=seq(WindowCut[1]/1000,WindowCut[2]/1000,10000),col.axis="black", las=1,cex.axis=1.4,lwd=4,pos=0)
    
  }
  if ((WindowCut[2]-WindowCut[1])<10000000 & (WindowCut[2]-WindowCut[1])>1000000) {
     axis(1, at=seq(WindowCut[1]/1000,WindowCut[2]/1000,1000), labels=NA,col.axis="black", las=1,cex.axis=1.1,lwd=4,pos=0, tck = -.03)
   axis(1,at=seq(WindowCut[1]/1000,WindowCut[2]/1000,1000), labels=seq(WindowCut[1]/1000,WindowCut[2]/1000,1000),col.axis="black", las=1,cex.axis=1.4,lwd=4,pos=0)
    
  }
  if ((WindowCut[2]-WindowCut[1])<1000000) {
   axis(1, at=seq(WindowCut[1]/1000,WindowCut[2]/1000,50), labels=NA,col.axis="black", las=1,cex.axis=1.1,lwd=4,pos=0, tck = -.03)
   axis(1,at=seq(WindowCut[1]/1000,WindowCut[2]/1000,50), labels=seq(WindowCut[1]/1000,WindowCut[2]/1000,50),col.axis="black", las=1,cex.axis=1.4,lwd=4,pos=0)
  }
  
  axis(2, at=seq(0,max(FstRange),0.1), labels=NA,col.axis="black", las=1,cex.axis=1,lwd=4,pos=WindowCut[1]/1000, tck = -.03)
  axis(2, at=seq(0,max(FstRange),0.1), labels=seq(0,max(FstRange),0.1),col.axis="black", las=1, cex.axis=1.4, lwd=4, pos=(WindowCut[1]/1000))
  lines(x=c(WindowCut[1]/1000,WindowCut[2]/1000), y=c(0,0),lwd=4)
  lines(x=c(WindowCut[2]/1000,WindowCut[2]/1000), y=c(0,FstRange[2]),lwd=4)
  lines(x=c(WindowCut[1]/1000,WindowCut[2]/1000), y=c(FstRange[2],FstRange[2]),lwd=4)
  lines(x=c(WindowCut[1]/1000,WindowCut[1]/1000), y=c(0,FstRange[2]),lwd=4)

  mtext('genome position (Kbp)',side=1,cex=1.4,line=2.7)
  mtext(expression('F'['ST']),side=2,cex=1.4,line=1.2)
  mtext(paste0("Dxy: pi_w"),side=2,cex=1.4,line=2.8)

  #mtext("YP1 x MP6",side=2,cex=1.2,line=0.92)
  thisMedian <- round(median(as.numeric(WindowData[,dataPlot])),3)
  FstMedian=substitute(paste('median'~'F'['ST']~'=',nn), list(nn=thisMedian))
  #text(FstMedian,x = max(range(temp.fst10[,"windowMid_graph"]))-30000000,y = 0.85,cex=1)
  
  #Find quantiles
  FstQuantiles <- quantile(WindowData[,dataPlot], na.rm=T, c(0.5, 0.95))
  WindowData_outliers <- WindowData[WindowData[,dataPlot]>=as.numeric(FstQuantiles[2]),]
  
  FstQuantiles_99Q <- quantile(WindowData[,dataPlot], na.rm=T, c(0.5, 0.99))
  WindowData_outliers_99Q <- WindowData[WindowData[,dataPlot]>=as.numeric(FstQuantiles_99Q[2]),]
 
  scaffs_outliers <- table(as.vector(WindowData_outliers[,"scaffold"]))
  scaffs_outliers_LG <- table(as.vector(WindowData_outliers[,"LG"]),as.vector(WindowData_outliers[,"cM"]))
  
  scaffs_outliers <- sort(scaffs_outliers[WindowData_outliers!=0])
  # plotting dashed horizontal quantile lines
  #lines(x=c(0,max(range(WindowData[,"windowMid_graph"]))),y=c(as.numeric(FstQuantiles_99Q[2]),as.numeric(FstQuantiles_99Q[2])),col = "black", lwd=1.5,lty = 3)
  #lines(x=c(0,max(range(WindowData[,"windowMid_graph"]))),y=c(as.numeric(FstQuantiles[2]),as.numeric(FstQuantiles[2])),col = "black", lwd=1.5,lty = 3)
  
  
  # WidthQuantiles <- quantile(clineData[,"width"], na.rm=T, c(0.5, 0.05))
  # lines(x=c(0,max(range(graphPos[,"windowMid_graph"]))),y=c(as.numeric(WidthQuantiles[1]),as.numeric(WidthQuantiles[1]))/1000,col = "black", lwd=1.5,lty = 3)
  # lines(x=c(0,max(range(graphPos[,"windowMid_graph"]))),y=c(as.numeric(WidthQuantiles[2]),as.numeric(WidthQuantiles[2]))/1000,col = "black", lwd=1.5,lty = 3)
  

  #tail(colnames(WindowData))
  #table(WindowData[,"ColourGene"])
  # background loci
  allChr_Fst_WG_elsewhere <- WindowData %>% filter(ColourGene==-9)
  
  # Chr 1 & CREMOSA 
  allChr_Fst_CREMOSA <- WindowData %>% filter(ColourGene=="CREMOSA1&2" | ColourGene=="CREMOSA_3")
  allChr_Fst_CREMOSAlinked <- WindowData %>% filter(ColourGene=="CREMOSA_1&2_linked" | ColourGene=="CREMOSA_3_linked") 
  
  # Chr 2 & FLAVIA/Aurina 
  allChr_Fst_Chr3 <- WindowData %>% filter(LG==2)
  allChr_Fst_FLAVIA <- WindowData %>% filter(ColourGene=="FLAVIA")
  allChr_Fst_AURINA <- WindowData %>% filter(ColourGene=="AURINA")
  allChr_Fst_FLAVIAlinked <- WindowData %>% filter(ColourGene=="FLAVIA_linked") 
  allChr_Fst_AURINAlinked <- WindowData %>% filter(ColourGene=="AURINA_linked")

  # Chr 3 
  allChr_Fst_Chr3 <- WindowData %>% filter(LG==3)
  allChr_Fst_Chr3_elsewhere <- WindowData %>% filter(LG==3 & ColourGene==-9)
  
  # Chr 4 & SULF  
  allChr_Fst_Chr4_elsewhere <- WindowData %>% filter(LG==4 & ColourGene==-9)
  allChr_Fst_SULF <- WindowData %>% filter(ColourGene=="SULF")
  allChr_Fst_SULFlinked <- WindowData %>% filter(ColourGene=="SULF_linked") 
  
  # Chr 5 & RUBIA 
  # 721 clines outside FLAVIA
  allChr_Fst_Chr5_elsewhere <- WindowData %>% filter(LG==5 & ColourGene==-9)
  allChr_Fst_RUBIA <- WindowData %>% filter(ColourGene=="RUBIA")
  allChr_Fst_RUBIAlinked <- WindowData %>% filter(ColourGene=="RUBIA_linked")

  # Chr 6 and ROS EL 
  allChr_Fst_Chr6_elsewhere <- WindowData %>% filter(LG==6 & ColourGene==-9)

  allChr_Fst_ROS1 <- WindowData %>% filter(ColourGene=="ROS_1")
  allChr_Fst_ROS2 <- WindowData %>% filter(ColourGene=="ROS_2")
  allChr_Fst_ROS3 <- WindowData %>% filter(ColourGene=="ROS_3")
  allChr_Fst_EL <- WindowData %>% filter(ColourGene=="ELUTA")
  allChr_Fst_ROSelLinked <- WindowData %>% filter(ColourGene=="ROS1_linked" | ColourGene=="ROS_EL_barrier" | ColourGene=="EL_linked")
  
  # range(allChr_Fst_ROSelLinked[,"windowMid_graph"])
  
  # Chr 7 
  allChr_Fst_Chr7_elsewhere <- WindowData %>% filter(LG==7 & ColourGene==-9)

  # Chr 8 
  allChr_Fst_Chr8 <- WindowData %>% filter(LG==8)
  # combine all the 'elsewhere' loci not linked to known colour genes
  
  # plot gene regions from gff file annotation (in pink)
  if (gffPlot ==T) {
    if (thisLG == 1) {
      # thisLG ==1
      thisLG_gff <- gffdata[gffdata[,"seqid"]=='GWHBJVT00000001',]
      # WindowCut=c(700000,1100000)
      thisLG_gff_cut <- thisLG_gff[thisLG_gff[,"start"]>WindowCut[1] & thisLG_gff[,"end"]<WindowCut[2],]
      all_genes_start <- unique(thisLG_gff_cut[,"start"])
      for (thisOne in 1:length(all_genes_start)) {
        # thisOne <- 1
        startPosition <- all_genes_start[thisOne]
        thisGene <- thisLG_gff_cut[thisLG_gff_cut[,"start"]==startPosition,]
        if ("CDS" %in% thisGene[,"type"]) {
          thisCDS_start <- thisGene[thisGene[,"type"]=="CDS","start"]
          thisCDS_end <- thisGene[thisGene[,"type"]=="CDS","end"]
          # CREMOSA 1 - a gene in same position in gff
          if (startPosition == 928205) {
              cat("\n in gff in CREMOSA 1, gene Accession: ", thisGene[thisGene[,"type"]=="gene","Accession"])
          }
          if (startPosition == 921293) {
              cat("\n in gff in CREMOSA 2 gene Accession: ", thisGene[thisGene[,"type"]=="gene","Accession"])
          }
          if (startPosition == 994948) {
              cat("\n in gff in CREMOSA 3 gene Accession: ", thisGene[thisGene[,"type"]=="gene","Accession"])
          }
          rect(xleft=(thisCDS_start/1000),xright=(thisCDS_end/1000),ybottom=0.02,ytop=FstRange[2]-0.02, col='lightpink',lty = "solid", lwd=2, border='lightpink')
        }
      }
    }
    if (thisLG == 2) {
    thisLG_gff <- gffdata[gffdata[,"seqid"]=='GWHBJVT00000002',]
    # WindowCut=c(700000,1100000)
    thisLG_gff_cut <- thisLG_gff[thisLG_gff[,"start"]>WindowCut[1] & thisLG_gff[,"end"]<WindowCut[2],]
    all_genes_start <- unique(thisLG_gff_cut[,"start"])

    for (thisOne in 1:length(all_genes_start)) {
      # thisOne <- 1
      startPosition <- all_genes_start[thisOne]
      thisGene <- thisLG_gff_cut[thisLG_gff_cut[,"start"]==startPosition,]
      if ("CDS" %in% thisGene[,"type"]) {
        if (startPosition == 1051685) {
          cat("\n in gff in AURINA, gene Accession: ", thisGene[thisGene[,"type"]=="gene","Accession"])
        }
        if (startPosition == 53468062) {
          cat("\n in gff in FLAVIA, gene Accession: ", thisGene[thisGene[,"type"]=="gene","Accession"])
        }
        thisCDS_start <- thisGene[thisGene[,"type"]=="CDS","start"]
        thisCDS_end <- thisGene[thisGene[,"type"]=="CDS","end"]
        rect(xleft=(thisCDS_start/1000),xright=(thisCDS_end/1000),ybottom=0.02,ytop=FstRange[2]-0.02, col='lightpink',lty = "solid", lwd=2, border='lightpink')
      }
      
    }
  }
    if (thisLG == 3) {
    thisLG_gff <- gffdata[gffdata[,"seqid"]=='GWHBJVT00000003',]
    # WindowCut=c(700000,1100000)
    thisLG_gff_cut <- thisLG_gff[thisLG_gff[,"start"]>WindowCut[1] & thisLG_gff[,"end"]<WindowCut[2],]
    all_genes_start <- unique(thisLG_gff_cut[,"start"])

    for (thisOne in 1:length(all_genes_start)) {
      # thisOne <- 1
      thisGene <- thisLG_gff_cut[thisLG_gff_cut[,"start"]==startPosition,]
      if ("CDS" %in% thisGene[,"type"]) {
        thisCDS_start <- thisGene[thisGene[,"type"]=="CDS","start"]
        thisCDS_end <- thisGene[thisGene[,"type"]=="CDS","end"]
        rect(xleft=(thisCDS_start/1000),xright=(thisCDS_end/1000),ybottom=0.02,ytop=FstRange[2]-0.02, col='lightpink',lty = "solid", lwd=2, border='lightpink')
      }
    }
  }  
    if (thisLG == 4) {
    thisLG_gff <- gffdata[gffdata[,"seqid"]=='GWHBJVT00000004',]
    # WindowCut=c(700000,1100000)
    thisLG_gff_cut <- thisLG_gff[thisLG_gff[,"start"]>WindowCut[1] & thisLG_gff[,"end"]<WindowCut[2],]
    all_genes_start <- unique(thisLG_gff_cut[,"start"])

    for (thisOne in 1:length(all_genes_start)) {
      # thisOne <- 1
      startPosition <- all_genes_start[thisOne]
      thisGene <- thisLG_gff_cut[thisLG_gff_cut[,"start"]==startPosition,]
      if ("CDS" %in% thisGene[,"type"]) {
        if (startPosition == 38348572) {
          cat("\n in gff in SULF, gene Accession: ", thisGene[thisGene[,"type"]=="gene","Accession"])
        }
        thisCDS_start <- thisGene[thisGene[,"type"]=="CDS","start"]
        thisCDS_end <- thisGene[thisGene[,"type"]=="CDS","end"]
        rect(xleft=(thisCDS_start/1000),xright=(thisCDS_end/1000),ybottom=0.02,ytop=FstRange[2]-0.02, col='lightpink',lty = "solid", lwd=2, border='lightpink')
      }
    }
  }
    if (thisLG == 5) {
    thisLG_gff <- gffdata[gffdata[,"seqid"]=='GWHBJVT00000005',]
    # WindowCut=c(700000,1100000)
    thisLG_gff_cut <- thisLG_gff[thisLG_gff[,"start"]>WindowCut[1] & thisLG_gff[,"end"]<WindowCut[2],]
    all_genes_start <- unique(thisLG_gff_cut[,"start"])
    #fileName <- paste0("gff_",thisLG,"_",WindowCut[1],".txt")
    #write.file(thisLG_gff_cut,fileName)
    
    # CREMOSA 1 - a gene in same position in gff
    #928205 %in% all_genes_start
    
    for (thisOne in 1:length(all_genes_start)) {
      # thisOne <- 1
      startPosition <- all_genes_start[thisOne]
      thisGene <- thisLG_gff_cut[thisLG_gff_cut[,"start"]==startPosition,]
      if ("CDS" %in% thisGene[,"type"]) {
        if (startPosition == 6269966) {
          cat("\n in gff in RUBIA, gene Accession: ", thisGene[thisGene[,"type"]=="gene","Accession"])
        }
        thisCDS_start <- thisGene[thisGene[,"type"]=="CDS","start"]
        thisCDS_end <- thisGene[thisGene[,"type"]=="CDS","end"]
        rect(xleft=(thisCDS_start/1000),xright=(thisCDS_end/1000),ybottom=0.02,ytop=FstRange[2]-0.02, col='lightpink',lty = "solid", lwd=2, border='lightpink')
      }
    }
  }
    if (thisLG == 6) {
    thisLG_gff <- gffdata[gffdata[,"seqid"]=='GWHBJVT00000006',]
    # WindowCut=c(700000,1100000)
    thisLG_gff_cut <- thisLG_gff[thisLG_gff[,"start"]>WindowCut[1] & thisLG_gff[,"end"]<WindowCut[2],]
    all_genes_start <- unique(thisLG_gff_cut[,"start"])
    for (thisOne in 1:length(all_genes_start)) {
      # thisOne <- 1
      startPosition <- all_genes_start[thisOne]
      thisGene <- thisLG_gff_cut[thisLG_gff_cut[,"start"]==startPosition,]
      if ("CDS" %in% thisGene[,"type"]) {
         if (startPosition == 52889323) {
            cat("\n in gff in ROS1, gene Accession: ", thisGene[thisGene[,"type"]=="gene","Accession"])
         }
        if (startPosition == 52912351) {
          cat("\n in gff in ROS2, gene Accession: ", thisGene[thisGene[,"type"]=="gene","Accession"])
        }
        if (startPosition == 53060761) {
          cat("\n in gff in ELUTA, gene Accession: ", thisGene[thisGene[,"type"]=="gene","Accession"])
        }
      thisCDS_start <- thisGene[thisGene[,"type"]=="CDS","start"]
      thisCDS_end <- thisGene[thisGene[,"type"]=="CDS","end"]
      rect(xleft=(thisCDS_start/1000),xright=(thisCDS_end/1000),ybottom=0.02,ytop=FstRange[2]-0.02, col='lightpink',lty = "solid", lwd=2, border='lightpink')
      }
    }
  }
    if (thisLG == 7) {
      thisLG_gff <- gffdata[gffdata[,"seqid"]=='GWHBJVT00000007',]
      # WindowCut=c(700000,1100000)
      thisLG_gff_cut <- thisLG_gff[thisLG_gff[,"start"]>WindowCut[1] & thisLG_gff[,"end"]<WindowCut[2],]
      all_genes_start <- unique(thisLG_gff_cut[,"start"])
      for (thisOne in 1:length(all_genes_start)) {
        # thisOne <- 1
        startPosition <- all_genes_start[thisOne]
        thisGene <- thisLG_gff_cut[thisLG_gff_cut[,"start"]==startPosition,]
        if ("CDS" %in% thisGene[,"type"]) {
          thisCDS_start <- thisGene[thisGene[,"type"]=="CDS","start"]
          thisCDS_end <- thisGene[thisGene[,"type"]=="CDS","end"]
          rect(xleft=(thisCDS_start/1000),xright=(thisCDS_end/1000),ybottom=0.02,ytop=FstRange[2]-0.02, col='lightpink',lty = "solid", lwd=2, border='lightpink')
        }
      }
  }
    if (thisLG == 8) {
    thisLG_gff <- gffdata[gffdata[,"seqid"]=='GWHBJVT00000008',]
    # WindowCut=c(700000,1100000)
    thisLG_gff_cut <- thisLG_gff[thisLG_gff[,"start"]>WindowCut[1] & thisLG_gff[,"end"]<WindowCut[2],]
    all_genes_start <- unique(thisLG_gff_cut[,"start"])
    for (thisOne in 1:length(all_genes_start)) {
      # thisOne <- 1
      startPosition <- all_genes_start[thisOne]
      thisGene <- thisLG_gff_cut[thisLG_gff_cut[,"start"]==startPosition,]
      if ("CDS" %in% thisGene[,"type"]) {
        thisCDS_start <- thisGene[thisGene[,"type"]=="CDS","start"]
        thisCDS_end <- thisGene[thisGene[,"type"]=="CDS","end"]
        rect(xleft=(thisCDS_start/1000),xright=(thisCDS_end/1000),ybottom=0.02,ytop=FstRange[2]-0.02, col='lightpink',lty = "solid", lwd=2, border='lightpink')
      }
    }
  }
  }
 
  # Plot gene regions (in green)
  # From JIC
  # plotColGeneRegions <- T
  if (plotColGeneRegions ==T) {
    if (thisLG == 1) {
      # CRE
      rect(xleft=(928205/1000),xright=(930025/1000),ybottom=0,ytop=FstRange[2], col='lightgreen',lty = "solid", lwd=2, border='lightgreen')
       rect(xleft=(921293/1000),xright=(925659/1000),ybottom=0,ytop=FstRange[2], col='lightgreen',lty = "solid", lwd=2, border='lightgreen')
       rect(xleft=(994948/1000),xright=(995277/1000),ybottom=0,ytop=FstRange[2], col='lightgreen',lty = "solid", lwd=2, border='lightgreen')
    }
    if (thisLG == 2) {
      # AURINA
      rect(xleft=(1051685/1000),xright=(1053653/1000),ybottom=0.02,ytop=FstRange[2]-0.02, col='lightgreen',lty = "solid", lwd=2, border='lightgreen')
       rect(xleft=(53468062/1000),xright=(53469435/1000),ybottom=0.02,ytop=FstRange[2]-0.02, col='lightgreen',lty = "solid", lwd=2, border='lightgreen')
    }
    if (thisLG == 4) {
      # SULF
      rect(xleft=(38348572/1000),xright=(38350336/1000),ybottom=0.02,ytop=FstRange[2]-0.02, col='lightgreen',lty = "solid", lwd=2, border='lightgreen')
    }
    # RUBIA
    if (thisLG == 5) {
      rect(xleft=(6269966/1000),xright=(6272151/1000),ybottom=0.02,ytop=FstRange[2]-0.02, col='lightgreen',lty = "solid", lwd=2, border='lightgreen')
    }
    if (thisLG == 6) {
      # ROS 1
      rect(xleft=(52889323/1000),xright=(52889737/1000),ybottom=0.02,ytop=FstRange[2]-0.02, col='lightgreen',lty = "solid", lwd=2, border='lightgreen')
      # ROS 2
      rect(xleft=(52912351/1000),xright=(52912777/1000),ybottom=0.02,ytop=FstRange[2]-0.02, col='lightgreen',lty = "solid", lwd=2, border='lightgreen')
      # ELUTA
        rect(xleft=(53060761/1000),xright=(53062505/1000),ybottom=0.02,ytop=FstRange[2]-0.02, col='lightgreen',lty = "solid", lwd=2, border='lightgreen')
    }
  }
  
  # # Text on plot with chromosome information
  # if ((WindowCut[2]-WindowCut[1])>1000000) {
  #   text(x=((WindowCut[1]/1000)+6000),y=FstRange[2]-0.05,paste0("Chr ",thisLG),cex=1.5)
  #   #text(x=((WindowCut[1]/1000)+3000),y=FstRange[2]-0.08,paste0(WindowCut[1],":",WindowCut[2]),cex=1.5)
  # }
  # if ((WindowCut[2]-WindowCut[1])<1000000) {
  #   text(x=((WindowCut[1]/1000)+20),y=FstRange[2]-0.05,paste0("Chr ",thisLG,),cex=1.5)
  #   #text(x=((WindowCut[1]/1000)+30),y=FstRange[2]-0.1,paste0(as.character(WindowCut[1],":",WindowCut[2])),cex=1.5)
  # }
  # 
  
  # plot lines Fst
  # background
  
    allChr_Fst_WG_elsewhere <- WindowData %>% filter(LG==thisLG)
    #allChr_Fst_WG_elsewhere[,"LG"]
   # points(allChr_Fst_WG_elsewhere[,"windowMid_graph"]/1000, allChr_Fst_WG_elsewhere[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="background","pointColour"], lwd=0.7, pch=21,cex=0.7)
  
    allChr_Fst_WG_elsewhere_cut <- allChr_Fst_WG_elsewhere %>% filter(windowMid>WindowCut[1] & windowMid<WindowCut[2])

    # Fst
    # lwdFst <- 1.2
   lines(allChr_Fst_WG_elsewhere_cut[,"windowMid"]/1000, allChr_Fst_WG_elsewhere_cut[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="background","pointColour"], lwd=lwdFst, pch=2,cex=0)
   # Dxy DxyPlot,piWPlot
   if (!is.na(DxyPlot)) {
     lines(allChr_Fst_WG_elsewhere_cut[,"windowMid"]/1000, allChr_Fst_WG_elsewhere_cut[,DxyPlot]*10,       col='blue',bg=colourTable[colourTable[,"Region"]=="background","pointColour"], lwd=lwdFst*0.5, pch=2,cex=0)
   }
    # Pi_w
    if (!is.na(piWPlot)) {
       lines(allChr_Fst_WG_elsewhere_cut[,"windowMid"]/1000, allChr_Fst_WG_elsewhere_cut[,piWPlot]*10,       col='red',bg=colourTable[colourTable[,"Region"]=="background","pointColour"], lwd=lwdFst*0.5, pch=2,cex=0)
    }
 
  # commence Clines plot
  clineData_filter <- clineData[clineData[,pool_deltaP]>=deltaP,]

  print(nrow(clineData_filter))
  # remove negative cline widths & those >12km wide
  # filterNegativeClines =T
  if (filterNegativeClines==T) {
      clineData_filter <- clineData_filter[clineData_filter[,ClineEst[2]]>0 & clineData_filter[,ClineEst[2]]<12000,]
  }
  clineData_filter[,ClineEst[2]] <- as.numeric(clineData_filter[,ClineEst[2]])/1000

  # WidthQuantiles <- quantile(clineData[,"width"], na.rm=T, c(0.5, 0.05))
  # lines(x=c(0,max(range(graphPos[,"windowMid_graph"]))),y=c(as.numeric(WidthQuantiles[1]),as.numeric(WidthQuantiles[1]))/1000,col = "black", lwd=1.5,lty = 3)
  # lines(x=c(0,max(range(graphPos[,"windowMid_graph"]))),y=c(as.numeric(WidthQuantiles[2]),as.numeric(WidthQuantiles[2]))/1000,col = "black", lwd=1.5,lty = 3)
  
  # background loci
  allChr_Clines_WG_elsewhere <- clineData_filter %>% filter(ColourGene==-9)
  
  # Chr 1 & CREMOSA 
  allChr_Clines_Chr1_elsewhere <- clineData_filter %>% filter(LG==1 & ColourGene==-9)
  allChr_Clines_CREMOSA <- clineData_filter %>% filter(ColourGene=="CREMOSA" & distanceToGene =="inGene")
  allChr_Clines_CREMOSAlinked <- clineData_filter %>% filter(ColourGene=="CREMOSA") %>% filter(distanceToGene %in% linkedAreas)
  
  # Chr 2 & FLAVIA/Aurina 
  allChr_Clines_Chr2_elsewhere <- clineData_filter %>% filter(LG==2 & ColourGene==-9)
  allChr_Clines_FLAVIA <- clineData_filter %>% filter(ColourGene=="FLAVIA" & distanceToGene =="inGene")
  allChr_Clines_FLAVIAlinked <- clineData_filter %>% filter(ColourGene=="FLAVIA") %>% filter(distanceToGene %in% linkedAreas) 
  allChr_Clines_AURINAlinked <- clineData_filter %>% filter(ColourGene=="AURINA") %>% filter(distanceToGene %in% linkedAreas) 

  # Chr 3 
  allChr_Clines_Chr3_elsewhere <- clineData_filter %>% filter(LG==3 & ColourGene==-9)
  
  # Chr 4 & SULF  
  allChr_Clines_Chr4_elsewhere <- clineData_filter %>% filter(LG==4 & ColourGene==-9)
  allChr_Clines_SULF <- clineData_filter %>% filter(ColourGene=="SULF" & distanceToGene =="inGene")
  allChr_Clines_SULFlinked <- clineData_filter %>% filter(ColourGene=="SULF") %>% filter(distanceToGene %in% linkedAreas) 
  
  # Chr 5 & RUBIA 
  # 721 clines outside FLAVIA
  allChr_Clines_Chr5_elsewhere <- clineData_filter %>% filter(LG==5 & ColourGene==-9)
  allChr_Clines_RUBIA <- clineData_filter %>% filter(ColourGene=="RUBIA" & distanceToGene =="inGene")
  allChr_Clines_RUBIAlinked <- clineData_filter %>% filter(ColourGene=="RUBIA") %>% filter(distanceToGene %in% linkedAreas)

  # Chr 6 and ROS EL 
  allChr_Clines_Chr6_elsewhere <- clineData_filter %>% filter(LG==6 & ColourGene==-9)
  
  allChr_Clines_ROS1 <- clineData_filter %>% filter(ColourGene=="ROS1" & distanceToGene =="inGene")
  allChr_Clines_ROS2 <- clineData_filter %>% filter(ColourGene=="ROS2" & distanceToGene =="inGene")
  # allChr_Clines_ROS3 <- clineData_filter %>% filter(ColourGene=="ROS3" & distanceToGene =="inGene") # none in ROS3
  allChr_Clines_EL <- clineData_filter %>% filter(ColourGene=="ELUTA" & distanceToGene =="inGene")
  allChr_Clines_ROSelLinked <- clineData_filter %>% filter(ColourGene=="ROSEA" | ColourGene=="ROS1" | ColourGene=="ROS2" | ColourGene=="ELUTA") %>% filter(distanceToGene %in% linkedAreas) 
  
  # Chr 7 
  allChr_Clines_Chr7_elsewhere <- clineData_filter %>% filter(LG==7 & ColourGene==-9)

  # Chr 8 
  allChr_Clines_Chr8 <- clineData_filter %>% filter(LG==8)
  
  # combine all the 'elsewhere' loci not linked to known colour genes
  par(new=T)
  plot(x=NA,xlab="",ylab="", main="", las=2, bty="n",cex=0.5,yaxt="n",type='n',xaxt="n",
       xlim=c((WindowCut[1]/1000),(WindowCut[2]/1000)), ylim=rev(c(-1.2,14.2)),
       cex.axis=1.4,cex.lab=1.4,col=rgb(100,100,100,120,maxColorValue=255), pch=16)
  axis(4, at=rev(c(0,2,4,6,8,10,12,14)), labels=NA,col.axis="black", las=1,cex.axis=1,lwd=4,pos=(WindowCut[2]/1000), tck = -.03)
  axis(4, at=rev(c(0,2,4,6,8,10,12,14)), labels=rev(c(0,2,4,6,8,10,12,14)), col.axis="black", las=1, cex.axis=1.4, lwd=4, pos=(WindowCut[2]/1000))
  mtext("cline width",side=4,cex=1.4,line=1.5)
  
  # add points for clines
   
   if (thisLG == 1) {
     # colourTable[colourTable[,"Region"]=="background","pointColour"]
      points(allChr_Clines_Chr1_elsewhere[,"position"]/1000, allChr_Clines_Chr1_elsewhere[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="background","pointColour"], lwd=0.7, pch=21,cex=clineCexBackground)
      points(allChr_Clines_CREMOSAlinked[,"position"]/1000, allChr_Clines_CREMOSAlinked[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="CREMOSAlinked","pointColour"], lwd=0.9, pch=21,cex=clineCexGene)
      points(allChr_Clines_CREMOSA[,"position"]/1000, allChr_Clines_CREMOSA[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="CREMOSA","pointColour"], lwd=0.9, pch=21,cex=clineCexGene)
      
   }
   if (thisLG == 2) {
      points(allChr_Clines_Chr2_elsewhere[,"position"]/1000, allChr_Clines_Chr2_elsewhere[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="background","pointColour"], lwd=0.7, pch=21,cex=clineCexBackground)
       points(allChr_Clines_AURINAlinked[,"position"]/1000, allChr_Clines_AURINAlinked[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="AURINAlinked","pointColour"], lwd=0.9, pch=21,cex=clineCexGene)
      #points(allChr_Clines_AURINA[,"position"]/1000, allChr_Clines_AURINA[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="AURINA","pointColour"], lwd=0.9, pch=23,cex=clineCexGene)
      points(allChr_Clines_FLAVIAlinked[,"position"]/1000, allChr_Clines_FLAVIAlinked[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="FLAVIAlinked","pointColour"], lwd=0.9, pch=21,cex=clineCexGene)
      points(allChr_Clines_FLAVIA[,"position"]/1000, allChr_Clines_FLAVIA[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="FLAVIA","pointColour"], lwd=0.9, pch=21,cex=clineCexGene)
      
    }
    if (thisLG == 3) {
      points(allChr_Clines_Chr3_elsewhere[,"position"]/1000, allChr_Clines_Chr3_elsewhere[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="background","pointColour"], lwd=0.7, pch=21,cex=clineCexBackground)
    }
    if (thisLG == 4) {
      points(allChr_Clines_Chr4_elsewhere[,"position"]/1000, allChr_Clines_Chr4_elsewhere[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="background","pointColour"], lwd=0.7, pch=21,cex=clineCexBackground)
      # SULF
     points(allChr_Clines_SULFlinked[,"position"]/1000, allChr_Clines_SULFlinked[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="SULFlinked","pointColour"], lwd=0.7, pch=21,cex=clineCexGene)
     
     # SULF KASP
      points((38316531/1000), 14.1, col='black',bg='black', lwd=0.7, pch=21,cex=clineCexGene)
      points((38397179/1000), 14.1, col='black',bg='black', lwd=0.7, pch=21,cex=clineCexGene)

     # KASP marker positions at SULF
# Chr4:38,316,531
# Chr4:38,397,179

    }
     
  if (thisLG == 5) {
  # background
      points(allChr_Clines_Chr5_elsewhere[,"position"]/1000, allChr_Clines_Chr5_elsewhere[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="background","pointColour"], lwd=0.7, pch=21,cex=clineCexBackground)
  # RUBIA
      points(allChr_Clines_RUBIAlinked[,"position"]/1000, allChr_Clines_RUBIAlinked[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="RUBIAlinked","pointColour"], lwd=0.9, pch=21,cex=clineCexGene)
      points(allChr_Clines_RUBIA[,"position"]/1000, allChr_Clines_RUBIA[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="RUBIA","pointColour"], lwd=0.9, pch=21,cex=clineCexGene)
      
      # KASPpos <- c(6286923,7600)
      # plot KASP cline as black point
      if (all(!is.na(KASPpos))) {
         points((6286923/1000), 7.6, col='black',bg='black', lwd=0.7, pch=21,cex=clineCexGene)
      }
    }
  
  if (thisLG == 6) {
  # background
      points(allChr_Clines_Chr6_elsewhere[,"position"]/1000, allChr_Clines_Chr6_elsewhere[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="background","pointColour"], lwd=0.7, pch=21,cex=clineCexBackground)
      # ROS EL
   points(allChr_Clines_ROSelLinked[,"position"]/1000, allChr_Clines_ROSelLinked[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="ROSelLinked","pointColour"], lwd=0.7, pch=21,cex=clineCexGene)
         points(allChr_Clines_EL[,"position"]/1000, allChr_Clines_EL[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="ELUTA","pointColour"], lwd=0.7, pch=21,cex=clineCexGene)
            points(allChr_Clines_ROS1[,"position"]/1000, allChr_Clines_ROS1[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="ROS","pointColour"], lwd=0.7, pch=21,cex=clineCexGene)
      points(allChr_Clines_ROS2[,"position"]/1000, allChr_Clines_ROS2[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="ROS","pointColour"], lwd=0.7, pch=21,cex=clineCexGene)
  }
     
  if (thisLG == 7) {
      points(allChr_Clines_Chr7_elsewhere[,"position"]/1000, allChr_Clines_Chr7_elsewhere[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="background","pointColour"], lwd=0.7, pch=21,cex=clineCexBackground)
    
  }
  if (thisLG == 8) {
      points(allChr_Clines_Chr8_elsewhere[,"position"]/1000, allChr_Clines_Chr8_elsewhere[,ClineEst[2]],       col='black',bg=colourTable[colourTable[,"Region"]=="background","pointColour"], lwd=0.7, pch=21,cex=clineCexBackground)
    
  }
}

```

```{r function for genome scans zoomed plots Differential Gene Expression}
genomeScanZoomDFgeneExp <- function(DFgeneExpData,WindowData,thisLG,WindowCut,dataPlot,DFgeneExpRange,gffdata,gffPlot,plotColGeneRegions,colourTable,startPlot,plotPos1,cexBackground,cexGene) {
  # par(mfrow=c(1,1))
  # clineData <- allChr_Clines_filterDpth15; WindowData <- allChr_Windows; deltaP <- 0.9; pool_deltaP <- "AFD_15_20"; startPlot = F; gffdata <- gff.data
  # plotPos1 = c(0.04, 0.96, 0.3, 0.5) ; thisLG <- 5; WindowCut=c(6100000,6400000); FstRange <- c(0,0.5); dataPlot <- "log2FoldChange"
  # ClineEst <- c('centre','width'); filterNegativeClines = T; colourTable=colourChoice
  # cexBackground <- 1.2; cexGene <- 1.3; gffPlot =T
  # cat("\nclinal loci start n=",nrow(clineData))
  # DFgeneExp <- RNAseqData
  # DFgeneExpData <- RNAseqData; DFgeneExpRange <- c(-4,4)
  # note that the windowCut object is the range of positions to zoom in on. It has to be based on original positions on each LG and not on the graphing positions, so it lines up with gene annotations
  # colnames(allChr_Windows)[1:12]
  # allChr_Fst_RUBIA[,3145:3153]
  # allChr_Fst_RUBIAlinked[,3145:3153]
  # WindowCut=c(6100000,6400000) for RUBIA
  

  par(mar=c(2,2,1,1))
  par(oma=c(1,1,1,1))
  par(fig = plotPos1, mar=c(2,2,1,1),new=((startPlot==TRUE)==FALSE))
  plot(as.numeric(DFgeneExpData[,"start"])/1000, as.numeric(DFgeneExpData[,dataPlot]), xlab="", ylab="", main="", las=1, bty="n",cex=0.5,type='n',yaxt='n',lwd=2,xaxt='n', xlim=c((WindowCut[1]/1000),(WindowCut[2]/1000)), ylim=c(DFgeneExpRange[1],DFgeneExpRange[2]),
       cex.axis=1.4,cex.lab=1.4,col=rgb(100,100,100,120,maxColorValue=255), pch=16)
     
  # layout(matrix(1:2,ncol=2), width = c(2,1),height = c(1,1))
  # par(fig = c(0.03,0.97,0,1), mar=c(2,2,1,1))
  # plot(as.numeric(DFgeneExpData[,"start"])/1000, as.numeric(DFgeneExpData[,dataPlot]), xlab="", ylab="", main="", las=1, bty="n",cex=0.5,type='n',yaxt='n',lwd=2,xaxt='n', xlim=c((WindowCut[1]/1000),(WindowCut[2]/1000)), ylim=c(DFgeneExpRange[1],DFgeneExpRange[2]),
  #      cex.axis=1.4,cex.lab=1.4,col=rgb(100,100,100,120,maxColorValue=255), pch=16)
  # 
  
  if ((WindowCut[2]-WindowCut[1])>10000000) {
     axis(1, at=seq(WindowCut[1]/1000,WindowCut[2]/1000,10000), labels=NA,col.axis="black", las=1,cex.axis=1.1,lwd=4,pos=DFgeneExpRange[1], tck = -.03)
   axis(1,at=seq(WindowCut[1]/1000,WindowCut[2]/1000,10000), labels=seq(WindowCut[1]/1000,WindowCut[2]/1000,10000),col.axis="black", las=1,cex.axis=1.4,lwd=4,pos=DFgeneExpRange[1])
    
  }
  if ((WindowCut[2]-WindowCut[1])<10000000 & (WindowCut[2]-WindowCut[1])>1000000) {
     axis(1, at=seq(WindowCut[1]/1000,WindowCut[2]/1000,1000), labels=NA,col.axis="black", las=1,cex.axis=1.1,lwd=4,pos=DFgeneExpRange[1], tck = -.03)
   axis(1,at=seq(WindowCut[1]/1000,WindowCut[2]/1000,1000), labels=seq(WindowCut[1]/1000,WindowCut[2]/1000,1000),col.axis="black", las=1,cex.axis=1.4,lwd=4,pos=DFgeneExpRange[1])
    
  }
  if ((WindowCut[2]-WindowCut[1])<1000000) {
   axis(1, at=seq(WindowCut[1]/1000,WindowCut[2]/1000,50), labels=NA,col.axis="black", las=1,cex.axis=1.1,lwd=4,pos=DFgeneExpRange[1], tck = -.03)
   axis(1,at=seq(WindowCut[1]/1000,WindowCut[2]/1000,50), labels=seq(WindowCut[1]/1000,WindowCut[2]/1000,50),col.axis="black", las=1,cex.axis=1.4,lwd=4,pos=DFgeneExpRange[1])
  }
  
  axis(2, at=seq(min(DFgeneExpRange),max(DFgeneExpRange),1), labels=NA,col.axis="black", las=1,cex.axis=1,lwd=4,pos=WindowCut[1]/1000, tck = -.03)
  axis(2, at=seq(min(DFgeneExpRange),max(DFgeneExpRange),1), labels=seq(min(DFgeneExpRange),max(DFgeneExpRange),1),col.axis="black", las=1, cex.axis=1.4, lwd=4, pos=(WindowCut[1]/1000))
  lines(x=c(WindowCut[2]/1000,WindowCut[2]/1000), y=c(min(DFgeneExpRange),max(DFgeneExpRange)),lwd=4)
  lines(x=c(WindowCut[1]/1000,WindowCut[2]/1000), y=c(DFgeneExpRange[2],DFgeneExpRange[2]),lwd=4)
  lines(x=c(WindowCut[1]/1000,WindowCut[1]/1000), y=c(DFgeneExpRange[1],DFgeneExpRange[2]),lwd=4)

  mtext('genome position (Kbp)',side=1,cex=1.4,line=2.7)
  mtext('log2FC',side=2,cex=1.4,line=1.2)
  #mtext(paste0("Dxy: pi_w"),side=2,cex=1.4,line=2.8)

  # plot gene regions from gff file annotation (in pink)
  if (gffPlot ==T) {
    if (thisLG == 1) {
      # thisLG ==1
      thisLG_gff <- gffdata[gffdata[,"seqid"]=='GWHBJVT00000001',]
      # WindowCut=c(700000,1100000)
      thisLG_gff_cut <- thisLG_gff[thisLG_gff[,"start"]>WindowCut[1] & thisLG_gff[,"end"]<WindowCut[2],]
      all_genes_start <- unique(thisLG_gff_cut[,"start"])
      for (thisOne in 1:length(all_genes_start)) {
        # thisOne <- 1
        startPosition <- all_genes_start[thisOne]
        thisGene <- thisLG_gff_cut[thisLG_gff_cut[,"start"]==startPosition,]
        if ("CDS" %in% thisGene[,"type"]) {
          thisCDS_start <- thisGene[thisGene[,"type"]=="CDS","start"]
          thisCDS_end <- thisGene[thisGene[,"type"]=="CDS","end"]
          # CREMOSA 1 - a gene in same position in gff
          if (startPosition == 928205) {
              cat("\n in gff in CREMOSA 1, gene Accession: ", thisGene[thisGene[,"type"]=="gene","Accession"])
          }
          if (startPosition == 921293) {
              cat("\n in gff in CREMOSA 2 gene Accession: ", thisGene[thisGene[,"type"]=="gene","Accession"])
          }
          if (startPosition == 994948) {
              cat("\n in gff in CREMOSA 3 gene Accession: ", thisGene[thisGene[,"type"]=="gene","Accession"])
          }
          rect(xleft=(thisCDS_start/1000),xright=(thisCDS_end/1000),ybottom=DFgeneExpRange[1]+0.06,ytop=DFgeneExpRange[2]-0.06, col='lightpink',lty = "solid", lwd=2, border='lightpink')
        }
      }
    }
    if (thisLG == 2) {
    thisLG_gff <- gffdata[gffdata[,"seqid"]=='GWHBJVT00000002',]
    # WindowCut=c(700000,1100000)
    thisLG_gff_cut <- thisLG_gff[thisLG_gff[,"start"]>WindowCut[1] & thisLG_gff[,"end"]<WindowCut[2],]
    all_genes_start <- unique(thisLG_gff_cut[,"start"])

    for (thisOne in 1:length(all_genes_start)) {
      # thisOne <- 1
      startPosition <- all_genes_start[thisOne]
      thisGene <- thisLG_gff_cut[thisLG_gff_cut[,"start"]==startPosition,]
      if ("CDS" %in% thisGene[,"type"]) {
        if (startPosition == 1051685) {
          cat("\n in gff in AURINA, gene Accession: ", thisGene[thisGene[,"type"]=="gene","Accession"])
        }
        if (startPosition == 53468062) {
          cat("\n in gff in FLAVIA, gene Accession: ", thisGene[thisGene[,"type"]=="gene","Accession"])
        }
        thisCDS_start <- thisGene[thisGene[,"type"]=="CDS","start"]
        thisCDS_end <- thisGene[thisGene[,"type"]=="CDS","end"]
        rect(xleft=(thisCDS_start/1000),xright=(thisCDS_end/1000),ybottom=DFgeneExpRange[1]+0.06,ytop=DFgeneExpRange[2]-0.06, col='lightpink',lty = "solid", lwd=2, border='lightpink')
      }
      
    }
  }
    if (thisLG == 3) {
    thisLG_gff <- gffdata[gffdata[,"seqid"]=='GWHBJVT00000003',]
    # WindowCut=c(700000,1100000)
    thisLG_gff_cut <- thisLG_gff[thisLG_gff[,"start"]>WindowCut[1] & thisLG_gff[,"end"]<WindowCut[2],]
    all_genes_start <- unique(thisLG_gff_cut[,"start"])

    for (thisOne in 1:length(all_genes_start)) {
      # thisOne <- 1
      thisGene <- thisLG_gff_cut[thisLG_gff_cut[,"start"]==startPosition,]
      if ("CDS" %in% thisGene[,"type"]) {
        thisCDS_start <- thisGene[thisGene[,"type"]=="CDS","start"]
        thisCDS_end <- thisGene[thisGene[,"type"]=="CDS","end"]
        rect(xleft=(thisCDS_start/1000),xright=(thisCDS_end/1000),ybottom=DFgeneExpRange[1]+0.06,ytop=DFgeneExpRange[2]-0.06, col='lightpink',lty = "solid", lwd=2, border='lightpink')
      }
    }
  }  
    if (thisLG == 4) {
    thisLG_gff <- gffdata[gffdata[,"seqid"]=='GWHBJVT00000004',]
    # WindowCut=c(700000,1100000)
    thisLG_gff_cut <- thisLG_gff[thisLG_gff[,"start"]>WindowCut[1] & thisLG_gff[,"end"]<WindowCut[2],]
    all_genes_start <- unique(thisLG_gff_cut[,"start"])

    for (thisOne in 1:length(all_genes_start)) {
      # thisOne <- 1
      startPosition <- all_genes_start[thisOne]
      thisGene <- thisLG_gff_cut[thisLG_gff_cut[,"start"]==startPosition,]
      if ("CDS" %in% thisGene[,"type"]) {
        if (startPosition == 38348572) {
          cat("\n in gff in SULF, gene Accession: ", thisGene[thisGene[,"type"]=="gene","Accession"])
        }
        thisCDS_start <- thisGene[thisGene[,"type"]=="CDS","start"]
        thisCDS_end <- thisGene[thisGene[,"type"]=="CDS","end"]
        rect(xleft=(thisCDS_start/1000),xright=(thisCDS_end/1000),ybottom=DFgeneExpRange[1]+0.06,ytop=DFgeneExpRange[2]-0.06, col='lightpink',lty = "solid", lwd=2, border='lightpink')
      }
    }
  }
    if (thisLG == 5) {
    thisLG_gff <- gffdata[gffdata[,"seqid"]=='GWHBJVT00000005',]
    # WindowCut=c(700000,1100000)
    thisLG_gff_cut <- thisLG_gff[thisLG_gff[,"start"]>WindowCut[1] & thisLG_gff[,"end"]<WindowCut[2],]
    all_genes_start <- unique(thisLG_gff_cut[,"start"])
    #fileName <- paste0("gff_",thisLG,"_",WindowCut[1],".txt")
    #write.file(thisLG_gff_cut,fileName)
    
    # CREMOSA 1 - a gene in same position in gff
    #928205 %in% all_genes_start
    
    for (thisOne in 1:length(all_genes_start)) {
      # thisOne <- 1
      startPosition <- all_genes_start[thisOne]
      thisGene <- thisLG_gff_cut[thisLG_gff_cut[,"start"]==startPosition,]
      if ("CDS" %in% thisGene[,"type"]) {
        if (startPosition == 6269966) {
          cat("\n in gff in RUBIA, gene Accession: ", thisGene[thisGene[,"type"]=="gene","Accession"])
        }
        thisCDS_start <- thisGene[thisGene[,"type"]=="CDS","start"]
        thisCDS_end <- thisGene[thisGene[,"type"]=="CDS","end"]
        rect(xleft=(thisCDS_start/1000),xright=(thisCDS_end/1000),ybottom=DFgeneExpRange[1]+0.06,ytop=DFgeneExpRange[2]-0.06, col='lightpink',lty = "solid", lwd=2, border='lightpink')
      }
    }
  }
    if (thisLG == 6) {
    thisLG_gff <- gffdata[gffdata[,"seqid"]=='GWHBJVT00000006',]
    # WindowCut=c(700000,1100000)
    thisLG_gff_cut <- thisLG_gff[thisLG_gff[,"start"]>WindowCut[1] & thisLG_gff[,"end"]<WindowCut[2],]
    all_genes_start <- unique(thisLG_gff_cut[,"start"])
    for (thisOne in 1:length(all_genes_start)) {
      # thisOne <- 1
      startPosition <- all_genes_start[thisOne]
      thisGene <- thisLG_gff_cut[thisLG_gff_cut[,"start"]==startPosition,]
      if ("CDS" %in% thisGene[,"type"]) {
         if (startPosition == 52889323) {
            cat("\n in gff in ROS1, gene Accession: ", thisGene[thisGene[,"type"]=="gene","Accession"])
         }
        if (startPosition == 52912351) {
          cat("\n in gff in ROS2, gene Accession: ", thisGene[thisGene[,"type"]=="gene","Accession"])
        }
        if (startPosition == 53060761) {
          cat("\n in gff in ELUTA, gene Accession: ", thisGene[thisGene[,"type"]=="gene","Accession"])
        }
      thisCDS_start <- thisGene[thisGene[,"type"]=="CDS","start"]
      thisCDS_end <- thisGene[thisGene[,"type"]=="CDS","end"]
      rect(xleft=(thisCDS_start/1000),xright=(thisCDS_end/1000),ybottom=DFgeneExpRange[1]+0.06,ytop=DFgeneExpRange[2]-0.06, col='lightpink',lty = "solid", lwd=2, border='lightpink')
      }
    }
  }
    if (thisLG == 7) {
      thisLG_gff <- gffdata[gffdata[,"seqid"]=='GWHBJVT00000007',]
      # WindowCut=c(700000,1100000)
      thisLG_gff_cut <- thisLG_gff[thisLG_gff[,"start"]>WindowCut[1] & thisLG_gff[,"end"]<WindowCut[2],]
      all_genes_start <- unique(thisLG_gff_cut[,"start"])
      for (thisOne in 1:length(all_genes_start)) {
        # thisOne <- 1
        startPosition <- all_genes_start[thisOne]
        thisGene <- thisLG_gff_cut[thisLG_gff_cut[,"start"]==startPosition,]
        if ("CDS" %in% thisGene[,"type"]) {
          thisCDS_start <- thisGene[thisGene[,"type"]=="CDS","start"]
          thisCDS_end <- thisGene[thisGene[,"type"]=="CDS","end"]
          rect(xleft=(thisCDS_start/1000),xright=(thisCDS_end/1000),ybottom=DFgeneExpRange[1]+0.06,ytop=DFgeneExpRange[2]-0.06, col='lightpink',lty = "solid", lwd=2, border='lightpink')
        }
      }
  }
    if (thisLG == 8) {
    thisLG_gff <- gffdata[gffdata[,"seqid"]=='GWHBJVT00000008',]
    # WindowCut=c(700000,1100000)
    thisLG_gff_cut <- thisLG_gff[thisLG_gff[,"start"]>WindowCut[1] & thisLG_gff[,"end"]<WindowCut[2],]
    all_genes_start <- unique(thisLG_gff_cut[,"start"])
    for (thisOne in 1:length(all_genes_start)) {
      # thisOne <- 1
      startPosition <- all_genes_start[thisOne]
      thisGene <- thisLG_gff_cut[thisLG_gff_cut[,"start"]==startPosition,]
      if ("CDS" %in% thisGene[,"type"]) {
        thisCDS_start <- thisGene[thisGene[,"type"]=="CDS","start"]
        thisCDS_end <- thisGene[thisGene[,"type"]=="CDS","end"]
        rect(xleft=(thisCDS_start/1000),xright=(thisCDS_end/1000),ybottom=DFgeneExpRange[1]+0.06,ytop=DFgeneExpRange[2]-0.06, col='lightpink',lty = "solid", lwd=2, border='lightpink')
      }
    }
  }
  }
 
  # Plot gene regions (in green)
  # From JIC
  # plotColGeneRegions <- T
  if (plotColGeneRegions ==T) {
    if (thisLG == 1) {
      # CRE
      rect(xleft=(928205/1000),xright=(930025/1000),ybottom=DFgeneExpRange[1]+0.06,ytop=DFgeneExpRange[2]-0.06, col='lightgreen',lty = "solid", lwd=2, border='lightgreen')
       rect(xleft=(921293/1000),xright=(925659/1000),ybottom=DFgeneExpRange[1]+0.06,ytop=DFgeneExpRange[2]-0.06, col='lightgreen',lty = "solid", lwd=2, border='lightgreen')
       rect(xleft=(994948/1000),xright=(995277/1000),ybottom=DFgeneExpRange[1]+0.06,ytop=DFgeneExpRange[2]-0.06, col='lightgreen',lty = "solid", lwd=2, border='lightgreen')
    }
     if (thisLG == 2) {
      # AURINA
      rect(xleft=(1051685/1000),xright=(1053653/1000),ybottom=DFgeneExpRange[1]+0.06,ytop=DFgeneExpRange[2]-0.06, col='lightgreen',lty = "solid", lwd=2, border='lightgreen')
       rect(xleft=(53468062/1000),xright=(53469435/1000),ybottom=DFgeneExpRange[1]+0.06,ytop=DFgeneExpRange[2]-0.06, col='lightgreen',lty = "solid", lwd=2, border='lightgreen')
    }
    #
    if (thisLG == 4) {
      # SULF
      rect(xleft=(38348572/1000),xright=(38350336/1000),ybottom=DFgeneExpRange[1]+0.06,ytop=DFgeneExpRange[2]-0.06, col='lightgreen',lty = "solid", lwd=2, border='lightgreen')
    }
    # RUBIA
    if (thisLG == 5) {
      rect(xleft=(6269966/1000),xright=(6272151/1000),ybottom=DFgeneExpRange[1]+0.06,ytop=DFgeneExpRange[2]-0.06, col='lightgreen',lty = "solid", lwd=2, border='lightgreen')
    }
    if (thisLG == 6) {
      # ROS 1
      rect(xleft=(52889323/1000),xright=(52889737/1000),ybottom=DFgeneExpRange[1]+0.06,ytop=DFgeneExpRange[2]-0.06, col='lightgreen',lty = "solid", lwd=2, border='lightgreen')
      # ROS 2
      rect(xleft=(52912351/1000),xright=(52912777/1000),ybottom=DFgeneExpRange[1]+0.06,ytop=DFgeneExpRange[2]-0.06, col='lightgreen',lty = "solid", lwd=2, border='lightgreen')
      # ELUTA
        rect(xleft=(53060761/1000),xright=(53062505/1000),ybottom=DFgeneExpRange[1]+0.06,ytop=DFgeneExpRange[2]-0.06, col='lightgreen',lty = "solid", lwd=2, border='lightgreen')
    }
  }
  # zero dashed line
  lines(x=c(WindowCut[1]/1000,WindowCut[2]/1000), y=c(0,0),lwd=2,lty=2)

  # Text on plot with chromosome information
  #if ((WindowCut[2]-WindowCut[1])>1000000) {
   # text(x=((WindowCut[1]/1000)+6000),y=FstRange[2]-0.05,paste0("Chr ",thisLG),cex=1.5)
    #text(x=((WindowCut[1]/1000)+3000),y=FstRange[2]-0.08,paste0(WindowCut[1],":",WindowCut[2]),cex=1.5)
#  }
 # if ((WindowCut[2]-WindowCut[1])<1000000) {
  #  text(x=((WindowCut[1]/1000)+20),y=FstRange[2]-0.05,paste0("Chr ",thisLG),cex=1.5)
  #  #text(x=((WindowCut[1]/1000)+30),y=FstRange[2]-0.1,paste0(as.character(WindowCut[1],":",WindowCut[2])),cex=1.5)
  #}
  
  
  # plot differential gene expression 
  # remove NA
  DFgeneExpData_Log2FC <- DFgeneExpData %>% filter(!is.na(log2FoldChange))
  # add mid position for plotting
  midPosition <- ((DFgeneExpData_Log2FC[,"end"]-DFgeneExpData_Log2FC[,"start"])/2)+DFgeneExpData_Log2FC[,"start"]
  # log10_pval
  log10_pval <- (log(DFgeneExpData_Log2FC[,"padj"],10))*-1
  # heat pallete column
  log10_pval_heatcolour <- rev(heat.colors(5))[as.numeric(cut(log10_pval,breaks = 5))]
  # wesanderson pallete column
  log10_pval_Zissou1colour <- wes_palette("Zissou1", 10, type = "continuous")[as.numeric(cut(log10_pval,breaks = 10))]
  
  DFgeneExpData_Log2FC <- cbind(DFgeneExpData_Log2FC,midPosition,log10_pval,log10_pval_heatcolour,log10_pval_Zissou1colour)
  # # non significant ones
  DFgeneExpData_Log2FC_ns <- DFgeneExpData_Log2FC %>% filter(padj>0.05)
  DFgeneExpData_Log2FC_sig <- DFgeneExpData_Log2FC %>% filter(padj<=0.05)

  # # 0.05 < p < 0.01
  # DFgeneExpData_Log2FC_weakP <- DFgeneExpData_Log2FC %>% filter(padj>0.01 & padj<0.05)
  # # 0.01 < p < 0.001
  # DFgeneExpData_Log2FC_strongP <- DFgeneExpData_Log2FC %>% filter(padj<0.001)
  
  # ggplot(DFgeneExpData_Log2FC, aes(midPosition/1000, log2FoldChange))+
  #   geom_point(aes(color = log10_pval)) +
  #   scale_color_viridis(option = "D")+
  #   theme_minimal() +
  #   theme(legend.position = "bottom")
  

   # points(DFgeneExpData_Log2FC[,"midPosition"]/1000,DFgeneExpData_Log2FC[,"log2FoldChange"], col=DFgeneExpData_Log2FC$log10_pval_Zissou1colour,bg=DFgeneExpData_Log2FC$log10_pval_Zissou1colour, lwd=0.9, pch=21,cex=cexGene)

    points(DFgeneExpData_Log2FC_ns[,"midPosition"]/1000,DFgeneExpData_Log2FC_ns[,"log2FoldChange"], col='black',bg='lightgrey', lwd=0.9, pch=21,cex=cexGene)
    points(DFgeneExpData_Log2FC_sig[,"midPosition"]/1000,DFgeneExpData_Log2FC_sig[,"log2FoldChange"], col='black',bg=DFgeneExpData_Log2FC_sig$log10_pval_heatcolour, lwd=0.9, pch=21,cex=cexGene)
    
# startPlot <-F
#   par(fig = c(0.9,1,0.7,1), mar=c(0.5,0.5,1,1),new=((startPlot==TRUE)==FALSE))
# 
# legend_image <- as.raster(matrix(heat.colors(5), ncol=1))
# plot(c(0,2),c(0,8),type = 'n', axes = F,xlab = '', ylab = '', main = '-log_10 pvalue')
# text(x=1.5, y = seq(0,8), labels = seq(0,8))
# rasterImage(legend_image, 0, 0, 1,8)
#   
# 
# # plot points
#   points(DFgeneExpData_Log2FC_ns[,"midPosition"]/1000,DFgeneExpData_Log2FC_ns[,"log2FoldChange"], col='black',bg="lightgrey", lwd=0.9, pch=21,cex=cexGene)
# points(DFgeneExpData_Log2FC_weakP[,"midPosition"]/1000,DFgeneExpData_Log2FC_weakP[,"log2FoldChange"], col='black',bg="orange", lwd=0.9, pch=21,cex=cexGene)
# points(DFgeneExpData_Log2FC_strongP[,"midPosition"]/1000,DFgeneExpData_Log2FC_strongP[,"log2FoldChange"], col='black',bg="red", lwd=0.9, pch=21,cex=cexGene)
#     # Fst
#     # lwdFst <- 1.2
#    lines(allChr_Fst_WG_elsewhere_cut[,"windowMid"]/1000, allChr_Fst_WG_elsewhere_cut[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="background","pointColour"], lwd=lwdFst, pch=2,cex=0)
#   
#    # Dxy
#    lines(allChr_Fst_WG_elsewhere_cut[,"windowMid"]/1000, allChr_Fst_WG_elsewhere_cut[,"dXYfromMeanFstAdj_15_20"],       col='blue',bg=colourTable[colourTable[,"Region"]=="background","pointColour"], lwd=lwdFst, pch=2,cex=0)
#    # pi_w
#    lines(allChr_Fst_WG_elsewhere_cut[,"windowMid"]/1000, allChr_Fst_WG_elsewhere_cut[,"piBarAdj_15_20"],       col='red',bg=colourTable[colourTable[,"Region"]=="background","pointColour"], lwd=lwdFst, pch=2,cex=0)
   
  # # CRE and linked
  #  points(allChr_Fst_CREMOSAlinked[,"windowMid"]/1000, allChr_Fst_CREMOSAlinked[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="CREMOSAlinked","pointColour"], lwd=0.7, pch=21,cex=0.7)
  #   points(allChr_Fst_CREMOSA[,"windowMid"]/1000, allChr_Fst_CREMOSA[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="CREMOSA","pointColour"], lwd=0.7, pch=21,cex=0.7)
  # 
  # #  FLAVIA/Aurina 
  #  points(allChr_Fst_FLAVIAlinked[,"windowMid"]/1000, allChr_Fst_FLAVIAlinked[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="FLAVIAlinked","pointColour"], lwd=0.7, pch=21,cex=0.7)
  #  points(allChr_Fst_AURINAlinked[,"windowMid"]/1000, allChr_Fst_AURINAlinked[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="AURINAlinked","pointColour"], lwd=0.7, pch=21,cex=0.7)
  #   points(allChr_Fst_FLAVIA[,"windowMid"]/1000, allChr_Fst_FLAVIA[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="FLAVIA","pointColour"], lwd=0.7, pch=21,cex=0.7)
  #  points(allChr_Fst_AURINA[,"windowMid"]/1000, allChr_Fst_AURINA[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="AURINA","pointColour"], lwd=0.7, pch=21,cex=0.7)
  #   
  #  # SULF
  #    points(allChr_Fst_SULFlinked[,"windowMid"]/1000, allChr_Fst_SULFlinked[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="SULFlinked","pointColour"], lwd=0.7, pch=21,cex=0.7)
  #   points(allChr_Fst_SULF[,"windowMid"]/1000, allChr_Fst_SULF[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="SULFlinked","pointColour"], lwd=0.7, pch=21,cex=0.7)
  #    
  #   # RUBIA
  #   # range(allChr_Fst_RUBIAlinked[,"windowMid_graph"])
  #   # range(allChr_Fst_RUBIA[,"windowMid_graph"])
  #  points(allChr_Fst_RUBIAlinked[,"windowMid"]/1000, allChr_Fst_RUBIAlinked[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="RUBIAlinked","pointColour"], lwd=0.7, pch=21,cex=1)
  # points(allChr_Fst_RUBIA[,"windowMid"]/1000, allChr_Fst_RUBIA[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="RUBIAlinked","pointColour"], lwd=0.7, pch=21,cex=2)
  #  
  #     # ROS EL
  #  points(allChr_Fst_ROSelLinked[,"windowMid"]/1000, allChr_Fst_ROSelLinked[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="ROSelLinked","pointColour"], lwd=0.7, pch=21,cex=0.7)
  #        points(allChr_Fst_EL[,"windowMid"]/1000, allChr_Fst_EL[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="ELUTA","pointColour"], lwd=0.7, pch=21,cex=0.7)
  #           points(allChr_Fst_ROS1[,"windowMid"]/1000, allChr_Fst_ROS1[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="ROS","pointColour"], lwd=0.7, pch=21,cex=0.7)
  #     points(allChr_Fst_ROS2[,"windowMid"]/1000, allChr_Fst_ROS2[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="ROS","pointColour"], lwd=0.7, pch=21,cex=0.7)
  # points(allChr_Fst_ROS3[,"windowMid"]/1000, allChr_Fst_ROS3[,dataPlot],       col='black',bg=colourTable[colourTable[,"Region"]=="ROS","pointColour"], lwd=0.7, pch=21,cex=0.7)
  # 
   
   
    
  }

```

```{r genome scan zooms for LG1}
# LG 1
# colourChoice[colourChoice[,"Region"]=="background",2] <- c((rgb(220,220,220,180,maxColorValue=255)))

range(allChr_Windows[allChr_Windows[,"LG"]==1,"windowStart"])
nrow(allChr_Clines_filterDpth15[allChr_Clines_filterDpth15["LG"]==1,])

# make background cline width points a little more grey
# previous colour "#DCDCDC32" 
colourChoice[colourChoice[,"Region"]=="background","pointColour"] <- "lightgrey" 

# All of LG1
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=1,WindowCut=c(0,71870000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.4),lwdFst = 1.2,DxyPlot=NA,piWPlot= NA,deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=F,plotColGeneRegions=T,colourTable = colourChoice, startPlot =T, plotPos1 = c(0.01, 1.0, 0.80, 1), clineCexBackground = 1.2, clineCexGene=1.3)

# note. black line = Fst, blue line Dxy, red line = pi_w, 

#CREMOSA
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=1,
                            WindowCut=c(700000,1100000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.4),lwdFst = 3,DxyPlot="dXYfromMeanFstAdj_15_20",piWPlot= "piBarAdj_15_20",deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data, gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1.0, 0.55, 0.75), clineCexBackground = 1.2, clineCexGene=1.3)

  #DxyPlot ="dXYfromMeanFstAdj_15_20"
  # piWPlot <- "piBarAdj_15_20"

# High Fst region 1
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=1, WindowCut=c(47000000,48000000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.4),lwdFst = 3,DxyPlot="dXYfromMeanFstAdj_15_20",piWPlot= "piBarAdj_15_20",deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1.0, 0.3, 0.5), clineCexBackground = 1.2, clineCexGene=1.3)

# High Fst region 2
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=1, WindowCut=c(50000000,51000000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.4),lwdFst = 3,DxyPlot="dXYfromMeanFstAdj_15_20",piWPlot= "piBarAdj_15_20",deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1.0, 0.05, 0.25), clineCexBackground = 1.2, clineCexGene=1.3)

# PDF version
pdf("FigSx_GenomeScanPanel_Chr1_Pool15_20.pdf",width = 20, height = 27, pointsize = 25)

# All of LG1

range(allChr_Windows[allChr_Windows[,"LG"]==1,"windowStart"])
nrow(allChr_Clines_filterDpth15[allChr_Clines_filterDpth15["LG"]==1,])
# All of LG1
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=1,WindowCut=c(0,71870000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.4),lwdFst = 1.2,DxyPlot=NA,piWPlot= NA,deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=F,plotColGeneRegions=T,colourTable = colourChoice, startPlot =T, plotPos1 = c(0.01, 1.0, 0.80, 1), clineCexBackground = 1.2, clineCexGene=1.3)

# note. black line = Fst, blue line Dxy, red line = pi_w, 

#CREMOSA
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=1,
                            WindowCut=c(700000,1100000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.4),lwdFst = 5,DxyPlot="dXYfromMeanFstAdj_15_20",piWPlot= "piBarAdj_15_20",deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data, gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1.0, 0.55, 0.75), clineCexBackground = 1.2, clineCexGene=1.3)

  #DxyPlot ="dXYfromMeanFstAdj_15_20"
  # piWPlot <- "piBarAdj_15_20"


# High Fst region 1
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=1, WindowCut=c(47000000,48000000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.4),lwdFst = 5,DxyPlot="dXYfromMeanFstAdj_15_20",piWPlot= "piBarAdj_15_20",deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1.0, 0.3, 0.5), clineCexBackground = 1.2, clineCexGene=1.3)

# High Fst region 2
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=1, WindowCut=c(50000000,51000000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.4),lwdFst = 5,DxyPlot="dXYfromMeanFstAdj_15_20",piWPlot= "piBarAdj_15_20",deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1.0, 0.05, 0.25), clineCexBackground = 1.2, clineCexGene=1.3)

dev.off()

# PDF version
pdf("FigSx_GenomeScanPanel_Chr1_Pool16_19.pdf",width = 20, height = 27, pointsize = 25)

# All of LG1

range(allChr_Windows[allChr_Windows[,"LG"]==1,"windowStart"])
nrow(allChr_Clines_filterDpth15[allChr_Clines_filterDpth15["LG"]==1,])
# All of LG1
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=1,WindowCut=c(0,71870000),dataPlot="FstfromMeanPiAdj_16_19",FstRange = c(0,0.4),lwdFst = 1.2,DxyPlot=NA,piWPlot= NA,deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=F,plotColGeneRegions=T,colourTable = colourChoice, startPlot =T, plotPos1 = c(0.01, 1.0, 0.80, 1), clineCexBackground = 1.2, clineCexGene=1.3)

# note. black line = Fst, blue line Dxy, red line = pi_w, 

#CREMOSA
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=1,
                            WindowCut=c(700000,1100000),dataPlot="FstfromMeanPiAdj_16_19",FstRange = c(0,0.4),lwdFst = 5,DxyPlot="dXYfromMeanFstAdj_16_19",piWPlot= "piBarAdj_16_19",deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data, gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1.0, 0.55, 0.75), clineCexBackground = 1.2, clineCexGene=1.3)

  #DxyPlot ="dXYfromMeanFstAdj_15_20"
  # piWPlot <- "piBarAdj_15_20"

# High Fst region 1
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=1, WindowCut=c(47000000,48000000),dataPlot="FstfromMeanPiAdj_16_19",FstRange = c(0,0.4),lwdFst = 5,DxyPlot="dXYfromMeanFstAdj_16_19",piWPlot= "piBarAdj_16_19",deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1.0, 0.3, 0.5), clineCexBackground = 1.2, clineCexGene=1.3)

# High Fst region 2
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=1, WindowCut=c(50000000,51000000),dataPlot="FstfromMeanPiAdj_16_19",FstRange = c(0,0.4),lwdFst = 5,DxyPlot="dXYfromMeanFstAdj_16_19",piWPlot= "piBarAdj_16_19",deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1.0, 0.05, 0.25), clineCexBackground = 1.2, clineCexGene=1.3)

dev.off()

```

```{r genome scan zooms for LG2}

range(allChr_Windows[allChr_Windows[,"LG"]==2,"windowStart"])
nrow(allChr_Clines_filterDpth15[allChr_Clines_filterDpth15["LG"]==2,])

# All of LG2
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=2,WindowCut=c(0,80000000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.9),lwdFst = 1.2,DxyPlot=NA,piWPlot= NA,deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=F,plotColGeneRegions=T,colourTable = colourChoice, startPlot =T, plotPos1 = c(0.01, 1.0, 0.80, 1), clineCexBackground = 1.2, clineCexGene=1.4)

#AURINA
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=2,
WindowCut=c(700000,1200000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.6),lwdFst = 5,DxyPlot="dXYfromMeanFstAdj_15_20",piWPlot= "piBarAdj_15_20",deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1.0, 0.55, 0.75), clineCexBackground = 1.2, clineCexGene=1.4)

# FLAVIA
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=2,
WindowCut=c(52500000,54500000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.9),lwdFst = 5,DxyPlot="dXYfromMeanFstAdj_15_20",piWPlot= "piBarAdj_15_20",deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1.0, 0.3, 0.5), clineCexBackground = 1.2, clineCexGene=1.4)


# pdf version pool 15 and 20
# PDF version
pdf("FigSx_GenomeScanPanel_Chr2_Pool15_20.pdf",width = 20, height = 27, pointsize = 25)

# All of LG2
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=2,WindowCut=c(0,80000000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.9),lwdFst = 1.2,DxyPlot=NA,piWPlot= NA,deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=F,plotColGeneRegions=T,colourTable = colourChoice, startPlot =T, plotPos1 = c(0.01, 1.0, 0.80, 1), clineCexBackground = 1.2, clineCexGene=1.4)

#AURINA
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=2,
WindowCut=c(700000,1200000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.6),lwdFst = 5,DxyPlot="dXYfromMeanFstAdj_15_20",piWPlot= "piBarAdj_15_20",deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1.0, 0.55, 0.75), clineCexBackground = 1.2, clineCexGene=1.4)

# FLAVIA
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=2,
WindowCut=c(52500000,54500000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.9),lwdFst = 5,DxyPlot="dXYfromMeanFstAdj_15_20",piWPlot= "piBarAdj_15_20",deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1.0, 0.3, 0.5), clineCexBackground = 1.2, clineCexGene=1.4)

dev.off()

# PDF version
pdf("FigSx_GenomeScanPanel_Chr2_Pool16_19.pdf",width = 20, height = 27, pointsize = 25)

# All of LG2
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=2,WindowCut=c(0,80000000),dataPlot="FstfromMeanPiAdj_16_19",FstRange = c(0,0.9),lwdFst = 1.2,DxyPlot=NA,piWPlot= NA,deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=F,plotColGeneRegions=T,colourTable = colourChoice, startPlot =T, plotPos1 = c(0.01, 1.0, 0.80, 1), clineCexBackground = 1.2, clineCexGene=1.4)

#AURINA
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=2,
WindowCut=c(700000,1200000),dataPlot="FstfromMeanPiAdj_16_19",FstRange = c(0,0.6),lwdFst = 5,DxyPlot="dXYfromMeanFstAdj_16_19",piWPlot= "piBarAdj_16_19",deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1.0, 0.55, 0.75), clineCexBackground = 1.2, clineCexGene=1.4)

# FLAVIA
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=2,
WindowCut=c(52500000,54500000),dataPlot="FstfromMeanPiAdj_16_19",FstRange = c(0,0.9),lwdFst = 5,DxyPlot="dXYfromMeanFstAdj_16_19",piWPlot= "piBarAdj_16_19",deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1.0, 0.3, 0.5), clineCexBackground = 1.2, clineCexGene=1.4)
dev.off()

```

```{r genome scan zooms for LG3}

# LG 3

range(allChr_Windows[allChr_Windows[,"LG"]==3,"windowStart"])
nrow(allChr_Clines_filterDpth15[allChr_Clines_filterDpth15["LG"]==3,])
# All of LG3
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=3,WindowCut=c(0,70000000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.4),lwdFst = 1.2,DxyPlot=NA,piWPlot= NA,deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=F,plotColGeneRegions=T,colourTable = colourChoice, startPlot =T, plotPos1 = c(0.01, 1.0, 0.80, 1), clineCexBackground = 1.2, clineCexGene=1.4)

pdf("FigSx_GenomeScanPanel_Chr3_Pool15_20.pdf",width = 20, height = 27, pointsize = 25)
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=3,WindowCut=c(0,70000000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.4),lwdFst = 1.2,DxyPlot=NA,piWPlot= NA,deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=F,plotColGeneRegions=T,colourTable = colourChoice, startPlot =T, plotPos1 = c(0.01, 1.0, 0.80, 1), clineCexBackground = 1.2, clineCexGene=1.4)
dev.off()


```

```{r genome scan zooms for LG4}

# LG 4
colourChoice[colourChoice[,"Region"]=="background",2] <- c((rgb(220,220,220,180,maxColorValue=255)))

range(allChr_Windows[allChr_Windows[,"LG"]==4,"windowStart"])
nrow(allChr_Clines_filterDpth15[allChr_Clines_filterDpth15["LG"]==4,])
# All of LG4
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=4,WindowCut=c(0,60000000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.4),lwdFst = 1.2,DxyPlot=NA,piWPlot= NA,deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=F,plotColGeneRegions=T,colourTable = colourChoice, startPlot =T, plotPos1 = c(0.01, 1.0, 0.80, 1), clineCexBackground = 1.2, clineCexGene=1.4)

# SULF region
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=4,
WindowCut=c(38000000,39000000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.6),lwdFst = 5,DxyPlot="dXYfromMeanFstAdj_15_20",piWPlot= "piBarAdj_15_20",deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1.0, 0.55, 0.75), clineCexBackground = 1.2, clineCexGene=1.4)

# KASP marker positions at SULF
# Chr4:38,316,531
# Chr4:38,397,179
# place cirlce for KASP locations

# pdf version
pdf("FigSx_GenomeScanPanel_Chr4_Pool15_20.pdf",width = 20, height = 27, pointsize = 25)

# All of LG4
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=4,WindowCut=c(0,60000000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.4),lwdFst = 1.2,DxyPlot=NA,piWPlot= NA,deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=F,plotColGeneRegions=T,colourTable = colourChoice, startPlot =T, plotPos1 = c(0.01, 1.0, 0.80, 1), clineCexBackground = 1.2, clineCexGene=1.4)

# SULF region
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=4,
WindowCut=c(38000000,39000000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.6),lwdFst = 5,DxyPlot="dXYfromMeanFstAdj_15_20",piWPlot= "piBarAdj_15_20",deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1.0, 0.55, 0.75), clineCexBackground = 1.2, clineCexGene=1.4)
dev.off()

```

This code chunk below produces fig 6a,b,c

```{r Fig 6 genome scan zooms for LG5}
ClineEst <- c("centre","width")

# RUBIA & clines data
# another cluster of clines on LG5
clineData_filterLG5 <- allChr_Clines_filterDpth15[allChr_Clines_filterDpth15[,"AFD_15_20"]>=0.9 & allChr_Clines_filterDpth15[,"LG"]==5,]
cat("\nclinal loci after delta p filter n=",nrow(clineData_filterLG5))
clineData_filterLG5 <- clineData_filterLG5[clineData_filterLG5[,ClineEst[2]]>0 & clineData_filterLG5[,ClineEst[2]]<12000,]
cat("\nclinal loci after negative cline filter n=",nrow(clineData_filterLG5))

allChr_Clines_Chr5_elsewhere <- clineData_filterLG5 %>% filter(LG==5 & ColourGene==-9)
nrow(allChr_Clines_Chr5_elsewhere)
allChr_Clines_RUBIA <- clineData_filterLG5 %>% filter(ColourGene=="RUBIA" & distanceToGene =="inGene")
nrow(allChr_Clines_RUBIA)
allChr_Clines_RUBIAlinked <- clineData_filterLG5 %>% filter(ColourGene=="RUBIA") %>% filter(distanceToGene %in% linkedAreas)
nrow(allChr_Clines_RUBIAlinked)

# how many clines within 200kb of FLA
sum(clineData_filterLG5[,"position"]>6100000 & clineData_filterLG5[,"position"]< 6400000)

# LG 5
#colourChoice[colourChoice[,"Region"]=="background",2] <- c((rgb(220,220,220,210,maxColorValue=255)))

range(allChr_Windows[allChr_Windows[,"LG"]==5,"windowStart"])

# All of LG 5
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=5,WindowCut=c(0,70950000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.5),lwdFst = 1.2,DxyPlot=NA,piWPlot= NA,deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=F,plotColGeneRegions=T,colourTable = colourChoice, startPlot =T, plotPos1 = c(0.01, 1.0, 0.80, 1), clineCexBackground = 1.2, clineCexGene=1.4)

# 300KB region around RUBIA 
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=5,
WindowCut=c(6100000,6400000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.4),lwdFst = 5,DxyPlot="dXYfromMeanFstAdj_15_20",piWPlot= "piBarAdj_15_20",deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1.0, 0.55, 0.75), clineCexBackground = 1.2, clineCexGene=1.4)

 # 300kb DFGE
genomeScanZoomDFgeneExp(DFgeneExpData = RNAseqData, WindowData=allChr_Windows, thisLG=5, WindowCut=c(6100000,6400000), dataPlot="log2FoldChange",DFgeneExpRange = c(-4,4), gffdata = gff.data,gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1, 0.3, 0.5), cexBackground = 1.2, cexGene=1.4)


# # High Fst region 
# genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=5, WindowCut=c(25000000,30000000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.4),lwdFst = 1.6,deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.04, 0.96, 0.3, 0.5), clineCexBackground = 1.2, clineCexGene=1.4)


# PDF version pools 15 x 20 
pdf("FigSx_GenomeScanPanel_Chr5_Pool15_20.pdf",width = 20, height = 27, pointsize = 25)

# All of LG 5
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=5,WindowCut=c(0,70950000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.5),lwdFst = 1.2,DxyPlot=NA,piWPlot= NA,deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=F,plotColGeneRegions=T,colourTable = colourChoice, startPlot =T, plotPos1 = c(0.01, 1.0, 0.80, 1), clineCexBackground = 1.2, clineCexGene=1.4)

# 300KB region around RUBIA 
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=5,
WindowCut=c(6100000,6400000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.4),lwdFst = 5,DxyPlot="dXYfromMeanFstAdj_15_20",piWPlot= "piBarAdj_15_20",deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1.0, 0.55, 0.75), clineCexBackground = 1.2, clineCexGene=1.4)

 # 300kb DFGE
genomeScanZoomDFgeneExp(DFgeneExpData = RNAseqData, WindowData=allChr_Windows, thisLG=5, WindowCut=c(6100000,6400000), dataPlot="log2FoldChange",DFgeneExpRange = c(-4,4), gffdata = gff.data,gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1, 0.3, 0.5), cexBackground = 1.2, cexGene=1.4)

dev.off()

# PDF version pools 16 x 19 
pdf("FigSx_GenomeScanPanel_Chr5_Pool16_19.pdf",width = 20, height = 27, pointsize = 25)

# All of LG 5
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=5,WindowCut=c(0,70950000),dataPlot="FstfromMeanPiAdj_16_19",FstRange = c(0,0.5),lwdFst = 1.2,DxyPlot=NA,piWPlot= NA,deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=F,plotColGeneRegions=T,colourTable = colourChoice, startPlot =T, plotPos1 = c(0.01, 1.0, 0.80, 1), clineCexBackground = 1.2, clineCexGene=1.4)

# 300KB region around RUBIA 
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=5,
WindowCut=c(6100000,6400000),dataPlot="FstfromMeanPiAdj_16_19",FstRange = c(0,0.4),lwdFst = 5,DxyPlot="dXYfromMeanFstAdj_16_19",piWPlot= "piBarAdj_16_19",deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1.0, 0.55, 0.75), clineCexBackground = 1.2, clineCexGene=1.4)

 # 300kb DFGE
genomeScanZoomDFgeneExp(DFgeneExpData = RNAseqData, WindowData=allChr_Windows, thisLG=5, WindowCut=c(6100000,6400000), dataPlot="log2FoldChange",DFgeneExpRange = c(-4,4), gffdata = gff.data,gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1, 0.3, 0.5), cexBackground = 1.2, cexGene=1.4)

dev.off()

```

```{r Fig 6 genome scan zooms for LG5 extra zoom around FLS and RUB}
ClineEst <- c("centre","width")

# RUBIA & clines data
# another cluster of clines on LG5
clineData_filterLG5 <- allChr_Clines_filterDpth15[allChr_Clines_filterDpth15[,"AFD_15_20"]>=0.9 & allChr_Clines_filterDpth15[,"LG"]==5,]
cat("\nclinal loci after delta p filter n=",nrow(clineData_filterLG5))
clineData_filterLG5 <- clineData_filterLG5[clineData_filterLG5[,ClineEst[2]]>0 & clineData_filterLG5[,ClineEst[2]]<12000,]
cat("\nclinal loci after negative cline filter n=",nrow(clineData_filterLG5))

allChr_Clines_Chr5_elsewhere <- clineData_filterLG5 %>% filter(LG==5 & ColourGene==-9)
nrow(allChr_Clines_Chr5_elsewhere)
allChr_Clines_RUBIA <- clineData_filterLG5 %>% filter(ColourGene=="RUBIA" & distanceToGene =="inGene")
nrow(allChr_Clines_RUBIA)
allChr_Clines_RUBIAlinked <- clineData_filterLG5 %>% filter(ColourGene=="RUBIA") %>% filter(distanceToGene %in% linkedAreas)
nrow(allChr_Clines_RUBIAlinked)

# LG 5
#colourChoice[colourChoice[,"Region"]=="background",2] <- c((rgb(220,220,220,210,maxColorValue=255)))

range(allChr_Windows[allChr_Windows[,"LG"]==5,"windowStart"])

# All of LG 5
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=5,WindowCut=c(0,70950000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.5),lwdFst = 1.2,DxyPlot=NA,piWPlot= NA,deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=F,plotColGeneRegions=T,colourTable = colourChoice, startPlot =T, plotPos1 = c(0.01, 1.0, 0.80, 1), clineCexBackground = 1.2, clineCexGene=1.4,KASPpos=c(6286923,7600))

# 300KB region around RUBIA 
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=5,
WindowCut=c(6270000,6290000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.4),lwdFst = 5,DxyPlot="dXYfromMeanFstAdj_15_20",piWPlot= "piBarAdj_15_20",deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1.0, 0.55, 0.75), clineCexBackground = 1.8, clineCexGene=2,KASPpos=c(6286923,7600))

 # 300kb DFGE
genomeScanZoomDFgeneExp(DFgeneExpData = RNAseqData, WindowData=allChr_Windows, thisLG=5, WindowCut=c(6260000,6300000), dataPlot="log2FoldChange",DFgeneExpRange = c(-4,4), gffdata = gff.data,gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1, 0.3, 0.5), cexBackground = 1.2, cexGene=1.4)


# # High Fst region 
# genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=5, WindowCut=c(25000000,30000000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.4),lwdFst = 1.6,deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.04, 0.96, 0.3, 0.5), clineCexBackground = 1.2, clineCexGene=1.4)


# PDF version pools 15 x 20 
pdf("FigSx_GenomeScanPanel_Chr5_Pool15_20_extraZoom.pdf",width = 13, height = 27, pointsize = 25)

# All of LG 5
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=5,WindowCut=c(0,70950000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.5),lwdFst = 1.2,DxyPlot=NA,piWPlot= NA,deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=F,plotColGeneRegions=T,colourTable = colourChoice, startPlot =T, plotPos1 = c(0.01, 1.0, 0.80, 1), clineCexBackground = 1.2, clineCexGene=1.4,KASPpos=c(6286923,7600))

# 300KB region around RUBIA 
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=5,
WindowCut=c(6265000,6290000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.4),lwdFst = 0,DxyPlot="dXYfromMeanFstAdj_15_20",piWPlot= "piBarAdj_15_20",deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1.0, 0.55, 0.75), clineCexBackground = 2.2, clineCexGene=2.4,KASPpos=c(6286923,7600))

 # 300kb DFGE
genomeScanZoomDFgeneExp(DFgeneExpData = RNAseqData, WindowData=allChr_Windows, thisLG=5, WindowCut=c(6265000,6300000), dataPlot="log2FoldChange",DFgeneExpRange = c(-4,4), gffdata = gff.data,gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1, 0.3, 0.5), cexBackground = 1.8, cexGene=2)

dev.off()

# PDF version pools 16 x 19 
pdf("FigSx_GenomeScanPanel_Chr5_Pool16_19_extraZoom.pdf",width = 20, height = 27, pointsize = 25)

# All of LG 5
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=5,WindowCut=c(0,70950000),dataPlot="FstfromMeanPiAdj_16_19",FstRange = c(0,0.5),lwdFst = 1.2,DxyPlot=NA,piWPlot= NA,deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=F,plotColGeneRegions=T,colourTable = colourChoice, startPlot =T, plotPos1 = c(0.01, 1.0, 0.80, 1), clineCexBackground = 1.2, clineCexGene=1.4)

# 300KB region around RUBIA 
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=5,
WindowCut=c(6260000,6300000),dataPlot="FstfromMeanPiAdj_16_19",FstRange = c(0,0.4),lwdFst = 5,DxyPlot="dXYfromMeanFstAdj_16_19",piWPlot= "piBarAdj_16_19",deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1.0, 0.55, 0.75), clineCexBackground = 1.2, clineCexGene=1.4)

 # 300kb DFGE
genomeScanZoomDFgeneExp(DFgeneExpData = RNAseqData, WindowData=allChr_Windows, thisLG=5, WindowCut=c(6260000,6300000), dataPlot="log2FoldChange",DFgeneExpRange = c(-4,4), gffdata = gff.data,gffPlot=T,plotColGeneRegions=T,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1, 0.3, 0.5), cexBackground = 1.2, cexGene=1.4)

dev.off()

```

```{r genome scan zooms for LG6}
ClineEst <- c("centre","width")
# Rubia in sliding windows
allChr_Fst_ROS[,3145:3153]
allChr_Fst_ROSelLinked[,3145:3153]

# LG 6
colourChoice[colourChoice[,"Region"]=="background",2] <- c((rgb(220,220,220,200,maxColorValue=255)))

range(allChr_Windows[allChr_Windows[,"LG"]==6,"windowStart"])
allChr_Clines_filterDpth15[allChr_Clines_filterDpth15[,"LG"]==6,"position"]

genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=6,WindowCut=c(0,60000000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.4),deltaP=0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,colourTable = colourChoice, startPlot =T, plotPos1 = c(0.01, 1.0, 0.80, 1), clineCexBackground = 1.2, clineCexGene=1.4)

# 300KB region around ROS1 52,889,323-52,889,737 , ELUTA 53,060,761-53,062,505
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15, WindowData=allChr_Windows,thisLG=6,
  WindowCut=c(52000000,54000000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.4),deltaP=0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1.0, 0.55, 0.75), clineCexBackground = 1.2, clineCexGene=1.4)

# High Fst region - all clines here only at deltaP > 0.8
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=6, WindowCut=c(47000000,50000000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.4),deltaP=0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,colourTable = colourChoice, startPlot=F, plotPos1 = c(0.01, 1.0, 0.3, 0.5), clineCexBackground = 1.2, clineCexGene=1.4)

```

```{r genome scan zooms for LG7}

# LG 7
colourChoice[colourChoice[,"Region"]=="background",2] <- c((rgb(220,220,220,180,maxColorValue=255)))

range(allChr_Windows[allChr_Windows[,"LG"]==7,"windowStart"])
nrow(allChr_Clines_filterDpth15[allChr_Clines_filterDpth15["LG"]==7,])
# All of LG7
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=7,WindowCut=c(0,60000000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.4),deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,colourTable = colourChoice, startPlot =T, plotPos1 = c(0.01, 1.0, 0.80, 1), clineCexBackground = 1.2, clineCexGene=1.4)

```

```{r genome scan zooms for LG8}

# LG 8
colourChoice[colourChoice[,"Region"]=="background",2] <- c((rgb(220,220,220,180,maxColorValue=255)))

range(allChr_Windows[allChr_Windows[,"LG"]==8,"windowStart"])
nrow(allChr_Clines_filterDpth15[allChr_Clines_filterDpth15["LG"]==8,])
# All of LG8
genomeScanZoomFstClineWidth(clineData=allChr_Clines_filterDpth15,WindowData=allChr_Windows,thisLG=8,WindowCut=c(0,60000000),dataPlot="FstfromMeanPiAdj_15_20",FstRange = c(0,0.4),deltaP =0.9, pool_deltaP ="AFD_15_20", ClineEst=c("centre","width"), filterNegativeClines=T, gffdata = gff.data,colourTable = colourChoice, startPlot =T, plotPos1 = c(0.01, 1.0, 0.80, 1), clineCexBackground = 1.2, clineCexGene=1.4)

```

# Genotype - Phenotype for RUBIA

Previous studies on **Antirrhinum majus** have identified several major effect genes directly responsible for phenotypic variation in flower colour differences between these phenotypes (ROSEA; Schwinn et al , Am4CTG; Ono et al 2008, SULF; Bradley et al., 2017 ELUTA; Tavares et al 2018). To confirm that the SNPs we selected within each of these known major effect genes were associated with phenotypic variation at the Planoles hybrid zone, we used the KASP SNP genotypes to retest the association between each SNP and components of flower colour variation. Although the aim here is not to conduct Genome Wide Association Scans (GWAS) to identify new causal loci, we also test for associations at other SNP loci showing steep geographic clines in allele frequencies.  

### METHODS  
*Flower colour phenotypes*  

We used a random selection of individuals from the Planoles hybrid zone pedigree which had good quality photographs of the flowers. Following Whibley et al., (2006), individuals were first categorized into six phenotype/genotype classes on the basis of the intensity and patterning of anthocyanin and aurone pigmentation across the flower (see Supplementary Sx). This classification system is based on the inferred underlying ROS/EL and SULF genotypes (Figure xx). We then stratified the random samples from the six major colour phenotypes (Magenta, Yellow, Pink, Weak Orange, Full Orange and White), to ensure similar numbers of the main phenotype classes were available for genotype-phenotype associations. All photographs were taken on a standard background with the same light source. Individuals randomly selected were first checked for replicate multilocus genotypes (see above). This resulted in a total of n = xx individuals for phenotyping.

Flower color measurements were taken in ImageJ (http://imagej.nih.gov/ij/). Images were white balanced using the macro `Chart_White_Balance` (http://imagejdocu.tudor.lu/doku.php?id=plugin:color:chart_white_balance:start) which operated consistently across flowers with the color chart contained in the flower photos. To properly white balance the photos, a line was drawn from the whitest part of the white swath from the upper color chart to the darkest part of the dark swath of the upper color chart. The macro then operated and white-balanced the photo. Measurements were taken from each of six flower measurement location sites. These measurements were taken using a standard circle with an area of 5480 pixels. The circle was placed in six standard locations (Figure Sx) using the plugin “RGB Measure” to obtain individual mean and standard deviation for the Red, Green, and Blue channels. 

Hue was calculated using these three colour channels. We first convert all RGB measures to 0-1 scale, then calculate Hue depending on the maximum channel as (i) Red Max: $Hue =[(G-B)/(max-min)]*60$, (ii) Green Max: $Hue= [2+(B-R)/(max-min)]*60$, (iii) Blue Max: $Hue= [4+[R-G]/(max-min)]*60$. 

As a measure of colourfullness of each region in proportion to its brightness, we calculated Saturation (Sat) as $Sat = [(max-min)/255]/[1-(2L-1)]$, where $L= [0.5(max-min)/255]$ and $max$ and $min$ refer to the maximum and minimum value amongst Red, Green and Blue channels (rescaled between 0 and 1).

We also calculated the standard deviation of the Grayscale (sd_GS), using the average of standard deviation values of Red, Green and Blue channels simply as $GS_sd = (sd_R+sd_G+sd_B)/3$.

Lastly, we calculated intensity density (Int) averaged over the three channels as $Int = (R+G+B)/3$.

![An example of the six regions of the front view of an Antirrhinum flower used to quantify patterns of floral pigmentation. The circle 2 is indicated as a dashed line because these measurements were not used in the analyses. This was due to some flowers having ventral petals that drooped over the 2nd measurement, making the spacing inaccurate and not comparable between flowers.](/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/Pedigree_DF/RawDataFiles/flowers/SixFlowerRegions.png)

We assessed these 4 traits representing colour and pattern (Hue, GS_sd, Int, Sat) in each of the six floral regions individually and in a multivariate framework using Principal Component Analysis (PCA) as implemented in R. Individual traits were transformed to ensure normality using normal quantile transformation. For the latter, we built two PCA models (i) with intensity (Int) included, (ii) without intensity. Excluding intensity measure were implemented to control for environmental variation, given that fine scale environmental conditions (e.g. shade) are known to influence the intensity of anthocyanin (magenta) expression in both flowers and leaves in **Antirrhinum**.

```{r Define-folder-locations-input-output, include=F, echo=F}

#folderLoc <- "/Users/dlfield/Seafile/Pedigree_DF/"
folderLoc <- "/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/Pedigree_DF/"

# contains the folder structure for input and output
folderStructure <- read.table(paste0(folderLoc,"/Scripts/R/","folderStructure_MQ.txt"),header=TRUE,fill=FALSE,na.strings = "?")
#setwd(folderLoc)
#folderStructure <- read.table("~/Dropbox/work/PostdocIST/projects_Rmd/SNPs/HZanalyses/folderStructure.txt",header=TRUE,fill=FALSE,na.strings = "?")

```
```{r Load-packages, include=F,echo=F}

library("dplyr")
library("grid")
library("cowplot")
library("Rmpfr")
library("nplr")
library("optimx")
library("maxLik")
library("data.table")
library("scatterplot3d")
#library("MHadaptive")
library("MCMCpack")
library("mcmc")
library("Hmisc")
library("ggplot2")
library("mapplots")
require(data.table)
library("RgoogleMaps")
library("psych")
library("calibrate")
library(RColorBrewer)
library("plotrix")
library("RANN")
library("stringdist")
library("pdist")
library("rmarkdown")
library(compiler)
library("plot3D")
library("png")
library("glmm")
library("gplots")
library("vcd")
library("LDcorSV")
library(viridis)
library(corpcor)
library(plotly)
library(terra)
library("raster")
#library("rasterVis")
library("rgdal")
library(sp)
library(maptools)
library(rgeos)
library(ggmap)
library(mapproj)
library(maps)
library(mapdata)
library("lubridate")
#library(R2G2)
library(reticulate)
library(kableExtra)
#library("LDheatmap")
#library(sequoia)
library(boot) # Bootstrap functions
library(car) # Companion to applied regression
library(dplyr) # Easy manipulation of dataset
library(ggplot2) # Advance graphic functions
library(nlme) # Linear mixed effect models
library(pastecs) # Some basic statistics functions
#library(plotly) # 3D plots
library(reshape2) # Re - structure the dataset
library(kinship2) # pedigree plots
#library(optiSel) # pedigree info
library(nadiv)
library(pedigree)
#library(asremlPlus)
library(pastecs)
library(igraph)
library(tnet)
#library(pedantics)
library(ggplot2)
library(MCMCglmm)
library(scales)
library(dplyr)
library(tidyr)
library(gridBase)
library(gridExtra)
library(grid)
library(lattice)
library(tinytex)

library(lattice)
library(DT)
library(grid)
library(png)
library(jpeg)
library(reticulate)
library(knitr)
library(imager)
library(leaflet)
library(tidyverse)
library(htmlwidgets)
library(tidygate)
library("remotes")
library("devtools")
library(cowplot)

library(magick)
library(rdrop2)
library(kableExtra)
library(broman)
library(broom)

library(colorspace)  # For HSV to HEX conversion
library(patchwork)
library(kableExtra)
library(gt)


```
```{r Source-my-functions, include=F,echo=F}
# load all of my functions
sourceDir <- function(path, trace = TRUE, ...) {
  # path <- folder_myFunctions
    for (nm in list.files(path, pattern = "[.][RrSsQq]$")) {
       if(trace) cat(nm,":")
       source(file.path(path, nm), ...)
       if(trace) cat("\n")
    }
 }
sourceDir(as.vector(folderStructure[folderStructure[,"FolderName"]=="folder_myFunctions","Location"]))
# file.sources = list.files("/Users/dfield/Documents/DavidWorkLaptop/PostdocIST/Projects/setupData_SNP_ecol/R/myFunctions", pattern="*.R$", full.names=TRUE,ignore.case=TRUE)
# sapply(file.sources,source,.GlobalEnv)
```
```{r import Taylors phenotype dataset}
# Planoles
flower <- read.csv("/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/poolSeq/version3.5/flowerPhenotypes/2014measurements.csv", header=T)

```
```{r import genotype data nucleotide coding as GenoEcol_nuc}
GenoEcol_nuc <- read.csv("/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/poolSeq/version3.5/KASP/FinalGenoEcol_short_nuc.csv",header=TRUE,fill=FALSE,na.strings = "?",stringsAsFactors = F)

```
```{r import other base data}
#SNPlociListUpdated <- readMyFile(folderStructure,"folder_base_genetic","shortSummarySNPsAdj.csv")
SNPlociListUpdated <- read.csv("/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/poolSeq/version3.5/KASP/Amajus_shortSummarySNPsAdj.csv",header=TRUE,fill=FALSE,na.strings = "?",stringsAsFactors = F)


# colnames(SNPlociListUpdated)
#SNPlociListUpdated <- read.csv(paste0(folderLoc,"/RawDataFiles/genetic/shortSummarySNPsAdj.csv"),header=TRUE,fill=FALSE,na.strings = "?",stringsAsFactors = F)

Antspec_compl <- readMyFile(folderStructure,"folder_processing_phase1QualScore","Antspec_compl.csv") 
summaryTable_antspec <- with(Antspec_compl[Antspec_compl[,"ThisYearLabel"]==1,], table(year))
# duplicate data
matchingMultiLoc <- readMyFile(folderStructure,"folder_processing_duplicates","matching_MultiLoc.csv")
locusErrors <- readMyFile(folderStructure,"folder_processing_duplicates","locusErrors.csv")
```

```{r calculate numbers not genotyped missing propMissing totalGenotyped for GenoEcol_nuc,echo=F}
#colnames(GenoEcol)
sumTable <- as.data.frame(table(GenoEcol_nuc[,"Population"]))

colnames(sumTable) <- c("Population","Sample Size")
phenoCatMain <- c("Y","FR","WO","FO","W","WR")
GenoEcol_nuc[,"phenoCat_final"] <- as.vector(GenoEcol_nuc[,"phenoCat_final"])

summaryTable <- with(GenoEcol_nuc[GenoEcol_nuc[,"phenoCat_final"]%in%phenoCatMain,], table(Population, phenoCat_final))
summaryTable <- summaryTable[,c(2,1,5,4,3,6)]
#kable(summaryTable,caption = "Number of samples genotyped across all populations and their phenotypes") %>%
#  kable_styling(bootstrap_options = "striped", full_width = F)

# how many SNP loci?
lociColumns <- colnames(GenoEcol_nuc)[which(colnames(GenoEcol_nuc)=="s1187_290152"):which(colnames(GenoEcol_nuc)=="s735_701957")]
numLoci <- length(lociColumns)
lociColumns <- which(colnames(GenoEcol_nuc)=="s1187_290152"):which(colnames(GenoEcol_nuc)=="s735_701957")
getMissing <- function(thisRow) {return(sum(thisRow==-9 | thisRow=="Bad"))}
getNGY <- function(thisRow) {return(sum(thisRow=="NGY"))}
getTotalGenotyped <- function(thisRow) {return(as.numeric(sum(thisRow=="NGY"| thisRow=="Bad" | thisRow==-9)))}
getTotalGenotypesAvailable <- function(thisRow) {return(sum(thisRow>=0))}

# turn -8 to -9
GenoEcol_nuc_cut <- GenoEcol_nuc[,lociColumns]
GenoEcol_nuc_cut[apply(GenoEcol_nuc_cut,2,function (x) x==-8)] <- -9
GenoEcol_nuc[,lociColumns] <- GenoEcol_nuc_cut

GenoEcol_nuc_new <- cbind(GenoEcol_nuc[,1:(lociColumns[1]-1)],GenoEcol_nuc_cut,GenoEcol_nuc[,c(which(colnames(GenoEcol_nuc)=="DataSet"):which(colnames(GenoEcol_nuc)=="PhenoComment"))])
missing <- apply(GenoEcol_nuc_new[,lociColumns],1, FUN = getMissing)
NGY <- apply(GenoEcol_nuc_new[,lociColumns],1, FUN = getNGY)
totals <- numLoci-(apply(GenoEcol_nuc_new[,lociColumns],1, FUN = getTotalGenotyped))
totals_available <- apply(GenoEcol_nuc_new[,lociColumns],1, FUN = getTotalGenotypesAvailable)
propMissing <- as.numeric(missing)/as.numeric(totals_available)
GenoEcol_nuc_new <- cbind(GenoEcol_nuc_new,missing,NGY,totals,propMissing)
colnames(GenoEcol_nuc_new)
lociColumns <- which(colnames(GenoEcol_nuc_new)=="s1187_290152"):which(colnames(GenoEcol_nuc_new)=="s735_701957")


```
```{r filtering GenoEcol_nuc to those with GPS and within Planoles valley}

# NEW 2023 leaving all genotype data (dont extract just genotype)
GenoEcol_nuc_new <- GenoEcol_nuc
# Remove an individual with missing location
GenoEcol_nuc_new <- GenoEcol_nuc_new %>% filter(Easting!=-9)

# Sorting individuals East-West
GenoEcol_nuc_new[,"Easting"] <- as.numeric(as.vector(GenoEcol_nuc_new[,"Easting"]))
GenoEcol_nuc_new <- arrange(GenoEcol_nuc_new, Easting)

# plot the whole region first
#par(mfrow=c(1,1))
#plotHZpedigree(GenoEcol_nuc_new,408000,439000,4684000,4692000,gridAdd=F,adjustMin=F,TitleStart="Planoles region all years")

# Pick out Planoles using Easting and Northing
GenoEcol_nuc_new <- GenoEcol_nuc_new %>% 
  filter(Easting>410000 & Easting<435000 & Northing>4684000 & Northing<4692000)

# Yellow side should be dominated by 0 or 1s (more hets on yellow flank)
# head(GenoEcol_Planoles[,"ros_assembly_543443"],40)
# Magenta side should be dominated by 2's
# tail(GenoEcol_Planoles[,"ros_assembly_543443"],40)

```
```{r filtering GenoEcol_nuc propMissingData}
#
#- Flavia: s316_93292, s316_257789 (these are the SNPs either side gene on LG 2 - 43.175cM)
#- Sulf: s91_39699, s91_78256 (genotyped the most), s91_78805 (s91_39699 left and s91_78256 right of Sulf LG 4 - 9.3cM)
#- ROS1: ros_assembly_541834 (drop),ros_assembly_542000 (drop), ros_assembly_543443 (best marker), ros_assembly_544601 (just outside)
#- ROS2: ros_assembly_567004
#- ROS3: ros_assembly_575837,ros_assembly_576271
#- Eluta: ros_assembly_715001 (most genotypes), ros_assembly_715015
# loci for hybrid index, note only keep one marker in close proximity because strong LD means they are not independent
theseDiagStrict <- c("s1187_290152","s829_8371","s316_93292","s316_257789","s91_39699","s91_78256","s91_78805","ros_assembly_543443","ros_assembly_715001")
               
# Full loci
#colnames(GenoEcol)
# useful objects in memory
# lociNames; lociColumns; GenoEcol[,"missing"]; GenoEcol[,"totals"]; GenoEcol[,"propMissing"]
# remove Hugos SNPs & the two dodgey SNPs Annabele flagged
badSNPs1 <- c("ros_assembly_542000","ros_assembly_541834_ROS1","ros_assembly_717957","s164_366822","s277_468670","ros_assembly_246866","ros_assembly_717957")
# second list of bad SNPs identified from latest reference genome
badSNPs2 <- as.vector(SNPlociListUpdated[SNPlociListUpdated[,"GoodMarker"]==0,"LocusName"])
# make an exception for the SULF marker for now
badSNPs2 <- badSNPs2[-(which(badSNPs2=="s91_78256"))]
# Good SNPs
GoodSNPs <- SNPlociListUpdated[,"LocusName"][(SNPlociListUpdated[,"LocusName"]%in%badSNPs2)==F]

# remove individuals with too much data missing
propRemoved <- sum((GenoEcol_nuc_new[,"propMissing"]<0.05)==F)/nrow(GenoEcol_nuc_new) # loose 31 inds
GenoEcol_nuc_new <- GenoEcol_nuc_new[GenoEcol_nuc_new[,"propMissing"]<0.05,]
GenoEcol_nuc_new_bad <- GenoEcol_nuc_new[GenoEcol_nuc_new[,"propMissing"]>=0.05,]

# remove bad loci
# badSNPs1 : these are a custom set of loci to remove based on suggestions from JIC or other information
# badSNPs2 : these are the set of loci with various problems identified in the excel spreadsheet shortSummarySNPsAdj.csv
GenoEcol_nuc_new_goodCols <- which(colnames(GenoEcol_nuc_new)%in%badSNPs1==F & colnames(GenoEcol_nuc_new)%in%badSNPs2==F)
GenoEcol_nuc_new <- GenoEcol_nuc_new[,GenoEcol_nuc_new_goodCols]

# re-establish which columns have locus data

# Note this is when the loci number is dropped from the 122 - 141 seen in 2009-2014 to around 106.
# this comes from a combination of loci flagged as bad, others flagged as trouble based on new reference genome and high error rates inferred from SNPPIT
lociColumns <- which(colnames(GenoEcol_nuc_new)=="s1187_290152"):which(colnames(GenoEcol_nuc_new)=="s735_701957")

# how many SNP loci?
lociNames <- colnames(GenoEcol_nuc_new)[which(colnames(GenoEcol_nuc_new)=="s1187_290152"):which(colnames(GenoEcol_nuc_new)=="s735_701957")]
numLoci <- length(lociColumns)
lociColumns <- which(colnames(GenoEcol_nuc_new)=="s1187_290152"):which(colnames(GenoEcol_nuc_new)=="s735_701957")

missing <- apply(GenoEcol_nuc_new[,lociColumns],1, FUN = getMissing)
NGY <- apply(GenoEcol_nuc_new[,lociColumns],1, FUN = getNGY)
totals <- numLoci-(apply(GenoEcol_nuc_new[,lociColumns],1, FUN = getTotalGenotyped))
totals_available <- apply(GenoEcol_nuc_new[,lociColumns],1, FUN = getTotalGenotypesAvailable)
propMissing <- as.numeric(missing)/as.numeric(totals_available)
# GenoEcol_nuc_new[,"missing"] <- missing
# GenoEcol_nuc_new[,"NGY"] <- NGY
# GenoEcol_nuc_new[,"totals"] <- totals
# GenoEcol_nuc_new[,"propMissing"] <- propMissing
# GenoEcol_nuc <- GenoEcol_nuc_new
rametIDs_list <- strsplit(as.character(GenoEcol_nuc[,"RametIDs"]),split=",")

```

*Flower colour phenotypes*

In addition to RGB colour space, we convert these values to HSV. 

- Hue remains as calculated above
- Saturation (Sat) converted to HSV Saturation S2:


```{r prepare phenotype data WITH intensity and run PCA}

# final values used here:

# normal quantile transformation (NQT) of mean of the sd of grayscale (SD)
# NQT_SD_1
# NQT_SD_3
# NQT_SD_4
# NQT_SD_5
# NQT_SD_6
# normal quantile transformation (NQT) of mean Hue
# NQT_Hue_1
# NQT_Hue_3
# NQT_Hue_4
# NQT_Hue_5
# NQT_Hue_6
# normal quantile transformation (NQT) of mean Intensity Density 
# NQT_Int_1
# NQT_Int_3
# NQT_Int_4
# NQT_Int_5
# NQT_Int_6
data.flower <- (flower[, c(235,237:240,241,243:246,247,249:252)])

# except remove intensity? Do analysis with and without
theseColsToUse <- colnames(data.flower)
theseColsToUse <- theseColsToUse[c(1:5,11:15)]

flower.phenotype <- flower[, 2]

# apply PCA - scale. = TRUE is highly advisable, but default is FALSE. 
flower.pca <- prcomp(data.flower,center = TRUE, scale. = TRUE) 
# pca2 contains just the first few principle components (ommitting those with low standard deviations)
flower.pca2 <- prcomp(data.flower, center = TRUE, scale. = TRUE, tol=.5)

summaryPCA <- summary(flower.pca)
# find loadings
# flower.pca$rotation

# relevant incantation
aload <- abs(flower.pca$rotation)
# sweep(aload, 2, colSums(aload), "/")
data.pca <- flower.pca$x

```
```{r integrate PCA scores WITH intensity with genotype dataset}
# need to find them
pca_IDs <- as.vector(flower[,"id1"])
#theColsTraits <- which(colnames(flower)=="Hue_1"):which(colnames(flower)=="NQT_Sat_6")
theColsTraits <- which(colnames(flower)=="Measurement_size.pixels."):which(colnames(flower)=="NQT_Sat_6")

getPos <- function(x) {return(unlist(thisPlant%in%x))}
PCAscores <- matrix(-1000,nrow(GenoEcol_nuc),5+length(theColsTraits))
colnames(PCAscores) <- c(colnames(flower)[theColsTraits],"TaylorSet","PC1","PC2","PC3","PC4")
for (thisRow in 1:length(pca_IDs)) {
  # thisRow <- 1
  thisPlant <- pca_IDs[thisRow]
  thisPos <- which(unlist(lapply(rametIDs_list,getPos))==T)
  PCAscores[thisPos,1:ncol(PCAscores)] <- c(as.numeric(flower[thisRow,theColsTraits]),1,as.numeric(data.pca[thisRow,1:4]))
}
GenoEcol_nuc_PCA <- cbind(GenoEcol_nuc,PCAscores)
GenoEcol_nuc_PCA <- GenoEcol_nuc_PCA[GenoEcol_nuc_PCA[,"PC1"]!=-1000,]

# GenoEcol_nuc_PCA$phenoCat
#range(flower.pca$x[,1])
#range(flower.pca$x[,2])
colOrder <- c("Y", "FR", "FO", "WR", "W", "WO")
GenoEcol_nuc_PCA <- GenoEcol_nuc_PCA[GenoEcol_nuc_PCA$phenoCat%in%colOrder,]

# remove dodgey FR in yellow space
GenoEcol_nuc_PCA <- GenoEcol_nuc_PCA[-which(GenoEcol_nuc_PCA$phenoCat=="FR" & GenoEcol_nuc_PCA$PC1>5),]

write.csv(GenoEcol_nuc_PCA,'GenoEcol_nuc_FlowerMeasures_PCA.csv',row.names=F)
```

```{r integrate flower data scores WITH intensity with genotype dataset}
# # need to find them
# pca_IDs <- as.vector(flower[,"id1"])
# #theColsTraits <- which(colnames(flower)=="Hue_1"):which(colnames(flower)=="NQT_Sat_6")
# theColsTraits <- which(colnames(flower)=="Measurement_size.pixels."):which(colnames(flower)=="V_6")
# 
# getPos <- function(x) {return(unlist(thisPlant%in%x))}
# PCAscores <- matrix(-1000,nrow(GenoEcol_nuc),5+length(theColsTraits))
# colnames(PCAscores) <- c(colnames(flower)[theColsTraits],"TaylorSet","PC1","PC2","PC3","PC4")
# for (thisRow in 1:length(pca_IDs)) {
#   # thisRow <- 1
#   thisPlant <- pca_IDs[thisRow]
#   thisPos <- which(unlist(lapply(rametIDs_list,getPos))==T)
#   PCAscores[thisPos,1:ncol(PCAscores)] <- c(as.numeric(flower[thisRow,theColsTraits]),1,as.numeric(data.pca[thisRow,1:4]))
# }
# 
# 
# GenoEcol_nuc_PCA <- cbind(GenoEcol_nuc,PCAscores)
# GenoEcol_nuc_PCA <- GenoEcol_nuc_PCA[GenoEcol_nuc_PCA[,"PC1"]!=-1000,]
# 
# # GenoEcol_nuc_PCA$phenoCat
# #range(flower.pca$x[,1])
# #range(flower.pca$x[,2])
# colOrder <- c("Y", "FR", "FO", "WR", "W", "WO")
# GenoEcol_nuc_PCA <- GenoEcol_nuc_PCA[GenoEcol_nuc_PCA$phenoCat%in%colOrder,]
# 
# # remove dodgey FR in yellow space
# GenoEcol_nuc_PCA <- GenoEcol_nuc_PCA[-which(GenoEcol_nuc_PCA$phenoCat=="FR" & GenoEcol_nuc_PCA$PC1>5),]
# 
# write.csv(GenoEcol_nuc_PCA,'GenoEcol_nuc_FlowerMeasures_PCA.csv',row.names=F)
```

```{r function for RGB to HSV}
# Function to convert RGB to HSV
rgb_to_hsv <- function(R, G, B) {
  # Normalize RGB values to [0,1]
  Rn <- R / 255
  Gn <- G / 255
  Bn <- B / 255
  
  # Compute max, min, and chroma
  max_val <- max(Rn, Gn, Bn)
  min_val <- min(Rn, Gn, Bn)
  chroma <- max_val - min_val
  
  # Compute Hue (Same as your formula)
  if (chroma == 0) {
    H <- 0  # Undefined Hue for grayscale colors
  } else if (max_val == Rn) {
    H <- ( (Gn - Bn) / chroma ) * 60
  } else if (max_val == Gn) {
    H <- (2 + (Bn - Rn) / chroma) * 60
  } else { # max_val == Bn
    H <- (4 + (Rn - Gn) / chroma) * 60
  }
  
  # Adjust Hue to ensure it's in the range [0,360]
  if (H < 0) {
    H <- H + 360
  }
  
  # Compute Saturation in HSV
  if (max_val == 0) {
    S <- 0  # Avoid division by zero
  } else {
    S <- chroma / max_val
  }
  
  # Compute Value (Brightness)
  V <- max_val
  
  # Return HSV values
  return(c(H = H, S = S, V = V))
}

```
```{r convert to HSV space}
#colnames(flower)

colsToGrab1 <- c("Mean.1R", "Mean.1G", "Mean.1B")
colsToGrab2 <- c("Mean.2R", "Mean.2G", "Mean.2B")
colsToGrab3 <- c("Mean.3R", "Mean.3G", "Mean.3B")
colsToGrab4 <- c("Mean.4R", "Mean.4G", "Mean.4B")
colsToGrab5 <- c("Mean.5R", "Mean.5G", "Mean.5B")
colsToGrab6 <- c("Mean.6R", "Mean.6G", "Mean.6B")

data.flower1 <- (GenoEcol_nuc_PCA[, colnames(GenoEcol_nuc_PCA)%in%colsToGrab1])
data.flower2 <- (GenoEcol_nuc_PCA[, colnames(GenoEcol_nuc_PCA)%in%colsToGrab2])
data.flower3 <- (GenoEcol_nuc_PCA[, colnames(GenoEcol_nuc_PCA)%in%colsToGrab3])
data.flower4 <- (GenoEcol_nuc_PCA[, colnames(GenoEcol_nuc_PCA)%in%colsToGrab4])
data.flower5 <- (GenoEcol_nuc_PCA[, colnames(GenoEcol_nuc_PCA)%in%colsToGrab5])
data.flower6 <- (GenoEcol_nuc_PCA[, colnames(GenoEcol_nuc_PCA)%in%colsToGrab6])


hsv_matrix1 <- t(apply(data.flower1, 1, function(row) rgb_to_hsv(row["Mean.1R"], row["Mean.1G"], row["Mean.1B"])))
hsv_matrix2 <- t(apply(data.flower2, 1, function(row) rgb_to_hsv(row["Mean.2R"], row["Mean.2G"], row["Mean.2B"])))
hsv_matrix3 <- t(apply(data.flower3, 1, function(row) rgb_to_hsv(row["Mean.3R"], row["Mean.3G"], row["Mean.3B"])))
hsv_matrix4 <- t(apply(data.flower4, 1, function(row) rgb_to_hsv(row["Mean.4R"], row["Mean.4G"], row["Mean.4B"])))
hsv_matrix5 <- t(apply(data.flower5, 1, function(row) rgb_to_hsv(row["Mean.5R"], row["Mean.5G"], row["Mean.5B"])))
hsv_matrix6 <- t(apply(data.flower6, 1, function(row) rgb_to_hsv(row["Mean.6R"], row["Mean.6G"], row["Mean.6B"])))
colnames(hsv_matrix1) <- c("H_1","S_1","V_1")
colnames(hsv_matrix2) <- c("H_2","S_2","V_2")
colnames(hsv_matrix3) <- c("H_3","S_3","V_3")
colnames(hsv_matrix4) <- c("H_4","S_4","V_4")
colnames(hsv_matrix5) <- c("H_5","S_5","V_5")
colnames(hsv_matrix6) <- c("H_6","S_6","V_6")
GenoEcol_nuc_PCA <- cbind(GenoEcol_nuc_PCA,hsv_matrix1,hsv_matrix2,hsv_matrix3,hsv_matrix4,hsv_matrix5,hsv_matrix6)


```


```{r PhenoCat and ROS box plots}

GenoEcol_nuc_PCA[,"s261_720757"] <- as.factor(GenoEcol_nuc_PCA[,"s261_720757"])
GenoEcol_nuc_PCA[,"ros_assembly_543443"] <- as.factor(GenoEcol_nuc_PCA[,"ros_assembly_543443"])
GenoEcol_nuc_PCA[,"phenoCat_final"] <- as.factor(GenoEcol_nuc_PCA[,"phenoCat_final"])

phenoCat_ROS_Hue6 <- ggplot(GenoEcol_nuc_PCA, aes(x=phenoCat_final, y=Hue_6)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        facet_wrap(~ros_assembly_543443,ncol=3) 
      

```

```{r RUBIArubia and ROSelrosEL Hue box plots}

# ros_assembly_543443 Allele 2 is pseudomajus
# s261_720757 Allele 2 is pseudomajus
GenoEcol_nuc_PCA[,"s261_720757"] <- as.factor(GenoEcol_nuc_PCA[,"s261_720757"])
GenoEcol_nuc_PCA[,"ros_assembly_543443"] <- as.factor(GenoEcol_nuc_PCA[,"ros_assembly_543443"])

# Can do new approach
RUB_RUB.ROSel_ROSel <- GenoEcol_nuc_PCA %>% filter(ros_assembly_543443==2 & ros_assembly_715001==2 & s261_720757==2)
RUB_rub.ROSel_ROSel <- GenoEcol_nuc_PCA %>% filter(ros_assembly_543443==2 & ros_assembly_715001==2 & s261_720757==1)
rub_rub.ROSel_ROSel <- GenoEcol_nuc_PCA %>% filter(ros_assembly_543443==2 & ros_assembly_715001==2 & s261_720757==0)
RubVariants_ROSel.ROSel_background <- rbind(RUB_RUB.ROSel_ROSel,RUB_rub.ROSel_ROSel,rub_rub.ROSel_ROSel)
RUBrosHaps <- c(rep("RUB_RUB.ROSel_ROSel",nrow(RUB_RUB.ROSel_ROSel)),rep("RUB_rub.ROSel_ROSel",nrow(RUB_rub.ROSel_ROSel)),rep("rub_rub.ROSel_ROSel",nrow(rub_rub.ROSel_ROSel)))
RubVariants_ROSel.ROSel_background <- cbind(RubVariants_ROSel.ROSel_background,RUBrosHaps)


RUB_RUB.ROSel_rosEL <- GenoEcol_nuc_PCA %>% filter(ros_assembly_543443==1 & ros_assembly_715001==1 & s261_720757==2)
RUB_rub.ROSel_rosEL <- GenoEcol_nuc_PCA %>% filter(ros_assembly_543443==1 & ros_assembly_715001==1 & s261_720757==1)
rub_rub.ROSel_rosEL <- GenoEcol_nuc_PCA %>% filter(ros_assembly_543443==1 & ros_assembly_715001==1 & s261_720757==0)
RubVariants_ROSel.rosEL_background <- rbind(RUB_RUB.ROSel_rosEL,RUB_rub.ROSel_rosEL,rub_rub.ROSel_rosEL)
RUBrosHaps <- c(rep("RUB_RUB.ROSel_rosEL",nrow(RUB_RUB.ROSel_rosEL)),rep("RUB_rub.ROSel_rosEL",nrow(RUB_rub.ROSel_rosEL)),rep("rub_rub.ROSel_rosEL",nrow(rub_rub.ROSel_rosEL)))
RubVariants_ROSel.rosEL_background <- cbind(RubVariants_ROSel.rosEL_background,RUBrosHaps)

RUB_RUB.rosEL_rosEL <- GenoEcol_nuc_PCA %>% filter(ros_assembly_543443==0 & ros_assembly_715001==0 & s261_720757==2)
RUB_rub.rosEL_rosEL <- GenoEcol_nuc_PCA %>% filter(ros_assembly_543443==0 & ros_assembly_715001==0 & s261_720757==1)
rub_rub.rosEL_rosEL <- GenoEcol_nuc_PCA %>% filter(ros_assembly_543443==0 & ros_assembly_715001==0 & s261_720757==0)
RubVariants_rosEL.rosEL_background <- rbind(RUB_RUB.rosEL_rosEL,RUB_rub.rosEL_rosEL,rub_rub.rosEL_rosEL)
RUBrosHaps <- c(rep("RUB_RUB.rosEL_rosEL",nrow(RUB_RUB.rosEL_rosEL)),rep("RUB_rub.rosEL_rosEL",nrow(RUB_rub.rosEL_rosEL)),rep("rub_rub.rosEL_rosEL",nrow(rub_rub.rosEL_rosEL)))
RubVariants_rosEL.rosEL_background <- cbind(RubVariants_rosEL.rosEL_background,RUBrosHaps)

ncol(RubVariants_ROSel.ROSel_background)
ncol(RubVariants_ROSel.rosEL_background)
ncol(RubVariants_rosEL.rosEL_background)

RubVariants_ALL_ROSel_backgrounds <- rbind(RubVariants_ROSel.ROSel_background,RubVariants_ROSel.rosEL_background,RubVariants_rosEL.rosEL_background)
RubVariants_ALL_ROSel_backgrounds[,"RUBrosHaps"] <- as.factor(RubVariants_ALL_ROSel_backgrounds[,"RUBrosHaps"])

write.csv(RubVariants_ALL_ROSel_backgrounds,"/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/poolSeq/version3.5/postAnalyses_R/RubVariants_ALL_ROSel_backgrounds.csv")

rub_summary <- RubVariants_ALL_ROSel_backgrounds %>% group_by(s261_720757,ros_assembly_543443) %>% tally()
rub_AllROSbackgrounds_plot_Hue6 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x = s261_720757, y = Hue_6)) +
    geom_boxplot(na.rm = TRUE) +  # Removes non-finite values safely
    theme(
        axis.text = element_text(size = 18),               
        axis.title.y = element_text(size = 20, vjust = 3, hjust = 0.5),  # Y-axis label
        axis.title.x = element_text(size = 20, 
                                    vjust = -2,              # Moves x-axis label down
                                    margin = margin(t = 10), # Small margin to keep spacing neat
                                    hjust = 0.5),             
        strip.text.x = element_text(size = 18),              
        plot.margin = margin(t = 20, r = 20, b = 60, l = 40)  # Increased bottom margin
    ) +
    stat_summary(fun = median, 
                 geom = "text", 
                 aes(label = after_stat(ymax)),  
                 vjust = -1, 
                 na.rm = TRUE) +  
    stat_summary(fun.data = function(x) data.frame(y = max(x, na.rm = TRUE) + 5, label = length(x)),
                 geom = "text", 
                 aes(label = after_stat(label)),
                 vjust = -0.5,
                 size = 5,
                 na.rm = TRUE) +  
    coord_cartesian(ylim = c(-50, 80)) +  
    facet_wrap(~ros_assembly_543443, ncol = 3)
     
rub_AllROSbackgrounds_plot_Hue5 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x = s261_720757, y = Hue_5)) +
    geom_boxplot(na.rm = TRUE) +  # Removes non-finite values safely
    theme(
        axis.text = element_text(size = 18),               
        axis.title.y = element_text(size = 20, vjust = 3, hjust = 0.5),  # Y-axis label
        axis.title.x = element_text(size = 20, 
                                    vjust = -2,              # Moves x-axis label down
                                    margin = margin(t = 10), # Small margin to keep spacing neat
                                    hjust = 0.5),             
        strip.text.x = element_text(size = 18),              
        plot.margin = margin(t = 20, r = 20, b = 60, l = 40)  # Increased bottom margin
    ) +
    stat_summary(fun = median, 
                 geom = "text", 
                 aes(label = after_stat(ymax)),  
                 vjust = -1, 
                 na.rm = TRUE) +  
    stat_summary(fun.data = function(x) data.frame(y = max(x, na.rm = TRUE) + 5, label = length(x)),
                 geom = "text", 
                 aes(label = after_stat(label)),
                 vjust = -0.5,
                 size = 5,
                 na.rm = TRUE) +  
    coord_cartesian(ylim = c(-50, 80)) +  
    facet_wrap(~ros_assembly_543443, ncol = 3)

rub_AllROSbackgrounds_plot_Hue4 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x = s261_720757, y = Hue_4)) +
    geom_boxplot(na.rm = TRUE) +  # Removes non-finite values safely
    theme(
        axis.text = element_text(size = 18),               
        axis.title.y = element_text(size = 20, vjust = 3, hjust = 0.5),  # Y-axis label
        axis.title.x = element_text(size = 20, 
                                    vjust = -2,              # Moves x-axis label down
                                    margin = margin(t = 10), # Small margin to keep spacing neat
                                    hjust = 0.5),             
        strip.text.x = element_text(size = 18),              
        plot.margin = margin(t = 20, r = 20, b = 60, l = 40)  # Increased bottom margin
    ) +
    stat_summary(fun = median, 
                 geom = "text", 
                 aes(label = after_stat(ymax)),  
                 vjust = -1, 
                 na.rm = TRUE) +  
    stat_summary(fun.data = function(x) data.frame(y = max(x, na.rm = TRUE) + 5, label = length(x)),
                 geom = "text", 
                 aes(label = after_stat(label)),
                 vjust = -0.5,
                 size = 5,
                 na.rm = TRUE) +  
    coord_cartesian(ylim = c(-50, 80)) +  
    facet_wrap(~ros_assembly_543443, ncol = 3)


rub_AllROSbackgrounds_plot_Hue3 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x = s261_720757, y = Hue_3)) +
    geom_boxplot(na.rm = TRUE) +  # Removes non-finite values safely
    theme(
        axis.text = element_text(size = 18),               
        axis.title.y = element_text(size = 20, vjust = 3, hjust = 0.5),  # Y-axis label
        axis.title.x = element_text(size = 20, 
                                    vjust = -2,              # Moves x-axis label down
                                    margin = margin(t = 10), # Small margin to keep spacing neat
                                    hjust = 0.5),             
        strip.text.x = element_text(size = 18),              
        plot.margin = margin(t = 20, r = 20, b = 60, l = 40)  # Increased bottom margin
    ) +
    stat_summary(fun = median, 
                 geom = "text", 
                 aes(label = after_stat(ymax)),  
                 vjust = -1, 
                 na.rm = TRUE) +  
    stat_summary(fun.data = function(x) data.frame(y = max(x, na.rm = TRUE) + 5, label = length(x)),
                 geom = "text", 
                 aes(label = after_stat(label)),
                 vjust = -0.5,
                 size = 5,
                 na.rm = TRUE) +  
    coord_cartesian(ylim = c(-50, 80)) +  
    facet_wrap(~ros_assembly_543443, ncol = 3)

rub_AllROSbackgrounds_plot_Hue2 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x = s261_720757, y = Hue_2)) +
    geom_boxplot(na.rm = TRUE) +  # Removes non-finite values safely
    theme(
        axis.text = element_text(size = 18),               
        axis.title.y = element_text(size = 20, vjust = 3, hjust = 0.5),  # Y-axis label
        axis.title.x = element_text(size = 20, 
                                    vjust = -2,              # Moves x-axis label down
                                    margin = margin(t = 10), # Small margin to keep spacing neat
                                    hjust = 0.5),             
        strip.text.x = element_text(size = 18),              
        plot.margin = margin(t = 20, r = 20, b = 60, l = 40)  # Increased bottom margin
    ) +
    stat_summary(fun = median, 
                 geom = "text", 
                 aes(label = after_stat(ymax)),  
                 vjust = -1, 
                 na.rm = TRUE) +  
    stat_summary(fun.data = function(x) data.frame(y = max(x, na.rm = TRUE) + 5, label = length(x)),
                 geom = "text", 
                 aes(label = after_stat(label)),
                 vjust = -0.5,
                 size = 5,
                 na.rm = TRUE) +  
    coord_cartesian(ylim = c(-50, 80)) +  
    facet_wrap(~ros_assembly_543443, ncol = 3)

rub_AllROSbackgrounds_plot_Hue1 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x = s261_720757, y = Hue_1)) +
    geom_boxplot(na.rm = TRUE) +  # Removes non-finite values safely
    theme(
        axis.text = element_text(size = 18),               
        axis.title.y = element_text(size = 20, vjust = 3, hjust = 0.5),  # Y-axis label
        axis.title.x = element_text(size = 20, 
                                    vjust = -2,              # Moves x-axis label down
                                    margin = margin(t = 10), # Small margin to keep spacing neat
                                    hjust = 0.5),             
        strip.text.x = element_text(size = 18),              
        plot.margin = margin(t = 20, r = 20, b = 60, l = 40)  # Increased bottom margin
    ) +
    stat_summary(fun = median, 
                 geom = "text", 
                 aes(label = after_stat(ymax)),  
                 vjust = -1, 
                 na.rm = TRUE) +  
    stat_summary(fun.data = function(x) data.frame(y = max(x, na.rm = TRUE) + 5, label = length(x)),
                 geom = "text", 
                 aes(label = after_stat(label)),
                 vjust = -0.5,
                 size = 5,
                 na.rm = TRUE) +  
    coord_cartesian(ylim = c(-50, 80)) +  
    facet_wrap(~ros_assembly_543443, ncol = 3)



plot_grid(rub_AllROSbackgrounds_plot_Hue1, rub_AllROSbackgrounds_plot_Hue2, rub_AllROSbackgrounds_plot_Hue3, rub_AllROSbackgrounds_plot_Hue4,rub_AllROSbackgrounds_plot_Hue5,rub_AllROSbackgrounds_plot_Hue6,labels = c('A', 'B', 'C', 'D','E','F'), align="hv")
```

```{r RUBIA linear model effect size for Hue_6}
# Fit the linear model
RubVariants_ALL_ROS_backgrounds <- RubVariants_ALL_ROSel_backgrounds
RubVariants_ALL_ROS_backgrounds[,"s261_720757"] <- as.numeric(as.vector(RubVariants_ALL_ROS_backgrounds[,"s261_720757"]))
RubVariants_ALL_ROS_backgrounds[,"ros_assembly_543443"] <- as.numeric(as.vector(RubVariants_ALL_ROS_backgrounds[,"ros_assembly_543443"]))

lm_result <- lm(Hue_4 ~ s261_720757, data = RubVariants_ALL_ROS_backgrounds)

ggplot(RubVariants_ALL_ROS_backgrounds, aes(x = s261_720757, y = Hue_4)) +
  geom_boxplot() +
  labs(title = "Genetic Effect Size Across RUBIA Genotypes",
       x = "s261_720757 Genotypes",
       y = "Hue_4")

# Print the summary of the linear model
summary(lm_result)
AIC(lm_result)
BIC(lm_result)

# residuals
residuals <- resid(lm_result)
plot(x = fitted(lm_result), y = residuals, xlab = "Fitted Values", ylab = "Residuals",
     main = "Residual Plot", pch = 16, col = "blue")
abline(h = 0, col = "red", lwd = 2)  # Add a horizontal line at y = 0

hist(residuals, breaks = 20, col = "lightblue", main = "Histogram of Residuals", xlab = "Residuals")

qqnorm(residuals)
qqline(residuals, col = "red")

sqrt_abs_res <- sqrt(abs(residuals))
plot(x = fitted(lm_result), y = sqrt_abs_res, xlab = "Fitted Values",
     ylab = "Square Root of Absolute Residuals", main = "Scale-Location Plot", pch = 16, col = "blue")
abline(h = mean(sqrt_abs_res), col = "red", lwd = 2)  # Add a horizontal line at the mean of the square root of absolute residuals

```

```{r RUBIA linear model with interaction with ROS1 effect size}
# linear model with interaction with ROS
# interaction with ROS
lm_result_interaction <- lm(Hue_4 ~ s261_720757 + ros_assembly_543443 + s261_720757:ros_assembly_543443, data = RubVariants_ALL_ROS_backgrounds)
summary(lm_result_interaction)
AIC(lm_result_interaction)
BIC(lm_result_interaction)

# residuals
residuals <- resid(lm_result_interaction)
plot(x = fitted(lm_result_interaction), y = residuals, xlab = "Fitted Values", ylab = "Residuals",
     main = "Residual Plot", pch = 16, col = "blue")
abline(h = 0, col = "red", lwd = 2)  # Add a horizontal line at y = 0

hist(residuals, breaks = 20, col = "lightblue", main = "Histogram of Residuals", xlab = "Residuals")

qqnorm(residuals)
qqline(residuals, col = "red")

sqrt_abs_res <- sqrt(abs(residuals))
plot(x = fitted(lm_result_interaction), y = sqrt_abs_res, xlab = "Fitted Values",
     ylab = "Square Root of Absolute Residuals", main = "Scale-Location Plot", pch = 16, col = "blue")
abline(h = mean(sqrt_abs_res), col = "red", lwd = 2)  # Add a horizontal line at the mean of the square root of absolute residuals


# Extract coefficient estimates and standard errors
coefficients <- coef(lm_result_interaction)
standard_errors <- sqrt(diag(vcov(lm_result_interaction)))  # Standard errors are the square root of the diagonal elements of the covariance matrix (vcov) of the model

# Assuming your model formula is: Genetic_Score ~ Genotype + Locus1 + Genotype:Locus1 + Locus2 + Genotype:Locus2
# The coefficients are ordered as: Intercept, Genotype2, Genotype3, Locus1, Genotype2:Locus1, Genotype3:Locus1, Locus2, Genotype2:Locus2, Genotype3:Locus2

# Effect size for s261_7207571 (main effect)
effect_size_s261_7207571 <- coefficients[[2]]

# Effect size for s261_7207571 when considering the interaction with Genotype2
effect_size_s261_7207571_Genotype2 <- coefficients[[2]] + coefficients[[3]]

# Effect size for s261_7207571 when considering the interaction with Genotype3
effect_size_s261_7207571_Genotype3 <- coefficients[[2]] + coefficients[[4]]

# Effect size for Locus2 (main effect)
effect_size_Locus2 <- coefficients["Locus2"]

# Effect size for Locus2 when considering the interaction with Genotype2
effect_size_Locus2_Genotype2 <- coefficients["Locus2"] + coefficients["Genotype2:Locus2"]

# Effect size for Locus2 when considering the interaction with Genotype3
effect_size_Locus2_Genotype3 <- coefficients["Locus2"] + coefficients["Genotype3:Locus2"]

```

```{r RUBIArubia and ROSelrosEL Sat box plots}

range(RubVariants_ALL_ROSel_backgrounds[,"Sat_6"])

rub_AllROSbackgrounds_plot_Sat1 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x=s261_720757, y=Sat_1)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(0.01,0.42) +
        facet_wrap(~ros_assembly_543443,ncol=3) 
rub_AllROSbackgrounds_plot_Sat2 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x=s261_720757, y=Sat_2)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(0.01,0.42) +
        facet_wrap(~ros_assembly_543443,ncol=3) 
rub_AllROSbackgrounds_plot_Sat3 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x=s261_720757, y=Sat_3)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(0.01,0.42) +
        facet_wrap(~ros_assembly_543443,ncol=3) 
rub_AllROSbackgrounds_plot_Sat4 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x=s261_720757, y=Sat_4)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(0.01,0.42) +
        facet_wrap(~ros_assembly_543443,ncol=3) 
rub_AllROSbackgrounds_plot_Sat5 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x=s261_720757, y=Sat_5)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(0.01,0.42) +
        facet_wrap(~ros_assembly_543443,ncol=3) 
rub_AllROSbackgrounds_plot_Sat6 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x=s261_720757, y=Sat_6)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(0.01,0.42) +
        facet_wrap(~ros_assembly_543443,ncol=3) 
plot_grid(rub_AllROSbackgrounds_plot_Sat1, rub_AllROSbackgrounds_plot_Sat2, rub_AllROSbackgrounds_plot_Sat3, rub_AllROSbackgrounds_plot_Sat4,rub_AllROSbackgrounds_plot_Sat5,rub_AllROSbackgrounds_plot_Sat6,labels = c('A', 'B', 'C', 'D','E','F'), align="hv")

```

```{r RUBIArubia and ROSelrosEL NQT_Sat box plots}

range(RubVariants_ALL_ROSel_backgrounds[,"NQT_Sat_6"])

rub_AllROSbackgrounds_plot_NQTSat1 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x=s261_720757, y=NQT_Sat_1)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(-3,3) +
        facet_wrap(~ros_assembly_543443,ncol=3) 
rub_AllROSbackgrounds_plot_NQTSat2 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x=s261_720757, y=NQT_Sat_2)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(-3,3) +
        facet_wrap(~ros_assembly_543443,ncol=3) 
rub_AllROSbackgrounds_plot_NQTSat3 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x=s261_720757, y=NQT_Sat_3)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(-3,3) +
        facet_wrap(~ros_assembly_543443,ncol=3) 
rub_AllROSbackgrounds_plot_NQTSat4 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x=s261_720757, y=NQT_Sat_4)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(-3,3) +
        facet_wrap(~ros_assembly_543443,ncol=3) 
rub_AllROSbackgrounds_plot_NQTSat5 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x=s261_720757, y=NQT_Sat_5)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(-3,3) +
        facet_wrap(~ros_assembly_543443,ncol=3) 
rub_AllROSbackgrounds_plot_NQTSat6 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x=s261_720757, y=NQT_Sat_6)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(-3,3) +
        facet_wrap(~ros_assembly_543443,ncol=3) 
plot_grid(rub_AllROSbackgrounds_plot_NQTSat1, rub_AllROSbackgrounds_plot_NQTSat2, rub_AllROSbackgrounds_plot_NQTSat3, rub_AllROSbackgrounds_plot_NQTSat4,rub_AllROSbackgrounds_plot_NQTSat5,rub_AllROSbackgrounds_plot_NQTSat6,labels = c('A', 'B', 'C', 'D','E','F'), align="hv")

```

```{r RUBIArubia and ROSelrosEL Int box plots}

range(RubVariants_ALL_ROSel_backgrounds[,"Int_6"])

rub_AllROSbackgrounds_plot_Int1 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x=s261_720757, y=Int_1)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(0,800) +
        facet_wrap(~ros_assembly_543443,ncol=3) 
rub_AllROSbackgrounds_plot_Int2 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x=s261_720757, y=Int_2)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(0,800) +
        facet_wrap(~ros_assembly_543443,ncol=3) 
rub_AllROSbackgrounds_plot_Int3 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x=s261_720757, y=Int_3)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(0,800) +
        facet_wrap(~ros_assembly_543443,ncol=3) 
rub_AllROSbackgrounds_plot_Int4 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x=s261_720757, y=Int_4)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(0,800) +
        facet_wrap(~ros_assembly_543443,ncol=3) 
rub_AllROSbackgrounds_plot_Int5 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x=s261_720757, y=Int_5)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(0,800) +
        facet_wrap(~ros_assembly_543443,ncol=3) 
rub_AllROSbackgrounds_plot_Int6 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x=s261_720757, y=Int_6)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(0,800) +
        facet_wrap(~ros_assembly_543443,ncol=3) 
plot_grid(rub_AllROSbackgrounds_plot_Int1, rub_AllROSbackgrounds_plot_Int2, rub_AllROSbackgrounds_plot_Int3, rub_AllROSbackgrounds_plot_Int4,rub_AllROSbackgrounds_plot_Int5,rub_AllROSbackgrounds_plot_Int6,labels = c('A', 'B', 'C', 'D','E','F'), align="hv")

```

# HSV 

```{r first linearise the Hue values}
# Step 1: Normalize Hue values to be between 0 and 360 (in case any are outside this range)
#data <- RubVariants_ALL_ROSel_backgrounds %>%
#  mutate(H_1 = ifelse(Hue < 0, Hue + 360, ifelse(Hue > 360, Hue - 360, Hue)))

#RubVariants_ALL_ROSel_backgrounds$Hue_linear <- NULL


# H_1 HSV convert to linear
# Step 2: Compute the median or mean Hue (central reference point)
median_H_1 <- median(RubVariants_ALL_ROSel_backgrounds$H_1)
# Step 3: Calculate the linear distance to the median, accounting for wraparound
RubVariants_ALL_ROSel_backgrounds <- RubVariants_ALL_ROSel_backgrounds %>%
  mutate(H_1_linear = ifelse(abs(H_1 - median_H_1) <= 180, H_1 - median_H_1, 
                             ifelse(H_1 < median_H_1, H_1 - median_H_1 + 360, H_1 - median_H_1 - 360)))
# H_2 HSV convert to linear
# Step 2: Compute the median or mean Hue (central reference point)
median_H_2 <- median(RubVariants_ALL_ROSel_backgrounds$H_2)
# Step 3: Calculate the linear distance to the median, accounting for wraparound
RubVariants_ALL_ROSel_backgrounds <- RubVariants_ALL_ROSel_backgrounds %>%
  mutate(H_2_linear = ifelse(abs(H_2 - median_H_2) <= 180, H_2 - median_H_2, 
                             ifelse(H_2 < median_H_2, H_2 - median_H_2 + 360, H_2 - median_H_2 - 360)))
# H_3 HSV convert to linear
# Step 2: Compute the median or mean Hue (central reference point)
median_H_3 <- median(RubVariants_ALL_ROSel_backgrounds$H_3)
# Step 3: Calculate the linear distance to the median, accounting for wraparound
RubVariants_ALL_ROSel_backgrounds <- RubVariants_ALL_ROSel_backgrounds %>%
  mutate(H_3_linear = ifelse(abs(H_3 - median_H_3) <= 180, H_3 - median_H_3, 
                             ifelse(H_3 < median_H_3, H_3 - median_H_3 + 360, H_3 - median_H_3 - 360)))
# H_4 HSV convert to linear
# Step 2: Compute the median or mean Hue (central reference point)
median_H_4 <- median(RubVariants_ALL_ROSel_backgrounds$H_4)
# Step 3: Calculate the linear distance to the median, accounting for wraparound
RubVariants_ALL_ROSel_backgrounds <- RubVariants_ALL_ROSel_backgrounds %>%
  mutate(H_4_linear = ifelse(abs(H_4 - median_H_4) <= 180, H_4 - median_H_4, 
                             ifelse(H_4 < median_H_4, H_4 - median_H_4 + 360, H_4 - median_H_4 - 360)))
# H_5 HSV convert to linear
# Step 2: Compute the median or mean Hue (central reference point)
median_H_5 <- median(RubVariants_ALL_ROSel_backgrounds$H_5)
# Step 3: Calculate the linear distance to the median, accounting for wraparound
RubVariants_ALL_ROSel_backgrounds <- RubVariants_ALL_ROSel_backgrounds %>%
  mutate(H_5_linear = ifelse(abs(H_5 - median_H_5) <= 180, H_5 - median_H_5, 
                             ifelse(H_5 < median_H_5, H_5 - median_H_5 + 360, H_5 - median_H_5 - 360)))

# H_5 HSV convert to linear
# Step 2: Compute the median or mean Hue (central reference point)
median_H_6 <- median(RubVariants_ALL_ROSel_backgrounds$H_6)
# Step 3: Calculate the linear distance to the median, accounting for wraparound
RubVariants_ALL_ROSel_backgrounds <- RubVariants_ALL_ROSel_backgrounds %>%
  mutate(H_6_linear = ifelse(abs(H_6 - median_H_6) <= 180, H_6 - median_H_6, 
                             ifelse(H_6 < median_H_6, H_6 - median_H_6 + 360, H_6 - median_H_6 - 360)))

```

```{r RUBIArubia and ROSelrosEL Hue in HSV box plots}
rub_summary <- RubVariants_ALL_ROSel_backgrounds %>% group_by(s261_720757,ros_assembly_543443) %>% tally()

# Create the plot
rub_AllROSbackgrounds_plot_H6 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x = s261_720757, y = H_6_linear)) +
    geom_boxplot(na.rm = TRUE) +  
    labs(
        y = "Hue 6",  
        x = expression(italic("RUB")^pse ~ "alleles")  
    ) +
    theme(
        axis.text = element_text(size = 18),               
        axis.title.y = element_text(size = 20, vjust = 3, hjust = 0.5),  
        axis.title.x = element_text(size = 20, 
                                    vjust = -2,              
                                    margin = margin(t = 10), 
                                    hjust = 0.5),             
        strip.text.x = element_text(size = 18),  
        strip.background = element_rect(fill = "gray90"),  
        plot.margin = margin(t = 20, r = 20, b = 60, l = 40)  
    ) +
    stat_summary(fun = median, 
                 geom = "text", 
                 aes(label = after_stat(ymax)),  
                 vjust = -1, 
                 na.rm = TRUE) +  
    stat_summary(fun.data = function(x) data.frame(y = max(x, na.rm = TRUE) + 5, label = length(x)),
                 geom = "text", 
                 aes(label = after_stat(label)),
                 vjust = -0.5,
                 size = 4,
                 na.rm = TRUE) +  
    coord_cartesian(ylim = c(-100, 100)) +  
    facet_wrap(~ros_assembly_543443, ncol = 3)

# # Create the separate label as a text element
# extra_label <- ggplot() + 
#     theme_void() + 
#     annotate("text", x = 1, y = 1, label = expression(italic("ROS")^pse ~ "alleles"), size = 7, fontface = "bold")
# 
# # Combine the label and plot using patchwork
# rub_AllROSbackgrounds_plot_H6 <- extra_label / rub_AllROSbackgrounds_plot_H6 + 
#     plot_layout(heights = c(0.05, 1))  # Adjust heights to position the label properly

     
rub_AllROSbackgrounds_plot_H5 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x = s261_720757, y = H_5_linear)) +
    geom_boxplot(na.rm = TRUE) +  # Removes non-finite values safely
    labs(
        y = "Hue 5",  
        x = expression(italic("RUB")^pse ~ "alleles")  
    ) +
    theme(
        axis.text = element_text(size = 18),               
        axis.title.y = element_text(size = 20, vjust = 3, hjust = 0.5),  # Y-axis label
        axis.title.x = element_text(size = 20, 
                                    vjust = -2,              # Moves x-axis label down
                                    margin = margin(t = 10), # Small margin to keep spacing neat
                                    hjust = 0.5),             
        strip.text.x = element_text(size = 18),              
        plot.margin = margin(t = 20, r = 20, b = 60, l = 40)  # Increased bottom margin
    ) +
    stat_summary(fun = median, 
                 geom = "text", 
                 aes(label = after_stat(ymax)),  
                 vjust = -1, 
                 na.rm = TRUE) +  
    stat_summary(fun.data = function(x) data.frame(y = max(x, na.rm = TRUE) + 5, label = length(x)),
                 geom = "text", 
                 aes(label = after_stat(label)),
                 vjust = -0.5,
                 size = 4,
                 na.rm = TRUE) +  
    coord_cartesian(ylim = c(-100, 100)) +  
    facet_wrap(~ros_assembly_543443, ncol = 3)

rub_AllROSbackgrounds_plot_H4 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x = s261_720757, y = H_4_linear)) +
    geom_boxplot(na.rm = TRUE) +  # Removes non-finite values safely
    labs(
        y = "Hue 4",  
        x = expression(italic("RUB")^pse ~ "alleles")  
    ) +
    theme(
        axis.text = element_text(size = 18),               
        axis.title.y = element_text(size = 20, vjust = 3, hjust = 0.5),  # Y-axis label
        axis.title.x = element_text(size = 20, 
                                    vjust = -2,              # Moves x-axis label down
                                    margin = margin(t = 10), # Small margin to keep spacing neat
                                    hjust = 0.5),             
        strip.text.x = element_text(size = 18),              
        plot.margin = margin(t = 20, r = 20, b = 60, l = 40)  # Increased bottom margin
    ) +
    stat_summary(fun = median, 
                 geom = "text", 
                 aes(label = after_stat(ymax)),  
                 vjust = -1, 
                 na.rm = TRUE) +  
    stat_summary(fun.data = function(x) data.frame(y = max(x, na.rm = TRUE) + 5, label = length(x)),
                 geom = "text", 
                 aes(label = after_stat(label)),
                 vjust = -0.5,
                 size = 4,
                 na.rm = TRUE) +  
    coord_cartesian(ylim = c(-100, 100)) +  
    facet_wrap(~ros_assembly_543443, ncol = 3)


rub_AllROSbackgrounds_plot_H3 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x = s261_720757, y = H_3_linear)) +
    geom_boxplot(na.rm = TRUE) +  # Removes non-finite values safely
    labs(
        y = "Hue 3",  
        x = expression(italic("RUB")^pse ~ "alleles")  
    ) +
    theme(
        axis.text = element_text(size = 18),               
        axis.title.y = element_text(size = 20, vjust = 3, hjust = 0.5),  # Y-axis label
        axis.title.x = element_text(size = 20, 
                                    vjust = -2,              # Moves x-axis label down
                                    margin = margin(t = 10), # Small margin to keep spacing neat
                                    hjust = 0.5),             
        strip.text.x = element_text(size = 18),              
        plot.margin = margin(t = 20, r = 20, b = 60, l = 40)  # Increased bottom margin
    ) +
    stat_summary(fun = median, 
                 geom = "text", 
                 aes(label = after_stat(ymax)),  
                 vjust = -1, 
                 na.rm = TRUE) +  
    stat_summary(fun.data = function(x) data.frame(y = max(x, na.rm = TRUE) + 5, label = length(x)),
                 geom = "text", 
                 aes(label = after_stat(label)),
                 vjust = -0.5,
                 size = 4,
                 na.rm = TRUE) +  
    coord_cartesian(ylim = c(-100, 100)) +  
    facet_wrap(~ros_assembly_543443, ncol = 3)

rub_AllROSbackgrounds_plot_H2 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x = s261_720757, y = H_2_linear)) +
    geom_boxplot(na.rm = TRUE) +  # Removes non-finite values safely
   labs(
        y = "Hue 2",  
        x = expression(italic("RUB")^pse ~ "alleles")  
    ) +
    theme(
        axis.text = element_text(size = 18),               
        axis.title.y = element_text(size = 20, vjust = 3, hjust = 0.5),  # Y-axis label
        axis.title.x = element_text(size = 20, 
                                    vjust = -2,              # Moves x-axis label down
                                    margin = margin(t = 10), # Small margin to keep spacing neat
                                    hjust = 0.5),             
        strip.text.x = element_text(size = 18),              
        plot.margin = margin(t = 20, r = 20, b = 60, l = 40)  # Increased bottom margin
    ) +
    stat_summary(fun = median, 
                 geom = "text", 
                 aes(label = after_stat(ymax)),  
                 vjust = -1, 
                 na.rm = TRUE) +  
    stat_summary(fun.data = function(x) data.frame(y = max(x, na.rm = TRUE) + 5, label = length(x)),
                 geom = "text", 
                 aes(label = after_stat(label)),
                 vjust = -0.5,
                 size = 4,
                 na.rm = TRUE) +  
    coord_cartesian(ylim = c(-100, 100)) +  
    facet_wrap(~ros_assembly_543443, ncol = 3)

rub_AllROSbackgrounds_plot_H1 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x = s261_720757, y = H_1_linear)) +
    geom_boxplot(na.rm = TRUE) +  # Removes non-finite values safely
    labs(
        y = "Hue 1",  
        x = expression(italic("RUB")^pse ~ "alleles")  
    ) +
    theme(
        axis.text = element_text(size = 18),               
        axis.title.y = element_text(size = 20, vjust = 3, hjust = 0.5),  # Y-axis label
        axis.title.x = element_text(size = 20, 
                                    vjust = -2,              # Moves x-axis label down
                                    margin = margin(t = 10), # Small margin to keep spacing neat
                                    hjust = 0.5),             
        strip.text.x = element_text(size = 18),              
        plot.margin = margin(t = 20, r = 20, b = 60, l = 40)  # Increased bottom margin
    ) +
    stat_summary(fun = median, 
                 geom = "text", 
                 aes(label = after_stat(ymax)),  
                 vjust = -1, 
                 na.rm = TRUE) +  
    stat_summary(fun.data = function(x) data.frame(y = max(x, na.rm = TRUE) + 5, label = length(x)),
                 geom = "text", 
                 aes(label = after_stat(label)),
                 vjust = -0.5,
                 size = 4,
                 na.rm = TRUE) +  
    coord_cartesian(ylim = c(-100, 100)) +  
    facet_wrap(~ros_assembly_543443, ncol = 3)



plot_grid(rub_AllROSbackgrounds_plot_H1, rub_AllROSbackgrounds_plot_H2, rub_AllROSbackgrounds_plot_H3, rub_AllROSbackgrounds_plot_H4,rub_AllROSbackgrounds_plot_H5,rub_AllROSbackgrounds_plot_H6,labels = c('A', 'B', 'C', 'D','E','F'), align="hv",label_size = 18)

pdf("HSV_Hue.pdf",width = 15, height = 10, pointsize = 25)
plot_grid(rub_AllROSbackgrounds_plot_H1, rub_AllROSbackgrounds_plot_H2, rub_AllROSbackgrounds_plot_H3, rub_AllROSbackgrounds_plot_H4,rub_AllROSbackgrounds_plot_H5,rub_AllROSbackgrounds_plot_H6,labels = c('A', 'B', 'C', 'D','E','F'), align="hv",label_size = 18)
dev.off()


pdf("HSV_Hue_4.pdf",width = 13, height = 10, pointsize = 25)

ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x = s261_720757, y = H_4_linear)) +
    geom_boxplot(na.rm = TRUE) +  # Removes non-finite values safely
    labs(
        y = "Hue 4",  
        x = expression(italic("RUB")^pse ~ "alleles")  
    ) +
    theme(
        axis.text = element_text(size = 44),               
        axis.title.y = element_text(size = 70, vjust = 3, hjust = 0.5),  # Y-axis label
        axis.title.x = element_text(size = 70, 
                                    vjust = -2,              # Moves x-axis label down
                                    margin = margin(t = 10), # Small margin to keep spacing neat
                                    hjust = 0.5),             
        strip.text.x = element_text(size = 50),              
        plot.margin = margin(t = 20, r = 20, b = 60, l = 40)  # Increased bottom margin
    ) +
    stat_summary(fun = median, 
                 geom = "text", 
                 aes(label = after_stat(ymax)),  
                 vjust = -1, 
                 na.rm = TRUE) +  
    stat_summary(fun.data = function(x) data.frame(y = max(x, na.rm = TRUE) + 5, label = length(x)),
                 geom = "text", 
                 aes(label = after_stat(label)),
                 vjust = -0.5,
                 size = 12,
                 na.rm = TRUE) +  
    coord_cartesian(ylim = c(-100, 50)) +  
    facet_wrap(~ros_assembly_543443, ncol = 3)

dev.off()
```

```{r RUBIArubia and ROSelrosEL Saturation in HSV box plots}

range(RubVariants_ALL_ROSel_backgrounds[,"S_1"])


# Create the plot
rub_AllROSbackgrounds_plot_S6 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x = s261_720757, y = S_6)) +
    geom_boxplot(na.rm = TRUE) +  
    labs(
        y = "Sat 6",  
        x = expression(italic("RUB")^pse ~ "alleles")  
    ) +
    theme(
        axis.text = element_text(size = 18),               
        axis.title.y = element_text(size = 20, vjust = 3, hjust = 0.5),  
        axis.title.x = element_text(size = 20, 
                                    vjust = -2,              
                                    margin = margin(t = 10), 
                                    hjust = 0.5),             
        strip.text.x = element_text(size = 18),  
        strip.background = element_rect(fill = "gray90"),  
        plot.margin = margin(t = 20, r = 20, b = 60, l = 40)  
    ) +
    stat_summary(fun = median, 
                 geom = "text", 
                 aes(label = after_stat(ymax)),  
                 vjust = -1, 
                 na.rm = TRUE) +  
    stat_summary(fun.data = function(x) data.frame(y = max(x, na.rm = TRUE) + 5, label = length(x)),
                 geom = "text", 
                 aes(label = after_stat(label)),
                 vjust = -0.5,
                 size = 4,
                 na.rm = TRUE) +  
    coord_cartesian(ylim = c(0.0, 1)) +  
    facet_wrap(~ros_assembly_543443, ncol = 3)

# # Create the separate label as a text element
# extra_label <- ggplot() + 
#     theme_void() + 
#     annotate("text", x = 1, y = 1, label = expression(italic("ROS")^pse ~ "alleles"), size = 7, fontface = "bold")
# 
# # Combine the label and plot using patchwork
# rub_AllROSbackgrounds_plot_H6 <- extra_label / rub_AllROSbackgrounds_plot_H6 + 
#     plot_layout(heights = c(0.05, 1))  # Adjust heights to position the label properly

     
rub_AllROSbackgrounds_plot_S5 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x = s261_720757, y = S_5)) +
    geom_boxplot(na.rm = TRUE) +  # Removes non-finite values safely
    labs(
        y = "Sat 5",  
        x = expression(italic("RUB")^pse ~ "alleles")  
    ) +
    theme(
        axis.text = element_text(size = 18),               
        axis.title.y = element_text(size = 20, vjust = 3, hjust = 0.5),  # Y-axis label
        axis.title.x = element_text(size = 20, 
                                    vjust = -2,              # Moves x-axis label down
                                    margin = margin(t = 10), # Small margin to keep spacing neat
                                    hjust = 0.5),             
        strip.text.x = element_text(size = 18),              
        plot.margin = margin(t = 20, r = 20, b = 60, l = 40)  # Increased bottom margin
    ) +
    stat_summary(fun = median, 
                 geom = "text", 
                 aes(label = after_stat(ymax)),  
                 vjust = -1, 
                 na.rm = TRUE) +  
    stat_summary(fun.data = function(x) data.frame(y = max(x, na.rm = TRUE) + 5, label = length(x)),
                 geom = "text", 
                 aes(label = after_stat(label)),
                 vjust = -0.5,
                 size = 4,
                 na.rm = TRUE) +  
    coord_cartesian(ylim = c(0.0, 1)) +  
    facet_wrap(~ros_assembly_543443, ncol = 3)

rub_AllROSbackgrounds_plot_S4 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x = s261_720757, y = S_4)) +
    geom_boxplot(na.rm = TRUE) +  # Removes non-finite values safely
    labs(
        y = "Sat 4",  
        x = expression(italic("RUB")^pse ~ "alleles")  
    ) +
    theme(
        axis.text = element_text(size = 18),               
        axis.title.y = element_text(size = 20, vjust = 3, hjust = 0.5),  # Y-axis label
        axis.title.x = element_text(size = 20, 
                                    vjust = -2,              # Moves x-axis label down
                                    margin = margin(t = 10), # Small margin to keep spacing neat
                                    hjust = 0.5),             
        strip.text.x = element_text(size = 18),              
        plot.margin = margin(t = 20, r = 20, b = 60, l = 40)  # Increased bottom margin
    ) +
    stat_summary(fun = median, 
                 geom = "text", 
                 aes(label = after_stat(ymax)),  
                 vjust = -1, 
                 na.rm = TRUE) +  
    stat_summary(fun.data = function(x) data.frame(y = max(x, na.rm = TRUE) + 5, label = length(x)),
                 geom = "text", 
                 aes(label = after_stat(label)),
                 vjust = -0.5,
                 size = 4,
                 na.rm = TRUE) +  
    coord_cartesian(ylim = c(0.0, 1)) +  
    facet_wrap(~ros_assembly_543443, ncol = 3)


rub_AllROSbackgrounds_plot_S3 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x = s261_720757, y = S_3)) +
    geom_boxplot(na.rm = TRUE) +  # Removes non-finite values safely
    labs(
        y = "Sat 3",  
        x = expression(italic("RUB")^pse ~ "alleles")  
    ) +
    theme(
        axis.text = element_text(size = 18),               
        axis.title.y = element_text(size = 20, vjust = 3, hjust = 0.5),  # Y-axis label
        axis.title.x = element_text(size = 20, 
                                    vjust = -2,              # Moves x-axis label down
                                    margin = margin(t = 10), # Small margin to keep spacing neat
                                    hjust = 0.5),             
        strip.text.x = element_text(size = 18),              
        plot.margin = margin(t = 20, r = 20, b = 60, l = 40)  # Increased bottom margin
    ) +
    stat_summary(fun = median, 
                 geom = "text", 
                 aes(label = after_stat(ymax)),  
                 vjust = -1, 
                 na.rm = TRUE) +  
    stat_summary(fun.data = function(x) data.frame(y = max(x, na.rm = TRUE) + 5, label = length(x)),
                 geom = "text", 
                 aes(label = after_stat(label)),
                 vjust = -0.5,
                 size = 4,
                 na.rm = TRUE) +  
    coord_cartesian(ylim = c(0.0, 1)) +  
    facet_wrap(~ros_assembly_543443, ncol = 3)

rub_AllROSbackgrounds_plot_S2 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x = s261_720757, y = S_2)) +
    geom_boxplot(na.rm = TRUE) +  # Removes non-finite values safely
   labs(
        y = "Sat 2",  
        x = expression(italic("RUB")^pse ~ "alleles")  
    ) +
    theme(
        axis.text = element_text(size = 18),               
        axis.title.y = element_text(size = 20, vjust = 3, hjust = 0.5),  # Y-axis label
        axis.title.x = element_text(size = 20, 
                                    vjust = -2,              # Moves x-axis label down
                                    margin = margin(t = 10), # Small margin to keep spacing neat
                                    hjust = 0.5),             
        strip.text.x = element_text(size = 18),              
        plot.margin = margin(t = 20, r = 20, b = 60, l = 40)  # Increased bottom margin
    ) +
    stat_summary(fun = median, 
                 geom = "text", 
                 aes(label = after_stat(ymax)),  
                 vjust = -1, 
                 na.rm = TRUE) +  
    stat_summary(fun.data = function(x) data.frame(y = max(x, na.rm = TRUE) + 5, label = length(x)),
                 geom = "text", 
                 aes(label = after_stat(label)),
                 vjust = -0.5,
                 size = 4,
                 na.rm = TRUE) +  
    coord_cartesian(ylim = c(0.0, 1)) +  
    facet_wrap(~ros_assembly_543443, ncol = 3)

rub_AllROSbackgrounds_plot_S1 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x = s261_720757, y = S_1)) +
    geom_boxplot(na.rm = TRUE) +  # Removes non-finite values safely
    labs(
        y = "Sat 1",  
        x = expression(italic("RUB")^pse ~ "alleles")  
    ) +
    theme(
        axis.text = element_text(size = 18),               
        axis.title.y = element_text(size = 20, vjust = 3, hjust = 0.5),  # Y-axis label
        axis.title.x = element_text(size = 20, 
                                    vjust = -2,              # Moves x-axis label down
                                    margin = margin(t = 10), # Small margin to keep spacing neat
                                    hjust = 0.5),             
        strip.text.x = element_text(size = 18),              
        plot.margin = margin(t = 20, r = 20, b = 60, l = 40)  # Increased bottom margin
    ) +
    stat_summary(fun = median, 
                 geom = "text", 
                 aes(label = after_stat(ymax)),  
                 vjust = -1, 
                 na.rm = TRUE) +  
    stat_summary(fun.data = function(x) data.frame(y = max(x, na.rm = TRUE) + 5, label = length(x)),
                 geom = "text", 
                 aes(label = after_stat(label)),
                 vjust = -0.5,
                 size = 4,
                 na.rm = TRUE) +  
    coord_cartesian(ylim = c(0.0, 1)) +  
    facet_wrap(~ros_assembly_543443, ncol = 3)

pdf("FlowerPheno_HSV_Saturation.pdf",width = 15, height = 10, pointsize = 25)

plot_grid(rub_AllROSbackgrounds_plot_S1, rub_AllROSbackgrounds_plot_S2, rub_AllROSbackgrounds_plot_S3, rub_AllROSbackgrounds_plot_S4,rub_AllROSbackgrounds_plot_S5,rub_AllROSbackgrounds_plot_S6,labels = c('A', 'B', 'C', 'D','E','F'), align="hv",label_size = 18)
dev.off()
```

```{r RUBIArubia and ROSelrosEL Value in HSV box plots}

range(RubVariants_ALL_ROSel_backgrounds[,"V_1"])

# Create the plot
rub_AllROSbackgrounds_plot_V6 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x = s261_720757, y = V_6)) +
    geom_boxplot(na.rm = TRUE) +  
    labs(
        y = "Value 6",  
        x = expression(italic("RUB")^pse ~ "alleles")  
    ) +
    theme(
        axis.text = element_text(size = 18),               
        axis.title.y = element_text(size = 20, vjust = 3, hjust = 0.5),  
        axis.title.x = element_text(size = 20, 
                                    vjust = -2,              
                                    margin = margin(t = 10), 
                                    hjust = 0.5),             
        strip.text.x = element_text(size = 18),  
        strip.background = element_rect(fill = "gray90"),  
        plot.margin = margin(t = 20, r = 20, b = 60, l = 40)  
    ) +
    stat_summary(fun = median, 
                 geom = "text", 
                 aes(label = after_stat(ymax)),  
                 vjust = -1, 
                 na.rm = TRUE) +  
    stat_summary(fun.data = function(x) data.frame(y = max(x, na.rm = TRUE) + 5, label = length(x)),
                 geom = "text", 
                 aes(label = after_stat(label)),
                 vjust = -0.5,
                 size = 4,
                 na.rm = TRUE) +  
    coord_cartesian(ylim = c(0.25, 1)) +  
    facet_wrap(~ros_assembly_543443, ncol = 3)

# # Create the separate label as a text element
# extra_label <- ggplot() + 
#     theme_void() + 
#     annotate("text", x = 1, y = 1, label = expression(italic("ROS")^pse ~ "alleles"), size = 7, fontface = "bold")
# 
# # Combine the label and plot using patchwork
# rub_AllROSbackgrounds_plot_H6 <- extra_label / rub_AllROSbackgrounds_plot_H6 + 
#     plot_layout(heights = c(0.05, 1))  # Adjust heights to position the label properly

     
rub_AllROSbackgrounds_plot_V5 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x = s261_720757, y = V_5)) +
    geom_boxplot(na.rm = TRUE) +  # Removes non-finite values safely
    labs(
        y = "Value 5",  
        x = expression(italic("RUB")^pse ~ "alleles")  
    ) +
    theme(
        axis.text = element_text(size = 18),               
        axis.title.y = element_text(size = 20, vjust = 3, hjust = 0.5),  # Y-axis label
        axis.title.x = element_text(size = 20, 
                                    vjust = -2,              # Moves x-axis label down
                                    margin = margin(t = 10), # Small margin to keep spacing neat
                                    hjust = 0.5),             
        strip.text.x = element_text(size = 18),              
        plot.margin = margin(t = 20, r = 20, b = 60, l = 40)  # Increased bottom margin
    ) +
    stat_summary(fun = median, 
                 geom = "text", 
                 aes(label = after_stat(ymax)),  
                 vjust = -1, 
                 na.rm = TRUE) +  
    stat_summary(fun.data = function(x) data.frame(y = max(x, na.rm = TRUE) + 5, label = length(x)),
                 geom = "text", 
                 aes(label = after_stat(label)),
                 vjust = -0.5,
                 size = 4,
                 na.rm = TRUE) +  
    coord_cartesian(ylim = c(0.25, 1)) +  
    facet_wrap(~ros_assembly_543443, ncol = 3)

rub_AllROSbackgrounds_plot_V4 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x = s261_720757, y = V_4)) +
    geom_boxplot(na.rm = TRUE) +  # Removes non-finite values safely
    labs(
        y = "Value 4",  
        x = expression(italic("RUB")^pse ~ "alleles")  
    ) +
    theme(
        axis.text = element_text(size = 18),               
        axis.title.y = element_text(size = 20, vjust = 3, hjust = 0.5),  # Y-axis label
        axis.title.x = element_text(size = 20, 
                                    vjust = -2,              # Moves x-axis label down
                                    margin = margin(t = 10), # Small margin to keep spacing neat
                                    hjust = 0.5),             
        strip.text.x = element_text(size = 18),              
        plot.margin = margin(t = 20, r = 20, b = 60, l = 40)  # Increased bottom margin
    ) +
    stat_summary(fun = median, 
                 geom = "text", 
                 aes(label = after_stat(ymax)),  
                 vjust = -1, 
                 na.rm = TRUE) +  
    stat_summary(fun.data = function(x) data.frame(y = max(x, na.rm = TRUE) + 5, label = length(x)),
                 geom = "text", 
                 aes(label = after_stat(label)),
                 vjust = -0.5,
                 size = 4,
                 na.rm = TRUE) +  
    coord_cartesian(ylim = c(0.25, 1)) +  
    facet_wrap(~ros_assembly_543443, ncol = 3)


rub_AllROSbackgrounds_plot_V3 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x = s261_720757, y = V_3)) +
    geom_boxplot(na.rm = TRUE) +  # Removes non-finite values safely
    labs(
        y = "Value 3",  
        x = expression(italic("RUB")^pse ~ "alleles")  
    ) +
    theme(
        axis.text = element_text(size = 18),               
        axis.title.y = element_text(size = 20, vjust = 3, hjust = 0.5),  # Y-axis label
        axis.title.x = element_text(size = 20, 
                                    vjust = -2,              # Moves x-axis label down
                                    margin = margin(t = 10), # Small margin to keep spacing neat
                                    hjust = 0.5),             
        strip.text.x = element_text(size = 18),              
        plot.margin = margin(t = 20, r = 20, b = 60, l = 40)  # Increased bottom margin
    ) +
    stat_summary(fun = median, 
                 geom = "text", 
                 aes(label = after_stat(ymax)),  
                 vjust = -1, 
                 na.rm = TRUE) +  
    stat_summary(fun.data = function(x) data.frame(y = max(x, na.rm = TRUE) + 5, label = length(x)),
                 geom = "text", 
                 aes(label = after_stat(label)),
                 vjust = -0.5,
                 size = 4,
                 na.rm = TRUE) +  
    coord_cartesian(ylim = c(0.25, 1)) +  
    facet_wrap(~ros_assembly_543443, ncol = 3)

rub_AllROSbackgrounds_plot_V2 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x = s261_720757, y = V_2)) +
    geom_boxplot(na.rm = TRUE) +  # Removes non-finite values safely
   labs(
        y = "Value 2",  
        x = expression(italic("RUB")^pse ~ "alleles")  
    ) +
    theme(
        axis.text = element_text(size = 18),               
        axis.title.y = element_text(size = 20, vjust = 3, hjust = 0.5),  # Y-axis label
        axis.title.x = element_text(size = 20, 
                                    vjust = -2,              # Moves x-axis label down
                                    margin = margin(t = 10), # Small margin to keep spacing neat
                                    hjust = 0.5),             
        strip.text.x = element_text(size = 18),              
        plot.margin = margin(t = 20, r = 20, b = 60, l = 40)  # Increased bottom margin
    ) +
    stat_summary(fun = median, 
                 geom = "text", 
                 aes(label = after_stat(ymax)),  
                 vjust = -1, 
                 na.rm = TRUE) +  
    stat_summary(fun.data = function(x) data.frame(y = max(x, na.rm = TRUE) + 5, label = length(x)),
                 geom = "text", 
                 aes(label = after_stat(label)),
                 vjust = -0.5,
                 size = 4,
                 na.rm = TRUE) +  
    coord_cartesian(ylim = c(0.25, 1)) +  
    facet_wrap(~ros_assembly_543443, ncol = 3)

rub_AllROSbackgrounds_plot_V1 <- ggplot(RubVariants_ALL_ROSel_backgrounds, aes(x = s261_720757, y = V_1)) +
    geom_boxplot(na.rm = TRUE) +  # Removes non-finite values safely
    labs(
        y = "Value 1",  
        x = expression(italic("RUB")^pse ~ "alleles")  
    ) +
    theme(
        axis.text = element_text(size = 18),               
        axis.title.y = element_text(size = 20, vjust = 3, hjust = 0.5),  # Y-axis label
        axis.title.x = element_text(size = 20, 
                                    vjust = -2,              # Moves x-axis label down
                                    margin = margin(t = 10), # Small margin to keep spacing neat
                                    hjust = 0.5),             
        strip.text.x = element_text(size = 18),              
        plot.margin = margin(t = 20, r = 20, b = 60, l = 40)  # Increased bottom margin
    ) +
    stat_summary(fun = median, 
                 geom = "text", 
                 aes(label = after_stat(ymax)),  
                 vjust = -1, 
                 na.rm = TRUE) +  
    stat_summary(fun.data = function(x) data.frame(y = max(x, na.rm = TRUE) + 5, label = length(x)),
                 geom = "text", 
                 aes(label = after_stat(label)),
                 vjust = -0.5,
                 size = 4,
                 na.rm = TRUE) +  
    coord_cartesian(ylim = c(0.25, 1)) +  
    facet_wrap(~ros_assembly_543443, ncol = 3)



pdf("FlowerPheno_HSV_Value.pdf",width = 15, height = 10, pointsize = 25)

plot_grid(rub_AllROSbackgrounds_plot_V1, rub_AllROSbackgrounds_plot_V2, rub_AllROSbackgrounds_plot_V3, rub_AllROSbackgrounds_plot_V4,rub_AllROSbackgrounds_plot_V5,rub_AllROSbackgrounds_plot_V6,labels = c('A', 'B', 'C', 'D','E','F'), align="hv",label_size = 18)
dev.off()
```

```{r RUBIA linear model effect size for Hue_6}
# Fit the linear model
RubVariants_ALL_ROS_backgrounds <- RubVariants_ALL_ROSel_backgrounds
RubVariants_ALL_ROS_backgrounds[,"s261_720757"] <- as.numeric(as.vector(RubVariants_ALL_ROS_backgrounds[,"s261_720757"]))
RubVariants_ALL_ROS_backgrounds[,"ros_assembly_543443"] <- as.numeric(as.vector(RubVariants_ALL_ROS_backgrounds[,"ros_assembly_543443"]))

lm_result <- lm(H_4 ~ s261_720757, data = RubVariants_ALL_ROS_backgrounds)

ggplot(RubVariants_ALL_ROS_backgrounds, aes(x = s261_720757, y = H_4)) +
  geom_boxplot() +
  labs(title = "Genetic Effect Size Across RUBIA Genotypes",
       x = "s261_720757 Genotypes",
       y = "Hue_4")

# Print the summary of the linear model
summary(lm_result)
AIC(lm_result)
BIC(lm_result)

# residuals
residuals <- resid(lm_result)
plot(x = fitted(lm_result), y = residuals, xlab = "Fitted Values", ylab = "Residuals",
     main = "Residual Plot", pch = 16, col = "blue")
abline(h = 0, col = "red", lwd = 2)  # Add a horizontal line at y = 0

hist(residuals, breaks = 20, col = "lightblue", main = "Histogram of Residuals", xlab = "Residuals")

qqnorm(residuals)
qqline(residuals, col = "red")

sqrt_abs_res <- sqrt(abs(residuals))
plot(x = fitted(lm_result), y = sqrt_abs_res, xlab = "Fitted Values",
     ylab = "Square Root of Absolute Residuals", main = "Scale-Location Plot", pch = 16, col = "blue")
abline(h = mean(sqrt_abs_res), col = "red", lwd = 2)  # Add a horizontal line at the mean of the square root of absolute residuals

```

Main statistical model in paper in H4, but H5 also significant (the rest are not)

```{r RUBIA linear model with interaction with ROS1 effect size H4}
# linear model with interaction with ROS
# interaction with ROS
RubVariants_ALL_ROS_backgrounds$s261_720757 <- as.factor(RubVariants_ALL_ROS_backgrounds$s261_720757)
RubVariants_ALL_ROS_backgrounds$ros_assembly_543443 <- as.factor(RubVariants_ALL_ROS_backgrounds$ros_assembly_543443)

lm_result_interaction_H4 <- lm(H_4 ~ s261_720757 + ros_assembly_543443 + s261_720757:ros_assembly_543443, data = RubVariants_ALL_ROS_backgrounds)
summary(lm_result_interaction_H4)
AIC(lm_result_interaction_H4)
BIC(lm_result_interaction_H4)

par(mfrow = c(2, 2))  # Set plotting layout for diagnostics
plot(lm_result_interaction_H4)  # Residual plots

# comparison of models shows the interaction is a better model fit 
lm_no_interaction <- lm(H_4 ~ s261_720757 + ros_assembly_543443, data = RubVariants_ALL_ROS_backgrounds)
anova(lm_no_interaction, lm_result_interaction_H4)

# residuals
residuals <- resid(lm_result_interaction)
plot(x = fitted(lm_result_interaction), y = residuals, xlab = "Fitted Values", ylab = "Residuals",
     main = "Residual Plot", pch = 16, col = "blue")
abline(h = 0, col = "red", lwd = 2)  # Add a horizontal line at y = 0

hist(residuals, breaks = 20, col = "lightblue", main = "Histogram of Residuals", xlab = "Residuals")

qqnorm(residuals)
qqline(residuals, col = "red")

sqrt_abs_res <- sqrt(abs(residuals))
plot(x = fitted(lm_result_interaction), y = sqrt_abs_res, xlab = "Fitted Values",
     ylab = "Square Root of Absolute Residuals", main = "Scale-Location Plot", pch = 16, col = "blue")
abline(h = mean(sqrt_abs_res), col = "red", lwd = 2)  # Add a horizontal line at the mean of the square root of absolute residuals


# Extract coefficient estimates and standard errors
coefficients <- coef(lm_result_interaction)
standard_errors <- sqrt(diag(vcov(lm_result_interaction)))  # Standard errors are the square root of the diagonal elements of the covariance matrix (vcov) of the model

# Assuming your model formula is: Genetic_Score ~ Genotype + Locus1 + Genotype:Locus1 + Locus2 + Genotype:Locus2
# The coefficients are ordered as: Intercept, Genotype2, Genotype3, Locus1, Genotype2:Locus1, Genotype3:Locus1, Locus2, Genotype2:Locus2, Genotype3:Locus2

# Effect size for s261_7207571 (main effect)
effect_size_s261_7207571 <- coefficients[[2]]

# Effect size for s261_7207571 when considering the interaction with Genotype2
effect_size_s261_7207571_Genotype2 <- coefficients[[2]] + coefficients[[3]]

# Effect size for s261_7207571 when considering the interaction with Genotype3
effect_size_s261_7207571_Genotype3 <- coefficients[[2]] + coefficients[[4]]

# Effect size for Locus2 (main effect)
effect_size_Locus2 <- coefficients["Locus2"]

# Effect size for Locus2 when considering the interaction with Genotype2
effect_size_Locus2_Genotype2 <- coefficients["Locus2"] + coefficients["Genotype2:Locus2"]

# Effect size for Locus2 when considering the interaction with Genotype3
effect_size_Locus2_Genotype3 <- coefficients["Locus2"] + coefficients["Genotype3:Locus2"]


# nice output
lm_result_interaction
model_table <- tidy(lm_result_interaction)
print(model_table)

library(stargazer)
# Generate formatted table
stargazer(lm_result_interaction, type = "text", 
          title = "Regression Results", 
          digits = 3, 
          star.cutoffs = c(0.05, 0.01, 0.001))

# Format results using gt
# tidy(lm_result_interaction) %>%
#   gt() %>%
#   tab_header(title = "Regression Interaction Results") %>%
#   fmt_number(columns = c(estimate, std.error, statistic, p.value), decimals = 3)

tidy(lm_result_interaction_H4) %>%
  kable(format = "html", digits = 5) %>%
  kable_styling(full_width = FALSE, bootstrap_options = c("striped", "hover"))

# new attempt to look at effect sizes

```

```{r RUBIA linear model with interaction with ROS1 effect size H5}
# linear model with interaction with ROS
# interaction with ROS
lm_result_interaction_H5 <- lm(H_5 ~ s261_720757 + ros_assembly_543443 + s261_720757:ros_assembly_543443, data = RubVariants_ALL_ROS_backgrounds)
summary(lm_result_interaction)
AIC(lm_result_interaction)
BIC(lm_result_interaction)

# residuals
residuals <- resid(lm_result_interaction)
plot(x = fitted(lm_result_interaction), y = residuals, xlab = "Fitted Values", ylab = "Residuals",
     main = "Residual Plot", pch = 16, col = "blue")
abline(h = 0, col = "red", lwd = 2)  # Add a horizontal line at y = 0

hist(residuals, breaks = 20, col = "lightblue", main = "Histogram of Residuals", xlab = "Residuals")

qqnorm(residuals)
qqline(residuals, col = "red")

sqrt_abs_res <- sqrt(abs(residuals))
plot(x = fitted(lm_result_interaction), y = sqrt_abs_res, xlab = "Fitted Values",
     ylab = "Square Root of Absolute Residuals", main = "Scale-Location Plot", pch = 16, col = "blue")
abline(h = mean(sqrt_abs_res), col = "red", lwd = 2)  # Add a horizontal line at the mean of the square root of absolute residuals


# Extract coefficient estimates and standard errors
coefficients <- coef(lm_result_interaction)
standard_errors <- sqrt(diag(vcov(lm_result_interaction)))  # Standard errors are the square root of the diagonal elements of the covariance matrix (vcov) of the model

# Assuming your model formula is: Genetic_Score ~ Genotype + Locus1 + Genotype:Locus1 + Locus2 + Genotype:Locus2
# The coefficients are ordered as: Intercept, Genotype2, Genotype3, Locus1, Genotype2:Locus1, Genotype3:Locus1, Locus2, Genotype2:Locus2, Genotype3:Locus2

# Effect size for s261_7207571 (main effect)
effect_size_s261_7207571 <- coefficients[[2]]

# Effect size for s261_7207571 when considering the interaction with Genotype2
effect_size_s261_7207571_Genotype2 <- coefficients[[2]] + coefficients[[3]]

# Effect size for s261_7207571 when considering the interaction with Genotype3
effect_size_s261_7207571_Genotype3 <- coefficients[[2]] + coefficients[[4]]

# Effect size for Locus2 (main effect)
effect_size_Locus2 <- coefficients["Locus2"]

# Effect size for Locus2 when considering the interaction with Genotype2
effect_size_Locus2_Genotype2 <- coefficients["Locus2"] + coefficients["Genotype2:Locus2"]

# Effect size for Locus2 when considering the interaction with Genotype3
effect_size_Locus2_Genotype3 <- coefficients["Locus2"] + coefficients["Genotype3:Locus2"]

```

```{r RUBIA linear model with interaction with ROS1 effect size H6}
# linear model with interaction with ROS
# interaction with ROS
lm_result_interaction_H6 <- lm(H_6 ~ s261_720757 + ros_assembly_543443 + s261_720757:ros_assembly_543443, data = RubVariants_ALL_ROS_backgrounds)
summary(lm_result_interaction)
AIC(lm_result_interaction)
BIC(lm_result_interaction)

# residuals
residuals <- resid(lm_result_interaction)
plot(x = fitted(lm_result_interaction), y = residuals, xlab = "Fitted Values", ylab = "Residuals",
     main = "Residual Plot", pch = 16, col = "blue")
abline(h = 0, col = "red", lwd = 2)  # Add a horizontal line at y = 0

hist(residuals, breaks = 20, col = "lightblue", main = "Histogram of Residuals", xlab = "Residuals")

qqnorm(residuals)
qqline(residuals, col = "red")

sqrt_abs_res <- sqrt(abs(residuals))
plot(x = fitted(lm_result_interaction), y = sqrt_abs_res, xlab = "Fitted Values",
     ylab = "Square Root of Absolute Residuals", main = "Scale-Location Plot", pch = 16, col = "blue")
abline(h = mean(sqrt_abs_res), col = "red", lwd = 2)  # Add a horizontal line at the mean of the square root of absolute residuals


# Extract coefficient estimates and standard errors
coefficients <- coef(lm_result_interaction)
standard_errors <- sqrt(diag(vcov(lm_result_interaction)))  # Standard errors are the square root of the diagonal elements of the covariance matrix (vcov) of the model

# Assuming your model formula is: Genetic_Score ~ Genotype + Locus1 + Genotype:Locus1 + Locus2 + Genotype:Locus2
# The coefficients are ordered as: Intercept, Genotype2, Genotype3, Locus1, Genotype2:Locus1, Genotype3:Locus1, Locus2, Genotype2:Locus2, Genotype3:Locus2

# Effect size for s261_7207571 (main effect)
effect_size_s261_7207571 <- coefficients[[2]]

# Effect size for s261_7207571 when considering the interaction with Genotype2
effect_size_s261_7207571_Genotype2 <- coefficients[[2]] + coefficients[[3]]

# Effect size for s261_7207571 when considering the interaction with Genotype3
effect_size_s261_7207571_Genotype3 <- coefficients[[2]] + coefficients[[4]]

# Effect size for Locus2 (main effect)
effect_size_Locus2 <- coefficients["Locus2"]

# Effect size for Locus2 when considering the interaction with Genotype2
effect_size_Locus2_Genotype2 <- coefficients["Locus2"] + coefficients["Genotype2:Locus2"]

# Effect size for Locus2 when considering the interaction with Genotype3
effect_size_Locus2_Genotype3 <- coefficients["Locus2"] + coefficients["Genotype3:Locus2"]

```

```{r RUBIA linear model with interaction with ROS1 effect size H3}
# linear model with interaction with ROS
# interaction with ROS
lm_result_interaction_H3 <- lm(H_3 ~ s261_720757 + ros_assembly_543443 + s261_720757:ros_assembly_543443, data = RubVariants_ALL_ROS_backgrounds)
summary(lm_result_interaction)
AIC(lm_result_interaction)
BIC(lm_result_interaction)

# residuals
residuals <- resid(lm_result_interaction)
plot(x = fitted(lm_result_interaction), y = residuals, xlab = "Fitted Values", ylab = "Residuals",
     main = "Residual Plot", pch = 16, col = "blue")
abline(h = 0, col = "red", lwd = 2)  # Add a horizontal line at y = 0

hist(residuals, breaks = 20, col = "lightblue", main = "Histogram of Residuals", xlab = "Residuals")

qqnorm(residuals)
qqline(residuals, col = "red")

sqrt_abs_res <- sqrt(abs(residuals))
plot(x = fitted(lm_result_interaction), y = sqrt_abs_res, xlab = "Fitted Values",
     ylab = "Square Root of Absolute Residuals", main = "Scale-Location Plot", pch = 16, col = "blue")
abline(h = mean(sqrt_abs_res), col = "red", lwd = 2)  # Add a horizontal line at the mean of the square root of absolute residuals


# Extract coefficient estimates and standard errors
coefficients <- coef(lm_result_interaction)
standard_errors <- sqrt(diag(vcov(lm_result_interaction)))  # Standard errors are the square root of the diagonal elements of the covariance matrix (vcov) of the model

# Assuming your model formula is: Genetic_Score ~ Genotype + Locus1 + Genotype:Locus1 + Locus2 + Genotype:Locus2
# The coefficients are ordered as: Intercept, Genotype2, Genotype3, Locus1, Genotype2:Locus1, Genotype3:Locus1, Locus2, Genotype2:Locus2, Genotype3:Locus2

# Effect size for s261_7207571 (main effect)
effect_size_s261_7207571 <- coefficients[[2]]

# Effect size for s261_7207571 when considering the interaction with Genotype2
effect_size_s261_7207571_Genotype2 <- coefficients[[2]] + coefficients[[3]]

# Effect size for s261_7207571 when considering the interaction with Genotype3
effect_size_s261_7207571_Genotype3 <- coefficients[[2]] + coefficients[[4]]

# Effect size for Locus2 (main effect)
effect_size_Locus2 <- coefficients["Locus2"]

# Effect size for Locus2 when considering the interaction with Genotype2
effect_size_Locus2_Genotype2 <- coefficients["Locus2"] + coefficients["Genotype2:Locus2"]

# Effect size for Locus2 when considering the interaction with Genotype3
effect_size_Locus2_Genotype3 <- coefficients["Locus2"] + coefficients["Genotype3:Locus2"]

```

```{r RUBIA linear model with interaction with ROS1 effect size H2}
# linear model with interaction with ROS
# interaction with ROS
lm_result_interaction_H2 <- lm(H_2 ~ s261_720757 + ros_assembly_543443 + s261_720757:ros_assembly_543443, data = RubVariants_ALL_ROS_backgrounds)
summary(lm_result_interaction)
AIC(lm_result_interaction)
BIC(lm_result_interaction)

# residuals
residuals <- resid(lm_result_interaction)
plot(x = fitted(lm_result_interaction), y = residuals, xlab = "Fitted Values", ylab = "Residuals",
     main = "Residual Plot", pch = 16, col = "blue")
abline(h = 0, col = "red", lwd = 2)  # Add a horizontal line at y = 0

hist(residuals, breaks = 20, col = "lightblue", main = "Histogram of Residuals", xlab = "Residuals")

qqnorm(residuals)
qqline(residuals, col = "red")

sqrt_abs_res <- sqrt(abs(residuals))
plot(x = fitted(lm_result_interaction), y = sqrt_abs_res, xlab = "Fitted Values",
     ylab = "Square Root of Absolute Residuals", main = "Scale-Location Plot", pch = 16, col = "blue")
abline(h = mean(sqrt_abs_res), col = "red", lwd = 2)  # Add a horizontal line at the mean of the square root of absolute residuals


# Extract coefficient estimates and standard errors
coefficients <- coef(lm_result_interaction)
standard_errors <- sqrt(diag(vcov(lm_result_interaction)))  # Standard errors are the square root of the diagonal elements of the covariance matrix (vcov) of the model

# Assuming your model formula is: Genetic_Score ~ Genotype + Locus1 + Genotype:Locus1 + Locus2 + Genotype:Locus2
# The coefficients are ordered as: Intercept, Genotype2, Genotype3, Locus1, Genotype2:Locus1, Genotype3:Locus1, Locus2, Genotype2:Locus2, Genotype3:Locus2

# Effect size for s261_7207571 (main effect)
effect_size_s261_7207571 <- coefficients[[2]]

# Effect size for s261_7207571 when considering the interaction with Genotype2
effect_size_s261_7207571_Genotype2 <- coefficients[[2]] + coefficients[[3]]

# Effect size for s261_7207571 when considering the interaction with Genotype3
effect_size_s261_7207571_Genotype3 <- coefficients[[2]] + coefficients[[4]]

# Effect size for Locus2 (main effect)
effect_size_Locus2 <- coefficients["Locus2"]

# Effect size for Locus2 when considering the interaction with Genotype2
effect_size_Locus2_Genotype2 <- coefficients["Locus2"] + coefficients["Genotype2:Locus2"]

# Effect size for Locus2 when considering the interaction with Genotype3
effect_size_Locus2_Genotype3 <- coefficients["Locus2"] + coefficients["Genotype3:Locus2"]

```

```{r RUBIA linear model with interaction with ROS1 effect size H1}
# linear model with interaction with ROS
# interaction with ROS
lm_result_interaction_H1 <- lm(H_1 ~ s261_720757 + ros_assembly_543443 + s261_720757:ros_assembly_543443, data = RubVariants_ALL_ROS_backgrounds)
summary(lm_result_interaction)
AIC(lm_result_interaction)
BIC(lm_result_interaction)

# residuals
residuals <- resid(lm_result_interaction)
plot(x = fitted(lm_result_interaction), y = residuals, xlab = "Fitted Values", ylab = "Residuals",
     main = "Residual Plot", pch = 16, col = "blue")
abline(h = 0, col = "red", lwd = 2)  # Add a horizontal line at y = 0

hist(residuals, breaks = 20, col = "lightblue", main = "Histogram of Residuals", xlab = "Residuals")

qqnorm(residuals)
qqline(residuals, col = "red")

sqrt_abs_res <- sqrt(abs(residuals))
plot(x = fitted(lm_result_interaction), y = sqrt_abs_res, xlab = "Fitted Values",
     ylab = "Square Root of Absolute Residuals", main = "Scale-Location Plot", pch = 16, col = "blue")
abline(h = mean(sqrt_abs_res), col = "red", lwd = 2)  # Add a horizontal line at the mean of the square root of absolute residuals


# Extract coefficient estimates and standard errors
coefficients <- coef(lm_result_interaction)
standard_errors <- sqrt(diag(vcov(lm_result_interaction)))  # Standard errors are the square root of the diagonal elements of the covariance matrix (vcov) of the model

# Assuming your model formula is: Genetic_Score ~ Genotype + Locus1 + Genotype:Locus1 + Locus2 + Genotype:Locus2
# The coefficients are ordered as: Intercept, Genotype2, Genotype3, Locus1, Genotype2:Locus1, Genotype3:Locus1, Locus2, Genotype2:Locus2, Genotype3:Locus2

# Effect size for s261_7207571 (main effect)
effect_size_s261_7207571 <- coefficients[[2]]

# Effect size for s261_7207571 when considering the interaction with Genotype2
effect_size_s261_7207571_Genotype2 <- coefficients[[2]] + coefficients[[3]]

# Effect size for s261_7207571 when considering the interaction with Genotype3
effect_size_s261_7207571_Genotype3 <- coefficients[[2]] + coefficients[[4]]

# Effect size for Locus2 (main effect)
effect_size_Locus2 <- coefficients["Locus2"]

# Effect size for Locus2 when considering the interaction with Genotype2
effect_size_Locus2_Genotype2 <- coefficients["Locus2"] + coefficients["Genotype2:Locus2"]

# Effect size for Locus2 when considering the interaction with Genotype3
effect_size_Locus2_Genotype3 <- coefficients["Locus2"] + coefficients["Genotype3:Locus2"]

```

```{r combine model summary into one table with stargazer and gt}
stargazer(lm_result_interaction_H1, lm_result_interaction_H2, lm_result_interaction_H3, lm_result_interaction_H4, lm_result_interaction_H5, lm_result_interaction_H6, type = "text",title = "Regression Results for Six Models",digits = 6, star.cutoffs = c(0.05, 0.01, 0.001))


# Function to tidy each model
tidy_model_safe <- function(model, model_name) {
  tryCatch({
    # Tidy model results and add model name
    tidy(model) %>%
      mutate(Model = model_name)
  }, error = function(e) {
    # Catch errors and return a message if model tidying fails
    message(paste("Error with model:", model_name))
    return(NULL)  # Return NULL for problematic models
  })
}
model_table <- bind_rows(
  tidy_model_safe(lm_result_interaction_H1, "Model 1"),
  tidy_model_safe(lm_result_interaction_H2, "Model 2"),
  tidy_model_safe(lm_result_interaction_H3, "Model 3"),
  tidy_model_safe(lm_result_interaction_H4, "Model 4"),
  tidy_model_safe(lm_result_interaction_H5, "Model 5"),
  tidy_model_safe(lm_result_interaction_H6, "Model 6")
)
print(model_table)
stargazer(lm_result_interaction_H1, lm_result_interaction_H2, lm_result_interaction_H3, lm_result_interaction_H4, lm_result_interaction_H5, lm_result_interaction_H6, type = "text",
title = "Regression Results for Six Models",digits = 6, star.cutoffs = c(0.05, 0.01, 0.001))


# Format results using gt
model_table %>%
     gt() %>%
     tab_header(title = "Summary of Six Linear Regressions") %>%
     fmt_number(columns = c(estimate, std.error, statistic, p.value), decimals = 3) %>%
     cols_label(
         estimate = "Estimate", 
         std.error = "Std. Error", 
         statistic = "t-value", 
         p.value = "p-value"
     )
library(writexl)

# Assuming model_table is the table you want to export
write_xlsx(model_table, "model_results.xlsx")
```

Now for Saturation


```{r RUBIA linear model with interaction with ROS1 effect size S4}
# linear model with interaction with ROS
# interaction with ROS
RubVariants_ALL_ROS_backgrounds$s261_720757 <- as.factor(RubVariants_ALL_ROS_backgrounds$s261_720757)
RubVariants_ALL_ROS_backgrounds$ros_assembly_543443 <- as.factor(RubVariants_ALL_ROS_backgrounds$ros_assembly_543443)

lm_result_interaction_S4 <- lm(S_4 ~ s261_720757 + ros_assembly_543443 + s261_720757:ros_assembly_543443, data = RubVariants_ALL_ROS_backgrounds)
summary(lm_result_interaction_S4)
AIC(lm_result_interaction_S4)
BIC(lm_result_interaction_S4)

par(mfrow = c(2, 2))  # Set plotting layout for diagnostics
plot(lm_result_interaction_S4)  # Residual plots

# comparison of models shows the interaction is a better model fit 
lm_no_interaction <- lm(H_4 ~ s261_720757 + ros_assembly_543443, data = RubVariants_ALL_ROS_backgrounds)
anova(lm_no_interaction, lm_result_interaction_H4)

# residuals
residuals <- resid(lm_result_interaction)
plot(x = fitted(lm_result_interaction), y = residuals, xlab = "Fitted Values", ylab = "Residuals",
     main = "Residual Plot", pch = 16, col = "blue")
abline(h = 0, col = "red", lwd = 2)  # Add a horizontal line at y = 0

hist(residuals, breaks = 20, col = "lightblue", main = "Histogram of Residuals", xlab = "Residuals")

qqnorm(residuals)
qqline(residuals, col = "red")

sqrt_abs_res <- sqrt(abs(residuals))
plot(x = fitted(lm_result_interaction), y = sqrt_abs_res, xlab = "Fitted Values",
     ylab = "Square Root of Absolute Residuals", main = "Scale-Location Plot", pch = 16, col = "blue")
abline(h = mean(sqrt_abs_res), col = "red", lwd = 2)  # Add a horizontal line at the mean of the square root of absolute residuals


# Extract coefficient estimates and standard errors
coefficients <- coef(lm_result_interaction)
standard_errors <- sqrt(diag(vcov(lm_result_interaction)))  # Standard errors are the square root of the diagonal elements of the covariance matrix (vcov) of the model

# Assuming your model formula is: Genetic_Score ~ Genotype + Locus1 + Genotype:Locus1 + Locus2 + Genotype:Locus2
# The coefficients are ordered as: Intercept, Genotype2, Genotype3, Locus1, Genotype2:Locus1, Genotype3:Locus1, Locus2, Genotype2:Locus2, Genotype3:Locus2

# Effect size for s261_7207571 (main effect)
effect_size_s261_7207571 <- coefficients[[2]]

# Effect size for s261_7207571 when considering the interaction with Genotype2
effect_size_s261_7207571_Genotype2 <- coefficients[[2]] + coefficients[[3]]

# Effect size for s261_7207571 when considering the interaction with Genotype3
effect_size_s261_7207571_Genotype3 <- coefficients[[2]] + coefficients[[4]]

# Effect size for Locus2 (main effect)
effect_size_Locus2 <- coefficients["Locus2"]

# Effect size for Locus2 when considering the interaction with Genotype2
effect_size_Locus2_Genotype2 <- coefficients["Locus2"] + coefficients["Genotype2:Locus2"]

# Effect size for Locus2 when considering the interaction with Genotype3
effect_size_Locus2_Genotype3 <- coefficients["Locus2"] + coefficients["Genotype3:Locus2"]


# nice output
lm_result_interaction
model_table <- tidy(lm_result_interaction)
print(model_table)

library(stargazer)
# Generate formatted table
stargazer(lm_result_interaction, type = "text", 
          title = "Regression Results", 
          digits = 3, 
          star.cutoffs = c(0.05, 0.01, 0.001))

# Format results using gt
# tidy(lm_result_interaction) %>%
#   gt() %>%
#   tab_header(title = "Regression Interaction Results") %>%
#   fmt_number(columns = c(estimate, std.error, statistic, p.value), decimals = 3)

tidy(lm_result_interaction) %>%
  kable(format = "html", digits = 5) %>%
  kable_styling(full_width = FALSE, bootstrap_options = c("striped", "hover"))
```

```{r RUBIA linear model with interaction with ROS1 effect size S5}
# linear model with interaction with ROS
# interaction with ROS
lm_result_interaction_S5 <- lm(S_5 ~ s261_720757 + ros_assembly_543443 + s261_720757:ros_assembly_543443, data = RubVariants_ALL_ROS_backgrounds)
summary(lm_result_interaction_S5)
AIC(lm_result_interaction_S5)
BIC(lm_result_interaction_S5)

# residuals
residuals <- resid(lm_result_interaction_S5)
plot(x = fitted(lm_result_interaction_S5), y = residuals, xlab = "Fitted Values", ylab = "Residuals",
     main = "Residual Plot", pch = 16, col = "blue")
abline(h = 0, col = "red", lwd = 2)  # Add a horizontal line at y = 0

hist(residuals, breaks = 20, col = "lightblue", main = "Histogram of Residuals", xlab = "Residuals")

qqnorm(residuals)
qqline(residuals, col = "red")

sqrt_abs_res <- sqrt(abs(residuals))
plot(x = fitted(lm_result_interaction_S5), y = sqrt_abs_res, xlab = "Fitted Values",
     ylab = "Square Root of Absolute Residuals", main = "Scale-Location Plot", pch = 16, col = "blue")
abline(h = mean(sqrt_abs_res), col = "red", lwd = 2)  # Add a horizontal line at the mean of the square root of absolute residuals


# Extract coefficient estimates and standard errors
coefficients <- coef(lm_result_interaction_S5)
standard_errors <- sqrt(diag(vcov(lm_result_interaction_S5)))  # Standard errors are the square root of the diagonal elements of the covariance matrix (vcov) of the model

# Assuming your model formula is: Genetic_Score ~ Genotype + Locus1 + Genotype:Locus1 + Locus2 + Genotype:Locus2
# The coefficients are ordered as: Intercept, Genotype2, Genotype3, Locus1, Genotype2:Locus1, Genotype3:Locus1, Locus2, Genotype2:Locus2, Genotype3:Locus2

# Effect size for s261_7207571 (main effect)
effect_size_s261_7207571 <- coefficients[[2]]

# Effect size for s261_7207571 when considering the interaction with Genotype2
effect_size_s261_7207571_Genotype2 <- coefficients[[2]] + coefficients[[3]]

# Effect size for s261_7207571 when considering the interaction with Genotype3
effect_size_s261_7207571_Genotype3 <- coefficients[[2]] + coefficients[[4]]

# Effect size for Locus2 (main effect)
effect_size_Locus2 <- coefficients["Locus2"]

# Effect size for Locus2 when considering the interaction with Genotype2
effect_size_Locus2_Genotype2 <- coefficients["Locus2"] + coefficients["Genotype2:Locus2"]

# Effect size for Locus2 when considering the interaction with Genotype3
effect_size_Locus2_Genotype3 <- coefficients["Locus2"] + coefficients["Genotype3:Locus2"]

```

```{r RUBIA linear model with interaction with ROS1 effect size S6}
# linear model with interaction with ROS
# interaction with ROS
lm_result_interaction_S6 <- lm(S_6 ~ s261_720757 + ros_assembly_543443 + s261_720757:ros_assembly_543443, data = RubVariants_ALL_ROS_backgrounds)
summary(lm_result_interaction_S6)
AIC(lm_result_interaction_S6)
BIC(lm_result_interaction_S6)

# residuals
residuals <- resid(lm_result_interaction_S6)
plot(x = fitted(lm_result_interaction_S6), y = residuals, xlab = "Fitted Values", ylab = "Residuals",
     main = "Residual Plot", pch = 16, col = "blue")
abline(h = 0, col = "red", lwd = 2)  # Add a horizontal line at y = 0

hist(residuals, breaks = 20, col = "lightblue", main = "Histogram of Residuals", xlab = "Residuals")

qqnorm(residuals)
qqline(residuals, col = "red")

sqrt_abs_res <- sqrt(abs(residuals))
plot(x = fitted(lm_result_interaction), y = sqrt_abs_res, xlab = "Fitted Values",
     ylab = "Square Root of Absolute Residuals", main = "Scale-Location Plot", pch = 16, col = "blue")
abline(h = mean(sqrt_abs_res), col = "red", lwd = 2)  # Add a horizontal line at the mean of the square root of absolute residuals


# Extract coefficient estimates and standard errors
coefficients <- coef(lm_result_interaction)
standard_errors <- sqrt(diag(vcov(lm_result_interaction)))  # Standard errors are the square root of the diagonal elements of the covariance matrix (vcov) of the model

# Assuming your model formula is: Genetic_Score ~ Genotype + Locus1 + Genotype:Locus1 + Locus2 + Genotype:Locus2
# The coefficients are ordered as: Intercept, Genotype2, Genotype3, Locus1, Genotype2:Locus1, Genotype3:Locus1, Locus2, Genotype2:Locus2, Genotype3:Locus2

# Effect size for s261_7207571 (main effect)
effect_size_s261_7207571 <- coefficients[[2]]

# Effect size for s261_7207571 when considering the interaction with Genotype2
effect_size_s261_7207571_Genotype2 <- coefficients[[2]] + coefficients[[3]]

# Effect size for s261_7207571 when considering the interaction with Genotype3
effect_size_s261_7207571_Genotype3 <- coefficients[[2]] + coefficients[[4]]

# Effect size for Locus2 (main effect)
effect_size_Locus2 <- coefficients["Locus2"]

# Effect size for Locus2 when considering the interaction with Genotype2
effect_size_Locus2_Genotype2 <- coefficients["Locus2"] + coefficients["Genotype2:Locus2"]

# Effect size for Locus2 when considering the interaction with Genotype3
effect_size_Locus2_Genotype3 <- coefficients["Locus2"] + coefficients["Genotype3:Locus2"]

```

```{r RUBIA linear model with interaction with ROS1 effect size S3}
# linear model with interaction with ROS
# interaction with ROS
lm_result_interaction_S3 <- lm(S_3 ~ s261_720757 + ros_assembly_543443 + s261_720757:ros_assembly_543443, data = RubVariants_ALL_ROS_backgrounds)
summary(lm_result_interaction_S3)
AIC(lm_result_interaction_S3)
BIC(lm_result_interaction_S3)

# residuals
residuals <- resid(lm_result_interaction_S3)
plot(x = fitted(lm_result_interaction_S3), y = residuals, xlab = "Fitted Values", ylab = "Residuals",
     main = "Residual Plot", pch = 16, col = "blue")
abline(h = 0, col = "red", lwd = 2)  # Add a horizontal line at y = 0

hist(residuals, breaks = 20, col = "lightblue", main = "Histogram of Residuals", xlab = "Residuals")

qqnorm(residuals)
qqline(residuals, col = "red")

sqrt_abs_res <- sqrt(abs(residuals))
plot(x = fitted(lm_result_interaction), y = sqrt_abs_res, xlab = "Fitted Values",
     ylab = "Square Root of Absolute Residuals", main = "Scale-Location Plot", pch = 16, col = "blue")
abline(h = mean(sqrt_abs_res), col = "red", lwd = 2)  # Add a horizontal line at the mean of the square root of absolute residuals


# Extract coefficient estimates and standard errors
coefficients <- coef(lm_result_interaction)
standard_errors <- sqrt(diag(vcov(lm_result_interaction)))  # Standard errors are the square root of the diagonal elements of the covariance matrix (vcov) of the model

# Assuming your model formula is: Genetic_Score ~ Genotype + Locus1 + Genotype:Locus1 + Locus2 + Genotype:Locus2
# The coefficients are ordered as: Intercept, Genotype2, Genotype3, Locus1, Genotype2:Locus1, Genotype3:Locus1, Locus2, Genotype2:Locus2, Genotype3:Locus2

# Effect size for s261_7207571 (main effect)
effect_size_s261_7207571 <- coefficients[[2]]

# Effect size for s261_7207571 when considering the interaction with Genotype2
effect_size_s261_7207571_Genotype2 <- coefficients[[2]] + coefficients[[3]]

# Effect size for s261_7207571 when considering the interaction with Genotype3
effect_size_s261_7207571_Genotype3 <- coefficients[[2]] + coefficients[[4]]

# Effect size for Locus2 (main effect)
effect_size_Locus2 <- coefficients["Locus2"]

# Effect size for Locus2 when considering the interaction with Genotype2
effect_size_Locus2_Genotype2 <- coefficients["Locus2"] + coefficients["Genotype2:Locus2"]

# Effect size for Locus2 when considering the interaction with Genotype3
effect_size_Locus2_Genotype3 <- coefficients["Locus2"] + coefficients["Genotype3:Locus2"]

```

```{r RUBIA linear model with interaction with ROS1 effect size S2}
# linear model with interaction with ROS
# interaction with ROS
lm_result_interaction_S2 <- lm(S_2 ~ s261_720757 + ros_assembly_543443 + s261_720757:ros_assembly_543443, data = RubVariants_ALL_ROS_backgrounds)
summary(lm_result_interaction_S2)
AIC(lm_result_interaction_S2)
BIC(lm_result_interaction_S2)

# residuals
residuals <- resid(lm_result_interaction_S2)
plot(x = fitted(lm_result_interaction_S2), y = residuals, xlab = "Fitted Values", ylab = "Residuals",
     main = "Residual Plot", pch = 16, col = "blue")
abline(h = 0, col = "red", lwd = 2)  # Add a horizontal line at y = 0

hist(residuals, breaks = 20, col = "lightblue", main = "Histogram of Residuals", xlab = "Residuals")

qqnorm(residuals)
qqline(residuals, col = "red")

sqrt_abs_res <- sqrt(abs(residuals))
plot(x = fitted(lm_result_interaction), y = sqrt_abs_res, xlab = "Fitted Values",
     ylab = "Square Root of Absolute Residuals", main = "Scale-Location Plot", pch = 16, col = "blue")
abline(h = mean(sqrt_abs_res), col = "red", lwd = 2)  # Add a horizontal line at the mean of the square root of absolute residuals


# Extract coefficient estimates and standard errors
coefficients <- coef(lm_result_interaction)
standard_errors <- sqrt(diag(vcov(lm_result_interaction)))  # Standard errors are the square root of the diagonal elements of the covariance matrix (vcov) of the model

# Assuming your model formula is: Genetic_Score ~ Genotype + Locus1 + Genotype:Locus1 + Locus2 + Genotype:Locus2
# The coefficients are ordered as: Intercept, Genotype2, Genotype3, Locus1, Genotype2:Locus1, Genotype3:Locus1, Locus2, Genotype2:Locus2, Genotype3:Locus2

# Effect size for s261_7207571 (main effect)
effect_size_s261_7207571 <- coefficients[[2]]

# Effect size for s261_7207571 when considering the interaction with Genotype2
effect_size_s261_7207571_Genotype2 <- coefficients[[2]] + coefficients[[3]]

# Effect size for s261_7207571 when considering the interaction with Genotype3
effect_size_s261_7207571_Genotype3 <- coefficients[[2]] + coefficients[[4]]

# Effect size for Locus2 (main effect)
effect_size_Locus2 <- coefficients["Locus2"]

# Effect size for Locus2 when considering the interaction with Genotype2
effect_size_Locus2_Genotype2 <- coefficients["Locus2"] + coefficients["Genotype2:Locus2"]

# Effect size for Locus2 when considering the interaction with Genotype3
effect_size_Locus2_Genotype3 <- coefficients["Locus2"] + coefficients["Genotype3:Locus2"]

```

```{r RUBIA linear model with interaction with ROS1 effect size S1}
# linear model with interaction with ROS
# interaction with ROS
lm_result_interaction_S1 <- lm(S_1 ~ s261_720757 + ros_assembly_543443 + s261_720757:ros_assembly_543443, data = RubVariants_ALL_ROS_backgrounds)
summary(lm_result_interaction_S1)
AIC(lm_result_interaction_S1)
BIC(lm_result_interaction_S1)

# residuals
residuals <- resid(lm_result_interaction_S1)
plot(x = fitted(lm_result_interaction_S1), y = residuals, xlab = "Fitted Values", ylab = "Residuals",
     main = "Residual Plot", pch = 16, col = "blue")
abline(h = 0, col = "red", lwd = 2)  # Add a horizontal line at y = 0

hist(residuals, breaks = 20, col = "lightblue", main = "Histogram of Residuals", xlab = "Residuals")

qqnorm(residuals)
qqline(residuals, col = "red")

sqrt_abs_res <- sqrt(abs(residuals))
plot(x = fitted(lm_result_interaction), y = sqrt_abs_res, xlab = "Fitted Values",
     ylab = "Square Root of Absolute Residuals", main = "Scale-Location Plot", pch = 16, col = "blue")
abline(h = mean(sqrt_abs_res), col = "red", lwd = 2)  # Add a horizontal line at the mean of the square root of absolute residuals


# Extract coefficient estimates and standard errors
coefficients <- coef(lm_result_interaction)
standard_errors <- sqrt(diag(vcov(lm_result_interaction)))  # Standard errors are the square root of the diagonal elements of the covariance matrix (vcov) of the model

# Assuming your model formula is: Genetic_Score ~ Genotype + Locus1 + Genotype:Locus1 + Locus2 + Genotype:Locus2
# The coefficients are ordered as: Intercept, Genotype2, Genotype3, Locus1, Genotype2:Locus1, Genotype3:Locus1, Locus2, Genotype2:Locus2, Genotype3:Locus2

# Effect size for s261_7207571 (main effect)
effect_size_s261_7207571 <- coefficients[[2]]

# Effect size for s261_7207571 when considering the interaction with Genotype2
effect_size_s261_7207571_Genotype2 <- coefficients[[2]] + coefficients[[3]]

# Effect size for s261_7207571 when considering the interaction with Genotype3
effect_size_s261_7207571_Genotype3 <- coefficients[[2]] + coefficients[[4]]

# Effect size for Locus2 (main effect)
effect_size_Locus2 <- coefficients["Locus2"]

# Effect size for Locus2 when considering the interaction with Genotype2
effect_size_Locus2_Genotype2 <- coefficients["Locus2"] + coefficients["Genotype2:Locus2"]

# Effect size for Locus2 when considering the interaction with Genotype3
effect_size_Locus2_Genotype3 <- coefficients["Locus2"] + coefficients["Genotype3:Locus2"]

```

```{r combine model summary into one table with stargazer and gt}
stargazer(lm_result_interaction_S1, lm_result_interaction_S2, lm_result_interaction_S3, lm_result_interaction_S4, lm_result_interaction_S5, lm_result_interaction_S6, type = "text",
title = "Regression Results for Six Models",digits = 6, star.cutoffs = c(0.05, 0.01, 0.001))


# Function to tidy each model
tidy_model_safe <- function(model, model_name) {
  tryCatch({
    # Tidy model results and add model name
    tidy(model) %>%
      mutate(Model = model_name)
  }, error = function(e) {
    # Catch errors and return a message if model tidying fails
    message(paste("Error with model:", model_name))
    return(NULL)  # Return NULL for problematic models
  })
}
model_table_S <- bind_rows(
  tidy_model_safe(lm_result_interaction_S1, "Model 1"),
  tidy_model_safe(lm_result_interaction_S2, "Model 2"),
  tidy_model_safe(lm_result_interaction_S3, "Model 3"),
  tidy_model_safe(lm_result_interaction_S4, "Model 4"),
  tidy_model_safe(lm_result_interaction_S5, "Model 5"),
  tidy_model_safe(lm_result_interaction_S6, "Model 6")
)
print(model_table_S)
stargazer(lm_result_interaction_S1, lm_result_interaction_S2, lm_result_interaction_S3, lm_result_interaction_S4, lm_result_interaction_S5, lm_result_interaction_S6, type = "text",
title = "Regression Results for Six Models",digits = 6, star.cutoffs = c(0.05, 0.01, 0.001))


# Format results using gt
model_table_S %>%
     gt() %>%
     tab_header(title = "Summary of Six Linear Regressions") %>%
     fmt_number(columns = c(estimate, std.error, statistic, p.value), decimals = 3) %>%
     cols_label(
         estimate = "Estimate", 
         std.error = "Std. Error", 
         statistic = "t-value", 
         p.value = "p-value"
     )
library(writexl)

# Assuming model_table is the table you want to export
write_xlsx(model_table, "model_results.xlsx")
```


```{r colour checks}
# Assuming your dataframe is already loaded
df <- RubVariants_ALL_ROSel_backgrounds

# Function to convert HSV to HEX color
recreate_color_from_hsv <- function(H, S, V) {
  hsv(H / 360, S, V)  # H should be in degrees (0-360), S and V between 0-1
}

# Apply color conversion using H_4, S_4, V_4
df$color_hex <- mapply(recreate_color_from_hsv, df$H_4, df$S_4, df$V_4)

df$Red_final <- as.numeric(as.character(df$Red_final))
df$Yellow_final <- as.numeric(as.character(df$Yellow_final))

# Function to plot 50 individuals per batch
plot_batch <- function(data, batch_num) {
  start_idx <- (batch_num - 1) * 50 + 1
  end_idx <- min(batch_num * 50, nrow(data))
  
  batch_data <- data[start_idx:end_idx, ]
  batch_data$ID_order <- factor(batch_data$PlantID_final, levels = batch_data$PlantID_final)
  
  ggplot(batch_data, aes(x = 1, y = 1)) +
    # Color block
    geom_tile(aes(fill = color_hex), width = 0.9, height = 0.6) +
    scale_fill_identity() +
    
    # Text block below the color
    geom_text(aes(y = 0.1, label = paste0(
      PlantID_final, "\n",
      phenoCat_final, "\n",
      "Red: ", round(Red_final, 2), ", Yellow: ", round(Yellow_final, 2)
    )), 
    size = 2.5, vjust = 0, lineheight = 1.1, color = "black") +
    
    # Facet for layout
    facet_wrap(~ ID_order, ncol = 5) +
    
    theme_minimal() +
    theme(
      axis.text = element_blank(),
      axis.ticks = element_blank(),
      axis.title = element_blank(),
      panel.grid = element_blank(),
      strip.text = element_blank(),
      plot.title = element_text(size = 14, face = "bold", hjust = 0.5),
      panel.spacing = unit(0.5, "lines")
    ) +
    ggtitle(paste("Batch", batch_num, "- Individuals", start_idx, "to", end_idx))
}

# Plot the first batch (Individuals 1–50)
plot_batch(df, 1)

plot_batch(df, 2)   # Individuals 51–100
plot_batch(df, 4)   # Individuals 101–150
plot_batch(df, 5)   # Individuals 151–200



# check some photos
write.csv(RubVariants_ALL_ROSel_backgrounds,"RubVariants_ALL_ROSel_backgrounds.csv")
```

```{r find photos of Mean values for Hue 4}
# authenticate account details

drop_auth(new_user = T,cache = TRUE,rdstoken = '/Users/MQ10006146/dropBoxRds.txt')
acc_info <- drop_acc()
# extract display name
acc_info$name$display_name
# photo locations
# J 2009,
# K 2010, all files in one folder: /fieldSeason2010/Data_backups/Flower_Photos
# L 2011, all files in one folder: /fieldSeason2011/Data_backups/Flower_Photos
# M 2012, all files in one folder: /fieldSeason2012/Data_backups/Flower_Photos
# P 2013, Olympus camera grouped files in folders by date taken, for example: /fieldSeason2013/Data_backups/Flower_Photos/2013_06_08
# S 2014, Olympus camera grouped files in folders by date taken, for example: /fieldSeason2014/Data_backups/Flower_Photos/2014_06_08
# T 2015, Canon camera grouped files in folders by date taken but different format, for example: /fieldSeason2015/Data_backups/Flower_Photos/100_0906
# 2016 – 2019, same as 2015.

# doing Hue4 - no influence of white face phenotypes on magenta background

par(mfrow=c(1,1))
par(mar=c(0.1,0.1,0.1,0.1))

findPhotoMeans <- function(thisGenoPhenoData,thisColMeasure,thisHaplotype,numPhotos,thisPathSave) {
  # thisGenoPhenoData <- RubVariants_ALL_ROS_backgrounds
  # thisColMeasure <- "Hue_4"
  # thisHaplotype <- "RUB_RUB.ROS_ROS"
  # numPhotos <- 5
  # thisPathSave <- "/Users/dlfield/Library/CloudStorage/OneDrive-SharedLibraries-EdithCowanUniversity/Snapdragons_2022-03193-FIELD - General/6_DATA_ANALYSIS/poolSeq/version3.5/postAnalyses_R/flowerPhotos/Hue_4/"
  thisGenoPhenoData_subset <- thisGenoPhenoData %>% filter(RUBrosHaps==thisHaplotype)
  thisCol_mean <- mean(thisGenoPhenoData_subset[,thisColMeasure])
  dist_mean <- abs(thisCol_mean-thisGenoPhenoData_subset[,thisColMeasure])
  thisGenoPhenoData_subset[13,thisColMeasure]
  thisGenoPhenoData_ordered <- thisGenoPhenoData_subset[order(dist_mean),]
  thisGenoPhenoData_keep <- thisGenoPhenoData_ordered[1:numPhotos,]
  # thisGenoPhenoData_keep[,"RametIDs"]

  for (thisRow in 1:nrow(thisGenoPhenoData_keep)) {
    # colnames(GenoEcol_Planoles_Core_subset)
    # which(GenoEcol_Planoles_Core_subset[,"PlantID_final"]=="S4795")
    # thisRow <- 1
    thisData <- thisGenoPhenoData_keep[thisRow,]
    thisID <- thisGenoPhenoData_keep[thisRow,"PlantID_final"]
    thisLabel <- unlist(strsplit(thisID, ""))[[1]]
    thisLabelRamets <- thisGenoPhenoData_keep[thisRow,"RametIDs"]
    thisFace <- thisData[,"Face"]
    thisSide <- thisData[,"Side"]
    thisRed <- thisData[,"Red_final"]
    thisYellow <- thisData[,"Yellow_final"]
    thisPheno <- thisData[,"phenoCat_final"]
    thisHapName <- thisData[,"multiLoc_Hap_6by7"]
    thisHapName2 <- thisHaplotype
  
    annotateText <- paste0(thisLabelRamets,",[",thisRed,",",thisYellow,"],",thisHapName,", ",thisHaplotype,", ",thisColMeasure," mean = ",round(thisCol_mean,2),", ",thisColMeasure," = ",round(thisData[,thisColMeasure],2))
    annotateText2 <- thisHaplotype
  
    if (thisPheno!=0 | thisPheno!=-9) {
          if (thisPheno =="FR") {
            borderCol <-"magenta"
          }
          if (thisPheno =="Y") {
            borderCol <-"yellow"
            }
          if (thisPheno=="FO") {
             borderCol <-"orange"
          }
          if (thisPheno=="WR") {
            borderCol <-"pink"
          }
          if (thisPheno=="W") {
            borderCol <-"white"
          }
          if (thisPheno=="WO") {
            borderCol <- "wheat"
          }
        }
    cat("\n plant: ", paste0(thisRow,": ",thisID))
    cat("; ",thisLabelRamets)
    cat(", ",thisHapName)
    cat(", ",thisHapName2)
  
    if (thisLabel %in% c("K","L","M")) {
      
      
    }
    if (thisLabel %in% c("P","S")) {
      if (thisLabel == "P") {
        thisDropBoxSearch <- drop_search(query=paste0(thisFace,".JPG"),path = "/Antirrhinum_fieldData/fieldSeason2013/Data_backups/Flower_Photos", start = 0, max_results = 100)
        if (thisDropBoxSearch$start!=0) {
          cat(", Photo found!")
           thisPhoto_link <- drop_media(thisDropBoxSearch$matches[[1]]$metadata$path_display)$link
           finalImage <- image_read(thisPhoto_link) %>%
                      image_background("blue", flatten = TRUE) %>%
                      image_border(borderCol, "30x30") %>%
                      #image_scale("400") %>%
                      image_annotate(annotateText, color = "white", size = 70,gravity = "southwest")
                      #image_annotate(annotateText2, color = "white", size = 16,location="+50+100")
           plot(finalImage)
           image_write(finalImage,paste0(thisPathSave,thisHapName2,"_",thisID,".jpg"))
        }
        if (thisDropBoxSearch$start==0) {
          thisDropBoxSearch <- drop_search(query=paste0(thisFace,".JPG"),path = "/Antirrhinum_fieldData/fieldSeason2014/Data_backups/Flower_Photos", start = 0, max_results = 100)
          if (thisDropBoxSearch$start!=0) {
            cat(", Photo found!")
            thisPhoto_link <- drop_media(thisDropBoxSearch$matches[[1]]$metadata$path_display)$link
            finalImage <- image_read(thisPhoto_link) %>%
                      image_background("blue", flatten = TRUE) %>%
                      image_border(borderCol, "30x30") %>%
                      image_scale("400") %>%
                      image_annotate(annotateText, color = "white", size = 20,gravity = "southwest")
                      #image_annotate(annotateText2, color = "white", size = 22,location="+50+100")
            plot(finalImage)
          }
        }
      }
      if (thisLabel == "S") {
        thisDropBoxSearch <- drop_search(query=paste0(thisFace,".JPG"),path = "/Antirrhinum_fieldData/fieldSeason2014/Data_backups/Flower_Photos", start = 0, max_results = 100)
        if (thisDropBoxSearch$start!=0) {
           cat(", Photo found!")
           thisPhoto_link <- drop_media(thisDropBoxSearch$matches[[1]]$metadata$path_display)$link
           finalImage <- image_read(thisPhoto_link) %>%
                      image_background("blue", flatten = TRUE) %>%
                      image_border(borderCol, "30x30") %>%
                      #image_scale("400") %>%
                      image_annotate(annotateText, color = "white", size = 70,gravity = "southwest")
                      #image_annotate(annotateText2, color = "white", size = 16,location="+50+100")
           plot(finalImage)
           image_write(finalImage,paste0(thisPathSave,thisHapName2,"_",thisID,".jpg"))
        }
      }
    }
  # will need different code like Haralds or Lenkas to find path to other years
  
  }
  
}
unique(RUBrosHaps)

thisPath_Hue4 <- "/Users/dlfield/Library/CloudStorage/OneDrive-SharedLibraries-EdithCowanUniversity/Snapdragons_2022-03193-FIELD - General/6_DATA_ANALYSIS/poolSeq/version3.5/postAnalyses_R/flowerPhotos/Hue_4/"

findPhotoMeans(thisGenoPhenoData = RubVariants_ALL_ROS_backgrounds,thisColMeasure = "Hue_4",thisHaplotype = "RUB_RUB.ROS_ROS",numPhotos = 6,thisPathSave = thisPath_Hue4)

findPhotoMeans(thisGenoPhenoData = RubVariants_ALL_ROS_backgrounds,thisColMeasure = "Hue_4",thisHaplotype = "RUB_rub.ROS_ROS",numPhotos = 6,thisPathSave = thisPath_Hue4)

findPhotoMeans(thisGenoPhenoData = RubVariants_ALL_ROS_backgrounds,thisColMeasure = "Hue_4",thisHaplotype = "rub_rub.ROS_ROS",numPhotos = 6,thisPathSave = thisPath_Hue4)

findPhotoMeans(thisGenoPhenoData = RubVariants_ALL_ROS_backgrounds,thisColMeasure = "Hue_4",thisHaplotype = "RUB_RUB.ROS_ros",numPhotos = 6,thisPathSave = thisPath_Hue4)

findPhotoMeans(thisGenoPhenoData = RubVariants_ALL_ROS_backgrounds,thisColMeasure = "Hue_4",thisHaplotype = "RUB_rub.ROS_ros",numPhotos = 6,thisPathSave = thisPath_Hue4)

findPhotoMeans(thisGenoPhenoData = RubVariants_ALL_ROS_backgrounds,thisColMeasure = "Hue_4",thisHaplotype = "rub_rub.ROS_ros",numPhotos = 6,thisPathSave = thisPath_Hue4)

findPhotoMeans(thisGenoPhenoData = RubVariants_ALL_ROS_backgrounds,thisColMeasure = "Hue_4",thisHaplotype = "RUB_RUB.ros_ros",numPhotos = 6,thisPathSave = thisPath_Hue4)

findPhotoMeans(thisGenoPhenoData = RubVariants_ALL_ROS_backgrounds,thisColMeasure = "Hue_4",thisHaplotype = "RUB_rub.ros_ros",numPhotos = 6,thisPathSave = thisPath_Hue4)

findPhotoMeans(thisGenoPhenoData = RubVariants_ALL_ROS_backgrounds,thisColMeasure = "Hue_4",thisHaplotype = "rub_rub.ros_ros",numPhotos = 6,thisPathSave = thisPath_Hue4)

```

```{r find photos of specific phenotypes}

# load full data set 
GenoEcol <- read.csv(paste0(folderLoc,"/PostProcessedData/FinalGenoEcol_short_int.csv"),header=TRUE,fill=FALSE,na.strings = "?",stringsAsFactors = F)
colnames(GenoEcol)
table(GenoEcol[,"phenoCat_final"])

nrow(GenoEcol)

GenoEcol_oddPhenos <- GenoEcol[GenoEcol[,"phenoCat_final"] %in% c("Y","FR","W","WO","WR","FO","-9")==F,]
nrow(GenoEcol_oddPhenos)
# authenticate account details

drop_auth(new_user = T,cache = TRUE,rdstoken = '/Users/MQ10006146/dropBoxRds.txt')
acc_info <- drop_acc()
# extract display name
acc_info$name$display_name
# photo locations
# J 2009,
# K 2010, all files in one folder: /fieldSeason2010/Data_backups/Flower_Photos
# L 2011, all files in one folder: /fieldSeason2011/Data_backups/Flower_Photos
# M 2012, all files in one folder: /fieldSeason2012/Data_backups/Flower_Photos
# P 2013, Olympus camera grouped files in folders by date taken, for example: /fieldSeason2013/Data_backups/Flower_Photos/2013_06_08
# S 2014, Olympus camera grouped files in folders by date taken, for example: /fieldSeason2014/Data_backups/Flower_Photos/2014_06_08
# T 2015, Canon camera grouped files in folders by date taken but different format, for example: /fieldSeason2015/Data_backups/Flower_Photos/100_0906
# 2016 – 2019, same as 2015.

# doing Hue4 - no influence of white face phenotypes on magenta background

par(mfrow=c(1,1))
par(mar=c(0.1,0.1,0.1,0.1))

findPhoto <- function(thisGenoPhenoData,thisPhenoCat,thisYear,numPhotos,thisPathSave) {
  # thisGenoPhenoData <- GenoEcol_nuc_PCA
  # thisColMeasure <- "Hue_4"
  # thisHaplotype <- "RUB_RUB.ROS_ROS"
  # numPhotos <- 5
  # thisPathSave <- "/Users/dlfield/Library/CloudStorage/OneDrive-SharedLibraries-EdithCowanUniversity/Snapdragons_2022-03193-FIELD - General/6_DATA_ANALYSIS/poolSeq/version3.5/postAnalyses_R/flowerPhotos/Hue_4/"
  colnames(thisGenoPhenoData)
  table(thisGenoPhenoData[,"phenoCat_final"])
  thisGenoPhenoData_subset <- thisGenoPhenoData %>% filter(RUBrosHaps==thisHaplotype)
  thisCol_mean <- mean(thisGenoPhenoData_subset[,thisColMeasure])
  dist_mean <- abs(thisCol_mean-thisGenoPhenoData_subset[,thisColMeasure])
  thisGenoPhenoData_subset[13,thisColMeasure]
  thisGenoPhenoData_ordered <- thisGenoPhenoData_subset[order(dist_mean),]
  thisGenoPhenoData_keep <- thisGenoPhenoData_ordered[1:numPhotos,]
  # thisGenoPhenoData_keep[,"RametIDs"]

  for (thisRow in 1:nrow(thisGenoPhenoData_keep)) {
    # colnames(GenoEcol_Planoles_Core_subset)
    # which(GenoEcol_Planoles_Core_subset[,"PlantID_final"]=="S4795")
    # thisRow <- 1
    thisData <- thisGenoPhenoData_keep[thisRow,]
    thisID <- thisGenoPhenoData_keep[thisRow,"PlantID_final"]
    thisLabel <- unlist(strsplit(thisID, ""))[[1]]
    thisLabelRamets <- thisGenoPhenoData_keep[thisRow,"RametIDs"]
    thisFace <- thisData[,"Face"]
    thisSide <- thisData[,"Side"]
    thisRed <- thisData[,"Red_final"]
    thisYellow <- thisData[,"Yellow_final"]
    thisPheno <- thisData[,"phenoCat_final"]
    thisHapName <- thisData[,"multiLoc_Hap_6by7"]
    thisHapName2 <- thisHaplotype
  
    annotateText <- paste0(thisLabelRamets,",[",thisRed,",",thisYellow,"],",thisHapName,", ",thisHaplotype,", ",thisColMeasure," mean = ",round(thisCol_mean,2),", ",thisColMeasure," = ",round(thisData[,thisColMeasure],2))
    annotateText2 <- thisHaplotype
  
    if (thisPheno!=0 | thisPheno!=-9) {
          if (thisPheno =="FR") {
            borderCol <-"magenta"
          }
          if (thisPheno =="Y") {
            borderCol <-"yellow"
            }
          if (thisPheno=="FO") {
             borderCol <-"orange"
          }
          if (thisPheno=="WR") {
            borderCol <-"pink"
          }
          if (thisPheno=="W") {
            borderCol <-"white"
          }
          if (thisPheno=="WO") {
            borderCol <- "wheat"
          }
        }
    cat("\n plant: ", paste0(thisRow,": ",thisID))
    cat("; ",thisLabelRamets)
    cat(", ",thisHapName)
    cat(", ",thisHapName2)
  
    if (thisLabel %in% c("K","L","M")) {
      
      
    }
    if (thisLabel %in% c("P","S")) {
      if (thisLabel == "P") {
        thisDropBoxSearch <- drop_search(query=paste0(thisFace,".JPG"),path = "/Antirrhinum_fieldData/fieldSeason2013/Data_backups/Flower_Photos", start = 0, max_results = 100)
        if (thisDropBoxSearch$start!=0) {
          cat(", Photo found!")
           thisPhoto_link <- drop_media(thisDropBoxSearch$matches[[1]]$metadata$path_display)$link
           finalImage <- image_read(thisPhoto_link) %>%
                      image_background("blue", flatten = TRUE) %>%
                      image_border(borderCol, "30x30") %>%
                      #image_scale("400") %>%
                      image_annotate(annotateText, color = "white", size = 70,gravity = "southwest")
                      #image_annotate(annotateText2, color = "white", size = 16,location="+50+100")
           plot(finalImage)
           image_write(finalImage,paste0(thisPathSave,thisHapName2,"_",thisID,".jpg"))
        }
        if (thisDropBoxSearch$start==0) {
          thisDropBoxSearch <- drop_search(query=paste0(thisFace,".JPG"),path = "/Antirrhinum_fieldData/fieldSeason2014/Data_backups/Flower_Photos", start = 0, max_results = 100)
          if (thisDropBoxSearch$start!=0) {
            cat(", Photo found!")
            thisPhoto_link <- drop_media(thisDropBoxSearch$matches[[1]]$metadata$path_display)$link
            finalImage <- image_read(thisPhoto_link) %>%
                      image_background("blue", flatten = TRUE) %>%
                      image_border(borderCol, "30x30") %>%
                      image_scale("400") %>%
                      image_annotate(annotateText, color = "white", size = 20,gravity = "southwest")
                      #image_annotate(annotateText2, color = "white", size = 22,location="+50+100")
            plot(finalImage)
          }
        }
      }
      if (thisLabel == "S") {
        thisDropBoxSearch <- drop_search(query=paste0(thisFace,".JPG"),path = "/Antirrhinum_fieldData/fieldSeason2014/Data_backups/Flower_Photos", start = 0, max_results = 100)
        if (thisDropBoxSearch$start!=0) {
           cat(", Photo found!")
           thisPhoto_link <- drop_media(thisDropBoxSearch$matches[[1]]$metadata$path_display)$link
           finalImage <- image_read(thisPhoto_link) %>%
                      image_background("blue", flatten = TRUE) %>%
                      image_border(borderCol, "30x30") %>%
                      #image_scale("400") %>%
                      image_annotate(annotateText, color = "white", size = 70,gravity = "southwest")
                      #image_annotate(annotateText2, color = "white", size = 16,location="+50+100")
           plot(finalImage)
           image_write(finalImage,paste0(thisPathSave,thisHapName2,"_",thisID,".jpg"))
        }
      }
    }
  # will need different code like Haralds or Lenkas to find path to other years
  
  }
  
}
unique(RUBrosHaps)

thisPath_Hue4 <- "/Users/dlfield/Library/CloudStorage/OneDrive-SharedLibraries-EdithCowanUniversity/Snapdragons_2022-03193-FIELD - General/6_DATA_ANALYSIS/poolSeq/version3.5/postAnalyses_R/flowerPhotos/Hue_4/"

findPhotoMeans(thisGenoPhenoData = RubVariants_ALL_ROS_backgrounds,thisColMeasure = "Hue_4",thisHaplotype = "RUB_RUB.ROS_ROS",numPhotos = 6,thisPathSave = thisPath_Hue4)

findPhotoMeans(thisGenoPhenoData = RubVariants_ALL_ROS_backgrounds,thisColMeasure = "Hue_4",thisHaplotype = "RUB_rub.ROS_ROS",numPhotos = 6,thisPathSave = thisPath_Hue4)

findPhotoMeans(thisGenoPhenoData = RubVariants_ALL_ROS_backgrounds,thisColMeasure = "Hue_4",thisHaplotype = "rub_rub.ROS_ROS",numPhotos = 6,thisPathSave = thisPath_Hue4)

findPhotoMeans(thisGenoPhenoData = RubVariants_ALL_ROS_backgrounds,thisColMeasure = "Hue_4",thisHaplotype = "RUB_RUB.ROS_ros",numPhotos = 6,thisPathSave = thisPath_Hue4)

findPhotoMeans(thisGenoPhenoData = RubVariants_ALL_ROS_backgrounds,thisColMeasure = "Hue_4",thisHaplotype = "RUB_rub.ROS_ros",numPhotos = 6,thisPathSave = thisPath_Hue4)

findPhotoMeans(thisGenoPhenoData = RubVariants_ALL_ROS_backgrounds,thisColMeasure = "Hue_4",thisHaplotype = "rub_rub.ROS_ros",numPhotos = 6,thisPathSave = thisPath_Hue4)

findPhotoMeans(thisGenoPhenoData = RubVariants_ALL_ROS_backgrounds,thisColMeasure = "Hue_4",thisHaplotype = "RUB_RUB.ros_ros",numPhotos = 6,thisPathSave = thisPath_Hue4)

findPhotoMeans(thisGenoPhenoData = RubVariants_ALL_ROS_backgrounds,thisColMeasure = "Hue_4",thisHaplotype = "RUB_rub.ros_ros",numPhotos = 6,thisPathSave = thisPath_Hue4)

findPhotoMeans(thisGenoPhenoData = RubVariants_ALL_ROS_backgrounds,thisColMeasure = "Hue_4",thisHaplotype = "rub_rub.ros_ros",numPhotos = 6,thisPathSave = thisPath_Hue4)

```

```{r RUBIA and ROS NQT_Hue box plots}
range(RubVariants_ALL_ROS_backgrounds[,"NQT_Hue_6"])

rub_AllROSbackgrouns_plot_NQT_Hue6 <- ggplot(RubVariants_ALL_ROS_backgrounds, aes(x=s261_720757, y=NQT_Hue_6)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(-4,+4) +
        facet_wrap(~ros_assembly_543443,ncol=3) 
      
rub_AllROSbackgrouns_plot_NQT_Hue5 <- ggplot(RubVariants_ALL_ROS_backgrounds, aes(x=s261_720757, y=NQT_Hue_5)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(-4,+4) +
        facet_wrap(~ros_assembly_543443,ncol=3) 

rub_AllROSbackgrouns_plot_NQT_Hue4 <- ggplot(RubVariants_ALL_ROS_backgrounds, aes(x=s261_720757, y=NQT_Hue_4)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(-4,+4) +
        facet_wrap(~ros_assembly_543443,ncol=3) 

rub_AllROSbackgrouns_plot_NQT_Hue3 <- ggplot(RubVariants_ALL_ROS_backgrounds, aes(x=s261_720757, y=NQT_Hue_3)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(-4,+4) +
        facet_wrap(~ros_assembly_543443,ncol=3) 

rub_AllROSbackgrouns_plot_NQT_Hue2 <- ggplot(RubVariants_ALL_ROS_backgrounds, aes(x=s261_720757, y=NQT_Hue_2)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(-4,+4) +
        facet_wrap(~ros_assembly_543443,ncol=3) 

rub_AllROSbackgrouns_plot_NQT_Hue1 <- ggplot(RubVariants_ALL_ROS_backgrounds, aes(x=s261_720757, y=NQT_Hue_1)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(-4,+4) +
        facet_wrap(~ros_assembly_543443,ncol=3) 


plot_grid(rub_AllROSbackgrouns_plot_NQT_Hue1, rub_AllROSbackgrouns_plot_NQT_Hue2, rub_AllROSbackgrouns_plot_NQT_Hue3, rub_AllROSbackgrouns_plot_NQT_Hue4,rub_AllROSbackgrouns_plot_NQT_Hue5,rub_AllROSbackgrouns_plot_NQT_Hue6,labels = c('A', 'B', 'C', 'D','E','F'), align="hv")
```

```{r RUBIA and controlling for ROSel FLA/sulf haplotype with Hue box plots}

# ros_assembly_543443 Allele 2 is pseudomajus
# s261_720757 Allele 2 is pseudomajus
GenoEcol_nuc_PCA[,"s261_720757"] <- as.factor(GenoEcol_nuc_PCA[,"s261_720757"])
GenoEcol_nuc_PCA[,"multiLoc_Hap_24by7"] <- as.factor(GenoEcol_nuc_PCA[,"multiLoc_Hap_24by7"])

table(GenoEcol_nuc_PCA[,"multiLoc_Hap_24by7"])

# Controlling for ROSel background (no recombinants) and FLA/sulf (FLA recombinants allowed, SULF/sulf allowed) we need:
pseudoParental <- c('SULF.SULF;FLA_r.FLA_r/ROSel.ROSel', 'SULF.sulf;FLA_r.FLA_r/ROSel.ROSel', 'SULF.SULF;FLA_r.FLA_r/ROSel.ROSel','SULF.sulf;FLA.fla/ROSel.ROSel','SULF.sulf;fla.fla/ROSel.ROSel','SULF.SULF;fla.fla/ROSel.ROSel')
# picking out rosEL.rosEL (no rosel_r or ROSel_r), SULF.SULF & SULF.sulf, but agnostic to FLA haplotype
hybrid <- c('SULF.SULF;FLA_r.FLA_r/rosEL.ROSel', 'SULF.sulf;FLA_r.FLA_r/rosEL.ROSel', 'SULF.SULF;FLA_r.FLA_r/rosEL.ROSel','SULF.sulf;FLA.fla/rosEL.ROSel','SULF.sulf;fla.fla/rosEL.ROSel','SULF.SULF;fla.fla/rosEL.ROSel','SULF.sulf;fla.fla/rosEL.ROSel','SULF.SULF;fla.fla/rosEL.ROSel','SULF.SULF;FLA.FLA/rosEL.ROSel','SULF.SULF;FLA_r.fla/rosEL.ROSel','SULF.sulf;FLA_r.fla/rosEL.ROSel')

striatumParental <- c('sulf.sulf;FLA_r.FLA_r/rosEL.rosEL','sulf.sulf;FLA.fla/rosEL.rosEL','sulf.sulf;fla.fla/rosEL.rosEL','sulf.sulf;FLA_r.fla/rosEL.rosEL','sulf.sulf;FLA_r.FLA/rosEL.rosEL','sulf.sulf;FLA.FLA/rosEL.rosEL')

RUB_RUB.ROS_ROS <- GenoEcol_nuc_PCA %>% filter(multiLoc_Hap_24by7%in%pseudoParental & s261_720757==2)
RUB_rub.ROS_ROS <- GenoEcol_nuc_PCA %>% filter(multiLoc_Hap_24by7%in%pseudoParental & s261_720757==1)
rub_rub.ROS_ROS <- GenoEcol_nuc_PCA %>% filter(multiLoc_Hap_24by7%in%pseudoParental & s261_720757==0)
RubVariants_ROS.ROS_background <- rbind(RUB_RUB.ROS_ROS,RUB_rub.ROS_ROS,rub_rub.ROS_ROS)

RUB_RUB.ROS_ros <- GenoEcol_nuc_PCA %>% filter(multiLoc_Hap_24by7%in%hybrid & s261_720757==2)
RUB_rub.ROS_ros <- GenoEcol_nuc_PCA %>% filter(multiLoc_Hap_24by7%in%hybrid & s261_720757==1)
rub_rub.ROS_ros <- GenoEcol_nuc_PCA %>% filter(multiLoc_Hap_24by7%in%hybrid & s261_720757==0)
RubVariants_ROS.ros_background <- rbind(RUB_RUB.ROS_ros,RUB_rub.ROS_ros,rub_rub.ROS_ros)

RUB_RUB.ros_ros <- GenoEcol_nuc_PCA %>% filter(multiLoc_Hap_24by7%in%striatumParental & s261_720757==2)
RUB_rub.ros_ros <- GenoEcol_nuc_PCA %>% filter(multiLoc_Hap_24by7%in%striatumParental & s261_720757==1)
rub_rub.ros_ros <- GenoEcol_nuc_PCA %>% filter(multiLoc_Hap_24by7%in%striatumParental & s261_720757==0)
RubVariants_ros.ros_background <- rbind(RUB_RUB.ros_ros,RUB_rub.ros_ros,rub_rub.ros_ros)


RubVariants_ALL_ROS_backgrounds_haps <- rbind(RubVariants_ROS.ROS_background,RubVariants_ROS.ros_background,RubVariants_ros.ros_background)

rub_AllROSbackgrouns_plot_Hue6_haps <- ggplot(RubVariants_ALL_ROS_backgrounds_haps, aes(x=s261_720757, y=Hue_6)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(-30,60) +
        facet_wrap(~ros_assembly_543443,ncol=3) 
      
rub_AllROSbackgrouns_plot_Hue5_haps <- ggplot(RubVariants_ALL_ROS_backgrounds_haps, aes(x=s261_720757, y=Hue_5)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(-30,60) +
        facet_wrap(~ros_assembly_543443,ncol=3) 

rub_AllROSbackgrouns_plot_Hue4_haps <- ggplot(RubVariants_ALL_ROS_backgrounds_haps, aes(x=s261_720757, y=Hue_4)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(-30,60) +
        facet_wrap(~ros_assembly_543443,ncol=3) 

rub_AllROSbackgrouns_plot_Hue3_haps <- ggplot(RubVariants_ALL_ROS_backgrounds_haps, aes(x=s261_720757, y=Hue_3)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(-30,60) +
        facet_wrap(~ros_assembly_543443,ncol=3) 

rub_AllROSbackgrouns_plot_Hue2_haps <- ggplot(RubVariants_ALL_ROS_backgrounds_haps, aes(x=s261_720757, y=Hue_2)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(-30,60) +
        facet_wrap(~ros_assembly_543443,ncol=3) 

rub_AllROSbackgrouns_plot_Hue1_haps <- ggplot(RubVariants_ALL_ROS_backgrounds_haps, aes(x=s261_720757, y=Hue_1)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        ylim(-30,60) +
        facet_wrap(~ros_assembly_543443,ncol=3) 


plot_grid(rub_AllROSbackgrouns_plot_Hue1_haps, rub_AllROSbackgrouns_plot_Hue2_haps, rub_AllROSbackgrouns_plot_Hue3_haps, rub_AllROSbackgrouns_plot_Hue4_haps,rub_AllROSbackgrouns_plot_Hue5_haps,rub_AllROSbackgrouns_plot_Hue6_haps,labels = c('A', 'B', 'C', 'D','E','F'), align="hv")
```

```{r RUBIA and ROS Int box plots}

rub_AllROSbackgrouns_plot_Int6 <- ggplot(RubVariants_ALL_ROS_backgrounds, aes(x=s261_720757, y=Int_6)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        facet_wrap(~ros_assembly_543443,ncol=3) 
      
rub_AllROSbackgrouns_plot_Int5 <- ggplot(RubVariants_ALL_ROS_backgrounds, aes(x=s261_720757, y=Int_5)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        facet_wrap(~ros_assembly_543443,ncol=3) 

rub_AllROSbackgrouns_plot_Int4 <- ggplot(RubVariants_ALL_ROS_backgrounds, aes(x=s261_720757, y=Int_4)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        facet_wrap(~ros_assembly_543443,ncol=3) 

rub_AllROSbackgrouns_plot_Int3 <- ggplot(RubVariants_ALL_ROS_backgrounds, aes(x=s261_720757, y=Int_3)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        facet_wrap(~ros_assembly_543443,ncol=3) 

rub_AllROSbackgrouns_plot_Int2 <- ggplot(RubVariants_ALL_ROS_backgrounds, aes(x=s261_720757, y=Int_2)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        facet_wrap(~ros_assembly_543443,ncol=3) 

rub_AllROSbackgrouns_plot_Int1 <- ggplot(RubVariants_ALL_ROS_backgrounds, aes(x=s261_720757, y=Int_1)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        facet_wrap(~ros_assembly_543443,ncol=3) 


plot_grid(rub_AllROSbackgrouns_plot_Int1, rub_AllROSbackgrouns_plot_Int2, rub_AllROSbackgrouns_plot_Int3, rub_AllROSbackgrouns_plot_Int4,rub_AllROSbackgrouns_plot_Int5,rub_AllROSbackgrouns_plot_Int6,labels = c('A', 'B', 'C', 'D','E','F'), align="hv")
```

```{r RUBIA and ROS box plots controlling for FLA}

GenoEcol_nuc_PCA[,"s261_720757"] <- as.factor(GenoEcol_nuc_PCA[,"s261_720757"])
GenoEcol_nuc_PCA[,"ros_assembly_543443"] <- as.factor(GenoEcol_nuc_PCA[,"ros_assembly_543443"])

RUB_RUB.ROS_ROS <- GenoEcol_nuc_PCA %>% filter(ros_assembly_543443==2 & s261_720757==2)
RUB_rub.ROS_ROS <- GenoEcol_nuc_PCA %>% filter(ros_assembly_543443==2 & s261_720757==1)
rub_rub.ROS_ROS <- GenoEcol_nuc_PCA %>% filter(ros_assembly_543443==2 & s261_720757==0)
RubVariants_ROS.ROS_background <- rbind(RUB_RUB.ROS_ROS,RUB_rub.ROS_ROS,rub_rub.ROS_ROS)

RUB_RUB.ROS_ros <- GenoEcol_nuc_PCA %>% filter(ros_assembly_543443==1 & s261_720757==2)
RUB_rub.ROS_ros <- GenoEcol_nuc_PCA %>% filter(ros_assembly_543443==1 & s261_720757==1)
rub_rub.ROS_ros <- GenoEcol_nuc_PCA %>% filter(ros_assembly_543443==1 & s261_720757==0)
RubVariants_ROS.ros_background <- rbind(RUB_RUB.ROS_ros,RUB_rub.ROS_ros,rub_rub.ROS_ros)

RUB_RUB.ros_ros <- GenoEcol_nuc_PCA %>% filter(ros_assembly_543443==0 & s261_720757==2)
RUB_rub.ros_ros <- GenoEcol_nuc_PCA %>% filter(ros_assembly_543443==0 & s261_720757==1)
rub_rub.ros_ros <- GenoEcol_nuc_PCA %>% filter(ros_assembly_543443==0 & s261_720757==0)
RubVariants_ros.ros_background <- rbind(RUB_RUB.ros_ros,RUB_rub.ros_ros,rub_rub.ros_ros)

ncol(RubVariants_ROS.ROS_background)
ncol(RubVariants_ROS.ros_background)
ncol(RubVariants_ros.ros_background)

RubVariants_ALL_ROS_backgrounds <- rbind(RubVariants_ROS.ROS_background,RubVariants_ROS.ros_background,RubVariants_ros.ros_background)

rub_summary <- RubVariants_ALL_ROS_backgrounds %>% group_by(s261_720757,ros_assembly_543443) %>% tally()
rub_AllROSbackgrouns_plot_Hue6 <- ggplot(RubVariants_ALL_ROS_backgrounds, aes(x=s261_720757, y=Hue_6)) + 
        geom_boxplot() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        stat_summary(fun.y = median, fun.ymax = length,
               geom = "text", aes(label = ..ymax..), vjust = -1) +
        facet_wrap(~ros_assembly_543443,ncol=3) 
      

```


# Slow cline fitting

Finding the most appropriate transect orientation 

In order to collapse the 2D demes to 1D we need to select a heading for the transect. Different angles through the hybrid zone are trialed during prior to the full cline fitting to determine which has the steepest cline width and thus the best fit. For this I concentrate on two representative markers, a SNP in ROS1 and another in FLAVIA - s316_. 

For this procedure I run my cline fitting program for the symmetrical 5-parameter model (see details below) with a short search for the maximum log likelihood values (-lnL) using simulated annealing (SANN) (burnin = , iterations = ). In my experience, even very short SANN runs with ~1000 iterations cannot be improved much with long runs, although it may be the case that more accurate estimates of the likelihood surface and 95%CI could be acheived. I collect the max -lnL across a range of different gradients and intercepts while keeping the transect centred on ~0.5 allele freqeuncy region (for flower phenotypes) at the main core region. 

I assess transects from direct West heading ($270^o$) to NW ($315^o$) in 10 increments of $4.5^o$ (figure x). We can justify a limited search for the heading of the transect on the fact that plants are largely restricted to a prevailing East to West direction through the valley. Any further shifts of the direction of the transect would begin to significantly distort the position of demes.

starting transect: gradient= -0.22, intercept = 5500. Lets call the centre at x = 13000 metres. So y = 2530. Would like to have 5 increments towards ~$270^o$ (due West) and 5 increments in angle towards ~$315^o$ (NW). Assuming they all pass through this point find the new linear equations through basic trig with y-y1 = m(x-x1).

```{r importing data}
# folder structure file system pedigree
folderLoc <- "/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/Pedigree_DF/"
folderStructure <- read.table(paste0(folderLoc,"/Scripts/R/","folderStructure_MQ.txt"),header=TRUE,fill=FALSE,na.strings = "?")

# Import some additional data
GenoEcol <- read.csv("/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/Pedigree_DF/PostProcessedData/FinalGenoEcol_short_int.csv")
SNPlociListUpdated <- read.csv("/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/Pedigree_DF/RawDataFiles/genetic/Amajus_shortSummarySNPsAdj.csv")

# Pick out Planoles and parapatric pops using Easting and Northing
GenoEcol_Planoles <- GenoEcol %>% 
  dplyr::filter(Easting>410000 & Easting<435000 & Northing>4684000 & Northing<4692000)

PedigreeLoci_strict <- readMyFile(folderStructure,"folder_base_genetic","PedigreeLoci_strict.csv")
PedigreeLoci_strict[,1] <- as.vector(PedigreeLoci_strict[,1])

# KASP markers
# CREMOSA (near UDP) Chr 1 "s1187_290152" 
# Flavia Chr 2 - 43.175, "s316_93292", "s316_257789", 
# SULF Chr 4 "s91_39699", 
# RUBIA Chr 5 (near FLS) "s261_720757"
# ROS EL Chr 6 "ros_assembly_543443","ros_assembly_715015"

# make matrix for poolSeq information

poolLoc <- as.data.frame(matrix(0,7,8))
colnames(poolLoc) <- c("pool","Easting","Northing","XPos_deme","YPos_deme","X_adj","Y_adj","DistAlongTransect")
poolLoc[1:7,1] <- c("DemeKASP_Start","YP4","YP2","YP1","MP2","MP4","MP11")
# new positions 2023
poolLoc[2,2:3] <- c(411635.27,4690296.66) # YP4
poolLoc[3,2:3] <- c(422120.82,4686387.46) # YP2
poolLoc[4,2:3] <- c(421968.27,4686511.65) # YP1
poolLoc[5,2:3] <- c(424440.68,4686082.79) # MP2
poolLoc[6,2:3] <- c(425130.38,4685954.23) # MP4
poolLoc[7,2:3] <- c(431641.97,4686865.38) # MP11
poolLoc[,2] <- as.numeric(as.vector(poolLoc[,2]))
poolLoc[,3] <- as.numeric(as.vector(poolLoc[,3]))
poolLoc[,4] <- as.numeric(as.vector(poolLoc[,4]))
poolLoc[,5] <- as.numeric(as.vector(poolLoc[,5]))
poolLoc[,6] <- as.numeric(as.vector(poolLoc[,6]))
poolLoc[,7] <- as.numeric(as.vector(poolLoc[,7]))


# try plot with the same locus only (KASP ros_assembly_543443) in WGS pools then change to this below
# allChr_Clines_ROSEL[allChr_Clines_ROSEL[,"position"]==52889078,]

#HZ_50mDemes <- DemeSetup2D(HZdata=GenoEcol_Planoles,lociNames="s1187_290152",demeSize=50,minSample=10,removePhenoMiss=F,transectGradient = -0.20, transectIntercept = 4900,  adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",poolSeqPlot=T,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=T,FigName="Chr1",plotCentreLine=T,poolSeqClines=allChr_Clines_Chr6clines,poolSeqStats=poolLoc,colourTable=colourChoice)
# str(HZ_50mDemes)

# positions of poolSeq demes along transect
poolSeqStats <- HZ_50mDemes$ros_assembly_543443$poolSeqStats
```

```{r change working directory}
setwd("/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/slowClines/outputs")

# load workspace
# load("ClinesData.RData")
```

```{r Cline fitting scan optimal transect direction CREMOSA s1187_290152 at 200m - only runs in Terminal R}
# First start up new version of R in terminal
getwd()
setwd("/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/slowClines/outputs")
load("ClinesData.RData")

library(Rmpfr)

# clineFit_modelScan()
# New function 2024. Provide any number of model runs using the input table above lnLLsummary
# Flexible function to scan across either a set of loci, rep number for one locus,  different transect directions for one locus, different iteration lengths, deme sizes or any combination of the above. 
# To run simply enter into table the set of parameters to test.
# E.g 1, if a set of transect directions are to be tested for one locus, simply put the same locus name in all rows, but alter the gradient and intercept values.
# E.g 2, if optimal transect direction already established, then keep the gradient and intercept the same in all rows and instead have different values in rows for locus column or reps.

# setting up model summary table
lnLLsummary <- matrix(0,11,39)
colnames(lnLLsummary) <- c("locus","demeSize","numDemes","rep","iters","burnin","gradient","intercept","centre","width","p0","p1","Fst","maxLL","acceptRate","N_total","delta_p","centre_accept","width_accept","pL_accept","pR_accept","Fst_accept","lnLmax","centre_L95","centre_U95","width_L95","width_U95","pL_L95","pL_U95","pR_L95","pR_U95","Fst_L95","Fst_U95","Ngenes","Xvals","p","pointCol","curveCol","flipCline")

lnLLsummary[,"demeSize"] <- 200
lnLLsummary[,"rep"] <- 1
lnLLsummary[,"iters"] <- 10000
lnLLsummary[,"burnin"] <- 1000
lnLLsummary[,"flipCline"] <- F

# setting up locus and cline model fitting parameters
lnLLsummary[,"locus"] <- "s1187_290152"

# colour for CREMOSA
colourChoice
lnLLsummary[,"pointCol"] <- colourChoice[which(colourChoice[,"Region"]=="CREMOSA"),"pointColour"]
lnLLsummary[,"curveCol"] <- colourChoice[which(colourChoice[,"Region"]=="CREMOSA"),"lineColour"]

# starting best looking by eye
lnLLsummary[6,7:8] <- c(-0.197037,4900)
# shifting South in y increments of 1000m setting x to zero
lnLLsummary[6,7:8] <-  c(((4900-2240)/(0-13500)),4900)
lnLLsummary[7,7:8] <-  c(((3900-2240)/(0-13500)),3900)
lnLLsummary[8,7:8] <-  c(((2900-2240)/(0-13500)),2900)
lnLLsummary[9,7:8] <-  c(((1900-2240)/(0-13500)),1900)
lnLLsummary[10,7:8] <-  c(((900-2240)/(0-13500)),900)
lnLLsummary[11,7:8] <-  c(((-100-2240)/(0-13500)),-100)
# shifting North in y increments of 1000m setting x to zero
lnLLsummary[5,7:8] <-  c(((5900-2240)/(0-13500)),5900)
lnLLsummary[4,7:8] <-  c(((6900-2240)/(0-13500)),6900)
lnLLsummary[3,7:8] <-  c(((7900-2240)/(0-13500)),7900)
lnLLsummary[2,7:8] <-  c(((8900-2240)/(0-13500)),8900)
lnLLsummary[1,7:8] <-  c(((9900-2240)/(0-13500)),9900)
lnLLsummary <- lnLLsummary[1:11,]
lnLLsummary <- as.data.frame(lnLLsummary)

# plot map again but with many different transects
par(mfrow=c(1,1))
par(mar=c(3,3,3,1))
par(oma=c(3,3,3,1))
#ls(HZ_150mDemes)
#names(HZ_150mDemes$s1187_290152)
tempData <- HZ_150mDemes$s1187_290152$DemeMatrix2D_combine_noZero
plot(x=NULL,y=NULL,type='n',
    main="",
    xlim=c(-3000,25000),
    ylim=c(-1000,7400),
    xlab="X (metres)",ylab="Y (metres)")
mtext("East (m)",side=1,cex=1.5,line=2.5)
mtext("North (m)",side=2,cex=1.5,line=2.5)

for (thisCurve in 1:nrow(lnLLsummary)) {
  # thisCurve <- 1
  transectGradient <- as.numeric(as.vector(lnLLsummary[thisCurve,7]))
  transectIntercept <- as.numeric(as.vector(lnLLsummary[thisCurve,8]))
  curve(((transectGradient*x)+(transectIntercept)), from=-3000,
            to=25000, col="darkgray",add = TRUE,type = "l",lwd=2,lty=1)
}

# show the likely best transect in blue
curve(((-0.197037*x)+(4900)), from=-3000,
            to=25000, col="blue",add = TRUE,type = "l",lwd=2,lty=1)

# plotting pie charts
totalN <- (tempData[,"pCount"]+tempData[,"qCount"])/2
keep <- which(totalN>=10)
tempData <- tempData[keep,]
allthisP <- as.numeric(as.vector(tempData[,"pCount"]))
allthisQ <- as.numeric(as.vector(tempData[,"qCount"]))
allSum <- allthisP+allthisQ
allFreq <- allthisP/allSum

if (mean(allFreq[1:4])<0.6) {
      allthisP_temp <- allthisQ
      allthisQ_temp <- allthisP
      allthisQ <- allthisP_temp
      allthisP <- allthisQ_temp
    }
for (thisDeme2D in 1:nrow(tempData)) {
      # thisDeme2D <- 1
      thisXraw <- as.numeric(as.vector(tempData[thisDeme2D,"X_adj"]))
      thisYraw <- as.numeric(as.vector(tempData[thisDeme2D,"Y_adj"]))
      thisP <- as.numeric(allthisP[thisDeme2D])
      thisQ <- as.numeric(allthisQ[thisDeme2D])
      # pie plot size has to be tweaked depending on deme size
      # small pie plots, original was radius 69 for deme size 200
      #     if (sum(thisP,thisQ)>minNum) {
      #       add.pie(z=c(thisP,thisQ), x=thisXmid, y=thisYmid, radius=69,col=c((rgb(255,0,0,255,maxColorValue=255)),(rgb(255,255,0,255,maxColorValue=255))),labels="")
          #     }
          # larger pie plots - 120 works okay for deme size 400
      totalN <- thisP + thisQ
      #if ((totalN < minNum)==F) {
      add.pie(z=c(thisP,thisQ), x=thisXraw, y=thisYraw, radius=40,col=c((rgb(255,0,0,255,maxColorValue=255)),(rgb(255,255,0,255,maxColorValue=255))),labels="")
      #add.pie(z=c(thisP,thisQ), x=thisXraw, y=thisYraw, radius=40,col=c((rgb(255,255,0,255,maxColorValue=255)),(rgb(255,0,0,255,maxColorValue=255))),labels="")
          #}
          #}
}

clineFit_scan_s1187_290152 <- clineFit_modelScan(inData = GenoEcol_Planoles,modelSummary=lnLLsummary,
                                                 thisTrial ="optimalTransect",thisPop="Planoles",dpThresh=0.6,
                                                 removePhenoMiss,clineModel="symm_5p",adjustXY=F,pointsCol,gridAdd,
                                                 plotAllele1D,plotDem,addPie,SNPlociListUpdated,plotAxisLab=F,
                                                 demeNmin=10,demeNmax=60, returnFreq=1,   
                                                 startParams=c(10000,3000,0.01,0.99,15),fileFormat="png",  
                                                 plotMarkDownAllDemes=F,plotMarkDownZoomed=F,plotEastNorthing=T, 
                                                 tracePlots=T,plotClines=T,multipleCurves=F,multiplePoints=F)

plot(clineFit_scan_s1187_290152[,"width"],clineFit_scan_s1187_290152[,"maxLL"],)
clineFit_scan_s1187_290152_maxLL <- clineFit_scan_s1187_290152 %>% slice_min(maxLL, n = 1, with_ties = FALSE)

```

```{r Cline fitting scan optimal transect direction ros_assembly_543443 200m demes- only runs in Terminal R}

# setting up model summary table
lnLLsummary <- matrix(0,11,39)
colnames(lnLLsummary) <- c("locus","demeSize","numDemes","rep","iters","burnin","gradient","intercept","centre","width","p0","p1","Fst","maxLL","acceptRate","N_total","delta_p","centre_accept","width_accept","pL_accept","pR_accept","Fst_accept","lnLmax","centre_L95","centre_U95","width_L95","width_U95","pL_L95","pL_U95","pR_L95","pR_U95","Fst_L95","Fst_U95","Ngenes","Xvals","p","pointCol","curveCol","flipCline")

lnLLsummary[,"demeSize"] <- 200
lnLLsummary[,"rep"] <- 1
lnLLsummary[,"iters"] <- 10000
lnLLsummary[,"burnin"] <- 1000
lnLLsummary[,"flipCline"] <- F

lnLLsummary[,"locus"] <- "ros_assembly_543443"

# colour for ROS
lnLLsummary[,"pointCol"] <- colourChoice[which(colourChoice[,"Region"]=="ROS"),"pointColour"]
lnLLsummary[,"curveCol"] <- colourChoice[which(colourChoice[,"Region"]=="ROS"),"lineColour"]

# starting best looking by eye
lnLLsummary[6,7:8] <- c(-0.197037,4900)
# shifting South in y increments of 1000m setting x to zero
lnLLsummary[6,7:8] <-  c(((4900-2240)/(0-13500)),4900)
lnLLsummary[7,7:8] <-  c(((3900-2240)/(0-13500)),3900)
lnLLsummary[8,7:8] <-  c(((2900-2240)/(0-13500)),2900)
lnLLsummary[9,7:8] <-  c(((1900-2240)/(0-13500)),1900)
lnLLsummary[10,7:8] <-  c(((900-2240)/(0-13500)),900)
lnLLsummary[11,7:8] <-  c(((-100-2240)/(0-13500)),-100)
# shifting North in y increments of 1000m setting x to zero
lnLLsummary[5,7:8] <-  c(((5900-2240)/(0-13500)),5900)
lnLLsummary[4,7:8] <-  c(((6900-2240)/(0-13500)),6900)
lnLLsummary[3,7:8] <-  c(((7900-2240)/(0-13500)),7900)
lnLLsummary[2,7:8] <-  c(((8900-2240)/(0-13500)),8900)
lnLLsummary[1,7:8] <-  c(((9900-2240)/(0-13500)),9900)
lnLLsummary <- lnLLsummary[1:11,]
lnLLsummary <- as.data.frame(lnLLsummary)

# Cline Scan not working properly. Have to run internal loop of function
# clineFit_scan_ros_assembly_543443 <- clineFit_modelScan(inData = GenoEcol_Planoles,modelSummary=lnLLsummary,
#                                                  thisTrial ="optimalTransect",thisPop="Planoles",dpThresh=0.6,
#                                                  removePhenoMiss,clineModel="symm_5p",adjustXY=F,pointsCol,gridAdd,
#                                                  plotAllele1D,plotDem,addPie,SNPlociListUpdated,plotAxisLab=F,
#                                                  demeNmin=10,demeNmax=70, returnFreq=1,   
#                                                  startParams=c(10000,3000,0.1,0.9,15),fileFormat="png",  
#                                                  plotMarkDownAllDemes=F,plotMarkDownZoomed=F,plotEastNorthing=T, 
#                                                  tracePlots=T,plotClines=T)

thisLocus <- "ros_assembly_543443"
rm(modelSummary)
modelSummary <- lnLLsummary
 for (thisRow in 1:nrow(modelSummary)) {
    # thisRow <- 6
    # note, some parameters coming in from modelSummary table in new version 2024
    thisLocus <- as.vector(modelSummary[thisRow,"locus"])
    locusName <- thisLocus
    demeSize <- as.numeric(as.vector(modelSummary[thisRow,"demeSize"]))
    rep <- as.vector(modelSummary[thisRow,"rep"])
    iterations <- as.numeric(as.vector(modelSummary[thisRow,"iters"]))
    burnin <- as.numeric(as.vector(modelSummary[thisRow,"burnin"]))
    thisGradient <- as.numeric(as.vector(modelSummary[thisRow,"gradient"]))
    thisIntercept <- as.numeric(as.vector(modelSummary[thisRow,"intercept"]))
    if (thisRow==1) {
      DemeData <- DemeSetup2D(HZdata=inData,lociNames=thisLocus,demeSize=demeSize,minSample=demeNmin,
                              removePhenoMiss=F,transectGradient = thisGradient, transectIntercept = thisIntercept,  
                              adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), 
                              addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",
                              poolSeqPlot=F,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,
                              plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=T,FigName=paste0("_ModelRow_",thisRow),plotCentreLine=T,
                              poolSeqClines=allChr_Clines_Chr1clines,poolSeqStats=poolLoc,colourTable=colourChoice)
      thisDataLoc <- DemeData[[which(names(DemeData)==thisLocus)]]
    }
    if (thisRow>1) {
      if (thisLocus != modelSummary[thisRow-1,"locus"] | thisGradient!=as.vector(modelSummary[thisRow-1,"gradient"])) {
        rm(DemeData)
        DemeData <- DemeSetup2D(HZdata=inData,lociNames=thisLocus,demeSize=demeSize,minSample=demeNmin,
                                removePhenoMiss=F,transectGradient = thisGradient, transectIntercept = thisIntercept,  
                                adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), 
                                addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",
                                poolSeqPlot=T,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,
                                plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=T,FigName=paste0("_ModelRow_",thisRow),plotCentreLine=T,
                                poolSeqClines=allChr_Clines_Chr1clines,poolSeqStats=poolLoc,colourTable=colourChoice)
        thisDataLoc <- DemeData[[which(names(DemeData)==thisLocus)]]
      }
    }
    
    # NEW approach 2020 and updated again for 2024
    XCordinates <- thisDataLoc[["DemeMatrix2D_DistanceRefLine"]][,2]
    thisLoc_DemeMatrix2D_NoZero <- thisDataLoc[["DemeMatrix2D_combine_noZero"]]
    
    # mininum sample size
    sampleSize <- (thisLoc_DemeMatrix2D_NoZero[,"pCount"]+thisLoc_DemeMatrix2D_NoZero[,"qCount"])/2
    theseDemesGood <- which(sampleSize>=demeNmin)
    TotalSamples <- sum(sampleSize[sampleSize>=demeNmin])
    
    # SANN cline fit
    # gather values for demes
    Xvals <- XCordinates 
    pCount_raw <- thisLoc_DemeMatrix2D_NoZero$pCount
    qCount_raw <- thisLoc_DemeMatrix2D_NoZero$qCount
    Ngenes_raw <- pCount_raw + qCount_raw
    pCount <- pCount_raw
    qCount <- qCount_raw
    Ngenes <- Ngenes_raw
    obsVals <- cbind(pCount,qCount,Ngenes)
    
    # skip loci with weak allele frequency differences
    striatum_p <- mean(thisLoc_DemeMatrix2D_NoZero[1:3,"p"])
    pseudom_p <- mean(thisLoc_DemeMatrix2D_NoZero[(nrow(thisLoc_DemeMatrix2D_NoZero)-3):(nrow(thisLoc_DemeMatrix2D_NoZero)),"p"])
    deltaP <- abs(pseudom_p-striatum_p)
    #if (deltaP<0.6) {
    #  next
    #}
    
    thisLoc_Demes_good <- thisLoc_DemeMatrix2D_NoZero[theseDemesGood,]
    
    # allele frequencies
    alleleFreqs <- thisLoc_Demes_good[,"p"]
    
    # plot raw data
    par(mfrow=c(1,1),mar=c(5,5,4,2),oma=c(2,2,2,2)) # (bottom, left, top, right)
    #chartLabel <- markerLabelPieChart(locusName,SNPlociListUpdated)
    textSec <- paste(paste("demes=",length(theseDemesGood),sep=""),
                     paste("n=",TotalSamples,sep=""),sep=", ")
    plot(XCordinates,alleleFreqs, main=paste(locusName,textSec,sep="\n"), 
         cex.main=0.9,cex.axis=1.1,cex.lab=1.1,ylim=c(0,1), xlim=c(min(XCordinates)-500,max(XCordinates)+500), 
         type="n",pch=21,xlab = "distance (m)",ylab = "allele frequency",lwd=2)
    points(cbind(XCordinates,alleleFreqs),cex=1,pch=21,col='black',bg='grey')
    # gather values for demes
    pCount_raw <- as.numeric(as.vector(thisLoc_Demes_good[,"pCount"]))
    qCount_raw <- as.numeric(as.vector(thisLoc_Demes_good[,"qCount"]))
    Ngenes_raw <- pCount_raw + qCount_raw
    demeN <- (pCount_raw + qCount_raw)/2
    pCount <- pCount_raw
    qCount <- qCount_raw
    Ngenes <- Ngenes_raw
    p_raw <- pCount_raw/Ngenes_raw
    
    # reduce size of exceptionally large N for some demes (subsample) 
    # Note - otherwise computational problem with pochhammer function
    # demeMaxSize <- 90
    if (any(demeN > demeNmax)) {
      largeNDemes <- which(demeN > demeMaxSize)
      for (thisDeme in 1:length(largeNDemes)) {
        # thisDeme <- 2
        thisN <- Ngenes[largeNDemes[thisDeme]]
        thisP <- pCount[largeNDemes[thisDeme]]
        thisQ <- qCount[largeNDemes[thisDeme]]
        subsample <- sample(c(1,0),demeNmax,prob=c((thisP/thisN),(thisQ/thisN)),replace=T)
        pCount[largeNDemes[thisDeme]] <- sum(subsample==1)
        qCount[largeNDemes[thisDeme]] <- sum(subsample==0)
        Ngenes[largeNDemes[thisDeme]] <- sum(subsample==1) + sum(subsample==0)
      }
    }
    
    obsVals <- cbind(pCount,qCount,Ngenes)
    
    # # new adjusted allele frequency
    p_adj <- pCount/Ngenes
    # demeNmax of 50 looks better - highly correlated with original values
    # plot(p_raw,p_adj)

    modelSummary[thisRow,"numDemes"] <- length(pCount)
    modelSummary[thisRow,"N_total"] <- sum(Ngenes)/2
    modelSummary[thisRow,"Xvals"] <- paste(XCordinates,collapse=";")
    modelSummary[thisRow,"Ngenes"] <- paste(Ngenes,collapse=";")
    modelSummary[thisRow,"p"] <- paste(p_adj,collapse=";")
    modelSummary[thisRow,"delta_p"] <- deltaP
    
    # old routine from slowClines_RosEl 2018 Aug (which works)
    cat(c("..Cline fitting.."))
    SANNfit_thisLocus <- SANN_clinefit(fn = logLik_5par_SANN,obsVals, positions=XCordinates, 
                                       initialParam = startParams,scale=0.3,retvals=TRUE,nmax = iterations,
                                       verbose=TRUE,acceptscale = 1.05, rejectscale = (1/1.05), 
                                       retfreq=returnFreq)
    # SANNfit_thisLocus$retvals[which(SANNfit_thisLocus$retvals[,"val"]==max(SANNfit_thisLocus$retvals[,"val"])),]
    
    SANNfit_thisLocus_backup <- SANNfit_thisLocus
    #SANNfit_thisLocus <- SANNfit_thisLocus_backup
    # Burnin remove
    if (burnin>0) {
      SANNfit_thisLocus$retvals <- SANNfit_thisLocus$retvals[(round((burnin/returnFreq),0)):nrow(SANNfit_thisLocus$retvals),]
    }
    # around 50% accept rate good
    # thisRow <- 6
    acceptRate <- round(sum(SANNfit_thisLocus$retvals[,"accept"])/nrow(SANNfit_thisLocus$retvals),3)
    modelSummary[thisRow,"acceptRate"] <- acceptRate
    #modelSummary[thisRow,"locusName"] <- locusName
    # Best estimate
    modelSummary[thisRow,"lnLmax"] <- SANNfit_thisLocus$minimum[1]
    modelSummary[thisRow,"centre"] <- SANNfit_thisLocus$estimate[1]
    modelSummary[thisRow,"width"] <- SANNfit_thisLocus$estimate[2]
    modelSummary[thisRow,"pL"] <- SANNfit_thisLocus$estimate[3]
    modelSummary[thisRow,"pR"] <- SANNfit_thisLocus$estimate[4]
    modelSummary[thisRow,"p0"] <- SANNfit_thisLocus$estimate[3]
    modelSummary[thisRow,"p1"] <- SANNfit_thisLocus$estimate[4]
    modelSummary[thisRow,"Fst"] <- 1/(SANNfit_thisLocus$estimate[5]+1)
    
    # Best estimate from remaining values returned
    # Note, if thinning >1 and burnin >0, you may loose the max lnL in the returned values
    SANNfit_thisLocus_retvals <- as.data.frame(SANNfit_thisLocus$retvals)
    SANNfit_thisLocus_retvals_reject <- SANNfit_thisLocus_retvals[SANNfit_thisLocus_retvals[,"accept"]==0,]
    SANNfit_thisLocus_retvals_accept <- SANNfit_thisLocus_retvals[SANNfit_thisLocus_retvals[,"accept"]==1,]
    modelSummary[thisRow,"centre_accept"] <- SANNfit_thisLocus_retvals_accept[SANNfit_thisLocus_retvals_accept[,"val"]==max(SANNfit_thisLocus_retvals_accept[,"val"]),"p1"]
    modelSummary[thisRow,"width_accept"] <- SANNfit_thisLocus_retvals_accept[SANNfit_thisLocus_retvals_accept[,"val"]==max(SANNfit_thisLocus_retvals_accept[,"val"]),"p2"]
    modelSummary[thisRow,"pL_accept"] <- SANNfit_thisLocus_retvals_accept[SANNfit_thisLocus_retvals_accept[,"val"]==max(SANNfit_thisLocus_retvals_accept[,"val"]),"p3"]
    modelSummary[thisRow,"pR_accept"] <- SANNfit_thisLocus_retvals_accept[SANNfit_thisLocus_retvals_accept[,"val"]==max(SANNfit_thisLocus_retvals_accept[,"val"]),"p4"]
    modelSummary[thisRow,"Fst_accept"] <- (1/(SANNfit_thisLocus_retvals_accept[SANNfit_thisLocus_retvals_accept[,"val"]==max(SANNfit_thisLocus_retvals_accept[,"val"]),"p5"]+1))
    #alleleFreq_Demes <- exportAlleleFreqDemes(DemeData)
    #write.csv(as.data.frame(alleleFreq_Demes),"alleleFreq_Demes.csv",row.names=F)
    
    # trace plots
    tracePlot(SANNfit_thisLocus,locusName,acceptRate)
    
    # Likelihood surfaces and support limits
    modelSum_thisRow <- LL_credibleRegions(SANNfit_thisLocus,locusName=thisLocus,SANNfit_thisLocus_retvals_accept,modelSummary[thisRow,])
    
    # Output trace plots and likelihood surfaces to file
    if (tracePlots==T) {
      # Trace plots
      if (fileFormat=="eps") {
        lastPart_trace <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
        fileTraceName <- paste0(locusName,"_","TracePlots_",thisTrial,"_",lastPart_trace,".eps")
        cairo_ps(fileTraceName,width = 24, height = 9.8, pointsize = 14,antialias="default")
        tracePlot(SANNfit_thisLocus,locusName,acceptRate)
        dev.off()
      }
      if (fileFormat=="png") {
        lastPart_trace <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
        fileTraceName <- paste0(locusName,"_","TracePlots_",thisTrial,"_",lastPart_trace,".png")
        png(fileTraceName,width = 800, height = 600, pointsize = 14)
        tracePlot(SANNfit_thisLocus,locusName,acceptRate)
        dev.off()
      }
      # Likelihood surfaces and support limits
      if (fileFormat=="eps") {
        lastPart <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
        fileLLfitsName <- paste0(locusName,"_","LLfits_",thisTrial,"_",lastPart,".eps")
        cairo_ps(fileLLfitsName,width = 24, height = 9.8, pointsize = 14,antialias="default")
        LL_credible <- LL_credibleRegions(SANNfit_thisLocus,locusName,SANNfit_thisLocus_retvals_accept,modelSummary[thisRow,])
        dev.off()
      }
      if (fileFormat=="png") {
        lastPart <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
        fileLLfitsName <- paste0(locusName,"_","LLfits_",thisTrial,"_",lastPart,".png")
        png(fileLLfitsName,width = 800, height = 600, pointsize = 14,antialias="default")
        LL_credibleRegions(SANNfit_thisLocus,locusName,SANNfit_thisLocus_retvals_accept,modelSummary[thisRow,])
        dev.off()
      }
      # update cline fits to file
      # fileName <- paste(paste("summaryModelsAllloci",thisTrial,sep="_"),".csv",sep="")
      # write.csv(as.data.frame(summaryModelsAllloci),fileName,row.names=F)
    }
    if (plotClines==T) {
      # plot cline
      clinePlot(clineModel="symm_5p",modelSummaryInput=modelSum_thisRow,curveFit=T,cexMain=1.1,cexAxisPlot= 1.85,
                axisLineWdth=2.5,cexPoints=1.8,lwdCurve=4,refLocus=F,refPoints=F,HeaderDetails=T,FigLabel="",multipleCurves=F,
                multiplePoints=F,HaplotypeData=NA)      
      # as pdf
      lastPart_cline <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
      fileClinePlotName <- paste0(locusName,"_","ClinePlot_",thisTrial,"_",lastPart_cline,".pdf")
      pdf(fileClinePlotName,width = 15, height = 15,pointsize=12)
      clinePlot(clineModel="symm_5p",modelSummaryInput=modelSum_thisRow,curveFit=T,cexMain=1.1,cexAxisPlot= 2.1,
                axisLineWdth=3.8,cexPoints=3,lwdCurve=4.3,refLocus=F,refPoints=F,HeaderDetails=T,FigLabel="",HaplotypeData=NA)
      dev.off()
      # as png
      lastPart_cline <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
      fileClinePlotName <- paste0(locusName,"_","ClinePlot_",thisTrial,"_",lastPart_cline,".png")
      png(fileClinePlotName,width = 1000, height = 1000, pointsize = 14,antialias="default")
      clinePlot(clineModel="symm_5p",modelSummaryInput=modelSum_thisRow,curveFit=T,cexMain=1.1,cexAxisPlot= 2.1,
                axisLineWdth=3.8,cexPoints=3,lwdCurve=4.3,refLocus=F,refPoints=F,HeaderDetails=T,FigLabel="",HaplotypeData=NA)
      dev.off()
      obsVals_thisLocus <- NULL
      curveFit <- NULL
    }
    
   modelSum_thisRow[,"N_total"] <- sum(Ngenes)/2
   modelSum_thisRow[,"maxLL"] <- modelSum_thisRow[,"lnLmax"]
   modelSum_thisRow[,"p0"] <- modelSum_thisRow[,"pL"]
   modelSum_thisRow[,"p1"] <- modelSum_thisRow[,"pR"]
   modelSummary[thisRow,]<- modelSum_thisRow
   write.csv(as.data.frame(modelSummary),paste0(thisLocus,"_","modelSummary_",thisTrial,".csv"),row.names=F)
   # write.csv(as.data.frame(modelSummary),"SULF_BradleyetALfreqs.csv",row.names=F)
   
  }

clineFit_scan_ros_assembly_543443 <- modelSummary

plot(clineFit_scan_ros_assembly_543443[,"width"],clineFit_scan_ros_assembly_543443[,"maxLL"])
clineFit_scan_ros_assembly_543443_maxLL <- clineFit_scan_ros_assembly_543443 %>% slice_min(maxLL, n = 1, with_ties = FALSE)
```

```{r Cline fitting scan optimal transect direction FLA1 s316_93292 - only runs in Terminal R}

# setting up model summary table
lnLLsummary <- matrix(0,11,39)
colnames(lnLLsummary) <- c("locus","demeSize","numDemes","rep","iters","burnin","gradient","intercept","centre","width","p0","p1","Fst","maxLL","acceptRate","N_total","delta_p","centre_accept","width_accept","pL_accept","pR_accept","Fst_accept","lnLmax","centre_L95","centre_U95","width_L95","width_U95","pL_L95","pL_U95","pR_L95","pR_U95","Fst_L95","Fst_U95","Ngenes","Xvals","p","pointCol","curveCol","flipCline")

lnLLsummary[,"demeSize"] <- 200
lnLLsummary[,"rep"] <- 1
lnLLsummary[,"iters"] <- 10000
lnLLsummary[,"burnin"] <- 1000
lnLLsummary[,"flipCline"] <- F

lnLLsummary[,"locus"] <- "s316_93292"

# colour for s316_93292
lnLLsummary[,"pointCol"] <- colourChoice[which(colourChoice[,"Region"]=="FLAVIA"),"pointColour"]
lnLLsummary[,"curveCol"] <- colourChoice[which(colourChoice[,"Region"]=="FLAVIA"),"lineColour"]

# starting best looking by eye
lnLLsummary[6,7:8] <- c(-0.197037,4900)
# shifting South in y increments of 1000m setting x to zero
lnLLsummary[6,7:8] <-  c(((4900-2240)/(0-13500)),4900)
lnLLsummary[7,7:8] <-  c(((3900-2240)/(0-13500)),3900)
lnLLsummary[8,7:8] <-  c(((2900-2240)/(0-13500)),2900)
lnLLsummary[9,7:8] <-  c(((1900-2240)/(0-13500)),1900)
lnLLsummary[10,7:8] <-  c(((900-2240)/(0-13500)),900)
lnLLsummary[11,7:8] <-  c(((-100-2240)/(0-13500)),-100)
# shifting North in y increments of 1000m setting x to zero
lnLLsummary[5,7:8] <-  c(((5900-2240)/(0-13500)),5900)
lnLLsummary[4,7:8] <-  c(((6900-2240)/(0-13500)),6900)
lnLLsummary[3,7:8] <-  c(((7900-2240)/(0-13500)),7900)
lnLLsummary[2,7:8] <-  c(((8900-2240)/(0-13500)),8900)
lnLLsummary[1,7:8] <-  c(((9900-2240)/(0-13500)),9900)
lnLLsummary <- lnLLsummary[1:11,]
lnLLsummary <- as.data.frame(lnLLsummary)

# clineFit_scan_s316_93292 <- clineFit_modelScan(inData = GenoEcol_Planoles,modelSummary=lnLLsummary,
#                                                  thisTrial ="optimalTransect",thisPop="Planoles",dpThresh=0.6,
#                                                  removePhenoMiss,clineModel="symm_5p",adjustXY=F,pointsCol,gridAdd,
#                                                  plotAllele1D,plotDem,addPie,SNPlociListUpdated,plotAxisLab=F,
#                                                  demeNmin=10,demeNmax=70, returnFreq=1,   
#                                                  startParams=c(10000,3000,0.1,0.9,15),fileFormat="png",  
#                                                  plotMarkDownAllDemes=F,plotMarkDownZoomed=F,plotEastNorthing=T, 
#                                                  tracePlots=T,plotClines=T)


rm(modelSummary)
thisLocus <- "s316_93292"
modelSummary <- lnLLsummary
 for (thisRow in 1:nrow(modelSummary)) {
    # thisRow <- 6
    # note, some parameters coming in from modelSummary table in new version 2024
    thisLocus <- as.vector(modelSummary[thisRow,"locus"])
    locusName <- thisLocus
    demeSize <- as.numeric(as.vector(modelSummary[thisRow,"demeSize"]))
    rep <- as.vector(modelSummary[thisRow,"rep"])
    iterations <- as.numeric(as.vector(modelSummary[thisRow,"iters"]))
    burnin <- as.numeric(as.vector(modelSummary[thisRow,"burnin"]))
    thisGradient <- as.numeric(as.vector(modelSummary[thisRow,"gradient"]))
    thisIntercept <- as.numeric(as.vector(modelSummary[thisRow,"intercept"]))
    if (thisRow==1) {
      DemeData <- DemeSetup2D(HZdata=inData,lociNames=thisLocus,demeSize=demeSize,minSample=demeNmin,
                              removePhenoMiss=F,transectGradient = thisGradient, transectIntercept = thisIntercept,  
                              adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), 
                              addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",
                              poolSeqPlot=F,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,
                              plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=T,FigName=paste0("_ModelRow_",thisRow),plotCentreLine=T,
                              poolSeqClines=allChr_Clines_Chr1clines,poolSeqStats=poolLoc,colourTable=colourChoice)
      thisDataLoc <- DemeData[[which(names(DemeData)==thisLocus)]]
    }
    if (thisRow>1) {
      if (thisLocus != modelSummary[thisRow-1,"locus"] | thisGradient!=as.vector(modelSummary[thisRow-1,"gradient"])) {
        rm(DemeData)
        DemeData <- DemeSetup2D(HZdata=inData,lociNames=thisLocus,demeSize=demeSize,minSample=demeNmin,
                                removePhenoMiss=F,transectGradient = thisGradient, transectIntercept = thisIntercept,  
                                adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), 
                                addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",
                                poolSeqPlot=T,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,
                                plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=T,FigName=paste0("_ModelRow_",thisRow),plotCentreLine=T,
                                poolSeqClines=allChr_Clines_Chr1clines,poolSeqStats=poolLoc,colourTable=colourChoice)
        thisDataLoc <- DemeData[[which(names(DemeData)==thisLocus)]]
      }
    }
    
    # NEW approach 2020 and updated again for 2024
    XCordinates <- thisDataLoc[["DemeMatrix2D_DistanceRefLine"]][,2]
    thisLoc_DemeMatrix2D_NoZero <- thisDataLoc[["DemeMatrix2D_combine_noZero"]]
    
    # mininum sample size
    sampleSize <- (thisLoc_DemeMatrix2D_NoZero[,"pCount"]+thisLoc_DemeMatrix2D_NoZero[,"qCount"])/2
    theseDemesGood <- which(sampleSize>=demeNmin)
    TotalSamples <- sum(sampleSize[sampleSize>=demeNmin])
    
    # SANN cline fit
    # gather values for demes
    Xvals <- XCordinates 
    pCount_raw <- thisLoc_DemeMatrix2D_NoZero$pCount
    qCount_raw <- thisLoc_DemeMatrix2D_NoZero$qCount
    Ngenes_raw <- pCount_raw + qCount_raw
    pCount <- pCount_raw
    qCount <- qCount_raw
    Ngenes <- Ngenes_raw
    obsVals <- cbind(pCount,qCount,Ngenes)
    
    # skip loci with weak allele frequency differences
    striatum_p <- mean(thisLoc_DemeMatrix2D_NoZero[1:3,"p"])
    pseudom_p <- mean(thisLoc_DemeMatrix2D_NoZero[(nrow(thisLoc_DemeMatrix2D_NoZero)-3):(nrow(thisLoc_DemeMatrix2D_NoZero)),"p"])
    deltaP <- abs(pseudom_p-striatum_p)
    #if (deltaP<0.6) {
    #  next
    #}
    
    thisLoc_Demes_good <- thisLoc_DemeMatrix2D_NoZero[theseDemesGood,]
    
    # allele frequencies
    alleleFreqs <- thisLoc_Demes_good[,"p"]
    
    # plot raw data
    par(mfrow=c(1,1),mar=c(5,5,4,2),oma=c(2,2,2,2)) # (bottom, left, top, right)
    #chartLabel <- markerLabelPieChart(locusName,SNPlociListUpdated)
    textSec <- paste(paste("demes=",length(theseDemesGood),sep=""),
                     paste("n=",TotalSamples,sep=""),sep=", ")
    plot(XCordinates,alleleFreqs, main=paste(locusName,textSec,sep="\n"), 
         cex.main=0.9,cex.axis=1.1,cex.lab=1.1,ylim=c(0,1), xlim=c(min(XCordinates)-500,max(XCordinates)+500), 
         type="n",pch=21,xlab = "distance (m)",ylab = "allele frequency",lwd=2)
    points(cbind(XCordinates,alleleFreqs),cex=1,pch=21,col='black',bg='grey')
    # gather values for demes
    pCount_raw <- as.numeric(as.vector(thisLoc_Demes_good[,"pCount"]))
    qCount_raw <- as.numeric(as.vector(thisLoc_Demes_good[,"qCount"]))
    Ngenes_raw <- pCount_raw + qCount_raw
    demeN <- (pCount_raw + qCount_raw)/2
    pCount <- pCount_raw
    qCount <- qCount_raw
    Ngenes <- Ngenes_raw
    p_raw <- pCount_raw/Ngenes_raw
    
    # reduce size of exceptionally large N for some demes (subsample) 
    # Note - otherwise computational problem with pochhammer function
    # demeMaxSize <- 90
    if (any(demeN > demeNmax)) {
      largeNDemes <- which(demeN > demeMaxSize)
      for (thisDeme in 1:length(largeNDemes)) {
        # thisDeme <- 2
        thisN <- Ngenes[largeNDemes[thisDeme]]
        thisP <- pCount[largeNDemes[thisDeme]]
        thisQ <- qCount[largeNDemes[thisDeme]]
        subsample <- sample(c(1,0),demeNmax,prob=c((thisP/thisN),(thisQ/thisN)),replace=T)
        pCount[largeNDemes[thisDeme]] <- sum(subsample==1)
        qCount[largeNDemes[thisDeme]] <- sum(subsample==0)
        Ngenes[largeNDemes[thisDeme]] <- sum(subsample==1) + sum(subsample==0)
      }
    }
    
    obsVals <- cbind(pCount,qCount,Ngenes)
    
    # # new adjusted allele frequency
    p_adj <- pCount/Ngenes
    # demeNmax of 50 looks better - highly correlated with original values
    # plot(p_raw,p_adj)

    modelSummary[thisRow,"numDemes"] <- length(pCount)
    modelSummary[thisRow,"N_total"] <- sum(Ngenes)/2
    modelSummary[thisRow,"Xvals"] <- paste(XCordinates,collapse=";")
    modelSummary[thisRow,"Ngenes"] <- paste(Ngenes,collapse=";")
    modelSummary[thisRow,"p"] <- paste(p_adj,collapse=";")
    modelSummary[thisRow,"delta_p"] <- deltaP
    
    # old routine from slowClines_RosEl 2018 Aug (which works)
    cat(c("..Cline fitting.."))
    SANNfit_thisLocus <- SANN_clinefit(fn = logLik_5par_SANN,obsVals, positions=XCordinates, 
                                       initialParam = startParams,scale=0.3,retvals=TRUE,nmax = iterations,
                                       verbose=TRUE,acceptscale = 1.05, rejectscale = (1/1.05), 
                                       retfreq=returnFreq)
    # SANNfit_thisLocus$retvals[which(SANNfit_thisLocus$retvals[,"val"]==max(SANNfit_thisLocus$retvals[,"val"])),]
    
    SANNfit_thisLocus_backup <- SANNfit_thisLocus
    #SANNfit_thisLocus <- SANNfit_thisLocus_backup
    # Burnin remove
    if (burnin>0) {
      SANNfit_thisLocus$retvals <- SANNfit_thisLocus$retvals[(round((burnin/returnFreq),0)):nrow(SANNfit_thisLocus$retvals),]
    }
    # around 50% accept rate good
    # thisRow <- 6
    acceptRate <- round(sum(SANNfit_thisLocus$retvals[,"accept"])/nrow(SANNfit_thisLocus$retvals),3)
    modelSummary[thisRow,"acceptRate"] <- acceptRate
    #modelSummary[thisRow,"locusName"] <- locusName
    # Best estimate
    modelSummary[thisRow,"lnLmax"] <- SANNfit_thisLocus$minimum[1]
    modelSummary[thisRow,"centre"] <- SANNfit_thisLocus$estimate[1]
    modelSummary[thisRow,"width"] <- SANNfit_thisLocus$estimate[2]
    modelSummary[thisRow,"pL"] <- SANNfit_thisLocus$estimate[3]
    modelSummary[thisRow,"pR"] <- SANNfit_thisLocus$estimate[4]
    modelSummary[thisRow,"p0"] <- SANNfit_thisLocus$estimate[3]
    modelSummary[thisRow,"p1"] <- SANNfit_thisLocus$estimate[4]
    modelSummary[thisRow,"Fst"] <- 1/(SANNfit_thisLocus$estimate[5]+1)
    
    # Best estimate from remaining values returned
    # Note, if thinning >1 and burnin >0, you may loose the max lnL in the returned values
    SANNfit_thisLocus_retvals <- as.data.frame(SANNfit_thisLocus$retvals)
    SANNfit_thisLocus_retvals_reject <- SANNfit_thisLocus_retvals[SANNfit_thisLocus_retvals[,"accept"]==0,]
    SANNfit_thisLocus_retvals_accept <- SANNfit_thisLocus_retvals[SANNfit_thisLocus_retvals[,"accept"]==1,]
    modelSummary[thisRow,"centre_accept"] <- SANNfit_thisLocus_retvals_accept[SANNfit_thisLocus_retvals_accept[,"val"]==max(SANNfit_thisLocus_retvals_accept[,"val"]),"p1"]
    modelSummary[thisRow,"width_accept"] <- SANNfit_thisLocus_retvals_accept[SANNfit_thisLocus_retvals_accept[,"val"]==max(SANNfit_thisLocus_retvals_accept[,"val"]),"p2"]
    modelSummary[thisRow,"pL_accept"] <- SANNfit_thisLocus_retvals_accept[SANNfit_thisLocus_retvals_accept[,"val"]==max(SANNfit_thisLocus_retvals_accept[,"val"]),"p3"]
    modelSummary[thisRow,"pR_accept"] <- SANNfit_thisLocus_retvals_accept[SANNfit_thisLocus_retvals_accept[,"val"]==max(SANNfit_thisLocus_retvals_accept[,"val"]),"p4"]
    modelSummary[thisRow,"Fst_accept"] <- (1/(SANNfit_thisLocus_retvals_accept[SANNfit_thisLocus_retvals_accept[,"val"]==max(SANNfit_thisLocus_retvals_accept[,"val"]),"p5"]+1))
    #alleleFreq_Demes <- exportAlleleFreqDemes(DemeData)
    #write.csv(as.data.frame(alleleFreq_Demes),"alleleFreq_Demes.csv",row.names=F)
    
    # trace plots
    tracePlot(SANNfit_thisLocus,locusName,acceptRate)
    
    # Likelihood surfaces and support limits
    modelSum_thisRow <- LL_credibleRegions(SANNfit_thisLocus,locusName=thisLocus,SANNfit_thisLocus_retvals_accept,modelSummary[thisRow,])
    
    # Output trace plots and likelihood surfaces to file
    if (tracePlots==T) {
      # Trace plots
      if (fileFormat=="eps") {
        lastPart_trace <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
        fileTraceName <- paste0(locusName,"_","TracePlots_",thisTrial,"_",lastPart_trace,".eps")
        cairo_ps(fileTraceName,width = 24, height = 9.8, pointsize = 14,antialias="default")
        tracePlot(SANNfit_thisLocus,locusName,acceptRate)
        dev.off()
      }
      if (fileFormat=="png") {
        lastPart_trace <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
        fileTraceName <- paste0(locusName,"_","TracePlots_",thisTrial,"_",lastPart_trace,".png")
        png(fileTraceName,width = 800, height = 600, pointsize = 14)
        tracePlot(SANNfit_thisLocus,locusName,acceptRate)
        dev.off()
      }
      # Likelihood surfaces and support limits
      if (fileFormat=="eps") {
        lastPart <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
        fileLLfitsName <- paste0(locusName,"_","LLfits_",thisTrial,"_",lastPart,".eps")
        cairo_ps(fileLLfitsName,width = 24, height = 9.8, pointsize = 14,antialias="default")
        LL_credible <- LL_credibleRegions(SANNfit_thisLocus,locusName,SANNfit_thisLocus_retvals_accept,modelSummary[thisRow,])
        dev.off()
      }
      if (fileFormat=="png") {
        lastPart <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
        fileLLfitsName <- paste0(locusName,"_","LLfits_",thisTrial,"_",lastPart,".png")
        png(fileLLfitsName,width = 800, height = 600, pointsize = 14,antialias="default")
        LL_credibleRegions(SANNfit_thisLocus,locusName,SANNfit_thisLocus_retvals_accept,modelSummary[thisRow,])
        dev.off()
      }
      # update cline fits to file
      # fileName <- paste(paste("summaryModelsAllloci",thisTrial,sep="_"),".csv",sep="")
      # write.csv(as.data.frame(summaryModelsAllloci),fileName,row.names=F)
    }
    if (plotClines==T) {
      # plot cline
      clinePlot(clineModel="symm_5p",modelSummaryInput=modelSum_thisRow,curveFit=T,cexMain=1.1,cexAxisPlot= 1.85,
                axisLineWdth=2.5,cexPoints=1.8,lwdCurve=4,refLocus=F,refPoints=F,HeaderDetails=T,FigLabel="",multipleCurves=F,
                multiplePoints=F,HaplotypeData=NA)      
      # as pdf
      lastPart_cline <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
      fileClinePlotName <- paste0(locusName,"_","ClinePlot_",thisTrial,"_",lastPart_cline,".pdf")
      pdf(fileClinePlotName,width = 15, height = 15,pointsize=12)
      clinePlot(clineModel="symm_5p",modelSummaryInput=modelSum_thisRow,curveFit=T,cexMain=1.1,cexAxisPlot= 2.1,
                axisLineWdth=3.8,cexPoints=3,lwdCurve=4.3,refLocus=F,refPoints=F,HeaderDetails=T,FigLabel="",HaplotypeData=NA)
      dev.off()
      # as png
      lastPart_cline <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
      fileClinePlotName <- paste0(locusName,"_","ClinePlot_",thisTrial,"_",lastPart_cline,".png")
      png(fileClinePlotName,width = 1000, height = 1000, pointsize = 14,antialias="default")
      clinePlot(clineModel="symm_5p",modelSummaryInput=modelSum_thisRow,curveFit=T,cexMain=1.1,cexAxisPlot= 2.1,
                axisLineWdth=3.8,cexPoints=3,lwdCurve=4.3,refLocus=F,refPoints=F,HeaderDetails=T,FigLabel="",HaplotypeData=NA)
      dev.off()
      obsVals_thisLocus <- NULL
      curveFit <- NULL
    }
    
   modelSum_thisRow[,"N_total"] <- sum(Ngenes)/2
   modelSum_thisRow[,"maxLL"] <- modelSum_thisRow[,"lnLmax"]
   modelSum_thisRow[,"p0"] <- modelSum_thisRow[,"pL"]
   modelSum_thisRow[,"p1"] <- modelSum_thisRow[,"pR"]
   modelSummary[thisRow,]<- modelSum_thisRow
   write.csv(as.data.frame(modelSummary),paste0(thisLocus,"_","modelSummary_",thisTrial,".csv"),row.names=F)
   # write.csv(as.data.frame(modelSummary),"SULF_BradleyetALfreqs.csv",row.names=F)
   
  }

clineFit_scan_s316_93292 <- modelSummary

plot(clineFit_scan_s316_93292[,"width"],clineFit_scan_s316_93292[,"maxLL"])
clineFit_scan_s316_93292_maxLL <- clineFit_scan_s316_93292 %>% slice_min(maxLL, n = 1, with_ties = FALSE)
```

```{r Cline fitting scan optimal transect direction FLA2 s316_257789 - only runs in Terminal R}

# setting up model summary table
lnLLsummary <- matrix(0,11,39)
colnames(lnLLsummary) <- c("locus","demeSize","numDemes","rep","iters","burnin","gradient","intercept","centre","width","p0","p1","Fst","maxLL","acceptRate","N_total","delta_p","centre_accept","width_accept","pL_accept","pR_accept","Fst_accept","lnLmax","centre_L95","centre_U95","width_L95","width_U95","pL_L95","pL_U95","pR_L95","pR_U95","Fst_L95","Fst_U95","Ngenes","Xvals","p","pointCol","curveCol","flipCline")

lnLLsummary[,"demeSize"] <- 200
lnLLsummary[,"rep"] <- 1
lnLLsummary[,"iters"] <- 10000
lnLLsummary[,"burnin"] <- 1000
lnLLsummary[,"flipCline"] <- F

lnLLsummary[,"locus"] <- "s316_257789"

# colour for s316_257789
lnLLsummary[,"pointCol"] <- colourChoice[which(colourChoice[,"Region"]=="FLAVIAlinked"),"pointColour"]
lnLLsummary[,"curveCol"] <- colourChoice[which(colourChoice[,"Region"]=="FLAVIAlinked"),"lineColour"]

# starting best looking by eye
lnLLsummary[6,7:8] <- c(-0.197037,4900)
# shifting South in y increments of 1000m setting x to zero
lnLLsummary[6,7:8] <-  c(((4900-2240)/(0-13500)),4900)
lnLLsummary[7,7:8] <-  c(((3900-2240)/(0-13500)),3900)
lnLLsummary[8,7:8] <-  c(((2900-2240)/(0-13500)),2900)
lnLLsummary[9,7:8] <-  c(((1900-2240)/(0-13500)),1900)
lnLLsummary[10,7:8] <-  c(((900-2240)/(0-13500)),900)
lnLLsummary[11,7:8] <-  c(((-100-2240)/(0-13500)),-100)
# shifting North in y increments of 1000m setting x to zero
lnLLsummary[5,7:8] <-  c(((5900-2240)/(0-13500)),5900)
lnLLsummary[4,7:8] <-  c(((6900-2240)/(0-13500)),6900)
lnLLsummary[3,7:8] <-  c(((7900-2240)/(0-13500)),7900)
lnLLsummary[2,7:8] <-  c(((8900-2240)/(0-13500)),8900)
lnLLsummary[1,7:8] <-  c(((9900-2240)/(0-13500)),9900)
lnLLsummary <- lnLLsummary[1:11,]
lnLLsummary <- as.data.frame(lnLLsummary)

# clineFit_scan_s316_257789 <- clineFit_modelScan(inData = GenoEcol_Planoles,modelSummary=lnLLsummary,
#                                                  thisTrial ="optimalTransect",thisPop="Planoles",dpThresh=0.6,
#                                                  removePhenoMiss,clineModel="symm_5p",adjustXY=F,pointsCol,gridAdd,
#                                                  plotAllele1D,plotDem,addPie,SNPlociListUpdated,plotAxisLab=F,
#                                                  demeNmin=10,demeNmax=70, returnFreq=1,   
#                                                  startParams=c(10000,3000,0.1,0.9,15),fileFormat="png",  
#                                                  plotMarkDownAllDemes=F,plotMarkDownZoomed=F,plotEastNorthing=T, 
#                                                  tracePlots=T,plotClines=T)

rm(modelSummary)
thisLocus <- "s316_257789"
modelSummary <- lnLLsummary
 for (thisRow in 1:nrow(modelSummary)) {
    # thisRow <- 6
    # note, some parameters coming in from modelSummary table in new version 2024
    thisLocus <- as.vector(modelSummary[thisRow,"locus"])
    locusName <- thisLocus
    demeSize <- as.numeric(as.vector(modelSummary[thisRow,"demeSize"]))
    rep <- as.vector(modelSummary[thisRow,"rep"])
    iterations <- as.numeric(as.vector(modelSummary[thisRow,"iters"]))
    burnin <- as.numeric(as.vector(modelSummary[thisRow,"burnin"]))
    thisGradient <- as.numeric(as.vector(modelSummary[thisRow,"gradient"]))
    thisIntercept <- as.numeric(as.vector(modelSummary[thisRow,"intercept"]))
    if (thisRow==1) {
      DemeData <- DemeSetup2D(HZdata=inData,lociNames=thisLocus,demeSize=demeSize,minSample=demeNmin,
                              removePhenoMiss=F,transectGradient = thisGradient, transectIntercept = thisIntercept,  
                              adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), 
                              addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",
                              poolSeqPlot=F,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,
                              plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=T,FigName=paste0("_ModelRow_",thisRow),plotCentreLine=T,
                              poolSeqClines=allChr_Clines_Chr1clines,poolSeqStats=poolLoc,colourTable=colourChoice)
      thisDataLoc <- DemeData[[which(names(DemeData)==thisLocus)]]
    }
    if (thisRow>1) {
      if (thisLocus != modelSummary[thisRow-1,"locus"] | thisGradient!=as.vector(modelSummary[thisRow-1,"gradient"])) {
        rm(DemeData)
        DemeData <- DemeSetup2D(HZdata=inData,lociNames=thisLocus,demeSize=demeSize,minSample=demeNmin,
                                removePhenoMiss=F,transectGradient = thisGradient, transectIntercept = thisIntercept,  
                                adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), 
                                addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",
                                poolSeqPlot=T,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,
                                plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=T,FigName=paste0("_ModelRow_",thisRow),plotCentreLine=T,
                                poolSeqClines=allChr_Clines_Chr1clines,poolSeqStats=poolLoc,colourTable=colourChoice)
        thisDataLoc <- DemeData[[which(names(DemeData)==thisLocus)]]
      }
    }
    
    # NEW approach 2020 and updated again for 2024
    XCordinates <- thisDataLoc[["DemeMatrix2D_DistanceRefLine"]][,2]
    thisLoc_DemeMatrix2D_NoZero <- thisDataLoc[["DemeMatrix2D_combine_noZero"]]
    
    # mininum sample size
    sampleSize <- (thisLoc_DemeMatrix2D_NoZero[,"pCount"]+thisLoc_DemeMatrix2D_NoZero[,"qCount"])/2
    theseDemesGood <- which(sampleSize>=demeNmin)
    TotalSamples <- sum(sampleSize[sampleSize>=demeNmin])
    
    # SANN cline fit
    # gather values for demes
    Xvals <- XCordinates 
    pCount_raw <- thisLoc_DemeMatrix2D_NoZero$pCount
    qCount_raw <- thisLoc_DemeMatrix2D_NoZero$qCount
    Ngenes_raw <- pCount_raw + qCount_raw
    pCount <- pCount_raw
    qCount <- qCount_raw
    Ngenes <- Ngenes_raw
    obsVals <- cbind(pCount,qCount,Ngenes)
    
    # skip loci with weak allele frequency differences
    striatum_p <- mean(thisLoc_DemeMatrix2D_NoZero[1:3,"p"])
    pseudom_p <- mean(thisLoc_DemeMatrix2D_NoZero[(nrow(thisLoc_DemeMatrix2D_NoZero)-3):(nrow(thisLoc_DemeMatrix2D_NoZero)),"p"])
    deltaP <- abs(pseudom_p-striatum_p)
    #if (deltaP<0.6) {
    #  next
    #}
    
    thisLoc_Demes_good <- thisLoc_DemeMatrix2D_NoZero[theseDemesGood,]
    
    # allele frequencies
    alleleFreqs <- thisLoc_Demes_good[,"p"]
    
    # plot raw data
    par(mfrow=c(1,1),mar=c(5,5,4,2),oma=c(2,2,2,2)) # (bottom, left, top, right)
    #chartLabel <- markerLabelPieChart(locusName,SNPlociListUpdated)
    textSec <- paste(paste("demes=",length(theseDemesGood),sep=""),
                     paste("n=",TotalSamples,sep=""),sep=", ")
    plot(XCordinates,alleleFreqs, main=paste(locusName,textSec,sep="\n"), 
         cex.main=0.9,cex.axis=1.1,cex.lab=1.1,ylim=c(0,1), xlim=c(min(XCordinates)-500,max(XCordinates)+500), 
         type="n",pch=21,xlab = "distance (m)",ylab = "allele frequency",lwd=2)
    points(cbind(XCordinates,alleleFreqs),cex=1,pch=21,col='black',bg='grey')
    # gather values for demes
    pCount_raw <- as.numeric(as.vector(thisLoc_Demes_good[,"pCount"]))
    qCount_raw <- as.numeric(as.vector(thisLoc_Demes_good[,"qCount"]))
    Ngenes_raw <- pCount_raw + qCount_raw
    demeN <- (pCount_raw + qCount_raw)/2
    pCount <- pCount_raw
    qCount <- qCount_raw
    Ngenes <- Ngenes_raw
    p_raw <- pCount_raw/Ngenes_raw
    
    # reduce size of exceptionally large N for some demes (subsample) 
    # Note - otherwise computational problem with pochhammer function
    # demeMaxSize <- 90
    if (any(demeN > demeNmax)) {
      largeNDemes <- which(demeN > demeMaxSize)
      for (thisDeme in 1:length(largeNDemes)) {
        # thisDeme <- 2
        thisN <- Ngenes[largeNDemes[thisDeme]]
        thisP <- pCount[largeNDemes[thisDeme]]
        thisQ <- qCount[largeNDemes[thisDeme]]
        subsample <- sample(c(1,0),demeNmax,prob=c((thisP/thisN),(thisQ/thisN)),replace=T)
        pCount[largeNDemes[thisDeme]] <- sum(subsample==1)
        qCount[largeNDemes[thisDeme]] <- sum(subsample==0)
        Ngenes[largeNDemes[thisDeme]] <- sum(subsample==1) + sum(subsample==0)
      }
    }
    
    obsVals <- cbind(pCount,qCount,Ngenes)
    
    # # new adjusted allele frequency
    p_adj <- pCount/Ngenes
    # demeNmax of 50 looks better - highly correlated with original values
    # plot(p_raw,p_adj)

    modelSummary[thisRow,"numDemes"] <- length(pCount)
    modelSummary[thisRow,"N_total"] <- sum(Ngenes)/2
    modelSummary[thisRow,"Xvals"] <- paste(XCordinates,collapse=";")
    modelSummary[thisRow,"Ngenes"] <- paste(Ngenes,collapse=";")
    modelSummary[thisRow,"p"] <- paste(p_adj,collapse=";")
    modelSummary[thisRow,"delta_p"] <- deltaP
    
    # old routine from slowClines_RosEl 2018 Aug (which works)
    cat(c("..Cline fitting.."))
    SANNfit_thisLocus <- SANN_clinefit(fn = logLik_5par_SANN,obsVals, positions=XCordinates, 
                                       initialParam = startParams,scale=0.3,retvals=TRUE,nmax = iterations,
                                       verbose=TRUE,acceptscale = 1.05, rejectscale = (1/1.05), 
                                       retfreq=returnFreq)
    # SANNfit_thisLocus$retvals[which(SANNfit_thisLocus$retvals[,"val"]==max(SANNfit_thisLocus$retvals[,"val"])),]
    
    SANNfit_thisLocus_backup <- SANNfit_thisLocus
    #SANNfit_thisLocus <- SANNfit_thisLocus_backup
    # Burnin remove
    if (burnin>0) {
      SANNfit_thisLocus$retvals <- SANNfit_thisLocus$retvals[(round((burnin/returnFreq),0)):nrow(SANNfit_thisLocus$retvals),]
    }
    # around 50% accept rate good
    # thisRow <- 6
    acceptRate <- round(sum(SANNfit_thisLocus$retvals[,"accept"])/nrow(SANNfit_thisLocus$retvals),3)
    modelSummary[thisRow,"acceptRate"] <- acceptRate
    #modelSummary[thisRow,"locusName"] <- locusName
    # Best estimate
    modelSummary[thisRow,"lnLmax"] <- SANNfit_thisLocus$minimum[1]
    modelSummary[thisRow,"centre"] <- SANNfit_thisLocus$estimate[1]
    modelSummary[thisRow,"width"] <- SANNfit_thisLocus$estimate[2]
    modelSummary[thisRow,"pL"] <- SANNfit_thisLocus$estimate[3]
    modelSummary[thisRow,"pR"] <- SANNfit_thisLocus$estimate[4]
    modelSummary[thisRow,"p0"] <- SANNfit_thisLocus$estimate[3]
    modelSummary[thisRow,"p1"] <- SANNfit_thisLocus$estimate[4]
    modelSummary[thisRow,"Fst"] <- 1/(SANNfit_thisLocus$estimate[5]+1)
    
    # Best estimate from remaining values returned
    # Note, if thinning >1 and burnin >0, you may loose the max lnL in the returned values
    SANNfit_thisLocus_retvals <- as.data.frame(SANNfit_thisLocus$retvals)
    SANNfit_thisLocus_retvals_reject <- SANNfit_thisLocus_retvals[SANNfit_thisLocus_retvals[,"accept"]==0,]
    SANNfit_thisLocus_retvals_accept <- SANNfit_thisLocus_retvals[SANNfit_thisLocus_retvals[,"accept"]==1,]
    modelSummary[thisRow,"centre_accept"] <- SANNfit_thisLocus_retvals_accept[SANNfit_thisLocus_retvals_accept[,"val"]==max(SANNfit_thisLocus_retvals_accept[,"val"]),"p1"]
    modelSummary[thisRow,"width_accept"] <- SANNfit_thisLocus_retvals_accept[SANNfit_thisLocus_retvals_accept[,"val"]==max(SANNfit_thisLocus_retvals_accept[,"val"]),"p2"]
    modelSummary[thisRow,"pL_accept"] <- SANNfit_thisLocus_retvals_accept[SANNfit_thisLocus_retvals_accept[,"val"]==max(SANNfit_thisLocus_retvals_accept[,"val"]),"p3"]
    modelSummary[thisRow,"pR_accept"] <- SANNfit_thisLocus_retvals_accept[SANNfit_thisLocus_retvals_accept[,"val"]==max(SANNfit_thisLocus_retvals_accept[,"val"]),"p4"]
    modelSummary[thisRow,"Fst_accept"] <- (1/(SANNfit_thisLocus_retvals_accept[SANNfit_thisLocus_retvals_accept[,"val"]==max(SANNfit_thisLocus_retvals_accept[,"val"]),"p5"]+1))
    #alleleFreq_Demes <- exportAlleleFreqDemes(DemeData)
    #write.csv(as.data.frame(alleleFreq_Demes),"alleleFreq_Demes.csv",row.names=F)
    
    # trace plots
    tracePlot(SANNfit_thisLocus,locusName,acceptRate)
    
    # Likelihood surfaces and support limits
    modelSum_thisRow <- LL_credibleRegions(SANNfit_thisLocus,locusName=thisLocus,SANNfit_thisLocus_retvals_accept,modelSummary[thisRow,])
    
    # Output trace plots and likelihood surfaces to file
    if (tracePlots==T) {
      # Trace plots
      if (fileFormat=="eps") {
        lastPart_trace <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
        fileTraceName <- paste0(locusName,"_","TracePlots_",thisTrial,"_",lastPart_trace,".eps")
        cairo_ps(fileTraceName,width = 24, height = 9.8, pointsize = 14,antialias="default")
        tracePlot(SANNfit_thisLocus,locusName,acceptRate)
        dev.off()
      }
      if (fileFormat=="png") {
        lastPart_trace <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
        fileTraceName <- paste0(locusName,"_","TracePlots_",thisTrial,"_",lastPart_trace,".png")
        png(fileTraceName,width = 800, height = 600, pointsize = 14)
        tracePlot(SANNfit_thisLocus,locusName,acceptRate)
        dev.off()
      }
      # Likelihood surfaces and support limits
      if (fileFormat=="eps") {
        lastPart <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
        fileLLfitsName <- paste0(locusName,"_","LLfits_",thisTrial,"_",lastPart,".eps")
        cairo_ps(fileLLfitsName,width = 24, height = 9.8, pointsize = 14,antialias="default")
        LL_credible <- LL_credibleRegions(SANNfit_thisLocus,locusName,SANNfit_thisLocus_retvals_accept,modelSummary[thisRow,])
        dev.off()
      }
      if (fileFormat=="png") {
        lastPart <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
        fileLLfitsName <- paste0(locusName,"_","LLfits_",thisTrial,"_",lastPart,".png")
        png(fileLLfitsName,width = 800, height = 600, pointsize = 14,antialias="default")
        LL_credibleRegions(SANNfit_thisLocus,locusName,SANNfit_thisLocus_retvals_accept,modelSummary[thisRow,])
        dev.off()
      }
      # update cline fits to file
      # fileName <- paste(paste("summaryModelsAllloci",thisTrial,sep="_"),".csv",sep="")
      # write.csv(as.data.frame(summaryModelsAllloci),fileName,row.names=F)
    }
    if (plotClines==T) {
      # plot cline
      clinePlot(clineModel="symm_5p",modelSummaryInput=modelSum_thisRow,curveFit=T,cexMain=1.1,cexAxisPlot= 1.85,
                axisLineWdth=2.5,cexPoints=1.8,lwdCurve=4,refLocus=F,refPoints=F,HeaderDetails=T,FigLabel="",multipleCurves=F,
                multiplePoints=F,HaplotypeData=NA)      
      # as pdf
      lastPart_cline <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
      fileClinePlotName <- paste0(locusName,"_","ClinePlot_",thisTrial,"_",lastPart_cline,".pdf")
      pdf(fileClinePlotName,width = 15, height = 15,pointsize=12)
      clinePlot(clineModel="symm_5p",modelSummaryInput=modelSum_thisRow,curveFit=T,cexMain=1.1,cexAxisPlot= 2.1,
                axisLineWdth=3.8,cexPoints=3,lwdCurve=4.3,refLocus=F,refPoints=F,HeaderDetails=T,FigLabel="",HaplotypeData=NA)
      dev.off()
      # as png
      lastPart_cline <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
      fileClinePlotName <- paste0(locusName,"_","ClinePlot_",thisTrial,"_",lastPart_cline,".png")
      png(fileClinePlotName,width = 1000, height = 1000, pointsize = 14,antialias="default")
      clinePlot(clineModel="symm_5p",modelSummaryInput=modelSum_thisRow,curveFit=T,cexMain=1.1,cexAxisPlot= 2.1,
                axisLineWdth=3.8,cexPoints=3,lwdCurve=4.3,refLocus=F,refPoints=F,HeaderDetails=T,FigLabel="",HaplotypeData=NA)
      dev.off()
      obsVals_thisLocus <- NULL
      curveFit <- NULL
    }
    
   modelSum_thisRow[,"N_total"] <- sum(Ngenes)/2
   modelSum_thisRow[,"maxLL"] <- modelSum_thisRow[,"lnLmax"]
   modelSum_thisRow[,"p0"] <- modelSum_thisRow[,"pL"]
   modelSum_thisRow[,"p1"] <- modelSum_thisRow[,"pR"]
   modelSummary[thisRow,]<- modelSum_thisRow
   write.csv(as.data.frame(modelSummary),paste0(thisLocus,"_","modelSummary_",thisTrial,".csv"),row.names=F)
   # write.csv(as.data.frame(modelSummary),"SULF_BradleyetALfreqs.csv",row.names=F)
   
  }

clineFit_scan_s316_257789 <- modelSummary


plot(clineFit_scan_s316_257789[,"width"],clineFit_scan_s316_257789[,"maxLL"])
clineFit_scan_s316_257789_maxLL <- clineFit_scan_s316_257789 %>% slice_min(maxLL, n = 1, with_ties = FALSE)
```

```{r Cline fitting scan optimal transect direction SULF 1 s91_39699 - only runs in Terminal R}

# setting up model summary table
lnLLsummary <- matrix(0,11,39)
colnames(lnLLsummary) <- c("locus","demeSize","numDemes","rep","iters","burnin","gradient","intercept","centre","width","p0","p1","Fst","maxLL","acceptRate","N_total","delta_p","centre_accept","width_accept","pL_accept","pR_accept","Fst_accept","lnLmax","centre_L95","centre_U95","width_L95","width_U95","pL_L95","pL_U95","pR_L95","pR_U95","Fst_L95","Fst_U95","Ngenes","Xvals","p","pointCol","curveCol","flipCline")

lnLLsummary[,"demeSize"] <- 200
lnLLsummary[,"rep"] <- 1
lnLLsummary[,"iters"] <- 10000
lnLLsummary[,"burnin"] <- 1000
lnLLsummary[,"flipCline"] <- F

lnLLsummary[,"locus"] <- "s91_39699"

# colour for s91_39699
lnLLsummary[,"pointCol"] <- colourChoice[which(colourChoice[,"Region"]=="SULF"),"pointColour"]
lnLLsummary[,"curveCol"] <- colourChoice[which(colourChoice[,"Region"]=="SULF"),"lineColour"]

# starting best looking by eye
lnLLsummary[6,7:8] <- c(-0.197037,4900)
# shifting South in y increments of 1000m setting x to zero
lnLLsummary[6,7:8] <-  c(((4900-2240)/(0-13500)),4900)
lnLLsummary[7,7:8] <-  c(((3900-2240)/(0-13500)),3900)
lnLLsummary[8,7:8] <-  c(((2900-2240)/(0-13500)),2900)
lnLLsummary[9,7:8] <-  c(((1900-2240)/(0-13500)),1900)
lnLLsummary[10,7:8] <-  c(((900-2240)/(0-13500)),900)
lnLLsummary[11,7:8] <-  c(((-100-2240)/(0-13500)),-100)
# shifting North in y increments of 1000m setting x to zero
lnLLsummary[5,7:8] <-  c(((5900-2240)/(0-13500)),5900)
lnLLsummary[4,7:8] <-  c(((6900-2240)/(0-13500)),6900)
lnLLsummary[3,7:8] <-  c(((7900-2240)/(0-13500)),7900)
lnLLsummary[2,7:8] <-  c(((8900-2240)/(0-13500)),8900)
lnLLsummary[1,7:8] <-  c(((9900-2240)/(0-13500)),9900)
lnLLsummary <- lnLLsummary[1:11,]
lnLLsummary <- as.data.frame(lnLLsummary)

# starting parameters from best fit from Bradley et al
# 13181.99114	943.4209713	0.328065109	0.920328421	0.037661472

# Due to bug in 200m deme with interpreting comparison to poolSeq - the following line has been hashed out in DemeSetup2D_genos:
# poolSeqStats[which(theseRows2D==T),"DistAlongTransect"] <- as.vector(dist(cbind(xvals,yvals),method="euclidean"))

# this bug only impacts this locus, not sure why as yet. 

clineFit_scan_s91_39699 <- clineFit_modelScan(inData = GenoEcol_Planoles,modelSummary=lnLLsummary,
                                                 thisTrial ="optimalTransect",thisPop="Planoles",dpThresh=0.6,
                                                 removePhenoMiss,clineModel="symm_5p",adjustXY=F,pointsCol,gridAdd,
                                                 plotAllele1D,plotDem,addPie,SNPlociListUpdated,plotAxisLab=F,
                                                 demeNmin=10,demeNmax=60, returnFreq=1,   
                                                 startParams=c(13181,943,0.32,0.92,15),fileFormat="png",  
                                                 plotMarkDownAllDemes=F,plotMarkDownZoomed=F,plotEastNorthing=T, 
                                                 tracePlots=T,plotClines=T)
clineFit_scan_s91_39699 <- modelSummary
plot(clineFit_scan_s91_39699[,"width"],clineFit_scan_s91_39699[,"maxLL"])
clineFit_scan_s91_39699_maxLL <- clineFit_scan_s91_39699 %>% slice_min(maxLL, n = 1, with_ties = FALSE)
```

```{r Cline fitting scan optimal transect direction SULF 2 s91_78256 - only runs in Terminal R}

# setting up model summary table
lnLLsummary <- matrix(0,11,39)
colnames(lnLLsummary) <- c("locus","demeSize","numDemes","rep","iters","burnin","gradient","intercept","centre","width","p0","p1","Fst","maxLL","acceptRate","N_total","delta_p","centre_accept","width_accept","pL_accept","pR_accept","Fst_accept","lnLmax","centre_L95","centre_U95","width_L95","width_U95","pL_L95","pL_U95","pR_L95","pR_U95","Fst_L95","Fst_U95","Ngenes","Xvals","p","pointCol","curveCol","flipCline")

lnLLsummary[,"demeSize"] <- 150
lnLLsummary[,"rep"] <- 1
lnLLsummary[,"iters"] <- 10000
lnLLsummary[,"burnin"] <- 1000
lnLLsummary[,"flipCline"] <- F

lnLLsummary[,"locus"] <- "s91_78256"

# colour for s91_78256
lnLLsummary[,"pointCol"] <- colourChoice[which(colourChoice[,"Region"]=="SULF"),"pointColour"]
lnLLsummary[,"curveCol"] <- colourChoice[which(colourChoice[,"Region"]=="SULF"),"lineColour"]

# starting best looking by eye
lnLLsummary[6,7:8] <- c(-0.197037,4900)
# shifting South in y increments of 1000m setting x to zero
lnLLsummary[6,7:8] <-  c(((4900-2240)/(0-13500)),4900)
lnLLsummary[7,7:8] <-  c(((3900-2240)/(0-13500)),3900)
lnLLsummary[8,7:8] <-  c(((2900-2240)/(0-13500)),2900)
lnLLsummary[9,7:8] <-  c(((1900-2240)/(0-13500)),1900)
lnLLsummary[10,7:8] <-  c(((900-2240)/(0-13500)),900)
lnLLsummary[11,7:8] <-  c(((-100-2240)/(0-13500)),-100)
# shifting North in y increments of 1000m setting x to zero
lnLLsummary[5,7:8] <-  c(((5900-2240)/(0-13500)),5900)
lnLLsummary[4,7:8] <-  c(((6900-2240)/(0-13500)),6900)
lnLLsummary[3,7:8] <-  c(((7900-2240)/(0-13500)),7900)
lnLLsummary[2,7:8] <-  c(((8900-2240)/(0-13500)),8900)
lnLLsummary[1,7:8] <-  c(((9900-2240)/(0-13500)),9900)
lnLLsummary <- lnLLsummary[1:11,]
lnLLsummary <- as.data.frame(lnLLsummary)

# clineFit_scan_s91_78256 <- clineFit_modelScan(inData = GenoEcol_Planoles,modelSummary=lnLLsummary,
#                                                  thisTrial ="optimalTransect",thisPop="Planoles",dpThresh=0.6,
#                                                  removePhenoMiss,clineModel="symm_5p",adjustXY=F,pointsCol,gridAdd,
#                                                  plotAllele1D,plotDem,addPie,SNPlociListUpdated,plotAxisLab=F,
#                                                  demeNmin=10,demeNmax=50, returnFreq=1,   
#                                                  startParams=c(10000,3000,0.1,0.9,15),fileFormat="png",  
#                                                  plotMarkDownAllDemes=F,plotMarkDownZoomed=F,plotEastNorthing=T, 
#                                                  tracePlots=T,plotClines=T)

rm(modelSummary)
thisLocus <- "s91_78256"
modelSummary <- lnLLsummary
 for (thisRow in 1:nrow(modelSummary)) {
    # thisRow <- 6
    # note, some parameters coming in from modelSummary table in new version 2024
    thisLocus <- as.vector(modelSummary[thisRow,"locus"])
    locusName <- thisLocus
    demeSize <- as.numeric(as.vector(modelSummary[thisRow,"demeSize"]))
    rep <- as.vector(modelSummary[thisRow,"rep"])
    iterations <- as.numeric(as.vector(modelSummary[thisRow,"iters"]))
    burnin <- as.numeric(as.vector(modelSummary[thisRow,"burnin"]))
    thisGradient <- as.numeric(as.vector(modelSummary[thisRow,"gradient"]))
    thisIntercept <- as.numeric(as.vector(modelSummary[thisRow,"intercept"]))
    if (thisRow==1) {
      DemeData <- DemeSetup2D(HZdata=inData,lociNames=thisLocus,demeSize=demeSize,minSample=demeNmin,
                              removePhenoMiss=F,transectGradient = thisGradient, transectIntercept = thisIntercept,  
                              adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), 
                              addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",
                              poolSeqPlot=F,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,
                              plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=T,FigName=paste0("_ModelRow_",thisRow),plotCentreLine=T,
                              poolSeqClines=allChr_Clines_Chr1clines,poolSeqStats=poolLoc,colourTable=colourChoice)
      thisDataLoc <- DemeData[[which(names(DemeData)==thisLocus)]]
    }
    if (thisRow>1) {
      if (thisLocus != modelSummary[thisRow-1,"locus"] | thisGradient!=as.vector(modelSummary[thisRow-1,"gradient"])) {
        rm(DemeData)
        DemeData <- DemeSetup2D(HZdata=inData,lociNames=thisLocus,demeSize=demeSize,minSample=demeNmin,
                                removePhenoMiss=F,transectGradient = thisGradient, transectIntercept = thisIntercept,  
                                adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), 
                                addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",
                                poolSeqPlot=T,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,
                                plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=T,FigName=paste0("_ModelRow_",thisRow),plotCentreLine=T,
                                poolSeqClines=allChr_Clines_Chr1clines,poolSeqStats=poolLoc,colourTable=colourChoice)
        thisDataLoc <- DemeData[[which(names(DemeData)==thisLocus)]]
      }
    }
    
    # NEW approach 2020 and updated again for 2024
    XCordinates <- thisDataLoc[["DemeMatrix2D_DistanceRefLine"]][,2]
    thisLoc_DemeMatrix2D_NoZero <- thisDataLoc[["DemeMatrix2D_combine_noZero"]]
    
    # mininum sample size
    sampleSize <- (thisLoc_DemeMatrix2D_NoZero[,"pCount"]+thisLoc_DemeMatrix2D_NoZero[,"qCount"])/2
    theseDemesGood <- which(sampleSize>=demeNmin)
    TotalSamples <- sum(sampleSize[sampleSize>=demeNmin])
    
    # SANN cline fit
    # gather values for demes
    Xvals <- XCordinates 
    pCount_raw <- thisLoc_DemeMatrix2D_NoZero$pCount
    qCount_raw <- thisLoc_DemeMatrix2D_NoZero$qCount
    Ngenes_raw <- pCount_raw + qCount_raw
    pCount <- pCount_raw
    qCount <- qCount_raw
    Ngenes <- Ngenes_raw
    obsVals <- cbind(pCount,qCount,Ngenes)
    
    # skip loci with weak allele frequency differences
    striatum_p <- mean(thisLoc_DemeMatrix2D_NoZero[1:3,"p"])
    pseudom_p <- mean(thisLoc_DemeMatrix2D_NoZero[(nrow(thisLoc_DemeMatrix2D_NoZero)-3):(nrow(thisLoc_DemeMatrix2D_NoZero)),"p"])
    deltaP <- abs(pseudom_p-striatum_p)
    #if (deltaP<0.6) {
    #  next
    #}
    
    thisLoc_Demes_good <- thisLoc_DemeMatrix2D_NoZero[theseDemesGood,]
    
    # allele frequencies
    alleleFreqs <- thisLoc_Demes_good[,"p"]
    
    # plot raw data
    par(mfrow=c(1,1),mar=c(5,5,4,2),oma=c(2,2,2,2)) # (bottom, left, top, right)
    #chartLabel <- markerLabelPieChart(locusName,SNPlociListUpdated)
    textSec <- paste(paste("demes=",length(theseDemesGood),sep=""),
                     paste("n=",TotalSamples,sep=""),sep=", ")
    plot(XCordinates,alleleFreqs, main=paste(locusName,textSec,sep="\n"), 
         cex.main=0.9,cex.axis=1.1,cex.lab=1.1,ylim=c(0,1), xlim=c(min(XCordinates)-500,max(XCordinates)+500), 
         type="n",pch=21,xlab = "distance (m)",ylab = "allele frequency",lwd=2)
    points(cbind(XCordinates,alleleFreqs),cex=1,pch=21,col='black',bg='grey')
    # gather values for demes
    pCount_raw <- as.numeric(as.vector(thisLoc_Demes_good[,"pCount"]))
    qCount_raw <- as.numeric(as.vector(thisLoc_Demes_good[,"qCount"]))
    Ngenes_raw <- pCount_raw + qCount_raw
    demeN <- (pCount_raw + qCount_raw)/2
    pCount <- pCount_raw
    qCount <- qCount_raw
    Ngenes <- Ngenes_raw
    p_raw <- pCount_raw/Ngenes_raw
    
    # reduce size of exceptionally large N for some demes (subsample) 
    # Note - otherwise computational problem with pochhammer function
    # demeMaxSize <- 90
    if (any(demeN > demeNmax)) {
      largeNDemes <- which(demeN > demeMaxSize)
      for (thisDeme in 1:length(largeNDemes)) {
        # thisDeme <- 2
        thisN <- Ngenes[largeNDemes[thisDeme]]
        thisP <- pCount[largeNDemes[thisDeme]]
        thisQ <- qCount[largeNDemes[thisDeme]]
        subsample <- sample(c(1,0),demeNmax,prob=c((thisP/thisN),(thisQ/thisN)),replace=T)
        pCount[largeNDemes[thisDeme]] <- sum(subsample==1)
        qCount[largeNDemes[thisDeme]] <- sum(subsample==0)
        Ngenes[largeNDemes[thisDeme]] <- sum(subsample==1) + sum(subsample==0)
      }
    }
    
    obsVals <- cbind(pCount,qCount,Ngenes)
    
    # # new adjusted allele frequency
    p_adj <- pCount/Ngenes
    # demeNmax of 50 looks better - highly correlated with original values
    # plot(p_raw,p_adj)

    modelSummary[thisRow,"numDemes"] <- length(pCount)
    modelSummary[thisRow,"N_total"] <- sum(Ngenes)/2
    modelSummary[thisRow,"Xvals"] <- paste(XCordinates,collapse=";")
    modelSummary[thisRow,"Ngenes"] <- paste(Ngenes,collapse=";")
    modelSummary[thisRow,"p"] <- paste(p_adj,collapse=";")
    modelSummary[thisRow,"delta_p"] <- deltaP
    
    # old routine from slowClines_RosEl 2018 Aug (which works)
    cat(c("..Cline fitting.."))
    SANNfit_thisLocus <- SANN_clinefit(fn = logLik_5par_SANN,obsVals, positions=XCordinates, 
                                       initialParam = startParams,scale=0.3,retvals=TRUE,nmax = iterations,
                                       verbose=TRUE,acceptscale = 1.05, rejectscale = (1/1.05), 
                                       retfreq=returnFreq)
    # SANNfit_thisLocus$retvals[which(SANNfit_thisLocus$retvals[,"val"]==max(SANNfit_thisLocus$retvals[,"val"])),]
    
    SANNfit_thisLocus_backup <- SANNfit_thisLocus
    #SANNfit_thisLocus <- SANNfit_thisLocus_backup
    # Burnin remove
    if (burnin>0) {
      SANNfit_thisLocus$retvals <- SANNfit_thisLocus$retvals[(round((burnin/returnFreq),0)):nrow(SANNfit_thisLocus$retvals),]
    }
    # around 50% accept rate good
    # thisRow <- 6
    acceptRate <- round(sum(SANNfit_thisLocus$retvals[,"accept"])/nrow(SANNfit_thisLocus$retvals),3)
    modelSummary[thisRow,"acceptRate"] <- acceptRate
    #modelSummary[thisRow,"locusName"] <- locusName
    # Best estimate
    modelSummary[thisRow,"lnLmax"] <- SANNfit_thisLocus$minimum[1]
    modelSummary[thisRow,"centre"] <- SANNfit_thisLocus$estimate[1]
    modelSummary[thisRow,"width"] <- SANNfit_thisLocus$estimate[2]
    modelSummary[thisRow,"pL"] <- SANNfit_thisLocus$estimate[3]
    modelSummary[thisRow,"pR"] <- SANNfit_thisLocus$estimate[4]
    modelSummary[thisRow,"p0"] <- SANNfit_thisLocus$estimate[3]
    modelSummary[thisRow,"p1"] <- SANNfit_thisLocus$estimate[4]
    modelSummary[thisRow,"Fst"] <- 1/(SANNfit_thisLocus$estimate[5]+1)
    
    # Best estimate from remaining values returned
    # Note, if thinning >1 and burnin >0, you may loose the max lnL in the returned values
    SANNfit_thisLocus_retvals <- as.data.frame(SANNfit_thisLocus$retvals)
    SANNfit_thisLocus_retvals_reject <- SANNfit_thisLocus_retvals[SANNfit_thisLocus_retvals[,"accept"]==0,]
    SANNfit_thisLocus_retvals_accept <- SANNfit_thisLocus_retvals[SANNfit_thisLocus_retvals[,"accept"]==1,]
    modelSummary[thisRow,"centre_accept"] <- SANNfit_thisLocus_retvals_accept[SANNfit_thisLocus_retvals_accept[,"val"]==max(SANNfit_thisLocus_retvals_accept[,"val"]),"p1"]
    modelSummary[thisRow,"width_accept"] <- SANNfit_thisLocus_retvals_accept[SANNfit_thisLocus_retvals_accept[,"val"]==max(SANNfit_thisLocus_retvals_accept[,"val"]),"p2"]
    modelSummary[thisRow,"pL_accept"] <- SANNfit_thisLocus_retvals_accept[SANNfit_thisLocus_retvals_accept[,"val"]==max(SANNfit_thisLocus_retvals_accept[,"val"]),"p3"]
    modelSummary[thisRow,"pR_accept"] <- SANNfit_thisLocus_retvals_accept[SANNfit_thisLocus_retvals_accept[,"val"]==max(SANNfit_thisLocus_retvals_accept[,"val"]),"p4"]
    modelSummary[thisRow,"Fst_accept"] <- (1/(SANNfit_thisLocus_retvals_accept[SANNfit_thisLocus_retvals_accept[,"val"]==max(SANNfit_thisLocus_retvals_accept[,"val"]),"p5"]+1))
    #alleleFreq_Demes <- exportAlleleFreqDemes(DemeData)
    #write.csv(as.data.frame(alleleFreq_Demes),"alleleFreq_Demes.csv",row.names=F)
    
    # trace plots
    tracePlot(SANNfit_thisLocus,locusName,acceptRate)
    
    # Likelihood surfaces and support limits
    modelSum_thisRow <- LL_credibleRegions(SANNfit_thisLocus,locusName=thisLocus,SANNfit_thisLocus_retvals_accept,modelSummary[thisRow,])
    
    # Output trace plots and likelihood surfaces to file
    if (tracePlots==T) {
      # Trace plots
      if (fileFormat=="eps") {
        lastPart_trace <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
        fileTraceName <- paste0(locusName,"_","TracePlots_",thisTrial,"_",lastPart_trace,".eps")
        cairo_ps(fileTraceName,width = 24, height = 9.8, pointsize = 14,antialias="default")
        tracePlot(SANNfit_thisLocus,locusName,acceptRate)
        dev.off()
      }
      if (fileFormat=="png") {
        lastPart_trace <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
        fileTraceName <- paste0(locusName,"_","TracePlots_",thisTrial,"_",lastPart_trace,".png")
        png(fileTraceName,width = 800, height = 600, pointsize = 14)
        tracePlot(SANNfit_thisLocus,locusName,acceptRate)
        dev.off()
      }
      # Likelihood surfaces and support limits
      if (fileFormat=="eps") {
        lastPart <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
        fileLLfitsName <- paste0(locusName,"_","LLfits_",thisTrial,"_",lastPart,".eps")
        cairo_ps(fileLLfitsName,width = 24, height = 9.8, pointsize = 14,antialias="default")
        LL_credible <- LL_credibleRegions(SANNfit_thisLocus,locusName,SANNfit_thisLocus_retvals_accept,modelSummary[thisRow,])
        dev.off()
      }
      if (fileFormat=="png") {
        lastPart <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
        fileLLfitsName <- paste0(locusName,"_","LLfits_",thisTrial,"_",lastPart,".png")
        png(fileLLfitsName,width = 800, height = 600, pointsize = 14,antialias="default")
        LL_credibleRegions(SANNfit_thisLocus,locusName,SANNfit_thisLocus_retvals_accept,modelSummary[thisRow,])
        dev.off()
      }
      # update cline fits to file
      # fileName <- paste(paste("summaryModelsAllloci",thisTrial,sep="_"),".csv",sep="")
      # write.csv(as.data.frame(summaryModelsAllloci),fileName,row.names=F)
    }
    if (plotClines==T) {
      # plot cline
      clinePlot(clineModel="symm_5p",modelSummaryInput=modelSum_thisRow,curveFit=T,cexMain=1.1,cexAxisPlot= 1.85,
                axisLineWdth=2.5,cexPoints=1.8,lwdCurve=4,refLocus=F,refPoints=F,HeaderDetails=T,FigLabel="",multipleCurves=F,
                multiplePoints=F,HaplotypeData=NA)      
      # as pdf
      lastPart_cline <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
      fileClinePlotName <- paste0(locusName,"_","ClinePlot_",thisTrial,"_",lastPart_cline,".pdf")
      pdf(fileClinePlotName,width = 15, height = 15,pointsize=12)
      clinePlot(clineModel="symm_5p",modelSummaryInput=modelSum_thisRow,curveFit=T,cexMain=1.1,cexAxisPlot= 2.1,
                axisLineWdth=3.8,cexPoints=3,lwdCurve=4.3,refLocus=F,refPoints=F,HeaderDetails=T,FigLabel="",HaplotypeData=NA)
      dev.off()
      # as png
      lastPart_cline <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
      fileClinePlotName <- paste0(locusName,"_","ClinePlot_",thisTrial,"_",lastPart_cline,".png")
      png(fileClinePlotName,width = 1000, height = 1000, pointsize = 14,antialias="default")
      clinePlot(clineModel="symm_5p",modelSummaryInput=modelSum_thisRow,curveFit=T,cexMain=1.1,cexAxisPlot= 2.1,
                axisLineWdth=3.8,cexPoints=3,lwdCurve=4.3,refLocus=F,refPoints=F,HeaderDetails=T,FigLabel="",HaplotypeData=NA)
      dev.off()
      obsVals_thisLocus <- NULL
      curveFit <- NULL
    }
    
   modelSum_thisRow[,"N_total"] <- sum(Ngenes)/2
   modelSum_thisRow[,"maxLL"] <- modelSum_thisRow[,"lnLmax"]
   modelSum_thisRow[,"p0"] <- modelSum_thisRow[,"pL"]
   modelSum_thisRow[,"p1"] <- modelSum_thisRow[,"pR"]
   modelSummary[thisRow,]<- modelSum_thisRow
   write.csv(as.data.frame(modelSummary),paste0(thisLocus,"_","modelSummary_",thisTrial,".csv"),row.names=F)
   # write.csv(as.data.frame(modelSummary),"SULF_BradleyetALfreqs.csv",row.names=F)
   
  }

clineFit_scan_s91_78256 <- modelSummary


plot(clineFit_scan_s91_78256[,"width"],clineFit_scan_s91_78256[,"maxLL"])
clineFit_scan_s91_78256_maxLL <- clineFit_scan_s91_78256 %>% slice_min(maxLL, n = 1, with_ties = FALSE)
```

```{r testing SULF old data fits}

# positions & allele freq from Bradley et al

rawXCordinate <- c("910.2;1820.5;2730.7;13708.2;13751.2;13208.1;13403.5;13598.8;13794.1;21607.3;11145.5;12512.8;13880.1;14075.4;14270.7;14466.1;14661.4;14856.7;15052;11969.8;12165.1;13141.7;13337.1;13923;14118.4;15095;15485.7;10450.1;11622.1;13575.4;14161.3;14356.7;14552;14747.3;14942.7;9711.8;13813.7;14009;15181;15376.3;15571.6;15766.9;16157.6;15809.9;16591.2;16981.9;17220.2;18392.2;18782.8;17458.5;19216.5;19454.8;23752;18911.7;20083.7;20279;22818.3;19345.4;19736")

raw_p_adj <- c("0.227272727272727;0.3;0.25;0.208333333333333;0.229166666666667;0.153846153846154;0.333333333333333;0.192307692307692;0.25;0.939024390243902;0.321428571428571;0.444444444444444;0.289473684210526;0.392857142857143;0.68;0.555555555555556;0.615384615384615;0.952380952380952;0.958333333333333;0.478260869565217;0.4;0.375;0.307692307692308;0.678571428571429;0.596153846153846;0.964285714285714;0.90625;0.333333333333333;0.416666666666667;0.4;0.635416666666667;0.720930232558139;0.75;0.571428571428571;0.833333333333333;0.05;0.352941176470588;0.337837837837838;0.727272727272727;0.916666666666667;0.846153846153846;1;0.791666666666667;0.857142857142857;0.9;0.958333333333333;0.863636363636364;0.921052631578947;0.727272727272727;0.805555555555556;0.975;0.933333333333333;1;1;0.923076923076923;0.944444444444444;1;0.944444444444444;0.866666666666667")

raw_Ngenes <- c("22;20;20;24;48;26;30;26;20;82;28;36;38;28;50;72;26;42;24;92;20;24;26;28;52;28;32;54;24;20;96;86;64;28;24;20;34;74;22;36;26;28;24;28;60;24;44;38;22;36;40;30;36;40;52;54;20;36;30")

XCordinates <- as.numeric(as.vector(unlist(strsplit(as.vector(rawXCordinate),";"))))
p_adj <- as.numeric(as.vector(unlist(strsplit(as.vector(raw_p_adj),";"))))
Ngenes <- as.numeric(as.vector(unlist(strsplit(as.vector(raw_Ngenes),";"))))
pCount <- p_adj*Ngenes
qCount <- (1-p_adj)*Ngenes

# SANN cline fit
  # gather values for demes
positions <- XCordinates 
obsVals <- cbind(pCount,qCount,Ngenes)
initialParam <- c(13181,943,0.32,0.92,15)
initialParam <- c(14334.85759,1105.429436,0.293682582,0.913245607,15)

val <- fn(initialParam,Xvals=positions,obsVals)
# lnLL = -290, so its a worse fit with old data to old cline parameters than new data against old cline parameters
# then go into clineFit_modelScan and run Simulated annealing routine (SANN_clinefit)
# saved output to file SULF_BradleyetALfreqs.csv, but ignore the gradient and intercept values

Sulf_BradleyFit <- read.csv("SULF_BradleyetALfreqs.csv")
Sulf_BradleyFit[,"Xvals"] <- rawXCordinate
Sulf_BradleyFit[,"Ngenes"] <- raw_Ngenes
Sulf_BradleyFit[,"p"] <- raw_p_adj

clinePlot(clineModel="symm_5p",modelSummaryInput=Sulf_BradleyFit[1,],curveFit=T,cexMain=1.4,cexAxisPlot= 2.1,
        axisLineWdth=3.8,cexPoints=3,lwdCurve=4.3,refLocus=F,refPoints=F,HeaderDetails=T,FigLabel="",multiplePoints=F,multipleCurves=F,HaplotypeData=NA)

# option 2 - using new data but forcing cline properties from Bradley to test lnLL
# Double checking numbers in data set
# table(GenoEcol_Planoles[,"s91_39699"])

XCordinates <- as.numeric(as.vector(unlist(strsplit(as.vector(clineFit_scan_s91_39699[1,"Xvals"]),";"))))
p_adj <- as.numeric(as.vector(unlist(strsplit(as.vector(clineFit_scan_s91_39699[1,"p"]),";"))))
Ngenes <- as.numeric(as.vector(unlist(strsplit(as.vector(clineFit_scan_s91_39699[1,"Ngenes"]),";"))))
pCount <- p_adj*Ngenes
qCount <- (1-p_adj)*Ngenes

# SANN cline fit
  # gather values for demes
positions <- XCordinates 
obsVals <- cbind(pCount,qCount,Ngenes)
initialParam <- c(13181,943,0.32,0.92,15)
initialParam <- c(14334.85759,1105.429436,0.293682582,0.913245607,15)

val <- fn(initialParam,Xvals=positions,obsVals)
val
clineFit_scan_s91_39699_adj <- clineFit_scan_s91_39699
clineFit_scan_s91_39699_adj[,""]
clinePlot(clineModel="symm_5p",modelSummaryInput=clineFit_scan_s91_39699[1,],curveFit=T,cexMain=1.4,cexAxisPlot= 2.1,
        axisLineWdth=3.8,cexPoints=3,lwdCurve=4.3,refLocus=F,refPoints=F,HeaderDetails=T,FigLabel="",multiplePoints=F,multipleCurves=F,HaplotypeData=NA)

clinePlot(clineModel="symm_5p",modelSummaryInput=modelSummary[6,],curveFit=T,cexMain=1.4,cexAxisPlot= 2.1,
        axisLineWdth=3.8,cexPoints=3,lwdCurve=4.3,refLocus=F,refPoints=F,HeaderDetails=T,FigLabel="",multiplePoints=F,multipleCurves=F,HaplotypeData=NA)

# lnLL = -267.5622 with new data and old cline parameters from Bradley
# some error in fitting routine, because running model scan for this locus gives ~-500 lnLL across these
# re-running
 SANNfit_thisLocus <- SANN_clinefit(fn = logLik_5par_SANN,obsVals, positions=XCordinates, 
                                       initialParam = startParams,scale=0.3,retvals=TRUE,nmax = iterations,
                                       verbose=TRUE,acceptscale = 1.05, rejectscale = (1/1.05), 
                                       retfreq=returnFreq)

# then go into clineFit_modelScan and run Simulated annealing routine (SANN_clinefit)
# saved output to file SULF_BradleyetALfreqs.csv, but ignore the gradient and intercept values

```

```{r Cline fitting scan optimal transect direction RUBIA s261_720757 - only runs in Terminal R}

# setting up model summary table
lnLLsummary <- matrix(0,11,39)
colnames(lnLLsummary) <- c("locus","demeSize","numDemes","rep","iters","burnin","gradient","intercept","centre","width","p0","p1","Fst","maxLL","acceptRate","N_total","delta_p","centre_accept","width_accept","pL_accept","pR_accept","Fst_accept","lnLmax","centre_L95","centre_U95","width_L95","width_U95","pL_L95","pL_U95","pR_L95","pR_U95","Fst_L95","Fst_U95","Ngenes","Xvals","p","pointCol","curveCol","flipCline")

lnLLsummary[,"demeSize"] <- 200
lnLLsummary[,"rep"] <- 1
lnLLsummary[,"iters"] <- 10000
lnLLsummary[,"burnin"] <- 1000
lnLLsummary[,"flipCline"] <- F

lnLLsummary[,"locus"] <- "s261_720757"

# colour for s261_720757
lnLLsummary[,"pointCol"] <- colourChoice[which(colourChoice[,"Region"]=="RUBIA"),"pointColour"]
lnLLsummary[,"curveCol"] <- colourChoice[which(colourChoice[,"Region"]=="RUBIA"),"lineColour"]

# starting best looking by eye
lnLLsummary[6,7:8] <- c(-0.197037,4900)
# shifting South in y increments of 1000m setting x to zero
lnLLsummary[6,7:8] <-  c(((4900-2240)/(0-13500)),4900)
lnLLsummary[7,7:8] <-  c(((3900-2240)/(0-13500)),3900)
lnLLsummary[8,7:8] <-  c(((2900-2240)/(0-13500)),2900)
lnLLsummary[9,7:8] <-  c(((1900-2240)/(0-13500)),1900)
lnLLsummary[10,7:8] <-  c(((900-2240)/(0-13500)),900)
lnLLsummary[11,7:8] <-  c(((-100-2240)/(0-13500)),-100)
# shifting North in y increments of 1000m setting x to zero
lnLLsummary[5,7:8] <-  c(((5900-2240)/(0-13500)),5900)
lnLLsummary[4,7:8] <-  c(((6900-2240)/(0-13500)),6900)
lnLLsummary[3,7:8] <-  c(((7900-2240)/(0-13500)),7900)
lnLLsummary[2,7:8] <-  c(((8900-2240)/(0-13500)),8900)
lnLLsummary[1,7:8] <-  c(((9900-2240)/(0-13500)),9900)
lnLLsummary <- lnLLsummary[1:11,]
lnLLsummary <- as.data.frame(lnLLsummary)

# clineFit_scan_s261_720757 <- clineFit_modelScan(inData = GenoEcol_Planoles,modelSummary=lnLLsummary,
#                                                  thisTrial ="optimalTransect",thisPop="Planoles",dpThresh=0.6,
#                                                  removePhenoMiss,clineModel="symm_5p",adjustXY=F,pointsCol,gridAdd,
#                                                  plotAllele1D,plotDem,addPie,SNPlociListUpdated,plotAxisLab=F,
#                                                  demeNmin=10,demeNmax=50, returnFreq=1,   
#                                                  startParams=c(10000,3000,0.1,0.9,15),fileFormat="png",  
#                                                  plotMarkDownAllDemes=F,plotMarkDownZoomed=F,plotEastNorthing=T, 
#                                                  tracePlots=T,plotClines=T)

rm(modelSummary)
thisLocus <- "s261_720757"
modelSummary <- lnLLsummary
 for (thisRow in 1:nrow(modelSummary)) {
    # thisRow <- 6
    # note, some parameters coming in from modelSummary table in new version 2024
    thisLocus <- as.vector(modelSummary[thisRow,"locus"])
    locusName <- thisLocus
    demeSize <- as.numeric(as.vector(modelSummary[thisRow,"demeSize"]))
    rep <- as.vector(modelSummary[thisRow,"rep"])
    iterations <- as.numeric(as.vector(modelSummary[thisRow,"iters"]))
    burnin <- as.numeric(as.vector(modelSummary[thisRow,"burnin"]))
    thisGradient <- as.numeric(as.vector(modelSummary[thisRow,"gradient"]))
    thisIntercept <- as.numeric(as.vector(modelSummary[thisRow,"intercept"]))
    if (thisRow==1) {
      DemeData <- DemeSetup2D(HZdata=inData,lociNames=thisLocus,demeSize=demeSize,minSample=demeNmin,
                              removePhenoMiss=F,transectGradient = thisGradient, transectIntercept = thisIntercept,  
                              adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), 
                              addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",
                              poolSeqPlot=F,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,
                              plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=T,FigName=paste0("_ModelRow_",thisRow),plotCentreLine=T,
                              poolSeqClines=allChr_Clines_Chr1clines,poolSeqStats=poolLoc,colourTable=colourChoice)
      thisDataLoc <- DemeData[[which(names(DemeData)==thisLocus)]]
    }
    if (thisRow>1) {
      if (thisLocus != modelSummary[thisRow-1,"locus"] | thisGradient!=as.vector(modelSummary[thisRow-1,"gradient"])) {
        rm(DemeData)
        DemeData <- DemeSetup2D(HZdata=inData,lociNames=thisLocus,demeSize=demeSize,minSample=demeNmin,
                                removePhenoMiss=F,transectGradient = thisGradient, transectIntercept = thisIntercept,  
                                adjustXY=F, pointsCol = F, gridAdd=F,plotAllele1D = T, plotDem=c(3,3), 
                                addPie=T,SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,plotType="pdf",
                                poolSeqPlot=T,plotMap=F,noPlots=F,plotMarkDownAllDemes=F,plotEastNorthing=T,
                                plotMarkDownZoomed=F,plotTerrain=F,plotKASPrep=T,FigName=paste0("_ModelRow_",thisRow),plotCentreLine=T,
                                poolSeqClines=allChr_Clines_Chr1clines,poolSeqStats=poolLoc,colourTable=colourChoice)
        thisDataLoc <- DemeData[[which(names(DemeData)==thisLocus)]]
      }
    }
    
    # NEW approach 2020 and updated again for 2024
    XCordinates <- thisDataLoc[["DemeMatrix2D_DistanceRefLine"]][,2]
    thisLoc_DemeMatrix2D_NoZero <- thisDataLoc[["DemeMatrix2D_combine_noZero"]]
    
    # mininum sample size
    sampleSize <- (thisLoc_DemeMatrix2D_NoZero[,"pCount"]+thisLoc_DemeMatrix2D_NoZero[,"qCount"])/2
    theseDemesGood <- which(sampleSize>=demeNmin)
    TotalSamples <- sum(sampleSize[sampleSize>=demeNmin])
    
    # SANN cline fit
    # gather values for demes
    Xvals <- XCordinates 
    pCount_raw <- thisLoc_DemeMatrix2D_NoZero$pCount
    qCount_raw <- thisLoc_DemeMatrix2D_NoZero$qCount
    Ngenes_raw <- pCount_raw + qCount_raw
    pCount <- pCount_raw
    qCount <- qCount_raw
    Ngenes <- Ngenes_raw
    obsVals <- cbind(pCount,qCount,Ngenes)
    
    # skip loci with weak allele frequency differences
    striatum_p <- mean(thisLoc_DemeMatrix2D_NoZero[1:3,"p"])
    pseudom_p <- mean(thisLoc_DemeMatrix2D_NoZero[(nrow(thisLoc_DemeMatrix2D_NoZero)-3):(nrow(thisLoc_DemeMatrix2D_NoZero)),"p"])
    deltaP <- abs(pseudom_p-striatum_p)
    #if (deltaP<0.6) {
    #  next
    #}
    
    thisLoc_Demes_good <- thisLoc_DemeMatrix2D_NoZero[theseDemesGood,]
    
    # allele frequencies
    alleleFreqs <- thisLoc_Demes_good[,"p"]
    
    # plot raw data
    par(mfrow=c(1,1),mar=c(5,5,4,2),oma=c(2,2,2,2)) # (bottom, left, top, right)
    #chartLabel <- markerLabelPieChart(locusName,SNPlociListUpdated)
    textSec <- paste(paste("demes=",length(theseDemesGood),sep=""),
                     paste("n=",TotalSamples,sep=""),sep=", ")
    plot(XCordinates,alleleFreqs, main=paste(locusName,textSec,sep="\n"), 
         cex.main=0.9,cex.axis=1.1,cex.lab=1.1,ylim=c(0,1), xlim=c(min(XCordinates)-500,max(XCordinates)+500), 
         type="n",pch=21,xlab = "distance (m)",ylab = "allele frequency",lwd=2)
    points(cbind(XCordinates,alleleFreqs),cex=1,pch=21,col='black',bg='grey')
    # gather values for demes
    pCount_raw <- as.numeric(as.vector(thisLoc_Demes_good[,"pCount"]))
    qCount_raw <- as.numeric(as.vector(thisLoc_Demes_good[,"qCount"]))
    Ngenes_raw <- pCount_raw + qCount_raw
    demeN <- (pCount_raw + qCount_raw)/2
    pCount <- pCount_raw
    qCount <- qCount_raw
    Ngenes <- Ngenes_raw
    p_raw <- pCount_raw/Ngenes_raw
    
    # reduce size of exceptionally large N for some demes (subsample) 
    # Note - otherwise computational problem with pochhammer function
    # demeMaxSize <- 90
    if (any(demeN > demeNmax)) {
      largeNDemes <- which(demeN > demeMaxSize)
      for (thisDeme in 1:length(largeNDemes)) {
        # thisDeme <- 2
        thisN <- Ngenes[largeNDemes[thisDeme]]
        thisP <- pCount[largeNDemes[thisDeme]]
        thisQ <- qCount[largeNDemes[thisDeme]]
        subsample <- sample(c(1,0),demeNmax,prob=c((thisP/thisN),(thisQ/thisN)),replace=T)
        pCount[largeNDemes[thisDeme]] <- sum(subsample==1)
        qCount[largeNDemes[thisDeme]] <- sum(subsample==0)
        Ngenes[largeNDemes[thisDeme]] <- sum(subsample==1) + sum(subsample==0)
      }
    }
    
    obsVals <- cbind(pCount,qCount,Ngenes)
    
    # # new adjusted allele frequency
    p_adj <- pCount/Ngenes
    # demeNmax of 50 looks better - highly correlated with original values
    # plot(p_raw,p_adj)

    modelSummary[thisRow,"numDemes"] <- length(pCount)
    modelSummary[thisRow,"N_total"] <- sum(Ngenes)/2
    modelSummary[thisRow,"Xvals"] <- paste(XCordinates,collapse=";")
    modelSummary[thisRow,"Ngenes"] <- paste(Ngenes,collapse=";")
    modelSummary[thisRow,"p"] <- paste(p_adj,collapse=";")
    modelSummary[thisRow,"delta_p"] <- deltaP
    
    # old routine from slowClines_RosEl 2018 Aug (which works)
    cat(c("..Cline fitting.."))
    SANNfit_thisLocus <- SANN_clinefit(fn = logLik_5par_SANN,obsVals, positions=XCordinates, 
                                       initialParam = startParams,scale=0.3,retvals=TRUE,nmax = iterations,
                                       verbose=TRUE,acceptscale = 1.05, rejectscale = (1/1.05), 
                                       retfreq=returnFreq)
    # SANNfit_thisLocus$retvals[which(SANNfit_thisLocus$retvals[,"val"]==max(SANNfit_thisLocus$retvals[,"val"])),]
    
    SANNfit_thisLocus_backup <- SANNfit_thisLocus
    #SANNfit_thisLocus <- SANNfit_thisLocus_backup
    # Burnin remove
    if (burnin>0) {
      SANNfit_thisLocus$retvals <- SANNfit_thisLocus$retvals[(round((burnin/returnFreq),0)):nrow(SANNfit_thisLocus$retvals),]
    }
    # around 50% accept rate good
    # thisRow <- 6
    acceptRate <- round(sum(SANNfit_thisLocus$retvals[,"accept"])/nrow(SANNfit_thisLocus$retvals),3)
    modelSummary[thisRow,"acceptRate"] <- acceptRate
    #modelSummary[thisRow,"locusName"] <- locusName
    # Best estimate
    modelSummary[thisRow,"lnLmax"] <- SANNfit_thisLocus$minimum[1]
    modelSummary[thisRow,"centre"] <- SANNfit_thisLocus$estimate[1]
    modelSummary[thisRow,"width"] <- SANNfit_thisLocus$estimate[2]
    modelSummary[thisRow,"pL"] <- SANNfit_thisLocus$estimate[3]
    modelSummary[thisRow,"pR"] <- SANNfit_thisLocus$estimate[4]
    modelSummary[thisRow,"p0"] <- SANNfit_thisLocus$estimate[3]
    modelSummary[thisRow,"p1"] <- SANNfit_thisLocus$estimate[4]
    modelSummary[thisRow,"Fst"] <- 1/(SANNfit_thisLocus$estimate[5]+1)
    
    # Best estimate from remaining values returned
    # Note, if thinning >1 and burnin >0, you may loose the max lnL in the returned values
    SANNfit_thisLocus_retvals <- as.data.frame(SANNfit_thisLocus$retvals)
    SANNfit_thisLocus_retvals_reject <- SANNfit_thisLocus_retvals[SANNfit_thisLocus_retvals[,"accept"]==0,]
    SANNfit_thisLocus_retvals_accept <- SANNfit_thisLocus_retvals[SANNfit_thisLocus_retvals[,"accept"]==1,]
    modelSummary[thisRow,"centre_accept"] <- SANNfit_thisLocus_retvals_accept[SANNfit_thisLocus_retvals_accept[,"val"]==max(SANNfit_thisLocus_retvals_accept[,"val"]),"p1"]
    modelSummary[thisRow,"width_accept"] <- SANNfit_thisLocus_retvals_accept[SANNfit_thisLocus_retvals_accept[,"val"]==max(SANNfit_thisLocus_retvals_accept[,"val"]),"p2"]
    modelSummary[thisRow,"pL_accept"] <- SANNfit_thisLocus_retvals_accept[SANNfit_thisLocus_retvals_accept[,"val"]==max(SANNfit_thisLocus_retvals_accept[,"val"]),"p3"]
    modelSummary[thisRow,"pR_accept"] <- SANNfit_thisLocus_retvals_accept[SANNfit_thisLocus_retvals_accept[,"val"]==max(SANNfit_thisLocus_retvals_accept[,"val"]),"p4"]
    modelSummary[thisRow,"Fst_accept"] <- (1/(SANNfit_thisLocus_retvals_accept[SANNfit_thisLocus_retvals_accept[,"val"]==max(SANNfit_thisLocus_retvals_accept[,"val"]),"p5"]+1))
    #alleleFreq_Demes <- exportAlleleFreqDemes(DemeData)
    #write.csv(as.data.frame(alleleFreq_Demes),"alleleFreq_Demes.csv",row.names=F)
    
    # trace plots
    tracePlot(SANNfit_thisLocus,locusName,acceptRate)
    
    # Likelihood surfaces and support limits
    modelSum_thisRow <- LL_credibleRegions(SANNfit_thisLocus,locusName=thisLocus,SANNfit_thisLocus_retvals_accept,modelSummary[thisRow,])
    
    # Output trace plots and likelihood surfaces to file
    if (tracePlots==T) {
      # Trace plots
      if (fileFormat=="eps") {
        lastPart_trace <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
        fileTraceName <- paste0(locusName,"_","TracePlots_",thisTrial,"_",lastPart_trace,".eps")
        cairo_ps(fileTraceName,width = 24, height = 9.8, pointsize = 14,antialias="default")
        tracePlot(SANNfit_thisLocus,locusName,acceptRate)
        dev.off()
      }
      if (fileFormat=="png") {
        lastPart_trace <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
        fileTraceName <- paste0(locusName,"_","TracePlots_",thisTrial,"_",lastPart_trace,".png")
        png(fileTraceName,width = 800, height = 600, pointsize = 14)
        tracePlot(SANNfit_thisLocus,locusName,acceptRate)
        dev.off()
      }
      # Likelihood surfaces and support limits
      if (fileFormat=="eps") {
        lastPart <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
        fileLLfitsName <- paste0(locusName,"_","LLfits_",thisTrial,"_",lastPart,".eps")
        cairo_ps(fileLLfitsName,width = 24, height = 9.8, pointsize = 14,antialias="default")
        LL_credible <- LL_credibleRegions(SANNfit_thisLocus,locusName,SANNfit_thisLocus_retvals_accept,modelSummary[thisRow,])
        dev.off()
      }
      if (fileFormat=="png") {
        lastPart <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
        fileLLfitsName <- paste0(locusName,"_","LLfits_",thisTrial,"_",lastPart,".png")
        png(fileLLfitsName,width = 800, height = 600, pointsize = 14,antialias="default")
        LL_credibleRegions(SANNfit_thisLocus,locusName,SANNfit_thisLocus_retvals_accept,modelSummary[thisRow,])
        dev.off()
      }
      # update cline fits to file
      # fileName <- paste(paste("summaryModelsAllloci",thisTrial,sep="_"),".csv",sep="")
      # write.csv(as.data.frame(summaryModelsAllloci),fileName,row.names=F)
    }
    if (plotClines==T) {
      # plot cline
      clinePlot(clineModel="symm_5p",modelSummaryInput=modelSum_thisRow,curveFit=T,cexMain=1.1,cexAxisPlot= 1.85,
                axisLineWdth=2.5,cexPoints=1.8,lwdCurve=4,refLocus=F,refPoints=F,HeaderDetails=T,FigLabel="",multipleCurves=F,
                multiplePoints=F,HaplotypeData=NA)      
      # as pdf
      lastPart_cline <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
      fileClinePlotName <- paste0(locusName,"_","ClinePlot_",thisTrial,"_",lastPart_cline,".pdf")
      pdf(fileClinePlotName,width = 15, height = 15,pointsize=12)
      clinePlot(clineModel="symm_5p",modelSummaryInput=modelSum_thisRow,curveFit=T,cexMain=1.1,cexAxisPlot= 2.1,
                axisLineWdth=3.8,cexPoints=3,lwdCurve=4.3,refLocus=F,refPoints=F,HeaderDetails=T,FigLabel="",HaplotypeData=NA)
      dev.off()
      # as png
      lastPart_cline <- paste0("rep_",rep,"_gradient_",round(thisGradient,3),"_intercept_",thisIntercept)
      fileClinePlotName <- paste0(locusName,"_","ClinePlot_",thisTrial,"_",lastPart_cline,".png")
      png(fileClinePlotName,width = 1000, height = 1000, pointsize = 14,antialias="default")
      clinePlot(clineModel="symm_5p",modelSummaryInput=modelSum_thisRow,curveFit=T,cexMain=1.1,cexAxisPlot= 2.1,
                axisLineWdth=3.8,cexPoints=3,lwdCurve=4.3,refLocus=F,refPoints=F,HeaderDetails=T,FigLabel="",HaplotypeData=NA)
      dev.off()
      obsVals_thisLocus <- NULL
      curveFit <- NULL
    }
    
   modelSum_thisRow[,"N_total"] <- sum(Ngenes)/2
   modelSum_thisRow[,"maxLL"] <- modelSum_thisRow[,"lnLmax"]
   modelSum_thisRow[,"p0"] <- modelSum_thisRow[,"pL"]
   modelSum_thisRow[,"p1"] <- modelSum_thisRow[,"pR"]
   modelSummary[thisRow,]<- modelSum_thisRow
   write.csv(as.data.frame(modelSummary),paste0(thisLocus,"_","modelSummary_",thisTrial,".csv"),row.names=F)
   # write.csv(as.data.frame(modelSummary),"SULF_BradleyetALfreqs.csv",row.names=F)
   
  }

clineFit_scan_s261_720757 <- modelSummary


plot(clineFit_scan_s261_720757[,"width"],clineFit_scan_s261_720757[,"maxLL"])
clineFit_scan_s261_720757_maxLL <- clineFit_scan_s261_720757 %>% slice_min(maxLL, n = 1, with_ties = FALSE)
```

```{r save progress}
setwd("/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/slowClines/outputs")
#save.image("ClinesOptimalTransect.RData")


```

```{r for 200m demes re-import optimal transect directions for all loci}
# re-import optimal transect directions for each locus if needed
load("/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/slowClines/outputs/150mDemes/ClinesOptimalTransect.RData")

# OR can load from csv output files generated
clineFit_scan_s1187_290152 <- read.csv("/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/slowClines/outputs/200mDemes/s1187_290152_modelSummary_OptimalTransect.csv")

clineFit_scan_s316_93292 <- read.csv("/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/slowClines/outputs/200mDemes/s316_93292_modelSummary_optimalTransect.csv")

clineFit_scan_s316_257789 <- read.csv("/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/slowClines/outputs/200mDemes/s316_257789_modelSummary_OptimalTransect.csv")

clineFit_scan_s91_39699 <- read.csv("/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/slowClines/outputs/200mDemes/s91_39699_modelSummary_optimalTransect.csv")

clineFit_scan_s91_78256 <- read.csv("/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/slowClines/outputs/200mDemes/s91_78256_modelSummary_OptimalTransect.csv")

clineFit_scan_ros_assembly_543443 <- read.csv("/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/slowClines/outputs/200mDemes/ros_assembly_543443_modelSummary_OptimalTransect.csv")

clineFit_scan_s261_720757 <- read.csv("/Users/MQ10006146/Library/CloudStorage/OneDrive-MacquarieUniversity/Projects/Snapdragons_shared/Data_Analyses/slowClines/outputs/200mDemes/s261_720757_modelSummary_OptimalTransect.csv")

#GenoEcol <- read.csv("FinalGenoEcol_short_int.csv")
# combine matrices
ncol(clineFit_scan_s1187_290152)
ncol(clineFit_scan_ros_assembly_543443)
ncol(clineFit_scan_s316_93292)
ncol(clineFit_scan_s316_93292)
ncol(clineFit_scan_s316_257789)
ncol(clineFit_scan_s261_720757)
ncol(clineFit_scan_s91_39699)

# Fig Sx. 2 from FLA paper (maybe move to WGS clines) ggplot approach with melt
clineFit_scan_allLoci <- rbind(clineFit_scan_s1187_290152[,1:39],clineFit_scan_s316_93292[,1:39],clineFit_scan_s316_257789[,1:39],clineFit_scan_s91_39699[,1:39],clineFit_scan_s261_720757[,1:39],clineFit_scan_ros_assembly_543443[,1:39])
clineFit_scan_allLoci_brief <- clineFit_scan_allLoci[,c(1:15,37:39)]
clineFit_scan_allLoci_brief$width <- as.numeric(as.vector(clineFit_scan_allLoci_brief$width))
clineFit_scan_allLoci_brief$gradient <- as.numeric(as.vector(clineFit_scan_allLoci_brief$gradient))
clineFit_scan_allLoci_brief$maxLL <- as.numeric(as.vector(clineFit_scan_allLoci_brief$maxLL))


# write.csv(as.data.frame(clineFit_scan_allLoci),"clineFit_scan_allLoci.csv")

## testing best transect in relation to maxLL
me <- melt(clineFit_scan_allLoci_brief, id=c("locus","gradient","width"),measure.vars="maxLL")
me %>%
  group_by(locus) %>%
  summarise(max = max(value)) -> me.2

left_join(me, me.2) %>%
  mutate(color = value == max) %>%
  filter(color == TRUE) -> me.3
ggplot(data=me,aes(x = gradient,y=value),na.rm=T) + 
  geom_line() +
  geom_point() +
  geom_point(data=me.3, aes(x = gradient, y = value), color = "red") +
  #scale_color_manual(values = mycolors)+
  labs(x ="gradient of transect", y = "maxLL") +
  scale_x_continuous(name="gradient", breaks=seq(-0.5,0.2,0.1),limit=c(-0.6,0.2)) +
  theme(legend.position="none") +
  theme(axis.text=element_text(size=16)) +
  theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
  theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
  theme(strip.text.x = element_text(size=16)) +
  facet_wrap(~locus, scales = "free_y",ncol=1) # multifacetted plot with

## testing best transect in relation to steepest gradient
me <- melt(clineFit_scan_allLoci_brief, id=c("locus","gradient","maxLL"),measure.vars="width")
#me$value <- as.numeric(me$value)
#me$variable=as.numeric(levels(me$variable))[me$variable]

me %>%
  group_by(locus) %>%
  summarise(min = min(value)) -> me.2

left_join(me, me.2) %>%
  mutate(color = value == min) %>%
  filter(color == TRUE) -> me.3
ggplot(data=me,aes(x = gradient,y=value),na.rm=T) + 
  geom_line() +
  geom_point() +
  geom_point(data=me.3, aes(x = gradient, y = value), color = "red") +
  #scale_color_manual(values = mycolors)+
  labs(x ="gradient of transect", y = "cline width") +
  scale_x_continuous(name="gradient", breaks=seq(-0.5,0.2,0.1),limit=c(-0.5,0.2)) +
  #scale_y_continuous(name="width", breaks=seq(500,2500,500),limit=c(500,2500)) +
  theme(legend.position="none") +
  theme(axis.text=element_text(size=16)) +
  theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
  theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
  theme(strip.text.x = element_text(size=16)) +
  facet_wrap(~locus, scales = "free_y",ncol=1) # multifacetted plot with

me <- melt(clineFit_scan_allLoci_brief, id=c("locus","gradient","width"))
clineFit_scan_allLoci_brief_ros <- clineFit_scan_allLoci_brief[clineFit_scan_allLoci_brief$locus=="ros_assembly_543443",]
me %>%
  group_by(locus) %>%
  summarise(min = max(value)) -> me.2

left_join(me, me.2) %>%
  mutate(color = value == max) %>%
  filter(color == TRUE) -> me.3

ggplot(data=clineFit_scan_allLoci_brief_ros, aes(x = gradient, y = value)) + 
  geom_line() +
  geom_point(data=clineFit_scan_allLoci_brief_ros.3, aes(x = gradient, y = maxLL), color = "red") +
  facet_wrap(~variable, ncol=1, scales='free_y')

```

```{r evaluating models for best transect across all loci}

clineFit_scan_allLoci[,"locus"] <- as.factor(clineFit_scan_allLoci[,"locus"])
clineFit_scan_allLoci[,"maxLL"] <- as.numeric(as.vector(clineFit_scan_allLoci[,"maxLL"]))
clineFit_scan_allLoci[,"gradient"] <- as.numeric(as.vector(clineFit_scan_allLoci[,"gradient"]))
clineFit_scan_allLoci[,"width"] <- as.numeric(as.vector(clineFit_scan_allLoci[,"width"]))
write.csv(clineFit_scan_allLoci,"clineFit_scan_allLoci.csv")
colnames(clineFit_scan_allLoci)
clineFit_scan_allLoci_cut <- clineFit_scan_allLoci[c(1,7:14,37:39)]


clineFit_scan_allLoci_width_maxLL <- ggplot(clineFit_scan_allLoci, aes(x=width, y=maxLL)) + 
        geom_point() +
        theme(axis.text=element_text(size=16)) +
        theme(axis.title.y = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(axis.title.x = element_text(size=16,vjust=2.5,hjust=0.5)) + 
        theme(strip.text.x = element_text(size=16)) +
        facet_wrap(~locus,ncol=1, scales="free") 


#plot_grid(rub_AllROSbackgrouns_plot_Hue1_haps, rub_AllROSbackgrouns_plot_Hue2_haps, rub_AllROSbackgrouns_plot_Hue3_haps, rub_AllROSbackgrouns_plot_Hue4_haps,rub_AllROSbackgrouns_plot_Hue5_haps,rub_AllROSbackgrouns_plot_Hue6_haps,labels = c('A', 'B', 'C', 'D','E','F'), align="hv")
```

```{r cline plots for FLA manuscript}
ncol(clineFit_scan_s316_93292)
ncol(clineFit_scan_s316_257789)
clineFit_scan_FLA <- rbind(clineFit_scan_s316_93292[4,],clineFit_scan_s316_257789[4,])
# FLAup
clinePlot(clineModel="symm_5p",modelSummaryInput=clineFit_scan_FLA[1,],curveFit=T,cexMain=1.4,cexAxisPlot= 2.1,axisLineWdth=3.8,cexPoints=3,lwdCurve=4.3,refLocus=F,refPoints=F,HeaderDetails=T,FigLabel="",HaplotypeData=NA)

# FLA down
clinePlot(clineModel="symm_5p",modelSummaryInput=clineFit_scan_FLA[2,],curveFit=T,cexMain=1.4,cexAxisPlot= 2.1,axisLineWdth=3.8,cexPoints=3,lwdCurve=4.3,refLocus=F,refPoints=F,HeaderDetails=T,FigLabel="",multiplePoints=F,HaplotypeData=NA)

# both FLA together
clineFit_scan_FLA[2,"flipCline"] <- 1 

pdf("FLAclines.pdf",width = 15, height = 15,pointsize=12)
clinePlot(clineModel="symm_5p",modelSummaryInput=clineFit_scan_FLA,curveFit=T,cexMain=1.4,cexAxisPlot= 2.1,
        axisLineWdth=3.8,cexPoints=3,lwdCurve=4.3,refLocus=F,refPoints=F,HeaderDetails=T,FigLabel="",multiplePoints=T,multipleCurves=T,HaplotypeData=NA)
dev.off()

plot(clineFit_scan_FLA[,"width"],clineFit_scan_FLA[,"maxLL"])
clineFit_scan_FLA_maxLL <- clineFit_scan_FLA %>% slice_min(maxLL, n = 1, with_ties = FALSE)

# plot 1 for Des
clineFit_scan_FLA_SULF_pse <- rbind(clineFit_scan_s316_93292[4,1:39],clineFit_scan_s91_39699[4,1:39])

# One plot showing the pseudo-SULF cline and pseudo-fla clines across the Hybrid Zone. Both curves will be max on the right (magenta side), and so we can compare the widths and shapes directly.

clinePlot(clineModel="symm_5p",modelSummaryInput=clineFit_scan_FLA_SULF_pse,curveFit=T,cexMain=1.4,cexAxisPlot= 2.1, axisLineWdth=3.8,cexPoints=3,lwdCurve=4.3,refLocus=F,refPoints=F,HeaderDetails=T,FigLabel="",multiplePoints=T,multipleCurves=T,HaplotypeData=NA)

pdf("SULF_FLA_pse_clines.pdf",width = 15, height = 15,pointsize=12)
clinePlot(clineModel="symm_5p",modelSummaryInput=clineFit_scan_FLA_SULF_pse,curveFit=T,cexMain=1.4,cexAxisPlot= 2.1, axisLineWdth=3.8,cexPoints=3,lwdCurve=4.3,refLocus=F,refPoints=F,HeaderDetails=T,FigLabel="",multiplePoints=T,multipleCurves=T,HaplotypeData=NA)
dev.off()

# second version without SULF
pdf("FLA_pse_cline.pdf",width = 15, height = 15,pointsize=12)
clinePlot(clineModel="symm_5p",modelSummaryInput=clineFit_scan_FLA_SULF_pse[1,],curveFit=T,cexMain=1.4,cexAxisPlot= 2.1, axisLineWdth=3.8,cexPoints=3,lwdCurve=4.3,refLocus=F,refPoints=F,HeaderDetails=T,FigLabel="",multiplePoints=F,multipleCurves=F,HaplotypeData=NA)
dev.off()



#plot 2 for Des 
#One plot showing striatum-FLA & recombinant-FLA & pseudo-fla cline curves. These will have striatum max on left (yellow side), recombinant max in the middle, and pseudo max on the right (magenta side).

clineFit_scan_FLA_striatum_recombinant_FLA_pse_fla <- rbind(clineFit_scan_s316_93292[4,1:39],clineFit_scan_s316_257789[4,1:39])
clineFit_scan_FLA_striatum_recombinant_FLA_pse_fla[,"curveCol"]

colnames(clineFit_scan_FLA_striatum_recombinant_FLA_pse_fla)
clineFit_scan_FLA_striatum_recombinant_FLA_pse_fla[,"locus"]
clineFit_scan_FLA_striatum_recombinant_FLA_pse_fla[,"flipCline"]
clineFit_scan_FLA_striatum_recombinant_FLA_pse_fla[2,"flipCline"] <- 1
# demeData for recombinants same transect as above -0.345185185,	6900
HZ_Planoles_200mDemes_FLAr <- DemeSetup2D_haplos(HZdata=GenoEcol_Planoles,lociNames=c("Fla_2loc"),demeSize=200,
                                                 minSample=10,removePhenoMiss=F,transectGradient = -0.345185185,
                                                 transectIntercept = 6900,  adjustXY=F, pointsCol = F,
                                                 gridAdd=F,plotAllele1D = T,plotDem=c(3,3),addPie=T,
                                                 SNPlociListUpdated,verbose=F,plotAxisLab=F,plotPerp=T,
                                                 plotType="pdf",poolSeqPlot=T,plotMap=F,noPlots=F)

HZ_Planoles_200mDemes_FLAr[["Fla_2loc"]][["DemeMatrix2D_combine_noZero"]]

pdf("FLA_recombinant_clines.pdf",width = 15, height = 15,pointsize=12)
clinePlot(clineModel="symm_5p",modelSummaryInput=clineFit_scan_FLA_striatum_recombinant_FLA_pse_fla,curveFit=T,cexMain=1.4,cexAxisPlot= 2.1,axisLineWdth=3.8,cexPoints=3,lwdCurve=4.3,refLocus=F,refPoints=F,HeaderDetails=T,FigLabel="",multiplePoints=T,multipleCurves=T,HaplotypeData=HZ_Planoles_200mDemes_FLAr)
dev.off()

```

```{r cline plot for RUBIA and ROSEA clines for WGS clines manuscript}
# modify final colours for WGS clines paper
# modelSummaryInput
clineFit_scan_s261_720757[,"pointCol"] <- "#377EB8"
clineFit_scan_s261_720757[,"curveCol"] <- "#377EB8"
colourChoice[colourChoice[,"Region"]=="RUBIA",2] <- "#377EB8"
colourChoice[colourChoice[,"Region"]=="RUBIA",3] <- "#377EB8"
colourChoice[colourChoice[,"Region"]=="RUBIA",4] <- "#377EB8"
colourChoice[colourChoice[,"Region"]=="RUBIAlinked",2] <- "#377EB8"
colourChoice[colourChoice[,"Region"]=="RUBIAlinked",3] <- "#377EB8"
colourChoice[colourChoice[,"Region"]=="RUBIAlinked",4] <- "#377EB8"

pdf("RUBIAcline_WGS.pdf",width = 15, height = 15,pointsize=12)
clinePlot_WGS(clineModel="symm_5p",modelSummaryInput=clineFit_scan_s261_720757[4,],curveFit=T,cexMain=1.4,cexAxisPlot= 6.5,axisLineWdth=3.8,cexPoints=6,lwdCurve=16,refLocus=F,refPoints=F,HeaderDetails=T,FigLabel="",HaplotypeData=NA,poolSeqClines=allChr_Clines_Chr5clines,poolSeqStats=poolSeqStatsKeep,plotKeyGene=T,colourTable=colourChoice)
dev.off()


pdf("ROSEAcline.pdf",width = 15, height = 15,pointsize=12)
clinePlot_WGS(clineModel="symm_5p",modelSummaryInput=clineFit_scan_ros_assembly_543443[4,],curveFit=T,cexMain=1.4,cexAxisPlot= 6.5,axisLineWdth=3.8,cexPoints=6,lwdCurve=16,refLocus=F,refPoints=F,HeaderDetails=T,FigLabel="",HaplotypeData=NA,poolSeqClines=allChr_Clines_Chr6clines,poolSeqStats=poolSeqStatsKeep,plotKeyGene=T,colourTable=colourChoice)
dev.off()


```








## Supplementary figures 

```{r experimental plot FastCline vs Traditional example S Fig1}

clineCentre <- function(p,deltaPthresh,demeOrder,demeSpans) {
  # 6-(sum(c(0,0.25,0.45,0.55,0.75,1))+0.5)
  # deltaPthresh <- 0.8
  # p <- c(0,0.25,0.45,0.55,0.75,1)
  # demeOrder <- seq(1,6); demeSpans <- rep(1000,6)
  # demeSpans <- demeSpacing_Opt1
  p <- as.vector(unlist(p))
  # reordering demes based on position along transect (if required)
  p <- p[demeOrder]
  p0 <- p[1]
  p1 <- p[length(p)]
  deltaP <- abs(p0-p1)
  d_i = demeSpans 
  # reorder the spans - not required (spans fixed, just flipping p between pool 2 & 3)
  # d_i <- d_i[demeOrder]
  if (deltaP < deltaPthresh) {
    return (-9)
  }
  if (deltaP >= deltaPthresh) {
    numerCentre = p-rep(p0,length(p))
    denomCentre = rep(p1-p0,length(p))
    #sum(d_i) - sum((numerCentre/denomCentre)*rep(d_i))
    centre_output = sum(d_i) - sum((numerCentre/denomCentre) * d_i)
    return(centre_output)
  }
}
clineWidth <- function(p,deltaPthresh,demeOrder,demeSpans) {
  # deltaPthresh <- 0.8
  # p <- this_p
  # demeSpans <- demeSpacing_Opt1
  p <- as.vector(unlist(p))
  # reordering demes based on position along transect (if required)
  p <- p[demeOrder]
  p0 <- p[1]
  p1 <- p[length(p)]
  deltaP <- abs(p0-p1)
  d_i = demeSpans 
  # reorder the spans (not required here)
  # d_i <- d_i[demeOrder]
  if (deltaP < deltaPthresh) {
    return (-9)
  }
  if (deltaP >= deltaPthresh) {
    numerWidth = (p-rep(p0,length(p)))*(rep(p1,length(p))-p)
    denomWidth = rep((p1-p0)^2,length(p))
    width_output = 4*(sum((numerWidth/denomWidth)*rep(d_i)))
    return(width_output)
  }
}

# testing functions
demeSpacingTest <- rep(1000,6)
clineCentre(p=c(0,0.25,0.45,0.55,0.75,1),1,seq(1,6),rep(1000,6))


# deme spacing original
demeSpacing_Original <- c(6000,5000,1500,1400,3500,6000)
# deme spacing 1
demeSpacing_Opt1 <- c(5805,5893,1270,1500,3453,7251)
# deme spacing 2
demeSpacing_Opt2 <- c(4195,4195,4195,4195,4195,4195)
# deme spacing 3
demeSpacing_Opt3 <- c(10046,1270,1270,1500,1500,9586)

demeSpacing_ExampleFig1 <- c(6000,5000,1500,1400,3500,6000)


# traditional cline sigmoid 

pdf("FLAclines.pdf",width = 15, height = 15,pointsize=12)
clinePlot(clineModel="symm_5p",modelSummaryInput=clineFit_scan_FLA,curveFit=T,cexMain=1.4,cexAxisPlot= 2.1,
        axisLineWdth=3.8,cexPoints=3,lwdCurve=4.3,refLocus=F,refPoints=F,HeaderDetails=T,FigLabel="",multiplePoints=T,multipleCurves=T,HaplotypeData=NA)
dev.off()

```

# Figure 2. FastClines approach 
plot generated in a combination of powerpoint and in another script: FastClines_PowerSims.Rmd

# Fig Sx. SFS and JSFS

```{r SFS}
p_allSLF_hist <- hist(p_allSLF,breaks=seq(0,1,0.05),plot=F)
p_noSLF_hist <- hist(p_noSLF,breaks=seq(0,1,0.05),plot=F)
p_noSLF_freq <- p_noSLF_hist$counts/sum(p_noSLF_hist$counts)
p_allSLF_freq <- p_allSLF_hist$counts/sum(p_allSLF_hist$counts)
p_combined <- rbind(p_allSLF_freq,p_noSLF_freq)
# drop last 2 freq bins as nothing in there
p_combined <- p_combined[,1:18]
labelsSFS <- p_allSLF_hist$breaks
totalSites_noSLF = sum(p_noSLF_hist$counts)
totalSites_SLF = sum(p_allSLF_hist$counts)
if (thisPopNum==1) {
    titleMain <- "SFS striatum, "
}
if (thisPopNum==2) {
    titleMain <- "SFS pseudomajus, "
  }
  subTitle <- paste(paste("within SLFs n=",totalSites_SLF,sep=""),paste("outside SLFs n=",totalSites_noSLF,sep=""),sep=", ")
titleMain <- paste(titleMain,subTitle,sep="")
labelsSFS <- c(">0-0.05","0.5-0.1","0.1-0.15","0.15-0.2","0.2-0.25","0.25-0.3","0.3-0.35","0.35-0.4","0.4-0.45",
                     "0.45-0.5","0.5-0.55","0.55-0.6","0.6-0.65","0.65-0.7","0.7-0.75","0.75-0.8","0.8-0.85","0.85-0.9")
barplot(p_combined, main=paste(titleMain,mappingRange,sep="\n"), names.arg=labelsSFS,ylim=c(0,0.6),
              ylab = "frequency", cex.lab = 1.5,las=2,
              cex.main = 1.0, beside=TRUE, col=colours)
```

```{r JSFS}
# JSFS
    noSLF <- (1-SitesData[SitesData[,"SLF"]==-9,c("p_1","p_2")])
    SLF <- (1-SitesData[SitesData[,"SLF"]!=-9,c("p_1","p_2")])
    SLF_JSF <- hist2d(SLF, nbins=17, xlim=c(0,1),ylim=c(0,1),xlab="allele frequency - striatum",ylab="allele frequency - pseudomajus",
                        main=paste("JSFS - within SLFs",mappingRange,sep="\n"),cex=1.2,cex.lab=1.2,cex.axis=1.2,same.scale=T,
                        col=r, FUN=function(x) log(length(x)))
    segments(x0=0,x1=1,y0=0,y1=1,lwd=1,lty=3)
    noSLF_JSF <- hist2d(noSLF, nbins=20, xlim=c(0,1),ylim=c(0,1),xlab="allele frequency - striatum",ylab="allele frequency - pseudomajus",
                          main=paste("JSFS - outside SLFs",mappingRange,sep="\n"),cex=1.2,cex.lab=1.2,cex.axis=1.2,same.scale=T,
                          col=r, FUN=function(x) log(length(x)))
    segments(x0=0,x1=1,y0=0,y1=1,lwd=1,lty=3)
```
        
        